diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutDataListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutDataListener.h
index 6317086472..c1e2760afd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutDataListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutDataListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*
  * @file
  * alljoyn_aboutdatalistener provides a Listener implemented by AllJoyn apps and called
@@ -72,7 +73,7 @@ typedef struct _alljoyn_aboutdatalistener_handle* alljoyn_aboutdatalistener;
  *  - other error indicating failure
  */
 typedef QStatus (AJ_CALL * alljoyn_aboutdatalistener_getaboutdata_ptr)(const void* context,
-                                                                       alljoyn_msgarg msgArg,
+                                                                       _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_msgarg msgArg,
                                                                        const char* language);
 
 /**
@@ -104,7 +105,7 @@ typedef QStatus (AJ_CALL * alljoyn_aboutdatalistener_getaboutdata_ptr)(const voi
  * @return ER_OK if successful
  */
 typedef QStatus (AJ_CALL * alljoyn_aboutdatalistener_getannouncedaboutdata_ptr)(const void* context,
-                                                                                alljoyn_msgarg msgArg);
+                                                                                _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_msgarg msgArg);
 
 /**
  * Struct containing callbacks used for creation of an alljoyn_aboutdatalistener.
@@ -126,7 +127,7 @@ typedef struct {
  * This object will invoke the provided callbacks passing in the provided context.
  * @return The allocated alljoyn_aboutdatalistener.
  */
-extern AJ_API alljoyn_aboutdatalistener AJ_CALL alljoyn_aboutdatalistener_create(const alljoyn_aboutdatalistener_callbacks* callbacks,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_aboutdatalistener AJ_CALL alljoyn_aboutdatalistener_create(const alljoyn_aboutdatalistener_callbacks* callbacks,
                                                                                  const void* context);
 
 /**
@@ -134,7 +135,7 @@ extern AJ_API alljoyn_aboutdatalistener AJ_CALL alljoyn_aboutdatalistener_create
  *
  * @param listener The alljoyn_aboutdatalistener to be freed.
  */
-extern AJ_API void AJ_CALL alljoyn_aboutdatalistener_destroy(alljoyn_aboutdatalistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_aboutdatalistener_destroy(alljoyn_aboutdatalistener listener);
 
 
 #ifdef __cplusplus
