diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutData.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutData.h
index fedf02d7ea..b578989bdc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutData.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutData.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * alljoyn_aboutdata is responsible for storing data for the org.alljoyn.About interface.
@@ -49,7 +50,7 @@ typedef struct _alljoyn_aboutdata_handle* alljoyn_aboutdata;
  *
  * @return The allocated alljoyn_aboutdata.
  */
-extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_empty();
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_empty();
 
 /**
  * Allocate a new alljoyn_aboutdata object setting the default language.
@@ -58,7 +59,7 @@ extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_empty();
  *
  * @return The allocated alljoyn_aboutdata.
  */
-extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create(const char* defaultLanguage);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create(const char* defaultLanguage);
 
 /**
  * Allocate a new alljoyn_aboutdata object filling in the fields of the About data
@@ -75,7 +76,7 @@ extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create(const char* def
  *
  * @return The allocated alljoyn_aboutdata.
  */
-extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_full(const alljoyn_msgarg arg,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_full(const alljoyn_msgarg arg,
                                                                       const char* language);
 
 /**
@@ -83,7 +84,7 @@ extern AJ_API alljoyn_aboutdata AJ_CALL alljoyn_aboutdata_create_full(const allj
  *
  * @param data The alljoyn_aboutdata to be freed.
  */
-extern AJ_API void AJ_CALL alljoyn_aboutdata_destroy(alljoyn_aboutdata data);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_aboutdata_destroy(alljoyn_aboutdata data);
 
 /**
  * Initialize the About data held by the specified alljoyn_aboutdata object
@@ -100,7 +101,7 @@ extern AJ_API void AJ_CALL alljoyn_aboutdata_destroy(alljoyn_aboutdata data);
  *                         the About data fields.
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfromxml(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfromxml(alljoyn_aboutdata data,
                                                               const char* aboutDataXml);
 
 /**
@@ -113,7 +114,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfromxml(alljoyn_aboutdata
  *
  * @return true if all required field are listed for the given language
  */
-extern AJ_API bool AJ_CALL alljoyn_aboutdata_isvalid(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API bool AJ_CALL alljoyn_aboutdata_isvalid(alljoyn_aboutdata data,
                                                      const char* language);
 
 /**
@@ -129,7 +130,7 @@ extern AJ_API bool AJ_CALL alljoyn_aboutdata_isvalid(alljoyn_aboutdata data,
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfrommsgarg(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfrommsgarg(alljoyn_aboutdata data,
                                                                  const alljoyn_msgarg arg,
                                                                  const char* language);
 
@@ -150,7 +151,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_createfrommsgarg(alljoyn_aboutda
  * - ER_OK on success
  * - #ER_ABOUT_INVALID_ABOUTDATA_FIELD_VALUE if appId is not 128-bits (16 bytes) in size
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid(alljoyn_aboutdata data,
                                                          const uint8_t* appId,
                                                          const size_t num);
 
@@ -173,7 +174,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid(alljoyn_aboutdata data,
  *  - #ER_OK on success
  *  - #ER_ABOUT_INVALID_ABOUTDATA_FIELD_VALUE if the AppId is not a 128-bits (16 bytes) in size
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid_fromstring(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid_fromstring(alljoyn_aboutdata data,
                                                                     const char* appId);
 
 /**
@@ -189,7 +190,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappid_fromstring(alljoyn_abou
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappid(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappid(alljoyn_aboutdata data,
                                                          uint8_t** appId,
                                                          size_t* num);
 
@@ -205,7 +206,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappid(alljoyn_aboutdata data,
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdefaultlanguage(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdefaultlanguage(alljoyn_aboutdata data,
                                                                    const char* defaultLanguage);
 
 /**
@@ -216,7 +217,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdefaultlanguage(alljoyn_about
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdefaultlanguage(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdefaultlanguage(alljoyn_aboutdata data,
                                                                    char** defaultLanguage);
 /**
  * Set the DeviceName to the About data
@@ -236,7 +237,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdefaultlanguage(alljoyn_about
  *                                             and the default language is also
  *                                             not found.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdevicename(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdevicename(alljoyn_aboutdata data,
                                                               const char* deviceName,
                                                               const char* language);
 
@@ -254,7 +255,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdevicename(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdevicename(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdevicename(alljoyn_aboutdata data,
                                                               char** deviceName,
                                                               const char* language);
 
@@ -271,7 +272,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdevicename(alljoyn_aboutdata
  * @return ER_OK on success
  */
 
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdeviceid(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdeviceid(alljoyn_aboutdata data,
                                                             const char* deviceId);
 
 /**
@@ -286,7 +287,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdeviceid(alljoyn_aboutdata da
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdeviceid(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdeviceid(alljoyn_aboutdata data,
                                                             char** deviceId);
 
 /**
@@ -307,7 +308,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdeviceid(alljoyn_aboutdata da
  *                                             and the default language is also
  *                                             not found.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappname(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappname(alljoyn_aboutdata data,
                                                            const char* appName,
                                                            const char* language);
 
@@ -325,7 +326,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setappname(alljoyn_aboutdata dat
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappname(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappname(alljoyn_aboutdata data,
                                                            char** appName,
                                                            const char* language);
 
@@ -347,7 +348,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getappname(alljoyn_aboutdata dat
  *                                             and the default language is also
  *                                             not found.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmanufacturer(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmanufacturer(alljoyn_aboutdata data,
                                                                 const char* manufacturer,
                                                                 const char* language);
 
@@ -365,7 +366,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmanufacturer(alljoyn_aboutdat
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmanufacturer(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmanufacturer(alljoyn_aboutdata data,
                                                                 char** manufacturer,
                                                                 const char* language);
 
@@ -381,7 +382,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmanufacturer(alljoyn_aboutdat
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmodelnumber(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmodelnumber(alljoyn_aboutdata data,
                                                                const char* modelNumber);
 
 /**
@@ -396,7 +397,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setmodelnumber(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmodelnumber(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmodelnumber(alljoyn_aboutdata data,
                                                                char** modelNumber);
 
 /**
@@ -413,7 +414,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getmodelnumber(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupportedlanguage(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupportedlanguage(alljoyn_aboutdata data,
                                                                      const char* language);
 
 /**
@@ -429,7 +430,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupportedlanguage(alljoyn_abo
  * @return The number of languageTags returned or the total number of
  *         language tags if languageTags is NULL.
  */
-extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getsupportedlanguages(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getsupportedlanguages(alljoyn_aboutdata data,
                                                                      const char** languageTags,
                                                                      size_t num);
 
@@ -450,7 +451,7 @@ extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getsupportedlanguages(alljoyn_abo
  *  - #ER_ABOUT_DEFAULT_LANGUAGE_NOT_SPECIFIED if language tag was not specified and
  *                                             the default language is also not found.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdescription(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdescription(alljoyn_aboutdata data,
                                                                const char* description,
                                                                const char* language);
 
@@ -468,7 +469,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdescription(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdescription(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdescription(alljoyn_aboutdata data,
                                                                char** description,
                                                                const char* language);
 
@@ -487,7 +488,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdescription(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdateofmanufacture(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdateofmanufacture(alljoyn_aboutdata data,
                                                                      const char* dateOfManufacture);
 
 /**
@@ -505,7 +506,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setdateofmanufacture(alljoyn_abo
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdateofmanufacture(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdateofmanufacture(alljoyn_aboutdata data,
                                                                      char** dateOfManufacture);
 
 /**
@@ -520,7 +521,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getdateofmanufacture(alljoyn_abo
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsoftwareversion(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsoftwareversion(alljoyn_aboutdata data,
                                                                    const char* softwareVersion);
 
 /**
@@ -535,7 +536,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsoftwareversion(alljoyn_about
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsoftwareversion(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsoftwareversion(alljoyn_aboutdata data,
                                                                    char** softwareVersion);
 
 /**
@@ -554,7 +555,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsoftwareversion(alljoyn_about
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getajsoftwareversion(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getajsoftwareversion(alljoyn_aboutdata data,
                                                                      char** ajSoftwareVersion);
 
 /**
@@ -569,7 +570,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getajsoftwareversion(alljoyn_abo
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_sethardwareversion(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_sethardwareversion(alljoyn_aboutdata data,
                                                                    const char* hardwareVersion);
 
 /**
@@ -584,7 +585,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_sethardwareversion(alljoyn_about
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_gethardwareversion(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_gethardwareversion(alljoyn_aboutdata data,
                                                                    char** hardwareVersion);
 
 /**
@@ -599,7 +600,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_gethardwareversion(alljoyn_about
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupporturl(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupporturl(alljoyn_aboutdata data,
                                                               const char* supportUrl);
 
 /**
@@ -614,7 +615,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setsupporturl(alljoyn_aboutdata
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsupporturl(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsupporturl(alljoyn_aboutdata data,
                                                               char** supportUrl);
 
 /**
@@ -642,7 +643,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getsupporturl(alljoyn_aboutdata
  *                                             and the default language is also
  *                                             not found.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setfield(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setfield(alljoyn_aboutdata data,
                                                          const char* name,
                                                          alljoyn_msgarg value,
                                                          const char* language);
@@ -658,7 +659,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_setfield(alljoyn_aboutdata data,
  *
  * @return ER_OK on success
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getfield(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getfield(alljoyn_aboutdata data,
                                                          const char* name,
                                                          alljoyn_msgarg* value,
                                                          const char* language);
@@ -676,7 +677,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getfield(alljoyn_aboutdata data,
  * @return
  *  The number of fields returned or the total number of fields if the fields parameter is NULL
  */
-extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getfields(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getfields(alljoyn_aboutdata data,
                                                          const char** fields,
                                                          size_t num_fields);
 
@@ -694,7 +695,7 @@ extern AJ_API size_t AJ_CALL alljoyn_aboutdata_getfields(alljoyn_aboutdata data,
  *  - ER_ABOUT_ABOUTDATA_MISSING_REQUIRED_FIELD if a required field is missing
  *  - other error indicating failure
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getaboutdata(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getaboutdata(alljoyn_aboutdata data,
                                                              alljoyn_msgarg msgArg,
                                                              const char* language);
 
@@ -721,7 +722,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getaboutdata(alljoyn_aboutdata d
  *                    data.
  * @return ER_OK if successful
  */
-extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getannouncedaboutdata(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getannouncedaboutdata(alljoyn_aboutdata data,
                                                                       alljoyn_msgarg msgArg);
 
 /**
@@ -734,7 +735,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_aboutdata_getannouncedaboutdata(alljoyn_ab
  *  - true if the field is required to make an About announcement
  *  - false otherwise.  If the fieldName is unknown false will be returned
  */
-extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldrequired(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldrequired(alljoyn_aboutdata data,
                                                              const char* fieldName);
 
 /**
@@ -747,7 +748,7 @@ extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldrequired(alljoyn_aboutdata d
  *  - true if the field is part of the announce signal
  *  - false otherwise.  If the fieldName is unknown false will be returned
  */
-extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldannounced(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldannounced(alljoyn_aboutdata data,
                                                               const char* fieldName);
 
 /**
@@ -762,7 +763,7 @@ extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldannounced(alljoyn_aboutdata
  *  - true if the field is a localizable value
  *  - false otherwise. If the fieldName is unknown false will be returned.
  */
-extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldlocalized(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldlocalized(alljoyn_aboutdata data,
                                                               const char* fieldName);
 
 /**
@@ -775,7 +776,7 @@ extern AJ_API bool AJ_CALL alljoyn_aboutdata_isfieldlocalized(alljoyn_aboutdata
  *  - the signature of the field
  *  - NULL is field is unknown
  */
-extern AJ_API const char* AJ_CALL alljoyn_aboutdata_getfieldsignature(alljoyn_aboutdata data,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_aboutdata_getfieldsignature(alljoyn_aboutdata data,
                                                                       const char* fieldName);
 
 #ifdef __cplusplus
