diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutListener.h
index ac1c73d4f1..7ec86998fb 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * alljoyn_aboutlistener provides a callback for the org.alljoyn.About.Announce signal.
@@ -90,13 +91,13 @@ typedef struct {
  * @param context  Context to pass along to callback functions.
  * @return allocated alljoyn_aboutlistener
  */
-extern AJ_API alljoyn_aboutlistener AJ_CALL alljoyn_aboutlistener_create(const alljoyn_aboutlistener_callback* callback,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_aboutlistener AJ_CALL alljoyn_aboutlistener_create(const alljoyn_aboutlistener_callback* callback,
                                                                          const void* context);
 
 /**
  * Free an alljoyn_aboutlistener.
  */
-extern AJ_API void AJ_CALL alljoyn_aboutlistener_destroy(alljoyn_aboutlistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_aboutlistener_destroy(alljoyn_aboutlistener listener);
 
 #ifdef __cplusplus
 } /* extern "C" */
