diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIconProxy.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIconProxy.h
index 2ee931524e..160a38a68d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIconProxy.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIconProxy.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * alljoyn_abouticonproxy implements a proxy bus object used to interact with
@@ -45,16 +46,16 @@ typedef struct _alljoyn_abouticonproxy_handle* alljoyn_abouticonproxy;
  *
  * @return The allocated alljoyn_abouticonproxy.
  */
-extern AJ_API alljoyn_abouticonproxy AJ_CALL alljoyn_abouticonproxy_create(alljoyn_busattachment bus,
-                                                                           const char* busName,
-                                                                           alljoyn_sessionid sessionId);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_abouticonproxy AJ_CALL alljoyn_abouticonproxy_create(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_busattachment bus,
+                                                                           _WIN32META_ANNOTATION_("win32metadata:in") const char* busName,
+                                                                           _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionid sessionId);
 
 /**
  * Free an alljoyn_abouticonproxy object.
  *
  * @param proxy The alljoyn_abouticonproxy to be freed.
  */
-extern AJ_API void AJ_CALL alljoyn_abouticonproxy_destroy(alljoyn_abouticonproxy proxy);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_abouticonproxy_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticonproxy proxy);
 
 /**
  * This method makes multiple proxy bus object method calls to fill in the
@@ -67,8 +68,8 @@ extern AJ_API void AJ_CALL alljoyn_abouticonproxy_destroy(alljoyn_abouticonproxy
  *  - #ER_OK if successful
  *  - an error status indicating failure to get the icon content
  */
-extern AJ_API QStatus AJ_CALL alljoyn_abouticonproxy_geticon(alljoyn_abouticonproxy proxy,
-                                                             alljoyn_abouticon icon);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_abouticonproxy_geticon(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticonproxy proxy,
+                                                             _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon);
 
 /**
  * @param[in]  proxy   alljoyn_abouticonproxy object
@@ -78,8 +79,8 @@ extern AJ_API QStatus AJ_CALL alljoyn_abouticonproxy_geticon(alljoyn_abouticonpr
  * - #ER_OK if successful
  * - an error status indicating failure
  */
-extern AJ_API QStatus AJ_CALL alljoyn_abouticonproxy_getversion(alljoyn_abouticonproxy proxy,
-                                                                uint16_t* version);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_abouticonproxy_getversion(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticonproxy proxy,
+                                                                _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") uint16_t* version);
 
 #ifdef __cplusplus
 } /* extern "C" */
