diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIcon.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIcon.h
index 950fae5afb..037e38b3ab 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIcon.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AboutIcon.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * alljoyn_abouticon is a container that holds information about an About icon.
@@ -36,14 +37,14 @@ typedef struct _alljoyn_abouticon_handle* alljoyn_abouticon;
  * Allocate a new empty alljoyn_abouticon object.
  * @return The allocated alljoyn_abouticon.
  */
-extern AJ_API alljoyn_abouticon AJ_CALL alljoyn_abouticon_create();
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_abouticon AJ_CALL alljoyn_abouticon_create();
 
 /**
  * Free an alljoyn_abouticon object.
  *
  * @param icon the alljoyn_abouticon to be freed.
  */
-extern AJ_API void AJ_CALL alljoyn_abouticon_destroy(alljoyn_abouticon icon);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_abouticon_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon);
 
 /**
  * Returns a pointer to the icon's internal content data as well as the size of this data.
@@ -52,9 +53,9 @@ extern AJ_API void AJ_CALL alljoyn_abouticon_destroy(alljoyn_abouticon icon);
  * @param[out]  data  pointer to the icon content data
  * @param[out]  size  size (in number of bytes) of the icon content data
  */
-extern AJ_API void AJ_CALL alljoyn_abouticon_getcontent(alljoyn_abouticon icon,
-                                                        const uint8_t** data,
-                                                        size_t* size);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_abouticon_getcontent(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon,
+                                                        _WIN32META_ANNOTATION_("win32metadata:in") const uint8_t** data,
+                                                        _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") size_t* size);
 
 /**
  * Set the content for an icon.  The content must fit into a alljoyn_msgarg so the
@@ -79,11 +80,11 @@ extern AJ_API void AJ_CALL alljoyn_abouticon_getcontent(alljoyn_abouticon icon,
  *  - ER_BUS_BAD_VALUE if the data is to large to be marshaled
  *  - other status indicating failure
  */
-extern AJ_API QStatus AJ_CALL alljoyn_abouticon_setcontent(alljoyn_abouticon icon,
-                                                           const char* type,
-                                                           uint8_t* data,
-                                                           size_t csize,
-                                                           bool ownsData);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_abouticon_setcontent(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon,
+                                                           _WIN32META_ANNOTATION_("win32metadata:in") const char* type,
+                                                           _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") uint8_t* data,
+                                                           _WIN32META_ANNOTATION_("win32metadata:in") size_t csize,
+                                                           _WIN32META_ANNOTATION_("win32metadata:in") bool ownsData);
 
 /**
  * Gets the URL that contains the icon for the application.
@@ -92,9 +93,9 @@ extern AJ_API QStatus AJ_CALL alljoyn_abouticon_setcontent(alljoyn_abouticon ico
  * @param[out] type  the Media Type indicating the icon image type
  * @param[out] url   the URL that contains the location of the icon
  */
-extern AJ_API void AJ_CALL alljoyn_abouticon_geturl(alljoyn_abouticon icon,
-                                                    const char** type,
-                                                    const char** url);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_abouticon_geturl(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon,
+                                                    _WIN32META_ANNOTATION_("win32metadata:in") const char** type,
+                                                    _WIN32META_ANNOTATION_("win32metadata:in") const char** url);
 
 /**
  * Set a url that contain the icon for the application.
@@ -111,9 +112,9 @@ extern AJ_API void AJ_CALL alljoyn_abouticon_geturl(alljoyn_abouticon icon,
  *  - #ER_OK on success
  *  - other status indicating failure
  */
-extern AJ_API QStatus AJ_CALL alljoyn_abouticon_seturl(alljoyn_abouticon icon,
-                                                       const char* type,
-                                                       const char* url);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_abouticon_seturl(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon,
+                                                       _WIN32META_ANNOTATION_("win32metadata:in") const char* type,
+                                                       _WIN32META_ANNOTATION_("win32metadata:in") const char* url);
 
 /**
  * Clear the alljoyn_abouticon. This will change all strings to empty strings
@@ -121,7 +122,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_abouticon_seturl(alljoyn_abouticon icon,
  *
  * @param[in] icon  the alljoyn_abouticon object this call is made for
  */
-extern AJ_API void AJ_CALL alljoyn_abouticon_clear(alljoyn_abouticon icon);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_abouticon_clear(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon);
 
 /**
  * Add the IconContent from an alljoyn_msgarg. This will make a local copy of the
@@ -135,8 +136,8 @@ extern AJ_API void AJ_CALL alljoyn_abouticon_clear(alljoyn_abouticon icon);
  *   - ER_OK on success
  *   - status indicating failure otherwise
  */
-extern AJ_API QStatus AJ_CALL alljoyn_abouticon_setcontent_frommsgarg(alljoyn_abouticon icon,
-                                                                      const alljoyn_msgarg arg);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_abouticon_setcontent_frommsgarg(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_abouticon icon,
+                                                                      _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_msgarg arg);
 
 #ifdef __cplusplus
 } /* extern "C" */
