diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/activdbg.h b/generation/WinSDK/RecompiledIdlHeaders/um/activdbg.h
index 227d23806f..9949d72f9d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/activdbg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/activdbg.h
@@ -970,6 +970,7 @@ extern "C"{
 //=--------------------------------------------------------------------------=
 //
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #pragma comment(lib,"uuid.lib")
@@ -10930,9 +10931,13 @@ unsigned char * __RPC_USER  BSTR_UserMarshal(  __RPC__in unsigned long *, __RPC_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
 void                      __RPC_USER  BSTR_UserFree(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Windows_SupportedOS_WindowsXP_
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
 /* [local] */ HRESULT __stdcall IEnumDebugCodeContexts_Next_Proxy( 
