diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/AF_Irda.h b/generation/WinSDK/RecompiledIdlHeaders/um/AF_Irda.h
index 1717dfd8bd..e1c3ff4efd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/AF_Irda.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/AF_Irda.h
@@ -11,6 +11,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -131,7 +132,7 @@ typedef struct
     u_char          nRXPackets;     // Number packets in receive window
 } LM_IRPARMS,*PLM_IRPARMS;
 
-typedef struct _SOCKADDR_IRDA
+typedef struct   _SOCKADDR_IRDA
 {
 	u_short irdaAddressFamily;
 	u_char  irdaDeviceID[4];
