diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/AppxPackaging.h b/generation/WinSDK/RecompiledIdlHeaders/um/AppxPackaging.h
index 59e812731b..94b0ede09a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/AppxPackaging.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/AppxPackaging.h
@@ -771,2 +771,3 @@ extern "C"{
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #if (NTDDI_VERSION >= NTDDI_WIN7)
@@ -990,2 +991,3 @@ EXTERN_C const IID IID_IAppxFactory;
     MIDL_INTERFACE("beb94909-e451-438b-b5a7-d79e767b75d8")
+    _Windows_SupportedOS_Windows8_
     IAppxFactory : public IUnknown
@@ -1157,2 +1159,3 @@ EXTERN_C const IID IID_IAppxFactory2;
     MIDL_INTERFACE("f1346df2-c282-4e22-b918-743a929a8d55")
+    _Windows_SupportedOS_10240_
     IAppxFactory2 : public IUnknown
@@ -1508,2 +1511,3 @@ EXTERN_C const IID IID_IAppxPackageReader;
     MIDL_INTERFACE("b5c49650-99bc-481c-9a34-3d53a4106708")
+    _Windows_SupportedOS_Windows8_
     IAppxPackageReader : public IUnknown
@@ -1655,2 +1659,3 @@ EXTERN_C const IID IID_IAppxPackageWriter;
     MIDL_INTERFACE("9099e33b-246f-41e4-881a-008eb613f858")
+    _Windows_SupportedOS_Windows8_
     IAppxPackageWriter : public IUnknown
@@ -1767,2 +1772,3 @@ EXTERN_C const IID IID_IAppxPackageWriter2;
     MIDL_INTERFACE("2cf5c4fd-e54c-4ea5-ba4e-f8c4b105a8c8")
+    _Windows_SupportedOS_10240_
     IAppxPackageWriter2 : public IUnknown
@@ -1858,2 +1864,3 @@ EXTERN_C const IID IID_IAppxPackageWriter3;
     MIDL_INTERFACE("a83aacd3-41c0-4501-b8a3-74164f50b2fd")
+    _Windows_SupportedOS_10240_
     IAppxPackageWriter3 : public IUnknown
@@ -1953,2 +1960,3 @@ EXTERN_C const IID IID_IAppxFile;
     MIDL_INTERFACE("91df827b-94fd-468f-827b-57f41b2f6f2e")
+    _Windows_SupportedOS_Windows8_
     IAppxFile : public IUnknown
@@ -2225,2 +2233,3 @@ EXTERN_C const IID IID_IAppxFilesEnumerator;
     MIDL_INTERFACE("f007eeaf-9831-411c-9847-917cdc62d1fe")
+    _Windows_SupportedOS_Windows8_
     IAppxFilesEnumerator : public IUnknown
@@ -2338,2 +2347,3 @@ EXTERN_C const IID IID_IAppxBlockMapReader;
     MIDL_INTERFACE("5efec991-bca3-42d1-9ec2-e92d609ec22a")
+    _Windows_SupportedOS_Windows8_
     IAppxBlockMapReader : public IUnknown
@@ -2468,2 +2478,3 @@ EXTERN_C const IID IID_IAppxBlockMapFile;
     MIDL_INTERFACE("277672ac-4f63-42c1-8abc-beae3600eb59")
+    _Windows_SupportedOS_Windows8_
     IAppxBlockMapFile : public IUnknown
@@ -2746,2 +2757,3 @@ EXTERN_C const IID IID_IAppxBlockMapFilesEnumerator;
     MIDL_INTERFACE("02b856a2-4262-4070-bacb-1a8cbbc42305")
+    _Windows_SupportedOS_Windows8_
     IAppxBlockMapFilesEnumerator : public IUnknown
@@ -2859,2 +2871,3 @@ EXTERN_C const IID IID_IAppxBlockMapBlock;
     MIDL_INTERFACE("75cf3930-3244-4fe0-a8c8-e0bcb270b889")
+    _Windows_SupportedOS_Windows8_
     IAppxBlockMapBlock : public IUnknown
@@ -2963,2 +2976,3 @@ EXTERN_C const IID IID_IAppxBlockMapBlocksEnumerator;
     MIDL_INTERFACE("6b429b5b-36ef-479e-b9eb-0c1482b49e16")
+    _Windows_SupportedOS_Windows8_
     IAppxBlockMapBlocksEnumerator : public IUnknown
@@ -3076,2 +3090,3 @@ EXTERN_C const IID IID_IAppxManifestReader;
     MIDL_INTERFACE("4e1bd148-55a0-4480-a3d1-15544710637c")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestReader : public IUnknown
@@ -3271,2 +3286,3 @@ EXTERN_C const IID IID_IAppxManifestReader2;
     MIDL_INTERFACE("d06f67bc-b31d-4eba-a8af-638e73e77b4d")
+    _Windows_SupportedOS_Windows81_
     IAppxManifestReader2 : public IAppxManifestReader
@@ -3842,2 +3858,3 @@ EXTERN_C const IID IID_IAppxManifestReader5;
     MIDL_INTERFACE("8D7AE132-A690-4C00-B75A-6AAE1FEAAC80")
+    _Windows_SupportedOS_10240_
     IAppxManifestReader5 : public IUnknown
@@ -3929,2 +3946,3 @@ EXTERN_C const IID IID_IAppxManifestReader6;
     MIDL_INTERFACE("34DEACA4-D3C0-4E3E-B312-E42625E3807E")
+    _Windows_SupportedOS_10240_
     IAppxManifestReader6 : public IUnknown
@@ -5081,2 +5099,3 @@ EXTERN_C const IID IID_IAppxManifestOptionalPackageInfo;
     MIDL_INTERFACE("2634847D-5B5D-4FE5-A243-002FF95EDC7E")
+    _Windows_SupportedOS_10240_
     IAppxManifestOptionalPackageInfo : public IUnknown
@@ -5181,2 +5200,3 @@ EXTERN_C const IID IID_IAppxManifestMainPackageDependenciesEnumerator;
     MIDL_INTERFACE("A99C4F00-51D2-4F0F-BA46-7ED5255EBDFF")
+    _Windows_SupportedOS_10240_
     IAppxManifestMainPackageDependenciesEnumerator : public IUnknown
@@ -5294,2 +5314,3 @@ EXTERN_C const IID IID_IAppxManifestMainPackageDependency;
     MIDL_INTERFACE("05D0611C-BC29-46D5-97E2-84B9C79BD8AE")
+    _Windows_SupportedOS_10240_
     IAppxManifestMainPackageDependency : public IUnknown
@@ -5407,2 +5428,3 @@ EXTERN_C const IID IID_IAppxManifestPackageId;
     MIDL_INTERFACE("283ce2d7-7153-4a91-9649-7a0f7240945f")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestPackageId : public IUnknown
@@ -5589,2 +5611,3 @@ EXTERN_C const IID IID_IAppxManifestPackageId2;
     MIDL_INTERFACE("2256999d-d617-42f1-880e-0ba4542319d5")
+    _Windows_SupportedOS_10240_
     IAppxManifestPackageId2 : public IAppxManifestPackageId
@@ -5751,2 +5774,3 @@ EXTERN_C const IID IID_IAppxManifestProperties;
     MIDL_INTERFACE("03faf64d-f26f-4b2c-aaf7-8fe7789b8bca")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestProperties : public IUnknown
@@ -5972,2 +5996,3 @@ EXTERN_C const IID IID_IAppxManifestTargetDeviceFamily;
     MIDL_INTERFACE("9091B09B-C8D5-4F31-8687-A338259FAEFB")
+    _Windows_SupportedOS_10240_
     IAppxManifestTargetDeviceFamily : public IUnknown
@@ -6085,2 +6110,3 @@ EXTERN_C const IID IID_IAppxManifestPackageDependenciesEnumerator;
     MIDL_INTERFACE("b43bbcf9-65a6-42dd-bac0-8c6741e7f5a4")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestPackageDependenciesEnumerator : public IUnknown
@@ -6198,2 +6224,3 @@ EXTERN_C const IID IID_IAppxManifestPackageDependency;
     MIDL_INTERFACE("e4946b59-733e-43f0-a724-3bde4c1285a0")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestPackageDependency : public IUnknown
@@ -6311,2 +6338,3 @@ EXTERN_C const IID IID_IAppxManifestPackageDependency2;
     MIDL_INTERFACE("DDA0B713-F3FF-49D3-898A-2786780C5D98")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestPackageDependency2 : public IAppxManifestPackageDependency
@@ -6513,2 +6541,3 @@ EXTERN_C const IID IID_IAppxManifestResourcesEnumerator;
     MIDL_INTERFACE("de4dfbbd-881a-48bb-858c-d6f2baeae6ed")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestResourcesEnumerator : public IUnknown
@@ -6626,2 +6655,3 @@ EXTERN_C const IID IID_IAppxManifestDeviceCapabilitiesEnumerator;
     MIDL_INTERFACE("30204541-427b-4a1c-bacf-655bf463a540")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestDeviceCapabilitiesEnumerator : public IUnknown
@@ -6852,2 +6882,3 @@ EXTERN_C const IID IID_IAppxManifestApplicationsEnumerator;
     MIDL_INTERFACE("9eb8a55a-f04b-4d0d-808d-686185d4847a")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestApplicationsEnumerator : public IUnknown
@@ -6965,2 +6996,3 @@ EXTERN_C const IID IID_IAppxManifestApplication;
     MIDL_INTERFACE("5da89bf4-3773-46be-b650-7e744863b7e8")
+    _Windows_SupportedOS_Windows8_
     IAppxManifestApplication : public IUnknown
@@ -7295,2 +7327,3 @@ EXTERN_C const IID IID_IAppxBundleFactory;
     MIDL_INTERFACE("BBA65864-965F-4A5F-855F-F074BDBF3A7B")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleFactory : public IUnknown
@@ -7612,2 +7645,3 @@ EXTERN_C const IID IID_IAppxBundleWriter;
     MIDL_INTERFACE("EC446FE8-BFEC-4C64-AB4F-49F038F0C6D2")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleWriter : public IUnknown
@@ -7712,2 +7746,3 @@ EXTERN_C const IID IID_IAppxBundleWriter2;
     MIDL_INTERFACE("6D8FE971-01CC-49A0-B685-233851279962")
+    _Windows_SupportedOS_10240_
     IAppxBundleWriter2 : public IUnknown
@@ -7803,2 +7838,3 @@ EXTERN_C const IID IID_IAppxBundleWriter3;
     MIDL_INTERFACE("AD711152-F969-4193-82D5-9DDF2786D21A")
+    _Windows_SupportedOS_10240_
     IAppxBundleWriter3 : public IUnknown
@@ -7907,2 +7943,3 @@ EXTERN_C const IID IID_IAppxBundleWriter4;
     MIDL_INTERFACE("9CD9D523-5009-4C01-9882-DC029FBD47A3")
+    _Windows_SupportedOS_10240_
     IAppxBundleWriter4 : public IUnknown
@@ -8044,2 +8081,3 @@ EXTERN_C const IID IID_IAppxBundleReader;
     MIDL_INTERFACE("DD75B8C0-BA76-43B0-AE0F-68656A1DC5C8")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleReader : public IUnknown
@@ -8282,2 +8320,3 @@ EXTERN_C const IID IID_IAppxBundleManifestReader;
     MIDL_INTERFACE("CF0EBBC1-CC99-4106-91EB-E67462E04FB0")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleManifestReader : public IUnknown
@@ -8395,2 +8434,3 @@ EXTERN_C const IID IID_IAppxBundleManifestReader2;
     MIDL_INTERFACE("5517DF70-033F-4AF2-8213-87D766805C02")
+    _Windows_SupportedOS_10240_
     IAppxBundleManifestReader2 : public IUnknown
@@ -8482,2 +8522,3 @@ EXTERN_C const IID IID_IAppxBundleManifestPackageInfoEnumerator;
     MIDL_INTERFACE("F9B856EE-49A6-4E19-B2B0-6A2406D63A32")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleManifestPackageInfoEnumerator : public IUnknown
@@ -8595,2 +8636,3 @@ EXTERN_C const IID IID_IAppxBundleManifestPackageInfo;
     MIDL_INTERFACE("54CD06C1-268F-40BB-8ED2-757A9EBAEC8D")
+    _Windows_SupportedOS_Windows81_
     IAppxBundleManifestPackageInfo : public IUnknown
@@ -8747,2 +8789,3 @@ EXTERN_C const IID IID_IAppxBundleManifestPackageInfo2;
     MIDL_INTERFACE("44C2ACBC-B2CF-4CCB-BBDB-9C6DA8C3BC9E")
+    _Windows_SupportedOS_10240_
     IAppxBundleManifestPackageInfo2 : public IUnknown
@@ -9034,2 +9077,3 @@ EXTERN_C const IID IID_IAppxBundleManifestOptionalBundleInfoEnumerator;
     MIDL_INTERFACE("9A178793-F97E-46AC-AACA-DD5BA4C177C8")
+    _Windows_SupportedOS_10240_
     IAppxBundleManifestOptionalBundleInfoEnumerator : public IUnknown
@@ -9147,2 +9191,3 @@ EXTERN_C const IID IID_IAppxBundleManifestOptionalBundleInfo;
     MIDL_INTERFACE("515BF2E8-BCB0-4D69-8C48-E383147B6E12")
+    _Windows_SupportedOS_10240_
     IAppxBundleManifestOptionalBundleInfo : public IUnknown
@@ -9260,2 +9305,3 @@ EXTERN_C const IID IID_IAppxContentGroupFilesEnumerator;
     MIDL_INTERFACE("1a09a2fd-7440-44eb-8c84-848205a6a1cc")
+    _Windows_SupportedOS_10240_
     IAppxContentGroupFilesEnumerator : public IUnknown
@@ -9373,2 +9419,3 @@ EXTERN_C const IID IID_IAppxContentGroup;
     MIDL_INTERFACE("328f6468-c04f-4e3c-b6fa-6b8d27f3003a")
+    _Windows_SupportedOS_10240_
     IAppxContentGroup : public IUnknown
@@ -9473,2 +9520,3 @@ EXTERN_C const IID IID_IAppxContentGroupsEnumerator;
     MIDL_INTERFACE("3264e477-16d1-4d63-823e-7d2984696634")
+    _Windows_SupportedOS_10240_
     IAppxContentGroupsEnumerator : public IUnknown
@@ -9586,2 +9634,3 @@ EXTERN_C const IID IID_IAppxContentGroupMapReader;
     MIDL_INTERFACE("418726d8-dd99-4f5d-9886-157add20de01")
+    _Windows_SupportedOS_10240_
     IAppxContentGroupMapReader : public IUnknown
@@ -9686,2 +9735,3 @@ EXTERN_C const IID IID_IAppxSourceContentGroupMapReader;
     MIDL_INTERFACE("f329791d-540b-4a9f-bc75-3282b7d73193")
+    _Windows_SupportedOS_10240_
     IAppxSourceContentGroupMapReader : public IUnknown
@@ -9786,2 +9836,3 @@ EXTERN_C const IID IID_IAppxContentGroupMapWriter;
     MIDL_INTERFACE("d07ab776-a9de-4798-8c14-3db31e687c78")
+    _Windows_SupportedOS_10240_
     IAppxContentGroupMapWriter : public IUnknown
@@ -10323,2 +10374,3 @@ EXTERN_C const IID IID_IAppxEncryptionFactory;
     MIDL_INTERFACE("80E8E04D-8C88-44AE-A011-7CADF6FB2E72")
+    _Windows_SupportedOS_14393_
     IAppxEncryptionFactory : public IUnknown
@@ -10605,2 +10657,3 @@ EXTERN_C const IID IID_IAppxEncryptionFactory2;
     MIDL_INTERFACE("c1b11eee-c4ba-4ab2-a55d-d015fe8ff64f")
+    _Windows_SupportedOS_10240_
     IAppxEncryptionFactory2 : public IUnknown
@@ -10716,2 +10769,3 @@ EXTERN_C const IID IID_IAppxEncryptionFactory3;
     MIDL_INTERFACE("09edca37-cd64-47d6-b7e8-1cb11d4f7e05")
+    _Windows_SupportedOS_10240_
     IAppxEncryptionFactory3 : public IUnknown
@@ -10918,2 +10972,3 @@ EXTERN_C const IID IID_IAppxEncryptionFactory4;
     MIDL_INTERFACE("A879611F-12FD-41fe-85D5-06AE779BBAF5")
+    _Windows_SupportedOS_10240_
     IAppxEncryptionFactory4 : public IUnknown
@@ -11141,2 +11196,3 @@ EXTERN_C const IID IID_IAppxEncryptedPackageWriter;
     MIDL_INTERFACE("F43D0B0B-1379-40E2-9B29-682EA2BF42AF")
+    _Windows_SupportedOS_14393_
     IAppxEncryptedPackageWriter : public IUnknown
@@ -11245,2 +11301,3 @@ EXTERN_C const IID IID_IAppxEncryptedPackageWriter2;
     MIDL_INTERFACE("3E475447-3A25-40b5-8AD2-F953AE50C92D")
+    _Windows_SupportedOS_14393_
     IAppxEncryptedPackageWriter2 : public IUnknown
@@ -11340,2 +11397,3 @@ EXTERN_C const IID IID_IAppxEncryptedBundleWriter;
     MIDL_INTERFACE("80B0902F-7BF0-4117-B8C6-4279EF81EE77")
+    _Windows_SupportedOS_14393_
     IAppxEncryptedBundleWriter : public IUnknown
@@ -11440,2 +11498,3 @@ EXTERN_C const IID IID_IAppxEncryptedBundleWriter2;
     MIDL_INTERFACE("E644BE82-F0FA-42B8-A956-8D1CB48EE379")
+    _Windows_SupportedOS_10240_
     IAppxEncryptedBundleWriter2 : public IUnknown
@@ -11554,2 +11613,3 @@ EXTERN_C const IID IID_IAppxEncryptedBundleWriter3;
     MIDL_INTERFACE("0D34DEB3-5CAE-4DD3-977C-504932A51D31")
+    _Windows_SupportedOS_10240_
     IAppxEncryptedBundleWriter3 : public IUnknown
@@ -11670,2 +11730,3 @@ EXTERN_C const IID IID_IAppxPackageEditor;
     MIDL_INTERFACE("E2ADB6DC-5E71-4416-86B6-86E5F5291A6B")
+    _Windows_SupportedOS_10240_
     IAppxPackageEditor : public IUnknown
