diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h b/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
index 728afc6bb2..b9bda66ea8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/audioendpoints.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -85,10 +86,11 @@ EXTERN_C const IID IID_IAudioEndpointFormatControl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("784CFD40-9F89-456E-A1A6-873B006A664E")
+    MIDL_INTERFACE("784CFD40-9F89-456E-A1A6-873B006A664E") _Windows_SupportedOS_Windows7_ 
     IAudioEndpointFormatControl : public IUnknown
     {
     public:
+         
         virtual /* [helpstring] */ HRESULT STDMETHODCALLTYPE ResetToDefault( 
             /* [annotation][in] */ 
             _In_  DWORD ResetFlags) = 0;
