diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Audioclient.h b/generation/WinSDK/RecompiledIdlHeaders/um/Audioclient.h
index e668403065..7f0637229f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Audioclient.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Audioclient.h
@@ -181,6 +181,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #if 0
 typedef /* [hidden][restricted] */ LONGLONG REFERENCE_TIME;
 
@@ -201,7 +202,7 @@ enum _AUDCLNT_BUFFERFLAGS
         AUDCLNT_BUFFERFLAGS_TIMESTAMP_ERROR	= 0x4
     } ;
 typedef /* [v1_enum] */ 
-enum AUDCLNT_STREAMOPTIONS
+enum  AUDCLNT_STREAMOPTIONS
     {
         AUDCLNT_STREAMOPTIONS_NONE	= 0,
         AUDCLNT_STREAMOPTIONS_RAW	= 0x1,
@@ -244,7 +245,7 @@ EXTERN_C const IID IID_IAudioClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("1CB9AD4C-DBFA-4c32-B178-C2F568A703B2")
+    MIDL_INTERFACE("1CB9AD4C-DBFA-4c32-B178-C2F568A703B2") _Windows_SupportedOS_WindowsVista_ 
     IAudioClient : public IUnknown
     {
     public:
@@ -274,7 +275,7 @@ EXTERN_C const IID IID_IAudioClient;
             /* [annotation][out] */ 
             _Out_  UINT32 *pNumPaddingFrames) = 0;
         
-        virtual HRESULT STDMETHODCALLTYPE IsFormatSupported( 
+        _WIN32META_ANNOTATION_("win32metadata:can_return_multiple_success_values") _Win32_PreserveResult_ virtual HRESULT STDMETHODCALLTYPE IsFormatSupported( 
             /* [annotation][in] */ 
             _In_  AUDCLNT_SHAREMODE ShareMode,
             /* [annotation][in] */ 
@@ -498,7 +499,7 @@ EXTERN_C const IID IID_IAudioClient2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("726778CD-F60A-4eda-82DE-E47610CD78AA")
+    MIDL_INTERFACE("726778CD-F60A-4eda-82DE-E47610CD78AA") _Windows_SupportedOS_Windows8_ 
     IAudioClient2 : public IAudioClient
     {
     public:
@@ -781,7 +782,7 @@ EXTERN_C const IID IID_IAudioClient3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("7ED4EE07-8E67-4CD4-8C1A-2B7A5987AD42")
+    MIDL_INTERFACE("7ED4EE07-8E67-4CD4-8C1A-2B7A5987AD42") _Windows_SupportedOS_10240_ 
     IAudioClient3 : public IAudioClient2
     {
     public:
@@ -1083,7 +1084,7 @@ EXTERN_C const IID IID_IAudioRenderClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("F294ACFC-3146-4483-A7BF-ADDCA7C260E2")
+    MIDL_INTERFACE("F294ACFC-3146-4483-A7BF-ADDCA7C260E2") _Windows_SupportedOS_WindowsVista_ 
     IAudioRenderClient : public IUnknown
     {
     public:
@@ -1191,7 +1192,7 @@ EXTERN_C const IID IID_IAudioCaptureClient;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C8ADBD64-E71E-48a0-A4DE-185C395CD317")
+    MIDL_INTERFACE("C8ADBD64-E71E-48a0-A4DE-185C395CD317") _Windows_SupportedOS_WindowsVista_ 
     IAudioCaptureClient : public IUnknown
     {
     public:
@@ -1329,7 +1330,7 @@ EXTERN_C const IID IID_IAudioClock;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("CD63314F-3FBA-4a1b-812C-EF96358728E7")
+    MIDL_INTERFACE("CD63314F-3FBA-4a1b-812C-EF96358728E7") _Windows_SupportedOS_WindowsVista_ 
     IAudioClock : public IUnknown
     {
     public:
@@ -1455,7 +1456,7 @@ EXTERN_C const IID IID_IAudioClock2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("6f49ff73-6727-49ac-a008-d98cf5e70048")
+    MIDL_INTERFACE("6f49ff73-6727-49ac-a008-d98cf5e70048") _Windows_SupportedOS_Windows7_ 
     IAudioClock2 : public IUnknown
     {
     public:
@@ -1555,7 +1556,7 @@ EXTERN_C const IID IID_IAudioClockAdjustment;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("f6e4c0a0-46d9-4fb8-be21-57a3ef2b626c")
+    MIDL_INTERFACE("f6e4c0a0-46d9-4fb8-be21-57a3ef2b626c") _Windows_SupportedOS_Windows7_ 
     IAudioClockAdjustment : public IUnknown
     {
     public:
@@ -1651,7 +1652,7 @@ EXTERN_C const IID IID_ISimpleAudioVolume;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("87CE5498-68D6-44E5-9215-6DA47EF883D8")
+    MIDL_INTERFACE("87CE5498-68D6-44E5-9215-6DA47EF883D8") _Windows_SupportedOS_WindowsVista_ 
     ISimpleAudioVolume : public IUnknown
     {
     public:
@@ -2091,7 +2092,7 @@ EXTERN_C const IID IID_IAudioEffectsManager;
         virtual HRESULT STDMETHODCALLTYPE GetAudioEffects( 
             /* [annotation][size_is][size_is][out] */ 
             _Outptr_result_buffer_maybenull_(*numEffects)  AUDIO_EFFECT **effects,
-            /* [out] */ UINT32 *numEffects) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") UINT32 *numEffects) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetAudioEffectState( 
             /* [annotation][in] */ 
@@ -2217,7 +2218,7 @@ EXTERN_C const IID IID_IAudioStreamVolume;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("93014887-242D-4068-8A15-CF5E93B90FE3")
+    MIDL_INTERFACE("93014887-242D-4068-8A15-CF5E93B90FE3") _Windows_SupportedOS_WindowsVista_ 
     IAudioStreamVolume : public IUnknown
     {
     public:
@@ -2582,7 +2583,7 @@ EXTERN_C const IID IID_IChannelAudioVolume;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("1C158861-B533-4B30-B1CF-E853E51C59B8")
+    MIDL_INTERFACE("1C158861-B533-4B30-B1CF-E853E51C59B8") _Windows_SupportedOS_WindowsVista_ 
     IChannelAudioVolume : public IUnknown
     {
     public:
