diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AuthListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AuthListener.h
index 55925a68c9..1c42b2897a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AuthListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/AuthListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * This file defines the alljoyn_authlistener and related functions that provides
@@ -84,7 +85,7 @@ static const uint16_t ALLJOYN_CRED_ONE_TIME_PWD = 0x2001; /**< Indicates the cre
  *          complete.
  */
 typedef QCC_BOOL (AJ_CALL * alljoyn_authlistener_requestcredentials_ptr)(const void* context, const char* authMechanism, const char* peerName, uint16_t authCount,
-                                                                         const char* userName, uint16_t credMask, alljoyn_credentials credentials);
+                                                                         const char* userName, uint16_t credMask, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials credentials);
 
 /**
  * Authentication mechanism asynchronous request for credentials. If the user name is not an empty string
@@ -110,9 +111,9 @@ typedef QCC_BOOL (AJ_CALL * alljoyn_authlistener_requestcredentials_ptr)(const v
  *      - ER_OK if the request is handled.
  *      - ER_NOT_IMPLEMENTED if implementation not found (default)
  */
-typedef QStatus (AJ_CALL * alljoyn_authlistener_requestcredentialsasync_ptr)(const void* context, alljoyn_authlistener listener,
+typedef QStatus (AJ_CALL * alljoyn_authlistener_requestcredentialsasync_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener,
                                                                              const char* authMechanism, const char* peerName, uint16_t authCount,
-                                                                             const char* userName, uint16_t credMask, void* authContext);
+                                                                             const char* userName, uint16_t credMask, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* authContext);
 
 /**
  * Respond to a call to alljoyn_authlistener_requestcredentialsasync_ptr.
@@ -125,7 +126,7 @@ typedef QStatus (AJ_CALL * alljoyn_authlistener_requestcredentialsasync_ptr)(con
  * @return   Returns ER_OK if the credential verification response was expected. Returns an error status if
  *           the credentials verification response was not expected.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_authlistener_requestcredentialsresponse(alljoyn_authlistener listener, void* authContext, QCC_BOOL accept, alljoyn_credentials credentials);
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_authlistener_requestcredentialsresponse(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* authContext, _WIN32META_ANNOTATION_("win32metadata:in") QCC_BOOL accept, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials credentials);
 
 /**
  * Type for the VerifyCredentials callback.
@@ -143,7 +144,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_authlistener_requestcredentialsresponse(al
  *          credentials are being rejected.
  */
 typedef QCC_BOOL (AJ_CALL * alljoyn_authlistener_verifycredentials_ptr)(const void* context, const char* authMechanism, const char* peerName,
-                                                                        const alljoyn_credentials credentials);
+                                                                        _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials credentials);
 
 /**
  * Authentication mechanism asynchronous request for verification of credentials from a remote peer.
@@ -162,8 +163,8 @@ typedef QCC_BOOL (AJ_CALL * alljoyn_authlistener_verifycredentials_ptr)(const vo
  *  - ER_OK if the request is handled.
  *  - ER_NOT_IMPLEMENTED (default)
  */
-typedef QStatus (AJ_CALL * alljoyn_authlistener_verifycredentialsasync_ptr)(const void* context, alljoyn_authlistener listener,
-                                                                            const char* authMechanism, const char* peerName, const alljoyn_credentials credentials, void* authContext);
+typedef QStatus (AJ_CALL * alljoyn_authlistener_verifycredentialsasync_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener,
+                                                                            const char* authMechanism, const char* peerName, _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials credentials, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* authContext);
 
 /**
  * Respond to a call to alljoyn_authlistener_verifycredentialsasync_ptr.
@@ -175,7 +176,7 @@ typedef QStatus (AJ_CALL * alljoyn_authlistener_verifycredentialsasync_ptr)(cons
  * @return   Returns ER_OK if the credential verification response was expected. Returns an error status if
  *           the credentials verification response was not expected.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_authlistener_verifycredentialsresponse(alljoyn_authlistener listener, void* authContext, QCC_BOOL accept);
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_authlistener_verifycredentialsresponse(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* authContext, _WIN32META_ANNOTATION_("win32metadata:in") QCC_BOOL accept);
 /**
  * Type for the SecurityViolation callback.
  *
@@ -188,7 +189,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_authlistener_verifycredentialsresponse(all
  * @param status   A status code indicating the type of security violation.
  * @param msg      The message that cause the security violation.
  */
-typedef void (AJ_CALL * alljoyn_authlistener_securityviolation_ptr)(const void* context, QStatus status, const alljoyn_message msg);
+typedef void (AJ_CALL * alljoyn_authlistener_securityviolation_ptr)(const void* context, QStatus status, _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_message msg);
 
 /**
  * Type for the AuthenticationComplete callback.
@@ -279,7 +280,7 @@ typedef struct {
  *
  * @return Handle to newly allocated alljoyn_authlistener.
  */
-extern AJ_API alljoyn_authlistener AJ_CALL alljoyn_authlistener_create(const alljoyn_authlistener_callbacks* callbacks, const void* context);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_authlistener AJ_CALL alljoyn_authlistener_create(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_authlistener_callbacks* callbacks, _WIN32META_ANNOTATION_("win32metadata:in") const void* context);
 
 /**
  * Create an alljoyn_authlistener which will trigger the provided callbacks, passing along the provided context.
@@ -289,21 +290,21 @@ extern AJ_API alljoyn_authlistener AJ_CALL alljoyn_authlistener_create(const all
  *
  * @return Handle to newly allocated alljoyn_authlistener.
  */
-extern AJ_API alljoyn_authlistener AJ_CALL alljoyn_authlistenerasync_create(const alljoyn_authlistenerasync_callbacks* callbacks, const void* context);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_authlistener AJ_CALL alljoyn_authlistenerasync_create(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_authlistenerasync_callbacks* callbacks, _WIN32META_ANNOTATION_("win32metadata:in") const void* context);
 
 /**
  * Destroy an alljoyn_authlistener.
  *
  * @param listener alljoyn_authlistener to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_authlistener_destroy(alljoyn_authlistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_authlistener_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener);
 
 /**
  * Destroy an alljoyn_authlistener.
  *
  * @param listener alljoyn_authlistener to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_authlistenerasync_destroy(alljoyn_authlistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_authlistenerasync_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener);
 
 /**
  * Sets a shared secret to be returned by the listener during ECDHE_SPEKE authentication.
@@ -316,21 +317,21 @@ extern AJ_API void AJ_CALL alljoyn_authlistenerasync_destroy(alljoyn_authlistene
  *          - #ER_OK        If successful.
  *          - #ER_BAD_ARG_2 If the secret is shorter than 4 bytes.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_authlistener_setsharedsecret(alljoyn_authlistener listener, const uint8_t* sharedSecret, size_t sharedSecretSize);
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_authlistener_setsharedsecret(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_authlistener listener, _WIN32META_ANNOTATION_("win32metadata:in") const uint8_t* sharedSecret, _WIN32META_ANNOTATION_("win32metadata:in") size_t sharedSecretSize);
 
 /**
  * Create credentials
  *
  * @return Newly created credentials.
  */
-extern AJ_API alljoyn_credentials AJ_CALL alljoyn_credentials_create();
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_credentials AJ_CALL alljoyn_credentials_create();
 
 /**
  * Destroy credentials
  *
  * @param cred Credentials to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_destroy(alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred);
 
 /**
  * Tests if one or more credentials are set.
@@ -339,7 +340,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_destroy(alljoyn_credentials cred)
  * @param creds  A logical or of the credential bit values.
  * @return true if the credentials are set.
  */
-extern AJ_API QCC_BOOL AJ_CALL alljoyn_credentials_isset(const alljoyn_credentials cred, uint16_t creds);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API QCC_BOOL AJ_CALL alljoyn_credentials_isset(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") uint16_t creds);
 
 /**
  * Sets a requested password.
@@ -347,7 +348,7 @@ extern AJ_API QCC_BOOL AJ_CALL alljoyn_credentials_isset(const alljoyn_credentia
  * @param cred The credentials to set.
  * @param pwd  The password to set.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setpassword(alljoyn_credentials cred, const char* pwd);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setpassword(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") const char* pwd);
 
 /**
  * Sets a requested user name.
@@ -355,7 +356,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setpassword(alljoyn_credentials c
  * @param cred      The credentials to set.
  * @param userName  The user name to set.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setusername(alljoyn_credentials cred, const char* userName);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setusername(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") const char* userName);
 
 /**
  * Sets a requested public key certificate chain. The certificates must be PEM encoded.
@@ -363,7 +364,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setusername(alljoyn_credentials c
  * @param cred       The credentials to set.
  * @param certChain  The certificate chain to set.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setcertchain(alljoyn_credentials cred, const char* certChain);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setcertchain(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") const char* certChain);
 
 /**
  * Sets a requested private key. The private key must be PEM encoded and may be encrypted. If
@@ -372,7 +373,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setcertchain(alljoyn_credentials
  * @param cred The credentials to set.
  * @param pk   The private key to set.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setprivatekey(alljoyn_credentials cred, const char* pk);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setprivatekey(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") const char* pk);
 
 /**
  * Sets a logon entry. For example for the Secure Remote Password protocol in RFC 5054, a
@@ -383,7 +384,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setprivatekey(alljoyn_credentials
  * @param cred        The credentials to set.
  * @param logonEntry  The logon entry to set.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setlogonentry(alljoyn_credentials cred, const char* logonEntry);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setlogonentry(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") const char* logonEntry);
 
 /**
  * Sets an expiration time in seconds relative to the current time for the credentials. This value is optional and
@@ -394,7 +395,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setlogonentry(alljoyn_credentials
  * @param cred        The credentials to set.
  * @param expiration  The expiration time in seconds.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_setexpiration(alljoyn_credentials cred, uint32_t expiration);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_setexpiration(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred, _WIN32META_ANNOTATION_("win32metadata:in") uint32_t expiration);
 
 /**
  * Gets the password from this credentials instance.
@@ -402,7 +403,7 @@ extern AJ_API void AJ_CALL alljoyn_credentials_setexpiration(alljoyn_credentials
  * @param cred The credentials to query.
  * @return A password or an empty string.
  */
-extern AJ_API const char* AJ_CALL alljoyn_credentials_getpassword(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_credentials_getpassword(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Gets the user name from this credentials instance.
@@ -410,7 +411,7 @@ extern AJ_API const char* AJ_CALL alljoyn_credentials_getpassword(const alljoyn_
  * @param cred The credentials to query.
  * @return A user name or an empty string.
  */
-extern AJ_API const char* AJ_CALL alljoyn_credentials_getusername(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_credentials_getusername(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Gets the PEM encoded X509 certificate chain from this credentials instance.
@@ -418,7 +419,7 @@ extern AJ_API const char* AJ_CALL alljoyn_credentials_getusername(const alljoyn_
  * @param cred The credentials to query.
  * @return An X509 certificate chain or an empty string.
  */
-extern AJ_API const char* AJ_CALL alljoyn_credentials_getcertchain(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_credentials_getcertchain(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Gets the PEM encoded private key from this credentials instance.
@@ -426,7 +427,7 @@ extern AJ_API const char* AJ_CALL alljoyn_credentials_getcertchain(const alljoyn
  * @param cred The credentials to query.
  * @return An PEM encode private key or an empty string.
  */
-extern AJ_API const char* AJ_CALL alljoyn_credentials_getprivateKey(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_credentials_getprivateKey(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Gets a logon entry.
@@ -434,7 +435,7 @@ extern AJ_API const char* AJ_CALL alljoyn_credentials_getprivateKey(const alljoy
  * @param cred The credentials to query.
  * @return An encoded logon entry or an empty string.
  */
-extern AJ_API const char* AJ_CALL alljoyn_credentials_getlogonentry(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API const char* AJ_CALL alljoyn_credentials_getlogonentry(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Get the expiration time in seconds if it is set.
@@ -442,14 +443,14 @@ extern AJ_API const char* AJ_CALL alljoyn_credentials_getlogonentry(const alljoy
  * @param cred The credentials to query.
  * @return The expiration or the max 32 bit unsigned value if it was not set.
  */
-extern AJ_API uint32_t AJ_CALL alljoyn_credentials_getexpiration(const alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API uint32_t AJ_CALL alljoyn_credentials_getexpiration(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_credentials cred);
 
 /**
  * Clear the credentials.
  *
  * @param cred The credentials to clear.
  */
-extern AJ_API void AJ_CALL alljoyn_credentials_clear(alljoyn_credentials cred);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_credentials_clear(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_credentials cred);
 
 #ifdef __cplusplus
 } /* extern "C" */
