diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/avrfsdk.h b/generation/WinSDK/RecompiledIdlHeaders/um/avrfsdk.h
index a2c5a1128b..559e5a02d8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/avrfsdk.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/avrfsdk.h
@@ -24,6 +24,7 @@ Revision History:
 #ifndef _AVRFSDK_
 #define _AVRFSDK_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -40,7 +41,7 @@ extern "C" {
 
 #define AVRF_MAX_TRACES 32
 
-typedef struct _AVRF_BACKTRACE_INFORMATION {
+typedef struct   _AVRF_BACKTRACE_INFORMATION {
     ULONG   Depth;
     ULONG   Index;
     ULONG64 ReturnAddresses[AVRF_MAX_TRACES];
@@ -64,7 +65,7 @@ enum eHeapEnumerationLevel {
     HeapEnumerationStop = 0xFFFFFFFF,
 };
 
-typedef struct _AVRF_HEAP_ALLOCATION {
+typedef struct   _AVRF_HEAP_ALLOCATION {
     ULONG64 HeapHandle;
     ULONG64 UserAllocation;    
     ULONG64 UserAllocationSize;
@@ -84,7 +85,7 @@ enum eHANDLE_TRACE_OPERATIONS {
     OperationDbBADREF
 };
 
-typedef struct _AVRF_HANDLE_OPERATION {
+typedef struct   _AVRF_HANDLE_OPERATION {
     ULONG64 Handle;
     ULONG  ProcessId;
     ULONG  ThreadId;
@@ -106,9 +107,9 @@ enum eAvrfResourceTypes {
 
 typedef ULONG 
 (NTAPI * AVRF_RESOURCE_ENUMERATE_CALLBACK)(
-    PVOID ResourceDescription,
-    PVOID EnumerationContext,
-    PULONG EnumerationLevel
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PVOID ResourceDescription,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PVOID EnumerationContext,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PULONG EnumerationLevel
     );
 
 //
@@ -117,9 +118,9 @@ typedef ULONG
 
 typedef ULONG 
 (NTAPI * AVRF_HEAPALLOCATION_ENUMERATE_CALLBACK)(
-    PAVRF_HEAP_ALLOCATION HeapAllocation,
-    PVOID EnumerationContext,
-    PULONG EnumerationLevel
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PAVRF_HEAP_ALLOCATION HeapAllocation,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PVOID EnumerationContext,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PULONG EnumerationLevel
     );
 
 //
@@ -128,9 +129,9 @@ typedef ULONG
 
 typedef ULONG 
 (NTAPI * AVRF_HANDLEOPERATION_ENUMERATE_CALLBACK)(
-    PAVRF_HANDLE_OPERATION HandleOperation,
-    PVOID EnumerationContext,
-    PULONG EnumerationLevel
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PAVRF_HANDLE_OPERATION HandleOperation,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PVOID EnumerationContext,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PULONG EnumerationLevel
     );
 
 //
@@ -140,11 +141,11 @@ typedef ULONG
 #define AVRF_ENUM_RESOURCES_FLAGS_SUSPEND             0x00000001
 #define AVRF_ENUM_RESOURCES_FLAGS_DONT_RESOLVE_TRACES 0x00000002
 
-ULONG NTAPI
+_Win32_ImportLibrary_("verifier.dll") ULONG NTAPI
 VerifierEnumerateResource(
     HANDLE Process,
-    ULONG  Flags,    
-    ULONG  ResourceType,
+    _Win32_AssociatedEnum_(VERIFIER_ENUM_RESOURCE_FLAGS) ULONG  Flags,    
+    _Win32_AssociatedEnum_(eAvrfResourceTypes) ULONG  ResourceType,
     AVRF_RESOURCE_ENUMERATE_CALLBACK ResourceCallback,
     PVOID  EnumerationContext
     );
@@ -166,4 +167,19 @@ VerifierIsPerUserSettingsEnabled(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("AVRF_ENUM_RESOURCES_FLAGS_DONT_RESOLVE_TRACES")
+#undef AVRF_ENUM_RESOURCES_FLAGS_DONT_RESOLVE_TRACES
+#pragma push_macro("AVRF_ENUM_RESOURCES_FLAGS_SUSPEND")
+#undef AVRF_ENUM_RESOURCES_FLAGS_SUSPEND
+enum [[clang::flag_enum]] VERIFIER_ENUM_RESOURCE_FLAGS : unsigned int
+{
+    AVRF_ENUM_RESOURCES_FLAGS_DONT_RESOLVE_TRACES = 2U,
+    AVRF_ENUM_RESOURCES_FLAGS_SUSPEND = 1U,
+};
+#pragma pop_macro("AVRF_ENUM_RESOURCES_FLAGS_SUSPEND")
+#pragma pop_macro("AVRF_ENUM_RESOURCES_FLAGS_DONT_RESOLVE_TRACES")
+#endif
+
 #endif /*AVRFSDK*/
