diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/bits4_0.h b/generation/WinSDK/RecompiledIdlHeaders/um/bits4_0.h
index 7592bc21da434f6f0b8a69a38d5b701e10c7cfcd..f5c6094e989cbd700e412f8b1a8dd5fcedc39e69 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/bits4_0.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/bits4_0.h
@@ -103,6 +103,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -124,7 +125,7 @@ EXTERN_C const IID IID_IBitsTokenOptions;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("9a2584c3-f7d2-457a-9a5e-22b67bffc7d2")
-    IBitsTokenOptions : public IUnknown
+    _Windows_SupportedOS_Windows7_ IBitsTokenOptions : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE SetHelperTokenFlags( 
@@ -195,7 +196,7 @@ EXTERN_C const IID IID_IBitsTokenOptions;
         END_INTERFACE
     } IBitsTokenOptionsVtbl;
 
-    interface IBitsTokenOptions
+    interface _Windows_SupportedOS_Windows7_ IBitsTokenOptions
     {
         CONST_VTBL struct IBitsTokenOptionsVtbl *lpVtbl;
     };
@@ -253,7 +254,7 @@ EXTERN_C const IID IID_IBackgroundCopyFile4;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("ef7e0655-7888-4960-b0e5-730846e03492")
-    IBackgroundCopyFile4 : public IBackgroundCopyFile3
+    _Windows_SupportedOS_Windows7_ IBackgroundCopyFile4 : public IBackgroundCopyFile3
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetPeerDownloadStats( 
@@ -353,7 +354,7 @@ EXTERN_C const IID IID_IBackgroundCopyFile4;
         END_INTERFACE
     } IBackgroundCopyFile4Vtbl;
 
-    interface IBackgroundCopyFile4
+    interface _Windows_SupportedOS_Windows7_ IBackgroundCopyFile4
     {
         CONST_VTBL struct IBackgroundCopyFile4Vtbl *lpVtbl;
     };
