diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Bits3_0.h b/generation/WinSDK/RecompiledIdlHeaders/um/Bits3_0.h
index f397dc1bff9ab2ef13f4d26bab106eddd0d26267..a6dd82f5839020dfecc34300253cc5c1e3ac4cb2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Bits3_0.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Bits3_0.h
@@ -186,6 +186,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -205,7 +206,7 @@ EXTERN_C const IID IID_IBitsPeerCacheRecord;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdeaf-489e-11d9-a9cd-000d56965251")
-    IBitsPeerCacheRecord : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeerCacheRecord : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetId( 
@@ -296,7 +297,7 @@ EXTERN_C const IID IID_IBitsPeerCacheRecord;
         END_INTERFACE
     } IBitsPeerCacheRecordVtbl;
 
-    interface IBitsPeerCacheRecord
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeerCacheRecord
     {
         CONST_VTBL struct IBitsPeerCacheRecordVtbl *lpVtbl;
     };
@@ -360,7 +361,7 @@ EXTERN_C const IID IID_IEnumBitsPeerCacheRecords;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdea4-489e-11d9-a9cd-000d56965251")
-    IEnumBitsPeerCacheRecords : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IEnumBitsPeerCacheRecords : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE Next( 
@@ -443,7 +444,7 @@ EXTERN_C const IID IID_IEnumBitsPeerCacheRecords;
         END_INTERFACE
     } IEnumBitsPeerCacheRecordsVtbl;
 
-    interface IEnumBitsPeerCacheRecords
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IEnumBitsPeerCacheRecords
     {
         CONST_VTBL struct IEnumBitsPeerCacheRecordsVtbl *lpVtbl;
     };
@@ -501,7 +502,7 @@ EXTERN_C const IID IID_IBitsPeer;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdea2-489e-11d9-a9cd-000d56965251")
-    IBitsPeer : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeer : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetPeerName( 
@@ -556,7 +557,7 @@ EXTERN_C const IID IID_IBitsPeer;
         END_INTERFACE
     } IBitsPeerVtbl;
 
-    interface IBitsPeer
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeer
     {
         CONST_VTBL struct IBitsPeerVtbl *lpVtbl;
     };
@@ -608,7 +609,7 @@ EXTERN_C const IID IID_IEnumBitsPeers;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdea5-489e-11d9-a9cd-000d56965251")
-    IEnumBitsPeers : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IEnumBitsPeers : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE Next( 
@@ -691,7 +692,7 @@ EXTERN_C const IID IID_IEnumBitsPeers;
         END_INTERFACE
     } IEnumBitsPeersVtbl;
 
-    interface IEnumBitsPeers
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IEnumBitsPeers
     {
         CONST_VTBL struct IEnumBitsPeersVtbl *lpVtbl;
     };
@@ -752,7 +753,7 @@ EXTERN_C const IID IID_IBitsPeerCacheAdministration;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdead-489e-11d9-a9cd-000d56965251")
-    IBitsPeerCacheAdministration : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeerCacheAdministration : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetMaximumCacheSize( 
@@ -899,7 +900,7 @@ EXTERN_C const IID IID_IBitsPeerCacheAdministration;
         END_INTERFACE
     } IBitsPeerCacheAdministrationVtbl;
 
-    interface IBitsPeerCacheAdministration
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBitsPeerCacheAdministration
     {
         CONST_VTBL struct IBitsPeerCacheAdministrationVtbl *lpVtbl;
     };
@@ -987,7 +988,7 @@ EXTERN_C const IID IID_IBackgroundCopyJob4;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdeae-489e-11d9-a9cd-000d56965251")
-    IBackgroundCopyJob4 : public IBackgroundCopyJob3
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyJob4 : public IBackgroundCopyJob3
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE SetPeerCachingFlags( 
@@ -1343,7 +1344,7 @@ EXTERN_C const IID IID_IBackgroundCopyJob4;
         END_INTERFACE
     } IBackgroundCopyJob4Vtbl;
 
-    interface IBackgroundCopyJob4
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyJob4
     {
         CONST_VTBL struct IBackgroundCopyJob4Vtbl *lpVtbl;
     };
@@ -1539,7 +1540,7 @@ EXTERN_C const IID IID_IBackgroundCopyFile3;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdeaa-489e-11d9-a9cd-000d56965251")
-    IBackgroundCopyFile3 : public IBackgroundCopyFile2
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyFile3 : public IBackgroundCopyFile2
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetTemporaryName( 
@@ -1641,7 +1642,7 @@ EXTERN_C const IID IID_IBackgroundCopyFile3;
         END_INTERFACE
     } IBackgroundCopyFile3Vtbl;
 
-    interface IBackgroundCopyFile3
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyFile3
     {
         CONST_VTBL struct IBackgroundCopyFile3Vtbl *lpVtbl;
     };
@@ -1713,7 +1714,7 @@ EXTERN_C const IID IID_IBackgroundCopyCallback2;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("659cdeac-489e-11d9-a9cd-000d56965251")
-    IBackgroundCopyCallback2 : public IBackgroundCopyCallback
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyCallback2 : public IBackgroundCopyCallback
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE FileTransferred( 
@@ -1780,7 +1781,7 @@ EXTERN_C const IID IID_IBackgroundCopyCallback2;
         END_INTERFACE
     } IBackgroundCopyCallback2Vtbl;
 
-    interface IBackgroundCopyCallback2
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IBackgroundCopyCallback2
     {
         CONST_VTBL struct IBackgroundCopyCallback2Vtbl *lpVtbl;
     };
