diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothapis.h
index b36ff097..25a19adc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/bluetoothapis.h
@@ -5,6 +5,9 @@
 
 #pragma once
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -133,13 +136,16 @@ typedef HANDLE      HBLUETOOTH_RADIO_FIND;
 //      any other
 //          Success. The return handle is valid and phRadio points to a valid handle.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
+_Win32_RAIIFree_(BluetoothFindRadioClose, 0)
 _Must_inspect_result_
 _Success_(return != NULL)
 HBLUETOOTH_RADIO_FIND
 WINAPI
 BluetoothFindFirstRadio(
     _In_ const BLUETOOTH_FIND_RADIO_PARAMS * pbtfrp,
-    _Out_ HANDLE * phRadio
+    _Out_ _Win32_RAIIFree_(CloseHandle, 0) HANDLE * phRadio
     );
 
 //
@@ -172,13 +178,15 @@ BluetoothFindFirstRadio(
 //
 //          other Win32 errors
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 _Success_(return)
 BOOL
 WINAPI
 BluetoothFindNextRadio(
     _In_  HBLUETOOTH_RADIO_FIND hFind,
-    _Out_ HANDLE * phRadio
+    _Out_ _Win32_RAIIFree_(CloseHandle, 0) HANDLE * phRadio
     );
 
 //
@@ -199,6 +207,8 @@ BluetoothFindNextRadio(
 //          ERROR_INVALID_HANDLE
 //              The handle is NULL.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 BOOL
 WINAPI
 BluetoothFindRadioClose(
@@ -251,6 +261,8 @@ typedef struct _BLUETOOTH_RADIO_INFO {
 //
 //      other Win32 error codes.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -418,6 +430,9 @@ typedef HANDLE      HBLUETOOTH_DEVICE_FIND;
 //      any other value
 //          Success. The return handle is valid and pbtdi points to valid data.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
+_Win32_RAIIFree_(BluetoothFindDeviceClose, 0)
 _Must_inspect_result_
 _Success_(return != NULL)
 HBLUETOOTH_DEVICE_FIND
@@ -462,6 +477,8 @@ BluetoothFindFirstDevice(
 //
 //          other Win32 errors
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -488,6 +505,8 @@ BluetoothFindNextDevice(
 //          ERROR_INVALID_HANDLE
 //              The handle is NULL.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 BOOL
 WINAPI
 BluetoothFindDeviceClose(
@@ -532,6 +551,7 @@ BluetoothFindDeviceClose(
 //
 //      other error codes
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -567,6 +587,7 @@ BluetoothGetDeviceInfo(
 //
 //      other Win32 error codes.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -587,6 +608,7 @@ BluetoothUpdateDeviceRecord(
 //          The device was not found. If no Bluetooth radio is installed,
 //          the devices could not be enumerated or removed.
 //
+_Windows_SupportedOS_WindowsVista_
 DWORD
 WINAPI
 BluetoothRemoveDevice(
@@ -706,6 +728,8 @@ typedef struct _BLUETOOTH_SELECT_DEVICE_PARAMS {
 //
 //          other WIN32 errors
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -727,6 +751,7 @@ BluetoothSelectDevices(
 //      FALSE
 //          Nothing to free.
 //
+_Windows_SupportedOS_WindowsVista_
 BOOL
 WINAPI
 BluetoothSelectDevicesFree(
@@ -761,6 +786,8 @@ BluetoothSelectDevicesFree(
 //          Failure. The property page was not displayed. Check GetLastError
 //          for more details.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 BOOL
 WINAPI
 BluetoothDisplayDeviceProperties(
@@ -849,6 +876,7 @@ BluetoothDisplayDeviceProperties(
 //          { BTH_ERROR_UNSPECIFIED_ERROR,      ERROR_NOT_READY },
 //          { BTH_ERROR_LOCAL_HOST_TERMINATED_CONNECTION, ERROR_VC_DISCONNECTED },
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -967,6 +995,7 @@ typedef struct _BLUETOOTH_PASSKEY_INFO {
 //          { BTH_ERROR_UNSPECIFIED_ERROR,      ERROR_NOT_READY },
 //          { BTH_ERROR_LOCAL_HOST_TERMINATED_CONNECTION, ERROR_VC_DISCONNECTED },
 //
+_Windows_SupportedOS_WindowsXP_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1023,6 +1052,7 @@ BluetoothAuthenticateDeviceEx(
 //      other WIN32 error
 //          Failure. Return value is the error code.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1092,6 +1122,7 @@ BluetoothAuthenticateMultipleDevices(
 //      other WIN32 error
 //          The call failed.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1134,6 +1165,7 @@ BluetoothSetServiceState(
 //      other WIN32 errors
 //          The call failed.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -1175,6 +1207,7 @@ BluetoothEnumerateInstalledServices(
 //          State was not changed. If the caller specified NULL for hRadio, all
 //          of the radios did not accept the state change.
 //
+_Windows_SupportedOS_WindowsVista_
 BOOL
 WINAPI
 BluetoothEnableDiscovery(
@@ -1200,6 +1233,7 @@ BluetoothEnableDiscovery(
 //      FALSE
 //          No radios are discoverable.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -1236,6 +1270,7 @@ BluetoothIsDiscoverable(
 //          State was not changed. If the caller specified NULL for hRadio, all
 //          of the radios did not accept the state change.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -1262,6 +1297,7 @@ BluetoothEnableIncomingConnections(
 //      FALSE
 //          No radios are allowing incoming connections.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -1314,14 +1350,16 @@ typedef BOOL (CALLBACK *PFN_AUTHENTICATION_CALLBACK)(LPVOID pvParam, PBLUETOOTH_
 //      other Win32 error.
 //          Failure. The registration handle is invalid.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 DWORD
 WINAPI
 BluetoothRegisterForAuthentication(
     _In_opt_ const BLUETOOTH_DEVICE_INFO * pbtdi,
-    _Out_ HBLUETOOTH_AUTHENTICATION_REGISTRATION * phRegHandle,
-    _In_opt_ PFN_AUTHENTICATION_CALLBACK pfnCallback,
-    _In_opt_ PVOID pvParam
+    _Out_ _Win32_RAIIFree_(BluetoothUnregisterAuthentication, 0) HBLUETOOTH_AUTHENTICATION_REGISTRATION * phRegHandle,
+    _In_opt_ _Win32_Retained_ PFN_AUTHENTICATION_CALLBACK pfnCallback,
+    _In_opt_ _Win32_Retained_ PVOID pvParam
     );
 
 //
@@ -1372,14 +1410,15 @@ typedef BOOL (CALLBACK *PFN_AUTHENTICATION_CALLBACK_EX)(_In_opt_ LPVOID pvParam,
 //      other Win32 error.
 //          Failure. The registration handle is invalid.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
 BluetoothRegisterForAuthenticationEx(
       _In_opt_ const BLUETOOTH_DEVICE_INFO * pbtdiIn
-    , _Out_ HBLUETOOTH_AUTHENTICATION_REGISTRATION * phRegHandleOut
-    , _In_opt_ PFN_AUTHENTICATION_CALLBACK_EX pfnCallbackIn
-    , _In_opt_ PVOID pvParam
+    , _Out_ _Win32_RAIIFree_(BluetoothUnregisterAuthentication, 0) HBLUETOOTH_AUTHENTICATION_REGISTRATION * phRegHandleOut
+    , _In_opt_ _Win32_Retained_ PFN_AUTHENTICATION_CALLBACK_EX pfnCallbackIn
+    , _In_opt_ _Win32_Retained_ PVOID pvParam
     );
 
 #endif // >= SP1+KB942567
@@ -1407,6 +1446,8 @@ BluetoothRegisterForAuthenticationEx(
 //
 //          other Win32 errors.
 //
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 BOOL
 WINAPI
 BluetoothUnregisterAuthentication(
@@ -1446,6 +1487,7 @@ BluetoothUnregisterAuthentication(
 //
 //      other Win32 error codes
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1517,6 +1559,7 @@ typedef struct _BLUETOOTH_AUTHENTICATE_RESPONSE {
 //
 //      other Win32 error codes
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1637,6 +1680,7 @@ typedef struct _SDP_ELEMENT_DATA {
 //      ERROR_SUCCESS
 //          the sdp element was parsed correctly
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -1708,6 +1752,7 @@ typedef HANDLE HBLUETOOTH_CONTAINER_ELEMENT;
 // }
 //
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -1762,6 +1807,7 @@ BluetoothSdpGetContainerElementData(
 //      printf("record handle is 0x%x\n", data.data.uint32);
 // }
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 _Success_(return == ERROR_SUCCESS)
 DWORD
@@ -1849,6 +1895,7 @@ typedef struct _SDP_STRING_TYPE_DATA {
 //
 //      Other HRESULTs returned by COM
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 _Success_(return == 0)
 DWORD
@@ -1898,6 +1945,8 @@ typedef BOOL (CALLBACK *PFN_BLUETOOTH_ENUM_ATTRIBUTES_CALLBACK)(
 //
 #define BluetoothEnumAttributes BluetoothSdpEnumAttributes
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 _Must_inspect_result_
 BOOL
 WINAPI
@@ -1916,6 +1965,7 @@ BluetoothSdpEnumAttributes(
 // The following APIs are only available on Vista or later
 //
 
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 DWORD
 WINAPI
@@ -1948,6 +1998,7 @@ BluetoothSetLocalServiceInfo(
 //
 // Note this function is only exported in version 2.1 and later.
 //
+_Windows_SupportedOS_WindowsVista_
 _Must_inspect_result_
 BOOL
 WINAPI
