diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/BusListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/BusListener.h
index 629bddeeab..534677dc9c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/BusListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/BusListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * BusListener is an abstract base class (interface) implemented by users of the
@@ -175,14 +176,14 @@ typedef struct {
  *
  * @return Handle to newly allocated alljoyn_buslistener.
  */
-extern AJ_API alljoyn_buslistener AJ_CALL alljoyn_buslistener_create(const alljoyn_buslistener_callbacks* callbacks, const void* context);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_buslistener AJ_CALL alljoyn_buslistener_create(const alljoyn_buslistener_callbacks* callbacks, const void* context);
 
 /**
  * Destroy an alljoyn_buslistener.
  *
  * @param listener alljoyn_buslistener to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_buslistener_destroy(alljoyn_buslistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_buslistener_destroy(alljoyn_buslistener listener);
 
 #ifdef __cplusplus
 } /* extern "C" */
