diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/calendarapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/calendarapis.h
index 0ff5496277..9766ce2e4a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/calendarapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/calendarapis.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 #include <windows.h>
 #include <WinNls.h>
 
@@ -5,7 +6,7 @@
 extern "C" {
 #endif
 
-enum CALDATETIME_DATEUNIT {
+enum   CALDATETIME_DATEUNIT {
   EraUnit, 
   YearUnit, 
   MonthUnit, 
@@ -17,7 +18,7 @@ enum CALDATETIME_DATEUNIT {
   TickUnit 
 };
 
-typedef struct _caldatetime {
+typedef struct   _caldatetime {
   CALID CalId;
   UINT Era;
   UINT Year;
@@ -30,12 +31,14 @@ typedef struct _caldatetime {
   ULONG Tick;
 } CALDATETIME, *LPCALDATETIME;
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL GetCalendarSupportedDateRange(
   _In_ CALID Calendar,
   _Out_ LPCALDATETIME lpCalMinDateTime,
   _Out_ LPCALDATETIME lpCalMaxDateTime
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL GetCalendarDateFormatEx(
   _In_ LPCWSTR lpszLocale,
   _In_ DWORD dwFlags,
@@ -45,27 +48,32 @@ BOOL GetCalendarDateFormatEx(
   _In_ int cchDate
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL ConvertSystemTimeToCalDateTime(
   _In_ const SYSTEMTIME *lpSysTime,
   _In_ CALID calId,
   _Out_ LPCALDATETIME lpCalDateTime
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL UpdateCalendarDayOfWeek(
   _Inout_ LPCALDATETIME lpCalDateTime
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL AdjustCalendarDate(
   _Inout_ LPCALDATETIME lpCalDateTime,
   _In_ CALDATETIME_DATEUNIT calUnit,
   _Out_ INT amount
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL ConvertCalDateTimeToSystemTime(
   _In_ const LPCALDATETIME lpCalDateTime,
   _Out_ SYSTEMTIME *lpSysTime
 );
 
+   _Win32_ImportLibrary_("Kernel32.dll")
 BOOL IsCalendarLeapYear(
   _In_ CALID calId,
   _In_ UINT year,
