diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/callobj.h b/generation/WinSDK/RecompiledIdlHeaders/um/callobj.h
index 1aaef4223e..26c8749a3d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/callobj.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/callobj.h
@@ -108,6 +108,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -198,8 +199,8 @@ EXTERN_C const IID IID_ICallFrame;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("D573B4B0-894E-11d2-B8B6-00C04FB9618A")
-    ICallFrame : public IUnknown
+    MIDL_INTERFACE("D573B4B0-894E-11d2-B8B6-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
+      ICallFrame : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetInfo( 
@@ -542,7 +543,7 @@ EXTERN_C const IID IID_ICallFrame;
         END_INTERFACE
     } ICallFrameVtbl;
 
-    interface ICallFrame
+    interface _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_ ICallFrame
     {
         CONST_VTBL struct ICallFrameVtbl *lpVtbl;
     };
@@ -641,8 +642,8 @@ EXTERN_C const IID IID_ICallIndirect;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("D573B4B1-894E-11d2-B8B6-00C04FB9618A")
-    ICallIndirect : public IUnknown
+    MIDL_INTERFACE("D573B4B1-894E-11d2-B8B6-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
+      ICallIndirect : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE CallIndirect( 
@@ -749,7 +750,7 @@ EXTERN_C const IID IID_ICallIndirect;
         END_INTERFACE
     } ICallIndirectVtbl;
 
-    interface ICallIndirect
+    interface _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_ ICallIndirect
     {
         CONST_VTBL struct ICallIndirectVtbl *lpVtbl;
     };
@@ -803,7 +804,7 @@ EXTERN_C const IID IID_ICallInterceptor;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("60C7CA75-896D-11d2-B8B6-00C04FB9618A")
+    MIDL_INTERFACE("60C7CA75-896D-11d2-B8B6-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
     ICallInterceptor : public ICallIndirect
     {
     public:
@@ -958,7 +959,7 @@ EXTERN_C const IID IID_ICallFrameEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("FD5E0843-FC91-11d0-97D7-00C04FB9618A")
+    MIDL_INTERFACE("FD5E0843-FC91-11d0-97D7-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
     ICallFrameEvents : public IUnknown
     {
     public:
@@ -1045,7 +1046,7 @@ EXTERN_C const IID IID_ICallUnmarshal;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("5333B003-2E42-11d2-B89D-00C04FB9618A")
+    MIDL_INTERFACE("5333B003-2E42-11d2-B89D-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
     ICallUnmarshal : public IUnknown
     {
     public:
@@ -1193,7 +1194,7 @@ EXTERN_C const IID IID_ICallFrameWalker;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("08B23919-392D-11d2-B8A4-00C04FB9618A")
+    MIDL_INTERFACE("08B23919-392D-11d2-B8A4-00C04FB9618A") _Windows_SupportedOS_Windows2000_ 
     ICallFrameWalker : public IUnknown
     {
     public:
@@ -1397,7 +1398,7 @@ extern RPC_IF_HANDLE __MIDL_itf_callobj_0000_0007_v0_0_s_ifspec;
 /* interface ICallFrameAPIs */
 /* [local][uuid] */ 
 
-HRESULT __stdcall CoGetInterceptor( 
+_Windows_SupportedOS_Windows2000_ HRESULT __stdcall CoGetInterceptor( 
     /* [annotation][in] */ 
     _In_  REFIID iidIntercepted,
     /* [annotation][in] */ 
