diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ccgplugins.h b/generation/WinSDK/RecompiledIdlHeaders/um/ccgplugins.h
index eb5e609b4b709d9dacfa574c269f2a5d3c9bbeec..ad9102e999312c2e24d403a0141232c58e8ac1fe 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ccgplugins.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ccgplugins.h
@@ -32,6 +32,7 @@
 
 #ifndef __ccgplugins_h__
 #define __ccgplugins_h__
+#include "win32metadata_annotations.h"
 
 #if defined(_MSC_VER) && (_MSC_VER >= 1020)
 #pragma once
@@ -74,7 +75,7 @@ EXTERN_C const IID IID_ICcgDomainAuthCredentials;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("6ecda518-2010-4437-8bc3-46e752b7b172")
-    ICcgDomainAuthCredentials : public IUnknown
+    _Windows_SupportedOS_17763_ ICcgDomainAuthCredentials : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetPasswordCredentials( 
@@ -127,7 +128,7 @@ EXTERN_C const IID IID_ICcgDomainAuthCredentials;
         END_INTERFACE
     } ICcgDomainAuthCredentialsVtbl;
 
-    interface ICcgDomainAuthCredentials
+    interface _Windows_SupportedOS_17763_ ICcgDomainAuthCredentials
     {
         CONST_VTBL struct ICcgDomainAuthCredentialsVtbl *lpVtbl;
     };
