diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/cchannel.h b/generation/WinSDK/RecompiledIdlHeaders/um/cchannel.h
index a7032a870e..e304975e56 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/cchannel.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/cchannel.h
@@ -23,6 +23,7 @@ Revision History:
 #include <pchannel.h>
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -316,7 +317,7 @@ typedef VIRTUALCHANNELWRITE FAR * PVIRTUALCHANNELWRITE;
 /*                                                                          */
 /* Description: Virtual Channel entry points passed to VirtualChannelEntry  */
 /****************************************************************************/
-typedef struct tagCHANNEL_ENTRY_POINTS
+typedef struct   tagCHANNEL_ENTRY_POINTS
 {
     DWORD cbSize;
     DWORD protocolVersion;
