diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/cfapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/cfapi.h
index 8a77485d62..75605c938b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/cfapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/cfapi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation.  All rights reserved.
@@ -132,7 +133,7 @@ typedef struct CF_PLATFORM_INFO {
     DWORD IntegrationNumber;
 } CF_PLATFORM_INFO;
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetPlatformInfo (
     _Out_ CF_PLATFORM_INFO *PlatformVersion
     );
@@ -176,8 +177,8 @@ DEFINE_USHORT_ENUM( CF_HYDRATION_POLICY_MODIFIER );
 DEFINE_ENUM_FLAG_OPERATORS( CF_HYDRATION_POLICY_MODIFIER );
 
 typedef struct CF_HYDRATION_POLICY {
-    CF_HYDRATION_POLICY_PRIMARY_USHORT Primary;
-    CF_HYDRATION_POLICY_MODIFIER_USHORT Modifier;
+    _Win32_AssociatedEnum_(CF_HYDRATION_POLICY_PRIMARY) CF_HYDRATION_POLICY_PRIMARY_USHORT Primary;
+    _Win32_AssociatedEnum_(CF_HYDRATION_POLICY_MODIFIER) CF_HYDRATION_POLICY_MODIFIER_USHORT Modifier;
 } CF_HYDRATION_POLICY;
 
 typedef enum CF_POPULATION_POLICY_PRIMARY {
@@ -196,8 +197,8 @@ DEFINE_USHORT_ENUM( CF_POPULATION_POLICY_MODIFIER );
 DEFINE_ENUM_FLAG_OPERATORS( CF_POPULATION_POLICY_MODIFIER );
 
 typedef struct CF_POPULATION_POLICY {
-    CF_POPULATION_POLICY_PRIMARY_USHORT Primary;
-    CF_POPULATION_POLICY_MODIFIER_USHORT Modifier;
+    _Win32_AssociatedEnum_(CF_POPULATION_POLICY_PRIMARY) CF_POPULATION_POLICY_PRIMARY_USHORT Primary;
+    _Win32_AssociatedEnum_(CF_POPULATION_POLICY_MODIFIER) CF_POPULATION_POLICY_MODIFIER_USHORT Modifier;
 } CF_POPULATION_POLICY;
 
 typedef enum CF_PLACEHOLDER_MANAGEMENT_POLICY {
@@ -254,7 +255,7 @@ typedef struct CF_SYNC_REGISTRATION {
     GUID ProviderId;
 } CF_SYNC_REGISTRATION;
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfRegisterSyncRoot (
     _In_ LPCWSTR SyncRootPath,
     _In_ CONST CF_SYNC_REGISTRATION *Registration,
@@ -262,7 +263,7 @@ CfRegisterSyncRoot (
     _In_ CF_REGISTER_FLAGS RegisterFlags
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfUnregisterSyncRoot (
     _In_ LPCWSTR SyncRootPath
     );
@@ -422,7 +423,7 @@ typedef struct CF_CALLBACK_PARAMETERS {
             LARGE_INTEGER OptionalFileOffset;
             LARGE_INTEGER OptionalLength;
             LARGE_INTEGER LastDehydrationTime;
-            CF_CALLBACK_DEHYDRATION_REASON LastDehydrationReason;
+            _Win32_AssociatedEnum_(CF_CALLBACK_DEHYDRATION_REASON) CF_CALLBACK_DEHYDRATION_REASON LastDehydrationReason;
         } FetchData;
 
         struct {
@@ -518,27 +519,27 @@ typedef enum CF_CONNECT_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_CONNECT_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfConnectSyncRoot (
     _In_ LPCWSTR SyncRootPath,
     _In_ CONST CF_CALLBACK_REGISTRATION *CallbackTable,
     _In_opt_ LPCVOID CallbackContext,
     _In_ CF_CONNECT_FLAGS ConnectFlags,
-    _Out_ CF_CONNECTION_KEY *ConnectionKey
+    _Win32_RAIIFree_(CfDisconnectSyncRoot, -1, 0) _Out_ CF_CONNECTION_KEY *ConnectionKey
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfDisconnectSyncRoot (
     _In_ CF_CONNECTION_KEY ConnectionKey
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetTransferKey(
     _In_ HANDLE FileHandle,
     _Out_ CF_TRANSFER_KEY* TransferKey
     );
 
-STDAPI_(VOID)
+_Windows_SupportedOS_16299_ STDAPI_(VOID)
 CfReleaseTransferKey(
     _In_ HANDLE FileHandle,
     _Out_ CF_TRANSFER_KEY* TransferKey
@@ -700,20 +701,20 @@ typedef struct CF_OPERATION_PARAMETERS {
 
 #pragma warning( pop )
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfExecute(
     _In_ CONST CF_OPERATION_INFO *OpInfo,
     _Inout_  CF_OPERATION_PARAMETERS *OpParams
     );
 
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfUpdateSyncProviderStatus(
     _In_ CF_CONNECTION_KEY ConnectionKey,
     _In_ CF_SYNC_PROVIDER_STATUS ProviderStatus
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfQuerySyncProviderStatus(
     _In_ CF_CONNECTION_KEY ConnectionKey,
     _Out_ CF_SYNC_PROVIDER_STATUS *ProviderStatus
@@ -722,7 +723,7 @@ CfQuerySyncProviderStatus(
 #pragma endregion
 
 
-STDAPI
+_Windows_SupportedOS_17134_ STDAPI
 CfReportSyncStatus(
     _In_ LPCWSTR SyncRootPath,
     _In_opt_ CF_SYNC_STATUS *SyncStatus
@@ -742,7 +743,7 @@ typedef enum CF_CREATE_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_CREATE_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfCreatePlaceholders (
     _In_ LPCWSTR BaseDirectoryPath,
     _Inout_updates_(PlaceholderCount) CF_PLACEHOLDER_CREATE_INFO *PlaceholderArray,
@@ -761,29 +762,29 @@ typedef enum CF_OPEN_FILE_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_OPEN_FILE_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfOpenFileWithOplock (
     _In_ LPCWSTR FilePath,
     _In_ CF_OPEN_FILE_FLAGS Flags,
-    _Out_ PHANDLE ProtectedHandle
+    _Win32_RAIIFree_(CfCloseHandle) _Out_ PHANDLE ProtectedHandle
     );
 
-STDAPI_( BOOLEAN )
+_Windows_SupportedOS_16299_ STDAPI_( BOOLEAN )
 CfReferenceProtectedHandle (
     _In_ HANDLE ProtectedHandle
     );
 
-STDAPI_( HANDLE )
+_Windows_SupportedOS_16299_ _Win32_RAIIFree_(CloseHandle, -1, 0) STDAPI_( HANDLE )
 CfGetWin32HandleFromProtectedHandle (
     _In_ HANDLE ProtectedHandle
     );
 
-STDAPI_( VOID )
+_Windows_SupportedOS_16299_ STDAPI_( VOID )
 CfReleaseProtectedHandle(
     _In_ HANDLE ProtectedHandle
     );
 
-STDAPI_(VOID)
+_Windows_SupportedOS_16299_ STDAPI_(VOID)
 CfCloseHandle(
     _In_ HANDLE FileHandle
     );
@@ -804,7 +805,7 @@ typedef enum CF_CONVERT_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_CONVERT_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfConvertToPlaceholder (
     _In_ HANDLE FileHandle,
     _In_reads_bytes_opt_(FileIdentityLength) LPCVOID FileIdentity,
@@ -831,7 +832,7 @@ typedef enum CF_UPDATE_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_UPDATE_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfUpdatePlaceholder (
     _In_ HANDLE FileHandle,
     _In_opt_ CONST CF_FS_METADATA *FsMetadata,
@@ -850,7 +851,7 @@ typedef enum CF_REVERT_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_REVERT_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfRevertPlaceholder (
     _In_ HANDLE FileHandle,
     _In_ CF_REVERT_FLAGS RevertFlags,
@@ -863,7 +864,7 @@ typedef enum CF_HYDRATE_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_HYDRATE_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfHydratePlaceholder (
     _In_ HANDLE FileHandle,
     _In_ LARGE_INTEGER StartingOffset,
@@ -905,7 +906,7 @@ typedef enum CF_SET_PIN_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_SET_PIN_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfSetPinState (
     _In_ HANDLE FileHandle,
     _In_ CF_PIN_STATE PinState,
@@ -924,7 +925,7 @@ typedef enum CF_SET_IN_SYNC_FLAGS {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_SET_IN_SYNC_FLAGS );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfSetInSyncState (
     _In_ HANDLE FileHandle,
     _In_ CF_IN_SYNC_STATE InSyncState,
@@ -932,13 +933,13 @@ CfSetInSyncState (
     _Inout_opt_ USN *InSyncUsn
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfSetCorrelationVector (
     _In_ HANDLE FileHandle,
     _In_ const PCORRELATION_VECTOR CorrelationVector
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetCorrelationVector (
     _In_ HANDLE FileHandle,
     _Out_ PCORRELATION_VECTOR CorrelationVector
@@ -957,19 +958,19 @@ typedef enum CF_PLACEHOLDER_STATE {
 
 DEFINE_ENUM_FLAG_OPERATORS( CF_PLACEHOLDER_STATE );
 
-STDAPI_( CF_PLACEHOLDER_STATE )
+_Windows_SupportedOS_16299_ STDAPI_( CF_PLACEHOLDER_STATE )
 CfGetPlaceholderStateFromAttributeTag (
     _In_ DWORD FileAttributes,
     _In_ DWORD ReparseTag
     );
 
-STDAPI_( CF_PLACEHOLDER_STATE )
+_Windows_SupportedOS_16299_ STDAPI_( CF_PLACEHOLDER_STATE )
 CfGetPlaceholderStateFromFileInfo (
     _In_ LPCVOID InfoBuffer,
     _In_ FILE_INFO_BY_HANDLE_CLASS InfoClass
     );
 
-STDAPI_( CF_PLACEHOLDER_STATE )
+_Windows_SupportedOS_16299_ STDAPI_( CF_PLACEHOLDER_STATE )
 CfGetPlaceholderStateFromFindData (
     _In_ CONST WIN32_FIND_DATA *FindData
     );
@@ -1015,11 +1016,11 @@ typedef struct CF_PLACEHOLDER_STANDARD_INFO {
 
 } CF_PLACEHOLDER_STANDARD_INFO;
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetPlaceholderInfo (
     _In_ HANDLE FileHandle,
     _In_ CF_PLACEHOLDER_INFO_CLASS InfoClass,
-    _Out_bytecapcount_(InfoBufferLength) PVOID InfoBuffer,
+     _Out_bytecapcount_(InfoBufferLength) PVOID InfoBuffer,
     _In_ DWORD InfoBufferLength,
     _Out_opt_ PDWORD ReturnedLength
     );
@@ -1080,7 +1081,7 @@ typedef struct CF_SYNC_ROOT_STANDARD_INFO {
 } CF_SYNC_ROOT_STANDARD_INFO;
 
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetSyncRootInfoByPath(
     _In_ LPCWSTR FilePath,
     _In_ CF_SYNC_ROOT_INFO_CLASS InfoClass,
@@ -1089,7 +1090,7 @@ CfGetSyncRootInfoByPath(
     _Out_opt_ DWORD *ReturnedLength
     );
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetSyncRootInfoByHandle(
     _In_ HANDLE FileHandle,
     _In_ CF_SYNC_ROOT_INFO_CLASS InfoClass,
@@ -1106,13 +1107,13 @@ typedef enum CF_PLACEHOLDER_RANGE_INFO_CLASS {
 
 } CF_PLACEHOLDER_RANGE_INFO_CLASS;
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfGetPlaceholderRangeInfo (
     _In_ HANDLE FileHandle,
     _In_ CF_PLACEHOLDER_RANGE_INFO_CLASS InfoClass,
     _In_ LARGE_INTEGER StartingOffset,
     _In_ LARGE_INTEGER Length,
-    _Out_bytecapcount_(InfoBufferLength) PVOID InfoBuffer,
+     _Out_bytecapcount_(InfoBufferLength) PVOID InfoBuffer,
     _In_ DWORD InfoBufferLength,
     _Out_opt_ PDWORD ReturnedLength
     );
@@ -1140,7 +1141,7 @@ CfGetPlaceholderRangeInfoForHydration(
 
 #pragma region Progress APIs
 
-STDAPI
+_Windows_SupportedOS_16299_ STDAPI
 CfReportProviderProgress (
     _In_ CF_CONNECTION_KEY ConnectionKey,
     _In_ CF_TRANSFER_KEY TransferKey,
