diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/coml2api.h b/generation/WinSDK/RecompiledIdlHeaders/um/coml2api.h
index a818bd60b2..eec47f33c9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/coml2api.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/coml2api.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //+---------------------------------------------------------------------------
 //
 //  Microsoft Windows
@@ -81,61 +82,61 @@ typedef DWORD STGFMT;
 #define STGFMT_DOCUMENT         0
 
 // Structured storage APIs
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgCreateDocfile(
+  StgCreateDocfile(
     _In_opt_ _Null_terminated_ const WCHAR* pwcsName,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _Reserved_ DWORD reserved,
     _Outptr_ IStorage** ppstgOpen
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgCreateDocfileOnILockBytes(
+  StgCreateDocfileOnILockBytes(
     _In_ ILockBytes* plkbyt,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _In_ DWORD reserved,
     _Outptr_ IStorage** ppstgOpen
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgOpenStorage(
+  StgOpenStorage(
     _In_opt_ _Null_terminated_ const WCHAR* pwcsName,
     _In_opt_ IStorage* pstgPriority,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _In_opt_z_ SNB snbExclude,
     _In_ DWORD reserved,
     _Outptr_ IStorage** ppstgOpen
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgOpenStorageOnILockBytes(
+  StgOpenStorageOnILockBytes(
     _In_ ILockBytes* plkbyt,
     _In_opt_ IStorage* pstgPriority,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _In_opt_z_ SNB snbExclude,
     _Reserved_ DWORD reserved,
     _Outptr_ IStorage** ppstgOpen
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgIsStorageFile(
+  StgIsStorageFile(
     _In_ _Null_terminated_ const WCHAR* pwcsName
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgIsStorageILockBytes(
+  StgIsStorageILockBytes(
     _In_ ILockBytes* plkbyt
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgSetTimes(
+  StgSetTimes(
     _In_ _Null_terminated_ const WCHAR* lpszName,
     _In_opt_ const FILETIME* pctime,
     _In_opt_ const FILETIME* patime,
@@ -151,7 +152,7 @@ StgSetTimes(
 #define STGOPTIONS_VERSION 0
 #endif
 
-typedef struct tagSTGOPTIONS
+typedef struct   tagSTGOPTIONS
 {
     USHORT usVersion;            // Versions 1 and 2 supported
     USHORT reserved;             // must be 0 for padding
@@ -161,11 +162,11 @@ typedef struct tagSTGOPTIONS
 #endif
 } STGOPTIONS;
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgCreateStorageEx(
+  StgCreateStorageEx(
     _In_opt_ _Null_terminated_ const WCHAR* pwcsName,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _In_ DWORD stgfmt,
     _In_ DWORD grfAttrs,
     _Inout_opt_ STGOPTIONS* pStgOptions,
@@ -174,11 +175,11 @@ StgCreateStorageEx(
     _Outptr_ void** ppObjectOpen
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgOpenStorageEx(
+  StgOpenStorageEx(
     _In_ _Null_terminated_ const WCHAR* pwcsName,
-    _In_ DWORD grfMode,
+    _Win32_AssociatedEnum_(STGM) _In_ DWORD grfMode,
     _In_ DWORD stgfmt,
     _In_ DWORD grfAttrs,
     _Inout_opt_ STGOPTIONS* pStgOptions,
@@ -189,9 +190,9 @@ StgOpenStorageEx(
 
 #ifndef _STGCREATEPROPSTG_DEFINED_
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgCreatePropStg(
+  StgCreatePropStg(
     _In_ IUnknown* pUnk,
     _In_ REFFMTID fmtid,
     _In_ const CLSID* pclsid,
@@ -200,9 +201,9 @@ StgCreatePropStg(
     _Outptr_ IPropertyStorage** ppPropStg
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgOpenPropStg(
+  StgOpenPropStg(
     _In_ IUnknown* pUnk,
     _In_ REFFMTID fmtid,
     _In_ DWORD grfFlags,
@@ -210,9 +211,9 @@ StgOpenPropStg(
     _Outptr_ IPropertyStorage** ppPropStg
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-StgCreatePropSetStg(
+  StgCreatePropSetStg(
     _In_ IStorage* pStorage,
     _Reserved_ DWORD dwReserved,
     _Outptr_ IPropertySetStorage** ppPropSetStg
@@ -220,16 +221,16 @@ StgCreatePropSetStg(
 
 #define CCH_MAX_PROPSTG_NAME    31
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-FmtIdToPropStgName(
+  FmtIdToPropStgName(
     _In_ const FMTID* pfmtid,
     _Out_writes_(CCH_MAX_PROPSTG_NAME+1) LPOLESTR oszName
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-PropStgNameToFmtId(
+  PropStgNameToFmtId(
     _In_ const LPOLESTR oszName,
     _Out_ FMTID* pfmtid
     );
@@ -237,53 +238,132 @@ PropStgNameToFmtId(
 #endif // _STGCREATEPROPSTG_DEFINED_
 
 // Helper functions
-WINOLEAPI
-ReadClassStg(
+_Windows_SupportedOS_Windows2000_ WINOLEAPI
+  ReadClassStg(
     _In_ LPSTORAGE pStg,
     _Out_ CLSID  FAR * pclsid
     );
 
-WINOLEAPI
-WriteClassStg(
+_Windows_SupportedOS_Windows2000_ WINOLEAPI
+  WriteClassStg(
     _In_ LPSTORAGE pStg,
     _In_ REFCLSID rclsid
     );
 
-WINOLEAPI
-ReadClassStm(
+_Windows_SupportedOS_Windows2000_ WINOLEAPI
+  ReadClassStm(
     _In_ LPSTREAM pStm,
     _Out_ CLSID  FAR * pclsid
     );
 
-WINOLEAPI
-WriteClassStm(
+_Windows_SupportedOS_Windows2000_ WINOLEAPI
+  WriteClassStm(
     _In_ LPSTREAM pStm,
     _In_ REFCLSID rclsid
     );
 
 // Storage utility APIs
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-GetHGlobalFromILockBytes(
+  GetHGlobalFromILockBytes(
     _In_ LPLOCKBYTES plkbyt,
-    _Out_ HGLOBAL  FAR * phglobal
+    _Win32_RAIIFree_(GlobalFree, -1, 0) _Out_ HGLOBAL  FAR * phglobal
     );
 
-_Check_return_
+_Windows_SupportedOS_Windows2000_ _Check_return_
 WINOLEAPI
-CreateILockBytesOnHGlobal(
+  CreateILockBytesOnHGlobal(
     _In_opt_ HGLOBAL hGlobal,
     _In_ BOOL fDeleteOnRelease,
     _Outptr_ LPLOCKBYTES  FAR * pplkbyt
     );
 
 // ConvertTo APIs
-WINOLEAPI
-GetConvertStg(
+_Windows_SupportedOS_Windows2000_ WINOLEAPI
+  GetConvertStg(
     _In_ LPSTORAGE pStg
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("STGM_DIRECT")
+#undef STGM_DIRECT
+#pragma push_macro("STGM_TRANSACTED")
+#undef STGM_TRANSACTED
+#pragma push_macro("STGM_SIMPLE")
+#undef STGM_SIMPLE
+#pragma push_macro("STGM_READ")
+#undef STGM_READ
+#pragma push_macro("STGM_WRITE")
+#undef STGM_WRITE
+#pragma push_macro("STGM_READWRITE")
+#undef STGM_READWRITE
+#pragma push_macro("STGM_SHARE_DENY_NONE")
+#undef STGM_SHARE_DENY_NONE
+#pragma push_macro("STGM_SHARE_DENY_READ")
+#undef STGM_SHARE_DENY_READ
+#pragma push_macro("STGM_SHARE_DENY_WRITE")
+#undef STGM_SHARE_DENY_WRITE
+#pragma push_macro("STGM_SHARE_EXCLUSIVE")
+#undef STGM_SHARE_EXCLUSIVE
+#pragma push_macro("STGM_PRIORITY")
+#undef STGM_PRIORITY
+#pragma push_macro("STGM_DELETEONRELEASE")
+#undef STGM_DELETEONRELEASE
+#pragma push_macro("STGM_NOSCRATCH")
+#undef STGM_NOSCRATCH
+#pragma push_macro("STGM_CREATE")
+#undef STGM_CREATE
+#pragma push_macro("STGM_CONVERT")
+#undef STGM_CONVERT
+#pragma push_macro("STGM_FAILIFTHERE")
+#undef STGM_FAILIFTHERE
+#pragma push_macro("STGM_NOSNAPSHOT")
+#undef STGM_NOSNAPSHOT
+#pragma push_macro("STGM_DIRECT_SWMR")
+#undef STGM_DIRECT_SWMR
+enum [[clang::flag_enum]] STGM : unsigned int
+{
+    STGM_DIRECT = 0U,
+    STGM_TRANSACTED = 65536U,
+    STGM_SIMPLE = 134217728U,
+    STGM_READ = 0U,
+    STGM_WRITE = 1U,
+    STGM_READWRITE = 2U,
+    STGM_SHARE_DENY_NONE = 64U,
+    STGM_SHARE_DENY_READ = 48U,
+    STGM_SHARE_DENY_WRITE = 32U,
+    STGM_SHARE_EXCLUSIVE = 16U,
+    STGM_PRIORITY = 262144U,
+    STGM_DELETEONRELEASE = 67108864U,
+    STGM_NOSCRATCH = 1048576U,
+    STGM_CREATE = 4096U,
+    STGM_CONVERT = 131072U,
+    STGM_FAILIFTHERE = 0U,
+    STGM_NOSNAPSHOT = 2097152U,
+    STGM_DIRECT_SWMR = 4194304U,
+};
+#pragma pop_macro("STGM_DIRECT_SWMR")
+#pragma pop_macro("STGM_NOSNAPSHOT")
+#pragma pop_macro("STGM_FAILIFTHERE")
+#pragma pop_macro("STGM_CONVERT")
+#pragma pop_macro("STGM_CREATE")
+#pragma pop_macro("STGM_NOSCRATCH")
+#pragma pop_macro("STGM_DELETEONRELEASE")
+#pragma pop_macro("STGM_PRIORITY")
+#pragma pop_macro("STGM_SHARE_EXCLUSIVE")
+#pragma pop_macro("STGM_SHARE_DENY_WRITE")
+#pragma pop_macro("STGM_SHARE_DENY_READ")
+#pragma pop_macro("STGM_SHARE_DENY_NONE")
+#pragma pop_macro("STGM_READWRITE")
+#pragma pop_macro("STGM_WRITE")
+#pragma pop_macro("STGM_READ")
+#pragma pop_macro("STGM_SIMPLE")
+#pragma pop_macro("STGM_TRANSACTED")
+#pragma pop_macro("STGM_DIRECT")
+#endif
+
 #endif     // __COML2API_H__
