diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/computecore.h b/generation/WinSDK/RecompiledIdlHeaders/um/computecore.h
index cf2dca40b5..975944bc3c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/computecore.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/computecore.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 // ComputeCore.h: ApiSet Contract for ext-ms-win-hyperv-compute-l1
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
@@ -38,7 +39,7 @@ HcsEnumerateComputeSystemsInNamespace(
     );
 
 // Creates a new operation.
-HCS_OPERATION
+_Win32_RAIIFree_(HcsCloseOperation, -1, 0) HCS_OPERATION
 WINAPI
 HcsCreateOperation(
     _In_opt_ const void* context,
@@ -46,7 +47,7 @@ HcsCreateOperation(
     );
 
 // Creates a new operation that allows registration for event notifications
-HCS_OPERATION
+_Win32_RAIIFree_(HcsCloseOperation, -1, 0) HCS_OPERATION
 WINAPI
 HcsCreateOperationWithNotifications(
     _In_ HCS_OPERATION_OPTIONS eventTypes,
@@ -77,14 +78,14 @@ HcsSetOperationContext(
     );
 
 // Returns the handle to compute system associated with an operation.
-HCS_SYSTEM
+_Win32_RAIIFree_(HcsCloseComputeSystem, -1, 0) HCS_SYSTEM
 WINAPI
 HcsGetComputeSystemFromOperation(
     _In_ HCS_OPERATION operation
     );
 
 // Returns the handle to the process associated with an operation
-HCS_PROCESS
+_Win32_RAIIFree_(HcsCloseProcess, -1, 0) HCS_PROCESS
 WINAPI
 HcsGetProcessFromOperation(
     _In_ HCS_OPERATION operation
@@ -193,7 +194,7 @@ HcsCreateComputeSystem(
     _In_ PCWSTR configuration,
     _In_ HCS_OPERATION operation,
     _In_opt_ const SECURITY_DESCRIPTOR* securityDescriptor,
-    _Out_ HCS_SYSTEM* computeSystem
+    _Win32_RAIIFree_(HcsCloseComputeSystem, -1, 0) _Out_ HCS_SYSTEM* computeSystem
     );
 
 /// Creates a new compute system in a given namespace.
@@ -205,7 +206,7 @@ HcsCreateComputeSystemInNamespace(
     _In_ PCWSTR configuration,
     _In_ HCS_OPERATION operation,
     _In_opt_ const HCS_CREATE_OPTIONS* options,
-    _Out_ HCS_SYSTEM* computeSystem
+    _Win32_RAIIFree_(HcsCloseComputeSystem, -1, 0) _Out_ HCS_SYSTEM* computeSystem
     );
 
 /// Opens a handle to an existing compute system.
@@ -214,7 +215,7 @@ WINAPI
 HcsOpenComputeSystem(
     _In_ PCWSTR id,
     _In_ DWORD requestedAccess,
-    _Out_ HCS_SYSTEM* computeSystem
+    _Win32_RAIIFree_(HcsCloseComputeSystem, -1, 0) _Out_ HCS_SYSTEM* computeSystem
     );
 
 /// Opens a handle to an existing compute system in a given namespace.
@@ -224,7 +225,7 @@ HcsOpenComputeSystemInNamespace(
     _In_ PCWSTR idNamespace,
     _In_ PCWSTR id,
     _In_ DWORD requestedAccess,
-    _Out_ HCS_SYSTEM* computeSystem
+    _Win32_RAIIFree_(HcsCloseComputeSystem, -1, 0) _Out_ HCS_SYSTEM* computeSystem
     );
 
 /// Closes a handle to a compute system.
@@ -379,7 +380,7 @@ HcsCreateProcess(
     _In_ PCWSTR processParameters,
     _In_ HCS_OPERATION operation,
     _In_opt_ const SECURITY_DESCRIPTOR* securityDescriptor,
-    _Out_ HCS_PROCESS* process
+    _Win32_RAIIFree_(HcsCloseProcess, -1, 0) _Out_ HCS_PROCESS* process
     );
 
 /// Opens an existing process in a compute system
@@ -389,7 +390,7 @@ HcsOpenProcess(
     _In_ HCS_SYSTEM computeSystem,
     _In_ DWORD processId,
     _In_ DWORD requestedAccess,
-    _Out_ HCS_PROCESS* process
+    _Win32_RAIIFree_(HcsCloseProcess, -1, 0) _Out_ HCS_PROCESS* process
     );
 
 /// Closes the handle to a process in a compute system
