diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi2.h b/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi2.h
index c46976b..52956d0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi2.h
@@ -16,6 +16,7 @@
 #include <apiset.h>
 #include <apisetcconv.h>
 #include <minwindef.h>
+#include <win32metadata_annotations.h>
 #include <minwinbase.h>
 
 #include <wincontypes.h>
@@ -50,6 +51,77 @@ extern "C" {
 
 #define COMMON_LVB_SBCSDBCS        0x0300 // SBCS or DBCS flag.
 
+#ifdef WIN32METADATA
+#pragma push_macro("FOREGROUND_BLUE")
+#pragma push_macro("FOREGROUND_GREEN")
+#pragma push_macro("FOREGROUND_RED")
+#pragma push_macro("FOREGROUND_INTENSITY")
+#pragma push_macro("BACKGROUND_BLUE")
+#pragma push_macro("BACKGROUND_GREEN")
+#pragma push_macro("BACKGROUND_RED")
+#pragma push_macro("BACKGROUND_INTENSITY")
+#pragma push_macro("COMMON_LVB_LEADING_BYTE")
+#pragma push_macro("COMMON_LVB_TRAILING_BYTE")
+#pragma push_macro("COMMON_LVB_GRID_HORIZONTAL")
+#pragma push_macro("COMMON_LVB_GRID_LVERTICAL")
+#pragma push_macro("COMMON_LVB_GRID_RVERTICAL")
+#pragma push_macro("COMMON_LVB_REVERSE_VIDEO")
+#pragma push_macro("COMMON_LVB_UNDERSCORE")
+#pragma push_macro("COMMON_LVB_SBCSDBCS")
+#undef FOREGROUND_BLUE
+#undef FOREGROUND_GREEN
+#undef FOREGROUND_RED
+#undef FOREGROUND_INTENSITY
+#undef BACKGROUND_BLUE
+#undef BACKGROUND_GREEN
+#undef BACKGROUND_RED
+#undef BACKGROUND_INTENSITY
+#undef COMMON_LVB_LEADING_BYTE
+#undef COMMON_LVB_TRAILING_BYTE
+#undef COMMON_LVB_GRID_HORIZONTAL
+#undef COMMON_LVB_GRID_LVERTICAL
+#undef COMMON_LVB_GRID_RVERTICAL
+#undef COMMON_LVB_REVERSE_VIDEO
+#undef COMMON_LVB_UNDERSCORE
+#undef COMMON_LVB_SBCSDBCS
+enum [[clang::flag_enum]] CONSOLE_CHARACTER_ATTRIBUTES : WORD {
+    FOREGROUND_BLUE = 0x0001,
+    FOREGROUND_GREEN = 0x0002,
+    FOREGROUND_RED = 0x0004,
+    FOREGROUND_INTENSITY = 0x0008,
+    BACKGROUND_BLUE = 0x0010,
+    BACKGROUND_GREEN = 0x0020,
+    BACKGROUND_RED = 0x0040,
+    BACKGROUND_INTENSITY = 0x0080,
+    COMMON_LVB_LEADING_BYTE = 0x0100,
+    COMMON_LVB_TRAILING_BYTE = 0x0200,
+    COMMON_LVB_GRID_HORIZONTAL = 0x0400,
+    COMMON_LVB_GRID_LVERTICAL = 0x0800,
+    COMMON_LVB_GRID_RVERTICAL = 0x1000,
+    COMMON_LVB_REVERSE_VIDEO = 0x4000,
+    COMMON_LVB_UNDERSCORE = 0x8000,
+    COMMON_LVB_SBCSDBCS = 0x0300,
+};
+#pragma pop_macro("COMMON_LVB_SBCSDBCS")
+#pragma pop_macro("COMMON_LVB_UNDERSCORE")
+#pragma pop_macro("COMMON_LVB_REVERSE_VIDEO")
+#pragma pop_macro("COMMON_LVB_GRID_RVERTICAL")
+#pragma pop_macro("COMMON_LVB_GRID_LVERTICAL")
+#pragma pop_macro("COMMON_LVB_GRID_HORIZONTAL")
+#pragma pop_macro("COMMON_LVB_TRAILING_BYTE")
+#pragma pop_macro("COMMON_LVB_LEADING_BYTE")
+#pragma pop_macro("BACKGROUND_INTENSITY")
+#pragma pop_macro("BACKGROUND_RED")
+#pragma pop_macro("BACKGROUND_GREEN")
+#pragma pop_macro("BACKGROUND_BLUE")
+#pragma pop_macro("FOREGROUND_INTENSITY")
+#pragma pop_macro("FOREGROUND_RED")
+#pragma pop_macro("FOREGROUND_GREEN")
+#pragma pop_macro("FOREGROUND_BLUE")
+#endif
+
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -61,6 +133,8 @@ FillConsoleOutputCharacterA(
     _Out_ LPDWORD lpNumberOfCharsWritten
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -78,17 +152,21 @@ FillConsoleOutputCharacterW(
 #define FillConsoleOutputCharacter  FillConsoleOutputCharacterA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 FillConsoleOutputAttribute(
     _In_ HANDLE hConsoleOutput,
-    _In_ WORD wAttribute,
+    _In_ _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) WORD wAttribute,
     _In_ DWORD nLength,
     _In_ COORD dwWriteCoord,
     _Out_ LPDWORD lpNumberOfAttrsWritten
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -97,6 +175,10 @@ GenerateConsoleCtrlEvent(
     _In_ DWORD dwProcessGroupId
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
+_Win32_RAIIFree_(CloseHandle)
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")
 WINBASEAPI
 HANDLE
 WINAPI
@@ -108,6 +190,8 @@ CreateConsoleScreenBuffer(
     _Reserved_ LPVOID lpScreenBufferData
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -115,6 +199,8 @@ SetConsoleActiveScreenBuffer(
     _In_ HANDLE hConsoleOutput
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -122,6 +208,8 @@ FlushConsoleInputBuffer(
     _In_ HANDLE hConsoleInput
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -129,6 +217,8 @@ SetConsoleCP(
     _In_ UINT wCodePageID
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -141,6 +231,8 @@ typedef struct _CONSOLE_CURSOR_INFO {
     BOOL   bVisible;
 } CONSOLE_CURSOR_INFO, *PCONSOLE_CURSOR_INFO;
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -149,6 +241,8 @@ GetConsoleCursorInfo(
     _Out_ PCONSOLE_CURSOR_INFO lpConsoleCursorInfo
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -160,11 +254,13 @@ SetConsoleCursorInfo(
 typedef struct _CONSOLE_SCREEN_BUFFER_INFO {
     COORD dwSize;
     COORD dwCursorPosition;
-    WORD  wAttributes;
+    _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) WORD  wAttributes;
     SMALL_RECT srWindow;
     COORD dwMaximumWindowSize;
 } CONSOLE_SCREEN_BUFFER_INFO, *PCONSOLE_SCREEN_BUFFER_INFO;
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -177,14 +273,16 @@ typedef struct _CONSOLE_SCREEN_BUFFER_INFOEX {
     ULONG cbSize;
     COORD dwSize;
     COORD dwCursorPosition;
-    WORD wAttributes;
+    _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) WORD wAttributes;
     SMALL_RECT srWindow;
     COORD dwMaximumWindowSize;
-    WORD wPopupAttributes;
+    _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) WORD wPopupAttributes;
     BOOL bFullscreenSupported;
     COLORREF ColorTable[16];
 } CONSOLE_SCREEN_BUFFER_INFOEX, *PCONSOLE_SCREEN_BUFFER_INFOEX;
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -193,6 +291,8 @@ GetConsoleScreenBufferInfoEx(
     _Inout_ PCONSOLE_SCREEN_BUFFER_INFOEX lpConsoleScreenBufferInfoEx
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -201,6 +301,8 @@ SetConsoleScreenBufferInfoEx(
     _In_ PCONSOLE_SCREEN_BUFFER_INFOEX lpConsoleScreenBufferInfoEx
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -209,6 +311,8 @@ SetConsoleScreenBufferSize(
     _In_ COORD dwSize
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -217,6 +321,8 @@ SetConsoleCursorPosition(
     _In_ COORD dwCursorPosition
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 COORD
 WINAPI
@@ -224,14 +330,18 @@ GetLargestConsoleWindowSize(
     _In_ HANDLE hConsoleOutput
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetConsoleTextAttribute(
     _In_ HANDLE hConsoleOutput,
-    _In_ WORD wAttributes
+    _In_ _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) WORD wAttributes
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -241,6 +351,8 @@ SetConsoleWindowInfo(
     _In_ CONST SMALL_RECT* lpConsoleWindow
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -252,6 +364,8 @@ WriteConsoleOutputCharacterA(
     _Out_ LPDWORD lpNumberOfCharsWritten
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -269,17 +383,21 @@ WriteConsoleOutputCharacterW(
 #define WriteConsoleOutputCharacter  WriteConsoleOutputCharacterA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 WriteConsoleOutputAttribute(
     _In_ HANDLE hConsoleOutput,
-    _In_reads_(nLength) CONST WORD* lpAttribute,
+    _In_reads_(nLength) _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) CONST WORD* lpAttribute,
     _In_ DWORD nLength,
     _In_ COORD dwWriteCoord,
     _Out_ LPDWORD lpNumberOfAttrsWritten
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -291,6 +409,8 @@ ReadConsoleOutputCharacterA(
     _Out_ LPDWORD lpNumberOfCharsRead
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -308,17 +428,21 @@ ReadConsoleOutputCharacterW(
 #define ReadConsoleOutputCharacter  ReadConsoleOutputCharacterA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 ReadConsoleOutputAttribute(
     _In_ HANDLE hConsoleOutput,
-    _Out_writes_(nLength) LPWORD lpAttribute,
+    _Out_writes_(nLength) _Win32_AssociatedEnum_(CONSOLE_CHARACTER_ATTRIBUTES) LPWORD lpAttribute,
     _In_ DWORD nLength,
     _In_ COORD dwReadCoord,
     _Out_ LPDWORD lpNumberOfAttrsRead
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -329,6 +453,8 @@ WriteConsoleInputA(
     _Out_ LPDWORD lpNumberOfEventsWritten
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -345,6 +471,8 @@ WriteConsoleInputW(
 #define WriteConsoleInput  WriteConsoleInputA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -356,6 +484,8 @@ ScrollConsoleScreenBufferA(
     _In_ CONST CHAR_INFO* lpFill
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -373,6 +503,8 @@ ScrollConsoleScreenBufferW(
 #define ScrollConsoleScreenBuffer  ScrollConsoleScreenBufferA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -384,6 +516,8 @@ WriteConsoleOutputA(
     _Inout_ PSMALL_RECT lpWriteRegion
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -401,6 +535,8 @@ WriteConsoleOutputW(
 #define WriteConsoleOutput  WriteConsoleOutputA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -412,6 +548,8 @@ ReadConsoleOutputA(
     _Inout_ PSMALL_RECT lpReadRegion
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -429,6 +567,8 @@ ReadConsoleOutputW(
 #define ReadConsoleOutput  ReadConsoleOutputA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -437,6 +577,8 @@ GetConsoleTitleA(
     _In_ DWORD nSize
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -453,6 +595,8 @@ GetConsoleTitleW(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -461,6 +605,8 @@ GetConsoleOriginalTitleA(
     _In_ DWORD nSize
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -477,6 +623,8 @@ GetConsoleOriginalTitleW(
 
 #endif /* _WIN32_WINNT >= 0x0600 */
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -484,6 +632,8 @@ SetConsoleTitleA(
     _In_ LPCSTR lpConsoleTitle
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
