diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi.h
index 3754ec4..480cc17 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/consoleapi.h
@@ -17,6 +17,7 @@
 #include <apiset.h>
 #include <apisetcconv.h>
 #include <minwindef.h>
+#include <win32metadata_annotations.h>
 #include <minwinbase.h>
 
 #include <wincontypes.h>
@@ -28,6 +29,8 @@ extern "C" {
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -55,6 +58,7 @@ typedef enum ALLOC_CONSOLE_RESULT {
     ALLOC_CONSOLE_RESULT_EXISTING_CONSOLE = 2
 } ALLOC_CONSOLE_RESULT, *PALLOC_CONSOLE_RESULT;
 
+_Windows_SupportedOS_26100_
 WINBASEAPI
 HRESULT
 WINAPI
@@ -65,6 +69,8 @@ AllocConsoleWithOptions(
 
 #endif
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -74,6 +80,8 @@ FreeConsole(
 
 #if (_WIN32_WINNT >= 0x0500)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -85,6 +93,8 @@ AttachConsole(
 
 #endif /* _WIN32_WINNT >= 0x0500 */
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 UINT
 WINAPI
@@ -92,6 +102,8 @@ GetConsoleCP(
     VOID
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 UINT
 WINAPI
@@ -124,22 +136,93 @@ GetConsoleOutputCP(
 #define DISABLE_NEWLINE_AUTO_RETURN         0x0008
 #define ENABLE_LVB_GRID_WORLDWIDE           0x0010
 
+#ifdef WIN32METADATA
+#pragma push_macro("ENABLE_PROCESSED_INPUT")
+#pragma push_macro("ENABLE_LINE_INPUT")
+#pragma push_macro("ENABLE_ECHO_INPUT")
+#pragma push_macro("ENABLE_WINDOW_INPUT")
+#pragma push_macro("ENABLE_MOUSE_INPUT")
+#pragma push_macro("ENABLE_INSERT_MODE")
+#pragma push_macro("ENABLE_QUICK_EDIT_MODE")
+#pragma push_macro("ENABLE_EXTENDED_FLAGS")
+#pragma push_macro("ENABLE_AUTO_POSITION")
+#pragma push_macro("ENABLE_VIRTUAL_TERMINAL_INPUT")
+#pragma push_macro("ENABLE_PROCESSED_OUTPUT")
+#pragma push_macro("ENABLE_WRAP_AT_EOL_OUTPUT")
+#pragma push_macro("ENABLE_VIRTUAL_TERMINAL_PROCESSING")
+#pragma push_macro("DISABLE_NEWLINE_AUTO_RETURN")
+#pragma push_macro("ENABLE_LVB_GRID_WORLDWIDE")
+#undef ENABLE_PROCESSED_INPUT
+#undef ENABLE_LINE_INPUT
+#undef ENABLE_ECHO_INPUT
+#undef ENABLE_WINDOW_INPUT
+#undef ENABLE_MOUSE_INPUT
+#undef ENABLE_INSERT_MODE
+#undef ENABLE_QUICK_EDIT_MODE
+#undef ENABLE_EXTENDED_FLAGS
+#undef ENABLE_AUTO_POSITION
+#undef ENABLE_VIRTUAL_TERMINAL_INPUT
+#undef ENABLE_PROCESSED_OUTPUT
+#undef ENABLE_WRAP_AT_EOL_OUTPUT
+#undef ENABLE_VIRTUAL_TERMINAL_PROCESSING
+#undef DISABLE_NEWLINE_AUTO_RETURN
+#undef ENABLE_LVB_GRID_WORLDWIDE
+enum [[clang::flag_enum]] CONSOLE_MODE : DWORD {
+    ENABLE_PROCESSED_INPUT = 0x0001,
+    ENABLE_LINE_INPUT = 0x0002,
+    ENABLE_ECHO_INPUT = 0x0004,
+    ENABLE_WINDOW_INPUT = 0x0008,
+    ENABLE_MOUSE_INPUT = 0x0010,
+    ENABLE_INSERT_MODE = 0x0020,
+    ENABLE_QUICK_EDIT_MODE = 0x0040,
+    ENABLE_EXTENDED_FLAGS = 0x0080,
+    ENABLE_AUTO_POSITION = 0x0100,
+    ENABLE_VIRTUAL_TERMINAL_INPUT = 0x0200,
+    ENABLE_PROCESSED_OUTPUT = 0x0001,
+    ENABLE_WRAP_AT_EOL_OUTPUT = 0x0002,
+    ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004,
+    DISABLE_NEWLINE_AUTO_RETURN = 0x0008,
+    ENABLE_LVB_GRID_WORLDWIDE = 0x0010,
+};
+#pragma pop_macro("ENABLE_LVB_GRID_WORLDWIDE")
+#pragma pop_macro("DISABLE_NEWLINE_AUTO_RETURN")
+#pragma pop_macro("ENABLE_VIRTUAL_TERMINAL_PROCESSING")
+#pragma pop_macro("ENABLE_WRAP_AT_EOL_OUTPUT")
+#pragma pop_macro("ENABLE_PROCESSED_OUTPUT")
+#pragma pop_macro("ENABLE_VIRTUAL_TERMINAL_INPUT")
+#pragma pop_macro("ENABLE_AUTO_POSITION")
+#pragma pop_macro("ENABLE_EXTENDED_FLAGS")
+#pragma pop_macro("ENABLE_QUICK_EDIT_MODE")
+#pragma pop_macro("ENABLE_INSERT_MODE")
+#pragma pop_macro("ENABLE_MOUSE_INPUT")
+#pragma pop_macro("ENABLE_WINDOW_INPUT")
+#pragma pop_macro("ENABLE_ECHO_INPUT")
+#pragma pop_macro("ENABLE_LINE_INPUT")
+#pragma pop_macro("ENABLE_PROCESSED_INPUT")
+#endif
+
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetConsoleMode(
     _In_ HANDLE hConsoleHandle,
-    _Out_ LPDWORD lpMode
+    _Out_ _Win32_AssociatedEnum_(CONSOLE_MODE) LPDWORD lpMode
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetConsoleMode(
     _In_ HANDLE hConsoleHandle,
-    _In_ DWORD dwMode
+    _In_ _Win32_AssociatedEnum_(CONSOLE_MODE) DWORD dwMode
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -148,6 +231,8 @@ GetNumberOfConsoleInputEvents(
     _Out_ LPDWORD lpNumberOfEvents
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -159,6 +244,8 @@ ReadConsoleInputA(
     _Out_ _Deref_out_range_(<=, nLength) LPDWORD lpNumberOfEventsRead
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -184,6 +271,8 @@ ReadConsoleInputW(
 
 // begin_consoleapi_h
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -194,6 +283,8 @@ PeekConsoleInputA(
     _Out_ LPDWORD lpNumberOfEventsRead
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -217,6 +308,8 @@ typedef struct _CONSOLE_READCONSOLE_CONTROL {
     ULONG dwControlKeyState;
 } CONSOLE_READCONSOLE_CONTROL, *PCONSOLE_READCONSOLE_CONTROL;
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -229,6 +322,8 @@ ReadConsoleA(
     _In_opt_ PCONSOLE_READCONSOLE_CONTROL pInputControl
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -247,6 +342,8 @@ ReadConsoleW(
 #define ReadConsole  ReadConsoleA
 #endif // !UNICODE
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -258,6 +355,8 @@ WriteConsoleA(
     _Reserved_ LPVOID lpReserved
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -297,6 +396,8 @@ BOOL
     _In_ DWORD CtrlType
     );
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -316,6 +417,7 @@ SetConsoleCtrlHandler(
 // CreatePseudoConsole Flags
 #define PSEUDOCONSOLE_INHERIT_CURSOR (0x1)
 
+_Windows_SupportedOS_17763_
 WINBASEAPI
 HRESULT
 WINAPI
@@ -324,9 +426,10 @@ CreatePseudoConsole(
     _In_ HANDLE hInput,
     _In_ HANDLE hOutput,
     _In_ DWORD dwFlags,
-    _Out_ HPCON* phPC
+    _Out_ _Win32_RAIIFree_(ClosePseudoConsole) HPCON* phPC
     );
 
+_Windows_SupportedOS_17763_
 WINBASEAPI
 HRESULT
 WINAPI
@@ -335,6 +438,7 @@ ResizePseudoConsole(
     _In_ COORD size
     );
 
+_Windows_SupportedOS_17763_
 WINBASEAPI
 VOID
 WINAPI
@@ -346,6 +450,7 @@ ClosePseudoConsole(
 
 #if (NTDDI_VERSION >= NTDDI_WIN11_GE)
 
+_Windows_SupportedOS_26100_
 WINBASEAPI
 HRESULT
 WINAPI
