diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/cryptcat.h b/generation/WinSDK/RecompiledIdlHeaders/um/cryptcat.h
index 86c4754e10..a5493177e5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/cryptcat.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/cryptcat.h
@@ -1,9 +1,11 @@
+#include <win32metadata_annotations.h>
 #include <mscat.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+_Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_
 LPWSTR WINAPI CryptCATCDFEnumMembersByCDFTagEx(
   _In_    CRYPTCATCDF                  *pCDF,
   _Inout_ LPWSTR                       pwszPrevCDFTag,
@@ -13,6 +15,7 @@ LPWSTR WINAPI CryptCATCDFEnumMembersByCDFTagEx(
   _In_    LPVOID                       pvReserved
 );
 
+_Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_
 CRYPTCATATTRIBUTE* WINAPI CryptCATCDFEnumAttributesWithCDFTag(
   _In_ CRYPTCATCDF                  *pCDF,
   _In_ LPWSTR                       pwszMemberTag,
