diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/cscapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/cscapi.h
index e67a96293f..a9a306468d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/cscapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/cscapi.h
@@ -21,6 +21,7 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -34,6 +35,8 @@ Abstract:
 // If *pbRebootRequired returns TRUE, a system restart is necessary to 
 // complete the operation.
 //
+_Windows_SupportedOS_WindowsVista_ 
+
 STDAPI_(DWORD)
 OfflineFilesEnable(
     _In_ BOOL bEnable, 
@@ -45,6 +48,8 @@ OfflineFilesEnable(
 // Start Offline Files feature if it's available. 
 // If the feature is not available then a ERROR_NOT_SUPPORTED error is returned.
 //
+_Windows_SupportedOS_Windows8_ 
+
 STDAPI_(DWORD)
 OfflineFilesStart(
     );
@@ -58,6 +63,8 @@ OfflineFilesStart(
 //  - enabled/disabled describes the configured state as last successfully set by 
 //    OfflineFilesEnable().
 //
+_Windows_SupportedOS_WindowsVista_ 
+
 STDAPI_(DWORD)
 OfflineFilesQueryStatus(
     _Out_opt_ BOOL *pbActive,
@@ -78,7 +85,7 @@ OfflineFilesQueryStatus(
 //    If the status is inactive/enabled/available then the OfflineFilesStart function
 //    can be used to transition the Offline Files feature to active state (no reboot required). 
 //
-STDAPI_(DWORD)
+_Windows_SupportedOS_Windows8_ STDAPI_(DWORD)
 OfflineFilesQueryStatusEx(
     _Out_opt_ BOOL *pbActive,
     _Out_opt_ BOOL *pbEnabled,
