diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
index 4ac84c8883..dd00789e05 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1_3.h
@@ -23,6 +23,7 @@
 #endif // #ifndef _D2D1_SVG_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
@@ -223,7 +224,7 @@ typedef enum D2D1_PATCH_EDGE_MODE
 /// the GradientMeshPatch function or the GradientMeshPatchFromCoonsPatch function
 /// to create one.
 /// </summary>
-typedef struct D2D1_GRADIENT_MESH_PATCH
+typedef struct  D2D1_GRADIENT_MESH_PATCH
 {
     
     /// <summary>
@@ -1901,9 +1902,9 @@ typedef interface ID2D1InkStyle ID2D1InkStyle;
 
 typedef interface ID2D1Ink ID2D1Ink;
 
-typedef interface ID2D1GradientMesh ID2D1GradientMesh;
+_Windows_SupportedOS_Windows81_ typedef interface ID2D1GradientMesh ID2D1GradientMesh;
 
-typedef interface ID2D1ImageSource ID2D1ImageSource;
+_Windows_SupportedOS_10240_ typedef interface ID2D1ImageSource ID2D1ImageSource;
 
 typedef interface ID2D1ImageSourceFromWic ID2D1ImageSourceFromWic;
 
@@ -1911,17 +1912,17 @@ typedef interface ID2D1TransformedImageSource ID2D1TransformedImageSource;
 
 typedef interface ID2D1LookupTable3D ID2D1LookupTable3D;
 
-typedef interface ID2D1DeviceContext2 ID2D1DeviceContext2;
+_Windows_SupportedOS_10240_ typedef interface ID2D1DeviceContext2 ID2D1DeviceContext2;
 
 typedef interface ID2D1Device2 ID2D1Device2;
 
 typedef interface ID2D1Factory3 ID2D1Factory3;
 
-typedef interface ID2D1CommandSink2 ID2D1CommandSink2;
+_Windows_SupportedOS_Windows81_ typedef interface ID2D1CommandSink2 ID2D1CommandSink2;
 
-typedef interface ID2D1GdiMetafile1 ID2D1GdiMetafile1;
+_Windows_SupportedOS_10240_ typedef interface ID2D1GdiMetafile1 ID2D1GdiMetafile1;
 
-typedef interface ID2D1GdiMetafileSink1 ID2D1GdiMetafileSink1;
+_Windows_SupportedOS_10240_ typedef interface ID2D1GdiMetafileSink1 ID2D1GdiMetafileSink1;
 
 #endif
 
@@ -1932,8 +1933,8 @@ extern "C"
 #endif
 
 #if NTDDI_VERSION >= NTDDI_WINTHRESHOLD
-    void WINAPI 
-    D2D1GetGradientMeshInteriorPointsFromCoonsPatch(
+    _Windows_SupportedOS_10240_ void WINAPI 
+      D2D1GetGradientMeshInteriorPointsFromCoonsPatch(
         _In_ CONST D2D1_POINT_2F *pPoint0,
         _In_ CONST D2D1_POINT_2F *pPoint1,
         _In_ CONST D2D1_POINT_2F *pPoint2,
