diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
index 3f69243671..94920bf332 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d2d1effectauthor.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //---------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation.  All rights reserved.
 //
@@ -422,7 +423,7 @@ EXTERN_C CONST IID IID_ID2D1EffectContext;
 /// <summary>
 /// A transform uses this interface to write new vertices to a vertex buffer.
 /// </summary>
-interface DX_DECLARE_INTERFACE("9b8b1336-00a5-4668-92b7-ced5d8bf9b7b") ID2D1VertexBuffer  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("9b8b1336-00a5-4668-92b7-ced5d8bf9b7b") ID2D1VertexBuffer  : public IUnknown
 {
     
     STDMETHOD(Map)(
@@ -435,7 +436,7 @@ interface DX_DECLARE_INTERFACE("9b8b1336-00a5-4668-92b7-ced5d8bf9b7b") ID2D1Vert
 }; // interface ID2D1VertexBuffer
 
 
-interface DX_DECLARE_INTERFACE("688d15c3-02b0-438d-b13a-d1b44c32c39a") ID2D1ResourceTexture  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("688d15c3-02b0-438d-b13a-d1b44c32c39a") ID2D1ResourceTexture  : public IUnknown
 {
     
     /// <summary>
@@ -456,7 +457,7 @@ interface DX_DECLARE_INTERFACE("688d15c3-02b0-438d-b13a-d1b44c32c39a") ID2D1Reso
 /// A transform uses this interface to specify how to render a particular pass in
 /// D2D.
 /// </summary>
-interface DX_DECLARE_INTERFACE("519ae1bd-d19a-420d-b849-364f594776b7") ID2D1RenderInfo  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("519ae1bd-d19a-420d-b849-364f594776b7") ID2D1RenderInfo  : public IUnknown
 {
     
     /// <summary>
@@ -497,7 +498,7 @@ interface DX_DECLARE_INTERFACE("519ae1bd-d19a-420d-b849-364f594776b7") ID2D1Rend
 /// A transform uses this interface to specify how to render a particular pass using
 /// pixel and vertex shaders.
 /// </summary>
-interface DX_DECLARE_INTERFACE("693ce632-7f2f-45de-93fe-18d88b37aa21") ID2D1DrawInfo  : public ID2D1RenderInfo
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("693ce632-7f2f-45de-93fe-18d88b37aa21") ID2D1DrawInfo  : public ID2D1RenderInfo
 {
     
     /// <summary>
@@ -550,7 +551,7 @@ interface DX_DECLARE_INTERFACE("693ce632-7f2f-45de-93fe-18d88b37aa21") ID2D1Draw
 /// A transform uses this interface to specify how to render a particular pass using
 /// compute shader.
 /// </summary>
-interface DX_DECLARE_INTERFACE("5598b14b-9fd7-48b7-9bdb-8f0964eb38bc") ID2D1ComputeInfo  : public ID2D1RenderInfo
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("5598b14b-9fd7-48b7-9bdb-8f0964eb38bc") ID2D1ComputeInfo  : public ID2D1RenderInfo
 {
     
     /// <summary>
@@ -581,7 +582,7 @@ interface DX_DECLARE_INTERFACE("5598b14b-9fd7-48b7-9bdb-8f0964eb38bc") ID2D1Comp
 /// <summary>
 /// A base object which can be inserted into a transform graph.
 /// </summary>
-interface DX_DECLARE_INTERFACE("b2efe1e7-729f-4102-949f-505fa21bf666") ID2D1TransformNode  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows8_  DX_DECLARE_INTERFACE("b2efe1e7-729f-4102-949f-505fa21bf666") ID2D1TransformNode  : public IUnknown
 {
     
     /// <summary>
@@ -595,7 +596,7 @@ interface DX_DECLARE_INTERFACE("b2efe1e7-729f-4102-949f-505fa21bf666") ID2D1Tran
 /// <summary>
 /// The implementation of the actual graph.
 /// </summary>
-interface DX_DECLARE_INTERFACE("13d29038-c3e6-4034-9081-13b53a417992") ID2D1TransformGraph  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("13d29038-c3e6-4034-9081-13b53a417992") ID2D1TransformGraph  : public IUnknown
 {
     
     /// <summary>
@@ -671,7 +672,7 @@ interface DX_DECLARE_INTERFACE("13d29038-c3e6-4034-9081-13b53a417992") ID2D1Tran
 /// <summary>
 /// The interface implemented by a transform author.
 /// </summary>
-interface DX_DECLARE_INTERFACE("ef1a287d-342a-4f76-8fdb-da0d6ea9f92b") ID2D1Transform  : public ID2D1TransformNode
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("ef1a287d-342a-4f76-8fdb-da0d6ea9f92b") ID2D1Transform  : public ID2D1TransformNode
 {
     
     STDMETHOD(MapOutputRectToInputRects)(
@@ -699,7 +700,7 @@ interface DX_DECLARE_INTERFACE("ef1a287d-342a-4f76-8fdb-da0d6ea9f92b") ID2D1Tran
 /// <summary>
 /// The interface implemented by a transform author to provide a GPU-based effect.
 /// </summary>
-interface DX_DECLARE_INTERFACE("36bfdcb6-9739-435d-a30d-a653beff6a6f") ID2D1DrawTransform  : public ID2D1Transform
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("36bfdcb6-9739-435d-a30d-a653beff6a6f") ID2D1DrawTransform  : public ID2D1Transform
 {
     
     STDMETHOD(SetDrawInfo)(
@@ -712,7 +713,7 @@ interface DX_DECLARE_INTERFACE("36bfdcb6-9739-435d-a30d-a653beff6a6f") ID2D1Draw
 /// The interface implemented by a transform author to provide a Compute Shader
 /// based effect.
 /// </summary>
-interface DX_DECLARE_INTERFACE("0d85573c-01e3-4f7d-bfd9-0d60608bf3c3") ID2D1ComputeTransform  : public ID2D1Transform
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("0d85573c-01e3-4f7d-bfd9-0d60608bf3c3") ID2D1ComputeTransform  : public ID2D1Transform
 {
     
     STDMETHOD(SetComputeInfo)(
@@ -732,7 +733,7 @@ interface DX_DECLARE_INTERFACE("0d85573c-01e3-4f7d-bfd9-0d60608bf3c3") ID2D1Comp
 /// The interface implemented by a transform author to indicate that it should
 /// receive an analysis result callback.
 /// </summary>
-interface DX_DECLARE_INTERFACE("0359dc30-95e6-4568-9055-27720d130e93") ID2D1AnalysisTransform  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("0359dc30-95e6-4568-9055-27720d130e93") ID2D1AnalysisTransform  : public IUnknown
 {
     
     STDMETHOD(ProcessAnalysisResults)(
@@ -746,7 +747,7 @@ interface DX_DECLARE_INTERFACE("0359dc30-95e6-4568-9055-27720d130e93") ID2D1Anal
 /// The interface implemented by a transform author to provide a CPU based source
 /// effect.
 /// </summary>
-interface DX_DECLARE_INTERFACE("db1800dd-0c34-4cf9-be90-31cc0a5653e1") ID2D1SourceTransform  : public ID2D1Transform
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("db1800dd-0c34-4cf9-be90-31cc0a5653e1") ID2D1SourceTransform  : public ID2D1Transform
 {
     
     STDMETHOD(SetRenderInfo)(
@@ -765,7 +766,7 @@ interface DX_DECLARE_INTERFACE("db1800dd-0c34-4cf9-be90-31cc0a5653e1") ID2D1Sour
 /// Base interface for built-in transforms on which precision and caching may be
 /// controlled.
 /// </summary>
-interface DX_DECLARE_INTERFACE("1a799d8a-69f7-4e4c-9fed-437ccc6684cc") ID2D1ConcreteTransform  : public ID2D1TransformNode
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("1a799d8a-69f7-4e4c-9fed-437ccc6684cc") ID2D1ConcreteTransform  : public ID2D1TransformNode
 {
     
     /// <summary>
@@ -788,7 +789,7 @@ interface DX_DECLARE_INTERFACE("1a799d8a-69f7-4e4c-9fed-437ccc6684cc") ID2D1Conc
 /// <summary>
 /// An effect uses this interface to configure a blending operation.
 /// </summary>
-interface DX_DECLARE_INTERFACE("63ac0b32-ba44-450f-8806-7f4ca1ff2f1b") ID2D1BlendTransform  : public ID2D1ConcreteTransform
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("63ac0b32-ba44-450f-8806-7f4ca1ff2f1b") ID2D1BlendTransform  : public ID2D1ConcreteTransform
 {
     
     STDMETHOD_(void, SetDescription)(
@@ -804,7 +805,7 @@ interface DX_DECLARE_INTERFACE("63ac0b32-ba44-450f-8806-7f4ca1ff2f1b") ID2D1Blen
 /// <summary>
 /// An effect uses this interface to configure border generation.
 /// </summary>
-interface DX_DECLARE_INTERFACE("4998735c-3a19-473c-9781-656847e3a347") ID2D1BorderTransform  : public ID2D1ConcreteTransform
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("4998735c-3a19-473c-9781-656847e3a347") ID2D1BorderTransform  : public ID2D1ConcreteTransform
 {
     
     STDMETHOD_(void, SetExtendModeX)(
@@ -827,7 +828,7 @@ interface DX_DECLARE_INTERFACE("4998735c-3a19-473c-9781-656847e3a347") ID2D1Bord
 /// An effect uses this interface to offset an image without inserting a rendering
 /// pass.
 /// </summary>
-interface DX_DECLARE_INTERFACE("3fe6adea-7643-4f53-bd14-a0ce63f24042") ID2D1OffsetTransform  : public ID2D1TransformNode
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("3fe6adea-7643-4f53-bd14-a0ce63f24042") ID2D1OffsetTransform  : public ID2D1TransformNode
 {
     
     STDMETHOD_(void, SetOffset)(
@@ -842,7 +843,7 @@ interface DX_DECLARE_INTERFACE("3fe6adea-7643-4f53-bd14-a0ce63f24042") ID2D1Offs
 /// <summary>
 /// An effect uses this interface to alter the image rectangle of its input.
 /// </summary>
-interface DX_DECLARE_INTERFACE("90f732e2-5092-4606-a819-8651970baccd") ID2D1BoundsAdjustmentTransform  : public ID2D1TransformNode
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("90f732e2-5092-4606-a819-8651970baccd") ID2D1BoundsAdjustmentTransform  : public ID2D1TransformNode
 {
     
     STDMETHOD_(void, SetOutputBounds)(
@@ -859,7 +860,7 @@ interface DX_DECLARE_INTERFACE("90f732e2-5092-4606-a819-8651970baccd") ID2D1Boun
 /// This is the interface implemented by an effect author, along with the
 /// constructor and registration information.
 /// </summary>
-interface DX_DECLARE_INTERFACE("a248fd3f-3e6c-4e63-9f03-7f68ecc91db9") ID2D1EffectImpl  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DX_DECLARE_INTERFACE("a248fd3f-3e6c-4e63-9f03-7f68ecc91db9") ID2D1EffectImpl  : public IUnknown
 {
     
     /// <summary>
@@ -894,7 +895,7 @@ interface DX_DECLARE_INTERFACE("a248fd3f-3e6c-4e63-9f03-7f68ecc91db9") ID2D1Effe
 /// and any other operation tied to context which is not useful to the application
 /// facing API.
 /// </summary>
-interface DX_DECLARE_INTERFACE("3d9f916b-27dc-4ad7-b4f1-64945340f563") ID2D1EffectContext  : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows8_  DX_DECLARE_INTERFACE("3d9f916b-27dc-4ad7-b4f1-64945340f563") ID2D1EffectContext  : public IUnknown
 {
     
     STDMETHOD_(void, GetDpi)(
@@ -1040,41 +1041,41 @@ interface DX_DECLARE_INTERFACE("3d9f916b-27dc-4ad7-b4f1-64945340f563") ID2D1Effe
 #ifdef D2D_USE_C_DEFINITIONS
 
 
-typedef interface ID2D1VertexBuffer ID2D1VertexBuffer;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1VertexBuffer ID2D1VertexBuffer;
 
-typedef interface ID2D1ResourceTexture ID2D1ResourceTexture;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1ResourceTexture ID2D1ResourceTexture;
 
-typedef interface ID2D1RenderInfo ID2D1RenderInfo;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1RenderInfo ID2D1RenderInfo;
 
-typedef interface ID2D1DrawInfo ID2D1DrawInfo;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1DrawInfo ID2D1DrawInfo;
 
-typedef interface ID2D1ComputeInfo ID2D1ComputeInfo;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1ComputeInfo ID2D1ComputeInfo;
 
 typedef interface ID2D1TransformNode ID2D1TransformNode;
 
-typedef interface ID2D1TransformGraph ID2D1TransformGraph;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1TransformGraph ID2D1TransformGraph;
 
-typedef interface ID2D1Transform ID2D1Transform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1Transform ID2D1Transform;
 
-typedef interface ID2D1DrawTransform ID2D1DrawTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1DrawTransform ID2D1DrawTransform;
 
-typedef interface ID2D1ComputeTransform ID2D1ComputeTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1ComputeTransform ID2D1ComputeTransform;
 
-typedef interface ID2D1AnalysisTransform ID2D1AnalysisTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1AnalysisTransform ID2D1AnalysisTransform;
 
-typedef interface ID2D1SourceTransform ID2D1SourceTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1SourceTransform ID2D1SourceTransform;
 
-typedef interface ID2D1ConcreteTransform ID2D1ConcreteTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1ConcreteTransform ID2D1ConcreteTransform;
 
-typedef interface ID2D1BlendTransform ID2D1BlendTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1BlendTransform ID2D1BlendTransform;
 
-typedef interface ID2D1BorderTransform ID2D1BorderTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1BorderTransform ID2D1BorderTransform;
 
-typedef interface ID2D1OffsetTransform ID2D1OffsetTransform;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1OffsetTransform ID2D1OffsetTransform;
 
 typedef interface ID2D1BoundsAdjustmentTransform ID2D1BoundsAdjustmentTransform;
 
-typedef interface ID2D1EffectImpl ID2D1EffectImpl;
+_Windows_SupportedOS_Windows8_ typedef interface ID2D1EffectImpl ID2D1EffectImpl;
 
 typedef interface ID2D1EffectContext ID2D1EffectContext;
 
