diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d3d10effect.h b/generation/WinSDK/RecompiledIdlHeaders/um/d3d10effect.h
index 3f8a2fd8c5..1c9ee062bf 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d3d10effect.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d3d10effect.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 //////////////////////////////////////////////////////////////////////////////
 //
@@ -127,7 +128,7 @@ DEFINE_GUID(IID_ID3D10StateBlock,
 #undef INTERFACE
 #define INTERFACE ID3D10StateBlock
 
-DECLARE_INTERFACE_(ID3D10StateBlock, IUnknown)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10StateBlock, IUnknown)
 {
     STDMETHOD(QueryInterface)(THIS_ REFIID iid, LPVOID *ppv) PURE;
     STDMETHOD_(ULONG, AddRef)(THIS) PURE;
@@ -168,14 +169,14 @@ extern "C" {
 //
 //----------------------------------------------------------------------------
 
-HRESULT WINAPI D3D10StateBlockMaskUnion(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
-HRESULT WINAPI D3D10StateBlockMaskIntersect(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
-HRESULT WINAPI D3D10StateBlockMaskDifference(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
-HRESULT WINAPI D3D10StateBlockMaskEnableCapture(_Inout_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT RangeStart, UINT RangeLength);
-HRESULT WINAPI D3D10StateBlockMaskDisableCapture(_Inout_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT RangeStart, UINT RangeLength);
-HRESULT WINAPI D3D10StateBlockMaskEnableAll(_Out_ D3D10_STATE_BLOCK_MASK *pMask);
-HRESULT WINAPI D3D10StateBlockMaskDisableAll(_Out_ D3D10_STATE_BLOCK_MASK *pMask);
-BOOL    WINAPI D3D10StateBlockMaskGetSetting(_In_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT Entry);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskUnion(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskIntersect(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskDifference(_In_ D3D10_STATE_BLOCK_MASK *pA, _In_ D3D10_STATE_BLOCK_MASK *pB, _Out_ D3D10_STATE_BLOCK_MASK *pResult);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskEnableCapture(_Inout_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT RangeStart, UINT RangeLength);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskDisableCapture(_Inout_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT RangeStart, UINT RangeLength);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskEnableAll(_Out_ D3D10_STATE_BLOCK_MASK *pMask);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10StateBlockMaskDisableAll(_Out_ D3D10_STATE_BLOCK_MASK *pMask);
+_Win32_ImportLibrary_("d3d10.dll") BOOL    WINAPI D3D10StateBlockMaskGetSetting(_In_ D3D10_STATE_BLOCK_MASK *pMask, D3D10_DEVICE_STATE_TYPES StateType, UINT Entry);
 
 //----------------------------------------------------------------------------
 // D3D10CreateStateBlock
@@ -197,7 +198,7 @@ BOOL    WINAPI D3D10StateBlockMaskGetSetting(_In_ D3D10_STATE_BLOCK_MASK *pMask,
 //      bit mask
 //----------------------------------------------------------------------------
 
-HRESULT WINAPI D3D10CreateStateBlock(_In_ ID3D10Device *pDevice, _In_ D3D10_STATE_BLOCK_MASK *pStateBlockMask, _Out_ ID3D10StateBlock **ppStateBlock);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10CreateStateBlock(_In_ ID3D10Device *pDevice, _In_ D3D10_STATE_BLOCK_MASK *pStateBlockMask, _Out_ ID3D10StateBlock **ppStateBlock);
 
 #ifdef __cplusplus
 }
@@ -273,8 +274,8 @@ typedef struct _D3D10_EFFECT_TYPE_DESC
     LPCSTR  TypeName;               // Name of the type 
                                     // (e.g. "float4" or "MyStruct")
 
-    D3D10_SHADER_VARIABLE_CLASS    Class;  // (e.g. scalar, vector, object, etc.)
-    D3D10_SHADER_VARIABLE_TYPE     Type;   // (e.g. float, texture, vertexshader, etc.)
+    _Win32_AssociatedEnum_(D3D_SHADER_VARIABLE_CLASS) D3D10_SHADER_VARIABLE_CLASS    Class;  // (e.g. scalar, vector, object, etc.)
+    _Win32_AssociatedEnum_(D3D_SHADER_VARIABLE_TYPE) D3D10_SHADER_VARIABLE_TYPE     Type;   // (e.g. float, texture, vertexshader, etc.)
     
     UINT    Elements;               // Number of elements in this type
                                     // (0 if not an array) 
@@ -303,10 +304,10 @@ DEFINE_GUID(IID_ID3D10EffectType,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectType
 
-DECLARE_INTERFACE(ID3D10EffectType)
+DECLARE_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectType)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
-    STDMETHOD(GetDesc)(THIS_ D3D10_EFFECT_TYPE_DESC *pDesc) PURE;
+    STDMETHOD(GetDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_EFFECT_TYPE_DESC *pDesc) PURE;
     STDMETHOD_(ID3D10EffectType*, GetMemberTypeByIndex)(THIS_ UINT Index) PURE;
     STDMETHOD_(ID3D10EffectType*, GetMemberTypeByName)(THIS_ LPCSTR Name) PURE;
     STDMETHOD_(ID3D10EffectType*, GetMemberTypeBySemantic)(THIS_ LPCSTR Semantic) PURE;
@@ -370,7 +371,7 @@ typedef interface ID3D10EffectDepthStencilVariable ID3D10EffectDepthStencilVaria
 typedef interface ID3D10EffectRasterizerVariable ID3D10EffectRasterizerVariable;
 typedef interface ID3D10EffectSamplerVariable ID3D10EffectSamplerVariable;
 
-DECLARE_INTERFACE(ID3D10EffectVariable)
+DECLARE_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -419,7 +420,7 @@ DEFINE_GUID(IID_ID3D10EffectScalarVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectScalarVariable
 
-DECLARE_INTERFACE_(ID3D10EffectScalarVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectScalarVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -486,7 +487,7 @@ DEFINE_GUID(IID_ID3D10EffectVectorVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectVectorVariable
 
-DECLARE_INTERFACE_(ID3D10EffectVectorVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectVectorVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -520,21 +521,21 @@ DECLARE_INTERFACE_(ID3D10EffectVectorVariable, ID3D10EffectVariable)
     STDMETHOD(SetRawValue)(THIS_ _In_reads_bytes_(ByteCount) void *pData, UINT ByteOffset, UINT ByteCount) PURE;
     STDMETHOD(GetRawValue)(THIS_ _Out_writes_bytes_(ByteCount) void *pData, UINT ByteOffset, UINT ByteCount) PURE;
     
-    STDMETHOD(SetBoolVector) (THIS_ BOOL *pData) PURE;
-    STDMETHOD(SetIntVector)  (THIS_ int *pData) PURE;
-    STDMETHOD(SetFloatVector)(THIS_ float *pData) PURE;
+    STDMETHOD(SetBoolVector) (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") BOOL *pData) PURE;
+    STDMETHOD(SetIntVector)  (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") int *pData) PURE;
+    STDMETHOD(SetFloatVector)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
 
-    STDMETHOD(GetBoolVector) (THIS_ BOOL *pData) PURE;
-    STDMETHOD(GetIntVector)  (THIS_ int *pData) PURE;
-    STDMETHOD(GetFloatVector)(THIS_ float *pData) PURE;
+    STDMETHOD(GetBoolVector) (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") BOOL *pData) PURE;
+    STDMETHOD(GetIntVector)  (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") int *pData) PURE;
+    STDMETHOD(GetFloatVector)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
 
-    STDMETHOD(SetBoolVectorArray) (THIS_ BOOL *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(SetIntVectorArray)  (THIS_ int *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(SetFloatVectorArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(SetBoolVectorArray) (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") BOOL *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(SetIntVectorArray)  (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") int *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(SetFloatVectorArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
 
-    STDMETHOD(GetBoolVectorArray) (THIS_ BOOL *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(GetIntVectorArray)  (THIS_ int *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(GetFloatVectorArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(GetBoolVectorArray) (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") BOOL *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(GetIntVectorArray)  (THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") int *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(GetFloatVectorArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
 };
 
 //////////////////////////////////////////////////////////////////////////////
@@ -551,7 +552,7 @@ DEFINE_GUID(IID_ID3D10EffectMatrixVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectMatrixVariable
 
-DECLARE_INTERFACE_(ID3D10EffectMatrixVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectMatrixVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -585,17 +586,17 @@ DECLARE_INTERFACE_(ID3D10EffectMatrixVariable, ID3D10EffectVariable)
     STDMETHOD(SetRawValue)(THIS_ _In_reads_bytes_(ByteCount) void *pData, UINT ByteOffset, UINT ByteCount) PURE;
     STDMETHOD(GetRawValue)(THIS_ _Out_writes_bytes_(ByteCount) void *pData, UINT ByteOffset, UINT ByteCount) PURE;
     
-    STDMETHOD(SetMatrix)(THIS_ float *pData) PURE;
-    STDMETHOD(GetMatrix)(THIS_ float *pData) PURE;
+    STDMETHOD(SetMatrix)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
+    STDMETHOD(GetMatrix)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
     
-    STDMETHOD(SetMatrixArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(GetMatrixArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(SetMatrixArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(GetMatrixArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
     
-    STDMETHOD(SetMatrixTranspose)(THIS_ float *pData) PURE;
-    STDMETHOD(GetMatrixTranspose)(THIS_ float *pData) PURE;
+    STDMETHOD(SetMatrixTranspose)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
+    STDMETHOD(GetMatrixTranspose)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData) PURE;
     
-    STDMETHOD(SetMatrixTransposeArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
-    STDMETHOD(GetMatrixTransposeArray)(THIS_ float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(SetMatrixTransposeArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
+    STDMETHOD(GetMatrixTransposeArray)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") float *pData, UINT Offset, UINT Count) PURE;
 };
 
 //////////////////////////////////////////////////////////////////////////////
@@ -612,7 +613,7 @@ DEFINE_GUID(IID_ID3D10EffectStringVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectStringVariable
 
-DECLARE_INTERFACE_(ID3D10EffectStringVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectStringVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -665,7 +666,7 @@ DEFINE_GUID(IID_ID3D10EffectShaderResourceVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectShaderResourceVariable
 
-DECLARE_INTERFACE_(ID3D10EffectShaderResourceVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectShaderResourceVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -720,7 +721,7 @@ DEFINE_GUID(IID_ID3D10EffectRenderTargetViewVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectRenderTargetViewVariable
 
-DECLARE_INTERFACE_(ID3D10EffectRenderTargetViewVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectRenderTargetViewVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -775,7 +776,7 @@ DEFINE_GUID(IID_ID3D10EffectDepthStencilViewVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectDepthStencilViewVariable
 
-DECLARE_INTERFACE_(ID3D10EffectDepthStencilViewVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectDepthStencilViewVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
@@ -830,7 +831,7 @@ DEFINE_GUID(IID_ID3D10EffectConstantBuffer,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectConstantBuffer
 
-DECLARE_INTERFACE_(ID3D10EffectConstantBuffer, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectConstantBuffer, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -910,7 +911,7 @@ DEFINE_GUID(IID_ID3D10EffectShaderVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectShaderVariable
 
-DECLARE_INTERFACE_(ID3D10EffectShaderVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectShaderVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -967,7 +968,7 @@ DEFINE_GUID(IID_ID3D10EffectBlendVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectBlendVariable
 
-DECLARE_INTERFACE_(ID3D10EffectBlendVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectBlendVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -1001,7 +1002,7 @@ DECLARE_INTERFACE_(ID3D10EffectBlendVariable, ID3D10EffectVariable)
     STDMETHOD(GetRawValue)(THIS_ _Out_writes_bytes_(ByteCount) void *pData, UINT Offset, UINT ByteCount) PURE;
     
     STDMETHOD(GetBlendState)(THIS_ UINT Index, ID3D10BlendState **ppBlendState) PURE;
-    STDMETHOD(GetBackingStore)(THIS_ UINT Index, D3D10_BLEND_DESC *pBlendDesc) PURE;
+    STDMETHOD(GetBackingStore)(THIS_ UINT Index, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_BLEND_DESC *pBlendDesc) PURE;
 };
 
 //////////////////////////////////////////////////////////////////////////////
@@ -1018,7 +1019,7 @@ DEFINE_GUID(IID_ID3D10EffectDepthStencilVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectDepthStencilVariable
 
-DECLARE_INTERFACE_(ID3D10EffectDepthStencilVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectDepthStencilVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -1069,7 +1070,7 @@ DEFINE_GUID(IID_ID3D10EffectRasterizerVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectRasterizerVariable
 
-DECLARE_INTERFACE_(ID3D10EffectRasterizerVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectRasterizerVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -1120,7 +1121,7 @@ DEFINE_GUID(IID_ID3D10EffectSamplerVariable,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectSamplerVariable
 
-DECLARE_INTERFACE_(ID3D10EffectSamplerVariable, ID3D10EffectVariable)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectSamplerVariable, ID3D10EffectVariable)
 {
     STDMETHOD_(ID3D10EffectType*, GetType)(THIS) PURE;
     STDMETHOD(GetDesc)(THIS_ _Out_ D3D10_EFFECT_VARIABLE_DESC *pDesc) PURE;
@@ -1214,14 +1215,14 @@ DEFINE_GUID(IID_ID3D10EffectPass,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectPass
 
-DECLARE_INTERFACE(ID3D10EffectPass)
+DECLARE_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectPass)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
-    STDMETHOD(GetDesc)(THIS_ D3D10_PASS_DESC *pDesc) PURE;
+    STDMETHOD(GetDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_PASS_DESC *pDesc) PURE;
     
-    STDMETHOD(GetVertexShaderDesc)(THIS_ D3D10_PASS_SHADER_DESC *pDesc) PURE;
-    STDMETHOD(GetGeometryShaderDesc)(THIS_ D3D10_PASS_SHADER_DESC *pDesc) PURE;
-    STDMETHOD(GetPixelShaderDesc)(THIS_ D3D10_PASS_SHADER_DESC *pDesc) PURE;
+    STDMETHOD(GetVertexShaderDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_PASS_SHADER_DESC *pDesc) PURE;
+    STDMETHOD(GetGeometryShaderDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_PASS_SHADER_DESC *pDesc) PURE;
+    STDMETHOD(GetPixelShaderDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_PASS_SHADER_DESC *pDesc) PURE;
     
     STDMETHOD_(ID3D10EffectVariable*, GetAnnotationByIndex)(THIS_ UINT Index) PURE;
     STDMETHOD_(ID3D10EffectVariable*, GetAnnotationByName)(THIS_ LPCSTR Name) PURE;
@@ -1258,10 +1259,10 @@ DEFINE_GUID(IID_ID3D10EffectTechnique,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectTechnique
 
-DECLARE_INTERFACE(ID3D10EffectTechnique)
+DECLARE_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectTechnique)
 {
     STDMETHOD_(BOOL, IsValid)(THIS) PURE;
-    STDMETHOD(GetDesc)(THIS_ D3D10_TECHNIQUE_DESC *pDesc) PURE;
+    STDMETHOD(GetDesc)(THIS_ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") D3D10_TECHNIQUE_DESC *pDesc) PURE;
     
     STDMETHOD_(ID3D10EffectVariable*, GetAnnotationByIndex)(THIS_ UINT Index) PURE;
     STDMETHOD_(ID3D10EffectVariable*, GetAnnotationByName)(THIS_ LPCSTR Name) PURE;
@@ -1312,7 +1313,7 @@ DEFINE_GUID(IID_ID3D10Effect,
 #undef INTERFACE
 #define INTERFACE ID3D10Effect
 
-DECLARE_INTERFACE_(ID3D10Effect, IUnknown)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10Effect, IUnknown)
 {
     // IUnknown
     STDMETHOD(QueryInterface)(THIS_ REFIID iid, LPVOID *ppv) PURE;
@@ -1357,7 +1358,7 @@ DEFINE_GUID(IID_ID3D10EffectPool,
 #undef INTERFACE
 #define INTERFACE ID3D10EffectPool
 
-DECLARE_INTERFACE_(ID3D10EffectPool, IUnknown)
+DECLARE_INTERFACE_( _WIN32META_ANNOTATION_("win32metadata:agile") ID3D10EffectPool, IUnknown)
 {
     // IUnknown
     STDMETHOD(QueryInterface)(THIS_ REFIID iid, LPVOID *ppv) PURE;
@@ -1424,14 +1425,14 @@ extern "C" {
 //
 //----------------------------------------------------------------------------
 
-HRESULT WINAPI D3D10CompileEffectFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, LPCSTR pSrcFileName, _In_opt_ CONST D3D10_SHADER_MACRO *pDefines, 
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10CompileEffectFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, LPCSTR pSrcFileName, _In_opt_ CONST D3D10_SHADER_MACRO *pDefines, 
     _In_opt_ ID3D10Include *pInclude, UINT HLSLFlags, UINT FXFlags, 
     _Out_ ID3D10Blob **ppCompiledEffect, _Out_opt_ ID3D10Blob **ppErrors);
 
-HRESULT WINAPI D3D10CreateEffectFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, UINT FXFlags, _In_ ID3D10Device *pDevice, 
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10CreateEffectFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, UINT FXFlags, _In_ ID3D10Device *pDevice, 
     _In_opt_ ID3D10EffectPool *pEffectPool, _Out_ ID3D10Effect **ppEffect);
 
-HRESULT WINAPI D3D10CreateEffectPoolFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, UINT FXFlags, _In_ ID3D10Device *pDevice,
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10CreateEffectPoolFromMemory(_In_reads_bytes_(DataLength) void *pData, SIZE_T DataLength, UINT FXFlags, _In_ ID3D10Device *pDevice,
     _Out_ ID3D10EffectPool **ppEffectPool);
 
 
@@ -1449,7 +1450,7 @@ HRESULT WINAPI D3D10CreateEffectPoolFromMemory(_In_reads_bytes_(DataLength) void
 //      Returns a buffer containing the disassembled effect.
 //----------------------------------------------------------------------------
 
-HRESULT WINAPI D3D10DisassembleEffect(_In_ ID3D10Effect *pEffect, BOOL EnableColorCode, _Out_ ID3D10Blob **ppDisassembly);
+_Win32_ImportLibrary_("d3d10.dll") HRESULT WINAPI D3D10DisassembleEffect(_In_ ID3D10Effect *pEffect, BOOL EnableColorCode, _Out_ ID3D10Blob **ppDisassembly);
 
 #ifdef __cplusplus
 }
