diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_2.h b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_2.h
index b14095b907..1046a9a768 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_2.h
@@ -9,8 +9,13 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
+
 #endif
 
 /* verify that the <rpcsal.h> version is high enough to compile this file*/
@@ -170,6 +175,7 @@ EXTERN_C const IID IID_ID3D11DeviceContext2;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("420d5b32-b90c-4da4-bef0-359f6a24a83a")
+    _Windows_SupportedOS_Windows81_
     ID3D11DeviceContext2 : public ID3D11DeviceContext1
     {
     public:
@@ -2165,6 +2171,7 @@ EXTERN_C const IID IID_ID3D11Device2;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("9d06dffa-d1e5-4d07-83a8-1bb123f2f841")
+    _Windows_SupportedOS_Windows81_
     ID3D11Device2 : public ID3D11Device1
     {
     public:
