diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_3.h b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_3.h
index d638734975..a36c0ec21f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d3d11_3.h
@@ -9,8 +9,13 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
+
 #endif
 
 /* verify that the <rpcsal.h> version is high enough to compile this file*/
@@ -261,6 +266,7 @@ EXTERN_C const IID IID_ID3D11Texture2D1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("51218251-1E33-4617-9CCB-4D3A4367E7BB")
+    _Windows_SupportedOS_10240_
     ID3D11Texture2D1 : public ID3D11Texture2D
     {
     public:
@@ -503,6 +509,7 @@ EXTERN_C const IID IID_ID3D11Texture3D1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("0C711683-2853-4846-9BB0-F3E60639E46A")
+    _Windows_SupportedOS_10240_
     ID3D11Texture3D1 : public ID3D11Texture3D
     {
     public:
@@ -758,6 +765,7 @@ EXTERN_C const IID IID_ID3D11RasterizerState2;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("6fbd02fb-209f-46c4-b059-2ed15586a6ac")
+    _Windows_SupportedOS_10240_
     ID3D11RasterizerState2 : public ID3D11RasterizerState1
     {
     public:
@@ -1166,6 +1174,7 @@ EXTERN_C const IID IID_ID3D11ShaderResourceView1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("91308b87-9040-411d-8c67-c39253ce3802")
+    _Windows_SupportedOS_10240_
     ID3D11ShaderResourceView1 : public ID3D11ShaderResourceView
     {
     public:
@@ -1523,6 +1532,7 @@ EXTERN_C const IID IID_ID3D11RenderTargetView1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("ffbe2e23-f011-418a-ac56-5ceed7c5b94b")
+    _Windows_SupportedOS_10240_
     ID3D11RenderTargetView1 : public ID3D11RenderTargetView
     {
     public:
@@ -1868,6 +1878,7 @@ EXTERN_C const IID IID_ID3D11UnorderedAccessView1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("7b3b6153-a886-4544-ab37-6537c8500403")
+    _Windows_SupportedOS_10240_
     ID3D11UnorderedAccessView1 : public ID3D11UnorderedAccessView
     {
     public:
@@ -2058,6 +2069,7 @@ EXTERN_C const IID IID_ID3D11Query1;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("631b4766-36dc-461d-8db6-c47e13e60916")
+    _Windows_SupportedOS_10240_
     ID3D11Query1 : public ID3D11Query
     {
     public:
@@ -6320,6 +6332,7 @@ EXTERN_C const IID IID_ID3D11Device3;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("A05C8C37-D2C6-4732-B3A0-9CE0B0DC9AE6")
+    _Windows_SupportedOS_10240_
     ID3D11Device3 : public ID3D11Device2
     {
     public:
