diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/d3dcommon.h b/generation/WinSDK/RecompiledIdlHeaders/um/d3dcommon.h
index 09861ee173..556298021c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/d3dcommon.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/d3dcommon.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -503,7 +504,7 @@ typedef ID3D10Blob ID3DBlob;
 typedef ID3DBlob* LPD3DBLOB;
 #define IID_ID3DBlob IID_ID3D10Blob
 typedef void ( __stdcall *PFN_DESTRUCTION_CALLBACK )( 
-    void *pData);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void *pData);
 
 
 
@@ -521,7 +522,7 @@ EXTERN_C const IID IID_ID3DDestructionNotifier;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("a06eb39a-50da-425b-8c31-4eecd6c270f3")
+    MIDL_INTERFACE("a06eb39a-50da-425b-8c31-4eecd6c270f3") _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows7_ 
     ID3DDestructionNotifier : public IUnknown
     {
     public:
@@ -634,9 +635,9 @@ enum _D3D_INCLUDE_TYPE
 typedef interface ID3DInclude ID3DInclude;
 #undef INTERFACE
 #define INTERFACE ID3DInclude
-DECLARE_INTERFACE(ID3DInclude)
+DECLARE_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") ID3DInclude)
 {
-    STDMETHOD(Open)(THIS_ D3D_INCLUDE_TYPE IncludeType, LPCSTR pFileName, LPCVOID pParentData, LPCVOID *ppData, UINT *pBytes) PURE;
+    STDMETHOD(Open)(THIS_ D3D_INCLUDE_TYPE IncludeType, LPCSTR pFileName, LPCVOID pParentData, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPCVOID *ppData, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") UINT *pBytes) PURE;
     STDMETHOD(Close)(THIS_ LPCVOID pData) PURE;
 };
 typedef ID3DInclude* LPD3DINCLUDE;
