diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dciman.h b/generation/WinSDK/RecompiledIdlHeaders/um/dciman.h
index 1da64fa5f3..683a213e59 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dciman.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dciman.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /****************************************************************************
 
  DCIMAN.H
@@ -38,10 +39,14 @@ DECLARE_HANDLE(HWINWATCH);  // context handle for WinWatch instance
 /****************************************************************************
  ***************************************************************************/
 
+_Windows_SupportedOS_Windows2000_  
+
 extern HDC WINAPI DCIOpenProvider(void);
+_Windows_SupportedOS_Windows2000_  
 extern void WINAPI DCICloseProvider(HDC hdc);
 
-extern int WINAPI DCICreatePrimary(HDC hdc, LPDCISURFACEINFO FAR *lplpSurface);
+_Windows_SupportedOS_Windows2000_  
+extern int WINAPI DCICreatePrimary(HDC hdc,  LPDCISURFACEINFO FAR *lplpSurface);
 extern int WINAPI DCICreateOffscreen(HDC hdc, DWORD dwCompression, DWORD dwRedMask,
     DWORD dwGreenMask, DWORD dwBlueMask, DWORD dwWidth, DWORD dwHeight,
     DWORD dwDCICaps, DWORD dwBitCount, LPDCIOFFSCREEN FAR *lplpSurface);
@@ -78,8 +83,11 @@ extern BOOL WINAPI WinWatchNotify(HWINWATCH hWW, WINWATCHNOTIFYPROC NotifyCallba
 /****************************************************************************
  helper functions to call DCIMAN16.DLL
  ***************************************************************************/
+_Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
 extern void WINAPI DCIEndAccess(LPDCISURFACEINFO pdci);
+_Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
 extern DCIRVAL WINAPI DCIBeginAccess(LPDCISURFACEINFO pdci, int x, int y, int dx, int dy);
+_Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
 extern void WINAPI DCIDestroy(LPDCISURFACEINFO pdci);
 extern DCIRVAL WINAPI DCIDraw(LPDCIOFFSCREEN pdci);
 extern DCIRVAL WINAPI DCISetClipList(LPDCIOFFSCREEN pdci, LPRGNDATA prd);
