diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dcompanimation.h b/generation/WinSDK/RecompiledIdlHeaders/um/dcompanimation.h
index 3a46a8c0d2..9104aaaf15 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dcompanimation.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dcompanimation.h
@@ -19,6 +19,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -89,15 +90,21 @@ EXTERN_C const IID IID_IDCompositionAnimation;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("CBFD91D9-51B2-45e4-B3DE-D19CCFB863C5")
+    MIDL_INTERFACE("CBFD91D9-51B2-45e4-B3DE-D19CCFB863C5") _Windows_SupportedOS_Windows8_ 
     IDCompositionAnimation : public IUnknown
     {
     public:
+        
+        
         virtual HRESULT STDMETHODCALLTYPE Reset( void) = 0;
         
+        
+        
         virtual HRESULT STDMETHODCALLTYPE SetAbsoluteBeginTime( 
             LARGE_INTEGER beginTime) = 0;
         
+        
+        
         virtual HRESULT STDMETHODCALLTYPE AddCubic( 
             double beginOffset,
             float constantCoefficient,
@@ -105,6 +112,8 @@ EXTERN_C const IID IID_IDCompositionAnimation;
             float quadraticCoefficient,
             float cubicCoefficient) = 0;
         
+        
+        
         virtual HRESULT STDMETHODCALLTYPE AddSinusoidal( 
             double beginOffset,
             float bias,
@@ -112,10 +121,14 @@ EXTERN_C const IID IID_IDCompositionAnimation;
             float frequency,
             float phase) = 0;
         
+        
+        
         virtual HRESULT STDMETHODCALLTYPE AddRepeat( 
             double beginOffset,
             double durationToRepeat) = 0;
         
+        
+        
         virtual HRESULT STDMETHODCALLTYPE End( 
             double endOffset,
             float endValue) = 0;
