diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DhcpCSdk.h b/generation/WinSDK/RecompiledIdlHeaders/um/DhcpCSdk.h
index d044f5d159..36b694bda9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DhcpCSdk.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DhcpCSdk.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -189,7 +190,7 @@ typedef struct _DHCPAPI_PARAMS {                  // use this structure to reque
 typedef struct _DHCPAPI_PARAMS
 DHCPCAPI_PARAMS, *PDHCPCAPI_PARAMS, *LPDHCPCAPI_PARAMS;
 
-typedef struct _DHCPCAPI_PARAMS_ARARAY {          // array of params..
+typedef struct   _DHCPCAPI_PARAMS_ARARAY {          // array of params..
     ULONG                          nParams;       // size of array
     #ifdef __midl
         [size_is(nParams)] LPDHCPCAPI_PARAMS              Params;        // actual array
@@ -198,7 +199,7 @@ typedef struct _DHCPCAPI_PARAMS_ARARAY {          // array of params..
     #endif
 } DHCPCAPI_PARAMS_ARRAY, *PDHCPCAPI_PARAMS_ARRAY, *LPDHCPCAPI_PARAMS_ARRAY;
 
-typedef struct _DHCPCAPI_CLASSID {                // defines a client class id.
+typedef struct   _DHCPCAPI_CLASSID {                // defines a client class id.
     ULONG                          Flags;         // must be zero currently.
     #ifdef __midl
         [size_is(nBytesData)] LPBYTE                         Data;          // classid binary data.
@@ -215,21 +216,21 @@ typedef struct _DHCPCAPI_CLASSID {                // defines a client class id.
 #define     DHCPCAPI_REQUEST_CANCEL               0x08 // cancel request
 #define     DHCPCAPI_REQUEST_MASK                 0x0F // allowed flags..
 
-DWORD
+_Windows_SupportedOS_Windows2000_ DWORD
 APIENTRY
-DhcpCApiInitialize(
+  DhcpCApiInitialize(
     OUT     LPDWORD                Version
 );
 
-VOID
+_Windows_SupportedOS_Windows2000_ VOID
 APIENTRY
-DhcpCApiCleanup(
+  DhcpCApiCleanup(
     VOID
 );
 
-DWORD                                             // win32 status
+_Windows_SupportedOS_Windows2000_ DWORD                                             // win32 status
 APIENTRY
-DhcpRequestParams(                                // request parameters of client
+  DhcpRequestParams(                                // request parameters of client
     IN      DWORD                  Flags,         // must be DHCPCAPI_REQUEST_SYNCHRONOUS
     IN      LPVOID                 Reserved,      // this parameter is reserved
     _In_    LPWSTR                 AdapterName,   // adapter name to request for
@@ -247,9 +248,9 @@ DhcpRequestParams(                                // request parameters of clien
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
-DWORD                                             // win32 status
+_Windows_SupportedOS_Windows2000_ DWORD                                             // win32 status
 APIENTRY
-DhcpUndoRequestParams(                            // undo the effect of a persistent request -- currently undo from registry
+  DhcpUndoRequestParams(                            // undo the effect of a persistent request -- currently undo from registry
     _Reserved_ DWORD                Flags,         // must be zero, reserved
     _Reserved_ LPVOID               Reserved,      // this parameter is reserved
     _In_       LPWSTR               AdapterName,   // the name of the adpater to delete for
@@ -257,9 +258,9 @@ DhcpUndoRequestParams(                            // undo the effect of a persis
 );
 
 #define     DHCPCAPI_REGISTER_HANDLE_EVENT        0x01 // handle returned is to an event
-DWORD                                             // win32 status
+_Windows_SupportedOS_Windows2000_ DWORD                                             // win32 status
 APIENTRY
-DhcpRegisterParamChange(                          // notify if a parameter has changed
+  DhcpRegisterParamChange(                          // notify if a parameter has changed
     IN      DWORD                  Flags,         // must be zero, reserved
     _Reserved_ LPVOID                 Reserved,      // this parameter is reserved
     _In_       LPWSTR                 AdapterName,   // adapter of interest
@@ -269,17 +270,17 @@ DhcpRegisterParamChange(                          // notify if a parameter has c
 );
 
 #define     DHCPCAPI_DEREGISTER_HANDLE_EVENT      0x01 // de-register handle that is an event
-DWORD
+_Windows_SupportedOS_Windows2000_ DWORD
 APIENTRY
-DhcpDeRegisterParamChange(                        // undo the registration
+  DhcpDeRegisterParamChange(                        // undo the registration
     IN      DWORD                  Flags,         // MUST BE ZERO --> No flags yet.
     IN      LPVOID                 Reserved,      // MUST BE NULL --> Reserved
     IN      LPVOID                 Event          // handle to event returned by DhcpRegisterParamChange.
 );
 
-DWORD
+_Windows_SupportedOS_Windows2000_ DWORD
 APIENTRY
-DhcpRemoveDNSRegistrations(
+  DhcpRemoveDNSRegistrations(
     VOID
     );
 
