diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dmort.h b/generation/WinSDK/RecompiledIdlHeaders/um/dmort.h
index aa3bca83fc..306e5d8efc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dmort.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dmort.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //------------------------------------------------------------------------------
 // File: DMORt.h
 //
@@ -58,7 +59,7 @@ STDAPI MoCopyMediaType(DMO_MEDIA_TYPE *pmtDest, const DMO_MEDIA_TYPE *pmtSrc);
 // DMO_MEDIA_TYPE structures allocated by MoCreateMediaType() must be freed
 // by calling MoDeleteMediaType().
 //
-STDAPI MoCreateMediaType(DMO_MEDIA_TYPE **ppmt, DWORD cbFormat);
+STDAPI MoCreateMediaType(_Win32_RAIIFree_(MoDeleteMediaType) DMO_MEDIA_TYPE **ppmt, DWORD cbFormat);
 
 //
 // Frees any format block, releases any IUnknown, and deletes the
@@ -73,7 +74,7 @@ STDAPI MoDeleteMediaType(DMO_MEDIA_TYPE *pmt);
 // as well as a new format block for the target mediatype.  Trager mediatype
 // must later be freed using MoDeleteMediaType().
 //
-STDAPI MoDuplicateMediaType(DMO_MEDIA_TYPE **ppmtDest, const DMO_MEDIA_TYPE *pmtSrc);
+STDAPI MoDuplicateMediaType(_Win32_RAIIFree_(MoDeleteMediaType) DMO_MEDIA_TYPE **ppmtDest, const DMO_MEDIA_TYPE *pmtSrc);
 
 
 
