diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dpa_dsa.h b/generation/WinSDK/RecompiledIdlHeaders/um/dpa_dsa.h
index 394438f51b..59a2ba2595 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dpa_dsa.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dpa_dsa.h
@@ -23,6 +23,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -71,8 +72,14 @@ struct _DSA;
 typedef struct _DSA *HDSA;
 
 
+_Windows_SupportedOS_WindowsVista_ _Win32_RAIIFree_(DSA_Destroy, -1, 0) 
+
 WINCOMMCTRLAPI HDSA   WINAPI DSA_Create(int cbItem, int cItemGrow);
+_Windows_SupportedOS_WindowsVista_ 
+
 WINCOMMCTRLAPI BOOL   WINAPI DSA_Destroy(_Inout_opt_ HDSA hdsa);
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2003_
 WINCOMMCTRLAPI void   WINAPI DSA_DestroyCallback(_Inout_opt_ HDSA hdsa, _In_ PFNDAENUMCALLBACK pfnCB, _In_opt_ void *pData);
 #ifdef __cplusplus
 extern "C++" __inline void DSA_DestroyCallback(_Inout_opt_ HDSA hdsa, _In_ PFNDAENUMCALLBACKCONST pfnCB, _In_opt_ void *pData)
@@ -80,8 +87,8 @@ extern "C++" __inline void DSA_DestroyCallback(_Inout_opt_ HDSA hdsa, _In_ PFNDA
     DSA_DestroyCallback(hdsa, (PFNDAENUMCALLBACK)pfnCB, pData);
 }
 #endif
-WINCOMMCTRLAPI BOOL   WINAPI DSA_DeleteItem(_Inout_ HDSA hdsa, _In_ int i);
-WINCOMMCTRLAPI BOOL   WINAPI DSA_DeleteAllItems(_Inout_ HDSA hdsa);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DSA_DeleteItem(_Inout_ HDSA hdsa, _In_ int i);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DSA_DeleteAllItems(_Inout_ HDSA hdsa);
 WINCOMMCTRLAPI void   WINAPI DSA_EnumCallback(_In_ HDSA hdsa, _In_ PFNDAENUMCALLBACK pfnCB, _In_opt_ void *pData);
 #ifdef __cplusplus
 extern "C++" __inline void DSA_EnumCallback(_In_ HDSA hdsa, _In_ PFNDAENUMCALLBACKCONST pfnCB, _In_opt_ void *pData)
@@ -89,16 +96,16 @@ extern "C++" __inline void DSA_EnumCallback(_In_ HDSA hdsa, _In_ PFNDAENUMCALLBA
     DSA_EnumCallback(hdsa, (PFNDAENUMCALLBACK)pfnCB, pData);
 }
 #endif
-WINCOMMCTRLAPI int    WINAPI DSA_InsertItem(_Inout_ HDSA hdsa, _In_ int i, _In_ const void *pitem);
-WINCOMMCTRLAPI PVOID  WINAPI DSA_GetItemPtr(_In_ HDSA hdsa, _In_ int i);
-_Success_(return) WINCOMMCTRLAPI BOOL   WINAPI DSA_GetItem(_In_ HDSA hdsa, _In_ int i, _Out_writes_(_Inexpressible_(pdsa->cbItem)) void *pitem);
-_Success_(return) WINCOMMCTRLAPI BOOL   WINAPI DSA_SetItem(_Inout_ HDSA hdsa, _In_ int i, _In_ const void *pitem);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI int    WINAPI DSA_InsertItem(_Inout_ HDSA hdsa, _In_ int i, _In_ const void *pitem);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI PVOID  WINAPI DSA_GetItemPtr(_In_ HDSA hdsa, _In_ int i);
+_Windows_SupportedOS_WindowsVista_ _Success_(return) WINCOMMCTRLAPI BOOL   WINAPI DSA_GetItem(_In_ HDSA hdsa, _In_ int i, _Out_writes_(_Inexpressible_(pdsa->cbItem)) void *pitem);
+_Windows_SupportedOS_WindowsVista_ _Success_(return) WINCOMMCTRLAPI BOOL   WINAPI DSA_SetItem(_Inout_ HDSA hdsa, _In_ int i, _In_ const void *pitem);
 #define DSA_GetItemCount(hdsa)      (*(int *)(hdsa))
 #define DSA_AppendItem(hdsa, pitem) DSA_InsertItem(hdsa, DA_LAST, pitem)
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
-WINCOMMCTRLAPI HDSA   WINAPI DSA_Clone(_In_ HDSA hdsa);
-WINCOMMCTRLAPI ULONGLONG WINAPI DSA_GetSize(_In_opt_ HDSA hdsa);
+_Windows_SupportedOS_WindowsVista_ _Win32_RAIIFree_(DSA_Destroy, -1, 0) WINCOMMCTRLAPI HDSA   WINAPI DSA_Clone(_In_ HDSA hdsa);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI ULONGLONG WINAPI DSA_GetSize(_In_opt_ HDSA hdsa);
 WINCOMMCTRLAPI BOOL   WINAPI DSA_Sort(_Inout_ HDSA pdsa, _In_ PFNDACOMPARE pfnCompare, _In_ LPARAM lParam);
 #ifdef __cplusplus
 extern "C++" __inline BOOL DSA_Sort(_Inout_ HDSA hdsa, _In_ PFNDACOMPARECONST pfnCompare, _In_ LPARAM lParam)
@@ -121,10 +128,10 @@ struct _DPA;
 typedef struct _DPA *HDPA;
 
 
-WINCOMMCTRLAPI HDPA   WINAPI DPA_Create(int cItemGrow);
-WINCOMMCTRLAPI HDPA   WINAPI DPA_CreateEx(_In_ int cpGrow, _In_opt_ HANDLE hheap);
-WINCOMMCTRLAPI HDPA   WINAPI DPA_Clone(_In_ const HDPA hdpa, _Inout_opt_ HDPA hdpaNew);
-WINCOMMCTRLAPI BOOL   WINAPI DPA_Destroy(_Inout_opt_ HDPA hdpa);
+_Windows_SupportedOS_WindowsVista_ _Win32_RAIIFree_(DPA_Destroy, -1, 0) WINCOMMCTRLAPI HDPA   WINAPI DPA_Create(int cItemGrow);
+_Windows_SupportedOS_WindowsVista_ _Win32_RAIIFree_(DPA_Destroy, -1, 0) WINCOMMCTRLAPI HDPA   WINAPI DPA_CreateEx(_In_ int cpGrow, _In_opt_ HANDLE hheap);
+_Windows_SupportedOS_WindowsVista_ _Win32_RAIIFree_(DPA_Destroy, -1, 0) WINCOMMCTRLAPI HDPA   WINAPI DPA_Clone(_In_ const HDPA hdpa, _Inout_opt_ HDPA hdpaNew);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DPA_Destroy(_Inout_opt_ HDPA hdpa);
 WINCOMMCTRLAPI void   WINAPI DPA_DestroyCallback(_Inout_opt_ HDPA hdpa, _In_ PFNDAENUMCALLBACK pfnCB, _In_opt_ void *pData);
 #ifdef __cplusplus
 extern "C++" __inline void DPA_DestroyCallback(_In_ HDPA hdpa, _In_ PFNDAENUMCALLBACKCONST pfnCB, _In_opt_ void *pData)
@@ -132,8 +139,8 @@ extern "C++" __inline void DPA_DestroyCallback(_In_ HDPA hdpa, _In_ PFNDAENUMCAL
     DPA_DestroyCallback(hdpa, (PFNDAENUMCALLBACK)pfnCB, pData);
 }
 #endif
-WINCOMMCTRLAPI PVOID  WINAPI DPA_DeletePtr(_Inout_ HDPA hdpa, _In_ int i);
-WINCOMMCTRLAPI BOOL   WINAPI DPA_DeleteAllPtrs(_Inout_ HDPA hdpa);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI PVOID  WINAPI DPA_DeletePtr(_Inout_ HDPA hdpa, _In_ int i);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DPA_DeleteAllPtrs(_Inout_ HDPA hdpa);
 WINCOMMCTRLAPI void   WINAPI DPA_EnumCallback(_In_opt_ HDPA hdpa, _In_opt_ PFNDAENUMCALLBACK pfnCB, _In_opt_ void *pData);
 #ifdef __cplusplus
 extern "C++" __inline void DPA_EnumCallback(_In_ HDPA hdpa, _In_ PFNDAENUMCALLBACKCONST pfnCB, _In_opt_ void *pData)
@@ -141,11 +148,11 @@ extern "C++" __inline void DPA_EnumCallback(_In_ HDPA hdpa, _In_ PFNDAENUMCALLBA
     DPA_EnumCallback(hdpa, (PFNDAENUMCALLBACK)pfnCB, pData);
 }
 #endif
-WINCOMMCTRLAPI BOOL   WINAPI DPA_Grow(_Inout_ HDPA pdpa, _In_ int cp);
-WINCOMMCTRLAPI int    WINAPI DPA_InsertPtr(_Inout_ HDPA hdpa, _In_ int i, _In_opt_ void *p);
-WINCOMMCTRLAPI BOOL   WINAPI DPA_SetPtr(_Inout_ HDPA hdpa, _In_ int i, _In_opt_ void *p);
-WINCOMMCTRLAPI PVOID  WINAPI DPA_GetPtr(_In_ HDPA hdpa, _In_ INT_PTR i);
-WINCOMMCTRLAPI int    WINAPI DPA_GetPtrIndex(_In_ HDPA hdpa, _In_opt_ const void *p);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DPA_Grow(_Inout_ HDPA pdpa, _In_ int cp);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI int    WINAPI DPA_InsertPtr(_Inout_ HDPA hdpa, _In_ int i, _In_opt_ void *p);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL   WINAPI DPA_SetPtr(_Inout_ HDPA hdpa, _In_ int i, _In_opt_ void *p);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI PVOID  WINAPI DPA_GetPtr(_In_ HDPA hdpa, _In_ INT_PTR i);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI int    WINAPI DPA_GetPtrIndex(_In_ HDPA hdpa, _In_opt_ const void *p);
 
 #define DPA_GetPtrCount(hdpa)       (*(int *)(hdpa))
 #define DPA_SetPtrCount(hdpa, cItems) (*(int *)(hdpa) = (cItems))
@@ -155,7 +162,7 @@ WINCOMMCTRLAPI int    WINAPI DPA_GetPtrIndex(_In_ HDPA hdpa, _In_opt_ const void
 #define DPA_AppendPtr(hdpa, pitem)  DPA_InsertPtr(hdpa, DA_LAST, pitem)
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
-WINCOMMCTRLAPI ULONGLONG WINAPI DPA_GetSize(_In_opt_ HDPA hdpa);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI ULONGLONG WINAPI DPA_GetSize(_In_opt_ HDPA hdpa);
 #endif  // NTDDI_VISTA
 
 WINCOMMCTRLAPI BOOL   WINAPI DPA_Sort(_Inout_ HDPA hdpa, _In_ PFNDACOMPARE pfnCompare, _In_ LPARAM lParam);
@@ -181,7 +188,7 @@ extern "C++" __inline BOOL DPA_Sort(_Inout_ HDPA hdpa, _In_ PFNDACOMPARECONST pf
 // or some failure.
 //
 
-typedef struct _DPASTREAMINFO
+typedef struct   _DPASTREAMINFO
 {
     int    iPos;        // Index of item
     void *pvItem;
@@ -190,8 +197,8 @@ typedef struct _DPASTREAMINFO
 struct IStream;
 typedef HRESULT (CALLBACK *PFNDPASTREAM)(_In_ DPASTREAMINFO * pinfo, _In_ struct IStream * pstream, _In_opt_ void *pvInstData);
 
-WINCOMMCTRLAPI HRESULT WINAPI DPA_LoadStream(_Outptr_ HDPA * phdpa, _In_ PFNDPASTREAM pfn, _In_ struct IStream * pstream, _In_opt_ void *pvInstData);
-WINCOMMCTRLAPI HRESULT WINAPI DPA_SaveStream(_In_ HDPA hdpa, _In_ PFNDPASTREAM pfn, _In_ struct IStream * pstream, _In_opt_ void *pvInstData);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI HRESULT WINAPI DPA_LoadStream(_Win32_RAIIFree_(DPA_Destroy, -1, 0) _Outptr_ HDPA * phdpa, _In_ PFNDPASTREAM pfn, _In_ struct IStream * pstream, _In_opt_ void *pvInstData);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI HRESULT WINAPI DPA_SaveStream(_In_ HDPA hdpa, _In_ PFNDPASTREAM pfn, _In_ struct IStream * pstream, _In_opt_ void *pvInstData);
 
 // Merge two DPAs.  This takes two (optionally) presorted arrays and merges
 // the source array into the dest.  DPA_Merge uses the provided callbacks
@@ -223,8 +230,8 @@ WINCOMMCTRLAPI HRESULT WINAPI DPA_SaveStream(_In_ HDPA hdpa, _In_ PFNDPASTREAM p
 // the pointer of the merged item.  It's okay to simply use pvDest
 // as the returned pointer.
 //
-typedef void* (CALLBACK *PFNDPAMERGE)(_In_ UINT uMsg, _In_ void *pvDest, _In_ void *pvSrc, _In_ LPARAM lParam);
-typedef const void* (CALLBACK *PFNDPAMERGECONST)(_In_ UINT uMsg, _In_ const void *pvDest, _In_ const void *pvSrc, _In_ LPARAM lParam);
+typedef void* (CALLBACK *PFNDPAMERGE)(_Win32_AssociatedEnum_(DPAMM_MESSAGE) _In_ UINT uMsg, _In_ void *pvDest, _In_ void *pvSrc, _In_ LPARAM lParam);
+typedef const void* (CALLBACK *PFNDPAMERGECONST)(_Win32_AssociatedEnum_(DPAMM_MESSAGE) _In_ UINT uMsg, _In_ const void *pvDest, _In_ const void *pvSrc, _In_ LPARAM lParam);
 
 // Messages for merge callback
 #define DPAMM_MERGE     1
@@ -279,7 +286,7 @@ extern "C++" __inline int DPA_Search(_In_ HDPA hdpa, _In_opt_ const void *pFind,
 #endif // NO_COMMCTRL_DA
 
 
-WINCOMMCTRLAPI BOOL WINAPI Str_SetPtrW(_Inout_ LPWSTR * ppsz, _In_opt_ LPCWSTR psz);
+_Windows_SupportedOS_WindowsVista_ WINCOMMCTRLAPI BOOL WINAPI Str_SetPtrW(_Inout_ LPWSTR * ppsz, _In_opt_ LPCWSTR psz);
 
 
 
@@ -303,5 +310,24 @@ WINCOMMCTRLAPI BOOL WINAPI Str_SetPtrW(_Inout_ LPWSTR * ppsz, _In_opt_ LPCWSTR p
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("DPAMM_MERGE")
+#undef DPAMM_MERGE
+#pragma push_macro("DPAMM_DELETE")
+#undef DPAMM_DELETE
+#pragma push_macro("DPAMM_INSERT")
+#undef DPAMM_INSERT
+enum DPAMM_MESSAGE : unsigned int
+{
+    DPAMM_MERGE = 1U,
+    DPAMM_DELETE = 2U,
+    DPAMM_INSERT = 3U,
+};
+#pragma pop_macro("DPAMM_INSERT")
+#pragma pop_macro("DPAMM_DELETE")
+#pragma pop_macro("DPAMM_MERGE")
+#endif
+
 #endif  /* _INC_DPADSA */
 
