diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/drt.h b/generation/WinSDK/RecompiledIdlHeaders/um/drt.h
index e3841852cc..3a76f2e25d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/drt.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/drt.h
@@ -18,6 +18,9 @@
 #ifndef _DRT_H_
 #define _DRT_H_
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -425,6 +428,7 @@ typedef const DRT_EVENT_DATA *PCDRT_EVENT_DATA;
 //////////////////////////////////////////////////////////////////////////////
 // DRT Default Module Management APIs
 
+_Win32_ImportLibrary_("drtprov.dll")
 HRESULT WINAPI DrtCreatePnrpBootstrapResolver(
     _In_        BOOL   fPublish,
     _In_        PCWSTR pwzPeerName,
@@ -432,19 +436,23 @@ HRESULT WINAPI DrtCreatePnrpBootstrapResolver(
     _In_opt_    PCWSTR pwzPublishingIdentity,
     _Out_       DRT_BOOTSTRAP_PROVIDER** ppResolver);
 
+_Win32_ImportLibrary_("drtprov.dll")
 void WINAPI DrtDeletePnrpBootstrapResolver(
     _In_        DRT_BOOTSTRAP_PROVIDER* pResolver);
 
+_Win32_ImportLibrary_("drtprov.dll")
 HRESULT WINAPI DrtCreateDnsBootstrapResolver(
     _In_ USHORT port,
     _In_ PCWSTR pwszAddress,
     _Out_ DRT_BOOTSTRAP_PROVIDER** ppModule
     );
 
+_Win32_ImportLibrary_("drtprov.dll")
 void WINAPI DrtDeleteDnsBootstrapResolver(
     _In_ DRT_BOOTSTRAP_PROVIDER* pResolver
     );
 
+_Win32_ImportLibrary_("drttransport.dll")
 HRESULT WINAPI DrtCreateIpv6UdpTransport(
                 DRT_SCOPE scope,
                 ULONG dwScopeId,
@@ -452,43 +460,53 @@ HRESULT WINAPI DrtCreateIpv6UdpTransport(
     _Inout_     USHORT *pwPort,
     _Out_       HDRT_TRANSPORT *phTransport);
 
+_Win32_ImportLibrary_("drttransport.dll")
 HRESULT WINAPI DrtDeleteIpv6UdpTransport(
     _In_        HDRT_TRANSPORT hTransport);
 
+_Win32_ImportLibrary_("drtprov.dll")
 HRESULT WINAPI DrtCreateDerivedKeySecurityProvider(
     _In_        PCCERT_CONTEXT pRootCert,
     _In_opt_    PCCERT_CONTEXT pLocalCert,
     _Out_       DRT_SECURITY_PROVIDER** ppSecurityProvider);
 
+_Win32_ImportLibrary_("drtprov.dll")
 HRESULT WINAPI DrtCreateDerivedKey(
     _In_        PCCERT_CONTEXT pLocalCert,
     _Out_       DRT_DATA* pKey);
 
+_Win32_ImportLibrary_("drtprov.dll")
 void WINAPI DrtDeleteDerivedKeySecurityProvider(
     _In_        DRT_SECURITY_PROVIDER* pSecurityProvider);
 
+_Win32_ImportLibrary_("drtprov.dll")
 HRESULT WINAPI DrtCreateNullSecurityProvider(
     _Out_ DRT_SECURITY_PROVIDER** ppSecurityProvider);
 
+_Win32_ImportLibrary_("drtprov.dll")
 void WINAPI DrtDeleteNullSecurityProvider(
     _In_ DRT_SECURITY_PROVIDER* pSecurityProvider);
 
 //////////////////////////////////////////////////////////////////////////////
 // DRT Lifetime Management APIs
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtOpen(
     _In_        const DRT_SETTINGS* pSettings,
     _In_opt_    HANDLE hEvent,
     _In_opt_    const PVOID pvContext,
     _Out_       HDRT *phDrt);
 
+_Win32_ImportLibrary_("drt.dll")
 VOID WINAPI DrtClose(
     _In_        HDRT hDrt);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetEventDataSize(
     _In_        HDRT hDrt,
     _Out_       ULONG          *pulEventDataLen);
     
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetEventData(
     _In_        HDRT hDrt,
     _In_        ULONG          ulEventDataLen,
@@ -497,23 +515,27 @@ HRESULT WINAPI DrtGetEventData(
 //////////////////////////////////////////////////////////////////////////////
 // DRT Key Registration APIs
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtRegisterKey(
     _In_        HDRT hDrt,
     _In_        DRT_REGISTRATION *pRegistration,
     _In_opt_    PVOID pvKeyContext,
     _Out_       HDRT_REGISTRATION_CONTEXT *phKeyRegistration);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtUpdateKey(
     _In_        HDRT_REGISTRATION_CONTEXT hKeyRegistration,
     _In_        DRT_DATA *pAppData
     );
 
+_Win32_ImportLibrary_("drt.dll")
 VOID WINAPI DrtUnregisterKey(
     _In_        HDRT_REGISTRATION_CONTEXT hKeyRegistration);
 
 //////////////////////////////////////////////////////////////////////////////
 // DRT Search APIs
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtStartSearch(
     _In_        HDRT hDrt,
     _In_        DRT_DATA *pKey,
@@ -523,38 +545,46 @@ HRESULT WINAPI DrtStartSearch(
     _In_opt_    const PVOID pvContext,
     _Out_       HDRT_SEARCH_CONTEXT* hSearchContext);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtContinueSearch(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetSearchResultSize(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext,
     _Out_       ULONG               *pulSearchResultSize);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetSearchResult(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext,
     _Inout_     ULONG               ulSearchResultSize, 
     _Out_writes_bytes_all_(ulSearchResultSize) DRT_SEARCH_RESULT   *pSearchResult);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetSearchPathSize(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext,
     _Out_       ULONG               *pulSearchPathSize);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetSearchPath(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext,
     _Inout_     ULONG               ulSearchPathSize, 
     _Out_writes_bytes_all_(ulSearchPathSize) DRT_ADDRESS_LIST   *pSearchPath);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtEndSearch(
     _In_        HDRT_SEARCH_CONTEXT hSearchContext);
 
 //////////////////////////////////////////////////////////////////////////////
 // DRT Instance name related APIs
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetInstanceName(
     _In_        HDRT    hDrt,
     _In_        ULONG   ulcbInstanceNameSize,
     _Out_writes_bytes_all_(ulcbInstanceNameSize) PWSTR  pwzDrtInstanceName);
 
+_Win32_ImportLibrary_("drt.dll")
 HRESULT WINAPI DrtGetInstanceNameSize(
     _In_        HDRT    hDrt,
     _Out_       ULONG*  pulcbInstanceNameSize);
