diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DsGetDC.h b/generation/WinSDK/RecompiledIdlHeaders/um/DsGetDC.h
index ec178be31a..3785205855 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DsGetDC.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DsGetDC.h
@@ -27,6 +27,7 @@ Notes:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -127,7 +128,7 @@ extern "C" {
 // Structure returned from DsGetDcName
 //
 
-typedef struct _DOMAIN_CONTROLLER_INFOA {
+typedef struct   _DOMAIN_CONTROLLER_INFOA {
     LPSTR DomainControllerName;
     LPSTR DomainControllerAddress;
     ULONG DomainControllerAddressType;
@@ -139,7 +140,7 @@ typedef struct _DOMAIN_CONTROLLER_INFOA {
     LPSTR ClientSiteName;
 } DOMAIN_CONTROLLER_INFOA, *PDOMAIN_CONTROLLER_INFOA;
 
-typedef struct _DOMAIN_CONTROLLER_INFOW {
+typedef struct   _DOMAIN_CONTROLLER_INFOW {
 #ifdef MIDL_PASS
     [string,unique] wchar_t *DomainControllerName;
 #else // MIDL_PASS
@@ -223,6 +224,8 @@ typedef struct _DOMAIN_CONTROLLER_INFOW {
 // Function Prototypes
 //
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ 
+
 _Success_(return == ERROR_SUCCESS)
 DSGETDCAPI
 DWORD
@@ -233,9 +236,11 @@ DsGetDcNameA(
     _In_opt_ GUID *DomainGuid,
     _In_opt_ LPCSTR SiteName,
     _In_ ULONG Flags,
-    _Outptr_result_nullonfailure_ PDOMAIN_CONTROLLER_INFOA *DomainControllerInfo
+    _Outptr_result_nullonfailure_  PDOMAIN_CONTROLLER_INFOA *DomainControllerInfo
     );
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ 
+
 _Success_(return == ERROR_SUCCESS)
 DSGETDCAPI
 DWORD
@@ -246,7 +251,7 @@ DsGetDcNameW(
     _In_opt_ GUID *DomainGuid,
     _In_opt_ LPCWSTR SiteName,
     _In_ ULONG Flags,
-    _Outptr_result_nullonfailure_ PDOMAIN_CONTROLLER_INFOW *DomainControllerInfo
+    _Outptr_result_nullonfailure_  PDOMAIN_CONTROLLER_INFOW *DomainControllerInfo
     );
 
 #ifdef UNICODE
@@ -255,22 +260,26 @@ DsGetDcNameW(
 #define DsGetDcName DsGetDcNameA
 #endif // !UNICODE
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ 
+
 _Success_(return == ERROR_SUCCESS)
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetSiteNameA(
     _In_opt_ LPCSTR ComputerName,
-    _Outptr_result_nullonfailure_ LPSTR *SiteName
+    _Outptr_result_nullonfailure_  LPSTR *SiteName
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetSiteNameW(
     _In_opt_ LPCWSTR ComputerName,
-    _Outptr_result_nullonfailure_ LPWSTR *SiteName
+    _Outptr_result_nullonfailure_  LPWSTR *SiteName
     );
 
 #ifdef UNICODE
@@ -279,6 +288,8 @@ DsGetSiteNameW(
 #define DsGetSiteName DsGetSiteNameA
 #endif // !UNICODE
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -286,6 +297,8 @@ DsValidateSubnetNameW(
     _In_ LPCWSTR SubnetName
     );
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -305,7 +318,9 @@ DsValidateSubnetNameA(
 //
 #ifdef _WINSOCK2API_
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -313,10 +328,12 @@ DsAddressToSiteNamesW(
     _In_opt_ LPCWSTR ComputerName,
     _In_ DWORD EntryCount,
     _In_reads_(EntryCount) PSOCKET_ADDRESS SocketAddresses,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPWSTR **SiteNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPWSTR **SiteNames
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -324,7 +341,7 @@ DsAddressToSiteNamesA(
     _In_opt_ LPCSTR ComputerName,
     _In_ DWORD EntryCount,
     _In_reads_(EntryCount) PSOCKET_ADDRESS SocketAddresses,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPSTR **SiteNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPSTR **SiteNames
     );
 
 #ifdef UNICODE
@@ -333,7 +350,9 @@ DsAddressToSiteNamesA(
 #define DsAddressToSiteNames DsAddressToSiteNamesA
 #endif // !UNICODE
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -341,11 +360,13 @@ DsAddressToSiteNamesExW(
     _In_opt_ LPCWSTR ComputerName,
     _In_ DWORD EntryCount,
     _In_reads_(EntryCount) PSOCKET_ADDRESS SocketAddresses,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPWSTR **SiteNames,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPWSTR **SubnetNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPWSTR **SiteNames,
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPWSTR **SubnetNames
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -353,8 +374,8 @@ DsAddressToSiteNamesExA(
     _In_opt_ LPCSTR ComputerName,
     _In_ DWORD EntryCount,
     _In_reads_(EntryCount) PSOCKET_ADDRESS SocketAddresses,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPSTR **SiteNames,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount) LPSTR **SubnetNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPSTR **SiteNames,
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(EntryCount)  LPSTR **SubnetNames
     );
 
 #ifdef UNICODE
@@ -368,7 +389,7 @@ DsAddressToSiteNamesExA(
 // API to enumerate trusted domains
 //
 
-typedef struct _DS_DOMAIN_TRUSTSW {
+typedef struct   _DS_DOMAIN_TRUSTSW {
 
     //
     // Name of the trusted domain.
@@ -439,7 +460,7 @@ typedef struct _DS_DOMAIN_TRUSTSW {
 //
 // ANSI version of the above struct
 //
-typedef struct _DS_DOMAIN_TRUSTSA {
+typedef struct   _DS_DOMAIN_TRUSTSA {
     LPSTR NetbiosDomainName;
     LPSTR DnsDomainName;
     ULONG Flags;
@@ -458,25 +479,29 @@ typedef struct _DS_DOMAIN_TRUSTSA {
 #define PDS_DOMAIN_TRUSTS PDS_DOMAIN_TRUSTSA
 #endif // !UNICODE
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsEnumerateDomainTrustsW (
     _In_opt_ LPWSTR ServerName,
     _In_ ULONG Flags,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*DomainCount) PDS_DOMAIN_TRUSTSW *Domains,
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*DomainCount)  PDS_DOMAIN_TRUSTSW *Domains,
     _Out_ PULONG DomainCount
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsEnumerateDomainTrustsA (
     _In_opt_ LPSTR ServerName,
     _In_ ULONG Flags,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*DomainCount) PDS_DOMAIN_TRUSTSA *Domains,
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*DomainCount)  PDS_DOMAIN_TRUSTSA *Domains,
     _Out_ PULONG DomainCount
     );
 
@@ -493,7 +518,9 @@ DsEnumerateDomainTrustsA (
 
 #if defined(_NTLSA_) || defined(_NTSECAPI_)
 
-_Success_(return == ERROR_SUCCESS)
+_Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -501,13 +528,15 @@ DsGetForestTrustInformationW (
     _In_opt_ LPCWSTR ServerName,
     _In_opt_ LPCWSTR TrustedDomainName,
     _In_ DWORD Flags,
-    _Outptr_result_nullonfailure_ PLSA_FOREST_TRUST_INFORMATION *ForestTrustInfo
+    _Outptr_result_nullonfailure_  PLSA_FOREST_TRUST_INFORMATION *ForestTrustInfo
     );
 
 #define DS_GFTI_UPDATE_TDO      0x1     // Update TDO with information returned
 #define DS_GFTI_VALID_FLAGS     0x1     // All valid flags to DsGetForestTrustInformation
 
-_Success_(return == ERROR_SUCCESS)
+_Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -515,29 +544,33 @@ DsMergeForestTrustInformationW(
     _In_ LPCWSTR DomainName,
     _In_ PLSA_FOREST_TRUST_INFORMATION NewForestTrustInfo,
     _In_opt_ PLSA_FOREST_TRUST_INFORMATION OldForestTrustInfo,
-    _Outptr_result_nullonfailure_ PLSA_FOREST_TRUST_INFORMATION *MergedForestTrustInfo
+    _Outptr_result_nullonfailure_  PLSA_FOREST_TRUST_INFORMATION *MergedForestTrustInfo
     );
 
 #endif // _NTLSA_ || _NTSECAPI_
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetDcSiteCoverageW(
     _In_opt_ LPCWSTR ServerName,
     _Out_ PULONG EntryCount,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*EntryCount) LPWSTR **SiteNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*EntryCount)  LPWSTR **SiteNames
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetDcSiteCoverageA(
     _In_opt_ LPCSTR ServerName,
     _Out_ PULONG EntryCount,
-    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*EntryCount) LPSTR **SiteNames
+    _Outptr_result_nullonfailure_ _Outptr_result_buffer_(*EntryCount)  LPSTR **SiteNames
     );
 
 #ifdef UNICODE
@@ -546,6 +579,8 @@ DsGetDcSiteCoverageA(
 #define DsGetDcSiteCoverage DsGetDcSiteCoverageA
 #endif // !UNICODE
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -557,6 +592,8 @@ DsDeregisterDnsHostRecordsW (
     _In_ LPWSTR DnsHostName
     );
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -600,6 +637,8 @@ DsDeregisterDnsHostRecordsA (
             DS_WRITABLE_REQUIRED )
 
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -610,9 +649,11 @@ DsGetDcOpenW(
     _In_opt_ GUID *DomainGuid,
     _In_opt_ LPCWSTR DnsForestName,
     _In_ ULONG DcFlags,
-    _Out_ PHANDLE RetGetDcContext
+    _Out_ _Win32_RAIIFree_(DsGetDcCloseW) PHANDLE RetGetDcContext
     );
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 DWORD
 WINAPI
@@ -623,7 +664,7 @@ DsGetDcOpenA(
     _In_opt_ GUID *DomainGuid,
     _In_opt_ LPCSTR DnsForestName,
     _In_ ULONG DcFlags,
-    _Out_ PHANDLE RetGetDcContext
+    _Out_ _Win32_RAIIFree_(DsGetDcCloseW) PHANDLE RetGetDcContext
     );
 
 #ifdef UNICODE
@@ -632,26 +673,30 @@ DsGetDcOpenA(
 #define DsGetDcOpen DsGetDcOpenA
 #endif // !UNICODE
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetDcNextW(
     _In_ HANDLE GetDcContextHandle,
     _Out_opt_ PULONG SockAddressCount,
-    _Outptr_opt_result_buffer_(*SockAddressCount) LPSOCKET_ADDRESS *SockAddresses,
-    _Outptr_opt_result_nullonfailure_ LPWSTR *DnsHostName
+    _Outptr_opt_result_buffer_(*SockAddressCount)  LPSOCKET_ADDRESS *SockAddresses,
+    _Outptr_opt_result_nullonfailure_  LPWSTR *DnsHostName
     );
 
-_Success_(return == ERROR_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ _Success_(return == ERROR_SUCCESS)
+
+
 DSGETDCAPI
 DWORD
 WINAPI
 DsGetDcNextA(
     _In_ HANDLE GetDcContextHandle,
     _Out_opt_ PULONG SockAddressCount,
-    _Outptr_opt_result_buffer_(*SockAddressCount) LPSOCKET_ADDRESS *SockAddresses,
-    _Outptr_opt_result_nullonfailure_ LPSTR *DnsHostName
+    _Outptr_opt_result_buffer_(*SockAddressCount)  LPSOCKET_ADDRESS *SockAddresses,
+    _Outptr_opt_result_nullonfailure_  LPSTR *DnsHostName
     );
 
 #ifdef UNICODE
@@ -660,6 +705,8 @@ DsGetDcNextA(
 #define DsGetDcNext DsGetDcNextA
 #endif // !UNICODE
 
+_Windows_SupportedOS_WindowsVista_ 
+
 DSGETDCAPI
 VOID
 WINAPI
@@ -683,4 +730,19 @@ DsGetDcCloseW(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("DS_INET_ADDRESS")
+#undef DS_INET_ADDRESS
+#pragma push_macro("DS_NETBIOS_ADDRESS")
+#undef DS_NETBIOS_ADDRESS
+enum KERB_ADDRESS_TYPE : unsigned int
+{
+    DS_INET_ADDRESS = 1U,
+    DS_NETBIOS_ADDRESS = 2U,
+};
+#pragma pop_macro("DS_NETBIOS_ADDRESS")
+#pragma pop_macro("DS_INET_ADDRESS")
+#endif
+
 #endif // _DSGETDC_
