diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/DskQuota.h b/generation/WinSDK/RecompiledIdlHeaders/um/DskQuota.h
index bb349ec0bb..24cfe4b498 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/DskQuota.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/DskQuota.h
@@ -12,6 +12,7 @@
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -135,7 +136,7 @@ DEFINE_GUID(IID_IDiskQuotaEvents,
 //
 // Per-user quota information.
 //
-typedef struct DiskQuotaUserInformation {
+typedef struct   DiskQuotaUserInformation {
     LONGLONG QuotaUsed;
     LONGLONG QuotaThreshold;
     LONGLONG QuotaLimit;
@@ -329,12 +330,12 @@ DECLARE_INTERFACE_IID_(IDiskQuotaControl, IConnectionPointContainer, "7988B572-E
 
     STDMETHOD(AddUserSid)(THIS_
         PSID pUserSid,
-        DWORD fNameResolution,
+        _Win32_AssociatedEnum_(DISKQUOTA_USERNAME_RESOLVE) DWORD fNameResolution,
         PDISKQUOTA_USER *ppUser) PURE;
 
     STDMETHOD(AddUserName)(THIS_
         LPCWSTR pszLogonName,
-        DWORD fNameResolution,
+        _Win32_AssociatedEnum_(DISKQUOTA_USERNAME_RESOLVE) DWORD fNameResolution,
         PDISKQUOTA_USER *ppUser) PURE;
 
     STDMETHOD(DeleteUser)(THIS_
@@ -342,7 +343,7 @@ DECLARE_INTERFACE_IID_(IDiskQuotaControl, IConnectionPointContainer, "7988B572-E
 
     STDMETHOD(FindUserSid)(THIS_
         PSID pUserSid,
-        DWORD fNameResolution,
+        _Win32_AssociatedEnum_(DISKQUOTA_USERNAME_RESOLVE) DWORD fNameResolution,
         PDISKQUOTA_USER *ppUser) PURE;
 
     STDMETHOD(FindUserName)(THIS_
@@ -352,7 +353,7 @@ DECLARE_INTERFACE_IID_(IDiskQuotaControl, IConnectionPointContainer, "7988B572-E
     STDMETHOD(CreateEnumUsers)(THIS_
         PSID *rgpUserSids,
         DWORD cpSids,
-        DWORD fNameResolution,
+        _Win32_AssociatedEnum_(DISKQUOTA_USERNAME_RESOLVE) DWORD fNameResolution,
         PENUM_DISKQUOTA_USERS *ppEnum) PURE;
 
     STDMETHOD(CreateUserBatch)(THIS_
@@ -386,5 +387,24 @@ typedef IDiskQuotaEvents DISKQUOTA_EVENTS, *PDISKQUOTA_EVENTS;
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("DISKQUOTA_USERNAME_RESOLVE_ASYNC")
+#undef DISKQUOTA_USERNAME_RESOLVE_ASYNC
+#pragma push_macro("DISKQUOTA_USERNAME_RESOLVE_NONE")
+#undef DISKQUOTA_USERNAME_RESOLVE_NONE
+#pragma push_macro("DISKQUOTA_USERNAME_RESOLVE_SYNC")
+#undef DISKQUOTA_USERNAME_RESOLVE_SYNC
+enum DISKQUOTA_USERNAME_RESOLVE : unsigned int
+{
+    DISKQUOTA_USERNAME_RESOLVE_ASYNC = 2U,
+    DISKQUOTA_USERNAME_RESOLVE_NONE = 0U,
+    DISKQUOTA_USERNAME_RESOLVE_SYNC = 1U,
+};
+#pragma pop_macro("DISKQUOTA_USERNAME_RESOLVE_SYNC")
+#pragma pop_macro("DISKQUOTA_USERNAME_RESOLVE_NONE")
+#pragma pop_macro("DISKQUOTA_USERNAME_RESOLVE_ASYNC")
+#endif
+
 #endif // __DSKQUOTA_H
 
