diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
index 01a3b56cc5..8319325ac8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
@@ -9,20 +9,21 @@
 
 #ifndef DWRITE_3_H_INCLUDED
 #define DWRITE_3_H_INCLUDED
+#include "win32metadata_annotations.h"
 
 #pragma once
 
 #include <DWrite_2.h>
 
-interface IDWriteFontFaceReference;
-interface IDWriteFont3;
-interface IDWriteFontFace3;
-interface IDWriteFontSet;
+interface _Windows_SupportedOS_10240_  IDWriteFontFaceReference;
+interface _Windows_SupportedOS_10240_  IDWriteFont3;
+interface _Windows_SupportedOS_10240_  IDWriteFontFace3;
+interface _Windows_SupportedOS_10240_  IDWriteFontSet;
 interface IDWriteFontSetBuilder;
-interface IDWriteFontCollection1;
-interface IDWriteFontFamily1;
+interface _Windows_SupportedOS_Windows7_  IDWriteFontCollection1;
+interface _Windows_SupportedOS_10240_  IDWriteFontFamily1;
 interface IDWriteStringList;
-interface IDWriteFontDownloadQueue;
+interface _Windows_SupportedOS_Windows81_  IDWriteFontDownloadQueue;
 
 
 /// <summary>
@@ -61,7 +62,7 @@ interface IDWriteFontDownloadQueue;
 /// <summary>
 /// The font property enumeration identifies a string in a font.
 /// </summary>
-enum DWRITE_FONT_PROPERTY_ID
+enum   DWRITE_FONT_PROPERTY_ID
 {
     /// <summary>
     /// Unspecified font property identifier.
@@ -223,7 +224,7 @@ enum DWRITE_FONT_PROPERTY_ID
 /// Font property used for filtering font sets and
 /// building a font set with explicit properties.
 /// </summary>
-struct DWRITE_FONT_PROPERTY
+struct   DWRITE_FONT_PROPERTY
 {
     /// <summary>
     /// Specifies the requested font property, such as DWRITE_FONT_PROPERTY_ID_FAMILY_NAME.
@@ -253,7 +254,7 @@ struct DWRITE_FONT_PROPERTY
 /// <summary>
 /// Specifies the locality of a resource.
 /// </summary>
-enum DWRITE_LOCALITY
+enum   DWRITE_LOCALITY
 {
     /// <summary>
     /// The resource is remote, and information is unknown yet, including the file size and date.
@@ -279,7 +280,7 @@ enum DWRITE_LOCALITY
 /// <summary>
 /// Represents a method of rendering glyphs.
 /// </summary>
-enum DWRITE_RENDERING_MODE1
+enum   DWRITE_RENDERING_MODE1
 {
     /// <summary>
     /// Specifies that the rendering mode is determined automatically based on the font and size.
@@ -342,7 +343,7 @@ enum DWRITE_RENDERING_MODE1
 /// <summary>
 /// The interface that represents text rendering settings for glyph rasterization and filtering.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") IDWriteRenderingParams3 : public IDWriteRenderingParams2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_10240_  DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") IDWriteRenderingParams3 : public IDWriteRenderingParams2
 {
     /// <summary>
     /// Gets the rendering mode.
@@ -354,7 +355,7 @@ interface DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") IDWri
 /// <summary>
 /// The root factory interface for all DWrite objects.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") IDWriteFactory3 : public IDWriteFactory2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_10240_  DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") IDWriteFactory3 : public IDWriteFactory2
 {
     /// <summary>
     /// Creates a glyph run analysis object, which encapsulates information
@@ -535,7 +536,7 @@ interface DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") IDWri
 /// Unlike IDWriteFontFamily and IDWriteFontList, which are part of the IDWriteFontCollection heirarchy, font sets
 /// are unordered flat lists.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116B") IDWriteFontSet : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116B") IDWriteFontSet : public IUnknown
 {
     /// <summary>
     /// Get the number of total fonts in the set.
@@ -718,7 +719,7 @@ interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116B") IDWri
 /// <summary>
 /// Builder interface to add font face references and create a font set.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("2F642AFE-9C68-4F40-B8BE-457401AFCB3D") IDWriteFontSetBuilder : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("2F642AFE-9C68-4F40-B8BE-457401AFCB3D") IDWriteFontSetBuilder : public IUnknown
 {
     /// <summary>
     /// Adds a reference to a font to the set being built. The necessary
@@ -788,7 +789,7 @@ interface DWRITE_DECLARE_INTERFACE("2F642AFE-9C68-4F40-B8BE-457401AFCB3D") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") IDWriteFontCollection1 : public IDWriteFontCollection
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") IDWriteFontCollection1 : public IDWriteFontCollection
 {
     /// <summary>
     /// Get the underlying font set used by this collection.
@@ -822,7 +823,7 @@ interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") IDWri
 /// The IDWriteFontFamily interface represents a set of fonts that share the same design but are differentiated
 /// by weight, stretch, and style.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") IDWriteFontFamily1 : public IDWriteFontFamily
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") IDWriteFontFamily1 : public IDWriteFontFamily
 {
     /// <summary>
     /// Gets the current locality of a font given its zero-based index.
@@ -874,7 +875,7 @@ interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") IDWri
 /// <summary>
 /// The IDWriteFontList interface represents a list of fonts.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") IDWriteFontList1 : public IDWriteFontList
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_10240_  DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") IDWriteFontList1 : public IDWriteFontList
 {
     /// <summary>
     /// Gets the current locality of a font given its zero-based index.
@@ -929,7 +930,7 @@ interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") IDWri
 /// consists of a font file, font face index, and font face simulation. The file
 /// data may or may not be physically present on the local machine yet.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") IDWriteFontFaceReference : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") IDWriteFontFaceReference : public IUnknown
 {
     /// <summary>
     /// Creates a font face from the reference for use with layout,
@@ -1087,7 +1088,7 @@ interface DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") IDWri
 /// <summary>
 /// The IDWriteFont interface represents a font in a font collection.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") IDWriteFont3 : public IDWriteFont2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") IDWriteFont3 : public IDWriteFont2
 {
     /// <summary>
     /// Creates a font face object for the font.
@@ -1151,7 +1152,7 @@ interface DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") IDWri
 /// It contains font face type, appropriate file references and face identification data.
 /// Various font data such as metrics, names and glyph outlines is obtained from IDWriteFontFace.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("D37D7598-09BE-4222-A236-2081341CC1F2") IDWriteFontFace3 : public IDWriteFontFace2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("D37D7598-09BE-4222-A236-2081341CC1F2") IDWriteFontFace3 : public IDWriteFontFace2
 {
     /// <summary>
     /// Return a font face reference identifying this font.
@@ -1354,7 +1355,7 @@ interface DWRITE_DECLARE_INTERFACE("D37D7598-09BE-4222-A236-2081341CC1F2") IDWri
 /// font set. You can QueryInterface from an IDWriteLocalizedStrings to an
 /// IDWriteStringList.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("CFEE3140-1157-47CA-8B85-31BFCF3F2D0E") IDWriteStringList : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("CFEE3140-1157-47CA-8B85-31BFCF3F2D0E") IDWriteStringList : public IUnknown
 {
     /// <summary>
     /// Gets the number of strings.
@@ -1427,7 +1428,7 @@ interface DWRITE_DECLARE_INTERFACE("CFEE3140-1157-47CA-8B85-31BFCF3F2D0E") IDWri
 /// downloading thread, and objects must be prepared to handle calls on their methods
 /// from other threads at any time.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") IDWriteFontDownloadListener : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows81_  DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") IDWriteFontDownloadListener : public IUnknown
 {
     /// <summary>
     /// The DownloadCompleted method is called back on an arbitrary thread when a
@@ -1453,7 +1454,7 @@ interface DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") IDWri
 /// download completion. Callbacks to listeners will occur on the downloading thread, and objects must
 /// be must be able to handle calls on their methods from other threads at any time.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B71E6052-5AEA-4FA3-832E-F60D431F7E91") IDWriteFontDownloadQueue : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("B71E6052-5AEA-4FA3-832E-F60D431F7E91") IDWriteFontDownloadQueue : public IUnknown
 {
     /// <summary>
     /// Registers a client-defined listener object that receives download notifications.
@@ -1549,7 +1550,7 @@ interface DWRITE_DECLARE_INTERFACE("B71E6052-5AEA-4FA3-832E-F60D431F7E91") IDWri
 /// <summary>
 /// The GDI interop interface provides interoperability with GDI.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("4556BE70-3ABD-4F70-90BE-421780A6F515") IDWriteGdiInterop1 : public IDWriteGdiInterop
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("4556BE70-3ABD-4F70-90BE-421780A6F515") IDWriteGdiInterop1 : public IDWriteGdiInterop
 {
     /// <summary>
     /// Creates a font object that matches the properties specified by the LOGFONT structure.
@@ -1640,7 +1641,7 @@ struct DWRITE_LINE_METRICS1 : DWRITE_LINE_METRICS
 /// <summary>
 /// Specify whether DWRITE_FONT_METRICS::lineGap value should be part of the line metrics. 
 /// </summary>
-enum DWRITE_FONT_LINE_GAP_USAGE
+enum   DWRITE_FONT_LINE_GAP_USAGE
 {
     /// <summary>
     /// The usage of the font line gap depends on the method used for text layout.
@@ -1661,7 +1662,7 @@ enum DWRITE_FONT_LINE_GAP_USAGE
 /// <summary>
 /// The DWRITE_LINE_SPACING structure specifies the parameters used to specify how to manage space between lines.
 /// </summary>
-struct DWRITE_LINE_SPACING
+struct   DWRITE_LINE_SPACING
 {
     /// <summary>
     /// Method used to determine line spacing.
@@ -1702,7 +1703,7 @@ struct DWRITE_LINE_SPACING
     DWRITE_FONT_LINE_GAP_USAGE fontLineGapUsage;
 };
 
-interface DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") IDWriteTextFormat2 : public IDWriteTextFormat1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows81_  DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") IDWriteTextFormat2 : public IDWriteTextFormat1
 {
     /// <summary>
     /// Set line spacing.
@@ -1726,7 +1727,7 @@ interface DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") IDWri
     using IDWriteTextFormat1::GetLineSpacing;
 };
 
-interface DWRITE_DECLARE_INTERFACE("07DDCD52-020E-4DE8-AC33-6C953D83F92D") IDWriteTextLayout3 : public IDWriteTextLayout2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  _Windows_SupportedOS_Windows81_  DWRITE_DECLARE_INTERFACE("07DDCD52-020E-4DE8-AC33-6C953D83F92D") IDWriteTextLayout3 : public IDWriteTextLayout2
 {
     /// <summary>
     /// Invalidates the layout, forcing layout to remeasure before calling the
@@ -1872,7 +1873,7 @@ struct DWRITE_GLYPH_IMAGE_DATA
 /// <summary>
 /// Enumerator for an ordered collection of color glyph runs.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("7C5F86DA-C7A1-4F05-B8E1-55A179FE5A35") IDWriteColorGlyphRunEnumerator1 : public IDWriteColorGlyphRunEnumerator
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("7C5F86DA-C7A1-4F05-B8E1-55A179FE5A35") IDWriteColorGlyphRunEnumerator1 : public IDWriteColorGlyphRunEnumerator
 {
     /// <summary>
     /// Gets the current color glyph run.
@@ -1898,7 +1899,7 @@ interface DWRITE_DECLARE_INTERFACE("7C5F86DA-C7A1-4F05-B8E1-55A179FE5A35") IDWri
 /// It contains font face type, appropriate file references and face identification data.
 /// Various font data such as metrics, names and glyph outlines is obtained from IDWriteFontFace.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("27F2A904-4EB8-441D-9678-0563F53E3E2F") IDWriteFontFace4 : public IDWriteFontFace3
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("27F2A904-4EB8-441D-9678-0563F53E3E2F") IDWriteFontFace4 : public IDWriteFontFace3
 {
     /// <summary>
     /// Gets all the glyph image formats supported by the entire font (SVG, PNG, JPEG, ...).
@@ -1954,12 +1955,12 @@ interface DWRITE_DECLARE_INTERFACE("27F2A904-4EB8-441D-9678-0563F53E3E2F") IDWri
     /// </summary>
     /// <param name="glyphDataContext">Opaque context from ReadGlyphData.</param>
     STDMETHOD_(void, ReleaseGlyphImageData)(
-        void* glyphDataContext
+        _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* glyphDataContext
         ) PURE;
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("4B0B5BD3-0797-4549-8AC5-FE915CC53856") IDWriteFactory4 : public IDWriteFactory3
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("4B0B5BD3-0797-4549-8AC5-FE915CC53856") IDWriteFactory4 : public IDWriteFactory3
 {
     /// <summary>
     /// Translates a glyph run to a sequence of color glyph runs, which can be
@@ -2044,7 +2045,7 @@ interface DWRITE_DECLARE_INTERFACE("4B0B5BD3-0797-4549-8AC5-FE915CC53856") IDWri
 
 #if NTDDI_VERSION >= NTDDI_WIN10_RS2
 
-interface DWRITE_DECLARE_INTERFACE("3FF7715F-3CDC-4DC6-9B72-EC5621DCCAFD") IDWriteFontSetBuilder1 : public IDWriteFontSetBuilder
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("3FF7715F-3CDC-4DC6-9B72-EC5621DCCAFD") IDWriteFontSetBuilder1 : public IDWriteFontSetBuilder
 {
     /// <summary>
     /// Adds references to all the fonts in the specified font file. The method
@@ -2064,14 +2065,14 @@ interface DWRITE_DECLARE_INTERFACE("3FF7715F-3CDC-4DC6-9B72-EC5621DCCAFD") IDWri
 /// operation. A client can use the interface to wait for the operation to
 /// complete and to get the result.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("CE25F8FD-863B-4D13-9651-C1F88DC73FE2") IDWriteAsyncResult : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("CE25F8FD-863B-4D13-9651-C1F88DC73FE2") IDWriteAsyncResult : public IUnknown
 {
     /// <summary>
     /// The GetWaitHandleMethod method returns a handle that can be used to wait 
     /// for the asynchronous operation to complete. The handle remains valid
     /// until the interface is released.
     /// </summary>
-    STDMETHOD_(HANDLE, GetWaitHandle)() PURE;
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, GetWaitHandle)() PURE;
 
     /// <summary>
     /// The GetResult method returns the result of the asynchronous operation.
@@ -2107,7 +2108,7 @@ struct DWRITE_FILE_FRAGMENT
 /// <remarks>
 /// For more information, see the description of IDWriteRemoteFontFileLoader.
 /// </remarks>
-interface DWRITE_DECLARE_INTERFACE("4DB3757A-2C72-4ED9-B2B6-1ABABE1AFF9C") IDWriteRemoteFontFileStream : public IDWriteFontFileStream
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("4DB3757A-2C72-4ED9-B2B6-1ABABE1AFF9C") IDWriteRemoteFontFileStream : public IDWriteFontFileStream
 {
     /// <summary>
     /// GetLocalFileSize returns the number of bytes of the font file that are
@@ -2140,7 +2141,7 @@ interface DWRITE_DECLARE_INTERFACE("4DB3757A-2C72-4ED9-B2B6-1ABABE1AFF9C") IDWri
         UINT64 fileOffset,
         UINT64 fragmentSize,
         _Out_ BOOL* isLocal,
-        _Out_range_(0, fragmentSize) UINT64* partialSize
+        _WIN32META_ANNOTATION_("win32metadata:in") _Out_range_(0, fragmentSize) UINT64* partialSize
         ) PURE;
 
     /// <summary>
@@ -2199,7 +2200,7 @@ enum DWRITE_CONTAINER_TYPE
 /// DWRITE_E_REMOTEFONT. This error signifies to DWrite that it should add requests to the 
 /// font download queue.
 /// </remarks>
-interface DWRITE_DECLARE_INTERFACE("68648C83-6EDE-46C0-AB46-20083A887FDE") IDWriteRemoteFontFileLoader : public IDWriteFontFileLoader
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("68648C83-6EDE-46C0-AB46-20083A887FDE") IDWriteRemoteFontFileLoader : public IDWriteFontFileLoader
 {
     /// <summary>
     /// Creates a remote font file stream object that encapsulates an open file resource
@@ -2269,7 +2270,7 @@ interface DWRITE_DECLARE_INTERFACE("68648C83-6EDE-46C0-AB46-20083A887FDE") IDWri
 /// unregistering using IDWriteFactory::RegisterFontFileLoader and 
 /// IDWriteFactory::UnregisterFontFileLoader.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("DC102F47-A12D-4B1C-822D-9E117E33043F") IDWriteInMemoryFontFileLoader : public IDWriteFontFileLoader
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("DC102F47-A12D-4B1C-822D-9E117E33043F") IDWriteInMemoryFontFileLoader : public IDWriteFontFileLoader
 {
     /// <summary>
     /// The CreateInMemoryFontFileReference method creates a font file reference
@@ -2310,7 +2311,7 @@ interface DWRITE_DECLARE_INTERFACE("DC102F47-A12D-4B1C-822D-9E117E33043F") IDWri
 /// <summary>
 /// The root factory interface for all DWrite objects.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("958DB99A-BE2A-4F09-AF7D-65189803D1D3") IDWriteFactory5 : public IDWriteFactory4
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("958DB99A-BE2A-4F09-AF7D-65189803D1D3") IDWriteFactory5 : public IDWriteFactory4
 {
     /// <summary>
     /// Creates an empty font set builder to add font face references
@@ -2553,7 +2554,7 @@ DEFINE_ENUM_FLAG_OPERATORS(DWRITE_FONT_AXIS_ATTRIBUTES);
 #endif
 
 
-interface DWRITE_DECLARE_INTERFACE("F3744D80-21F7-42EB-B35D-995BC72FC223") IDWriteFactory6 : public IDWriteFactory5
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("F3744D80-21F7-42EB-B35D-995BC72FC223") IDWriteFactory6 : public IDWriteFactory5
 {
     /// <summary>
     /// Creates a reference to a specific font instance within a file.
@@ -2689,7 +2690,7 @@ interface DWRITE_DECLARE_INTERFACE("F3744D80-21F7-42EB-B35D-995BC72FC223") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("98EFF3A5-B667-479A-B145-E2FA5B9FDC29") IDWriteFontFace5 : public IDWriteFontFace4
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("98EFF3A5-B667-479A-B145-E2FA5B9FDC29") IDWriteFontFace5 : public IDWriteFontFace4
 {
     /// <summary>
     /// Get the number of axes defined by the font. This includes both static and variable axes.
@@ -2741,7 +2742,7 @@ interface DWRITE_DECLARE_INTERFACE("98EFF3A5-B667-479A-B145-E2FA5B9FDC29") IDWri
 /// <summary>
 /// Interface to return axis information for a font resource and create specific font face instances.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("1F803A76-6871-48E8-987F-B975551C50F2") IDWriteFontResource : public IUnknown
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("1F803A76-6871-48E8-987F-B975551C50F2") IDWriteFontResource : public IUnknown
 {
     /// <summary>
     /// Get the font file of the resource.
@@ -2914,7 +2915,7 @@ interface DWRITE_DECLARE_INTERFACE("1F803A76-6871-48E8-987F-B975551C50F2") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("C081FE77-2FD1-41AC-A5A3-34983C4BA61A") IDWriteFontFaceReference1 : public IDWriteFontFaceReference
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("C081FE77-2FD1-41AC-A5A3-34983C4BA61A") IDWriteFontFaceReference1 : public IDWriteFontFaceReference
 {
     /// <summary>
     /// Creates a font face from the reference for use with layout, shaping, or rendering.
@@ -2952,7 +2953,7 @@ interface DWRITE_DECLARE_INTERFACE("C081FE77-2FD1-41AC-A5A3-34983C4BA61A") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("EE5BA612-B131-463C-8F4F-3189B9401E45") IDWriteFontSetBuilder2 : public IDWriteFontSetBuilder1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("EE5BA612-B131-463C-8F4F-3189B9401E45") IDWriteFontSetBuilder2 : public IDWriteFontSetBuilder1
 {
     /// <summary>
     /// Adds a font to the set being built, with the caller supplying enough information to search on
@@ -3002,7 +3003,7 @@ interface DWRITE_DECLARE_INTERFACE("EE5BA612-B131-463C-8F4F-3189B9401E45") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("7E9FDA85-6C92-4053-BC47-7AE3530DB4D3") IDWriteFontSet1 : public IDWriteFontSet
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("7E9FDA85-6C92-4053-BC47-7AE3530DB4D3") IDWriteFontSet1 : public IDWriteFontSet
 {
     /// <summary>
     /// Generates a matching font set based on the requested inputs, ordered so that nearer matches are earlier.
@@ -3230,7 +3231,7 @@ interface DWRITE_DECLARE_INTERFACE("7E9FDA85-6C92-4053-BC47-7AE3530DB4D3") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("C0763A34-77AF-445A-B735-08C37B0A5BF5") IDWriteFontList2 : public IDWriteFontList1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("C0763A34-77AF-445A-B735-08C37B0A5BF5") IDWriteFontList2 : public IDWriteFontList1
 {
     /// <summary>
     /// Get the underlying font set used by this list.
@@ -3245,7 +3246,7 @@ interface DWRITE_DECLARE_INTERFACE("C0763A34-77AF-445A-B735-08C37B0A5BF5") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("3ED49E77-A398-4261-B9CF-C126C2131EF3") IDWriteFontFamily2 : public IDWriteFontFamily1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("3ED49E77-A398-4261-B9CF-C126C2131EF3") IDWriteFontFamily2 : public IDWriteFontFamily1
 {
     /// <summary>
     /// Gets a list of fonts in the font family ranked in order of how well they match the specified axis values.
@@ -3277,7 +3278,7 @@ interface DWRITE_DECLARE_INTERFACE("3ED49E77-A398-4261-B9CF-C126C2131EF3") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("514039C6-4617-4064-BF8B-92EA83E506E0") IDWriteFontCollection2 : public IDWriteFontCollection1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("514039C6-4617-4064-BF8B-92EA83E506E0") IDWriteFontCollection2 : public IDWriteFontCollection1
 {
     /// <summary>
     /// Creates a font family object given a zero-based font family index.
@@ -3338,7 +3339,7 @@ interface DWRITE_DECLARE_INTERFACE("514039C6-4617-4064-BF8B-92EA83E506E0") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("05A9BF42-223F-4441-B5FB-8263685F55E9") IDWriteTextLayout4 : public IDWriteTextLayout3
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("05A9BF42-223F-4441-B5FB-8263685F55E9") IDWriteTextLayout4 : public IDWriteTextLayout3
 {
     /// <summary>
     /// Set values for font axes over a range of text.
@@ -3395,7 +3396,7 @@ interface DWRITE_DECLARE_INTERFACE("05A9BF42-223F-4441-B5FB-8263685F55E9") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("6D3B5641-E550-430D-A85B-B7BF48A93427") IDWriteTextFormat3 : public IDWriteTextFormat2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("6D3B5641-E550-430D-A85B-B7BF48A93427") IDWriteTextFormat3 : public IDWriteTextFormat2
 {
     /// <summary>
     /// Set values for font axes of the format.
@@ -3447,7 +3448,7 @@ interface DWRITE_DECLARE_INTERFACE("6D3B5641-E550-430D-A85B-B7BF48A93427") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("2397599D-DD0D-4681-BD6A-F4F31EAADE77") IDWriteFontFallback1 : public IDWriteFontFallback
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("2397599D-DD0D-4681-BD6A-F4F31EAADE77") IDWriteFontFallback1 : public IDWriteFontFallback
 {
     /// <summary>
     /// Determines an appropriate font to use to render the range of text.
@@ -3477,7 +3478,7 @@ interface DWRITE_DECLARE_INTERFACE("2397599D-DD0D-4681-BD6A-F4F31EAADE77") IDWri
         _In_opt_z_ WCHAR const* baseFamilyName,
         _In_reads_(fontAxisValueCount) DWRITE_FONT_AXIS_VALUE const* fontAxisValues,
         UINT32 fontAxisValueCount,
-        _Deref_out_range_(0, textLength) UINT32* mappedLength,
+        _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") _Deref_out_range_(0, textLength) UINT32* mappedLength,
         _Out_ FLOAT* scale,
         _COM_Outptr_ IDWriteFontFace5** mappedFontFace
         ) PURE;
@@ -3489,7 +3490,7 @@ interface DWRITE_DECLARE_INTERFACE("2397599D-DD0D-4681-BD6A-F4F31EAADE77") IDWri
 
 #if NTDDI_VERSION >= NTDDI_WIN10_RS4
 
-interface DWRITE_DECLARE_INTERFACE("DC7EAD19-E54C-43AF-B2DA-4E2B79BA3F7F") IDWriteFontSet2 : public IDWriteFontSet1
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("DC7EAD19-E54C-43AF-B2DA-4E2B79BA3F7F") IDWriteFontSet2 : public IDWriteFontSet1
 {
     /// <summary>
     /// Gets the expiration event for the font set, if any. The expiration event is set on a system font set object if
@@ -3504,10 +3505,10 @@ interface DWRITE_DECLARE_INTERFACE("DC7EAD19-E54C-43AF-B2DA-4E2B79BA3F7F") IDWri
     /// object, and remains valid as long as the client holds a reference to the font set. The client can wait
     /// on the returned event or use RegisterWaitForSingleObject to request a callback when the event is set.
     /// </remarks>
-    STDMETHOD_(HANDLE, GetExpirationEvent)() PURE;
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, GetExpirationEvent)() PURE;
 };
 
-interface DWRITE_DECLARE_INTERFACE("A4D055A6-F9E3-4E25-93B7-9E309F3AF8E9") IDWriteFontCollection3 : public IDWriteFontCollection2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("A4D055A6-F9E3-4E25-93B7-9E309F3AF8E9") IDWriteFontCollection3 : public IDWriteFontCollection2
 {
     /// <summary>
     /// Gets the expiration event for the font collection, if any. The expiration event is set on a system font 
@@ -3523,10 +3524,10 @@ interface DWRITE_DECLARE_INTERFACE("A4D055A6-F9E3-4E25-93B7-9E309F3AF8E9") IDWri
     /// object, and remains valid as long as the client holds a reference to the font collection. The client can wait
     /// on the returned event or use RegisterWaitForSingleObject to request a callback when the event is set.
     /// </remarks>
-    STDMETHOD_(HANDLE, GetExpirationEvent)() PURE;
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, GetExpirationEvent)() PURE;
 };
 
-interface DWRITE_DECLARE_INTERFACE("35D0E0B3-9076-4D2E-A016-A91B568A06B4") IDWriteFactory7 : public IDWriteFactory6
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("35D0E0B3-9076-4D2E-A016-A91B568A06B4") IDWriteFactory7 : public IDWriteFactory6
 {
     /// <summary>
     /// Retrieves the set of system fonts.
@@ -3597,7 +3598,7 @@ enum DWRITE_FONT_SOURCE_TYPE
     DWRITE_FONT_SOURCE_TYPE_REMOTE_FONT_PROVIDER
 };
 
-interface DWRITE_DECLARE_INTERFACE("7C073EF2-A7F4-4045-8C32-8AB8AE640F90") IDWriteFontSet3 : public IDWriteFontSet2
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("7C073EF2-A7F4-4045-8C32-8AB8AE640F90") IDWriteFontSet3 : public IDWriteFontSet2
 {
     /// <summary>
     /// Gets the font source type of the specified font.
@@ -3632,7 +3633,7 @@ interface DWRITE_DECLARE_INTERFACE("7C073EF2-A7F4-4045-8C32-8AB8AE640F90") IDWri
 
 #if NTDDI_VERSION >= NTDDI_WIN10_MN
 
-interface DWRITE_DECLARE_INTERFACE("C4B1FE1B-6E84-47D5-B54C-A597981B06AD") IDWriteFontFace6 : public IDWriteFontFace5
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("C4B1FE1B-6E84-47D5-B54C-A597981B06AD") IDWriteFontFace6 : public IDWriteFontFace5
 {
     /// <summary>
     /// Creates a localized strings object that contains the family names for the font, indexed by locale name.
@@ -3669,7 +3670,7 @@ interface DWRITE_DECLARE_INTERFACE("C4B1FE1B-6E84-47D5-B54C-A597981B06AD") IDWri
 
 #if NTDDI_VERSION >= NTDDI_WIN10_NI
 
-interface DWRITE_DECLARE_INTERFACE("EEC175FC-BEA9-4C86-8B53-CCBDD7DF0C82") IDWriteFontSet4 : public IDWriteFontSet3
+interface _WIN32META_ANNOTATION_("win32metadata:agile")  DWRITE_DECLARE_INTERFACE("EEC175FC-BEA9-4C86-8B53-CCBDD7DF0C82") IDWriteFontSet4 : public IDWriteFontSet3
 {
     /// <summary>
     /// Computes derived font axis values from the specified font weight, stretch, style, and size.
@@ -3748,7 +3749,7 @@ struct DWRITE_BITMAP_DATA_BGRA32
 /// <summary>
 /// Encapsulates a bitmap which can be used for rendering glyphs.
 /// </summary>
-DWRITE_BEGIN_INTERFACE(IDWriteBitmapRenderTarget2, "C553A742-FC01-44DA-A66E-B8B9ED6C3995") : IDWriteBitmapRenderTarget1
+DWRITE_BEGIN_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") IDWriteBitmapRenderTarget2, "C553A742-FC01-44DA-A66E-B8B9ED6C3995") : IDWriteBitmapRenderTarget1
 {
     /// <summary>
     /// Gets the demensions and a pointer to the system memory bitmap encapsulated by this
@@ -4216,7 +4217,7 @@ struct DWRITE_PAINT_ELEMENT
             /// <summary>
             /// Specifies the compositing or blending mode.
             /// </summary>
-            DWRITE_COLOR_COMPOSITE_MODE mode;
+            _Win32_AssociatedEnum_(DWRITE_COLOR_COMPOSITE_MODE) DWRITE_COLOR_COMPOSITE_MODE mode;
         } composite;
     } paint;
 };
@@ -4227,7 +4228,7 @@ struct D2D1_GRADIENT_STOP;
 /// Interface used to read color glyph data for a specific font. A color glyph is
 /// represented as a visual tree of paint elements.
 /// </summary>
-DWRITE_BEGIN_INTERFACE(IDWritePaintReader, "8128E912-3B97-42A5-AB6C-24AAD3A86E54") : IUnknown
+DWRITE_BEGIN_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") IDWritePaintReader, "8128E912-3B97-42A5-AB6C-24AAD3A86E54") : IUnknown
 {
     /// <summary>
     /// Sets the current glyph and positions the reader on the root paint element of the
@@ -4395,7 +4396,7 @@ DWRITE_BEGIN_INTERFACE(IDWritePaintReader, "8128E912-3B97-42A5-AB6C-24AAD3A86E54
         ) PURE;
 };
 
-DWRITE_BEGIN_INTERFACE(IDWriteFontFace7, "3945B85B-BC95-40F7-B72C-8B73BFC7E13B") : IDWriteFontFace6
+DWRITE_BEGIN_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") IDWriteFontFace7, "3945B85B-BC95-40F7-B72C-8B73BFC7E13B") : IDWriteFontFace6
 {
     /// <summary>
     /// Returns the maximum paint feature level supported for the specified glyph image format.
@@ -4427,7 +4428,7 @@ DWRITE_BEGIN_INTERFACE(IDWriteFontFace7, "3945B85B-BC95-40F7-B72C-8B73BFC7E13B")
 };
 
 
-DWRITE_BEGIN_INTERFACE(IDWriteFactory8, "EE0A7FB5-DEF4-4C23-A454-C9C7DC878398") : IDWriteFactory7
+DWRITE_BEGIN_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") IDWriteFactory8, "EE0A7FB5-DEF4-4C23-A454-C9C7DC878398") : IDWriteFactory7
 {
     /// <summary>
     /// Translates a glyph run to a sequence of color glyph runs, which can be
@@ -4480,7 +4481,7 @@ DWRITE_BEGIN_INTERFACE(IDWriteFactory8, "EE0A7FB5-DEF4-4C23-A454-C9C7DC878398")
 /// <summary>
 /// Encapsulates a bitmap which can be used for rendering glyphs.
 /// </summary>
-DWRITE_BEGIN_INTERFACE(IDWriteBitmapRenderTarget3, "AEEC37DB-C337-40F1-8E2A-9A41B167B238") : IDWriteBitmapRenderTarget2
+DWRITE_BEGIN_INTERFACE( _WIN32META_ANNOTATION_("win32metadata:agile") IDWriteBitmapRenderTarget3, "AEEC37DB-C337-40F1-8E2A-9A41B167B238") : IDWriteBitmapRenderTarget2
 {
     /// <summary>
     /// Returns the paint feature level supported by this render target.
