diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_1.h b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_1.h
index 727b009a38..b1ade4511d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_1.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_1.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //+--------------------------------------------------------------------------
 //
 //  Copyright (c) Microsoft Corporation.  All rights reserved.
@@ -1008,14 +1009,14 @@ struct DWRITE_JUSTIFICATION_OPPORTUNITY
 };
 
 
-interface IDWriteTextAnalysisSource1;
-interface IDWriteTextAnalysisSink1;
-interface IDWriteRenderingParams1;
+interface _Windows_SupportedOS_Windows8_  IDWriteTextAnalysisSource1;
+interface _Windows_SupportedOS_Windows8_  IDWriteTextAnalysisSink1;
+interface _Windows_SupportedOS_Windows8_  IDWriteRenderingParams1;
 
 /// <summary>
 /// The root factory interface for all DWrite objects.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("30572f99-dac6-41db-a16e-0486307e606a") IDWriteFactory1 : public IDWriteFactory
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("30572f99-dac6-41db-a16e-0486307e606a") IDWriteFactory1 : public IDWriteFactory
 {
     /// <summary>
     /// Gets a font collection representing the set of end-user defined
@@ -1087,7 +1088,7 @@ interface DWRITE_DECLARE_INTERFACE("30572f99-dac6-41db-a16e-0486307e606a") IDWri
 /// It contains font face type, appropriate file references and face identification data.
 /// Various font data such as metrics, names and glyph outlines is obtained from IDWriteFontFace.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("a71efdb4-9fdb-4838-ad90-cfc3be8c3daf") IDWriteFontFace1 : public IDWriteFontFace
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("a71efdb4-9fdb-4838-ad90-cfc3be8c3daf") IDWriteFontFace1 : public IDWriteFontFace
 {
     /// <summary>
     /// Gets common metrics for the font in design units.
@@ -1345,7 +1346,7 @@ interface DWRITE_DECLARE_INTERFACE("a71efdb4-9fdb-4838-ad90-cfc3be8c3daf") IDWri
 /// <summary>
 /// The IDWriteFont interface represents a physical font in a font collection.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("acd16696-8c14-4f5d-877e-fe3fc1d32738") IDWriteFont1 : public IDWriteFont
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("acd16696-8c14-4f5d-877e-fe3fc1d32738") IDWriteFont1 : public IDWriteFont
 {
     /// <summary>
     /// Gets common metrics for the font in design units.
@@ -1419,7 +1420,7 @@ interface DWRITE_DECLARE_INTERFACE("94413cf4-a6fc-4248-8b50-6674348fcad3") IDWri
 /// <summary>
 /// Analyzes various text properties for complex script processing.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("80DAD800-E21F-4E83-96CE-BFCCE500DB7C") IDWriteTextAnalyzer1 : public IDWriteTextAnalyzer
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("80DAD800-E21F-4E83-96CE-BFCCE500DB7C") IDWriteTextAnalyzer1 : public IDWriteTextAnalyzer
 {
     /// <summary>
     /// Applies spacing between characters, properly adjusting glyph clusters
@@ -1799,7 +1800,7 @@ interface DWRITE_DECLARE_INTERFACE("B0D941A0-85E7-4D8B-9FD3-5CED9934482A") IDWri
 ///
 /// All coordinates are in device independent pixels (DIPs).
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("9064D822-80A7-465C-A986-DF65F78B8FEB") IDWriteTextLayout1 : public IDWriteTextLayout
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("9064D822-80A7-465C-A986-DF65F78B8FEB") IDWriteTextLayout1 : public IDWriteTextLayout
 {
     /// <summary>
     /// Enables/disables pair-kerning on the given range.
@@ -1896,7 +1897,7 @@ enum DWRITE_TEXT_ANTIALIAS_MODE
 /// <summary>
 /// Encapsulates a 32-bit device independent bitmap and device context, which can be used for rendering glyphs.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("791e8298-3ef3-4230-9880-c9bdecc42064") IDWriteBitmapRenderTarget1 : public IDWriteBitmapRenderTarget
+interface _Windows_SupportedOS_Windows8_  DWRITE_DECLARE_INTERFACE("791e8298-3ef3-4230-9880-c9bdecc42064") IDWriteBitmapRenderTarget1 : public IDWriteBitmapRenderTarget
 {
     /// <summary>
     /// Gets the current text antialiasing mode of the bitmap render target.
