diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
index 23e5b07171..16f0257a36 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_3.h
@@ -7,6 +7,10 @@
 //
 //----------------------------------------------------------------------------
 
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef DWRITE_3_H_INCLUDED
 #define DWRITE_3_H_INCLUDED
 
@@ -342,7 +346,7 @@ enum DWRITE_RENDERING_MODE1
 /// <summary>
 /// The interface that represents text rendering settings for glyph rasterization and filtering.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") IDWriteRenderingParams3 : public IDWriteRenderingParams2
+interface DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") _Windows_SupportedOS_10240_ IDWriteRenderingParams3 : public IDWriteRenderingParams2
 {
     /// <summary>
     /// Gets the rendering mode.
@@ -354,7 +358,7 @@ interface DWRITE_DECLARE_INTERFACE("B7924BAA-391B-412A-8C5C-E44CC2D867DC") IDWri
 /// <summary>
 /// The root factory interface for all DWrite objects.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") IDWriteFactory3 : public IDWriteFactory2
+interface DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") _Windows_SupportedOS_10240_ IDWriteFactory3 : public IDWriteFactory2
 {
     /// <summary>
     /// Creates a glyph run analysis object, which encapsulates information
@@ -535,7 +539,7 @@ interface DWRITE_DECLARE_INTERFACE("9A1B41C3-D3BB-466A-87FC-FE67556A3B65") IDWri
 /// Unlike IDWriteFontFamily and IDWriteFontList, which are part of the IDWriteFontCollection heirarchy, font sets
 /// are unordered flat lists.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116B") IDWriteFontSet : public IUnknown
+interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116B") _Windows_SupportedOS_10240_ IDWriteFontSet : public IUnknown
 {
     /// <summary>
     /// Get the number of total fonts in the set.
@@ -788,7 +792,7 @@ interface DWRITE_DECLARE_INTERFACE("2F642AFE-9C68-4F40-B8BE-457401AFCB3D") IDWri
 };
 
 
-interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") IDWriteFontCollection1 : public IDWriteFontCollection
+interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") _Windows_SupportedOS_Windows7_ IDWriteFontCollection1 : public IDWriteFontCollection
 {
     /// <summary>
     /// Get the underlying font set used by this collection.
@@ -822,7 +826,7 @@ interface DWRITE_DECLARE_INTERFACE("53585141-D9F8-4095-8321-D73CF6BD116C") IDWri
 /// The IDWriteFontFamily interface represents a set of fonts that share the same design but are differentiated
 /// by weight, stretch, and style.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") IDWriteFontFamily1 : public IDWriteFontFamily
+interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") _Windows_SupportedOS_10240_ IDWriteFontFamily1 : public IDWriteFontFamily
 {
     /// <summary>
     /// Gets the current locality of a font given its zero-based index.
@@ -874,7 +878,7 @@ interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADF") IDWri
 /// <summary>
 /// The IDWriteFontList interface represents a list of fonts.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") IDWriteFontList1 : public IDWriteFontList
+interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") _Windows_SupportedOS_10240_ IDWriteFontList1 : public IDWriteFontList
 {
     /// <summary>
     /// Gets the current locality of a font given its zero-based index.
@@ -929,7 +933,7 @@ interface DWRITE_DECLARE_INTERFACE("DA20D8EF-812A-4C43-9802-62EC4ABD7ADE") IDWri
 /// consists of a font file, font face index, and font face simulation. The file
 /// data may or may not be physically present on the local machine yet.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") IDWriteFontFaceReference : public IUnknown
+interface DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") _Windows_SupportedOS_10240_ IDWriteFontFaceReference : public IUnknown
 {
     /// <summary>
     /// Creates a font face from the reference for use with layout,
@@ -1087,7 +1091,7 @@ interface DWRITE_DECLARE_INTERFACE("5E7FA7CA-DDE3-424C-89F0-9FCD6FED58CD") IDWri
 /// <summary>
 /// The IDWriteFont interface represents a font in a font collection.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") IDWriteFont3 : public IDWriteFont2
+interface DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") _Windows_SupportedOS_10240_ IDWriteFont3 : public IDWriteFont2
 {
     /// <summary>
     /// Creates a font face object for the font.
@@ -1151,7 +1155,7 @@ interface DWRITE_DECLARE_INTERFACE("29748ED6-8C9C-4A6A-BE0B-D912E8538944") IDWri
 /// It contains font face type, appropriate file references and face identification data.
 /// Various font data such as metrics, names and glyph outlines is obtained from IDWriteFontFace.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("D37D7598-09BE-4222-A236-2081341CC1F2") IDWriteFontFace3 : public IDWriteFontFace2
+interface DWRITE_DECLARE_INTERFACE("D37D7598-09BE-4222-A236-2081341CC1F2") _Windows_SupportedOS_10240_ IDWriteFontFace3 : public IDWriteFontFace2
 {
     /// <summary>
     /// Return a font face reference identifying this font.
@@ -1427,7 +1431,7 @@ interface DWRITE_DECLARE_INTERFACE("CFEE3140-1157-47CA-8B85-31BFCF3F2D0E") IDWri
 /// downloading thread, and objects must be prepared to handle calls on their methods
 /// from other threads at any time.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") IDWriteFontDownloadListener : public IUnknown
+interface DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") _Windows_SupportedOS_Windows81_ IDWriteFontDownloadListener : public IUnknown
 {
     /// <summary>
     /// The DownloadCompleted method is called back on an arbitrary thread when a
@@ -1453,7 +1457,7 @@ interface DWRITE_DECLARE_INTERFACE("B06FE5B9-43EC-4393-881B-DBE4DC72FDA7") IDWri
 /// download completion. Callbacks to listeners will occur on the downloading thread, and objects must
 /// be must be able to handle calls on their methods from other threads at any time.
 /// </summary>
-interface DWRITE_DECLARE_INTERFACE("B71E6052-5AEA-4FA3-832E-F60D431F7E91") IDWriteFontDownloadQueue : public IUnknown
+interface DWRITE_DECLARE_INTERFACE("B71E6052-5AEA-4FA3-832E-F60D431F7E91") _Windows_SupportedOS_Windows81_ IDWriteFontDownloadQueue : public IUnknown
 {
     /// <summary>
     /// Registers a client-defined listener object that receives download notifications.
@@ -1702,7 +1706,7 @@ struct DWRITE_LINE_SPACING
     DWRITE_FONT_LINE_GAP_USAGE fontLineGapUsage;
 };
 
-interface DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") IDWriteTextFormat2 : public IDWriteTextFormat1
+interface DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") _Windows_SupportedOS_Windows81_ IDWriteTextFormat2 : public IDWriteTextFormat1
 {
     /// <summary>
     /// Set line spacing.
@@ -1726,7 +1730,7 @@ interface DWRITE_DECLARE_INTERFACE("F67E0EDD-9E3D-4ECC-8C32-4183253DFE70") IDWri
     using IDWriteTextFormat1::GetLineSpacing;
 };
 
-interface DWRITE_DECLARE_INTERFACE("07DDCD52-020E-4DE8-AC33-6C953D83F92D") IDWriteTextLayout3 : public IDWriteTextLayout2
+interface DWRITE_DECLARE_INTERFACE("07DDCD52-020E-4DE8-AC33-6C953D83F92D") _Windows_SupportedOS_Windows81_ IDWriteTextLayout3 : public IDWriteTextLayout2
 {
     /// <summary>
     /// Invalidates the layout, forcing layout to remeasure before calling the
