diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_2.h b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_2.h
index 9fb86ff4698f7cad406ef79b7255390bf1f697f6..845c0b45cb52bedeeff8163529dff0e01b526576 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/dwrite_2.h
@@ -9,6 +9,7 @@
 
 #ifndef DWRITE_2_H_INCLUDED
 #define DWRITE_2_H_INCLUDED
+#include "win32metadata_annotations.h"
 
 #pragma once
 
@@ -41,7 +42,7 @@ enum DWRITE_OPTICAL_ALIGNMENT
 /// <summary>
 /// Whether to enable grid-fitting of glyph outlines (a.k.a. hinting).
 /// </summary>
-enum DWRITE_GRID_FIT_MODE
+enum _Windows_SupportedOS_Windows81_ DWRITE_GRID_FIT_MODE
 {
     /// <summary>
     /// Choose grid fitting base on the font's gasp table information.
@@ -770,7 +771,7 @@ interface DWRITE_DECLARE_INTERFACE("d8b768ff-64bc-4e66-982b-ec8e87f693f7") IDWri
 /// layered on top of each other to produce a color representation of the
 /// given base glyph run.
 /// </summary>
-struct DWRITE_COLOR_GLYPH_RUN
+struct _Windows_SupportedOS_Windows81_ DWRITE_COLOR_GLYPH_RUN
 {
     /// <summary>
     /// Glyph run to render.
