diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/dxgi1_4.h b/generation/WinSDK/RecompiledIdlHeaders/shared/dxgi1_4.h
index e8534ebe..955e1ae2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/dxgi1_4.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/dxgi1_4.h
@@ -9,8 +9,13 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
+
 #endif
 
 /* verify that the <rpcsal.h> version is high enough to compile this file*/
@@ -112,6 +117,7 @@ EXTERN_C const IID IID_IDXGISwapChain3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_10240_
     MIDL_INTERFACE("94d99bdb-f1f8-4ab0-b236-7da0170edab1")
     IDXGISwapChain3 : public IDXGISwapChain2
     {
@@ -612,6 +618,7 @@ EXTERN_C const IID IID_IDXGIOutput4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_10240_
     MIDL_INTERFACE("dc7dca35-2196-414d-9F53-617884032a60")
     IDXGIOutput4 : public IDXGIOutput3
     {
