diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/encdec.h b/generation/WinSDK/RecompiledIdlHeaders/um/encdec.h
index 5685043d83..d901db0619 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/encdec.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/encdec.h
@@ -213,6 +213,7 @@ extern "C"{
 //--------------------------------------------------------------------------
 #pragma once
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 #pragma warning(push)
@@ -639,7 +640,7 @@ EXTERN_C const IID IID_IETFilterConfig;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4D1-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4D1-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IETFilterConfig : public IUnknown
     {
     public:
@@ -739,7 +740,7 @@ EXTERN_C const IID IID_IDTFilterConfig;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4D2-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4D2-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IDTFilterConfig : public IUnknown
     {
     public:
@@ -1222,7 +1223,7 @@ EXTERN_C const IID IID_IETFilter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4B1-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4B1-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IETFilter : public IUnknown
     {
     public:
@@ -1506,7 +1507,7 @@ EXTERN_C const IID IID_IDTFilter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4B2-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4B2-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IDTFilter : public IUnknown
     {
     public:
@@ -1708,13 +1709,13 @@ EXTERN_C const IID IID_IDTFilter2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4B4-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4B4-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IDTFilter2 : public IDTFilter
     {
     public:
         virtual /* [helpstring][id][propget] */ HRESULT STDMETHODCALLTYPE get_ChallengeUrl( 
             /* [annotation][out] */ 
-            _Out_  BSTR *pbstrChallengeUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *pbstrChallengeUrl) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE GetCurrLicenseExpDate( 
             /* [annotation][in] */ 
@@ -1907,7 +1908,7 @@ EXTERN_C const IID IID_IDTFilter3;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("513998cc-e929-4cdf-9fbd-bad1e0314866")
-    IDTFilter3 : public IDTFilter2
+    _Windows_SupportedOS_WindowsVista_ IDTFilter3 : public IDTFilter2
     {
     public:
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE GetProtectionType( 
@@ -2046,7 +2047,7 @@ EXTERN_C const IID IID_IDTFilter3;
         END_INTERFACE
     } IDTFilter3Vtbl;
 
-    interface IDTFilter3
+    interface _Windows_SupportedOS_WindowsVista_ IDTFilter3
     {
         CONST_VTBL struct IDTFilter3Vtbl *lpVtbl;
     };
@@ -2271,7 +2272,7 @@ EXTERN_C const IID IID_IXDSCodec;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C4C4C4B3-0049-4E2B-98FB-9537F6CE516D")
+    MIDL_INTERFACE("C4C4C4B3-0049-4E2B-98FB-9537F6CE516D") _Windows_SupportedOS_WindowsXP_ 
     IXDSCodec : public IUnknown
     {
     public:
@@ -2305,7 +2306,7 @@ EXTERN_C const IID IID_IXDSCodec;
             /* [annotation][out] */ 
             _Out_  long *pXDSTypePkt,
             /* [annotation][out] */ 
-            _Out_  BSTR *pBstrXDSPkt,
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *pBstrXDSPkt,
             /* [annotation][out] */ 
             _Out_  long *pPktSeqID,
             /* [annotation][out] */ 
