diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ErrorRep.h b/generation/WinSDK/RecompiledIdlHeaders/um/ErrorRep.h
index b3b706b1cc..babc134dc9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ErrorRep.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ErrorRep.h
@@ -18,6 +18,7 @@ extern "C" {
 #endif
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -38,9 +39,15 @@ typedef enum tagEFaultRepRetVal
     frrvErrDoubleFault,
 } EFaultRepRetVal;
 
+_Windows_SupportedOS_WindowsXP_ 
+
 EFaultRepRetVal APIENTRY ReportFault(_In_ LPEXCEPTION_POINTERS pep, _In_ DWORD dwOpt);
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ 
+
 BOOL            APIENTRY AddERExcludedApplicationA(_In_ LPCSTR szApplication);
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ 
+
 BOOL            APIENTRY AddERExcludedApplicationW(_In_ LPCWSTR wszApplication);
 
 
@@ -59,7 +66,7 @@ typedef EFaultRepRetVal (APIENTRY *pfn_ADDEREXCLUDEDAPPLICATIONW)(_In_ LPCWSTR);
 // 
 //+++++++++++++++++++++ Hang reporting api ++++++++++++++++++++++++++++++++
 //
-HRESULT 
+_Windows_SupportedOS_WindowsVista_ HRESULT 
 WINAPI
 WerReportHang(
     _In_ HWND hwndHungApp,
