diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/esent.h b/generation/WinSDK/RecompiledIdlHeaders/um/esent.h
index 56a8a639db..bc45341777 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/esent.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/esent.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++ BUILD Version: 0000     Increment this if a change has global effects
 
 Copyright (c) Microsoft Corporation. All rights reserved.
@@ -117,9 +118,9 @@ typedef wchar_t            JET_WCHAR;
 
 
 #if defined(_WIN64)
-    typedef JET_UINT64 JET_API_PTR;
+    _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_UINT64 JET_API_PTR;
 #else
-    typedef JET_UINT32 JET_API_PTR;
+    _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_UINT32 JET_API_PTR;
 #endif
 
 typedef _Return_type_success_( return >= 0 ) JET_INT32 JET_ERR;
@@ -129,12 +130,12 @@ typedef JET_UINT16         JET_LANGID;
 typedef JET_UINT16         JET_CP;
 
 
-typedef JET_API_PTR JET_HANDLE;     // backup file handle
-typedef JET_API_PTR JET_INSTANCE;   // Instance Identifier
-typedef JET_API_PTR JET_SESID;      // Session Identifier
-typedef JET_API_PTR JET_TABLEID;    // Table Identifier
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_HANDLE;     // backup file handle
+_WIN32META_ANNOTATION_("win32metadata:raii_free=JetTerm") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_INSTANCE;   // Instance Identifier
+_WIN32META_ANNOTATION_("win32metadata:raii_free=JetEndSession") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_SESID;      // Session Identifier
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_TABLEID;    // Table Identifier
 #if ( JET_VERSION >= 0x0501 )
-typedef JET_API_PTR JET_LS;         // Local Storage
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_LS;         // Local Storage
 #endif // JET_VERSION >= 0x0501
 
 typedef JET_UINT32 JET_COLUMNID;    // Column Identifier
@@ -3720,7 +3721,7 @@ JetInit2(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetInit3A(
     _Inout_opt_ JET_INSTANCE *  pinstance,
     _In_opt_ JET_RSTINFO_A *    prstInfo,
@@ -3734,7 +3735,7 @@ JetInit3A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetInit3W(
     _Inout_opt_ JET_INSTANCE *  pinstance,
     _In_opt_ JET_RSTINFO_W *    prstInfo,
@@ -3762,9 +3763,9 @@ JetInit3W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateInstanceA(
-    _Out_ JET_INSTANCE *    pinstance,
+    _Win32_RAIIFree_(JetTerm, 0) _Out_ JET_INSTANCE *    pinstance,
     _In_opt_ JET_PCSTR      szInstanceName );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT) */
@@ -3775,9 +3776,9 @@ JetCreateInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateInstanceW(
-    _Out_ JET_INSTANCE *    pinstance,
+    _Win32_RAIIFree_(JetTerm, 0) _Out_ JET_INSTANCE *    pinstance,
     _In_opt_ JET_PCWSTR     szInstanceName );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT) */
@@ -3797,9 +3798,9 @@ JetCreateInstanceW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateInstance2A(
-    _Out_ JET_INSTANCE *    pinstance,
+    _Win32_RAIIFree_(JetTerm, 0) _Out_ JET_INSTANCE *    pinstance,
     _In_opt_ JET_PCSTR      szInstanceName,
     _In_opt_ JET_PCSTR      szDisplayName,
     _In_ JET_GRBIT          grbit );
@@ -3812,9 +3813,9 @@ JetCreateInstance2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateInstance2W(
-    _Out_ JET_INSTANCE *    pinstance,
+    _Win32_RAIIFree_(JetTerm, 0) _Out_ JET_INSTANCE *    pinstance,
     _In_opt_ JET_PCWSTR     szInstanceName,
     _In_opt_ JET_PCWSTR     szDisplayName,
     _In_ JET_GRBIT          grbit );
@@ -3937,7 +3938,7 @@ JetStopBackupInstance(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetSystemParameterA(
     _Inout_opt_ JET_INSTANCE *  pinstance,
     _In_opt_ JET_SESID          sesid,
@@ -3953,7 +3954,7 @@ JetSetSystemParameterA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetSystemParameterW(
     _Inout_opt_ JET_INSTANCE *  pinstance,
     _In_opt_ JET_SESID          sesid,
@@ -3978,7 +3979,7 @@ JetSetSystemParameterW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetSystemParameterA(
     _In_ JET_INSTANCE                   instance,
     _In_opt_ JET_SESID                  sesid,
@@ -3995,7 +3996,7 @@ JetGetSystemParameterA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetSystemParameterW(
     _In_ JET_INSTANCE                   instance,
     _In_opt_ JET_SESID                  sesid,
@@ -4024,7 +4025,7 @@ JetGetSystemParameterW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetEnableMultiInstanceA(
     _In_reads_opt_( csetsysparam ) JET_SETSYSPARAM_A *  psetsysparam,
     _In_ JET_UINT32                                     csetsysparam,
@@ -4038,7 +4039,7 @@ JetEnableMultiInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetEnableMultiInstanceW(
     _In_reads_opt_( csetsysparam ) JET_SETSYSPARAM_W *  psetsysparam,
     _In_ JET_UINT32                                     csetsysparam,
@@ -4079,9 +4080,9 @@ JetGetThreadStats(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetBeginSessionA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API JetBeginSessionA(
     _In_ JET_INSTANCE   instance,
-    _Out_ JET_SESID *   psesid,
+    _Win32_RAIIFree_(JetEndSession, 0) _Out_ JET_SESID *   psesid,
     _In_opt_ JET_PCSTR  szUserName,
     _In_opt_ JET_PCSTR  szPassword );
 
@@ -4093,9 +4094,9 @@ JET_ERR JET_API JetBeginSessionA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetBeginSessionW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetBeginSessionW(
     _In_ JET_INSTANCE   instance,
-    _Out_ JET_SESID *   psesid,
+    _Win32_RAIIFree_(JetEndSession, 0) _Out_ JET_SESID *   psesid,
     _In_opt_ JET_PCWSTR szUserName,
     _In_opt_ JET_PCWSTR szPassword );
 
@@ -4115,7 +4116,7 @@ JET_ERR JET_API JetBeginSessionW(
 JET_ERR JET_API
 JetDupSession(
     _In_ JET_SESID      sesid,
-    _Out_ JET_SESID *   psesid );
+    _Win32_RAIIFree_(JetEndSession, 0) _Out_ JET_SESID *   psesid );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT) */
 #pragma endregion
@@ -4156,7 +4157,7 @@ JetIdle(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateDatabaseA(
     _In_ JET_SESID      sesid,
     _In_ JET_PCSTR      szFilename,
@@ -4172,7 +4173,7 @@ JetCreateDatabaseA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateDatabaseW(
     _In_ JET_SESID      sesid,
     _In_ JET_PCWSTR     szFilename,
@@ -4198,7 +4199,7 @@ JetCreateDatabaseW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateDatabase2A(
     _In_ JET_SESID              sesid,
     _In_ JET_PCSTR              szFilename,
@@ -4214,7 +4215,7 @@ JetCreateDatabase2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetCreateDatabase2W(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetCreateDatabase2W(
     _In_ JET_SESID              sesid,
     _In_ JET_PCWSTR             szFilename,
     _In_ const JET_UINT32       cpgDatabaseSizeMax,
@@ -4239,7 +4240,7 @@ JET_ERR JET_API JetCreateDatabase2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetAttachDatabaseA(
     _In_ JET_SESID  sesid,
     _In_ JET_PCSTR  szFilename,
@@ -4253,7 +4254,7 @@ JetAttachDatabaseA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetAttachDatabaseW(
     _In_ JET_SESID  sesid,
     _In_ JET_PCWSTR szFilename,
@@ -4276,7 +4277,7 @@ JetAttachDatabaseW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetAttachDatabase2A(
     _In_ JET_SESID              sesid,
     _In_ JET_PCSTR              szFilename,
@@ -4291,7 +4292,7 @@ JetAttachDatabase2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetAttachDatabase2W(
     _In_ JET_SESID              sesid,
     _In_ JET_PCWSTR             szFilename,
@@ -4316,7 +4317,7 @@ JetAttachDatabase2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDetachDatabaseA(
     _In_ JET_SESID  sesid,
     _In_opt_ JET_PCSTR  szFilename );
@@ -4329,7 +4330,7 @@ JetDetachDatabaseA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDetachDatabaseW(
     _In_ JET_SESID  sesid,
     _In_opt_ JET_PCWSTR szFilename );
@@ -4352,7 +4353,7 @@ JetDetachDatabaseW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDetachDatabase2A(
     _In_ JET_SESID  sesid,
     _In_opt_ JET_PCSTR  szFilename,
@@ -4366,7 +4367,7 @@ JetDetachDatabase2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDetachDatabase2W(
     _In_ JET_SESID  sesid,
     _In_opt_ JET_PCWSTR szFilename,
@@ -4392,7 +4393,7 @@ JetDetachDatabase2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetObjectInfoA(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -4411,7 +4412,7 @@ JetGetObjectInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetObjectInfoW(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -4440,7 +4441,7 @@ JetGetObjectInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetTableInfoA(
     _In_ JET_SESID                        sesid,
     _In_ JET_TABLEID                      tableid,
@@ -4456,7 +4457,7 @@ JetGetTableInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetTableInfoW(
     _In_ JET_SESID                        sesid,
     _In_ JET_TABLEID                      tableid,
@@ -4482,7 +4483,7 @@ JetGetTableInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateTableA(
     _In_ JET_SESID       sesid,
     _In_ JET_DBID        dbid,
@@ -4499,7 +4500,7 @@ JetCreateTableA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateTableW(
     _In_ JET_SESID       sesid,
     _In_ JET_DBID        dbid,
@@ -4525,7 +4526,7 @@ JetCreateTableW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateTableColumnIndexA(
     _In_ JET_SESID               sesid,
     _In_ JET_DBID                dbid,
@@ -4539,7 +4540,7 @@ JetCreateTableColumnIndexA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateTableColumnIndexW(
     _In_ JET_SESID               sesid,
     _In_ JET_DBID                dbid,
@@ -4564,7 +4565,7 @@ JetCreateTableColumnIndexW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateTableColumnIndex2A(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
@@ -4578,7 +4579,7 @@ JetCreateTableColumnIndex2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateTableColumnIndex2W(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
@@ -4600,13 +4601,13 @@ JetCreateTableColumnIndex2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateTableColumnIndex3A(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
     _Inout_ JET_TABLECREATE3_A *    ptablecreate );
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateTableColumnIndex3W(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
@@ -4627,7 +4628,7 @@ JetCreateTableColumnIndex3W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateTableColumnIndex4A(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
@@ -4639,7 +4640,7 @@ JetCreateTableColumnIndex4A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateTableColumnIndex4W(
     _In_ JET_SESID                  sesid,
     _In_ JET_DBID                   dbid,
@@ -4663,7 +4664,7 @@ JetCreateTableColumnIndex4W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDeleteTableA(
     _In_ JET_SESID  sesid,
     _In_ JET_DBID   dbid,
@@ -4677,7 +4678,7 @@ JetDeleteTableA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDeleteTableW(
     _In_ JET_SESID  sesid,
     _In_ JET_DBID   dbid,
@@ -4699,7 +4700,7 @@ JetDeleteTableW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetRenameTableA(
     _In_ JET_SESID  sesid,
     _In_ JET_DBID   dbid,
@@ -4714,7 +4715,7 @@ JetRenameTableA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetRenameTableW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetRenameTableW(
     _In_ JET_SESID  sesid,
     _In_ JET_DBID   dbid,
     _In_ JET_PCWSTR szName,
@@ -4738,7 +4739,7 @@ JET_ERR JET_API JetRenameTableW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetTableColumnInfoA(
     _In_ JET_SESID                        sesid,
     _In_ JET_TABLEID                      tableid,
@@ -4755,7 +4756,7 @@ JetGetTableColumnInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetGetTableColumnInfoW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetGetTableColumnInfoW(
     _In_ JET_SESID                        sesid,
     _In_ JET_TABLEID                      tableid,
     _In_opt_ JET_PCWSTR                   szColumnName,
@@ -4781,7 +4782,7 @@ JET_ERR JET_API JetGetTableColumnInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetColumnInfoA(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -4799,7 +4800,7 @@ JetGetColumnInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetGetColumnInfoW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetGetColumnInfoW(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
     _In_ JET_PCWSTR                       szTableName,
@@ -4826,7 +4827,7 @@ JET_ERR JET_API JetGetColumnInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetAddColumnA(
     _In_ JET_SESID                               sesid,
     _In_ JET_TABLEID                             tableid,
@@ -4844,7 +4845,7 @@ JetAddColumnA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetAddColumnW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API JetAddColumnW(
     _In_ JET_SESID                               sesid,
     _In_ JET_TABLEID                             tableid,
     _In_ JET_PCWSTR                              szColumnName,
@@ -4871,7 +4872,7 @@ JET_ERR JET_API JetAddColumnW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDeleteColumnA(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -4885,7 +4886,7 @@ JetDeleteColumnA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDeleteColumnW(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -4910,7 +4911,7 @@ JetDeleteColumnW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDeleteColumn2A(
     _In_ JET_SESID          sesid,
     _In_ JET_TABLEID        tableid,
@@ -4925,7 +4926,7 @@ JetDeleteColumn2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDeleteColumn2W(
     _In_ JET_SESID          sesid,
     _In_ JET_TABLEID        tableid,
@@ -4950,7 +4951,7 @@ JetDeleteColumn2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetRenameColumnA(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -4966,7 +4967,7 @@ JetRenameColumnA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetRenameColumnW(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -4996,7 +4997,7 @@ JetRenameColumnW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetColumnDefaultValueA(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -5014,7 +5015,7 @@ JetSetColumnDefaultValueA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetColumnDefaultValueW(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -5042,7 +5043,7 @@ JetSetColumnDefaultValueW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetTableIndexInfoA(
     _In_ JET_SESID                           sesid,
     _In_ JET_TABLEID                         tableid,
@@ -5059,7 +5060,7 @@ JetGetTableIndexInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetTableIndexInfoW(
     _In_ JET_SESID                           sesid,
     _In_ JET_TABLEID                         tableid,
@@ -5086,7 +5087,7 @@ JetGetTableIndexInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetIndexInfoA(
     _In_ JET_SESID                           sesid,
     _In_ JET_DBID                            dbid,
@@ -5104,7 +5105,7 @@ JetGetIndexInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetIndexInfoW(
     _In_ JET_SESID                           sesid,
     _In_ JET_DBID                            dbid,
@@ -5132,7 +5133,7 @@ JetGetIndexInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateIndexA(
     _In_ JET_SESID                      sesid,
     _In_ JET_TABLEID                    tableid,
@@ -5150,7 +5151,7 @@ JetCreateIndexA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateIndexW(
     _In_ JET_SESID                       sesid,
     _In_ JET_TABLEID                     tableid,
@@ -5178,7 +5179,7 @@ JetCreateIndexW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateIndex2A(
     _In_ JET_SESID                                  sesid,
     _In_ JET_TABLEID                                tableid,
@@ -5193,7 +5194,7 @@ JetCreateIndex2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateIndex2W(
     _In_ JET_SESID                                  sesid,
     _In_ JET_TABLEID                                tableid,
@@ -5215,14 +5216,14 @@ JetCreateIndex2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateIndex3A(
     _In_ JET_SESID                                   sesid,
     _In_ JET_TABLEID                                 tableid,
     _In_reads_( cIndexCreate ) JET_INDEXCREATE2_A *  pindexcreate,
     _In_ JET_UINT32                                  cIndexCreate );
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateIndex3W(
     _In_ JET_SESID                                   sesid,
     _In_ JET_TABLEID                                 tableid,
@@ -5245,7 +5246,7 @@ JetCreateIndex3W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCreateIndex4A(
     _In_ JET_SESID                                   sesid,
     _In_ JET_TABLEID                                 tableid,
@@ -5258,7 +5259,7 @@ JetCreateIndex4A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCreateIndex4W(
     _In_ JET_SESID                                   sesid,
     _In_ JET_TABLEID                                 tableid,
@@ -5283,7 +5284,7 @@ JetCreateIndex4W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDeleteIndexA(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5297,7 +5298,7 @@ JetDeleteIndexA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDeleteIndexW(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5377,7 +5378,7 @@ JetRollback(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API JetGetDatabaseInfoA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API JetGetDatabaseInfoA(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
     _Out_writes_bytes_( cbMax ) JET_PVOID pvResult,
@@ -5392,7 +5393,7 @@ JET_ERR JET_API JetGetDatabaseInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetDatabaseInfoW(
     _In_ JET_SESID                        sesid,
     _In_ JET_DBID                         dbid,
@@ -5418,7 +5419,7 @@ JetGetDatabaseInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetDatabaseFileInfoA(
     _In_ JET_PCSTR                        szDatabaseName,
     _Out_writes_bytes_( cbMax ) JET_PVOID pvResult,
@@ -5433,7 +5434,7 @@ JetGetDatabaseFileInfoA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetDatabaseFileInfoW(
     _In_ JET_PCWSTR                       szDatabaseName,
     _Out_writes_bytes_( cbMax ) JET_PVOID pvResult,
@@ -5458,7 +5459,7 @@ JetGetDatabaseFileInfoW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOpenDatabaseA(
     _In_ JET_SESID      sesid,
     _In_ JET_PCSTR      szFilename,
@@ -5474,7 +5475,7 @@ JetOpenDatabaseA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOpenDatabaseW(
     _In_ JET_SESID      sesid,
     _In_ JET_PCWSTR     szFilename,
@@ -5511,7 +5512,7 @@ JetCloseDatabase(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOpenTableA(
     _In_ JET_SESID                                  sesid,
     _In_ JET_DBID                                   dbid,
@@ -5529,7 +5530,7 @@ JetOpenTableA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOpenTableW(
     _In_ JET_SESID                                  sesid,
     _In_ JET_DBID                                   dbid,
@@ -5779,7 +5780,7 @@ JetDupCursor(
 #define JetGetCurrentIndexA JetGetCurrentIndex
 #endif
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetCurrentIndexA(
     _In_ JET_SESID                          sesid,
     _In_ JET_TABLEID                        tableid,
@@ -5794,7 +5795,7 @@ JetGetCurrentIndexA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetCurrentIndexW(
     _In_ JET_SESID                          sesid,
     _In_ JET_TABLEID                        tableid,
@@ -5819,7 +5820,7 @@ JetGetCurrentIndexW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetCurrentIndexA(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5833,7 +5834,7 @@ JetSetCurrentIndexA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetCurrentIndexW(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5856,7 +5857,7 @@ JetSetCurrentIndexW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetCurrentIndex2A(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5871,7 +5872,7 @@ JetSetCurrentIndex2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetCurrentIndex2W(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5896,7 +5897,7 @@ JetSetCurrentIndex2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetCurrentIndex3A(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5912,7 +5913,7 @@ JetSetCurrentIndex3A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetCurrentIndex3W(
     _In_ JET_SESID      sesid,
     _In_ JET_TABLEID    tableid,
@@ -5938,7 +5939,7 @@ JetSetCurrentIndex3W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetCurrentIndex4A(
     _In_ JET_SESID          sesid,
     _In_ JET_TABLEID        tableid,
@@ -5955,7 +5956,7 @@ JetSetCurrentIndex4A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetCurrentIndex4W(
     _In_ JET_SESID          sesid,
     _In_ JET_TABLEID        tableid,
@@ -6110,7 +6111,7 @@ JetGetSecondaryIndexBookmark(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetCompactA(
     _In_ JET_SESID              sesid,
     _In_ JET_PCSTR              szDatabaseSrc,
@@ -6127,7 +6128,7 @@ JetCompactA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetCompactW(
     _In_ JET_SESID              sesid,
     _In_ JET_PCWSTR             szDatabaseSrc,
@@ -6154,7 +6155,7 @@ JetCompactW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDefragmentA(
     _In_ JET_SESID              sesid,
     _In_ JET_DBID               dbid,
@@ -6171,7 +6172,7 @@ JetDefragmentA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDefragmentW(
     _In_ JET_SESID              sesid,
     _In_ JET_DBID               dbid,
@@ -6199,7 +6200,7 @@ JetDefragmentW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDefragment2A(
     _In_ JET_SESID              sesid,
     _In_ JET_DBID               dbid,
@@ -6217,7 +6218,7 @@ JetDefragment2A(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDefragment2W(
     _In_ JET_SESID              sesid,
     _In_ JET_DBID               dbid,
@@ -6245,7 +6246,7 @@ JetDefragment2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetDefragment3A(
     _In_ JET_SESID              sesid,
     _In_ JET_PCSTR              szDatabaseName,
@@ -6264,7 +6265,7 @@ JetDefragment3A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetDefragment3W(
     _In_ JET_SESID              sesid,
     _In_ JET_PCWSTR             szDatabaseName,
@@ -6294,7 +6295,7 @@ JetDefragment3W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetSetDatabaseSizeA(
     _In_ JET_SESID          sesid,
     _In_ JET_PCSTR          szDatabaseName,
@@ -6309,7 +6310,7 @@ JetSetDatabaseSizeA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetSetDatabaseSizeW(
     _In_ JET_SESID          sesid,
     _In_ JET_PCWSTR         szDatabaseName,
@@ -6497,7 +6498,7 @@ JetOpenTemporaryTable2(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetBackupA(
     _In_ JET_PCSTR      szBackupPath,
     _In_ JET_GRBIT      grbit,
@@ -6511,7 +6512,7 @@ JetBackupA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetBackupW(
     _In_ JET_PCWSTR     szBackupPath,
     _In_ JET_GRBIT      grbit,
@@ -6535,7 +6536,7 @@ JetBackupW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetBackupInstanceA(
     _In_ JET_INSTANCE   instance,
     _In_ JET_PCSTR      szBackupPath,
@@ -6550,7 +6551,7 @@ JetBackupInstanceA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetBackupInstanceW(
     _In_ JET_INSTANCE   instance,
     _In_ JET_PCWSTR     szBackupPath,
@@ -6577,7 +6578,7 @@ JetBackupInstanceW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetRestoreA(
     _In_ JET_PCSTR      szSource,
     _In_opt_ JET_PFNSTATUS  pfn );
@@ -6590,7 +6591,7 @@ JetRestoreA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetRestoreW(
     _In_ JET_PCWSTR     szSource,
     _In_opt_ JET_PFNSTATUS  pfn );
@@ -6613,7 +6614,7 @@ JetRestoreW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetRestore2A(
     _In_ JET_PCSTR      sz,
     _In_opt_ JET_PCSTR  szDest,
@@ -6627,7 +6628,7 @@ JetRestore2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetRestore2W(
     _In_ JET_PCWSTR     sz,
     _In_opt_ JET_PCWSTR szDest,
@@ -6652,7 +6653,7 @@ JetRestore2W(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetRestoreInstanceA(
     _In_ JET_INSTANCE   instance,
     _In_ JET_PCSTR      sz,
@@ -6667,7 +6668,7 @@ JetRestoreInstanceA(
 #pragma region Application Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetRestoreInstanceW(
     _In_ JET_INSTANCE   instance,
     _In_ JET_PCWSTR     sz,
@@ -6756,7 +6757,7 @@ JET_ERR JET_API JetBeginExternalBackupInstance(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetAttachInfoA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetAttachInfoA(
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
 #else
@@ -6773,7 +6774,7 @@ JetGetAttachInfoA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetAttachInfoW(
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR    wszzDatabases,
     _In_ JET_UINT32                                         cbMax,
@@ -6799,7 +6800,7 @@ JetGetAttachInfoW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetAttachInfoInstanceA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetAttachInfoInstanceA(
     _In_ JET_INSTANCE                                        instance,
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
@@ -6817,7 +6818,7 @@ JetGetAttachInfoInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetAttachInfoInstanceW(
     _In_ JET_INSTANCE                                       instance,
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR    szzDatabases,
@@ -6844,7 +6845,7 @@ JetGetAttachInfoInstanceW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOpenFileA(
     _In_ JET_PCSTR          szFileName,
     _Out_ JET_HANDLE *      phfFile,
@@ -6859,7 +6860,7 @@ JetOpenFileA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOpenFileW(
     _In_ JET_PCWSTR         szFileName,
     _Out_ JET_HANDLE *      phfFile,
@@ -6885,7 +6886,7 @@ JetOpenFileW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOpenFileInstanceA(
     _In_ JET_INSTANCE       instance,
     _In_ JET_PCSTR          szFileName,
@@ -6901,7 +6902,7 @@ JetOpenFileInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOpenFileInstanceW(
     _In_ JET_INSTANCE       instance,
     _In_ JET_PCWSTR         szFileName,
@@ -6986,7 +6987,7 @@ JetCloseFileInstance(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetLogInfoA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetLogInfoA(
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
 #else
@@ -7003,7 +7004,7 @@ JetGetLogInfoA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetLogInfoW(
         _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR    szzLogs,
         _In_ JET_UINT32                                         cbMax,
@@ -7029,7 +7030,7 @@ JetGetLogInfoW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetLogInfoInstanceA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetLogInfoInstanceA(
     _In_ JET_INSTANCE                                        instance,
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
@@ -7047,7 +7048,7 @@ JetGetLogInfoInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetLogInfoInstanceW(
     _In_ JET_INSTANCE                                         instance,
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR  wszzLogs,
@@ -7066,7 +7067,7 @@ JetGetLogInfoInstanceW(
 
 
 #define JET_BASE_NAME_LENGTH    3
-typedef struct
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize") 
 {
     JET_UINT32      cbSize;
     JET_UINT32      ulGenLow;
@@ -7074,7 +7075,7 @@ typedef struct
     JET_CHAR        szBaseName[ JET_BASE_NAME_LENGTH + 1 ];
 } JET_LOGINFO_A;
 
-typedef struct
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize") 
 {
     JET_UINT32      cbSize;
     JET_UINT32      ulGenLow;
@@ -7096,7 +7097,7 @@ typedef struct
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetLogInfoInstance2A(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetLogInfoInstance2A(
     _In_ JET_INSTANCE                                        instance,
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
@@ -7115,7 +7116,7 @@ JetGetLogInfoInstance2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetLogInfoInstance2W(
     _In_ JET_INSTANCE                                         instance,
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR  wszzLogs,
@@ -7142,7 +7143,7 @@ JetGetLogInfoInstance2W(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
 JET_ERR JET_API
-JetGetTruncateLogInfoInstanceA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") JetGetTruncateLogInfoInstanceA(
     _In_ JET_INSTANCE                                        instance,
 #if ( JET_VERSION < 0x0600 )
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PVOID pv,
@@ -7160,7 +7161,7 @@ JetGetTruncateLogInfoInstanceA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetTruncateLogInfoInstanceW(
     _In_ JET_INSTANCE                                         instance,
     _Out_writes_bytes_to_opt_( cbMax, *pcbActual ) JET_PWSTR  wszzLogs,
@@ -7237,7 +7238,7 @@ JetEndExternalBackupInstance2(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetExternalRestoreA(
     _In_ JET_PSTR                                   szCheckpointFilePath,
     _In_ JET_PSTR                                   szLogPath,
@@ -7256,7 +7257,7 @@ JetExternalRestoreA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetExternalRestoreW(
     _In_ JET_PWSTR                                  szCheckpointFilePath,
     _In_ JET_PWSTR                                  szLogPath,
@@ -7286,7 +7287,7 @@ JetExternalRestoreW(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetExternalRestore2A(
     _In_ JET_PSTR                                   szCheckpointFilePath,
     _In_ JET_PSTR                                   szLogPath,
@@ -7307,7 +7308,7 @@ JetExternalRestore2A(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetExternalRestore2W(
     _In_ JET_PWSTR                                  szCheckpointFilePath,
     _In_ JET_PWSTR                                  szLogPath,
@@ -7388,7 +7389,7 @@ typedef struct _JET_INSTANCE_INFO_W
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetGetInstanceInfoA(
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
     _Outptr_result_buffer_( *pcInstanceInfo ) JET_INSTANCE_INFO_A **  paInstanceInfo );
@@ -7401,7 +7402,7 @@ JetGetInstanceInfoA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetGetInstanceInfoW(
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
     _Outptr_result_buffer_( *pcInstanceInfo ) JET_INSTANCE_INFO_W **  paInstanceInfo );
@@ -7421,7 +7422,7 @@ JetGetInstanceInfoW(
 
 JET_ERR JET_API
 JetFreeBuffer(
-    _Pre_notnull_ JET_CHAR *  pbBuf );
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") _Pre_notnull_ JET_CHAR *  pbBuf );
 
 JET_ERR JET_API
 JetSetLS(
@@ -7444,7 +7445,7 @@ JetGetLS(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-typedef JET_API_PTR JET_OSSNAPID;   // Snapshot Session Identifier
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef JET_API_PTR JET_OSSNAPID;   // Snapshot Session Identifier
 
 JET_ERR JET_API
 JetOSSnapshotPrepare(
@@ -7463,7 +7464,7 @@ JetOSSnapshotPrepareInstance(
 #define JetOSSnapshotFreezeA JetOSSnapshotFreeze
 #endif
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOSSnapshotFreezeA(
     _In_ const JET_OSSNAPID                                           snapId,
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
@@ -7472,7 +7473,7 @@ JetOSSnapshotFreezeA(
 
 #if ( JET_VERSION >= 0x0600 )
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOSSnapshotFreezeW(
     _In_ const JET_OSSNAPID                                           snapId,
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
@@ -7531,7 +7532,7 @@ JetOSSnapshotTruncateLogInstance(
 #define JetOSSnapshotGetFreezeInfoA JetOSSnapshotGetFreezeInfo
 #endif
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:ansi") JET_ERR JET_API
 JetOSSnapshotGetFreezeInfoA(
     _In_ const JET_OSSNAPID                                           snapId,
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
@@ -7544,7 +7545,7 @@ JetOSSnapshotGetFreezeInfoA(
 #pragma region Desktop Family or Esent Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_ESENT)
 
-JET_ERR JET_API
+_WIN32META_ANNOTATION_("win32metadata:unicode") JET_ERR JET_API
 JetOSSnapshotGetFreezeInfoW(
     _In_ const JET_OSSNAPID                                           snapId,
     _Out_ JET_UINT32 *                                                pcInstanceInfo,
