diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ExDisp.h b/generation/WinSDK/RecompiledIdlHeaders/um/ExDisp.h
index 88cd584ea7..f9b42793dc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ExDisp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ExDisp.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -501,7 +502,7 @@ EXTERN_C const IID IID_IWebBrowser;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Type( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *Type) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *Type) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Left( 
             /* [annotation][retval][out] */ 
@@ -537,11 +538,11 @@ EXTERN_C const IID IID_IWebBrowser;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LocationName( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *LocationName) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *LocationName) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LocationURL( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *LocationURL) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *LocationURL) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Busy( 
             /* [annotation][retval][out] */ 
@@ -1052,7 +1053,7 @@ EXTERN_C const IID IID_IWebBrowserApp;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Name( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *Name) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *Name) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_HWND( 
             /* [annotation][retval][out] */ 
@@ -1060,11 +1061,11 @@ EXTERN_C const IID IID_IWebBrowserApp;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_FullName( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *FullName) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *FullName) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Path( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *Path) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *Path) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Visible( 
             /* [annotation][retval][out] */ 
@@ -1084,7 +1085,7 @@ EXTERN_C const IID IID_IWebBrowserApp;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_StatusText( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *StatusText) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *StatusText) = 0;
         
         virtual /* [helpcontext][helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_StatusText( 
             /* [annotation][in] */ 
@@ -1647,7 +1648,7 @@ EXTERN_C const IID IID_IWebBrowser2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("D30C1661-CDAF-11d0-8A3E-00C04FC9E26E")
+    MIDL_INTERFACE("D30C1661-CDAF-11d0-8A3E-00C04FC9E26E") _Windows_SupportedOS_WindowsXP_ 
     IWebBrowser2 : public IWebBrowserApp
     {
     public:
@@ -2781,7 +2782,7 @@ EXTERN_C const IID IID_IShellWindows;
             /* [annotation][in] */ 
             _In_  long hwnd,
             /* [annotation][in] */ 
-            _In_  int swClass,
+            _Win32_AssociatedEnum_(ShellWindowTypeConstants) _In_  int swClass,
             /* [annotation][out] */ 
             _Out_  long *plCookie) = 0;
         
@@ -2793,7 +2794,7 @@ EXTERN_C const IID IID_IShellWindows;
             /* [annotation][in] */ 
             _In_  VARIANT *pvarlocRoot,
             /* [annotation][in] */ 
-            _In_  int swClass,
+            _Win32_AssociatedEnum_(ShellWindowTypeConstants) _In_  int swClass,
             /* [annotation][out] */ 
             _Out_  long *plCookie) = 0;
         
@@ -2819,11 +2820,11 @@ EXTERN_C const IID IID_IShellWindows;
             /* [annotation][in] */ 
             _In_  VARIANT *pvarLocRoot,
             /* [annotation][in] */ 
-            _In_  int swClass,
+            _Win32_AssociatedEnum_(ShellWindowTypeConstants) _In_  int swClass,
             /* [annotation][out] */ 
             _Out_  long *phwnd,
             /* [annotation][in] */ 
-            _In_  int swfwOptions,
+            _Win32_AssociatedEnum_(ShellWindowFindWindowOptions) _In_  int swfwOptions,
             /* [annotation][retval][out] */ 
             _Out_retval_  IDispatch **ppdispOut) = 0;
         
@@ -3470,7 +3471,7 @@ EXTERN_C const IID IID_IShellUIHelper2;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE BrandImageUri( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrUri) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrUri) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE SkipTabsWelcome( void) = 0;
         
@@ -3492,7 +3493,7 @@ EXTERN_C const IID IID_IShellUIHelper2;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE DefaultSearchProvider( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrName) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrName) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE RunOnceRequiredSettingsComplete( 
             /* [annotation][in] */ 
@@ -3504,7 +3505,7 @@ EXTERN_C const IID IID_IShellUIHelper2;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE SearchGuideUrl( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrUrl) = 0;
         
     };
     
@@ -8054,19 +8055,19 @@ EXTERN_C const IID IID_IShellUIHelper8;
     public:
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE GetCVListData( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrResult) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrResult) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE GetCVListLocalData( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrResult) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrResult) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE GetEMIEListData( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrResult) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrResult) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE GetEMIEListLocalData( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pbstrResult) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pbstrResult) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE OpenFavoritesPane( void) = 0;
         
@@ -10654,7 +10655,7 @@ EXTERN_C const IID IID_IShellNameSpace;
         
         virtual /* [helpstring][id][propget] */ HRESULT STDMETHODCALLTYPE get_Columns( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *bstrColumns) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *bstrColumns) = 0;
         
         virtual /* [helpstring][id][propput] */ HRESULT STDMETHODCALLTYPE put_Columns( 
             /* [annotation][in] */ 
@@ -11139,11 +11140,11 @@ EXTERN_C const IID IID_IScriptErrorList;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE getErrorMsg( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pstr) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pstr) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE getErrorUrl( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *pstr) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *pstr) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE getAlwaysShowLockState( 
             /* [annotation][retval][out] */ 
