diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/FaxExt.h b/generation/WinSDK/RecompiledIdlHeaders/um/FaxExt.h
index 34bf0971a5..836656aebc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/FaxExt.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/FaxExt.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation.  All rights reserved.
@@ -57,7 +58,7 @@ FaxExtGetData (
     LPBYTE                     *ppData,         // (Out) Pointer to allocated data
     LPDWORD                     lpdwDataSize    // (Out) Pointer to data size
 );
-typedef DWORD (CALLBACK *PFAX_EXT_GET_DATA) (DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, LPBYTE *, LPDWORD);
+typedef DWORD (CALLBACK *PFAX_EXT_GET_DATA) (DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPBYTE *, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPDWORD);
 
 //
 // Prototype of FaxExtSetData
@@ -72,7 +73,7 @@ FaxExtSetData (
     LPBYTE                      pData,          // Pointer to  data
     DWORD                       dwDataSize      // Data size
 );
-typedef DWORD (CALLBACK *PFAX_EXT_SET_DATA) (HINSTANCE, DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, LPBYTE, DWORD);
+typedef DWORD (CALLBACK *PFAX_EXT_SET_DATA) (HINSTANCE, DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPBYTE, DWORD);
 
 HRESULT
 WINAPI
@@ -82,7 +83,7 @@ FaxExtConfigChange (
     LPBYTE      lpData,             // New configuration data
     DWORD       dwDataSize          // Size of new configuration data
 );
-typedef HRESULT (WINAPI *PFAX_EXT_CONFIG_CHANGE) (DWORD, LPCWSTR, LPBYTE, DWORD);
+typedef HRESULT (WINAPI *PFAX_EXT_CONFIG_CHANGE) (DWORD, LPCWSTR, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPBYTE, DWORD);
 
 //
 // Prototype of FaxExtRegisterForEvents
@@ -96,7 +97,7 @@ FaxExtRegisterForEvents (
     LPCWSTR                     lpcwstrDataGUID,// GUID of data
     PFAX_EXT_CONFIG_CHANGE      lpConfigChangeCallback
 );
-typedef HANDLE (CALLBACK *PFAX_EXT_REGISTER_FOR_EVENTS) (HINSTANCE, DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, PFAX_EXT_CONFIG_CHANGE);
+_Win32_RAIIFree_(CloseHandle, -1, 0) typedef HANDLE (CALLBACK *PFAX_EXT_REGISTER_FOR_EVENTS) (HINSTANCE, DWORD, FAX_ENUM_DEVICE_ID_SOURCE, LPCWSTR, PFAX_EXT_CONFIG_CHANGE);
 
 //
 // Prototype of FaxExtUnregisterForEvents
@@ -116,7 +117,7 @@ WINAPI
 FaxExtFreeBuffer (
     LPVOID lpvBuffer
 );
-typedef VOID (CALLBACK *PFAX_EXT_FREE_BUFFER) (LPVOID);
+typedef VOID (CALLBACK *PFAX_EXT_FREE_BUFFER) (_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPVOID);
 
 //
 // The extension should implement and export the following function:
