diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/featurestagingapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/featurestagingapi.h
index 35e58e4089..91c1182505 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/featurestagingapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/featurestagingapi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*********************************************************************************
 *                                                                                *
 * featurestagingapi.h -- ApiSet Contract for api-ms-win-core-featurestaging-l1 *
@@ -81,7 +82,7 @@ RecordFeatureError(
 EXTERN_C
 void
 SubscribeFeatureStateChangeNotification(
-    _Outptr_ FEATURE_STATE_CHANGE_SUBSCRIPTION* subscription,
+    _Win32_RAIIFree_(UnsubscribeFeatureStateChangeNotification, -1, 0) _Outptr_ FEATURE_STATE_CHANGE_SUBSCRIPTION* subscription,
     _In_ PFEATURE_STATE_CHANGE_CALLBACK callback,
     _In_opt_ void* context
     );
