diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/FhCfg.h b/generation/WinSDK/RecompiledIdlHeaders/um/FhCfg.h
index 86e8285db6..a2a03ddd76 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/FhCfg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/FhCfg.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -159,7 +160,7 @@ EXTERN_C const IID IID_IFhTarget;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("D87965FD-2BAD-4657-BD3B-9567EB300CED")
+    MIDL_INTERFACE("D87965FD-2BAD-4657-BD3B-9567EB300CED") _Windows_SupportedOS_Windows8_ 
     IFhTarget : public IUnknown
     {
     public:
@@ -167,7 +168,7 @@ EXTERN_C const IID IID_IFhTarget;
             /* [annotation][in] */ 
             _In_  FH_TARGET_PROPERTY_TYPE PropertyType,
             /* [annotation][out] */ 
-            _Out_  BSTR *PropertyValue) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *PropertyValue) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetNumericalProperty( 
             /* [annotation][in] */ 
@@ -276,7 +277,7 @@ EXTERN_C const IID IID_IFhScopeIterator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("3197ABCE-532A-44C6-8615-F3666566A720")
+    MIDL_INTERFACE("3197ABCE-532A-44C6-8615-F3666566A720") _Windows_SupportedOS_Windows8_ 
     IFhScopeIterator : public IUnknown
     {
     public:
@@ -284,7 +285,7 @@ EXTERN_C const IID IID_IFhScopeIterator;
         
         virtual HRESULT STDMETHODCALLTYPE GetItem( 
             /* [annotation][out] */ 
-            _Out_  BSTR *Item) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *Item) = 0;
         
     };
     
@@ -436,7 +437,7 @@ EXTERN_C const IID IID_IFhConfigMgr;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("6A5FEA5B-BF8F-4EE5-B8C3-44D8A0D7331C")
+    MIDL_INTERFACE("6A5FEA5B-BF8F-4EE5-B8C3-44D8A0D7331C") _Windows_SupportedOS_Windows8_ 
     IFhConfigMgr : public IUnknown
     {
     public:
@@ -508,7 +509,7 @@ EXTERN_C const IID IID_IFhConfigMgr;
             /* [annotation][out] */ 
             _Out_  DWORD *ProtectionState,
             /* [annotation][out] */ 
-            _Out_  BSTR *ProtectedUntilTime) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *ProtectedUntilTime) = 0;
         
     };
     
@@ -729,7 +730,7 @@ EXTERN_C const IID IID_IFhReassociation;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("6544A28A-F68D-47ac-91EF-16B2B36AA3EE")
+    MIDL_INTERFACE("6544A28A-F68D-47ac-91EF-16B2B36AA3EE") _Windows_SupportedOS_Windows8_ 
     IFhReassociation : public IUnknown
     {
     public:
@@ -747,9 +748,9 @@ EXTERN_C const IID IID_IFhReassociation;
             /* [annotation][in] */ 
             _In_  DWORD Index,
             /* [annotation][out] */ 
-            _Out_  BSTR *UserName,
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *UserName,
             /* [annotation][out] */ 
-            _Out_  BSTR *PcName,
+            _Win32_RAIIFree_(SysFreeString) _Out_  BSTR *PcName,
             /* [annotation][out] */ 
             _Out_  FILETIME *BackupTime) = 0;
         
