diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
index 0383dfc960..72ba37462c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/fibersapi.h
@@ -16,6 +16,7 @@
 #include <apiset.h>
 #include <apisetcconv.h>
 #include <minwindef.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -30,13 +31,20 @@ extern "C" {
 #define FLS_OUT_OF_INDEXES ((DWORD)0xFFFFFFFF)
 #endif
 
+_Win32_RAIIFree_(FlsFree, -1)
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2003_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
 FlsAlloc(
-    _In_opt_ PFLS_CALLBACK_FUNCTION lpCallback
+    _Win32_Retained_ _In_opt_ PFLS_CALLBACK_FUNCTION lpCallback
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2003_
+_Win32_SetLastError_
 WINBASEAPI
 PVOID
 WINAPI
@@ -44,14 +52,20 @@ FlsGetValue(
     _In_ DWORD dwFlsIndex
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2003_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 FlsSetValue(
     _In_ DWORD dwFlsIndex,
-    _In_opt_ PVOID lpFlsData
+    _Win32_Retained_ _In_opt_ PVOID lpFlsData
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2003_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -69,6 +83,8 @@ FlsFree(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2008_
 WINBASEAPI
 BOOL
 WINAPI
