diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
index e576a1a610..a458066213 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**************************************************************************
 *
 * Copyright (c) 2001 Microsoft Corporation
@@ -165,20 +166,20 @@ struct ColorCurveParams
 class CGpEffect;
 
 extern "C" {
-Status __stdcall
-GdipCreateEffect(const GUID guid, CGpEffect **effect);
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
+GdipCreateEffect(_WIN32META_ANNOTATION_("win32metadata:in") const GUID guid, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") CGpEffect **effect);
 
-Status __stdcall
-GdipDeleteEffect(CGpEffect *effect);
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
+GdipDeleteEffect(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") CGpEffect *effect);
 
-Status __stdcall
-GdipGetEffectParameterSize(CGpEffect *effect, UINT *size);
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
+GdipGetEffectParameterSize(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") CGpEffect *effect, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") UINT *size);
 
-Status __stdcall
-GdipSetEffectParameters(CGpEffect *effect, const VOID *params, const UINT size);
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
+GdipSetEffectParameters(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") CGpEffect *effect, _WIN32META_ANNOTATION_("win32metadata:in") const VOID *params, _WIN32META_ANNOTATION_("win32metadata:in") const UINT size);
 
-Status __stdcall
-GdipGetEffectParameters(CGpEffect *effect, UINT *size, VOID *params);
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
+GdipGetEffectParameters(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") CGpEffect *effect, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") UINT *size, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") VOID *params);
 }
 
 #ifndef _GDIPLUSEFFECTS_EXCLUDEOBJECTS
