diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
index e576a1a610..2585846a83 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/gdipluseffects.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**************************************************************************
 *
 * Copyright (c) 2001 Microsoft Corporation
@@ -165,19 +166,19 @@ struct ColorCurveParams
 class CGpEffect;
 
 extern "C" {
-Status __stdcall
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
 GdipCreateEffect(const GUID guid, CGpEffect **effect);
 
-Status __stdcall
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
 GdipDeleteEffect(CGpEffect *effect);
 
-Status __stdcall
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
 GdipGetEffectParameterSize(CGpEffect *effect, UINT *size);
 
-Status __stdcall
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
 GdipSetEffectParameters(CGpEffect *effect, const VOID *params, const UINT size);
 
-Status __stdcall
+_Win32_ImportLibrary_("gdiplus.dll") Status __stdcall
 GdipGetEffectParameters(CGpEffect *effect, UINT *size, VOID *params);
 }
 
