diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/getcurrentpackageinfo3.h b/generation/WinSDK/RecompiledIdlHeaders/um/getcurrentpackageinfo3.h
index f62023d932..bfd9079aa5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/getcurrentpackageinfo3.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/getcurrentpackageinfo3.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 #include <windows.h>
 #include <appmodel.h>
 
@@ -10,9 +11,9 @@ enum PackageInfo3Type
 	PackageInfo3Type_PackageInfoGeneration = 16,
 } PackageInfoType;
 
-HRESULT GetCurrentPackageInfo3(
+_Win32_ImportLibrary_("kernel32.dll") HRESULT GetCurrentPackageInfo3(
   _In_ UINT32                                 flags,
-  _In_ PackageInfo3Type                       packageInfoType,
+   _In_ PackageInfo3Type                       packageInfoType,
   _Inout_ UINT32                             *bufferLength,
   _Out_writes_bytes_opt_(*bufferLength) void *buffer,
   _Out_opt_ UINT32                           *count
