diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
index 54c1445d36..fc10cce0fd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
@@ -17,6 +17,7 @@
 #include <apisetcconv.h>
 #include <minwindef.h>
 #include <minwinbase.h>
+#include <win32metadata_annotations.h>
 
 #if _MSC_VER < 1900
 #define DECLSPEC_ALLOCATOR
@@ -40,7 +41,7 @@ extern "C" {
 // typdefs
 //
 
-typedef struct _HEAP_SUMMARY {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:supported_os=windows10.0.20348") _HEAP_SUMMARY {
     DWORD cb;
     SIZE_T cbAllocated;
     SIZE_T cbCommitted;
@@ -53,6 +54,9 @@ typedef PHEAP_SUMMARY LPHEAP_SUMMARY;
 // Prototypes
 //
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
+_Win32_RAIIFree_(HeapDestroy)
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -63,6 +67,8 @@ HeapCreate(
     _In_ SIZE_T dwMaximumSize
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -70,6 +76,7 @@ HeapDestroy(
     _In_ HANDLE hHeap
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 _Ret_maybenull_
 _Post_writable_byte_size_(dwBytes)
@@ -82,6 +89,7 @@ HeapAlloc(
     _In_ SIZE_T dwBytes
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 _Success_(return != 0)
 _Ret_maybenull_
@@ -96,6 +104,8 @@ HeapReAlloc(
     _In_ SIZE_T dwBytes
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -106,6 +116,7 @@ HeapFree(
     __drv_freesMem(Mem) _Frees_ptr_opt_ LPVOID lpMem
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 SIZE_T
 WINAPI
@@ -115,6 +126,9 @@ HeapSize(
     _In_ LPCVOID lpMem
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
+_WIN32META_ANNOTATION_("win32metadata:do_not_release")
 WINBASEAPI
 HANDLE
 WINAPI
@@ -122,6 +136,8 @@ GetProcessHeap(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 SIZE_T
 WINAPI
@@ -130,6 +146,8 @@ HeapCompact(
     _In_ DWORD dwFlags
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -146,6 +164,7 @@ HeapSetInformation(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 BOOL
 WINAPI
@@ -161,6 +180,7 @@ HeapValidate(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_WIN32META_ANNOTATION_("win32metadata:supported_os=windows10.0.20348")
 BOOL
 WINAPI
 HeapSummary(
@@ -175,14 +195,18 @@ HeapSummary(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
 GetProcessHeaps(
     _In_ DWORD NumberOfHeaps,
-    _Out_writes_to_(NumberOfHeaps,return) PHANDLE ProcessHeaps
+    _Out_writes_to_(NumberOfHeaps,return) _WIN32META_ANNOTATION_("win32metadata:do_not_release") PHANDLE ProcessHeaps
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -190,6 +214,8 @@ HeapLock(
     _In_ HANDLE hHeap
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -203,6 +229,8 @@ HeapUnlock(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -211,6 +239,8 @@ HeapWalk(
     _Inout_ LPPROCESS_HEAP_ENTRY lpEntry
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
