diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
index d72e8f8ca0..be64afdc4b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/heapapi.h
@@ -17,6 +17,7 @@
 #include <apisetcconv.h>
 #include <minwindef.h>
 #include <minwinbase.h>
+#include <win32metadata_annotations.h>
 
 #if _MSC_VER < 1900
 #define DECLSPEC_ALLOCATOR
@@ -40,7 +41,7 @@ extern "C" {
 // typdefs
 //
 
-typedef struct _HEAP_SUMMARY {
+typedef struct  _HEAP_SUMMARY {
     DWORD cb;
     SIZE_T cbAllocated;
     SIZE_T cbCommitted;
@@ -53,16 +54,21 @@ typedef PHEAP_SUMMARY LPHEAP_SUMMARY;
 // Prototypes
 //
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
+_Win32_RAIIFree_(HeapDestroy)
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
 WINAPI
 HeapCreate(
-    _In_ DWORD flOptions,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD flOptions,
     _In_ SIZE_T dwInitialSize,
     _In_ SIZE_T dwMaximumSize
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -70,6 +76,7 @@ HeapDestroy(
     _In_ HANDLE hHeap
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 _Ret_maybenull_
 _Post_writable_byte_size_(dwBytes)
@@ -78,10 +85,11 @@ LPVOID
 WINAPI
 HeapAlloc(
     _In_ HANDLE hHeap,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags,
     _In_ SIZE_T dwBytes
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 _Success_(return != 0)
 _Ret_maybenull_
@@ -91,30 +99,35 @@ LPVOID
 WINAPI
 HeapReAlloc(
     _Inout_ HANDLE hHeap,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags,
     _Frees_ptr_opt_ LPVOID lpMem,
     _In_ SIZE_T dwBytes
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
 WINAPI
 HeapFree(
     _Inout_ HANDLE hHeap,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags,
     __drv_freesMem(Mem) _Frees_ptr_opt_ LPVOID lpMem
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 SIZE_T
 WINAPI
 HeapSize(
     _In_ HANDLE hHeap,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags,
     _In_ LPCVOID lpMem
     );
 
+_WIN32META_ANNOTATION_("win32metadata:do_not_release") _Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -122,14 +135,18 @@ GetProcessHeap(
     VOID
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 SIZE_T
 WINAPI
 HeapCompact(
     _In_ HANDLE hHeap,
-    _In_ DWORD dwFlags
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -146,12 +163,13 @@ HeapSetInformation(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 BOOL
 WINAPI
 HeapValidate(
     _In_ HANDLE hHeap,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(HEAP_FLAGS) _In_ DWORD dwFlags,
     _In_opt_ LPCVOID lpMem
     );
 
@@ -161,6 +179,7 @@ HeapValidate(
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+
 BOOL
 WINAPI
 HeapSummary(
@@ -175,14 +194,18 @@ HeapSummary(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
 GetProcessHeaps(
     _In_ DWORD NumberOfHeaps,
-    _Out_writes_to_(NumberOfHeaps,return) PHANDLE ProcessHeaps
+    _WIN32META_ANNOTATION_("win32metadata:do_not_release") _Out_writes_to_(NumberOfHeaps,return) PHANDLE ProcessHeaps
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -190,6 +213,8 @@ HeapLock(
     _In_ HANDLE hHeap
     );
 
+_Win32_ImportLibrary_("kernel32.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -203,6 +228,8 @@ HeapUnlock(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -211,6 +238,8 @@ HeapWalk(
     _Inout_ LPPROCESS_HEAP_ENTRY lpEntry
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
