diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/hidclass.h b/generation/WinSDK/RecompiledIdlHeaders/shared/hidclass.h
index 6fcf801b8b..fb5d1a8f55 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/hidclass.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/hidclass.h
@@ -19,6 +19,7 @@ Environment:
 
 #include <basetyps.h>
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -207,6 +208,7 @@ typedef struct _HID_INTERFACE_NOTIFY_PNP
 #ifdef __HIDPI_H__
 
 _Must_inspect_result_
+_Windows_SupportedOS_Windows2000_
 typedef 
 NTSTATUS 
 (__stdcall *PHIDP_GETCAPS) (
