diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/HolographicSpaceInterop.h b/generation/WinSDK/RecompiledIdlHeaders/um/HolographicSpaceInterop.h
index 98328aba61..165f689e04 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/HolographicSpaceInterop.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/HolographicSpaceInterop.h
@@ -19,6 +19,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -85,10 +86,11 @@ EXTERN_C const IID IID_IHolographicSpaceInterop;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("5C4EE536-6A98-4B86-A170-587013D6FD4B")
+    MIDL_INTERFACE("5C4EE536-6A98-4B86-A170-587013D6FD4B") _Windows_SupportedOS_15063_ 
     IHolographicSpaceInterop : public IInspectable
     {
     public:
+        
         virtual HRESULT STDMETHODCALLTYPE CreateForWindow( 
             /* [annotation][in] */ 
             _In_  HWND window,
