diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/homepagesetting.h b/generation/WinSDK/RecompiledIdlHeaders/um/homepagesetting.h
index 0f150b877a21152df01434c825716f2e5fa21997..6f8b918812c0092415d67788a0b51169eee4f0c9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/homepagesetting.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/homepagesetting.h
@@ -90,6 +90,7 @@ extern "C"{
 //=--------------------------------------------------------------------------=
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -109,7 +110,7 @@ EXTERN_C const IID IID_IHomePageSetting;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("FDFC244F-18FA-4FF2-B08E-1D618F3FFBE4")
-    IHomePageSetting : public IUnknown
+    _Windows_SupportedOS_Windows7_ IHomePageSetting : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE SetHomePage( 
@@ -178,7 +179,7 @@ EXTERN_C const IID IID_IHomePageSetting;
         END_INTERFACE
     } IHomePageSettingVtbl;
 
-    interface IHomePageSetting
+    interface _Windows_SupportedOS_Windows7_ IHomePageSetting
     {
         CONST_VTBL struct IHomePageSettingVtbl *lpVtbl;
     };
