diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/http.h b/generation/WinSDK/RecompiledIdlHeaders/um/http.h
index 4178ef13d1..d3912dfa13 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/http.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/http.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /*++
@@ -2997,22 +2998,22 @@ typedef enum _HTTP_FEATURE_ID
 
 // NOTE: MUST be called once before all other APIs
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpInitialize(
     IN HTTPAPI_VERSION Version,
-    IN ULONG Flags,
+    _Win32_AssociatedEnum_(HTTP_INITIALIZE) IN ULONG Flags,
     _Reserved_ IN OUT PVOID pReserved
     );
 
 // NOTE: MUST be called after final API call returns.
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpTerminate(
-    IN ULONG Flags,
+    _Win32_AssociatedEnum_(HTTP_INITIALIZE) IN ULONG Flags,
     _Reserved_ IN OUT PVOID pReserved
     );
 
@@ -3027,11 +3028,11 @@ HttpTerminate(
 // HttpCreateHttpHandle() API.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCreateHttpHandle(
-    OUT PHANDLE RequestQueueHandle,
+    _Win32_RAIIFree_(CloseHandle, -1, 0) OUT PHANDLE RequestQueueHandle,
     _Reserved_ ULONG Reserved
     );
 
@@ -3044,7 +3045,7 @@ HttpCreateHttpHandle(
 // created by the HttpCreateRequestQueue API.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCreateRequestQueue(
@@ -3052,17 +3053,17 @@ HttpCreateRequestQueue(
     IN PCWSTR Name OPTIONAL,
     IN PSECURITY_ATTRIBUTES SecurityAttributes OPTIONAL,
     IN ULONG Flags OPTIONAL,
-    OUT PHANDLE RequestQueueHandle
+    _Win32_RAIIFree_(HttpCloseRequestQueue, 0) OUT PHANDLE RequestQueueHandle
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCloseRequestQueue(
     IN HANDLE RequestQueueHandle
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpSetRequestQueueProperty(
@@ -3074,7 +3075,7 @@ HttpSetRequestQueueProperty(
     _Reserved_ PVOID Reserved2
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpQueryRequestQueueProperty(
@@ -3114,7 +3115,7 @@ HttpQueryRequestProperty(
     _In_opt_ LPOVERLAPPED Overlapped
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpShutdownRequestQueue(
@@ -3127,7 +3128,7 @@ HttpShutdownRequestQueue(
 // SSL APIs.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpReceiveClientCertificate(
@@ -3137,7 +3138,7 @@ HttpReceiveClientCertificate(
     _Out_writes_bytes_to_(SslClientCertInfoSize, *BytesReceived) PHTTP_SSL_CLIENT_CERT_INFO SslClientCertInfo,
     IN ULONG SslClientCertInfoSize,
     _Out_opt_ PULONG BytesReceived,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
 #if _WIN32_WINNT >= 0x0600
@@ -3146,7 +3147,7 @@ HttpReceiveClientCertificate(
 // Server Session APIs.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCreateServerSession(
@@ -3155,14 +3156,14 @@ HttpCreateServerSession(
     _Reserved_ IN ULONG Reserved
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCloseServerSession(
     IN HTTP_SERVER_SESSION_ID ServerSessionId
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 _Success_(return == NO_ERROR)
 ULONG
 WINAPI
@@ -3174,7 +3175,7 @@ HttpQueryServerSessionProperty(
     _Out_opt_ PULONG ReturnLength
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpSetServerSessionProperty(
@@ -3190,7 +3191,7 @@ HttpSetServerSessionProperty(
 // Url Configuration APIs. Can only be used for V1 request queues.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpAddUrl(
@@ -3199,7 +3200,7 @@ HttpAddUrl(
     _Reserved_ PVOID Reserved
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpRemoveUrl(
@@ -3213,7 +3214,7 @@ HttpRemoveUrl(
 // Url Group APIs.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCreateUrlGroup(
@@ -3222,14 +3223,14 @@ HttpCreateUrlGroup(
     _Reserved_ IN ULONG Reserved
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCloseUrlGroup(
     IN HTTP_URL_GROUP_ID UrlGroupId
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpAddUrlToUrlGroup(
@@ -3239,7 +3240,7 @@ HttpAddUrlToUrlGroup(
     _Reserved_ IN ULONG Reserved
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpRemoveUrlFromUrlGroup(
@@ -3248,7 +3249,7 @@ HttpRemoveUrlFromUrlGroup(
     IN ULONG Flags
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpSetUrlGroupProperty(
@@ -3258,7 +3259,7 @@ HttpSetUrlGroupProperty(
     IN ULONG PropertyInformationLength
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 _Success_(return == NO_ERROR)
 ULONG
 WINAPI
@@ -3274,7 +3275,7 @@ HttpQueryUrlGroupProperty(
 
 #if _WIN32_WINNT >= _WIN32_WINNT_WIN8
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_Windows8_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpPrepareUrl(
@@ -3290,20 +3291,20 @@ HttpPrepareUrl(
 // HTTP Server I/O APIs.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpReceiveHttpRequest(
     IN HANDLE RequestQueueHandle,
     IN HTTP_REQUEST_ID RequestId,
-    IN ULONG Flags,
+    _Win32_AssociatedEnum_(HTTP_RECEIVE_HTTP_REQUEST_FLAGS) IN ULONG Flags,
     _Out_writes_bytes_to_(RequestBufferLength, *BytesReturned) PHTTP_REQUEST RequestBuffer,
     IN ULONG RequestBufferLength,
     _Out_opt_ PULONG BytesReturned,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpReceiveRequestEntityBody(
@@ -3313,7 +3314,7 @@ HttpReceiveRequestEntityBody(
     _Out_writes_bytes_to_(EntityBufferLength, *BytesReturned) PVOID EntityBuffer,
     IN ULONG EntityBufferLength,
     _Out_opt_ PULONG BytesReturned,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
 #if _WIN32_WINNT >= 0x0600
@@ -3393,7 +3394,7 @@ HttpSendResponseEntityBody(
 
 #endif // _WIN32_WINNT >= 0x0600
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_10240_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpDeclarePush(
@@ -3405,13 +3406,13 @@ HttpDeclarePush(
     _In_opt_ PHTTP_REQUEST_HEADERS Headers
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpWaitForDisconnect(
     IN HANDLE RequestQueueHandle,
     IN HTTP_CONNECTION_ID ConnectionId,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
 #if _WIN32_WINNT >= 0x0600
@@ -3423,24 +3424,24 @@ HttpWaitForDisconnectEx(
     IN HANDLE RequestQueueHandle,
     IN HTTP_CONNECTION_ID ConnectionId,
     _Reserved_ IN ULONG Reserved OPTIONAL,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpCancelHttpRequest(
     IN HANDLE RequestQueueHandle,
     IN HTTP_REQUEST_ID RequestId,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpWaitForDemandStart(
     IN HANDLE RequestQueueHandle,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
 BOOL
@@ -3477,17 +3478,17 @@ HttpFindUrlGroupId(
 // Cache manipulation APIs.
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpFlushResponseCache(
     IN HANDLE RequestQueueHandle,
     IN PCWSTR UrlPrefix,
     IN ULONG Flags,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpAddFragmentToCache(
@@ -3495,10 +3496,10 @@ HttpAddFragmentToCache(
     IN PCWSTR UrlPrefix,
     IN PHTTP_DATA_CHUNK DataChunk,
     IN PHTTP_CACHE_POLICY CachePolicy,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpReadFragmentFromCache(
@@ -3508,14 +3509,14 @@ HttpReadFragmentFromCache(
     _Out_writes_bytes_to_(BufferLength, *BytesRead) PVOID Buffer,
     IN ULONG BufferLength,
     _Out_opt_ PULONG BytesRead,
-    IN LPOVERLAPPED Overlapped OPTIONAL
+    _Win32_Retained_ IN LPOVERLAPPED Overlapped OPTIONAL
     );
 
 //
 // Server configuration APIs
 //
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsXP_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpSetServiceConfiguration(
@@ -3526,7 +3527,7 @@ HttpSetServiceConfiguration(
     _Reserved_ LPOVERLAPPED pOverlapped
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_15063_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpUpdateServiceConfiguration(
@@ -3537,7 +3538,7 @@ HttpUpdateServiceConfiguration(
     _Reserved_ LPOVERLAPPED Overlapped
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpDeleteServiceConfiguration(
@@ -3548,7 +3549,7 @@ HttpDeleteServiceConfiguration(
     _Reserved_ LPOVERLAPPED pOverlapped
     );
 
-HTTPAPI_LINKAGE
+_Windows_SupportedOS_WindowsVista_ HTTPAPI_LINKAGE
 ULONG
 WINAPI
 HttpQueryServiceConfiguration(
@@ -3582,5 +3583,34 @@ HttpGetExtension(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("HTTP_RECEIVE_REQUEST_FLAG_COPY_BODY")
+#undef HTTP_RECEIVE_REQUEST_FLAG_COPY_BODY
+#pragma push_macro("HTTP_RECEIVE_REQUEST_FLAG_FLUSH_BODY")
+#undef HTTP_RECEIVE_REQUEST_FLAG_FLUSH_BODY
+enum HTTP_RECEIVE_HTTP_REQUEST_FLAGS : unsigned int
+{
+    HTTP_RECEIVE_REQUEST_FLAG_COPY_BODY = 1U,
+    HTTP_RECEIVE_REQUEST_FLAG_FLUSH_BODY = 2U,
+};
+#pragma pop_macro("HTTP_RECEIVE_REQUEST_FLAG_FLUSH_BODY")
+#pragma pop_macro("HTTP_RECEIVE_REQUEST_FLAG_COPY_BODY")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("HTTP_INITIALIZE_CONFIG")
+#undef HTTP_INITIALIZE_CONFIG
+#pragma push_macro("HTTP_INITIALIZE_SERVER")
+#undef HTTP_INITIALIZE_SERVER
+enum [[clang::flag_enum]] HTTP_INITIALIZE : unsigned int
+{
+    HTTP_INITIALIZE_CONFIG = 2U,
+    HTTP_INITIALIZE_SERVER = 1U,
+};
+#pragma pop_macro("HTTP_INITIALIZE_SERVER")
+#pragma pop_macro("HTTP_INITIALIZE_CONFIG")
+#endif
+
 #endif // __HTTP_H__
 
