diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ImageHlp.h b/generation/WinSDK/RecompiledIdlHeaders/um/ImageHlp.h
index 3d23d23d79..53730703b7 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ImageHlp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ImageHlp.h
@@ -23,6 +23,7 @@ Revision History:
 #endif
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #define NONGAMESPARTITIONS WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WER
 
@@ -126,6 +127,8 @@ typedef struct _LOADED_IMAGE {
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 BindImage(
@@ -188,6 +191,8 @@ BOOL
     );
 
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 BindImageEx(
@@ -206,6 +211,8 @@ BindImageEx(
                                                 //  (same as NT 3.1->NT 4.0)
 #define BIND_REPORT_64BIT_VA   0x00000010
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ReBaseImage(
@@ -223,6 +230,8 @@ ReBaseImage(
     _In_ ULONG TimeStamp             // new timestamp for image, if non-zero
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ReBaseImage64(
@@ -271,6 +280,8 @@ ReBaseImage64(
 // Define checksum function prototypes.
 //
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 PIMAGE_NT_HEADERS
 IMAGEAPI
 CheckSumMappedFile (
@@ -280,6 +291,7 @@ CheckSumMappedFile (
     _Out_ PDWORD CheckSum
     );
 
+_Windows_SupportedOS_WindowsXP_
 DWORD
 IMAGEAPI
 MapFileAndCheckSumA (
@@ -288,6 +300,7 @@ MapFileAndCheckSumA (
     _Out_ PDWORD CheckSum
     );
 
+_Windows_SupportedOS_WindowsXP_
 DWORD
 IMAGEAPI
 MapFileAndCheckSumW (
@@ -302,6 +315,8 @@ MapFileAndCheckSumW (
 #define MapFileAndCheckSum  MapFileAndCheckSumA
 #endif // !UNICODE
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 GetImageConfigInformation(
@@ -309,6 +324,8 @@ GetImageConfigInformation(
     _Out_ PIMAGE_LOAD_CONFIG_DIRECTORY ImageConfigInformation
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 GetImageUnusedHeaderBytes(
@@ -316,6 +333,8 @@ GetImageUnusedHeaderBytes(
     _Out_ PDWORD SizeUnusedHeaderBytes
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SetImageConfigInformation(
@@ -336,6 +355,8 @@ typedef PVOID DIGEST_HANDLE;
 
 typedef BOOL (WINAPI *DIGEST_FUNCTION) (DIGEST_HANDLE refdata, PBYTE pData, DWORD dwLength);
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ImageGetDigestStream(
@@ -345,6 +366,8 @@ ImageGetDigestStream(
     _In_ DIGEST_HANDLE DigestHandle
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 _Success_(return != FALSE)
 BOOL
 IMAGEAPI
@@ -354,6 +377,8 @@ ImageAddCertificate(
     _Out_ PDWORD Index
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ImageRemoveCertificate(
@@ -361,6 +386,8 @@ ImageRemoveCertificate(
     _In_ DWORD Index
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 _Success_(return != FALSE)
 BOOL
 IMAGEAPI
@@ -372,6 +399,8 @@ ImageEnumerateCertificates(
     _In_opt_ DWORD IndexCount
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 _Success_(return != FALSE)
 BOOL
 IMAGEAPI
@@ -382,6 +411,8 @@ ImageGetCertificateData(
     _Inout_ PDWORD RequiredLength
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ImageGetCertificateHeader(
@@ -390,6 +421,8 @@ ImageGetCertificateHeader(
     _Inout_ LPWIN_CERTIFICATE Certificateheader
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 PLOADED_IMAGE
 IMAGEAPI
 ImageLoad(
@@ -397,12 +430,16 @@ ImageLoad(
     _In_opt_ PCSTR DllPath
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 ImageUnload(
     _Inout_ PLOADED_IMAGE LoadedImage
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 MapAndLoad(
@@ -413,12 +450,16 @@ MapAndLoad(
     _In_ BOOL ReadOnly
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 UnMapAndLoad(
     _Inout_ PLOADED_IMAGE LoadedImage
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 TouchFileTimes (
@@ -426,6 +467,8 @@ TouchFileTimes (
     _In_opt_ PSYSTEMTIME pSystemTime
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SplitSymbols (
@@ -435,6 +478,8 @@ SplitSymbols (
     _In_ ULONG Flags
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 UpdateDebugInfoFile(
@@ -444,6 +489,7 @@ UpdateDebugInfoFile(
     _In_ PIMAGE_NT_HEADERS32 NtHeaders
     );
 
+_Windows_SupportedOS_WindowsXP_
 BOOL
 IMAGEAPI
 UpdateDebugInfoFileEx(
@@ -470,6 +516,7 @@ typedef BOOL
     _In_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 SymFindDebugInfoFile(
@@ -487,6 +534,7 @@ typedef BOOL
     _In_ PVOID  CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 SymFindDebugInfoFileW(
@@ -497,6 +545,7 @@ SymFindDebugInfoFileW(
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindDebugInfoFile (
@@ -505,6 +554,7 @@ FindDebugInfoFile (
     _Out_writes_(MAX_PATH + 1) PSTR DebugFilePath
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindDebugInfoFileEx (
@@ -515,6 +565,7 @@ FindDebugInfoFileEx (
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindDebugInfoFileExW (
@@ -531,6 +582,7 @@ typedef BOOL
     _In_ PVOID context
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFindFileInPath(
@@ -552,6 +604,7 @@ typedef BOOL
     _In_ PVOID context
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFindFileInPathW(
@@ -574,6 +627,7 @@ typedef BOOL
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 SymFindExecutableImage(
@@ -591,6 +645,7 @@ typedef BOOL
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 SymFindExecutableImageW(
@@ -601,6 +656,7 @@ SymFindExecutableImageW(
     _In_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindExecutableImage(
@@ -609,6 +665,7 @@ FindExecutableImage(
     _Out_writes_(MAX_PATH + 1) PSTR ImageFilePath
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindExecutableImageEx(
@@ -619,6 +676,7 @@ FindExecutableImageEx(
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 HANDLE
 IMAGEAPI
 FindExecutableImageExW(
@@ -629,12 +687,14 @@ FindExecutableImageExW(
     _In_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 PIMAGE_NT_HEADERS
 IMAGEAPI
 ImageNtHeader (
     _In_ PVOID Base
     );
 
+_Win32_SetLastError_
 PVOID
 IMAGEAPI
 ImageDirectoryEntryToDataEx (
@@ -645,6 +705,7 @@ ImageDirectoryEntryToDataEx (
     _Out_opt_ PIMAGE_SECTION_HEADER *FoundHeader
     );
 
+_Win32_SetLastError_
 PVOID
 IMAGEAPI
 ImageDirectoryEntryToData (
@@ -654,6 +715,7 @@ ImageDirectoryEntryToData (
     _Out_ PULONG Size
     );
 
+_Win32_SetLastError_
 PIMAGE_SECTION_HEADER
 IMAGEAPI
 ImageRvaToSection(
@@ -662,6 +724,7 @@ ImageRvaToSection(
     _In_ ULONG Rva
     );
 
+_Win32_SetLastError_
 PVOID
 IMAGEAPI
 ImageRvaToVa(
@@ -721,6 +784,7 @@ typedef struct _IMAGE_DEBUG_INFORMATION {
 } IMAGE_DEBUG_INFORMATION, *PIMAGE_DEBUG_INFORMATION;
 
 
+_Win32_SetLastError_
 PIMAGE_DEBUG_INFORMATION
 IMAGEAPI
 MapDebugInformation(
@@ -730,6 +794,7 @@ MapDebugInformation(
     _In_ ULONG ImageBase
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 UnmapDebugInformation(
@@ -738,6 +803,7 @@ UnmapDebugInformation(
 
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SearchTreeForFile(
@@ -746,6 +812,7 @@ SearchTreeForFile(
     _Out_writes_(MAX_PATH + 1) PSTR OutputPathBuffer
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SearchTreeForFileW(
@@ -760,6 +827,7 @@ typedef BOOL
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumDirTree(
@@ -777,6 +845,7 @@ typedef BOOL
     _In_opt_ PVOID CallerData
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumDirTreeW(
@@ -788,6 +857,7 @@ EnumDirTreeW(
     _In_opt_ PVOID data
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 MakeSureDirectoryPathExists(
@@ -817,6 +887,7 @@ MakeSureDirectoryPathExists(
 #define UNDNAME_NO_ARGUMENTS             (0x2000)  // Don't undecorate arguments to function
 #define UNDNAME_NO_SPECIAL_SYMS          (0x4000)  // Don't undecorate special names (v-table, vcall, vector xxx, metatype, etc)
 
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 WINAPI
@@ -827,6 +898,7 @@ UnDecorateSymbolName(
     _In_ DWORD flags
     );
 
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 WINAPI
@@ -1387,6 +1459,7 @@ ImagehlpApiVersionEx(
     _In_ LPAPI_VERSION AppVersion
     );
 
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 GetTimestampForLoadedLibrary(
@@ -2039,6 +2112,7 @@ typedef struct _IMAGEHLP_JIT_SYMBOL_MAP {
 
 // If dbghelp ever needs to display graphical UI, it will use this as the parent window.
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetParentWindow(
@@ -2051,6 +2125,7 @@ SymGetParentWindow(
     _Out_ HWND * pHwnd
     );
 
+_Win32_SetLastError_
 PCHAR
 IMAGEAPI
 SymSetHomeDirectory(
@@ -2058,6 +2133,7 @@ SymSetHomeDirectory(
     _In_opt_ PCSTR dir
     );
 
+_Win32_SetLastError_
 PWSTR
 IMAGEAPI
 SymSetHomeDirectoryW(
@@ -2065,6 +2141,7 @@ SymSetHomeDirectoryW(
     _In_opt_ PCWSTR dir
     );
 
+_Win32_SetLastError_
 PCHAR
 IMAGEAPI
 SymGetHomeDirectory(
@@ -2073,6 +2150,7 @@ SymGetHomeDirectory(
     _In_ size_t size
     );
 
+_Win32_SetLastError_
 PWSTR
 IMAGEAPI
 SymGetHomeDirectoryW(
@@ -2093,6 +2171,7 @@ typedef struct _OMAP {
     ULONG  rvaTo;
 } OMAP, *POMAP;
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetOmaps(
@@ -2180,6 +2259,7 @@ SymGetOptions(
     VOID
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymCleanup(
@@ -2208,6 +2288,7 @@ SymSetExtendedOption(_In_ IMAGEHLP_EXTENDED_OPTIONS option, _In_ BOOL value);
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymMatchString(
@@ -2224,6 +2305,7 @@ SymMatchStringA(
     _In_ BOOL fCase
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymMatchStringW(
@@ -2241,6 +2323,7 @@ typedef BOOL
 // for backwards compatibility - don't use this
 #define PSYM_ENUMSOURCFILES_CALLBACK PSYM_ENUMSOURCEFILES_CALLBACK
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSourceFiles(
@@ -2257,6 +2340,7 @@ typedef BOOL
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSourceFilesW(
@@ -2267,6 +2351,7 @@ SymEnumSourceFilesW(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumerateModules64(
@@ -2275,6 +2360,7 @@ SymEnumerateModules64(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumerateModulesW64(
@@ -2286,6 +2372,7 @@ SymEnumerateModulesW64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymEnumerateModules SymEnumerateModules64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumerateModules(
@@ -2295,6 +2382,7 @@ SymEnumerateModules(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumerateLoadedModulesEx(
@@ -2303,6 +2391,7 @@ EnumerateLoadedModulesEx(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumerateLoadedModulesExW(
@@ -2311,6 +2400,7 @@ EnumerateLoadedModulesExW(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumerateLoadedModules64(
@@ -2319,6 +2409,7 @@ EnumerateLoadedModules64(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumerateLoadedModulesW64(
@@ -2330,6 +2421,7 @@ EnumerateLoadedModulesW64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define EnumerateLoadedModules EnumerateLoadedModules64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 EnumerateLoadedModules(
@@ -2339,6 +2431,7 @@ EnumerateLoadedModules(
     );
 #endif
 
+_Win32_SetLastError_
 PVOID
 IMAGEAPI
 SymFunctionTableAccess64(
@@ -2364,6 +2457,7 @@ SymFunctionTableAccess64AccessRoutines(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymFunctionTableAccess SymFunctionTableAccess64
 #else
+_Win32_SetLastError_
 PVOID
 IMAGEAPI
 SymFunctionTableAccess(
@@ -2387,6 +2481,7 @@ SymGetUnwindInfo(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetModuleInfo64(
@@ -2395,6 +2490,7 @@ SymGetModuleInfo64(
     _Out_ PIMAGEHLP_MODULE64 ModuleInfo
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetModuleInfoW64(
@@ -2407,6 +2503,7 @@ SymGetModuleInfoW64(
 #define SymGetModuleInfo   SymGetModuleInfo64
 #define SymGetModuleInfoW  SymGetModuleInfoW64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetModuleInfo(
@@ -2415,6 +2512,7 @@ SymGetModuleInfo(
     _Out_ PIMAGEHLP_MODULE ModuleInfo
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetModuleInfoW(
@@ -2424,6 +2522,7 @@ SymGetModuleInfoW(
     );
 #endif
 
+_Win32_SetLastError_
 DWORD64
 IMAGEAPI
 SymGetModuleBase64(
@@ -2434,6 +2533,7 @@ SymGetModuleBase64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetModuleBase SymGetModuleBase64
 #else
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 SymGetModuleBase(
@@ -2474,6 +2574,7 @@ typedef BOOL
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumLines(
@@ -2491,6 +2592,7 @@ typedef BOOL
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumLinesW(
@@ -2508,6 +2610,7 @@ SymEnumLinesW(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromAddr64(
@@ -2523,6 +2626,7 @@ SymGetLineFromAddr64(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromAddrW64(
@@ -2532,6 +2636,7 @@ SymGetLineFromAddrW64(
     _Out_ PIMAGEHLP_LINEW64 Line
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromInlineContext(
@@ -2543,6 +2648,7 @@ SymGetLineFromInlineContext(
     _Out_ PIMAGEHLP_LINE64 Line64
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromInlineContextW(
@@ -2554,6 +2660,7 @@ SymGetLineFromInlineContextW(
     _Out_ PIMAGEHLP_LINEW64 Line
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSourceLines(
@@ -2567,6 +2674,7 @@ SymEnumSourceLines(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSourceLinesW(
@@ -2608,6 +2716,7 @@ SymCompareInlineTrace(
     _In_ DWORD64 RetAddress2
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymQueryInlineTrace(
@@ -2632,6 +2741,7 @@ SymQueryInlineTrace(
 #define SymGetLineFromAddr SymGetLineFromAddr64
 #define SymGetLineFromAddrW SymGetLineFromAddrW64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromAddr(
@@ -2651,6 +2761,7 @@ SymGetLineFromAddrW(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromName64(
@@ -2662,6 +2773,7 @@ SymGetLineFromName64(
     _Inout_ PIMAGEHLP_LINE64 Line
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromNameW64(
@@ -2676,6 +2788,7 @@ SymGetLineFromNameW64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetLineFromName SymGetLineFromName64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineFromName(
@@ -2688,6 +2801,7 @@ SymGetLineFromName(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineNext64(
@@ -2695,6 +2809,7 @@ SymGetLineNext64(
     _Inout_ PIMAGEHLP_LINE64 Line
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineNextW64(
@@ -2705,6 +2820,7 @@ SymGetLineNextW64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetLineNext SymGetLineNext64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLineNext(
@@ -2720,6 +2836,7 @@ SymGetLineNextW(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLinePrev64(
@@ -2727,6 +2844,7 @@ SymGetLinePrev64(
     _Inout_ PIMAGEHLP_LINE64 Line
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLinePrevW64(
@@ -2737,6 +2855,7 @@ SymGetLinePrevW64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetLinePrev SymGetLinePrev64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetLinePrev(
@@ -2752,6 +2871,7 @@ SymGetLinePrevW(
     );
 #endif
 
+_Win32_SetLastError_
 ULONG
 IMAGEAPI
 SymGetFileLineOffsets64(
@@ -2762,6 +2882,7 @@ SymGetFileLineOffsets64(
     _In_ ULONG BufferLines
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymMatchFileName(
@@ -2771,6 +2892,7 @@ SymMatchFileName(
     _Outptr_opt_ PSTR *MatchStop
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymMatchFileNameW(
@@ -2780,6 +2902,7 @@ SymMatchFileNameW(
     _Outptr_opt_ PWSTR *MatchStop
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFile(
@@ -2791,6 +2914,7 @@ SymGetSourceFile(
     _In_ DWORD Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileW(
@@ -2802,6 +2926,7 @@ SymGetSourceFileW(
     _In_ DWORD Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileToken(
@@ -2824,6 +2949,7 @@ SymGetSourceFileTokenByTokenName(
     _Out_ DWORD *Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileChecksumW(
@@ -2836,6 +2962,7 @@ SymGetSourceFileChecksumW(
     _Out_ DWORD *pActualBytesWritten
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileChecksum(
@@ -2848,6 +2975,7 @@ SymGetSourceFileChecksum(
     _Out_ DWORD *pActualBytesWritten
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileTokenW(
@@ -2870,6 +2998,7 @@ SymGetSourceFileTokenByTokenNameW(
     _Out_ DWORD *Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileFromToken(
@@ -2891,6 +3020,7 @@ SymGetSourceFileFromTokenByTokenName(
     _In_ DWORD Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceFileFromTokenW(
@@ -2912,6 +3042,7 @@ SymGetSourceFileFromTokenByTokenNameW(
     _In_ DWORD Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceVarFromToken(
@@ -2923,6 +3054,7 @@ SymGetSourceVarFromToken(
     _In_ DWORD Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSourceVarFromTokenW(
@@ -2936,6 +3068,7 @@ SymGetSourceVarFromTokenW(
 
 typedef BOOL (CALLBACK *PENUMSOURCEFILETOKENSCALLBACK)(_In_ PVOID token,  _In_ size_t size);
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSourceFileTokens(
@@ -2950,6 +3083,7 @@ SymEnumSourceFileTokens(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymInitialize(
@@ -2958,6 +3092,7 @@ SymInitialize(
     _In_ BOOL fInvadeProcess
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymInitializeW(
@@ -2978,6 +3113,7 @@ SymInitializeW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSearchPath(
@@ -2986,6 +3122,7 @@ SymGetSearchPath(
     _In_ DWORD SearchPathLength
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSearchPathW(
@@ -2994,6 +3131,7 @@ SymGetSearchPathW(
     _In_ DWORD SearchPathLength
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetSearchPath(
@@ -3007,6 +3145,7 @@ SymSetSearchPath(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetSearchPathW(
@@ -3024,6 +3163,7 @@ SymSetSearchPathW(
 #define SLMFLAG_ALT_INDEX   0x2
 #define SLMFLAG_NO_SYMBOLS  0x4
 
+_Win32_SetLastError_
 DWORD64
 IMAGEAPI
 SymLoadModuleEx(
@@ -3043,6 +3183,7 @@ SymLoadModuleEx(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 DWORD64
 IMAGEAPI
 SymLoadModuleExW(
@@ -3062,6 +3203,7 @@ SymLoadModuleExW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymUnloadModule64(
@@ -3072,6 +3214,7 @@ SymUnloadModule64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymUnloadModule SymUnloadModule64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymUnloadModule(
@@ -3080,6 +3223,7 @@ SymUnloadModule(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymUnDName64(
@@ -3091,6 +3235,7 @@ SymUnDName64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymUnDName SymUnDName64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymUnDName(
@@ -3100,6 +3245,7 @@ SymUnDName(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRegisterCallback64(
@@ -3108,6 +3254,7 @@ SymRegisterCallback64(
     _In_ ULONG64 UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRegisterCallbackW64(
@@ -3116,6 +3263,7 @@ SymRegisterCallbackW64(
     _In_ ULONG64 UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRegisterFunctionEntryCallback64(
@@ -3128,6 +3276,7 @@ SymRegisterFunctionEntryCallback64(
 #define SymRegisterCallback SymRegisterCallback64
 #define SymRegisterFunctionEntryCallback SymRegisterFunctionEntryCallback64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRegisterCallback(
@@ -3136,6 +3285,7 @@ SymRegisterCallback(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRegisterFunctionEntryCallback(
@@ -3233,6 +3383,7 @@ typedef struct _IMAGEHLP_STACK_FRAME
 typedef VOID IMAGEHLP_CONTEXT, *PIMAGEHLP_CONTEXT;
 
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetContext(
@@ -3241,6 +3392,7 @@ SymSetContext(
     _In_opt_ PIMAGEHLP_CONTEXT Context
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetScopeFromAddr(
@@ -3248,6 +3400,7 @@ SymSetScopeFromAddr(
     _In_ ULONG64 Address
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetScopeFromInlineContext(
@@ -3256,6 +3409,7 @@ SymSetScopeFromInlineContext(
     _In_ ULONG InlineContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSetScopeFromIndex(
@@ -3270,6 +3424,7 @@ typedef BOOL
     _In_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumProcesses(
@@ -3283,6 +3438,7 @@ SymEnumProcesses(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromAddr(
@@ -3292,6 +3448,7 @@ SymFromAddr(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromAddrW(
@@ -3307,6 +3464,7 @@ SymFromAddrW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromInlineContext(
@@ -3317,6 +3475,7 @@ SymFromInlineContext(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromInlineContextW(
@@ -3327,6 +3486,7 @@ SymFromInlineContextW(
     _Inout_ PSYMBOL_INFOW Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromToken(
@@ -3336,6 +3496,7 @@ SymFromToken(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromTokenW(
@@ -3345,6 +3506,7 @@ SymFromTokenW(
     _Inout_ PSYMBOL_INFOW Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymNext(
@@ -3352,6 +3514,7 @@ SymNext(
     _Inout_ PSYMBOL_INFO si
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymNextW(
@@ -3359,6 +3522,7 @@ SymNextW(
     _Inout_ PSYMBOL_INFOW siw
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymPrev(
@@ -3366,6 +3530,7 @@ SymPrev(
     _Inout_ PSYMBOL_INFO si
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymPrevW(
@@ -3379,6 +3544,7 @@ SymPrevW(
 // expressions.  That way you can search across modules
 // and differentiate between identically named symbols.
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromName(
@@ -3387,6 +3553,7 @@ SymFromName(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromNameW(
@@ -3405,6 +3572,7 @@ typedef BOOL
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbols(
@@ -3415,6 +3583,7 @@ SymEnumSymbols(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbolsEx(
@@ -3433,6 +3602,7 @@ typedef BOOL
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbolsW(
@@ -3443,6 +3613,7 @@ SymEnumSymbolsW(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbolsExW(
@@ -3454,6 +3625,7 @@ SymEnumSymbolsExW(
     _In_ DWORD Options
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbolsForAddr(
@@ -3463,6 +3635,7 @@ SymEnumSymbolsForAddr(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumSymbolsForAddrW(
@@ -3477,6 +3650,7 @@ SymEnumSymbolsForAddrW(
 #define SYMSEARCH_GLOBALSONLY   0X04    // search only for global symbols
 #define SYMSEARCH_ALLITEMS      0X08    // search for everything in the pdb, not just normal scoped symbols
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSearch(
@@ -3491,6 +3665,7 @@ SymSearch(
     _In_ DWORD Options
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSearchW(
@@ -3505,6 +3680,7 @@ SymSearchW(
     _In_ DWORD Options
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetScope(
@@ -3514,6 +3690,7 @@ SymGetScope(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetScopeW(
@@ -3523,6 +3700,7 @@ SymGetScopeW(
     _Inout_ PSYMBOL_INFOW Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromIndex(
@@ -3532,6 +3710,7 @@ SymFromIndex(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymFromIndexW(
@@ -3601,6 +3780,7 @@ typedef struct _TI_GET_DISCRIMINATEDUNION_TAG_RANGES_PARAMS {
     DISCRIMINATEDUNION_TAG_VALUE Range[1];
 } TI_GET_DISCRIMINATEDUNION_TAG_RANGES_PARAMS;
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetTypeInfo(
@@ -3635,6 +3815,7 @@ typedef struct _IMAGEHLP_GET_TYPE_INFO_PARAMS {
     OUT PULONG64 ReqsValid OPTIONAL;
 } IMAGEHLP_GET_TYPE_INFO_PARAMS, *PIMAGEHLP_GET_TYPE_INFO_PARAMS;
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetTypeInfoEx(
@@ -3643,6 +3824,7 @@ SymGetTypeInfoEx(
     _Inout_ PIMAGEHLP_GET_TYPE_INFO_PARAMS Params
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumTypes(
@@ -3652,6 +3834,7 @@ SymEnumTypes(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumTypesW(
@@ -3661,6 +3844,7 @@ SymEnumTypesW(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumTypesByName(
@@ -3671,6 +3855,7 @@ SymEnumTypesByName(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymEnumTypesByNameW(
@@ -3681,6 +3866,7 @@ SymEnumTypesByNameW(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetTypeFromName(
@@ -3690,6 +3876,7 @@ SymGetTypeFromName(
     _Inout_ PSYMBOL_INFO Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetTypeFromNameW(
@@ -3699,6 +3886,7 @@ SymGetTypeFromNameW(
     _Inout_ PSYMBOL_INFOW Symbol
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymAddSymbol(
@@ -3710,6 +3898,7 @@ SymAddSymbol(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymAddSymbolW(
@@ -3721,6 +3910,7 @@ SymAddSymbolW(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymDeleteSymbol(
@@ -3731,6 +3921,7 @@ SymDeleteSymbol(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymDeleteSymbolW(
@@ -3747,6 +3938,7 @@ SymDeleteSymbolW(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymRefreshModuleList(
@@ -3759,6 +3951,7 @@ SymRefreshModuleList(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymAddSourceStream(
@@ -3783,6 +3976,7 @@ SymAddSourceStreamA(
 
 typedef BOOL (WINAPI *SYMADDSOURCESTREAMA)(HANDLE, ULONG64, PCSTR, PBYTE, size_t);
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymAddSourceStreamW(
@@ -3793,6 +3987,7 @@ SymAddSourceStreamW(
     _In_ size_t Size
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvIsStoreW(
@@ -3800,6 +3995,7 @@ SymSrvIsStoreW(
     _In_ PCWSTR path
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvIsStore(
@@ -3807,6 +4003,7 @@ SymSrvIsStore(
     _In_ PCSTR path
     );
 
+_Win32_SetLastError_
 PCSTR
 IMAGEAPI
 SymSrvDeltaName(
@@ -3817,6 +4014,7 @@ SymSrvDeltaName(
     _In_ PCSTR File2
     );
 
+_Win32_SetLastError_
 PCWSTR
 IMAGEAPI
 SymSrvDeltaNameW(
@@ -3827,6 +4025,7 @@ SymSrvDeltaNameW(
     _In_ PCWSTR File2
     );
 
+_Win32_SetLastError_
 PCSTR
 IMAGEAPI
 SymSrvGetSupplement(
@@ -3836,6 +4035,7 @@ SymSrvGetSupplement(
     _In_ PCSTR File
     );
 
+_Win32_SetLastError_
 PCWSTR
 IMAGEAPI
 SymSrvGetSupplementW(
@@ -3845,6 +4045,7 @@ SymSrvGetSupplementW(
     _In_ PCWSTR File
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexes(
@@ -3855,6 +4056,7 @@ SymSrvGetFileIndexes(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexesW(
@@ -3865,6 +4067,7 @@ SymSrvGetFileIndexesW(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexStringW(
@@ -3876,6 +4079,7 @@ SymSrvGetFileIndexStringW(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexString(
@@ -3913,6 +4117,7 @@ typedef struct {
     DWORD age;
 } SYMSRV_INDEX_INFOW, *PSYMSRV_INDEX_INFOW;
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexInfo(
@@ -3921,6 +4126,7 @@ SymSrvGetFileIndexInfo(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymSrvGetFileIndexInfoW(
@@ -3929,6 +4135,7 @@ SymSrvGetFileIndexInfoW(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 PCSTR
 IMAGEAPI
 SymSrvStoreSupplement(
@@ -3939,6 +4146,7 @@ SymSrvStoreSupplement(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 PCWSTR
 IMAGEAPI
 SymSrvStoreSupplementW(
@@ -3949,6 +4157,7 @@ SymSrvStoreSupplementW(
     _In_ DWORD Flags
     );
 
+_Win32_SetLastError_
 PCSTR
 IMAGEAPI
 SymSrvStoreFile(
@@ -3964,6 +4173,7 @@ SymSrvStoreFile(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 PCWSTR
 IMAGEAPI
 SymSrvStoreFileW(
@@ -3989,6 +4199,7 @@ typedef enum {
     sfMax
 } IMAGEHLP_SF_TYPE;
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymbolFile(
@@ -4002,6 +4213,7 @@ SymGetSymbolFile(
     _In_ size_t cDbgFile
     );
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymbolFileW(
@@ -4053,6 +4265,7 @@ DbgHelpCreateUserDumpW(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(NONGAMESPARTITIONS | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymFromAddr64(
@@ -4066,6 +4279,7 @@ SymGetSymFromAddr64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetSymFromAddr SymGetSymFromAddr64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymFromAddr(
@@ -4088,6 +4302,7 @@ SymGetSymFromAddr(
 // expressions.  That way you can search across modules
 // and differentiate between identically named symbols.
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymFromName64(
@@ -4099,6 +4314,7 @@ SymGetSymFromName64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymGetSymFromName SymGetSymFromName64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymFromName(
@@ -4393,6 +4609,7 @@ SymEnumSym(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 DBHLP_DEPRECIATED
 BOOL
 IMAGEAPI
@@ -4403,6 +4620,7 @@ SymEnumerateSymbols64(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 DBHLP_DEPRECIATED
 BOOL
 IMAGEAPI
@@ -4418,6 +4636,7 @@ SymEnumerateSymbolsW64(
 #define SymEnumerateSymbols SymEnumerateSymbols64
 #define SymEnumerateSymbolsW SymEnumerateSymbolsW64
 #else
+_Win32_SetLastError_
 DBHLP_DEPRECIATED
 BOOL
 IMAGEAPI
@@ -4428,6 +4647,7 @@ SymEnumerateSymbols(
     _In_opt_ PVOID UserContext
     );
 
+_Win32_SetLastError_
 DBHLP_DEPRECIATED
 BOOL
 IMAGEAPI
@@ -4441,6 +4661,7 @@ SymEnumerateSymbolsW(
 
 // use SymLoadModuleEx
 
+_Win32_SetLastError_
 DWORD64
 IMAGEAPI
 SymLoadModule64(
@@ -4455,6 +4676,7 @@ SymLoadModule64(
 #if !defined(_IMAGEHLP_SOURCE_) && defined(_IMAGEHLP64)
 #define SymLoadModule SymLoadModule64
 #else
+_Win32_SetLastError_
 DWORD
 IMAGEAPI
 SymLoadModule(
@@ -4467,6 +4689,7 @@ SymLoadModule(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymNext64(
@@ -4485,6 +4708,7 @@ SymGetSymNextW64(
 #define SymGetSymNext SymGetSymNext64
 #define SymGetSymNextW SymGetSymNextW64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymNext(
@@ -4500,6 +4724,7 @@ SymGetSymNextW(
     );
 #endif
 
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymPrev64(
@@ -4518,6 +4743,7 @@ SymGetSymPrevW64(
 #define SymGetSymPrev SymGetSymPrev64
 #define SymGetSymPrevW SymGetSymPrevW64
 #else
+_Win32_SetLastError_
 BOOL
 IMAGEAPI
 SymGetSymPrev(
