diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/immdev.h b/generation/WinSDK/RecompiledIdlHeaders/um/immdev.h
index 5295f27001..ae7b582925 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/immdev.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/immdev.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**********************************************************************/
 /*      immdev.h - Input Method Manager definitions                   */
 /*                 for IME developers                                 */
@@ -27,8 +28,8 @@ extern "C" {
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
 #if (WINVER >= 0x040A)
-DECLARE_HANDLE(HIMC);
-DECLARE_HANDLE(HIMCC);
+DECLARE_HANDLE(HIMC) _WIN32META_ANNOTATION_("win32metadata:raii_free=ImmDestroyContext")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(HIMCC) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
 #else
 typedef DWORD   HIMC;
 typedef DWORD   HIMCC;
@@ -909,7 +910,7 @@ typedef struct tagSOFTKBDDATA {
     WORD        wCode[1][256];
 } SOFTKBDDATA, *PSOFTKBDDATA, NEAR *NPSOFTKBDDATA, FAR * LPSOFTKBDDATA;
 
-BOOL WINAPI ImmGetHotKey(IN DWORD, _Out_ LPUINT lpuModifiers, _Out_ LPUINT lpuVKey, _Out_ LPHKL phKL);
+BOOL WINAPI ImmGetHotKey(IN DWORD, _Out_ LPUINT lpuModifiers, _Out_ LPUINT lpuVKey, _Win32_RAIIFree_(UnloadKeyboardLayout, -1, 0) _Out_ LPHKL phKL);
 BOOL WINAPI ImmSetHotKey(IN DWORD, IN UINT, IN UINT, IN HKL);
 BOOL WINAPI ImmGenerateMessage(IN HIMC);
 
@@ -921,8 +922,8 @@ BOOL WINAPI ImmGenerateMessage(IN HIMC);
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
-LRESULT WINAPI ImmRequestMessageA(IN HIMC, IN WPARAM, IN LPARAM);
-LRESULT WINAPI ImmRequestMessageW(IN HIMC, IN WPARAM, IN LPARAM);
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ LRESULT WINAPI ImmRequestMessageA(IN HIMC, IN WPARAM, IN LPARAM);
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ LRESULT WINAPI ImmRequestMessageW(IN HIMC, IN WPARAM, IN LPARAM);
 #ifdef UNICODE
 #define ImmRequestMessage  ImmRequestMessageW
 #else
