diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/inked.h b/generation/WinSDK/RecompiledIdlHeaders/um/inked.h
index 395086642e..3c02261a28 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/inked.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/inked.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -365,7 +366,7 @@ EXTERN_C const IID IID_IInkEdit;
             /* [in] */ __RPC__in_opt IInkRecognizer *newVal) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Factoid( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pVal) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pVal) = 0;
         
         virtual /* [helpcontext][helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_Factoid( 
             /* [in] */ __RPC__in BSTR newVal) = 0;
@@ -420,7 +421,7 @@ EXTERN_C const IID IID_IInkEdit;
             /* [in] */ __RPC__in_opt IFontDisp *ppFont) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Text( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrText) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrText) = 0;
         
         virtual /* [helpcontext][helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_Text( 
             /* [in] */ __RPC__in BSTR pbstrText) = 0;
@@ -525,7 +526,7 @@ EXTERN_C const IID IID_IInkEdit;
             /* [in] */ VARIANT pvarSelCharOffset) = 0;
         
         virtual /* [helpcontext][helpstring][defaultbind][displaybind][bindable][propget][id] */ HRESULT STDMETHODCALLTYPE get_TextRTF( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrTextRTF) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrTextRTF) = 0;
         
         virtual /* [helpcontext][helpstring][displaybind][bindable][propput][id] */ HRESULT STDMETHODCALLTYPE put_TextRTF( 
             /* [in] */ __RPC__in BSTR pbstrTextRTF) = 0;
@@ -543,13 +544,13 @@ EXTERN_C const IID IID_IInkEdit;
             /* [in] */ long plSelLength) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_SelText( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrSelText) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrSelText) = 0;
         
         virtual /* [helpcontext][helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_SelText( 
             /* [in] */ __RPC__in BSTR pbstrSelText) = 0;
         
         virtual /* [helpcontext][helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_SelRTF( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrSelRTF) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrSelRTF) = 0;
         
         virtual /* [helpcontext][helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_SelRTF( 
             /* [in] */ __RPC__in BSTR pbstrSelRTF) = 0;
