diff --git a/generation/WinSDK/RecompiledIdlHeaders/winrt/inspectable.h b/generation/WinSDK/RecompiledIdlHeaders/winrt/inspectable.h
index 9b79b49792..1a5eb0ad3a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/winrt/inspectable.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/winrt/inspectable.h
@@ -19,6 +19,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -79,7 +80,7 @@ extern "C"{
 typedef /* [unique] */  __RPC_unique_pointer IInspectable *LPINSPECTABLE;
 
 typedef /* [v1_enum] */ 
-enum TrustLevel
+enum _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ TrustLevel
     {
         BaseTrust	= 0,
         PartialTrust	= ( BaseTrust + 1 ) ,
@@ -106,13 +107,19 @@ EXTERN_C const IID IID_IInspectable;
     IInspectable : public IUnknown
     {
     public:
+        _Windows_SupportedOS_Windows8_
+        _Windows_SupportedOS_Server2012_
         virtual HRESULT STDMETHODCALLTYPE GetIids( 
             /* [out] */ __RPC__out ULONG *iidCount,
-            /* [size_is][size_is][out] */ __RPC__deref_out_ecount_full_opt(*iidCount) IID **iids) = 0;
+            /* [size_is][size_is][out] */ __RPC__deref_out_ecount_full_opt(*iidCount) _Win32_RAIIFree_(CoTaskMemFree) IID **iids) = 0;
         
+        _Windows_SupportedOS_Windows8_
+        _Windows_SupportedOS_Server2012_
         virtual HRESULT STDMETHODCALLTYPE GetRuntimeClassName( 
-            /* [out] */ __RPC__deref_out_opt HSTRING *className) = 0;
+            /* [out] */ __RPC__deref_out_opt _Win32_RAIIFree_(WindowsDeleteString) HSTRING *className) = 0;
         
+        _Windows_SupportedOS_Windows8_
+        _Windows_SupportedOS_Server2012_
         virtual HRESULT STDMETHODCALLTYPE GetTrustLevel( 
             /* [out] */ __RPC__out TrustLevel *trustLevel) = 0;
         
@@ -144,12 +151,12 @@ EXTERN_C const IID IID_IInspectable;
         HRESULT ( STDMETHODCALLTYPE *GetIids )( 
             __RPC__in IInspectable * This,
             /* [out] */ __RPC__out ULONG *iidCount,
-            /* [size_is][size_is][out] */ __RPC__deref_out_ecount_full_opt(*iidCount) IID **iids);
+            /* [size_is][size_is][out] */ __RPC__deref_out_ecount_full_opt(*iidCount) _Win32_RAIIFree_(CoTaskMemFree) IID **iids);
         
         DECLSPEC_XFGVIRT(IInspectable, GetRuntimeClassName)
         HRESULT ( STDMETHODCALLTYPE *GetRuntimeClassName )( 
             __RPC__in IInspectable * This,
-            /* [out] */ __RPC__deref_out_opt HSTRING *className);
+            /* [out] */ __RPC__deref_out_opt _Win32_RAIIFree_(WindowsDeleteString) HSTRING *className);
         
         DECLSPEC_XFGVIRT(IInspectable, GetTrustLevel)
         HRESULT ( STDMETHODCALLTYPE *GetTrustLevel )( 
@@ -227,14 +234,30 @@ extern RPC_IF_HANDLE __MIDL_itf_inspectable_0000_0001_v0_0_s_ifspec;
 
 /* Additional Prototypes for ALL interfaces */
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned long             __RPC_USER  HSTRING_UserSize(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned char * __RPC_USER  HSTRING_UserMarshal(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 void                      __RPC_USER  HSTRING_UserFree(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned long             __RPC_USER  HSTRING_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned char * __RPC_USER  HSTRING_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 unsigned char * __RPC_USER  HSTRING_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HSTRING * ); 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 void                      __RPC_USER  HSTRING_UserFree64(     __RPC__in unsigned long *, __RPC__in HSTRING * ); 
 
 /* end of Additional Prototypes */
