diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ioapiset.h b/generation/WinSDK/RecompiledIdlHeaders/um/ioapiset.h
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ioapiset.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ioapiset.h
@@ -18,6 +18,9 @@
 #include <minwindef.h>
 #include <minwinbase.h>
 
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -25,6 +28,7 @@ extern "C" {
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -36,6 +40,7 @@ CreateIoCompletionPort(
     _In_ DWORD NumberOfConcurrentThreads
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -49,6 +54,7 @@ GetQueuedCompletionStatus(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -63,6 +69,7 @@ GetQueuedCompletionStatusEx(
 
 #endif // _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -79,6 +86,7 @@ PostQueuedCompletionStatus(
 #pragma region  Desktop Family or OneCore Family or Application Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -99,6 +107,7 @@ DeviceIoControl(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -111,6 +120,7 @@ GetOverlappedResult(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -127,6 +137,7 @@ CancelIoEx(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -134,6 +145,7 @@ CancelIo(
     _In_ HANDLE hFile
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -153,6 +165,7 @@ GetOverlappedResultEx(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
