diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/joystickapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/joystickapi.h
index 57d6e6d0f6..38e5973185 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/joystickapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/joystickapi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /********************************************************************************
 *                                                                               *
 * joystickapi.h -- ApiSet Contract for api-ms-win-mm-joystick-l1-1-0            *
@@ -129,7 +130,7 @@ extern "C" {
 /* joystick device capabilities data structure */
 #ifdef _WIN32
 
-typedef struct tagJOYCAPSA {
+typedef struct   tagJOYCAPSA {
     WORD    wMid;                /* manufacturer ID */
     WORD    wPid;                /* product ID */
     CHAR    szPname[MAXPNAMELEN];/* product name (NULL terminated string) */
@@ -157,7 +158,7 @@ typedef struct tagJOYCAPSA {
     CHAR    szOEMVxD[MAX_JOYSTICKOEMVXDNAME]; /* OEM VxD in use */
 #endif
 } JOYCAPSA, *PJOYCAPSA, *NPJOYCAPSA, *LPJOYCAPSA;
-typedef struct tagJOYCAPSW {
+typedef struct   tagJOYCAPSW {
     WORD    wMid;                /* manufacturer ID */
     WORD    wPid;                /* product ID */
     WCHAR   szPname[MAXPNAMELEN];/* product name (NULL terminated string) */
@@ -298,7 +299,7 @@ typedef struct joycaps_tag {
 #endif
 
 /* joystick information data structure */
-typedef struct joyinfo_tag {
+typedef struct   joyinfo_tag {
     UINT wXpos;                 /* x position */
     UINT wYpos;                 /* y position */
     UINT wZpos;                 /* z position */
@@ -306,7 +307,7 @@ typedef struct joyinfo_tag {
 } JOYINFO, *PJOYINFO, NEAR *NPJOYINFO, FAR *LPJOYINFO;
 
 #if (WINVER >= 0x0400)
-typedef struct joyinfoex_tag {
+typedef struct   joyinfoex_tag {
     DWORD dwSize;                /* size of structure */
     DWORD dwFlags;               /* flags to indicate what to return */
     DWORD dwXpos;                /* x position */
@@ -326,20 +327,20 @@ typedef struct joyinfoex_tag {
 /* joystick function prototypes */
 
 #if (WINVER >= 0x0400)
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyGetPosEx(
+  joyGetPosEx(
     _In_ UINT uJoyID,
     _Out_ LPJOYINFOEX pji
     );
 
 #endif /* WINVER >= 0x0400 */
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 UINT
 WINAPI
-joyGetNumDevs(
+  joyGetNumDevs(
     void
     );
 
@@ -348,16 +349,16 @@ joyGetNumDevs(
 WINMMAPI
 MMRESULT
 WINAPI
-joyGetDevCapsA(
+  joyGetDevCapsA(
     _In_ UINT_PTR uJoyID,
     _Out_writes_bytes_(cbjc) LPJOYCAPSA pjc,
     _In_ UINT cbjc
     );
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyGetDevCapsW(
+  joyGetDevCapsW(
     _In_ UINT_PTR uJoyID,
     _Out_writes_bytes_(cbjc) LPJOYCAPSW pjc,
     _In_ UINT cbjc
@@ -373,52 +374,52 @@ joyGetDevCapsW(
 MMRESULT WINAPI joyGetDevCaps(UINT uJoyID, LPJOYCAPS pjc, UINT cbjc);
 #endif
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyGetPos(
+  joyGetPos(
     _In_ UINT uJoyID,
     _Out_ LPJOYINFO pji
     );
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyGetThreshold(
+  joyGetThreshold(
     _In_ UINT uJoyID,
     _Out_ LPUINT puThreshold
     );
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyReleaseCapture(
+  joyReleaseCapture(
     _In_ UINT uJoyID
     );
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joySetCapture(
+  joySetCapture(
     _In_ HWND hwnd,
     _In_ UINT uJoyID,
     _In_ UINT uPeriod,
     _In_ BOOL fChanged
     );
 
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joySetThreshold(
+  joySetThreshold(
     _In_ UINT uJoyID,
     _In_ UINT uThreshold
     );
 
 #if (WINVER >= 0x0400)
-WINMMAPI
+_Windows_SupportedOS_Windows2000_ WINMMAPI
 MMRESULT
 WINAPI
-joyConfigChanged(
+  joyConfigChanged(
     _In_ DWORD dwFlags
     );
 
