diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/KeyStoreListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/KeyStoreListener.h
index bb62b99091..c0d2643268 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/KeyStoreListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/KeyStoreListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * The alljoyn_keystorelistener handles requests to load or store the key store.
@@ -58,7 +59,7 @@ typedef struct _alljoyn_keystorelistener_handle*            alljoyn_keystorelist
  *      - An error status otherwise
  *
  */
-typedef QStatus (AJ_CALL * alljoyn_keystorelistener_loadrequest_ptr)(const void* context, alljoyn_keystorelistener listener, alljoyn_keystore keyStore);
+typedef QStatus (AJ_CALL * alljoyn_keystorelistener_loadrequest_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystorelistener listener, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystore keyStore);
 
 /**
  * Type for the StoreRequest callback.
@@ -76,7 +77,7 @@ typedef QStatus (AJ_CALL * alljoyn_keystorelistener_loadrequest_ptr)(const void*
  *      - #ER_OK if the store request was satisfied
  *      - An error status otherwise
  */
-typedef QStatus (AJ_CALL * alljoyn_keystorelistener_storerequest_ptr)(const void* context, alljoyn_keystorelistener listener, alljoyn_keystore keyStore);
+typedef QStatus (AJ_CALL * alljoyn_keystorelistener_storerequest_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystorelistener listener, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystore keyStore);
 
 /**
  * Type for the AcquireExclusiveLock callback.
@@ -90,7 +91,7 @@ typedef QStatus (AJ_CALL * alljoyn_keystorelistener_storerequest_ptr)(const void
  *      - #ER_OK if the store request was satisfied
  *      - An error status otherwise
  */
-typedef QStatus (AJ_CALL * alljoyn_keystorelistener_acquireexclusivelock_ptr)(const void* context, alljoyn_keystorelistener listener);
+typedef QStatus (AJ_CALL * alljoyn_keystorelistener_acquireexclusivelock_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystorelistener listener);
 
 /**
  * Type for the ReleaseExclusiveLock callback.
@@ -100,7 +101,7 @@ typedef QStatus (AJ_CALL * alljoyn_keystorelistener_acquireexclusivelock_ptr)(co
  * @param context    The context pointer passed into the alljoyn_keystorelistener_with_synchronization_create function.
  * @param listener   The listener that is responsible for the ReleaseExclusiveLock callback.
  */
-typedef void (AJ_CALL * alljoyn_keystorelistener_releaseexclusivelock_ptr)(const void* context, alljoyn_keystorelistener listener);
+typedef void (AJ_CALL * alljoyn_keystorelistener_releaseexclusivelock_ptr)(const void* context, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_keystorelistener listener);
 
 /**
  * Structure used during alljoyn_keystorelistener_create to provide callbacks into C.
@@ -154,7 +155,7 @@ typedef struct {
  *
  * @return an allocated alljoyn_keystorelistener
  */
-extern AJ_API alljoyn_keystorelistener AJ_CALL alljoyn_keystorelistener_create(const alljoyn_keystorelistener_callbacks* callbacks,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_keystorelistener AJ_CALL alljoyn_keystorelistener_create(const alljoyn_keystorelistener_callbacks* callbacks,
                                                                                const void* context);
 
 /**
@@ -165,14 +166,14 @@ extern AJ_API alljoyn_keystorelistener AJ_CALL alljoyn_keystorelistener_create(c
  *
  * @return an allocated alljoyn_keystorelistener
  */
-extern AJ_API alljoyn_keystorelistener AJ_CALL alljoyn_keystorelistener_with_synchronization_create(const alljoyn_keystorelistener_with_synchronization_callbacks* callbacks,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_keystorelistener AJ_CALL alljoyn_keystorelistener_with_synchronization_create(const alljoyn_keystorelistener_with_synchronization_callbacks* callbacks,
                                                                                                     void* context);
 /**
  * Destroy an alljoyn_keystorelistener
  *
  * @param listener The alljoyn_keystorelistener to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_keystorelistener_destroy(alljoyn_keystorelistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_keystorelistener_destroy(alljoyn_keystorelistener listener);
 
 /**
  * Put keys into the key store from an encrypted byte string.
@@ -187,7 +188,7 @@ extern AJ_API void AJ_CALL alljoyn_keystorelistener_destroy(alljoyn_keystorelist
  *      - An error status otherwise
  *
  */
-extern AJ_API QStatus AJ_CALL alljoyn_keystorelistener_putkeys(alljoyn_keystorelistener listener, alljoyn_keystore keyStore,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_keystorelistener_putkeys(alljoyn_keystorelistener listener, alljoyn_keystore keyStore,
                                                                const char* source, const char* password);
 
 /**
@@ -219,7 +220,7 @@ extern AJ_API QStatus AJ_CALL alljoyn_keystorelistener_putkeys(alljoyn_keystorel
  *        to hold the sink
  *      - An error status otherwise
  */
-extern AJ_API QStatus AJ_CALL alljoyn_keystorelistener_getkeys(alljoyn_keystorelistener listener, alljoyn_keystore keyStore,
+_Win32_ImportLibrary_("msajapi.dll") _Win32_AssociatedEnum_(QStatus) extern AJ_API QStatus AJ_CALL alljoyn_keystorelistener_getkeys(alljoyn_keystorelistener listener, alljoyn_keystore keyStore,
                                                                char* sink, size_t* sink_sz);
 
 #ifdef __cplusplus
