diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/ksproxy.h b/generation/WinSDK/RecompiledIdlHeaders/shared/ksproxy.h
index 1f0b101176..b28108f774 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/ksproxy.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/ksproxy.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation. All rights reserved.
@@ -321,7 +322,7 @@ interface DECLSPEC_UUID("8da64899-c0d9-11d0-8413-0000f822fe8a") IKsAllocator;
 #define INTERFACE IKsAllocator
 DECLARE_INTERFACE_(IKsAllocator, IUnknown)
 {
-    STDMETHOD_(HANDLE, KsGetAllocatorHandle)(
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, KsGetAllocatorHandle)(
         THIS
     ) PURE;
     STDMETHOD_(KSALLOCATORMODE, KsGetAllocatorMode)(
@@ -353,7 +354,7 @@ DECLARE_INTERFACE_(IKsAllocatorEx, IKsAllocator)
         THIS_
         _In_ HANDLE AllocatorHandle
     ) PURE;
-    STDMETHOD_(HANDLE, KsCreateAllocatorAndGetHandle)(
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, KsCreateAllocatorAndGetHandle)(
         THIS_
         _In_ IKsPin*   KsPin
     ) PURE;
@@ -484,10 +485,10 @@ DECLARE_INTERFACE_(IKsPinPipe, IUnknown)
 //
 // very useful methods for tracing.
 //
-    STDMETHOD_(PWCHAR, KsGetPinName)(
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") STDMETHOD_(PWCHAR, KsGetPinName)(
         THIS
     ) PURE;
-    STDMETHOD_(PWCHAR, KsGetFilterName)(
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") STDMETHOD_(PWCHAR, KsGetFilterName)(
         THIS
     ) PURE;
 };
@@ -598,7 +599,7 @@ interface DECLSPEC_UUID("423c13a2-2070-11d0-9ef7-00aa00a216a1") IKsObject;
 #define INTERFACE IKsObject
 DECLARE_INTERFACE_(IKsObject, IUnknown)
 {
-    STDMETHOD_(HANDLE, KsGetObjectHandle)(
+    _Win32_RAIIFree_(CloseHandle, -1, 0) STDMETHOD_(HANDLE, KsGetObjectHandle)(
         THIS
     ) PURE;
 };
@@ -645,7 +646,7 @@ WINAPI
 KsOpenDefaultDevice(
     _In_ REFGUID Category,
     _In_ ACCESS_MASK Access,
-    _Out_ PHANDLE DeviceHandle
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE DeviceHandle
     );
 KSDDKAPI
 HRESULT
