diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
index c8e8e79cbe..b096cf986a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/libloaderapi2.h
@@ -17,6 +17,7 @@
 #include <apisetcconv.h>
 #include <minwindef.h>
 #include <minwinbase.h>
+#include "win32metadata_annotations.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -33,7 +34,7 @@ HMODULE
 WINAPI
 LoadPackagedLibrary(
     _In_ LPCWSTR lpwLibFileName,
-    _Reserved_ DWORD Reserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ DWORD Reserved
     );
 
 #endif // _WIN32_WINNT >= 0x0602
@@ -46,6 +47,9 @@ LoadPackagedLibrary(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Windows_SupportedOS_10240_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -53,7 +57,7 @@ QueryOptionalDelayLoadedAPI(
     _In_ HMODULE hParentModule,
     _In_ LPCSTR lpDllName,
     _In_ LPCSTR lpProcName,
-    _Reserved_ DWORD Reserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ DWORD Reserved
     );
 
 #endif // _WIN32_WINNT >= 0x0602
