diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
index 0150fb5596..4eaad77e32 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMat.h
@@ -33,6 +33,7 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -91,7 +92,7 @@ extern "C" {
 
 
 
-typedef struct _AT_INFO {
+typedef struct   _AT_INFO {
     DWORD_PTR   JobTime;
     DWORD   DaysOfMonth;
     UCHAR   DaysOfWeek;
@@ -99,7 +100,7 @@ typedef struct _AT_INFO {
     LPWSTR  Command;
 } AT_INFO, *PAT_INFO, *LPAT_INFO;
 
-typedef struct _AT_ENUM {
+typedef struct   _AT_ENUM {
     DWORD   JobId;
     DWORD_PTR   JobTime;
     DWORD   DaysOfMonth;
@@ -108,13 +109,15 @@ typedef struct _AT_ENUM {
     LPWSTR  Command;
 } AT_ENUM, *PAT_ENUM, *LPAT_ENUM;
 
+_Windows_SupportedOS_Windows2000_  
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobAdd(
     IN      LPCWSTR         Servername  OPTIONAL,
-    IN      LPBYTE          Buffer,
-    OUT     LPDWORD         JobId
+    _WIN32META_ANNOTATION_("win32metadata:out") IN      LPBYTE          Buffer,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT     LPDWORD         JobId
     );
 
+_Windows_SupportedOS_Windows2000_  
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobDel(
     IN      LPCWSTR         Servername  OPTIONAL,
@@ -122,21 +125,23 @@ NetScheduleJobDel(
     IN      DWORD           MaxJobId
     );
 
+_Windows_SupportedOS_Windows2000_  
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobEnum(
     IN      LPCWSTR         Servername              OPTIONAL,
-    OUT     LPBYTE *        PointerToBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT      LPBYTE *        PointerToBuffer,
     IN      DWORD           PrefferedMaximumLength,
-    OUT     LPDWORD         EntriesRead,
-    OUT     LPDWORD         TotalEntries,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT     LPDWORD         EntriesRead,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT     LPDWORD         TotalEntries,
     IN OUT  LPDWORD         ResumeHandle
     );
 
+_Windows_SupportedOS_Windows2000_  
 NET_API_STATUS NET_API_FUNCTION
 NetScheduleJobGetInfo(
     IN      LPCWSTR         Servername              OPTIONAL,
     IN      DWORD           JobId,
-    OUT     LPBYTE *        PointerToBuffer
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT      LPBYTE *        PointerToBuffer
     );
 
 #ifdef __cplusplus
