diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LMaudit.h b/generation/WinSDK/RecompiledIdlHeaders/um/LMaudit.h
index b80506543f..b83e68d93f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LMaudit.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LMaudit.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++ BUILD Version: 0003    // Increment this if a change has global effects
 
 Copyright (c) 1991-1999  Microsoft Corporation
@@ -74,21 +75,21 @@ NET_API_STATUS NET_API_FUNCTION
 NetAuditRead (
     IN  LPCWSTR  server OPTIONAL,
     IN  LPCWSTR  service OPTIONAL,  // WARNING: buggy support before LM 2.0C!!
-    IN  LPHLOG   auditloghandle,
+    _WIN32META_ANNOTATION_("win32metadata:out") IN  LPHLOG   auditloghandle,
     IN  DWORD    offset,
     IN  LPDWORD  reserved1 OPTIONAL,
     IN  DWORD   reserved2,
     IN  DWORD   offsetflag,
-    OUT LPBYTE  *bufptr,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT LPBYTE  *bufptr,
     IN  DWORD   prefmaxlen,
-    OUT LPDWORD bytesread,
-    OUT LPDWORD totalavailable
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT LPDWORD bytesread,
+    _WIN32META_ANNOTATION_("win32metadata:in") OUT LPDWORD totalavailable
     );
 
 NET_API_STATUS NET_API_FUNCTION
 NetAuditWrite (
     IN  DWORD    type,
-    IN  LPBYTE   buf,
+    _WIN32META_ANNOTATION_("win32metadata:out") IN  LPBYTE   buf,
     IN  DWORD    numbytes,
     IN  LPCWSTR  service OPTIONAL,
     IN  LPBYTE   reserved OPTIONAL
