diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/LsaLookup.h b/generation/WinSDK/RecompiledIdlHeaders/um/LsaLookup.h
index 86e45aa74e..28afa4b10c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/LsaLookup.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/LsaLookup.h
@@ -21,6 +21,7 @@ Abstract:
 #define _LSALOOKUP_
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family or OneCore Family 
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -49,7 +50,7 @@ typedef struct _LSA_UNICODE_STRING {
 typedef struct _LSA_STRING {
     USHORT Length;
     USHORT MaximumLength;
-    PCHAR Buffer;
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") PCHAR Buffer;
 } LSA_STRING, *PLSA_STRING;
 
 typedef struct _LSA_OBJECT_ATTRIBUTES {
@@ -67,7 +68,7 @@ typedef struct _LSA_OBJECT_ATTRIBUTES {
 // The following data type is used to identify a domain
 //
 
-typedef struct _LSA_TRUST_INFORMATION {
+typedef struct   _LSA_TRUST_INFORMATION {
 
     LSA_UNICODE_STRING Name;
     PSID Sid;
@@ -86,7 +87,7 @@ typedef struct _LSA_TRUST_INFORMATION {
 // describe the domains referenced in the lookup operation.
 //
 
-typedef struct _LSA_REFERENCED_DOMAIN_LIST {
+typedef struct   _LSA_REFERENCED_DOMAIN_LIST {
 
     ULONG Entries;
     PLSA_TRUST_INFORMATION Domains;
@@ -108,7 +109,7 @@ typedef struct _LSA_REFERENCED_DOMAIN_LIST {
 //
 
 #if (_WIN32_WINNT >= 0x0501)
-typedef struct _LSA_TRANSLATED_SID2 {
+typedef struct   _LSA_TRANSLATED_SID2 {
 
     SID_NAME_USE Use;
     PSID         Sid;
@@ -139,7 +140,7 @@ typedef struct _LSA_TRANSLATED_SID2 {
 // describe the domains referenced in the lookup operation.
 //
 
-typedef struct _LSA_TRANSLATED_NAME {
+typedef struct   _LSA_TRANSLATED_NAME {
 
     SID_NAME_USE Use;
     LSA_UNICODE_STRING Name;
@@ -169,7 +170,7 @@ typedef struct _LSA_TRANSLATED_NAME {
 // (corresponds to the PolicyAccountDomainInformation information class).
 //
 
-typedef struct _POLICY_ACCOUNT_DOMAIN_INFO {
+typedef struct   _POLICY_ACCOUNT_DOMAIN_INFO {
 
     LSA_UNICODE_STRING DomainName;
     PSID DomainSid;
@@ -188,7 +189,7 @@ typedef struct _POLICY_ACCOUNT_DOMAIN_INFO {
 // information class
 //
 
-typedef struct _POLICY_DNS_DOMAIN_INFO
+typedef struct   _POLICY_DNS_DOMAIN_INFO
 {
     LSA_UNICODE_STRING Name;
     LSA_UNICODE_STRING DnsDomainName;
