diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mileffects.h b/generation/WinSDK/RecompiledIdlHeaders/um/mileffects.h
index 292e3ef5ac..78544e9437 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mileffects.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mileffects.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -320,7 +321,7 @@ EXTERN_C const IID IID_IMILBitmapEffectConnectorInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("F66D2E4B-B46B-42FC-859E-3DA0ECDB3C43")
+    MIDL_INTERFACE("F66D2E4B-B46B-42FC-859E-3DA0ECDB3C43") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectConnectorInfo : public IUnknown
     {
     public:
@@ -450,7 +451,7 @@ EXTERN_C const IID IID_IMILBitmapEffectConnectionsInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("476B538A-C765-4237-BA4A-D6A880FF0CFC")
+    MIDL_INTERFACE("476B538A-C765-4237-BA4A-D6A880FF0CFC") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectConnectionsInfo : public IUnknown
     {
     public:
@@ -584,7 +585,7 @@ EXTERN_C const IID IID_IMILBitmapEffectConnections;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("C2B5D861-9B1A-4374-89B0-DEC4874D6A81")
+    MIDL_INTERFACE("C2B5D861-9B1A-4374-89B0-DEC4874D6A81") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectConnections : public IUnknown
     {
     public:
@@ -692,7 +693,7 @@ EXTERN_C const IID IID_IMILBitmapEffect;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("8A6FF321-C944-4A1B-9944-9954AF301258")
+    MIDL_INTERFACE("8A6FF321-C944-4A1B-9944-9954AF301258") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffect : public IUnknown
     {
     public:
@@ -817,7 +818,7 @@ EXTERN_C const IID IID_IMILBitmapEffectImpl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("CC2468F2-9936-47BE-B4AF-06B5DF5DBCBB")
+    MIDL_INTERFACE("CC2468F2-9936-47BE-B4AF-06B5DF5DBCBB") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectImpl : public IUnknown
     {
     public:
@@ -1018,7 +1019,7 @@ EXTERN_C const IID IID_IMILBitmapEffectGroup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("2F952360-698A-4ac6-81A1-BCFDF08EB8E8")
+    MIDL_INTERFACE("2F952360-698A-4ac6-81A1-BCFDF08EB8E8") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectGroup : public IUnknown
     {
     public:
@@ -1139,7 +1140,7 @@ EXTERN_C const IID IID_IMILBitmapEffectGroupImpl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("78FED518-1CFC-4807-8B85-6B6E51398F62")
+    MIDL_INTERFACE("78FED518-1CFC-4807-8B85-6B6E51398F62") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectGroupImpl : public IUnknown
     {
     public:
@@ -1252,7 +1253,7 @@ EXTERN_C const IID IID_IMILBitmapEffectRenderContext;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("12A2EC7E-2D33-44b2-B334-1ABB7846E390")
+    MIDL_INTERFACE("12A2EC7E-2D33-44b2-B334-1ABB7846E390") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectRenderContext : public IUnknown
     {
     public:
@@ -1438,7 +1439,7 @@ EXTERN_C const IID IID_IMILBitmapEffectRenderContextImpl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("4D25ACCB-797D-4fd2-B128-DFFEFF84FCC3")
+    MIDL_INTERFACE("4D25ACCB-797D-4fd2-B128-DFFEFF84FCC3") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectRenderContextImpl : public IUnknown
     {
     public:
@@ -1577,7 +1578,7 @@ EXTERN_C const IID IID_IMILBitmapEffectFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("33A9DF34-A403-4EC7-B07E-BC0682370845")
+    MIDL_INTERFACE("33A9DF34-A403-4EC7-B07E-BC0682370845") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectFactory : public IUnknown
     {
     public:
@@ -1694,7 +1695,7 @@ EXTERN_C const IID IID_IMILBitmapEffectPrimitive;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("67E31025-3091-4dfc-98D6-DD494551461D")
+    MIDL_INTERFACE("67E31025-3091-4dfc-98D6-DD494551461D") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectPrimitive : public IUnknown
     {
     public:
@@ -1898,10 +1899,10 @@ EXTERN_C const IID IID_IMILBitmapEffectPrimitiveImpl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("CE41E00B-EFA6-44e7-B007-DD042E3AE126")
+    MIDL_INTERFACE("CE41E00B-EFA6-44e7-B007-DD042E3AE126") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectPrimitiveImpl : public IUnknown
     {
-    public:
+    _Win32_PreserveResult_ public:
         virtual HRESULT STDMETHODCALLTYPE IsDirty( 
             /* [annotation][in] */ 
             _In_  ULONG uiOutputIndex,
@@ -2006,7 +2007,7 @@ EXTERN_C const IID IID_IMILBitmapEffects;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("51AC3DCE-67C5-448b-9180-AD3EABDDD5DD")
+    MIDL_INTERFACE("51AC3DCE-67C5-448b-9180-AD3EABDDD5DD") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffects : public IUnknown
     {
     public:
@@ -2134,7 +2135,7 @@ EXTERN_C const IID IID_IMILBitmapEffectConnector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("F59567B3-76C1-4d47-BA1E-79F955E350EF")
+    MIDL_INTERFACE("F59567B3-76C1-4d47-BA1E-79F955E350EF") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectConnector : public IMILBitmapEffectConnectorInfo
     {
     public:
@@ -2273,7 +2274,7 @@ EXTERN_C const IID IID_IMILBitmapEffectInputConnector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("A9B4ECAA-7A3C-45e7-8573-F4B81B60DD6C")
+    MIDL_INTERFACE("A9B4ECAA-7A3C-45e7-8573-F4B81B60DD6C") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectInputConnector : public IMILBitmapEffectConnector
     {
     public:
@@ -2431,7 +2432,7 @@ EXTERN_C const IID IID_IMILBitmapEffectOutputConnector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("92957AAD-841B-4866-82EC-8752468B07FD")
+    MIDL_INTERFACE("92957AAD-841B-4866-82EC-8752468B07FD") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectOutputConnector : public IMILBitmapEffectConnector
     {
     public:
@@ -2593,7 +2594,7 @@ EXTERN_C const IID IID_IMILBitmapEffectOutputConnectorImpl;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("21FAE777-8B39-4bfa-9F2D-F3941ED36913")
+    MIDL_INTERFACE("21FAE777-8B39-4bfa-9F2D-F3941ED36913") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectOutputConnectorImpl : public IUnknown
     {
     public:
@@ -2693,7 +2694,7 @@ EXTERN_C const IID IID_IMILBitmapEffectInteriorInputConnector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("20287E9E-86A2-4e15-953D-EB1438A5B842")
+    MIDL_INTERFACE("20287E9E-86A2-4e15-953D-EB1438A5B842") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectInteriorInputConnector : public IUnknown
     {
     public:
@@ -2780,7 +2781,7 @@ EXTERN_C const IID IID_IMILBitmapEffectInteriorOutputConnector;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("00BBB6DC-ACC9-4bfc-B344-8BEE383DFEFA")
+    MIDL_INTERFACE("00BBB6DC-ACC9-4bfc-B344-8BEE383DFEFA") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectInteriorOutputConnector : public IUnknown
     {
     public:
@@ -2867,7 +2868,7 @@ EXTERN_C const IID IID_IMILBitmapEffectEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("2E880DD8-F8CE-457b-8199-D60BB3D7EF98")
+    MIDL_INTERFACE("2E880DD8-F8CE-457b-8199-D60BB3D7EF98") _Windows_SupportedOS_WindowsXP_ 
     IMILBitmapEffectEvents : public IUnknown
     {
     public:
