diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/minidumpapiset.h b/generation/WinSDK/RecompiledIdlHeaders/um/minidumpapiset.h
index f3e7cbb510..9472805866 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/minidumpapiset.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/minidumpapiset.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 // Copyright (C) Microsoft Corporation. All rights reserved.
 
 #if defined(_MSC_VER)
@@ -238,7 +239,7 @@ typedef struct _MINIDUMP_SYSTEM_INFO {
     // taken from the SYSTEM_INFO structure obtained by GetSystemInfo( ).
     //
 
-    USHORT ProcessorArchitecture;
+    _Win32_AssociatedEnum_(PROCESSOR_ARCHITECTURE) USHORT ProcessorArchitecture;
     USHORT ProcessorLevel;
     USHORT ProcessorRevision;
 
@@ -259,7 +260,7 @@ typedef struct _MINIDUMP_SYSTEM_INFO {
     ULONG32 MajorVersion;
     ULONG32 MinorVersion;
     ULONG32 BuildNumber;
-    ULONG32 PlatformId;
+    _Win32_AssociatedEnum_(VER_PLATFORM) ULONG32 PlatformId;
 
     //
     // RVA to a CSDVersion string in the string table.
@@ -586,7 +587,7 @@ typedef struct _XSTATE_CONFIG_FEATURE_MSC_INFO
 
 typedef struct _MINIDUMP_MISC_INFO {
     ULONG32 SizeOfInfo;
-    ULONG32 Flags1;
+    _Win32_AssociatedEnum_(MINIDUMP_MISC_INFO_FLAGS) ULONG32 Flags1;
     ULONG32 ProcessId;
     ULONG32 ProcessCreateTime;
     ULONG32 ProcessUserTime;
@@ -693,7 +694,7 @@ typedef struct _MINIDUMP_MEMORY_INFO {
     ULONG32 AllocationProtect;
     ULONG32 __alignment1;
     ULONG64 RegionSize;
-    ULONG32 State;
+    _Win32_AssociatedEnum_(VIRTUAL_ALLOCATION_TYPE) ULONG32 State;
     ULONG32 Protect;
     ULONG32 Type;
     ULONG32 __alignment2;
@@ -743,7 +744,7 @@ typedef struct _MINIDUMP_THREAD_NAME_LIST {
 
 typedef struct _MINIDUMP_THREAD_INFO {
     ULONG32 ThreadId;
-    ULONG32 DumpFlags;
+    _Win32_AssociatedEnum_(MINIDUMP_THREAD_INFO_DUMP_FLAGS) ULONG32 DumpFlags;
     ULONG32 DumpError;
     ULONG32 ExitStatus;
     ULONG64 CreateTime;
@@ -1160,7 +1161,7 @@ typedef enum _THREAD_WRITE_FLAGS {
 } THREAD_WRITE_FLAGS;
 
 typedef struct _MINIDUMP_MODULE_CALLBACK {
-    PWCHAR FullPath;
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") PWCHAR FullPath;
     ULONG64 BaseOfImage;
     ULONG SizeOfImage;
     ULONG CheckSum;
@@ -1362,7 +1363,11 @@ typedef struct _MINIDUMP_CALLBACK_OUTPUT {
 // with the virtual protection attribute of PAGE_WRITECOMBINE.
 //
 
-typedef enum _MINIDUMP_TYPE {
+typedef enum
+#if defined(WIN32METADATA) && defined(__clang__)
+[[clang::flag_enum]]
+#endif
+ _MINIDUMP_TYPE {
     MiniDumpNormal                         = 0x00000000,
     MiniDumpWithDataSegs                   = 0x00000001,
     MiniDumpWithFullMemory                 = 0x00000002,
@@ -1471,7 +1476,7 @@ typedef struct _MINIDUMP_CALLBACK_INFORMATION {
 
 BOOL
 WINAPI
-MiniDumpWriteDump(
+_Win32_SetLastError_ MiniDumpWriteDump(
     _In_ HANDLE hProcess,
     _In_ DWORD ProcessId,
     _In_ HANDLE hFile,
@@ -1523,6 +1528,51 @@ MiniDumpReadDumpStream(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("MINIDUMP_THREAD_INFO_ERROR_THREAD")
+#undef MINIDUMP_THREAD_INFO_ERROR_THREAD
+#pragma push_macro("MINIDUMP_THREAD_INFO_EXITED_THREAD")
+#undef MINIDUMP_THREAD_INFO_EXITED_THREAD
+#pragma push_macro("MINIDUMP_THREAD_INFO_INVALID_CONTEXT")
+#undef MINIDUMP_THREAD_INFO_INVALID_CONTEXT
+#pragma push_macro("MINIDUMP_THREAD_INFO_INVALID_INFO")
+#undef MINIDUMP_THREAD_INFO_INVALID_INFO
+#pragma push_macro("MINIDUMP_THREAD_INFO_INVALID_TEB")
+#undef MINIDUMP_THREAD_INFO_INVALID_TEB
+#pragma push_macro("MINIDUMP_THREAD_INFO_WRITING_THREAD")
+#undef MINIDUMP_THREAD_INFO_WRITING_THREAD
+enum MINIDUMP_THREAD_INFO_DUMP_FLAGS : unsigned int
+{
+    MINIDUMP_THREAD_INFO_ERROR_THREAD = 1U,
+    MINIDUMP_THREAD_INFO_EXITED_THREAD = 4U,
+    MINIDUMP_THREAD_INFO_INVALID_CONTEXT = 16U,
+    MINIDUMP_THREAD_INFO_INVALID_INFO = 8U,
+    MINIDUMP_THREAD_INFO_INVALID_TEB = 32U,
+    MINIDUMP_THREAD_INFO_WRITING_THREAD = 2U,
+};
+#pragma pop_macro("MINIDUMP_THREAD_INFO_WRITING_THREAD")
+#pragma pop_macro("MINIDUMP_THREAD_INFO_INVALID_TEB")
+#pragma pop_macro("MINIDUMP_THREAD_INFO_INVALID_INFO")
+#pragma pop_macro("MINIDUMP_THREAD_INFO_INVALID_CONTEXT")
+#pragma pop_macro("MINIDUMP_THREAD_INFO_EXITED_THREAD")
+#pragma pop_macro("MINIDUMP_THREAD_INFO_ERROR_THREAD")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("MINIDUMP_MISC1_PROCESS_ID")
+#undef MINIDUMP_MISC1_PROCESS_ID
+#pragma push_macro("MINIDUMP_MISC1_PROCESS_TIMES")
+#undef MINIDUMP_MISC1_PROCESS_TIMES
+enum [[clang::flag_enum]] MINIDUMP_MISC_INFO_FLAGS : unsigned int
+{
+    MINIDUMP_MISC1_PROCESS_ID = 1U,
+    MINIDUMP_MISC1_PROCESS_TIMES = 2U,
+};
+#pragma pop_macro("MINIDUMP_MISC1_PROCESS_TIMES")
+#pragma pop_macro("MINIDUMP_MISC1_PROCESS_ID")
+#endif
+
 #endif /* _MINIDUMP_H */
 
 // HEXTRACT: end_dbghelp hide-line end_imagehlp
