diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h b/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
index 962985fd8a..01589c0ea0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
@@ -1,4 +1,7 @@
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 //*@@@+++@@@@******************************************************************
 //
@@ -3054,10 +3057,16 @@ typedef MPEG1WAVEFORMAT FAR            *LPMPEG1WAVEFORMAT;
 
 // WAVE_FORMAT_MPEGLAYER3 format sructure
 //
+#ifdef WIN32METADATA
+enum class MPEGLAYER3WAVEFORMAT_FLAGS : int;
+#else
+typedef DWORD MPEGLAYER3WAVEFORMAT_FLAGS;
+#endif
+
 typedef struct mpeglayer3waveformat_tag {
   WAVEFORMATEX  wfx;
   WORD          wID;
-  DWORD         fdwFlags;
+  MPEGLAYER3WAVEFORMAT_FLAGS fdwFlags;
   WORD          nBlockSize;
   WORD          nFramesPerBlock;
   WORD          nCodecDelay;
@@ -3077,6 +3086,24 @@ typedef MPEGLAYER3WAVEFORMAT FAR     *LPMPEGLAYER3WAVEFORMAT;
 #define MPEGLAYER3_FLAG_PADDING_ON       0x00000001
 #define MPEGLAYER3_FLAG_PADDING_OFF      0x00000002
 
+#ifdef WIN32METADATA
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_ISO")
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_ON")
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_OFF")
+#undef MPEGLAYER3_FLAG_PADDING_ISO
+#undef MPEGLAYER3_FLAG_PADDING_ON
+#undef MPEGLAYER3_FLAG_PADDING_OFF
+enum class MPEGLAYER3WAVEFORMAT_FLAGS : int
+{
+    MPEGLAYER3_FLAG_PADDING_ISO = 0,
+    MPEGLAYER3_FLAG_PADDING_ON = 1,
+    MPEGLAYER3_FLAG_PADDING_OFF = 2,
+};
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_OFF")
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_ON")
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_ISO")
+#endif
+
 /* ==========================================================================
 
       For WAVE_FORMAT_MPEG_HEAAC (0x1610)
@@ -3811,4 +3838,3 @@ cpp_quote("#endif /* 0 */")
 
 #endif /* __midl */
 
-
