diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h b/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
index 55920c8e3f..b051360d1e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/mmreg.h
@@ -1,4 +1,5 @@
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 //*@@@+++@@@@******************************************************************
 //
@@ -3057,7 +3058,7 @@ typedef MPEG1WAVEFORMAT FAR            *LPMPEG1WAVEFORMAT;
 typedef struct mpeglayer3waveformat_tag {
   WAVEFORMATEX  wfx;
   WORD          wID;
-  DWORD         fdwFlags;
+  _Win32_AssociatedEnum_(MPEGLAYER3WAVEFORMAT_FLAGS) DWORD         fdwFlags;
   WORD          nBlockSize;
   WORD          nFramesPerBlock;
   WORD          nCodecDelay;
@@ -3153,7 +3154,7 @@ typedef MPEGLAYER3WAVEFORMAT FAR     *LPMPEGLAYER3WAVEFORMAT;
  }
 */
 
-typedef struct heaacwaveinfo_tag {
+typedef struct   heaacwaveinfo_tag {
 
     // Defines core AAC properties. See description above. WAVEFORMATEX is of size 18 bytes.
     WAVEFORMATEX wfx;
@@ -3203,7 +3204,7 @@ typedef HEAACWAVEINFO FAR  *LPHEAACWAVEINFO;
 //
 // This structure describes the format block for wStructType=0
 //
-typedef struct heaacwaveformat_tag {
+typedef struct   heaacwaveformat_tag {
 
     HEAACWAVEINFO wfInfo; // This structure has a size of 30 bytes
     BYTE          pbAudioSpecificConfig[1];  // First byte of AudioSpecificConfig()
@@ -3446,7 +3447,7 @@ typedef CSIMAADPCMWAVEFORMAT  FAR  *LPCSIMAADPCMWAVEFORMAT;
 #define WAVE_FILTER_UNKNOWN         0x0000
 #define WAVE_FILTER_DEVELOPMENT    (0xFFFF)
 
-typedef struct wavefilter_tag {
+typedef struct   wavefilter_tag {
     DWORD   cbStruct;           /* Size of the filter in bytes */
     DWORD   dwFilterTag;        /* filter type */
     DWORD   fdwFilter;          /* Flags for the filter (Universal Dfns) */
@@ -3809,6 +3810,25 @@ typedef struct {
 #pragma pack(pop)  /* Assume byte packing throughout */
 cpp_quote("#endif /* 0 */")
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_ISO")
+#undef MPEGLAYER3_FLAG_PADDING_ISO
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_ON")
+#undef MPEGLAYER3_FLAG_PADDING_ON
+#pragma push_macro("MPEGLAYER3_FLAG_PADDING_OFF")
+#undef MPEGLAYER3_FLAG_PADDING_OFF
+enum MPEGLAYER3WAVEFORMAT_FLAGS : unsigned int
+{
+    MPEGLAYER3_FLAG_PADDING_ISO = 0U,
+    MPEGLAYER3_FLAG_PADDING_ON = 1U,
+    MPEGLAYER3_FLAG_PADDING_OFF = 2U,
+};
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_OFF")
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_ON")
+#pragma pop_macro("MPEGLAYER3_FLAG_PADDING_ISO")
+#endif
+
 #endif /* __midl */
 
 
