diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MpegType.h b/generation/WinSDK/RecompiledIdlHeaders/um/MpegType.h
index 68a85d0e1a..88b51839dd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MpegType.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MpegType.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //------------------------------------------------------------------------------
 // File: MPEGType.h
 //
@@ -65,54 +66,67 @@ typedef struct tagAM_MPEGSYSTEMTYPE
 
 DECLARE_INTERFACE_(IMpegAudioDecoder, IUnknown) {
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_FrequencyDivider) (THIS_
                            _Out_ unsigned long *pDivider   /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_FrequencyDivider) (THIS_
                            unsigned long Divider     /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_DecoderAccuracy) (THIS_
                            _Out_ unsigned long *pAccuracy  /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_DecoderAccuracy) (THIS_
                            unsigned long Accuracy    /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_Stereo) (THIS_
                            _Out_ unsigned long *pStereo    /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_Stereo) (THIS_
                            unsigned long Stereo      /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_DecoderWordSize) (THIS_
                            _Out_ unsigned long *pWordSize  /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_DecoderWordSize) (THIS_
                            unsigned long WordSize    /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_IntegerDecode) (THIS_
                            _Out_ unsigned long *pIntDecode /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_IntegerDecode) (THIS_
                            unsigned long IntDecode   /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_DualMode) (THIS_
                            unsigned long *pIntDecode /* [out] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(put_DualMode) (THIS_
                            unsigned long IntDecode   /* [in] */
                            ) PURE;
 
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD(get_AudioFormat) (THIS_
                            _Out_ MPEG1WAVEFORMAT *lpFmt    /* [out] */
                            ) PURE;
