diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h b/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
index 6aa2b19df6..5e2336a281 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Mq.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
  
 /*++
 
@@ -2244,8 +2245,8 @@ VOID
     QUEUEHANDLE hSource,
     DWORD dwTimeout,
     DWORD dwAction,
-    MQMSGPROPS* pMessageProps,
-    LPOVERLAPPED lpOverlapped,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") MQMSGPROPS* pMessageProps,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPOVERLAPPED lpOverlapped,
     HANDLE hCursor
     );
 
@@ -2276,7 +2277,7 @@ MQLocateBegin(
     _In_opt_ MQRESTRICTION* pRestriction,
     _In_ MQCOLUMNSET* pColumns,
     _In_ MQSORTSET* pSort,
-    _Out_ PHANDLE phEnum
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE phEnum
     );
 
 HRESULT
@@ -2343,7 +2344,7 @@ HRESULT
 APIENTRY
 MQCreateCursor(
     _In_ QUEUEHANDLE hQueue,
-    _Out_ PHANDLE phCursor
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE phCursor
     );
 
 HRESULT
@@ -2386,7 +2387,7 @@ HRESULT
 APIENTRY
 MQSetQueueSecurity(
     _In_ LPCWSTR lpwcsFormatName,
-    _In_ SECURITY_INFORMATION SecurityInformation,
+    _Win32_AssociatedEnum_(OBJECT_SECURITY_INFORMATION) _In_ SECURITY_INFORMATION SecurityInformation,
     _In_opt_ PSECURITY_DESCRIPTOR pSecurityDescriptor
     );
 
@@ -2445,7 +2446,7 @@ APIENTRY
 MQGetSecurityContext(
     _In_reads_bytes_opt_(dwCertBufferLength) PVOID lpCertBuffer,
     _In_ DWORD dwCertBufferLength,
-    _Out_ HANDLE* phSecurityContext
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ HANDLE* phSecurityContext
     );
 
 HRESULT
@@ -2453,7 +2454,7 @@ APIENTRY
 MQGetSecurityContextEx(
     _In_reads_bytes_opt_(dwCertBufferLength) PVOID lpCertBuffer,
     _In_ DWORD dwCertBufferLength,
-    _Out_ HANDLE* phSecurityContext
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ HANDLE* phSecurityContext
     );
 
 VOID
@@ -2535,5 +2536,7 @@ MQMoveMessage(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+
 #endif // __MQ_H__
 
