diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MSAJTransport.h b/generation/WinSDK/RecompiledIdlHeaders/um/MSAJTransport.h
index aacc5ee37c..345ee0066b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MSAJTransport.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MSAJTransport.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 // AllJoyn APIs to connect, send/receive data, 
 // set/enum events for Named Pipe Transport.     
 //
@@ -22,17 +23,17 @@ extern "C" {
 
 #define ALLJOYN_NAMED_PIPE_CONNECT_SPEC     L"npipe:"
 
-HANDLE
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") _Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE
 WINAPI
 AllJoynConnectToBus(
     _In_opt_ PCWSTR connectionSpec);
 
-BOOL
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") BOOL
 WINAPI
 AllJoynCloseBusHandle(
     _In_ HANDLE busHandle);
 
-BOOL
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") BOOL
 WINAPI 
 AllJoynSendToBus(
     _In_ HANDLE connectedBusHandle,
@@ -41,7 +42,7 @@ AllJoynSendToBus(
     _Out_opt_ PDWORD bytesTransferred,
     _Inout_ PVOID reserved);
 
-BOOL
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") BOOL
 WINAPI 
 AllJoynReceiveFromBus(
     _In_ HANDLE connectedBusHandle,
@@ -55,14 +56,14 @@ AllJoynReceiveFromBus(
 #define ALLJOYN_WRITE_READY     0x2
 #define ALLJOYN_DISCONNECTED    0x4
 
-BOOL
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") BOOL
 WINAPI 
 AllJoynEventSelect(
     _In_ HANDLE connectedBusHandle,
     _In_ HANDLE eventHandle,
     _In_ DWORD eventTypes);
 
-BOOL
+_Windows_SupportedOS_10240_ _Win32_SetLastError_ _Win32_ImportLibrary_("msajapi.dll") BOOL
 WINAPI 
 AllJoynEnumEvents(
     _In_ HANDLE connectedBusHandle,
@@ -76,14 +77,14 @@ AllJoynEnumEvents(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
-HANDLE
+_Win32_ImportLibrary_("msajapi.dll") _Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE
 WINAPI
 AllJoynCreateBus(
     _In_ DWORD outBufferSize,
     _In_ DWORD inBufferSize,
     _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes);
 
-DWORD
+_Win32_ImportLibrary_("msajapi.dll") DWORD
 WINAPI
 AllJoynAcceptBusConnection(
     _In_ HANDLE serverBusHandle,
