diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/msfeeds.h b/generation/WinSDK/RecompiledIdlHeaders/um/msfeeds.h
index 93c8a04aa6..378a6b28b8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/msfeeds.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/msfeeds.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -3812,7 +3813,7 @@ EXTERN_C const IID IID_IFeedsManager;
             /* [annotation][in] */ 
             _In_  BSTR feedXmlIn,
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *feedXmlOut) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *feedXmlOut) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_ItemCountLimit( 
             /* [annotation][retval][out] */ 
@@ -4339,7 +4340,7 @@ EXTERN_C const IID IID_IFeedFolder;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Name( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *folderName) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *folderName) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE Rename( 
             /* [annotation][in] */ 
@@ -4347,7 +4348,7 @@ EXTERN_C const IID IID_IFeedFolder;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Path( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *folderPath) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *folderPath) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE Move( 
             /* [annotation][in] */ 
@@ -5079,11 +5080,11 @@ EXTERN_C const IID IID_IFeed;
             /* [annotation][in] */ 
             _In_  FEEDS_XML_INCLUDE_FLAGS includeFlags,
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *xml) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *xml) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Name( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *name) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *name) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE Rename( 
             /* [annotation][in] */ 
@@ -5091,7 +5092,7 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Url( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *feedUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *feedUrl) = 0;
         
         virtual /* [helpstring][propput][id] */ HRESULT STDMETHODCALLTYPE put_Url( 
             /* [annotation][in] */ 
@@ -5099,11 +5100,11 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LocalId( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *feedGuid) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *feedGuid) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Path( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *path) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *path) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE Move( 
             /* [annotation][in] */ 
@@ -5147,7 +5148,7 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LocalEnclosurePath( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *path) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *path) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Items( 
             /* [annotation][retval][out] */ 
@@ -5161,19 +5162,19 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Title( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *title) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *title) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Description( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *description) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *description) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Link( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *homePage) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *homePage) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Image( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *imageUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *imageUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LastBuildDate( 
             /* [annotation][retval][out] */ 
@@ -5189,11 +5190,11 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Language( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *language) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *language) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Copyright( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *copyright) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *copyright) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_MaxItemCount( 
             /* [annotation][retval][out] */ 
@@ -5227,7 +5228,7 @@ EXTERN_C const IID IID_IFeed;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_DownloadUrl( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *feedUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *feedUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_IsList( 
             /* [annotation][retval][out] */ 
@@ -5803,11 +5804,11 @@ EXTERN_C const IID IID_IFeed2;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Username( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *username) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *username) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Password( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *password) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *password) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE SetCredentials( 
             /* [annotation][in] */ 
@@ -6670,23 +6671,23 @@ EXTERN_C const IID IID_IFeedItem;
             /* [annotation][in] */ 
             _In_  FEEDS_XML_INCLUDE_FLAGS includeFlags,
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *xml) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *xml) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Title( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *title) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *title) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Link( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *linkUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *linkUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Guid( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *itemGuid) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *itemGuid) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Description( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *description) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *description) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_PubDate( 
             /* [annotation][retval][out] */ 
@@ -6694,11 +6695,11 @@ EXTERN_C const IID IID_IFeedItem;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Comments( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *comments) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *comments) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Author( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *author) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *author) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Enclosure( 
             /* [annotation][retval][out] */ 
@@ -6724,7 +6725,7 @@ EXTERN_C const IID IID_IFeedItem;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_DownloadUrl( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *itemUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *itemUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LastDownloadTime( 
             /* [annotation][retval][out] */ 
@@ -7328,11 +7329,11 @@ EXTERN_C const IID IID_IFeedEnclosure;
     public:
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Url( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *enclosureUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *enclosureUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Type( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *mimeType) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *mimeType) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Length( 
             /* [annotation][retval][out] */ 
@@ -7352,7 +7353,7 @@ EXTERN_C const IID IID_IFeedEnclosure;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_LocalPath( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *localPath) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *localPath) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Parent( 
             /* [annotation][retval][out] */ 
@@ -7360,11 +7361,11 @@ EXTERN_C const IID IID_IFeedEnclosure;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_DownloadUrl( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *enclosureUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *enclosureUrl) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_DownloadMimeType( 
             /* [annotation][retval][out] */ 
-            _Out_retval_  BSTR *mimeType) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_retval_  BSTR *mimeType) = 0;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE RemoveFile( void) = 0;
         
