diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MsHtmHst.h b/generation/WinSDK/RecompiledIdlHeaders/um/MsHtmHst.h
index 0e58a78962..dda2be641c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MsHtmHst.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MsHtmHst.h
@@ -132,6 +132,7 @@ extern "C"{
 //=--------------------------------------------------------------------------=
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -429,11 +430,11 @@ extern RPC_IF_HANDLE __MIDL_itf_mshtmhst_0000_0002_v0_0_s_ifspec;
 /* interface IDocHostUIHandler */
 /* [local][unique][uuid][object] */ 
 
-typedef struct _DOCHOSTUIINFO
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")    _DOCHOSTUIINFO
     {
     ULONG cbSize;
-    DWORD dwFlags;
-    DWORD dwDoubleClick;
+    _Win32_AssociatedEnum_(DOCHOSTUIFLAG) DWORD dwFlags;
+    _Win32_AssociatedEnum_(DOCHOSTUIDBLCLK) DWORD dwDoubleClick;
     OLECHAR *pchHostCss;
     OLECHAR *pchHostNS;
     } 	DOCHOSTUIINFO;
