diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mssip.h b/generation/WinSDK/RecompiledIdlHeaders/um/mssip.h
index 9ba32f8775..4237617a7f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mssip.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mssip.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //+-------------------------------------------------------------------------
 //
 //  Microsoft Windows - Internet Security
@@ -75,7 +76,7 @@ typedef CRYPT_HASH_BLOB             CRYPT_DIGEST_DATA;
 //  the ENTIRE structure to binary zero before the FIRST call is made.  Do
 //  not initialize it BETWEEN calls!
 //
-typedef struct SIP_SUBJECTINFO_
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SIP_SUBJECTINFO_
 {
     DWORD                       cbSize;         // set to sizeof(SIP_SUBJECTINFO)
     GUID                        *pgSubjectType; // subject type
@@ -192,7 +193,7 @@ typedef struct MS_ADDINFO_DETACHEDSIG_
 //
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
-typedef struct _SIP_CAP_SET_V2
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIP_CAP_SET_V2
 {
     DWORD cbSize;           //Size of this struct
     DWORD dwVersion;        //The SIP API version being used, default to 2.
@@ -202,7 +203,7 @@ typedef struct _SIP_CAP_SET_V2
 
 #if (NTDDI_VERSION >= NTDDI_WINBLUE)
 
-typedef struct _SIP_CAP_SET_V3
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIP_CAP_SET_V3
 {
     DWORD cbSize;           //Size of this struct
     DWORD dwVersion;        //The SIP API version being used, default to 2.
@@ -275,19 +276,19 @@ typedef struct SIP_INDIRECT_DATA_
 //                                      enough to hold the data.  pcbSignedDataMsg
 //                                      contains the required size.
 //
-extern BOOL WINAPI CryptSIPGetSignedDataMsg(
-                                IN      SIP_SUBJECTINFO *pSubjectInfo,
-                                OUT     DWORD           *pdwEncodingType,
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPGetSignedDataMsg(
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:in") _Win32_AssociatedEnum_(CERT_QUERY_ENCODING_TYPE) OUT     DWORD           *pdwEncodingType,
                                 IN      DWORD           dwIndex,
                                 IN OUT  DWORD           *pcbSignedDataMsg,
-                                OUT     BYTE            *pbSignedDataMsg);
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     BYTE            *pbSignedDataMsg);
 
 typedef BOOL (WINAPI * pCryptSIPGetSignedDataMsg)(
-                                IN      SIP_SUBJECTINFO *pSubjectInfo,
-                                OUT     DWORD           *pdwEncodingType,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     DWORD           *pdwEncodingType,
                                 IN      DWORD           dwIndex,
                                 IN OUT  DWORD           *pcbSignedDataMsg,
-                                OUT     BYTE            *pbSignedDataMsg);
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     BYTE            *pbSignedDataMsg);
 
 
 //////////////////////////////////////////////////////////////////////////////
@@ -321,19 +322,19 @@ typedef BOOL (WINAPI * pCryptSIPGetSignedDataMsg)(
 //                                      indirect data (hash).  Then, call the
 //                                      'put' function again.
 //
-extern BOOL WINAPI CryptSIPPutSignedDataMsg(
-                                IN      SIP_SUBJECTINFO *pSubjectInfo,
-                                IN      DWORD           dwEncodingType,
-                                OUT     DWORD           *pdwIndex,
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPPutSignedDataMsg(
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO *pSubjectInfo,
+                                _Win32_AssociatedEnum_(CERT_QUERY_ENCODING_TYPE) IN      DWORD           dwEncodingType,
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     DWORD           *pdwIndex,
                                 IN      DWORD           cbSignedDataMsg,
-                                IN      BYTE            *pbSignedDataMsg);
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      BYTE            *pbSignedDataMsg);
 
 typedef BOOL (WINAPI * pCryptSIPPutSignedDataMsg)(
-                                IN      SIP_SUBJECTINFO *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO *pSubjectInfo,
                                 IN      DWORD           dwEncodingType,
-                                OUT     DWORD           *pdwIndex,
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     DWORD           *pdwIndex,
                                 IN      DWORD           cbSignedDataMsg,
-                                IN      BYTE            *pbSignedDataMsg);
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      BYTE            *pbSignedDataMsg);
 
 //////////////////////////////////////////////////////////////////////////////
 //
@@ -356,15 +357,15 @@ typedef BOOL (WINAPI * pCryptSIPPutSignedDataMsg)(
 //      ERROR_BAD_FORMAT:               file/data format is not correct
 //                                      for the requested SIP.
 //
-extern BOOL WINAPI CryptSIPCreateIndirectData(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPCreateIndirectData(
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
                                 IN OUT  DWORD               *pcbIndirectData,
-                                OUT     SIP_INDIRECT_DATA   *pIndirectData);
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     SIP_INDIRECT_DATA   *pIndirectData);
 
 typedef BOOL (WINAPI * pCryptSIPCreateIndirectData)(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
                                 IN OUT  DWORD               *pcbIndirectData,
-                                OUT     SIP_INDIRECT_DATA   *pIndirectData);
+                                _WIN32META_ANNOTATION_("win32metadata:in") OUT     SIP_INDIRECT_DATA   *pIndirectData);
 
 
 
@@ -389,13 +390,13 @@ typedef BOOL (WINAPI * pCryptSIPCreateIndirectData)(
 //      ERROR_INVALID_PARAMETER:        bad argument passed in
 //      ERROR_BAD_FORMAT:               file/data format is not correct
 //                                      for the requested SIP.
-extern BOOL WINAPI CryptSIPVerifyIndirectData(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
-                                IN      SIP_INDIRECT_DATA   *pIndirectData);
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPVerifyIndirectData(
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_INDIRECT_DATA   *pIndirectData);
 
 typedef BOOL (WINAPI * pCryptSIPVerifyIndirectData)(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
-                                IN      SIP_INDIRECT_DATA   *pIndirectData);
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_INDIRECT_DATA   *pIndirectData);
 
 
 //////////////////////////////////////////////////////////////////////////////
@@ -415,12 +416,12 @@ typedef BOOL (WINAPI * pCryptSIPVerifyIndirectData)(
 //      ERROR_BAD_FORMAT:               file/data format is not correct
 //                                      for the requested SIP.
 //
-extern BOOL WINAPI CryptSIPRemoveSignedDataMsg(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPRemoveSignedDataMsg(
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
                                 IN      DWORD               dwIndex);
 
 typedef BOOL (WINAPI * pCryptSIPRemoveSignedDataMsg)(
-                                IN      SIP_SUBJECTINFO     *pSubjectInfo,
+                                _WIN32META_ANNOTATION_("win32metadata:out") IN      SIP_SUBJECTINFO     *pSubjectInfo,
                                 IN      DWORD               dwIndex);
 
 
@@ -431,7 +432,7 @@ typedef BOOL (WINAPI * pCryptSIPRemoveSignedDataMsg)(
 // SIP_DISPATCH_INFO
 //----------------------------------------------------------------------------
 //
-typedef struct SIP_DISPATCH_INFO_
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SIP_DISPATCH_INFO_
 {
     DWORD                           cbSize;     // = sizeof(SIP_DISPATCH_INFO)
     HANDLE                          hSIP;       // used internal
@@ -449,10 +450,10 @@ typedef struct SIP_DISPATCH_INFO_
 // out the pgSubject with the appropiate GUID.
 //
 typedef BOOL (WINAPI *pfnIsFileSupported)(IN  HANDLE  hFile,
-                                   OUT GUID    *pgSubject);
+                                   _WIN32META_ANNOTATION_("win32metadata:in") OUT GUID    *pgSubject);
 
 typedef BOOL (WINAPI *pfnIsFileSupportedName)(IN WCHAR *pwszFileName,
-                                       OUT GUID *pgSubject);
+                                       _WIN32META_ANNOTATION_("win32metadata:in") OUT GUID *pgSubject);
 
 typedef struct SIP_ADD_NEWPROVIDER_
 {
@@ -490,7 +491,7 @@ typedef struct SIP_ADD_NEWPROVIDER_
 //      TRUE:                           No fatal errors
 //      FALSE:                          Errors occured.  See GetLastError()
 //
-extern BOOL WINAPI CryptSIPLoad(IN const GUID               *pgSubject,     // GUID for the requried sip
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPLoad(IN const GUID               *pgSubject,     // GUID for the requried sip
                                 IN DWORD                    dwFlags,        // Reserved - MUST BE ZERO
                                 IN OUT SIP_DISPATCH_INFO    *pSipDispatch); // Table of functions
 
@@ -505,9 +506,9 @@ extern BOOL WINAPI CryptSIPLoad(IN const GUID               *pgSubject,     // G
 //      TRUE:                           No fatal errors
 //      FALSE:                          Errors occured.  See GetLastError()
 //
-extern BOOL WINAPI CryptSIPRetrieveSubjectGuid(IN LPCWSTR FileName,   // wide file name
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPRetrieveSubjectGuid(IN LPCWSTR FileName,   // wide file name
                                                IN OPTIONAL HANDLE hFileIn,     // or handle of open file
-                                               OUT GUID *pgSubject);           // defined SIP's GUID
+                                               _WIN32META_ANNOTATION_("win32metadata:in") OUT GUID *pgSubject);           // defined SIP's GUID
 
 //////////////////////////////////////////////////////////////////////////////
 //
@@ -522,9 +523,9 @@ extern BOOL WINAPI CryptSIPRetrieveSubjectGuid(IN LPCWSTR FileName,   // wide fi
 //      TRUE:                           No fatal errors
 //      FALSE:                          Errors occured.  See GetLastError()
 //
-extern BOOL WINAPI CryptSIPRetrieveSubjectGuidForCatalogFile(IN LPCWSTR FileName,   // wide file name
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPRetrieveSubjectGuidForCatalogFile(IN LPCWSTR FileName,   // wide file name
                                                              IN OPTIONAL HANDLE hFileIn,     // or handle of open file
-                                                             OUT GUID *pgSubject);           // defined SIP's GUID
+                                                             _WIN32META_ANNOTATION_("win32metadata:in") OUT GUID *pgSubject);           // defined SIP's GUID
 
 
 //////////////////////////////////////////////////////////////////////////////
@@ -536,7 +537,7 @@ extern BOOL WINAPI CryptSIPRetrieveSubjectGuidForCatalogFile(IN LPCWSTR FileName
 //      TRUE:                           No fatal errors
 //      FALSE:                          Errors occured.  See GetLastError()
 //
-extern BOOL WINAPI CryptSIPAddProvider(IN SIP_ADD_NEWPROVIDER *psNewProv);
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPAddProvider(_WIN32META_ANNOTATION_("win32metadata:out") IN SIP_ADD_NEWPROVIDER *psNewProv);
 
 //////////////////////////////////////////////////////////////////////////////
 //
@@ -547,7 +548,7 @@ extern BOOL WINAPI CryptSIPAddProvider(IN SIP_ADD_NEWPROVIDER *psNewProv);
 //      TRUE:                           No fatal errors
 //      FALSE:                          Errors occured.  See GetLastError()
 //
-extern BOOL WINAPI CryptSIPRemoveProvider(IN GUID *pgProv);
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ extern BOOL WINAPI CryptSIPRemoveProvider(_WIN32META_ANNOTATION_("win32metadata:out") IN GUID *pgProv);
 
 //////////////////////////////////////////////////////////////////////////////
 //
@@ -560,7 +561,7 @@ extern BOOL WINAPI CryptSIPRemoveProvider(IN GUID *pgProv);
 //      FALSE:                          Errors occured.  See GetLastError()
 //  
 #if (NTDDI_VERSION >= NTDDI_WIN8)
-extern BOOL WINAPI CryptSIPGetCaps(_In_ SIP_SUBJECTINFO * pSubjInfo,
+_Windows_SupportedOS_Windows8_ extern BOOL WINAPI CryptSIPGetCaps(_In_ SIP_SUBJECTINFO * pSubjInfo,
                                    _Inout_ SIP_CAP_SET *pCaps);
 
 typedef BOOL (WINAPI *pCryptSIPGetCaps)(_In_ SIP_SUBJECTINFO * pSubjInfo,
@@ -595,4 +596,47 @@ typedef
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIG_APPEND")
+#undef SIG_APPEND
+#pragma push_macro("SPC_INC_PE_RESOURCES_FLAG")
+#undef SPC_INC_PE_RESOURCES_FLAG
+#pragma push_macro("SPC_INC_PE_DEBUG_INFO_FLAG")
+#undef SPC_INC_PE_DEBUG_INFO_FLAG
+#pragma push_macro("SPC_INC_PE_IMPORT_ADDR_TABLE_FLAG")
+#undef SPC_INC_PE_IMPORT_ADDR_TABLE_FLAG
+#pragma push_macro("SPC_EXC_PE_PAGE_HASHES_FLAG")
+#undef SPC_EXC_PE_PAGE_HASHES_FLAG
+#pragma push_macro("SPC_INC_PE_PAGE_HASHES_FLAG")
+#undef SPC_INC_PE_PAGE_HASHES_FLAG
+#pragma push_macro("SPC_DIGEST_GENERATE_FLAG")
+#undef SPC_DIGEST_GENERATE_FLAG
+#pragma push_macro("SPC_DIGEST_SIGN_FLAG")
+#undef SPC_DIGEST_SIGN_FLAG
+#pragma push_macro("SPC_DIGEST_SIGN_EX_FLAG")
+#undef SPC_DIGEST_SIGN_EX_FLAG
+enum [[clang::flag_enum]] SIGNER_SIGN_FLAGS : unsigned int
+{
+    SIG_APPEND = 4096U,
+    SPC_INC_PE_RESOURCES_FLAG = 128U,
+    SPC_INC_PE_DEBUG_INFO_FLAG = 64U,
+    SPC_INC_PE_IMPORT_ADDR_TABLE_FLAG = 32U,
+    SPC_EXC_PE_PAGE_HASHES_FLAG = 16U,
+    SPC_INC_PE_PAGE_HASHES_FLAG = 256U,
+    SPC_DIGEST_GENERATE_FLAG = 512U,
+    SPC_DIGEST_SIGN_FLAG = 1024U,
+    SPC_DIGEST_SIGN_EX_FLAG = 16384U,
+};
+#pragma pop_macro("SPC_DIGEST_SIGN_EX_FLAG")
+#pragma pop_macro("SPC_DIGEST_SIGN_FLAG")
+#pragma pop_macro("SPC_DIGEST_GENERATE_FLAG")
+#pragma pop_macro("SPC_INC_PE_PAGE_HASHES_FLAG")
+#pragma pop_macro("SPC_EXC_PE_PAGE_HASHES_FLAG")
+#pragma pop_macro("SPC_INC_PE_IMPORT_ADDR_TABLE_FLAG")
+#pragma pop_macro("SPC_INC_PE_DEBUG_INFO_FLAG")
+#pragma pop_macro("SPC_INC_PE_RESOURCES_FLAG")
+#pragma pop_macro("SIG_APPEND")
+#endif
+
 #endif // MSSIP_H
