diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/mssign.h b/generation/WinSDK/RecompiledIdlHeaders/um/mssign.h
index 1fcfeaee21..17b4254773 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/mssign.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/mssign.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 #include <wincrypt.h>
 
 #ifdef __cplusplus
@@ -66,7 +67,7 @@ typedef HRESULT (WINAPI *PFN_AUTHENTICODE_DIGEST_SIGN_WITHFILEHANDLE)(
     _Out_ PCRYPT_DATA_BLOB pSignedDigest 
     );
 
-typedef struct _SIGNER_ATTR_AUTHCODE {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_ATTR_AUTHCODE {
     DWORD   cbSize;
     BOOL    fCommercial;
     BOOL    fIndividual;
@@ -74,7 +75,7 @@ typedef struct _SIGNER_ATTR_AUTHCODE {
     LPCWSTR pwszInfo;
 } SIGNER_ATTR_AUTHCODE, *PSIGNER_ATTR_AUTHCODE;
 
-typedef struct _SIGNER_BLOB_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_BLOB_INFO {
     DWORD   cbSize;
     GUID    *pGuidSubject;
     DWORD   cbBlob;
@@ -82,23 +83,23 @@ typedef struct _SIGNER_BLOB_INFO {
     LPCWSTR pwszDisplayName;
 } SIGNER_BLOB_INFO, *PSIGNER_BLOB_INFO;
 
-typedef struct _SIGNER_CERT_STORE_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_CERT_STORE_INFO {
     DWORD          cbSize;
     PCCERT_CONTEXT pSigningCert;
-    DWORD          dwCertPolicy;
+    _Win32_AssociatedEnum_(SIGNER_CERT_POLICY) DWORD          dwCertPolicy;
     HCERTSTORE     hCertStore;
 } SIGNER_CERT_STORE_INFO, *PSIGNER_CERT_STORE_INFO;
 
-typedef struct _SIGNER_SPC_CHAIN_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_SPC_CHAIN_INFO {
     DWORD      cbSize;
     LPCWSTR    pwszSpcFile;
     DWORD      dwCertPolicy;
     HCERTSTORE hCertStore;
 } SIGNER_SPC_CHAIN_INFO, *PSIGNER_SPC_CHAIN_INFO;
 
-typedef struct _SIGNER_CERT {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_CERT {
     DWORD cbSize;
-    DWORD dwCertChoice;
+    _Win32_AssociatedEnum_(SIGNER_CERT_CHOICE) DWORD dwCertChoice;
     union {
         LPCWSTR                pwszSpcFile;
         SIGNER_CERT_STORE_INFO *pCertStoreInfo;
@@ -107,13 +108,13 @@ typedef struct _SIGNER_CERT {
     HWND  hwnd;
 } SIGNER_CERT, *PSIGNER_CERT;
 
-typedef struct _SIGNER_CONTEXT {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_CONTEXT {
     DWORD cbSize;
     DWORD cbBlob;
     BYTE  *pbBlob;
 } SIGNER_CONTEXT, *PSIGNER_CONTEXT;
 
-typedef struct _SIGNER_DIGEST_SIGN_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_DIGEST_SIGN_INFO {
     DWORD cbSize;                                                
     DWORD dwDigestSignChoice;                                    
     union {
@@ -128,41 +129,41 @@ typedef struct _SIGNER_DIGEST_SIGN_INFO {
     DWORD dwReserved3;
 } SIGNER_DIGEST_SIGN_INFO, *PSIGNER_DIGEST_SIGN_INFO;
 
-typedef struct SIGNER_DIGEST_SIGN_INFO_V1 {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SIGNER_DIGEST_SIGN_INFO_V1 {
     DWORD cbSize;
     PFN_AUTHENTICODE_DIGEST_SIGN pfnAuthenticodeDigestSign;
     PCRYPT_DATA_BLOB pMetadataBlob;
 } SIGNER_DIGEST_SIGN_INFO_V1, *PSIGNER_DIGEST_SIGN_INFO_V1;
 
-typedef struct SIGNER_DIGEST_SIGN_INFO_V2 {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SIGNER_DIGEST_SIGN_INFO_V2 {
     DWORD cbSize;                                                
     PFN_AUTHENTICODE_DIGEST_SIGN pfnAuthenticodeDigestSign;      
     PFN_AUTHENTICODE_DIGEST_SIGN_EX pfnAuthenticodeDigestSignEx; 
     PCRYPT_DATA_BLOB pMetadataBlob;                              
 } SIGNER_DIGEST_SIGN_INFO_V2, *PSIGNER_DIGEST_SIGN_INFO_V2;
 
-typedef struct _SIGNER_FILE_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_FILE_INFO {
     DWORD   cbSize;
     LPCWSTR pwszFileName;
     HANDLE  hFile;
 } SIGNER_FILE_INFO, *PSIGNER_FILE_INFO;
 
-typedef struct _SIGNER_PROVIDER_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_PROVIDER_INFO {
     DWORD   cbSize;
     LPCWSTR pwszProviderName;
     DWORD   dwProviderType;
     DWORD   dwKeySpec;
-    DWORD   dwPvkChoice;
+    _Win32_AssociatedEnum_(SIGNER_PRIVATE_KEY_CHOICE) DWORD   dwPvkChoice;
     union {
         LPWSTR pwszPvkFileName;
         LPWSTR pwszKeyContainer;
     };
 } SIGNER_PROVIDER_INFO, *PSIGNER_PROVIDER_INFO;
 
-typedef struct _SIGNER_SIGNATURE_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_SIGNATURE_INFO {
     DWORD             cbSize;
     ALG_ID            algidHash;
-    DWORD             dwAttrChoice;
+    _Win32_AssociatedEnum_(SIGNER_SIGNATURE_ATTRIBUTE_CHOICE) DWORD             dwAttrChoice;
     union {
         SIGNER_ATTR_AUTHCODE *pAttrAuthcode;
     };
@@ -170,23 +171,23 @@ typedef struct _SIGNER_SIGNATURE_INFO {
     PCRYPT_ATTRIBUTES psUnauthenticated;
 } SIGNER_SIGNATURE_INFO, *PSIGNER_SIGNATURE_INFO;
 
-typedef struct _SIGNER_SUBJECT_INFO {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _SIGNER_SUBJECT_INFO {
     DWORD cbSize;
     DWORD *pdwIndex;
-    DWORD dwSubjectChoice;
+    _Win32_AssociatedEnum_(SIGNER_SUBJECT_CHOICE) DWORD dwSubjectChoice;
     union {
         SIGNER_FILE_INFO *pSignerFileInfo;
         SIGNER_BLOB_INFO *pSignerBlobInfo;
     };
 } SIGNER_SUBJECT_INFO, *PSIGNER_SUBJECT_INFO;
 
-HRESULT WINAPI SignError(void);
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignError(void);
 
-HRESULT WINAPI SignerFreeSignerContext(
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerFreeSignerContext(
     _In_ SIGNER_CONTEXT *pSignerContext
     );
 
-HRESULT WINAPI SignerSign(
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerSign(
     _In_     SIGNER_SUBJECT_INFO   *pSubjectInfo,
     _In_     SIGNER_CERT           *pSignerCert,
     _In_     SIGNER_SIGNATURE_INFO *pSignatureInfo,
@@ -196,8 +197,8 @@ HRESULT WINAPI SignerSign(
     _In_opt_ LPVOID                pSipData
     );
 
-HRESULT WINAPI SignerSignEx(
-    _In_     DWORD                 dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerSignEx(
+    _Win32_AssociatedEnum_(SIGNER_SIGN_FLAGS) _In_     DWORD                 dwFlags,
     _In_     SIGNER_SUBJECT_INFO   *pSubjectInfo,
     _In_     SIGNER_CERT           *pSignerCert,
     _In_     SIGNER_SIGNATURE_INFO *pSignatureInfo,
@@ -208,29 +209,29 @@ HRESULT WINAPI SignerSignEx(
     _Out_    SIGNER_CONTEXT        **ppSignerContext
     );
 
-HRESULT WINAPI SignerSignEx2(
-    _In_       DWORD                  dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerSignEx2(
+    _Win32_AssociatedEnum_(SIGNER_SIGN_FLAGS) _In_       DWORD                  dwFlags,
     _In_       SIGNER_SUBJECT_INFO    *pSubjectInfo,
     _In_       SIGNER_CERT            *pSignerCert,
     _In_       SIGNER_SIGNATURE_INFO  *pSignatureInfo,
     _In_opt_   SIGNER_PROVIDER_INFO   *pProviderInfo,
-    _In_opt_   DWORD                  dwTimestampFlags,
+    _Win32_AssociatedEnum_(SIGNER_TIMESTAMP_FLAGS) _In_opt_   DWORD                  dwTimestampFlags,
     _In_opt_   PCSTR                  pszTimestampAlgorithmOid,
     _In_opt_   PCWSTR                 pwszHttpTimeStamp,
     _In_opt_   PCRYPT_ATTRIBUTES      psRequest,
     _In_opt_   PVOID                  pSipData,
     _Out_      SIGNER_CONTEXT         **ppSignerContext,
     _In_opt_   PCERT_STRONG_SIGN_PARA pCryptoPolicy,
-    _Reserved_ PVOID                  pReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ PVOID                  pReserved
     );
 
-HRESULT WINAPI SignerSignEx3(
-    _In_       DWORD                  dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerSignEx3(
+    _Win32_AssociatedEnum_(SIGNER_SIGN_FLAGS) _In_       DWORD                  dwFlags,
     _In_       SIGNER_SUBJECT_INFO    *pSubjectInfo,
     _In_       SIGNER_CERT            *pSignerCert,
     _In_       SIGNER_SIGNATURE_INFO  *pSignatureInfo,
     _In_opt_   SIGNER_PROVIDER_INFO   *pProviderInfo,
-    _In_opt_   DWORD                  dwTimestampFlags,
+    _Win32_AssociatedEnum_(SIGNER_TIMESTAMP_FLAGS) _In_opt_   DWORD                  dwTimestampFlags,
     _In_opt_   PCSTR                  pszTimestampAlgorithmOid,
     _In_opt_   PCWSTR                 pwszHttpTimeStamp,
     _In_opt_   PCRYPT_ATTRIBUTES      psRequest,
@@ -238,18 +239,18 @@ HRESULT WINAPI SignerSignEx3(
     _Out_      SIGNER_CONTEXT         **ppSignerContext,
     _In_opt_   PCERT_STRONG_SIGN_PARA pCryptoPolicy,
     _In_opt_   SIGNER_DIGEST_SIGN_INFO  *pDigestSignInfo,
-    _Reserved_ PVOID                  pReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ PVOID                  pReserved
     );
 
-HRESULT WINAPI SignerTimeStamp(
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerTimeStamp(
     _In_     SIGNER_SUBJECT_INFO *pSubjectInfo,
     _In_     LPCWSTR             pwszHttpTimeStamp,
     _In_opt_ PCRYPT_ATTRIBUTES   psRequest,
     _In_opt_ LPVOID              pSipData
     );
 
-HRESULT WINAPI SignerTimeStampEx(
-    _Reserved_ DWORD               dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerTimeStampEx(
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ DWORD               dwFlags,
     _In_       SIGNER_SUBJECT_INFO *pSubjectInfo,
     _In_       LPCWSTR             pwszHttpTimeStamp,
     _In_       PCRYPT_ATTRIBUTES   psRequest,
@@ -257,8 +258,8 @@ HRESULT WINAPI SignerTimeStampEx(
     _Out_      SIGNER_CONTEXT      **ppSignerContext 
     );
 
-HRESULT WINAPI SignerTimeStampEx2(
-    _Reserved_ DWORD               dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerTimeStampEx2(
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Win32_AssociatedEnum_(SIGNER_TIMESTAMP_FLAGS) _Reserved_ DWORD               dwFlags,
     _In_       SIGNER_SUBJECT_INFO *pSubjectInfo,
     _In_       LPCWSTR             pwszHttpTimeStamp,
     _In_       ALG_ID              dwAlgId,
@@ -267,8 +268,8 @@ HRESULT WINAPI SignerTimeStampEx2(
     _Out_      SIGNER_CONTEXT      **ppSignerContext 
     );
 
-HRESULT WINAPI SignerTimeStampEx3(
-    _In_       DWORD                  dwFlags,
+_Win32_ImportLibrary_("mssign32.dll") HRESULT WINAPI SignerTimeStampEx3(
+    _Win32_AssociatedEnum_(SIGNER_TIMESTAMP_FLAGS) _In_       DWORD                  dwFlags,
     _In_       DWORD                  dwIndex,
     _In_       SIGNER_SUBJECT_INFO    *pSubjectInfo,
     _In_       PCWSTR                 pwszHttpTimeStamp,
@@ -277,9 +278,106 @@ HRESULT WINAPI SignerTimeStampEx3(
     _In_opt_   PVOID                  pSipData,
     _Out_      SIGNER_CONTEXT         **ppSignerContext,
     _In_opt_   PCERT_STRONG_SIGN_PARA pCryptoPolicy,
-    _Reserved_ PVOID                  pReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ PVOID                  pReserved
     );
 
 #ifdef __cplusplus
 }
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIGNER_TIMESTAMP_AUTHENTICODE")
+#undef SIGNER_TIMESTAMP_AUTHENTICODE
+#pragma push_macro("SIGNER_TIMESTAMP_RFC3161")
+#undef SIGNER_TIMESTAMP_RFC3161
+enum SIGNER_TIMESTAMP_FLAGS : unsigned int
+{
+    SIGNER_TIMESTAMP_AUTHENTICODE = 1U,
+    SIGNER_TIMESTAMP_RFC3161 = 2U,
+};
+#pragma pop_macro("SIGNER_TIMESTAMP_RFC3161")
+#pragma pop_macro("SIGNER_TIMESTAMP_AUTHENTICODE")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIGNER_SUBJECT_FILE")
+#undef SIGNER_SUBJECT_FILE
+#pragma push_macro("SIGNER_SUBJECT_BLOB")
+#undef SIGNER_SUBJECT_BLOB
+enum SIGNER_SUBJECT_CHOICE : unsigned int
+{
+    SIGNER_SUBJECT_FILE = 1U,
+    SIGNER_SUBJECT_BLOB = 2U,
+};
+#pragma pop_macro("SIGNER_SUBJECT_BLOB")
+#pragma pop_macro("SIGNER_SUBJECT_FILE")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIGNER_NO_ATTR")
+#undef SIGNER_NO_ATTR
+#pragma push_macro("SIGNER_AUTHCODE_ATTR")
+#undef SIGNER_AUTHCODE_ATTR
+enum SIGNER_SIGNATURE_ATTRIBUTE_CHOICE : unsigned int
+{
+    SIGNER_NO_ATTR = 0U,
+    SIGNER_AUTHCODE_ATTR = 1U,
+};
+#pragma pop_macro("SIGNER_AUTHCODE_ATTR")
+#pragma pop_macro("SIGNER_NO_ATTR")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("PVK_TYPE_FILE_NAME")
+#undef PVK_TYPE_FILE_NAME
+#pragma push_macro("PVK_TYPE_KEYCONTAINER")
+#undef PVK_TYPE_KEYCONTAINER
+enum SIGNER_PRIVATE_KEY_CHOICE : unsigned int
+{
+    PVK_TYPE_FILE_NAME = 1U,
+    PVK_TYPE_KEYCONTAINER = 2U,
+};
+#pragma pop_macro("PVK_TYPE_KEYCONTAINER")
+#pragma pop_macro("PVK_TYPE_FILE_NAME")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIGNER_CERT_POLICY_STORE")
+#undef SIGNER_CERT_POLICY_STORE
+#pragma push_macro("SIGNER_CERT_POLICY_CHAIN")
+#undef SIGNER_CERT_POLICY_CHAIN
+#pragma push_macro("SIGNER_CERT_POLICY_SPC")
+#undef SIGNER_CERT_POLICY_SPC
+#pragma push_macro("SIGNER_CERT_POLICY_CHAIN_NO_ROOT")
+#undef SIGNER_CERT_POLICY_CHAIN_NO_ROOT
+enum [[clang::flag_enum]] SIGNER_CERT_POLICY : unsigned int
+{
+    SIGNER_CERT_POLICY_STORE = 1U,
+    SIGNER_CERT_POLICY_CHAIN = 2U,
+    SIGNER_CERT_POLICY_SPC = 4U,
+    SIGNER_CERT_POLICY_CHAIN_NO_ROOT = 8U,
+};
+#pragma pop_macro("SIGNER_CERT_POLICY_CHAIN_NO_ROOT")
+#pragma pop_macro("SIGNER_CERT_POLICY_SPC")
+#pragma pop_macro("SIGNER_CERT_POLICY_CHAIN")
+#pragma pop_macro("SIGNER_CERT_POLICY_STORE")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SIGNER_CERT_SPC_FILE")
+#undef SIGNER_CERT_SPC_FILE
+#pragma push_macro("SIGNER_CERT_STORE")
+#undef SIGNER_CERT_STORE
+#pragma push_macro("SIGNER_CERT_SPC_CHAIN")
+#undef SIGNER_CERT_SPC_CHAIN
+enum SIGNER_CERT_CHOICE : unsigned int
+{
+    SIGNER_CERT_SPC_FILE = 1U,
+    SIGNER_CERT_STORE = 2U,
+    SIGNER_CERT_SPC_CHAIN = 3U,
+};
+#pragma pop_macro("SIGNER_CERT_SPC_CHAIN")
+#pragma pop_macro("SIGNER_CERT_STORE")
+#pragma pop_macro("SIGNER_CERT_SPC_FILE")
+#endif
+
 #endif
\ No newline at end of file
