diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
index 848fdbd97f..015687e804 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
@@ -22,6 +22,9 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if !defined(_WINSOCK_DEPRECATED_BY)
 #if ((defined(_WINSOCK_DEPRECATED_NO_WARNINGS) || defined(BUILD_WINDOWS)) && !defined(_WINSOCK_DEPRECATE_WARNINGS)) || defined(MIDL_PASS)
@@ -125,6 +128,7 @@ extern "C" {
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Win32_SetLastError_
 _WINSOCK_DEPRECATED_BY("WSARecv()")
 #if(_WIN32_WINNT < 0x0600)
 int
