diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
index e067444153..d5df2b2750 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/MSWSock.h
@@ -22,6 +22,7 @@ Revision History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #if !defined(_WINSOCK_DEPRECATED_BY)
 #if ((defined(_WINSOCK_DEPRECATED_NO_WARNINGS) || defined(BUILD_WINDOWS)) && !defined(_WINSOCK_DEPRECATE_WARNINGS)) || defined(MIDL_PASS)
@@ -136,7 +137,7 @@ WSARecvEx(
     _Inout_ int FAR *flags
     );
 #else //(_WIN32_WINNT < 0x0600)
-INT
+_Windows_SupportedOS_Windows2000_ _Win32_SetLastError_ _Win32_ImportLibrary_("mswsock.dll") INT
 PASCAL FAR
 WSARecvEx(
     _In_ SOCKET s,
@@ -148,7 +149,7 @@ WSARecvEx(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
-typedef struct _TRANSMIT_FILE_BUFFERS {
+typedef struct   _TRANSMIT_FILE_BUFFERS {
     LPVOID Head;
     DWORD HeadLength;
     LPVOID Tail;
@@ -162,6 +163,7 @@ typedef struct _TRANSMIT_FILE_BUFFERS {
 #define TF_USE_SYSTEM_THREAD  0x10
 #define TF_USE_KERNEL_APC     0x20
 
+_Windows_SupportedOS_Server2003_
 BOOL
 PASCAL FAR
 TransmitFile (
@@ -174,6 +176,7 @@ TransmitFile (
     _In_  DWORD dwReserved
     );
 
+_Windows_SupportedOS_Server2003_
 BOOL
 PASCAL FAR
 AcceptEx (
@@ -187,6 +190,7 @@ AcceptEx (
     _Inout_ LPOVERLAPPED lpOverlapped
     );
 
+_Windows_SupportedOS_Server2003_
 VOID
 PASCAL FAR
 GetAcceptExSockaddrs (
@@ -395,7 +399,7 @@ typedef struct _NLA_BLOB {
         // header.type -> NLA_CONNECTIVITY
         struct {
             NLA_CONNECTIVITY_TYPE type;
-            NLA_INTERNET internet;
+            _Win32_AssociatedEnum_(NLA_INTERNET) NLA_INTERNET internet;
         } connectivity;
 
         // header.type -> NLA_ICS
@@ -635,7 +639,7 @@ typedef INT (PASCAL FAR * LPFN_RIONOTIFY)(
 );
 
 typedef RIO_BUFFERID (PASCAL FAR * LPFN_RIOREGISTERBUFFER)(
-    _In_ PCHAR DataBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _In_ PCHAR DataBuffer,
     _In_ DWORD DataLength
 );
 
@@ -650,7 +654,7 @@ typedef BOOL (PASCAL FAR * LPFN_RIORESIZEREQUESTQUEUE) (
     _In_ DWORD MaxOutstandingSend
 );
 
-typedef struct _RIO_EXTENSION_FUNCTION_TABLE {
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _RIO_EXTENSION_FUNCTION_TABLE {
     DWORD cbSize;
 
     LPFN_RIORECEIVE RIOReceive;
