diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h b/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
index ca51682f69..012288e91c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/netioapi.h
@@ -55,6 +55,7 @@ Notes:
 #define _NETIOAPI_H_
 #pragma once
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
@@ -114,7 +115,7 @@ extern "C" {
 
 #define NETIOAPI_API NETIO_STATUS NETIOAPI_API_
 
-typedef enum _MIB_NOTIFICATION_TYPE {
+typedef enum   _MIB_NOTIFICATION_TYPE {
     //
     // ParameterChange.
     //
@@ -151,7 +152,7 @@ typedef enum _MIB_NOTIFICATION_TYPE {
 
 #include <ntddndis.h>
 
-typedef struct _MIB_IF_ROW2 {
+typedef struct   _MIB_IF_ROW2 {
     //
     // Key structure.  Sorted by preference.
     //
@@ -218,12 +219,12 @@ typedef struct _MIB_IF_ROW2 {
     ULONG64 OutQLen;
 } MIB_IF_ROW2, *PMIB_IF_ROW2;
 
-typedef struct _MIB_IF_TABLE2 {
+typedef struct   _MIB_IF_TABLE2 {
     ULONG NumEntries;
     MIB_IF_ROW2 Table[ANY_SIZE];
 } MIB_IF_TABLE2, *PMIB_IF_TABLE2;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -270,7 +271,7 @@ typedef enum _MIB_IF_ENTRY_LEVEL {
     MibIfEntryNormalWithoutStatistics = 2
 } MIB_IF_ENTRY_LEVEL, *PMIB_IF_ENTRY_LEVEL;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_15063_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -312,7 +313,7 @@ Notes:
 --*/
 #endif
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -350,7 +351,7 @@ typedef enum _MIB_IF_TABLE_LEVEL {
     #endif
 } MIB_IF_TABLE_LEVEL, *PMIB_IF_TABLE_LEVEL;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -391,7 +392,7 @@ Notes:
 //
 // The MIB structure for Network layer Interface management routines.
 //
-typedef struct _MIB_IPINTERFACE_ROW {
+typedef struct   _MIB_IPINTERFACE_ROW {
     //
     // Key Structure;
     //
@@ -457,27 +458,27 @@ typedef struct _MIB_IPINTERFACE_ROW {
     BOOLEAN DisableDefaultRoutes;
 } MIB_IPINTERFACE_ROW, *PMIB_IPINTERFACE_ROW;
 
-typedef struct _MIB_IPINTERFACE_TABLE {
+typedef struct   _MIB_IPINTERFACE_TABLE {
     ULONG NumEntries;
     MIB_IPINTERFACE_ROW Table[ANY_SIZE];
 } MIB_IPINTERFACE_TABLE, *PMIB_IPINTERFACE_TABLE;
 
-typedef struct _MIB_IFSTACK_ROW {
+typedef struct   _MIB_IFSTACK_ROW {
     NET_IFINDEX HigherLayerInterfaceIndex;
     NET_IFINDEX LowerLayerInterfaceIndex;
 } MIB_IFSTACK_ROW, *PMIB_IFSTACK_ROW;
 
-typedef struct _MIB_INVERTEDIFSTACK_ROW {
+typedef struct   _MIB_INVERTEDIFSTACK_ROW {
     NET_IFINDEX LowerLayerInterfaceIndex;
     NET_IFINDEX HigherLayerInterfaceIndex;
 } MIB_INVERTEDIFSTACK_ROW, *PMIB_INVERTEDIFSTACK_ROW;
 
-typedef struct _MIB_IFSTACK_TABLE {
+typedef struct   _MIB_IFSTACK_TABLE {
     ULONG NumEntries;
     MIB_IFSTACK_ROW Table[ANY_SIZE];
 } MIB_IFSTACK_TABLE, *PMIB_IFSTACK_TABLE;
 
-typedef struct _MIB_INVERTEDIFSTACK_TABLE {
+typedef struct   _MIB_INVERTEDIFSTACK_TABLE {
     ULONG NumEntries;
     MIB_INVERTEDIFSTACK_ROW Table[ANY_SIZE];
 } MIB_INVERTEDIFSTACK_TABLE, *PMIB_INVERTEDIFSTACK_TABLE;
@@ -490,13 +491,13 @@ VOID
     _In_ MIB_NOTIFICATION_TYPE NotificationType
     );
 
-typedef struct _MIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES {
+typedef struct   _MIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES {
     NL_BANDWIDTH_INFORMATION InboundBandwidthInformation;
     NL_BANDWIDTH_INFORMATION OutboundBandwidthInformation;
 } MIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES,
   *PMIB_IP_NETWORK_CONNECTION_BANDWIDTH_ESTIMATES;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -504,7 +505,7 @@ GetIfStackTable(
     _Outptr_ PMIB_IFSTACK_TABLE *Table
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -512,7 +513,7 @@ GetInvertedIfStackTable(
     _Outptr_ PMIB_INVERTEDIFSTACK_TABLE *Table
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -547,7 +548,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -586,7 +587,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
 NETIOAPI_API_
@@ -615,7 +616,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -667,7 +668,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -703,7 +704,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_Windows8_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -748,7 +749,7 @@ Return Value:
 //
 // The structure for unicast IP Address management.
 //
-typedef struct _MIB_UNICASTIPADDRESS_ROW {
+typedef struct   _MIB_UNICASTIPADDRESS_ROW {
     //
     // Key Structure.
     //
@@ -774,7 +775,7 @@ typedef struct _MIB_UNICASTIPADDRESS_ROW {
     LARGE_INTEGER CreationTimeStamp;
 } MIB_UNICASTIPADDRESS_ROW, *PMIB_UNICASTIPADDRESS_ROW;
 
-typedef struct _MIB_UNICASTIPADDRESS_TABLE {
+typedef struct   _MIB_UNICASTIPADDRESS_TABLE {
     ULONG NumEntries;
     MIB_UNICASTIPADDRESS_ROW Table[ANY_SIZE];
 } MIB_UNICASTIPADDRESS_TABLE, *PMIB_UNICASTIPADDRESS_TABLE;
@@ -793,7 +794,7 @@ VOID
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -829,7 +830,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -860,7 +861,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -900,7 +901,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -946,7 +947,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
 NETIOAPI_API_
@@ -983,7 +984,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1042,7 +1043,7 @@ VOID
     _In_ PMIB_UNICASTIPADDRESS_TABLE AddressTable
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1060,7 +1061,7 @@ NotifyStableUnicastIpAddressTable(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1100,7 +1101,7 @@ Notes:
 // Anycast address management routines.
 //
 
-typedef struct _MIB_ANYCASTIPADDRESS_ROW {
+typedef struct   _MIB_ANYCASTIPADDRESS_ROW {
     //
     // Key Structure.
     //
@@ -1114,12 +1115,12 @@ typedef struct _MIB_ANYCASTIPADDRESS_ROW {
     SCOPE_ID ScopeId;
 } MIB_ANYCASTIPADDRESS_ROW, *PMIB_ANYCASTIPADDRESS_ROW;
 
-typedef struct _MIB_ANYCASTIPADDRESS_TABLE {
+typedef struct   _MIB_ANYCASTIPADDRESS_TABLE {
     ULONG NumEntries;
     MIB_ANYCASTIPADDRESS_ROW Table[ANY_SIZE];
 } MIB_ANYCASTIPADDRESS_TABLE, *PMIB_ANYCASTIPADDRESS_TABLE;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1150,7 +1151,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1181,7 +1182,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1215,7 +1216,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1259,7 +1260,7 @@ Notes:
 //
 // Multicast address management routines.
 //
-typedef struct _MIB_MULTICASTIPADDRESS_ROW {
+typedef struct   _MIB_MULTICASTIPADDRESS_ROW {
     //
     // Key Structure.
     //
@@ -1273,12 +1274,12 @@ typedef struct _MIB_MULTICASTIPADDRESS_ROW {
     SCOPE_ID ScopeId;
 } MIB_MULTICASTIPADDRESS_ROW, *PMIB_MULTICASTIPADDRESS_ROW;
 
-typedef struct _MIB_MULTICASTIPADDRESS_TABLE {
+typedef struct   _MIB_MULTICASTIPADDRESS_TABLE {
     ULONG NumEntries;
     MIB_MULTICASTIPADDRESS_ROW Table[ANY_SIZE];
 } MIB_MULTICASTIPADDRESS_TABLE, *PMIB_MULTICASTIPADDRESS_TABLE;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1312,7 +1313,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1362,12 +1363,12 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-typedef struct _IP_ADDRESS_PREFIX {
+typedef struct  _IP_ADDRESS_PREFIX {
     SOCKADDR_INET Prefix;
     UINT8 PrefixLength;
 } IP_ADDRESS_PREFIX, *PIP_ADDRESS_PREFIX;
 
-typedef struct _MIB_IPFORWARD_ROW2 {
+typedef struct   _MIB_IPFORWARD_ROW2 {
     //
     // Key Structure.
     //
@@ -1397,7 +1398,7 @@ typedef struct _MIB_IPFORWARD_ROW2 {
     NL_ROUTE_ORIGIN Origin;
 } MIB_IPFORWARD_ROW2, *PMIB_IPFORWARD_ROW2;
 
-typedef struct _MIB_IPFORWARD_TABLE2 {
+typedef struct   _MIB_IPFORWARD_TABLE2 {
     ULONG NumEntries;
     MIB_IPFORWARD_ROW2 Table[ANY_SIZE];
 } MIB_IPFORWARD_TABLE2, *PMIB_IPFORWARD_TABLE2;
@@ -1416,7 +1417,7 @@ VOID
     _In_ MIB_NOTIFICATION_TYPE NotificationType
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1453,7 +1454,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1491,7 +1492,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1548,7 +1549,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1590,7 +1591,7 @@ Notes:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1635,7 +1636,7 @@ Notes:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 VOID
 NETIOAPI_API_
@@ -1664,7 +1665,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1716,7 +1717,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1757,7 +1758,7 @@ Notes:
 // Path management routines.
 //
 
-typedef struct _MIB_IPPATH_ROW {
+typedef struct   _MIB_IPPATH_ROW {
     //
     // Key.
     //
@@ -1803,13 +1804,13 @@ typedef struct _MIB_IPPATH_ROW {
 
 } MIB_IPPATH_ROW, *PMIB_IPPATH_ROW;
 
-typedef struct _MIB_IPPATH_TABLE {
+typedef struct   _MIB_IPPATH_TABLE {
     ULONG NumEntries;
     MIB_IPPATH_ROW Table[ANY_SIZE];
 } MIB_IPPATH_TABLE, *PMIB_IPPATH_TABLE;
 
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1840,7 +1841,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1874,7 +1875,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1917,7 +1918,7 @@ Notes:
 // ARP and IPv6 Neighbor management routines.
 //
 
-typedef struct _MIB_IPNET_ROW2 {
+typedef struct   _MIB_IPNET_ROW2 {
     //
     // Key Struture.
     //
@@ -1950,12 +1951,12 @@ typedef struct _MIB_IPNET_ROW2 {
     } ReachabilityTime;
 } MIB_IPNET_ROW2, *PMIB_IPNET_ROW2;
 
-typedef struct _MIB_IPNET_TABLE2 {
+typedef struct   _MIB_IPNET_TABLE2 {
     ULONG NumEntries;
     MIB_IPNET_ROW2 Table[ANY_SIZE];
 } MIB_IPNET_TABLE2, *PMIB_IPNET_TABLE2;
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -1987,7 +1988,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2019,7 +2020,7 @@ Notes:
 --*/
 
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2056,7 +2057,7 @@ Return Value:
 --*/
 
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2090,7 +2091,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2129,7 +2130,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2170,7 +2171,7 @@ Notes:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2221,7 +2222,7 @@ VOID
     _Inout_ MIB_NOTIFICATION_TYPE NotificationType
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2232,7 +2233,7 @@ NotifyTeredoPortChange(
     _Inout_ HANDLE *NotificationHandle
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2279,7 +2280,7 @@ Return Value:
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
 CancelMibChangeNotify2(
@@ -2308,7 +2309,7 @@ Notes:
 
 --*/
 
-IPHLPAPI_DLL_LINKAGE
+_Windows_SupportedOS_WindowsVista_ IPHLPAPI_DLL_LINKAGE
 VOID
 NETIOAPI_API_
 FreeMibTable(
@@ -2336,7 +2337,7 @@ Return Value:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2401,7 +2402,7 @@ Return Value:
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2429,7 +2430,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2457,7 +2458,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2485,7 +2486,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2513,7 +2514,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2544,7 +2545,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2575,7 +2576,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2603,7 +2604,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2631,7 +2632,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2662,7 +2663,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2690,7 +2691,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2718,7 +2719,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2748,7 +2749,7 @@ Return Value:
 
 #define IF_NAMESIZE NDIS_IF_MAX_STRING_SIZE
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 NET_IFINDEX
 NETIOAPI_API_
@@ -2771,7 +2772,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_WindowsVista_ _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 PCHAR
 NETIOAPI_API_
@@ -2825,7 +2826,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2880,7 +2881,7 @@ Permissions:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2937,7 +2938,7 @@ Permissions:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -2993,7 +2994,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3041,7 +3042,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3078,7 +3079,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3111,7 +3112,7 @@ Return Value:
 
 #pragma warning(pop)
 
-IPHLPAPI_DLL_LINKAGE
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
 ConvertLengthToIpv4Mask(
     _In_ ULONG MaskLength,
@@ -3137,7 +3138,7 @@ Return Value:
 
 --*/
 
-IPHLPAPI_DLL_LINKAGE
+_Windows_SupportedOS_WindowsVista_ _Win32_AssociatedEnum_(WIN32_ERROR) IPHLPAPI_DLL_LINKAGE
 NETIOAPI_API
 ConvertIpv4MaskToLength(
     _In_ ULONG Mask,
@@ -3253,7 +3254,7 @@ typedef struct _DNS_SETTINGS2
     ULONG64 SettingFlags;
 } DNS_SETTINGS2;
 
-typedef struct _DNS_DOH_SERVER_SETTINGS
+typedef struct  _DNS_DOH_SERVER_SETTINGS
 {
 #ifdef MIDL_PASS
     [unique, string] PWSTR Template;
@@ -3275,7 +3276,7 @@ typedef struct _DNS_DOT_SERVER_SETTINGS
     USHORT Port;
 } DNS_DOT_SERVER_SETTINGS;
 
-typedef enum _DNS_SERVER_PROPERTY_TYPE
+typedef enum   _DNS_SERVER_PROPERTY_TYPE
 {
     DnsServerInvalidProperty = 0,
     DnsServerDohProperty,
@@ -3296,7 +3297,7 @@ typedef union _DNS_SERVER_PROPERTY_TYPES
 } DNS_SERVER_PROPERTY_TYPES;
 #endif
 
-typedef struct _DNS_SERVER_PROPERTY
+typedef struct  _DNS_SERVER_PROPERTY
 {
     ULONG Version;
     ULONG ServerIndex;
@@ -3370,7 +3371,7 @@ typedef struct _DNS_INTERFACE_SETTINGS4
     ULONG EncryptedDnsAdapterFlags;
 } DNS_INTERFACE_SETTINGS4;
 
-NETIOAPI_API
+_Win32_AssociatedEnum_(WIN32_ERROR) NETIOAPI_API
 GetDnsSettings(
     _Inout_ DNS_SETTINGS *Settings
     );
@@ -3417,7 +3418,7 @@ Return Value:
 
 --*/
 
-NETIOAPI_API
+_Win32_AssociatedEnum_(WIN32_ERROR) NETIOAPI_API
 SetDnsSettings(
     _In_ const DNS_SETTINGS *Settings
     );
@@ -3441,7 +3442,7 @@ Return Value:
 
 --*/
 
-NETIOAPI_API
+_Win32_AssociatedEnum_(WIN32_ERROR) NETIOAPI_API
 GetInterfaceDnsSettings(
     _In_ GUID Interface,
     _Inout_ DNS_INTERFACE_SETTINGS *Settings
@@ -3491,7 +3492,7 @@ Return Value:
 
 --*/
 
-NETIOAPI_API
+_Win32_AssociatedEnum_(WIN32_ERROR) NETIOAPI_API
 SetInterfaceDnsSettings(
     _In_ GUID Interface,
     _In_ const DNS_INTERFACE_SETTINGS *Settings
@@ -3518,7 +3519,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_19041_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3544,7 +3545,7 @@ Return Value:
 
 --*/
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_19041_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3582,7 +3583,7 @@ VOID
     _In_ NL_NETWORK_CONNECTIVITY_HINT ConnectivityHint
     );
 
-_IRQL_requires_max_(PASSIVE_LEVEL)
+_Windows_SupportedOS_19041_ _Win32_AssociatedEnum_(WIN32_ERROR) _IRQL_requires_max_(PASSIVE_LEVEL)
 IPHLPAPI_DLL_LINKAGE
 _NETIOAPI_SUCCESS_
 NETIOAPI_API
@@ -3590,7 +3591,7 @@ NotifyNetworkConnectivityHintChange(
     _In_ PNETWORK_CONNECTIVITY_HINT_CHANGE_CALLBACK Callback,
     _In_opt_ PVOID CallerContext,
     _In_ BOOLEAN InitialNotification,
-    _Out_ PHANDLE NotificationHandle
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE NotificationHandle
     );
 /*++
 
