diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/NotificationActivationCallback.h b/generation/WinSDK/RecompiledIdlHeaders/um/NotificationActivationCallback.h
index f1018eb11f..a60009480a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/NotificationActivationCallback.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/NotificationActivationCallback.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -68,7 +69,7 @@ extern "C"{
 #include <winapifamily.h>
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 #if (NTDDI_VERSION >= NTDDI_WIN10)
-typedef struct NOTIFICATION_USER_INPUT_DATA
+typedef struct _Windows_SupportedOS_10240_ NOTIFICATION_USER_INPUT_DATA
     {
     LPCWSTR Key;
     LPCWSTR Value;
@@ -94,6 +95,7 @@ EXTERN_C const IID IID_INotificationActivationCallback;
     INotificationActivationCallback : public IUnknown
     {
     public:
+        _Windows_SupportedOS_10240_
         virtual HRESULT STDMETHODCALLTYPE Activate( 
             /* [annotation][string][in] */ 
             _In_  LPCWSTR appUserModelId,
