diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ntenclv.h b/generation/WinSDK/RecompiledIdlHeaders/um/ntenclv.h
index 26882f4210..d38af989fa 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ntenclv.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ntenclv.h
@@ -17,11 +17,16 @@ Abstract:
 #ifndef _NTENCLV_
 #define _NTENCLV_
 
+#include <win32metadata_annotations.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef enum ENCLAVE_SEALING_IDENTITY_POLICY {
+typedef enum
+    
+    
+    ENCLAVE_SEALING_IDENTITY_POLICY {
     ENCLAVE_IDENTITY_POLICY_SEAL_INVALID = 0,
     ENCLAVE_IDENTITY_POLICY_SEAL_EXACT_CODE,
     ENCLAVE_IDENTITY_POLICY_SEAL_SAME_PRIMARY_CODE,
@@ -41,7 +46,10 @@ typedef enum ENCLAVE_SEALING_IDENTITY_POLICY {
 #pragma pack(push)
 #pragma pack(1)
 
-typedef struct ENCLAVE_IDENTITY {
+typedef struct
+    
+    
+    ENCLAVE_IDENTITY {
     UINT8 OwnerId[IMAGE_ENCLAVE_LONG_ID_LENGTH];
     UINT8 UniqueId[IMAGE_ENCLAVE_LONG_ID_LENGTH];
     UINT8 AuthorId[IMAGE_ENCLAVE_LONG_ID_LENGTH];
@@ -50,6 +58,7 @@ typedef struct ENCLAVE_IDENTITY {
     UINT32 EnclaveSvn;
     UINT32 SecureKernelSvn;
     UINT32 PlatformSvn;
+    
     UINT32 Flags;
     UINT32 SigningLevel;
     UINT32 EnclaveType;
@@ -59,11 +68,24 @@ typedef struct ENCLAVE_IDENTITY {
 #define ENCLAVE_FLAG_DYNAMIC_DEBUG_ENABLED      0x00000002
 #define ENCLAVE_FLAG_DYNAMIC_DEBUG_ACTIVE       0x00000004
 
+#ifdef WIN32METADATA
+enum [[clang::flag_enum]]
+    _Win32_AssociatedConstant_(ENCLAVE_FLAG_FULL_DEBUG_ENABLED)
+    _Win32_AssociatedConstant_(ENCLAVE_FLAG_DYNAMIC_DEBUG_ENABLED)
+    _Win32_AssociatedConstant_(ENCLAVE_FLAG_DYNAMIC_DEBUG_ACTIVE)
+    ENCLAVE_IDENTITY_FLAGS : unsigned int
+{
+};
+#endif
+
 #define VBS_ENCLAVE_REPORT_PKG_HEADER_VERSION_CURRENT (1)
 
 #define VBS_ENCLAVE_REPORT_SIGNATURE_SCHEME_SHA256_RSA_PSS_SHA256 (1)
 
-typedef struct VBS_ENCLAVE_REPORT_PKG_HEADER {
+typedef struct
+    
+    
+    VBS_ENCLAVE_REPORT_PKG_HEADER {
     UINT32 PackageSize;
     UINT32 Version;
     UINT32 SignatureScheme;
@@ -76,14 +98,21 @@ typedef struct VBS_ENCLAVE_REPORT_PKG_HEADER {
 
 #define ENCLAVE_REPORT_DATA_LENGTH      64
 
-typedef struct VBS_ENCLAVE_REPORT {
+typedef struct
+    
+    
+    VBS_ENCLAVE_REPORT {
     UINT32 ReportSize;
     UINT32 ReportVersion;
     UINT8 EnclaveData[ENCLAVE_REPORT_DATA_LENGTH];
     ENCLAVE_IDENTITY EnclaveIdentity;    
 } VBS_ENCLAVE_REPORT;
 
-typedef struct VBS_ENCLAVE_REPORT_VARDATA_HEADER {
+typedef struct
+    
+    
+    VBS_ENCLAVE_REPORT_VARDATA_HEADER {
+    
     UINT32 DataType;
     UINT32 Size;
 } VBS_ENCLAVE_REPORT_VARDATA_HEADER;
@@ -91,7 +120,19 @@ typedef struct VBS_ENCLAVE_REPORT_VARDATA_HEADER {
 #define VBS_ENCLAVE_VARDATA_INVALID     0x00000000
 #define VBS_ENCLAVE_VARDATA_MODULE      0x00000001
 
-typedef struct VBS_ENCLAVE_REPORT_MODULE {
+#ifdef WIN32METADATA
+enum
+    _Win32_AssociatedConstant_(VBS_ENCLAVE_VARDATA_INVALID)
+    _Win32_AssociatedConstant_(VBS_ENCLAVE_VARDATA_MODULE)
+    VBS_ENCLAVE_REPORT_DATA_TYPE : unsigned int
+{
+};
+#endif
+
+typedef struct
+    
+    
+    VBS_ENCLAVE_REPORT_MODULE {
     VBS_ENCLAVE_REPORT_VARDATA_HEADER Header;
     UINT8 UniqueId[IMAGE_ENCLAVE_LONG_ID_LENGTH];
     UINT8 AuthorId[IMAGE_ENCLAVE_LONG_ID_LENGTH];
@@ -104,7 +145,10 @@ typedef struct VBS_ENCLAVE_REPORT_MODULE {
 #pragma pack(pop)
 #pragma warning(pop)
 
-typedef struct ENCLAVE_INFORMATION {
+typedef struct
+    
+    
+    ENCLAVE_INFORMATION {
     ULONG EnclaveType;
     ULONG Reserved;
     PVOID BaseAddress;
