diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecPKG.h b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecPKG.h
index c55db34243..5863e66e1b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/NTSecPKG.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/NTSecPKG.h
@@ -19,6 +19,9 @@ Revision History:
 #define _NTSECPKG_
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -1266,7 +1269,7 @@ typedef NTSTATUS
 typedef NTSTATUS
 (NTAPI LSA_DUPLICATE_HANDLE)(
     _In_ HANDLE     SourceHandle,
-    _Out_ PHANDLE   DestionationHandle);
+    _Out_ _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE   DestionationHandle);
 
 
 typedef NTSTATUS
@@ -1278,6 +1281,7 @@ typedef NTSTATUS
     );
 
 
+_Win32_RAIIFree_(CloseHandle, -1, 0)
 typedef HANDLE
 (NTAPI LSA_CREATE_THREAD)(
     _In_ SEC_ATTRS SecurityAttributes,
@@ -1336,7 +1340,7 @@ typedef NTSTATUS
     _In_ PUNICODE_STRING AuthorityName,
     _In_opt_ PUNICODE_STRING Workstation,
     _In_opt_ PUNICODE_STRING ProfilePath,
-    _Out_ PHANDLE Token,
+    _Out_ _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE Token,
     _Out_ PNTSTATUS SubStatus
     );
 
@@ -1357,7 +1361,7 @@ typedef NTSTATUS
     _In_opt_ PUNICODE_STRING ProfilePath,
     _In_ PVOID SessionInformation,
     _In_ SECPKG_SESSIONINFO_TYPE SessionInformationType,
-    _Out_ PHANDLE Token,
+    _Out_ _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE Token,
     _Out_ PNTSTATUS SubStatus
     );
 
@@ -1591,8 +1595,19 @@ typedef NTSTATUS
 #define NOTIFY_CLASS_DOMAIN_CHANGE      3
 #define NOTIFY_CLASS_REGISTRY_CHANGE    4
 
+#ifdef WIN32METADATA
+enum SECPKG_PACKAGE_CHANGE_TYPE : int
+{
+    SECPKG_PACKAGE_CHANGE_LOAD = 0,
+    SECPKG_PACKAGE_CHANGE_UNLOAD = 1,
+    SECPKG_PACKAGE_CHANGE_SELECT = 2,
+};
+#else
+typedef ULONG SECPKG_PACKAGE_CHANGE_TYPE;
+#endif
+
 typedef struct _SECPKG_EVENT_PACKAGE_CHANGE {
-    ULONG   ChangeType;
+    SECPKG_PACKAGE_CHANGE_TYPE ChangeType;
     LSA_SEC_HANDLE  PackageId;
     SECURITY_STRING PackageName;
 } SECPKG_EVENT_PACKAGE_CHANGE, * PSECPKG_EVENT_PACKAGE_CHANGE;
@@ -1637,7 +1652,7 @@ typedef
 NTSTATUS
 (NTAPI LSA_OPEN_TOKEN_BY_LOGON_ID)(
     _In_ PLUID LogonId,
-    _Out_ HANDLE *RetTokenHandle
+    _Out_ _Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE *RetTokenHandle
     );
 
 typedef
@@ -2445,7 +2460,7 @@ typedef NTSTATUS
 typedef NTSTATUS
 (NTAPI SpGetContextTokenFn)(
     _In_ LSA_SEC_HANDLE ContextHandle,
-    _Out_ PHANDLE ImpersonationToken
+    _Out_ _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE ImpersonationToken
     );
 
 
@@ -2454,7 +2469,7 @@ typedef NTSTATUS
     _In_        LSA_SEC_HANDLE phContext,             // (in) context to export
     _In_        ULONG          fFlags,                // (in) option flags
     _Out_       PSecBuffer     pPackedContext,        // (out) marshalled context
-    _Out_opt_   PHANDLE        pToken                 // (out, optional) token handle for impersonation
+    _Out_opt_   _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE        pToken        // (out, optional) token handle for impersonation
     );
 
 typedef NTSTATUS
@@ -2744,7 +2759,7 @@ typedef NTSTATUS
 typedef NTSTATUS
 (NTAPI KspGetTokenFn)(
     _In_ LSA_SEC_HANDLE ContextId,
-    _Out_opt_ PHANDLE ImpersonationToken,
+    _Out_opt_ _Win32_RAIIFree_(CloseHandle, -1, 0) PHANDLE ImpersonationToken,
     _Outptr_opt_ PACCESS_TOKEN * RawToken
     );
 
