diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/oledb.h b/generation/WinSDK/RecompiledIdlHeaders/um/oledb.h
index 46f6243fba..6960915772 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/oledb.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/oledb.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -2480,7 +2481,7 @@ typedef struct tagDBPROPINFO
     LPOLESTR pwszDescription;
     DBPROPID dwPropertyID;
     DBPROPFLAGS dwFlags;
-    VARTYPE vtType;
+    _Win32_AssociatedEnum_(VARENUM) VARTYPE vtType;
     VARIANT vValues;
     } 	DBPROPINFO;
 
@@ -13095,9 +13096,9 @@ EXTERN_C const IID IID_IErrorLookup;
             /* [annotation][in] */ 
             _In_  LCID lcid,
             /* [annotation][out] */ 
-            _Outptr_result_maybenull_z_  BSTR *pbstrSource,
+            _Win32_RAIIFree_(SysFreeString) _Outptr_result_maybenull_z_  BSTR *pbstrSource,
             /* [annotation][out] */ 
-            _Outptr_result_maybenull_z_  BSTR *pbstrDescription) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Outptr_result_maybenull_z_  BSTR *pbstrDescription) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE GetHelpInfo( 
             /* [annotation][in] */ 
@@ -13107,7 +13108,7 @@ EXTERN_C const IID IID_IErrorLookup;
             /* [annotation][in] */ 
             _In_  LCID lcid,
             /* [annotation][out] */ 
-            _Outptr_result_maybenull_  BSTR *pbstrHelpFile,
+            _Win32_RAIIFree_(SysFreeString) _Outptr_result_maybenull_  BSTR *pbstrHelpFile,
             /* [annotation][out] */ 
             _Out_  DWORD *pdwHelpContext) = 0;
         
@@ -13312,7 +13313,7 @@ EXTERN_C const IID IID_ISQLErrorInfo;
     public:
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE GetSQLInfo( 
             /* [annotation][out] */ 
-            _Outptr_  BSTR *pbstrSQLState,
+            _Win32_RAIIFree_(SysFreeString) _Outptr_  BSTR *pbstrSQLState,
             /* [annotation][out] */ 
             _Out_  LONG *plNativeError) = 0;
         
