diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/opmxbox.h b/generation/WinSDK/RecompiledIdlHeaders/um/opmxbox.h
index 54f5fafd53..be0dd79daa 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/opmxbox.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/opmxbox.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //
 //---------------------------------------------------------------------------
 //
@@ -40,18 +41,18 @@ typedef enum OPM_HDCP_STATUS
 //
 // Request to Enable particular HDCP version
 //
-HRESULT OPMXboxEnableHDCP(
+_Win32_ImportLibrary_("opmxbox.dll") HRESULT OPMXboxEnableHDCP(
     OPM_HDCP_TYPE HDCPType);
 
 //
 // Retrieve HDCP on/off status
 //
-HRESULT OPMXboxGetHDCPStatus(OPM_HDCP_STATUS *pHDCPStatus);
+_Win32_ImportLibrary_("opmxbox.dll") HRESULT OPMXboxGetHDCPStatus(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") OPM_HDCP_STATUS *pHDCPStatus);
 
 //
 // Retrieve HDCP on/off status and type
 //
-HRESULT OPMXboxGetHDCPStatusAndType(OPM_HDCP_STATUS *pHDCPStatus, OPM_HDCP_TYPE *pHDCPType);
+_Win32_ImportLibrary_("opmxbox.dll") HRESULT OPMXboxGetHDCPStatusAndType(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") OPM_HDCP_STATUS *pHDCPStatus, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") OPM_HDCP_TYPE *pHDCPType);
 
 #ifdef __cplusplus 
 }
