diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/PasswordManager.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/PasswordManager.h
index 83669b61ba..8da53e2bce 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/PasswordManager.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/PasswordManager.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file PasswordManager.h is the top-level object that provides the interface to
  * set credentials used for the authentication of thin clients.
@@ -41,7 +42,7 @@ extern "C" {
  * @return
  *      - #ER_OK  if the credentials was successfully set.
  */
-extern AJ_API QStatus AJ_CALL alljoyn_passwordmanager_setcredentials(const char* authMechanism, const char* password);
+_Win32_ImportLibrary_("msajapi.dll")  extern AJ_API QStatus AJ_CALL alljoyn_passwordmanager_setcredentials(_WIN32META_ANNOTATION_("win32metadata:in") const char* authMechanism, _WIN32META_ANNOTATION_("win32metadata:in") const char* password);
 
 #ifdef __cplusplus
 } /* extern "C" */
