diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/PatchApi.h b/generation/WinSDK/RecompiledIdlHeaders/um/PatchApi.h
index 638fd924df..00ebbc3aff 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/PatchApi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/PatchApi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 // Copyright (c) Microsoft Corporation.  All rights reserved.
 //
 //  patchapi.h
@@ -462,7 +463,7 @@ ApplyPatchToFileByBuffers(
 //  application time thus decreasing total bytes necessary to download.
 //
 
-BOOL
+_Win32_ImportLibrary_("mspatcha.dll") BOOL
 PATCHAPI
 GetFilePatchSignatureA(
     _In_                                  LPCSTR FileName,
@@ -476,7 +477,7 @@ GetFilePatchSignatureA(
     _Out_writes_bytes_( SignatureBufferSize )   LPSTR  SignatureBuffer
     );
 
-BOOL
+_Win32_ImportLibrary_("mspatcha.dll") BOOL
 PATCHAPI
 GetFilePatchSignatureW(
     _In_                                  LPCWSTR FileName,
@@ -490,7 +491,7 @@ GetFilePatchSignatureW(
     _Out_writes_bytes_( SignatureBufferSize )   LPWSTR  SignatureBuffer
     );
 
-BOOL
+_Win32_ImportLibrary_("mspatcha.dll") BOOL
 PATCHAPI
 GetFilePatchSignatureByHandle(
     _In_                                  HANDLE  FileHandle,
@@ -504,7 +505,7 @@ GetFilePatchSignatureByHandle(
     _Out_writes_bytes_( SignatureBufferSize )   LPSTR   SignatureBuffer
     );
 
-BOOL
+_Win32_ImportLibrary_("mspatcha.dll") BOOL
 PATCHAPI
 GetFilePatchSignatureByBuffer(
     _Inout_updates_bytes_( FileSize )            PBYTE   FileBufferWritable,   // modified!
@@ -528,7 +529,7 @@ GetFilePatchSignatureByBuffer(
 //  indicate that the buffer was modified in the process of normalization.
 //
 
-INT
+_Win32_ImportLibrary_("mspatcha.dll") INT
 WINAPI
 NormalizeFileForPatchSignature(
     _Inout_updates_bytes_( FileSize )          PVOID FileBuffer,       // modified!
