diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/playlist.h b/generation/WinSDK/RecompiledIdlHeaders/um/playlist.h
index eeac8755d7..460fbfdfd6 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/playlist.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/playlist.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -131,7 +132,7 @@ EXTERN_C const IID IID_IAMPlayListItem;
         
         virtual HRESULT STDMETHODCALLTYPE GetSourceURL( 
             /* [in] */ DWORD dwSourceIndex,
-            /* [out] */ __RPC__deref_out_opt BSTR *pbstrURL) = 0;
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrURL) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetSourceStart( 
             /* [in] */ DWORD dwSourceIndex,
@@ -151,14 +152,14 @@ EXTERN_C const IID IID_IAMPlayListItem;
         
         virtual HRESULT STDMETHODCALLTYPE GetSourceStartMarkerName( 
             /* [in] */ DWORD dwSourceIndex,
-            /* [out] */ __RPC__deref_out_opt BSTR *pbstrStartMarker) = 0;
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrStartMarker) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetSourceEndMarkerName( 
             /* [in] */ DWORD dwSourceIndex,
-            /* [out] */ __RPC__deref_out_opt BSTR *pbstrEndMarker) = 0;
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrEndMarker) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetLinkURL( 
-            /* [out] */ __RPC__deref_out_opt BSTR *pbstrURL) = 0;
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrURL) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetScanDuration( 
             /* [in] */ DWORD dwSourceIndex,
