diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/pnrpns.h b/generation/WinSDK/RecompiledIdlHeaders/um/pnrpns.h
index ff2dc8a3f0..a5ec836720 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/pnrpns.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/pnrpns.h
@@ -17,6 +17,7 @@ Abstract:
 #ifndef _PNRPNS_H_
 #define _PNRPNS_H_
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -48,7 +49,7 @@ Abstract:
 #pragma warning(disable:4201) // nameless struct/union
 
 
-typedef struct _PNRPINFO_V1
+typedef struct  _PNRPINFO_V1
 {
     DWORD                       dwSize;                 // size of this struct
     LPWSTR                      lpwszIdentity;          // identity name string
