diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/PrintManagerInterop.h b/generation/WinSDK/RecompiledIdlHeaders/um/PrintManagerInterop.h
index a6202fb958..b92f09067a 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/PrintManagerInterop.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/PrintManagerInterop.h
@@ -19,6 +19,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -90,6 +91,8 @@ EXTERN_C const IID IID_IPrintManagerInterop;
     IPrintManagerInterop : public IInspectable
     {
     public:
+        _Windows_SupportedOS_Windows8_
+        _Windows_SupportedOS_Server2012_
         virtual HRESULT STDMETHODCALLTYPE GetForWindow( 
             /* [annotation][in] */ 
             _In_  HWND appWindow,
@@ -98,6 +101,8 @@ EXTERN_C const IID IID_IPrintManagerInterop;
             /* [annotation][iid_is][retval][out] */ 
             _COM_Outptr_retval_  void **printManager) = 0;
         
+        _Windows_SupportedOS_Windows8_
+        _Windows_SupportedOS_Server2012_
         virtual HRESULT STDMETHODCALLTYPE ShowPrintUIForWindowAsync( 
             /* [annotation][in] */ 
             _In_  HWND appWindow,
