diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h b/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
index ab0ef556ea..aa5511020d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prnasnot.h
@@ -15,6 +15,7 @@ Revision History:
 --*/
 #ifndef _PRINTASYNCNOTIFY_H_
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -57,14 +58,14 @@ DEFINE_GUID(PRINT_PORT_MONITOR_NOTIFY_CHANNEL, 0x25df3b0e, 0x74a9, 0x47f5, 0x80,
 extern "C" {
 #endif
 
-typedef enum
+typedef enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_
 {
     kPerUser,
     kAllUsers,
 
 } PrintAsyncNotifyUserFilter;
 
-typedef enum
+typedef enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_
 {
     kBiDirectional,
     kUniDirectional
@@ -93,6 +94,8 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyDataObject, IUnknown)
         THIS
         ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
     STDMETHOD(AcquireData)(
          THIS_
          _Outptr_opt_result_buffer_(*pSize) BYTE             **ppNotificationData,
@@ -100,6 +103,8 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyDataObject, IUnknown)
          _Outptr_opt_                   PrintAsyncNotificationType **ppSchema
          ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
     STDMETHOD(ReleaseData)(
         THIS
         ) PURE;
@@ -125,11 +130,17 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyChannel, IUnknown)
         THIS
         ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
+    _Win32_PreserveResult_
     STDMETHOD(SendNotification)(
          THIS_
          _In_ IPrintAsyncNotifyDataObject *pData
          ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
+    _Win32_PreserveResult_
     STDMETHOD(CloseChannel)(
          THIS_
          _In_ IPrintAsyncNotifyDataObject *pData
@@ -155,12 +166,18 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyCallback, IUnknown)
         THIS
         ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
+    _Win32_PreserveResult_
     STDMETHOD(OnEventNotify)(
          THIS_
          _In_ IPrintAsyncNotifyChannel    *pChannel,
          _In_ IPrintAsyncNotifyDataObject *pData
          ) PURE;
 
+    _Windows_SupportedOS_WindowsVista_
+    _Windows_SupportedOS_Server2008_
+    _Win32_PreserveResult_
     STDMETHOD(ChannelClosed)(
          THIS_
          _In_ IPrintAsyncNotifyChannel    *pChannel,
@@ -170,25 +187,34 @@ DECLARE_INTERFACE_(IPrintAsyncNotifyCallback, IUnknown)
 
 #undef INTERFACE
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2008_
 HRESULT
 WINAPI
+_Win32_PreserveResult_
 RegisterForPrintAsyncNotifications(
     _In_opt_ PCWSTR                            pszName,
     _In_     PrintAsyncNotificationType*       pNotificationType,
              PrintAsyncNotifyUserFilter        eUserFilter,
              PrintAsyncNotifyConversationStyle eConversationStyle,
     _In_     IPrintAsyncNotifyCallback*        pCallback,
-    _Out_    HANDLE*                           phNotify
+    _Out_    _Win32_RAIIFree_(UnRegisterForPrintAsyncNotifications) HANDLE*                           phNotify
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2008_
 HRESULT
 WINAPI
+_Win32_PreserveResult_
 UnRegisterForPrintAsyncNotifications(
     _In_     HANDLE
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2008_
 HRESULT
 WINAPI
+_Win32_PreserveResult_
 CreatePrintAsyncNotifyChannel(
     _In_opt_ PCWSTR                            pszName,
     _In_     PrintAsyncNotificationType*       pNotificationType,
@@ -198,7 +224,7 @@ CreatePrintAsyncNotifyChannel(
     _Out_    IPrintAsyncNotifyChannel**        ppIAsynchNotification
     );
 
-typedef enum
+typedef enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_
 {
     CHANNEL_CLOSED_BY_SERVER                    = 0x01,
     CHANNEL_CLOSED_BY_ANOTHER_LISTENER          = 0x02,
