diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prnasntp.h b/generation/WinSDK/RecompiledIdlHeaders/um/prnasntp.h
index deb35ff187..6787c57a1e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prnasntp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prnasntp.h
@@ -18,6 +18,7 @@ Revision History:
 
 #ifndef _PRINTASYNCNOTIFYP_H_
 #define _PRINTASYNCNOTIFYP_H_
+#include <win32metadata_annotations.h>
 
 #include <winapifamily.h>
 
@@ -475,7 +476,7 @@ RouterRegisterForPrintAsyncNotifications(
     _In_  PrintAsyncNotifyUserFilter         eNotifyFilter,
     _In_  PrintAsyncNotifyConversationStyle  eConversationStyle,
     _In_  IPrintAsyncNotifyCallback          *pCallback,
-    _Out_writes_bytes_(sizeof(HANDLE))
+    _Win32_RAIIFree_(RouterUnregisterForPrintAsyncNotifications) _Out_writes_bytes_(sizeof(HANDLE))
           HANDLE                             *phNotify
     );
 
