diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/prntvpt.h b/generation/WinSDK/RecompiledIdlHeaders/um/prntvpt.h
index a080747413..680afb78f0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/prntvpt.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/prntvpt.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -63,13 +64,13 @@ error in the driver.
 */
 #define E_PRINTDEVICECAPABILITIES_FORMAT       (0x80040006)
 
-typedef enum tagEDefaultDevmodeType
+typedef enum   tagEDefaultDevmodeType
 {
     kUserDefaultDevmode,
     kPrinterDefaultDevmode
 } EDefaultDevmodeType;
 
-typedef enum 
+typedef enum   
 {
     kPTPageScope,
     kPTDocumentScope,
@@ -77,85 +78,97 @@ typedef enum
 } EPrintTicketScope;
 
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTQuerySchemaVersionSupport(
             _In_        PCWSTR       pszPrinterName,
             _Out_       DWORD       *pMaxVersion
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTOpenProvider( 
             _In_        PCWSTR       pszPrinterName,
                         DWORD        dwVersion,
-            _Out_       HPTPROVIDER *phProvider
+            _Win32_RAIIFree_(PTCloseProvider, -1, 0) _Out_       HPTPROVIDER *phProvider
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTOpenProviderEx( 
             _In_  PCWSTR       pszPrinterName,
                   DWORD        dwMaxVersion,
                   DWORD        dwPrefVersion,
-            _Out_ HPTPROVIDER *phProvider,
+            _Win32_RAIIFree_(PTCloseProvider, -1, 0) _Out_ HPTPROVIDER *phProvider,
             _Out_ DWORD       *pUsedVersion
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTCloseProvider(
             _In_  HPTPROVIDER hProvider
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTReleaseMemory(
     _In_ PVOID pBuffer
     );
     
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTGetPrintCapabilities( 
             _In_            HPTPROVIDER         hProvider,
             _In_opt_        IStream            *pPrintTicket,
             _Inout_         IStream            *pCapabilities,
-            _Out_opt_       BSTR               *pbstrErrorMessage
+            _Out_opt_       _Win32_RAIIFree_(SysFreeString) BSTR               *pbstrErrorMessage
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_15063_ HRESULT WINAPI
+ 
 PTGetPrintDeviceCapabilities(
     _In_            HPTPROVIDER         hProvider,
     _In_opt_        IStream            *pPrintTicket,
     _Inout_         IStream            *pDeviceCapabilities,
-    _Out_opt_       BSTR               *pbstrErrorMessage
+    _Win32_RAIIFree_(SysFreeString) _Out_opt_       BSTR               *pbstrErrorMessage
     );
 
-HRESULT WINAPI
+_Windows_SupportedOS_15063_ HRESULT WINAPI
+ 
 PTGetPrintDeviceResources(
     _In_            HPTPROVIDER         hProvider,
     _In_            LPCWSTR             pszLocaleName,
     _In_opt_        IStream            *pPrintTicket,
     _Inout_         IStream            *pDeviceResources,
-    _Out_opt_       BSTR               *pbstrErrorMessage
+    _Win32_RAIIFree_(SysFreeString) _Out_opt_       BSTR               *pbstrErrorMessage
 );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
+
 PTMergeAndValidatePrintTicket(
             _In_            HPTPROVIDER        hProvider,
             _In_            IStream           *pBaseTicket,
             _In_opt_        IStream           *pDeltaTicket,
             _In_            EPrintTicketScope  scope,
             _Inout_         IStream           *pResultTicket, 
-            _Out_opt_       BSTR              *pbstrErrorMessage 
+            _Out_opt_       _Win32_RAIIFree_(SysFreeString) BSTR              *pbstrErrorMessage 
             );
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTConvertPrintTicketToDevMode( 
             _In_                     HPTPROVIDER          hProvider,
             _In_                     IStream             *pPrintTicket,
                                      EDefaultDevmodeType  baseDevmodeType,
             _In_                     EPrintTicketScope  scope,
             _Out_                    ULONG               *pcbDevmode,
-            _Outptr_result_bytebuffer_(*pcbDevmode) OUT PDEVMODE *ppDevmode,
-            _Out_opt_                BSTR                *pbstrErrorMessage 
+            _Outptr_result_bytebuffer_(*pcbDevmode) OUT  PDEVMODE *ppDevmode,
+            _Out_opt_                _Win32_RAIIFree_(SysFreeString) BSTR                *pbstrErrorMessage 
             );
         
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsXP_ HRESULT WINAPI
+ 
 PTConvertDevModeToPrintTicket( 
             _In_        HPTPROVIDER        hProvider,
                         ULONG              cbDevmode,
