diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/processthreadsapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/processthreadsapi.h
index 0d0863017f..0ae487980e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/processthreadsapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/processthreadsapi.h
@@ -15,6 +15,7 @@
 
 #include <apiset.h>
 #include <apisetcconv.h>
+#include <win32metadata_annotations.h>
 #include <minwindef.h>
 #include <minwinbase.h>
 
@@ -44,7 +45,7 @@ typedef struct _STARTUPINFOA {
     DWORD   dwXCountChars;
     DWORD   dwYCountChars;
     DWORD   dwFillAttribute;
-    DWORD   dwFlags;
+    _Win32_AssociatedEnum_(STARTUPINFOW_FLAGS) DWORD   dwFlags;
     WORD    wShowWindow;
     WORD    cbReserved2;
     LPBYTE  lpReserved2;
@@ -64,7 +65,7 @@ typedef struct _STARTUPINFOW {
     DWORD   dwXCountChars;
     DWORD   dwYCountChars;
     DWORD   dwFillAttribute;
-    DWORD   dwFlags;
+    _Win32_AssociatedEnum_(STARTUPINFOW_FLAGS) DWORD   dwFlags;
     WORD    wShowWindow;
     WORD    cbReserved2;
     LPBYTE  lpReserved2;
@@ -88,11 +89,14 @@ typedef LPSTARTUPINFOA LPSTARTUPINFO;
 
 #if (_WIN32_WINNT >= 0x0400) || (_WIN32_WINDOWS > 0x0400)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
 QueueUserAPC(
-    _In_ PAPCFUNC pfnAPC,
+    _In_  PAPCFUNC pfnAPC,
     _In_ HANDLE hThread,
     _In_ ULONG_PTR dwData
     );
@@ -119,11 +123,12 @@ typedef struct _APC_CALLBACK_DATA {
     ULONG_PTR Reserved1;
 } APC_CALLBACK_DATA, *PAPC_CALLBACK_DATA;
 
+
 WINBASEAPI
 BOOL
 WINAPI
 QueueUserAPC2(
-    _In_ PAPCFUNC ApcRoutine,
+    _In_  PAPCFUNC ApcRoutine,
     _In_ HANDLE Thread,
     _In_ ULONG_PTR Data,
     _In_ QUEUE_USER_APC_FLAGS Flags
@@ -131,6 +136,9 @@ QueueUserAPC2(
 
 #endif /* NTDDI_VERSION >= NTDDI_WIN10_MN */
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -142,6 +150,9 @@ GetProcessTimes(
     _Out_ LPFILETIME lpUserTime
     );
 
+_WIN32META_ANNOTATION_("win32metadata:do_not_release") _Windows_SupportedOS_WindowsXP_ 
+
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 HANDLE
 WINAPI
@@ -149,6 +160,8 @@ GetCurrentProcess(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
 WINBASEAPI
 DWORD
 WINAPI
@@ -156,6 +169,8 @@ GetCurrentProcessId(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
 WINBASEAPI
 DECLSPEC_NORETURN
 VOID
@@ -164,6 +179,10 @@ ExitProcess(
     _In_ UINT uExitCode
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 BOOL
 WINAPI
@@ -172,6 +191,9 @@ TerminateProcess(
     _In_ UINT uExitCode
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -180,6 +202,8 @@ GetExitCodeProcess(
     _Out_ LPDWORD lpExitCode
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
 WINBASEAPI
 BOOL
 WINAPI
@@ -187,6 +211,10 @@ SwitchToThread(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) 
+
+_Win32_SetLastError_
+
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -194,9 +222,9 @@ WINAPI
 CreateThread(
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ SIZE_T dwStackSize,
-    _In_ LPTHREAD_START_ROUTINE lpStartAddress,
-    _In_opt_ __drv_aliasesMem LPVOID lpParameter,
-    _In_ DWORD dwCreationFlags,
+    _In_  LPTHREAD_START_ROUTINE lpStartAddress,
+    _In_opt_ __drv_aliasesMem  LPVOID lpParameter,
+    _Win32_AssociatedEnum_(THREAD_CREATION_FLAGS) _In_ DWORD dwCreationFlags,
     _Out_opt_ LPDWORD lpThreadId
     );
 
@@ -205,6 +233,10 @@ CreateThread(
 
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) 
+
+_Win32_SetLastError_
+
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -213,8 +245,8 @@ CreateRemoteThread(
     _In_ HANDLE hProcess,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ SIZE_T dwStackSize,
-    _In_ LPTHREAD_START_ROUTINE lpStartAddress,
-    _In_opt_ LPVOID lpParameter,
+    _In_  LPTHREAD_START_ROUTINE lpStartAddress,
+    _In_opt_  LPVOID lpParameter,
     _In_ DWORD dwCreationFlags,
     _Out_opt_ LPDWORD lpThreadId
     );
@@ -225,6 +257,9 @@ CreateRemoteThread(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) 
+
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 HANDLE
 WINAPI
@@ -232,6 +267,9 @@ GetCurrentThread(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 DWORD
 WINAPI
@@ -239,24 +277,34 @@ GetCurrentThreadId(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) 
+
+_Win32_SetLastError_
+
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
 WINAPI
 OpenThread(
-    _In_ DWORD dwDesiredAccess,
+    _Win32_AssociatedEnum_(THREAD_ACCESS_RIGHTS) _In_ DWORD dwDesiredAccess,
     _In_ BOOL bInheritHandle,
     _In_ DWORD dwThreadId
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetThreadPriority(
     _In_ HANDLE hThread,
-    _In_ int nPriority
+    _Win32_AssociatedEnum_(THREAD_PRIORITY) _In_ int nPriority
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -265,6 +313,9 @@ SetThreadPriorityBoost(
     _In_ BOOL bDisablePriorityBoost
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -273,6 +324,9 @@ GetThreadPriorityBoost(
     _Out_ PBOOL pDisablePriorityBoost
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -280,6 +334,8 @@ GetThreadPriority(
     _In_ HANDLE hThread
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
 WINBASEAPI
 DECLSPEC_NORETURN
 VOID
@@ -293,6 +349,9 @@ ExitThread(
 
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -307,6 +366,9 @@ TerminateThread(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0)
 BOOL
@@ -316,6 +378,9 @@ GetExitCodeThread(
     _Out_ LPDWORD lpExitCode
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -323,6 +388,9 @@ SuspendThread(
     _In_ HANDLE hThread
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -334,6 +402,10 @@ ResumeThread(
 #define TLS_OUT_OF_INDEXES ((DWORD)0xFFFFFFFF)
 #endif
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 _Must_inspect_result_
 WINBASEAPI
 DWORD
@@ -342,6 +414,10 @@ TlsAlloc(
     VOID
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 LPVOID
 WINAPI
@@ -349,6 +425,10 @@ TlsGetValue(
     _In_ DWORD dwTlsIndex
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 BOOL
 WINAPI
@@ -357,6 +437,10 @@ TlsSetValue(
     _In_opt_ LPVOID lpTlsValue
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 BOOL
 WINAPI
@@ -364,6 +448,9 @@ TlsFree(
     _In_ DWORD dwTlsIndex
     );
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -373,13 +460,16 @@ CreateProcessA(
     _In_opt_ LPSECURITY_ATTRIBUTES lpProcessAttributes,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ BOOL bInheritHandles,
-    _In_ DWORD dwCreationFlags,
+    _Win32_AssociatedEnum_(PROCESS_CREATION_FLAGS) _In_ DWORD dwCreationFlags,
     _In_opt_ LPVOID lpEnvironment,
     _In_opt_ LPCSTR lpCurrentDirectory,
     _In_ LPSTARTUPINFOA lpStartupInfo,
     _Out_ LPPROCESS_INFORMATION lpProcessInformation
     );
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -389,7 +479,7 @@ CreateProcessW(
     _In_opt_ LPSECURITY_ATTRIBUTES lpProcessAttributes,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ BOOL bInheritHandles,
-    _In_ DWORD dwCreationFlags,
+    _Win32_AssociatedEnum_(PROCESS_CREATION_FLAGS) _In_ DWORD dwCreationFlags,
     _In_opt_ LPVOID lpEnvironment,
     _In_opt_ LPCWSTR lpCurrentDirectory,
     _In_ LPSTARTUPINFOW lpStartupInfo,
@@ -408,6 +498,9 @@ CreateProcessW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -416,6 +509,9 @@ SetProcessShutdownParameters(
     _In_ DWORD dwFlags
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -429,6 +525,8 @@ GetProcessVersion(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ 
+
 WINBASEAPI
 VOID
 WINAPI
@@ -446,6 +544,9 @@ GetStartupInfoW(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -456,7 +557,7 @@ CreateProcessAsUserW(
     _In_opt_ LPSECURITY_ATTRIBUTES lpProcessAttributes,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ BOOL bInheritHandles,
-    _In_ DWORD dwCreationFlags,
+    _Win32_AssociatedEnum_(PROCESS_CREATION_FLAGS) _In_ DWORD dwCreationFlags,
     _In_opt_ LPVOID lpEnvironment,
     _In_opt_ LPCWSTR lpCurrentDirectory,
     _In_ LPSTARTUPINFOW lpStartupInfo,
@@ -487,6 +588,8 @@ CreateProcessAsUserW(
 #if !defined(MIDL_PASS)
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_WIN32META_ANNOTATION_("win32metadata:do_not_release") _Win32_ImportLibrary_("forceinline") 
+
 FORCEINLINE
 HANDLE
 GetCurrentProcessToken (
@@ -496,6 +599,8 @@ GetCurrentProcessToken (
     return (HANDLE)(LONG_PTR) -4;
 }
 
+_WIN32META_ANNOTATION_("win32metadata:do_not_release") _Win32_ImportLibrary_("forceinline") 
+
 FORCEINLINE
 HANDLE
 GetCurrentThreadToken (
@@ -505,6 +610,8 @@ GetCurrentThreadToken (
     return (HANDLE)(LONG_PTR) -5;
 }
 
+_WIN32META_ANNOTATION_("win32metadata:do_not_release") _Win32_ImportLibrary_("forceinline") 
+
 FORCEINLINE
 HANDLE
 GetCurrentThreadEffectiveToken (
@@ -517,6 +624,9 @@ GetCurrentThreadEffectiveToken (
 #endif // (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 #endif // !defined(MIDL_PASS)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINADVAPI
 _Must_inspect_result_
 BOOL
@@ -532,13 +642,16 @@ SetThreadToken(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
 OpenProcessToken(
     _In_ HANDLE ProcessHandle,
-    _In_ DWORD DesiredAccess,
-    _Outptr_ PHANDLE TokenHandle
+    _Win32_AssociatedEnum_(TOKEN_ACCESS_MASK) _In_ DWORD DesiredAccess,
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Outptr_  PHANDLE TokenHandle
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
@@ -547,24 +660,33 @@ OpenProcessToken(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
 OpenThreadToken(
     _In_ HANDLE ThreadHandle,
-    _In_ DWORD DesiredAccess,
+    _Win32_AssociatedEnum_(TOKEN_ACCESS_MASK) _In_ DWORD DesiredAccess,
     _In_ BOOL OpenAsSelf,
-    _Outptr_ PHANDLE TokenHandle
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Outptr_  PHANDLE TokenHandle
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetPriorityClass(
     _In_ HANDLE hProcess,
-    _In_ DWORD dwPriorityClass
+    _Win32_AssociatedEnum_(PROCESS_CREATION_FLAGS) _In_ DWORD dwPriorityClass
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -578,6 +700,8 @@ GetPriorityClass(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 BOOL
 WINAPI
@@ -591,6 +715,9 @@ SetThreadStackGuarantee(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -605,7 +732,7 @@ ProcessIdToSessionId(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-typedef struct _PROC_THREAD_ATTRIBUTE_LIST *PPROC_THREAD_ATTRIBUTE_LIST, *LPPROC_THREAD_ATTRIBUTE_LIST;
+_WIN32META_ANNOTATION_("win32metadata:raii_free=DeleteProcThreadAttributeList") typedef struct _PROC_THREAD_ATTRIBUTE_LIST *PPROC_THREAD_ATTRIBUTE_LIST, *LPPROC_THREAD_ATTRIBUTE_LIST;
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
@@ -615,6 +742,7 @@ typedef struct _PROC_THREAD_ATTRIBUTE_LIST *PPROC_THREAD_ATTRIBUTE_LIST, *LPPROC
 
 #if (_WIN32_WINNT >= 0x0501)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -626,6 +754,9 @@ GetProcessId(
 
 #if (_WIN32_WINNT >= 0x0502)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -637,6 +768,8 @@ GetThreadId(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 VOID
 WINAPI
@@ -654,6 +787,9 @@ FlushProcessWriteBuffers(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -661,17 +797,22 @@ GetProcessIdOfThread(
     _In_ HANDLE Thread
     );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
 WINAPI
 InitializeProcThreadAttributeList(
-    _Out_writes_bytes_to_opt_(*lpSize,*lpSize) LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
+    _Win32_RAIIFree_(DeleteProcThreadAttributeList) _Out_writes_bytes_to_opt_(*lpSize,*lpSize) LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
     _In_ DWORD dwAttributeCount,
-    _Reserved_ DWORD dwFlags,
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ DWORD dwFlags,
     _When_(lpAttributeList == nullptr,_Out_) _When_(lpAttributeList != nullptr,_Inout_) PSIZE_T lpSize
     );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 VOID
 WINAPI
@@ -681,6 +822,9 @@ DeleteProcThreadAttributeList(
 
 #define PROC_THREAD_ATTRIBUTE_REPLACE_VALUE     0x00000001
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -688,7 +832,7 @@ UpdateProcThreadAttribute(
     _Inout_ LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
     _In_ DWORD dwFlags,
     _In_ DWORD_PTR Attribute,
-    _In_reads_bytes_opt_(cbSize) PVOID lpValue,
+    _In_reads_bytes_opt_(cbSize)  PVOID lpValue,
     _In_ SIZE_T cbSize,
     _Out_writes_bytes_opt_(cbSize) PVOID lpPreviousValue,
     _In_opt_ PSIZE_T lpReturnSize
@@ -696,6 +840,7 @@ UpdateProcThreadAttribute(
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_VB)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -709,6 +854,8 @@ SetProcessDynamicEHContinuationTargets(
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_FE)
 
+
+
 WINBASEAPI
 BOOL
 WINAPI
@@ -728,20 +875,31 @@ SetProcessDynamicEnforcedCetCompatibleRanges(
 
 #define PROCESS_AFFINITY_ENABLE_AUTO_UPDATE 0x00000001UL
 
+#ifdef WIN32METADATA
+#pragma push_macro("PROCESS_AFFINITY_ENABLE_AUTO_UPDATE")
+#undef PROCESS_AFFINITY_ENABLE_AUTO_UPDATE
+enum PROCESS_AFFINITY_AUTO_UPDATE_FLAGS : DWORD {
+    PROCESS_AFFINITY_ENABLE_AUTO_UPDATE = 0x00000001UL,
+};
+#pragma pop_macro("PROCESS_AFFINITY_ENABLE_AUTO_UPDATE")
+#endif
+
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetProcessAffinityUpdateMode(
     _In_ HANDLE hProcess,
-    _In_ DWORD dwFlags
+    _In_ _Win32_AssociatedEnum_(PROCESS_AFFINITY_AUTO_UPDATE_FLAGS) DWORD dwFlags
     );
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 QueryProcessAffinityUpdateMode(
     _In_ HANDLE hProcess,
-    _Out_opt_ LPDWORD lpdwFlags
+    _Out_opt_ _Win32_AssociatedEnum_(PROCESS_AFFINITY_AUTO_UPDATE_FLAGS) LPDWORD lpdwFlags
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM) */
@@ -752,6 +910,9 @@ QueryProcessAffinityUpdateMode(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_RAIIFree_(CloseHandle, -1, 0) _Windows_SupportedOS_Windows7_
+_Win32_SetLastError_
+
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -760,8 +921,8 @@ CreateRemoteThreadEx(
     _In_ HANDLE hProcess,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ SIZE_T dwStackSize,
-    _In_ LPTHREAD_START_ROUTINE lpStartAddress,
-    _In_opt_ LPVOID lpParameter,
+    _In_  LPTHREAD_START_ROUTINE lpStartAddress,
+    _In_opt_  LPVOID lpParameter,
     _In_ DWORD dwCreationFlags,
     _In_opt_ LPPROC_THREAD_ATTRIBUTE_LIST lpAttributeList,
     _Out_opt_ LPDWORD lpThreadId
@@ -777,6 +938,8 @@ CreateRemoteThreadEx(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Windows_SupportedOS_Windows8_ 
+
 WINBASEAPI
 VOID
 WINAPI
@@ -787,6 +950,9 @@ GetCurrentThreadStackLimits(
 
 #endif
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -803,6 +969,9 @@ GetThreadContext(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -821,6 +990,9 @@ GetProcessMitigationPolicy(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -837,6 +1009,9 @@ SetThreadContext(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -856,6 +1031,9 @@ SetProcessMitigationPolicy(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -871,6 +1049,9 @@ FlushInstructionCache(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -882,20 +1063,27 @@ GetThreadTimes(
     _Out_ LPFILETIME lpUserTime
     );
 
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) 
+
+_Win32_SetLastError_
+
 WINBASEAPI
 HANDLE
 WINAPI
 OpenProcess(
-    _In_ DWORD dwDesiredAccess,
+    _Win32_AssociatedEnum_(PROCESS_ACCESS_RIGHTS) _In_ DWORD dwDesiredAccess,
     _In_ BOOL bInheritHandle,
     _In_ DWORD dwProcessId
     );
 
+_Windows_SupportedOS_Windows2000_ 
+
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 BOOL
 WINAPI
 IsProcessorFeaturePresent(
-    _In_ DWORD ProcessorFeature
+    _Win32_AssociatedEnum_(PROCESSOR_FEATURE_ID) _In_ DWORD ProcessorFeature
     );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM  | WINAPI_PARTITION_GAMES) */
@@ -906,6 +1094,7 @@ IsProcessorFeaturePresent(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -924,6 +1113,8 @@ GetProcessHandleCount(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 DWORD
 WINAPI
@@ -941,6 +1132,8 @@ GetCurrentProcessorNumber(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_Windows7_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -950,6 +1143,8 @@ SetThreadIdealProcessorEx(
     _Out_opt_ PPROCESSOR_NUMBER lpPreviousIdealProcessor
     );
 
+_Windows_SupportedOS_Windows7_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -958,6 +1153,7 @@ GetThreadIdealProcessorEx(
     _Out_ PPROCESSOR_NUMBER lpIdealProcessor
     );
 
+_Windows_SupportedOS_Windows7_
 WINBASEAPI
 VOID
 WINAPI
@@ -975,6 +1171,9 @@ GetCurrentProcessorNumberEx(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINXP)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -983,6 +1182,9 @@ GetProcessPriorityBoost(
     _Out_ PBOOL pDisablePriorityBoost
     );
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1001,6 +1203,7 @@ SetProcessPriorityBoost(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINXP)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1018,6 +1221,7 @@ GetThreadIOPendingFlag(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINXP)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1049,9 +1253,12 @@ typedef enum _THREAD_INFORMATION_CLASS {
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
 typedef struct _MEMORY_PRIORITY_INFORMATION {
-    ULONG MemoryPriority;
+    _Win32_AssociatedEnum_(MEMORY_PRIORITY) ULONG MemoryPriority;
 } MEMORY_PRIORITY_INFORMATION, *PMEMORY_PRIORITY_INFORMATION;
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1062,6 +1269,9 @@ GetThreadInformation(
     _In_ DWORD ThreadInformationSize
     );
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1098,6 +1308,8 @@ typedef struct _THREAD_POWER_THROTTLING_STATE {
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WINBLUE)
 
+_Windows_SupportedOS_Windows81_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1110,6 +1322,8 @@ IsProcessCritical(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN10)
 
+_Windows_SupportedOS_Windows81_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1119,6 +1333,7 @@ SetProtectedPolicy(
     _Out_opt_ PULONG_PTR OldPolicyValue
     );
 
+_Windows_SupportedOS_Windows81_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1135,6 +1350,9 @@ QueryProtectedPolicy(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -1183,7 +1401,7 @@ DEFINE_ENUM_FLAG_OPERATORS(MACHINE_ATTRIBUTES);
 #endif
 
 typedef struct _PROCESS_MACHINE_INFORMATION {
-    USHORT ProcessMachine;
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) USHORT ProcessMachine;
     USHORT Res0;
     MACHINE_ATTRIBUTES MachineAttributes;
 } PROCESS_MACHINE_INFORMATION;
@@ -1229,7 +1447,7 @@ typedef struct _PROCESS_POWER_THROTTLING_STATE {
 } PROCESS_POWER_THROTTLING_STATE, *PPROCESS_POWER_THROTTLING_STATE;
 
 typedef struct PROCESS_PROTECTION_LEVEL_INFORMATION {
-    DWORD ProtectionLevel;
+    _Win32_AssociatedEnum_(PROCESS_PROTECTION_LEVEL) DWORD ProtectionLevel;
 } PROCESS_PROTECTION_LEVEL_INFORMATION;
 
 #define PROCESS_LEAP_SECOND_INFO_FLAG_ENABLE_SIXTY_SECOND   0x1
@@ -1243,6 +1461,9 @@ typedef struct _PROCESS_LEAP_SECOND_INFO {
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1253,6 +1474,9 @@ SetProcessInformation(
     _In_ DWORD ProcessInformationSize
     );
 
+_Windows_SupportedOS_Windows8_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1267,6 +1491,9 @@ GetProcessInformation(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN10)
 
+
+
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1275,9 +1502,12 @@ GetSystemCpuSetInformation(
     _In_ ULONG BufferLength,
     _Always_(_Out_) PULONG ReturnedLength,
     _In_opt_ HANDLE Process,
-    _Reserved_ ULONG Flags
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_ ULONG Flags
     );
 
+
+
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1288,6 +1518,8 @@ GetProcessDefaultCpuSets(
     _Always_(_Out_) PULONG RequiredIdCount
     );
 
+
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1297,6 +1529,9 @@ SetProcessDefaultCpuSets(
     _In_ ULONG CpuSetIdCount
     );
 
+
+
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1307,6 +1542,8 @@ GetThreadSelectedCpuSets(
     _Always_(_Out_) PULONG RequiredIdCount
     );
 
+
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1324,6 +1561,9 @@ SetThreadSelectedCpuSets(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -1334,7 +1574,7 @@ CreateProcessAsUserA(
     _In_opt_ LPSECURITY_ATTRIBUTES lpProcessAttributes,
     _In_opt_ LPSECURITY_ATTRIBUTES lpThreadAttributes,
     _In_ BOOL bInheritHandles,
-    _In_ DWORD dwCreationFlags,
+    _Win32_AssociatedEnum_(PROCESS_CREATION_FLAGS) _In_ DWORD dwCreationFlags,
     _In_opt_ LPVOID lpEnvironment,
     _In_opt_ LPCSTR lpCurrentDirectory,
     _In_ LPSTARTUPINFOA lpStartupInfo,
@@ -1345,6 +1585,9 @@ CreateProcessAsUserA(
 #define CreateProcessAsUser  CreateProcessAsUserA
 #endif
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1361,6 +1604,7 @@ GetProcessShutdownParameters(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN10_FE)
 
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1380,6 +1624,7 @@ SetProcessDefaultCpuSetMasks(
     _In_ USHORT CpuSetMaskCount
     );
 
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1390,6 +1635,7 @@ GetThreadSelectedCpuSetMasks(
     _Out_ PUSHORT RequiredMaskCount
     );
 
+
 _Success_(return != FALSE)
 BOOL
 WINAPI
@@ -1419,6 +1665,8 @@ GetMachineTypeAttributes(
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM) */
 #pragma endregion
 
+_Windows_SupportedOS_14393_ 
+
 WINBASEAPI
 HRESULT
 WINAPI
@@ -1427,12 +1675,14 @@ SetThreadDescription(
     _In_ PCWSTR lpThreadDescription
     );
 
+_Windows_SupportedOS_14393_ 
+
 WINBASEAPI
 HRESULT
 WINAPI
 GetThreadDescription(
     _In_ HANDLE hThread,
-    _Outptr_result_z_ PWSTR* ppszThreadDescription
+    _Outptr_result_z_  PWSTR* ppszThreadDescription
     );
 
 #pragma region Application Family or OneCore Family or Games Family
@@ -1440,6 +1690,7 @@ GetThreadDescription(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN10)
 
+_Win32_ImportLibrary_("kernel32.dll")
 WINBASEAPI
 LPVOID
 WINAPI
