diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/PropIdlBase.h b/generation/WinSDK/RecompiledIdlHeaders/um/PropIdlBase.h
index 4d7505f2da..0e3cedb390 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/PropIdlBase.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/PropIdlBase.h
@@ -9,6 +9,10 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
 #endif
@@ -496,6 +500,7 @@ EXTERN_C const IID IID_IPropertyStorage;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_Windows2000_
     MIDL_INTERFACE("00000138-0000-0000-C000-000000000046")
     IPropertyStorage : public IUnknown
     {
@@ -776,6 +781,7 @@ EXTERN_C const IID IID_IPropertySetStorage;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_Windows2000_
     MIDL_INTERFACE("0000013A-0000-0000-C000-000000000046")
     IPropertySetStorage : public IUnknown
     {
@@ -928,6 +934,7 @@ EXTERN_C const IID IID_IEnumSTATPROPSTG;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_Windows2000_
     MIDL_INTERFACE("00000139-0000-0000-C000-000000000046")
     IEnumSTATPROPSTG : public IUnknown
     {
@@ -1077,6 +1084,7 @@ EXTERN_C const IID IID_IEnumSTATPROPSETSTG;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
+    _Windows_SupportedOS_Windows2000_
     MIDL_INTERFACE("0000013B-0000-0000-C000-000000000046")
     IEnumSTATPROPSETSTG : public IUnknown
     {
@@ -1246,14 +1254,22 @@ unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal(  __RPC__in unsigned long *,
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
 void                      __RPC_USER  LPSAFEARRAY_UserFree(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
+_Windows_SupportedOS_WindowsXP_
 unsigned long             __RPC_USER  BSTR_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  BSTR_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  BSTR_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out BSTR * ); 
+_Windows_SupportedOS_WindowsXP_
 void                      __RPC_USER  BSTR_UserFree64(     __RPC__in unsigned long *, __RPC__in BSTR * ); 
 
+_Windows_SupportedOS_WindowsXP_
 unsigned long             __RPC_USER  LPSAFEARRAY_UserSize64(     __RPC__in unsigned long *, unsigned long            , __RPC__in LPSAFEARRAY * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  LPSAFEARRAY_UserMarshal64(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in LPSAFEARRAY * ); 
+_Windows_SupportedOS_WindowsXP_
 unsigned char * __RPC_USER  LPSAFEARRAY_UserUnmarshal64(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out LPSAFEARRAY * ); 
+_Windows_SupportedOS_WindowsXP_
 void                      __RPC_USER  LPSAFEARRAY_UserFree64(     __RPC__in unsigned long *, __RPC__in LPSAFEARRAY * ); 
 
 /* [local] */ HRESULT STDMETHODCALLTYPE IEnumSTATPROPSTG_Next_Proxy( 
