diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/propidl.h b/generation/WinSDK/RecompiledIdlHeaders/um/propidl.h
index 4ff6ae0d6e..2e11c45142 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/propidl.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/propidl.h
@@ -105,6 +105,7 @@ extern "C"{
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 #ifndef _PROPIDLBASE_
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
@@ -1342,17 +1343,17 @@ inline void PropVariantInit (_Out_ PROPVARIANT * pvar )
 
 #ifndef _SERIALIZEDPROPERTYVALUE_DEFINED_
 #define _SERIALIZEDPROPERTYVALUE_DEFINED_
-typedef struct tagSERIALIZEDPROPERTYVALUE
+typedef struct   tagSERIALIZEDPROPERTYVALUE
 {
     DWORD	dwType;
     BYTE	rgb[1];
 } SERIALIZEDPROPERTYVALUE;
 #endif
 
-EXTERN_C
+_Windows_SupportedOS_Windows2000_ EXTERN_C
 _Success_(TRUE)  /* Raises status on failure */
 SERIALIZEDPROPERTYVALUE* __stdcall
-StgConvertVariantToProperty(
+  StgConvertVariantToProperty(
             _In_ const PROPVARIANT* pvar,
             _In_ USHORT CodePage,
             _Out_writes_bytes_opt_(*pcb) SERIALIZEDPROPERTYVALUE* pprop,
@@ -1364,10 +1365,10 @@ StgConvertVariantToProperty(
 #ifdef __cplusplus
 class PMemoryAllocator;
 
-EXTERN_C
+_Windows_SupportedOS_Windows2000_ EXTERN_C
 _Success_(TRUE)  /* Raises status on failure */
 BOOLEAN __stdcall
-StgConvertPropertyToVariant(
+  StgConvertPropertyToVariant(
             _In_ const SERIALIZEDPROPERTYVALUE* pprop,
             _In_ USHORT CodePage,
             _Out_ PROPVARIANT* pvar,
