diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RasDlg.h b/generation/WinSDK/RecompiledIdlHeaders/um/RasDlg.h
index bd65a37388..21a1f05cfd 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RasDlg.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RasDlg.h
@@ -24,6 +24,7 @@ Abstract:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -249,23 +250,29 @@ typedef BOOL (WINAPI *RasCustomEntryDlgFn) (
 
 /* RAS common dialog API prototypes.
 */
-BOOL APIENTRY RasPhonebookDlgA(
+_Windows_SupportedOS_Windows2000_ BOOL APIENTRY RasPhonebookDlgA(
     _In_opt_ LPSTR lpszPhonebook, _In_opt_ LPSTR lpszEntry, LPRASPBDLGA lpInfo );
 
+_Windows_SupportedOS_Windows2000_ 
+
 BOOL APIENTRY RasPhonebookDlgW(
     _In_opt_ LPWSTR lpszPhonebook, _In_opt_ LPWSTR lpszEntry, LPRASPBDLGW lpInfo );
 
+_Windows_SupportedOS_Windows2000_ 
+
 BOOL APIENTRY RasEntryDlgA(
     _In_opt_ LPSTR lpszPhonebook, _In_opt_ LPSTR lpszEntry, LPRASENTRYDLGA lpInfo );
 
-BOOL APIENTRY RasEntryDlgW(
+_Windows_SupportedOS_Windows2000_ BOOL APIENTRY RasEntryDlgW(
     _In_opt_ LPWSTR lpszPhonebook, _In_opt_ LPWSTR lpszEntry, LPRASENTRYDLGW lpInfo );
 
+_Windows_SupportedOS_Windows2000_ 
+
 BOOL APIENTRY RasDialDlgA(
     _In_opt_ LPSTR lpszPhonebook, _In_opt_ LPSTR lpszEntry, _In_opt_ LPSTR lpszPhoneNumber,
     _Inout_ LPRASDIALDLG lpInfo );
 
-BOOL APIENTRY RasDialDlgW(
+_Windows_SupportedOS_Windows2000_ BOOL APIENTRY RasDialDlgW(
     _In_opt_ LPWSTR lpszPhonebook, _In_opt_ LPWSTR lpszEntry, _In_opt_ LPWSTR lpszPhoneNumber,
     _Inout_ LPRASDIALDLG lpInfo );
 
