diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/recapis.h b/generation/WinSDK/RecompiledIdlHeaders/um/recapis.h
index 3969ac6c39..61041107f3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/recapis.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/recapis.h
@@ -10,6 +10,7 @@
 //--------------------------------------------------------------------------
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -70,49 +71,89 @@ typedef IID CLSID;
 ////////////////////////
 // IRecognizer
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI CreateRecognizer(CLSID *pCLSID, HRECOGNIZER *phrec);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyRecognizer(HRECOGNIZER hrec);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetRecoAttributes(HRECOGNIZER hrec, RECO_ATTRS* pRecoAttrs);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI CreateContext(HRECOGNIZER hrec, HRECOCONTEXT *phrc);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyContext(HRECOCONTEXT hrc);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetResultPropertyList(HRECOGNIZER hrec, ULONG* pPropertyCount, GUID*pPropertyGuid);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetPreferredPacketDescription(HRECOGNIZER hrec, PACKET_DESCRIPTION* pPacketDescription);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetUnicodeRanges(HRECOGNIZER hrec, ULONG *pcRanges, CHARACTER_RANGE *pcr);
 
 ////////////////////////
 // IRecoContext
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AddStroke(HRECOCONTEXT hrc, const PACKET_DESCRIPTION* pPacketDesc, ULONG cbPacket, const BYTE *pPacket, const XFORM *pXForm);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetBestResultString(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_opt_ (*pcSize) WCHAR* pwcBestResult);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyAlternate(HRECOALT hrcalt);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetGuide(HRECOCONTEXT hrc, const RECO_GUIDE* pGuide, ULONG iIndex);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI GetGuide(HRECOCONTEXT hrc, RECO_GUIDE* pGuide, ULONG *piIndex);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AdviseInkChange(HRECOCONTEXT hrc, BOOL bNewStroke);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetCACMode(HRECOCONTEXT hrc, int iMode);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI EndInkInput(HRECOCONTEXT hrc);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI CloneContext(HRECOCONTEXT hrc, HRECOCONTEXT* pCloneHrc);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI ResetContext(HRECOCONTEXT hrc);
+_Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI Process(HRECOCONTEXT hrc, BOOL *pbPartialProcessing);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetFactoid(HRECOCONTEXT hrc, ULONG cwcFactoid, const WCHAR *pwcFactoid);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetFlags(HRECOCONTEXT hrc, DWORD dwFlags);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetLatticePtr(HRECOCONTEXT hrc, RECO_LATTICE **ppLattice);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetTextContext(HRECOCONTEXT hrc, ULONG cwcBefore, _In_reads_(cwcBefore) const WCHAR *pwcBefore, ULONG cwcAfter, _In_reads_(cwcAfter) const WCHAR *pwcAfter);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetEnabledUnicodeRanges(HRECOCONTEXT hrc, ULONG *pcRanges, CHARACTER_RANGE *pcr);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetEnabledUnicodeRanges(HRECOCONTEXT hrc, ULONG cRanges, CHARACTER_RANGE *pcr);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetContextPropertyList(HRECOCONTEXT hrc, ULONG *pcProperties, GUID *pPropertyGUIDS);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetContextPropertyValue(HRECOCONTEXT hrc, GUID *pGuid, ULONG *pcbSize, BYTE *pProperty);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI SetContextPropertyValue(HRECOCONTEXT hrc, GUID *pGuid, ULONG cbSize, BYTE *pProperty);
+_Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI IsStringSupported(HRECOCONTEXT hrc, ULONG wcString, const WCHAR *pwcString);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetWordList(HRECOCONTEXT hrc, HRECOWORDLIST hwl);
+_Win32_ImportLibrary_("inkobjcore.dll") _Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetContextPreferenceFlags(HRECOCONTEXT hrc, DWORD *pdwContextPreferenceFlags);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetRightSeparator(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_(*pcSize) OPTIONAL WCHAR* pwcRightSeparator);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetLeftSeparator(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_(*pcSize) OPTIONAL WCHAR* pwcLeftSeparator);
 
 ////////////////////////
 // IRecoWordList
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyWordList(HRECOWORDLIST hwl);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AddWordsToWordList(HRECOWORDLIST hwl, _NullNull_terminated_ WCHAR * pwcWords);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI MakeWordList(HRECOGNIZER hrec, _NullNull_terminated_ WCHAR *pBuffer, HRECOWORDLIST *phwl);
 
 #ifdef __cplusplus
@@ -180,44 +221,73 @@ extern "C" {
 ////////////////////////
 // IRecognizer
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI CreateRecognizer(CLSID *pCLSID, HRECOGNIZER *phrec);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyRecognizer(HRECOGNIZER hrec);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetRecoAttributes(HRECOGNIZER hrec, RECO_ATTRS* pRecoAttrs);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI CreateContext(HRECOGNIZER hrec, HRECOCONTEXT *phrc);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyContext(HRECOCONTEXT hrc);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetResultPropertyList(HRECOGNIZER hrec, ULONG* pPropertyCount, GUID*pPropertyGuid);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetUnicodeRanges(HRECOGNIZER hrec, ULONG *pcRanges, CHARACTER_RANGE *pcr);
 
 ////////////////////////
 // IRecoContext
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AddStroke(HRECOCONTEXT hrc, const PACKET_DESCRIPTION* pPacketDesc, ULONG cbPacket, const BYTE *pPacket, const XFORM *pXForm);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetBestResultString(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_opt_(*pcSize) WCHAR* pwcBestResult);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetGuide(HRECOCONTEXT hrc, const RECO_GUIDE* pGuide, ULONG iIndex);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AdviseInkChange(HRECOCONTEXT hrc, BOOL bNewStroke);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI EndInkInput(HRECOCONTEXT hrc);
+_Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI Process(HRECOCONTEXT hrc, BOOL *pbPartialProcessing);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetFactoid(HRECOCONTEXT hrc, ULONG cwcFactoid, const WCHAR *pwcFactoid);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetFlags(HRECOCONTEXT hrc, DWORD dwFlags);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetLatticePtr(HRECOCONTEXT hrc, RECO_LATTICE **ppLattice);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetTextContext(HRECOCONTEXT hrc, ULONG cwcBefore, _In_reads_(cwcBefore) const WCHAR *pwcBefore, ULONG cwcAfter, _In_reads_(cwcAfter) const WCHAR *pwcAfter);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetEnabledUnicodeRanges(HRECOCONTEXT hrc, ULONG cRanges, CHARACTER_RANGE *pcr);
+_Windows_SupportedOS_WindowsXP_
+_Win32_PreserveResult_
 HRESULT WINAPI IsStringSupported(HRECOCONTEXT hrc, ULONG wcString, const WCHAR *pwcString);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI SetWordList(HRECOCONTEXT hrc, HRECOWORDLIST hwl);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetRightSeparator(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_(*pcSize) OPTIONAL WCHAR* pwcRightSeparator);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetLeftSeparator(HRECOCONTEXT hrc, _Inout_ ULONG *pcSize, _Out_writes_(*pcSize) OPTIONAL WCHAR* pwcLeftSeparator);
 
 ////////////////////////
 // IRecoWordList
 ////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI DestroyWordList(HRECOWORDLIST hwl);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI AddWordsToWordList(HRECOWORDLIST hwl, _NullNull_terminated_ WCHAR * pwcWords);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI MakeWordList(HRECOGNIZER hrec, _NullNull_terminated_ WCHAR *pBuffer, HRECOWORDLIST *phwl);
 
 ////////////////////////////
 // Recognizer Broker APIs
 ////////////////////////////
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI GetAllRecognizers(CLSID** recognizerClsids, ULONG* count);
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI LoadCachedAttributes(CLSID clsid, RECO_ATTRS *pRecoAttributes);
 #ifdef __cplusplus
 }
