diff --git a/generation/WinSDK/RecompiledIdlHeaders/winrt/rometadata.h b/generation/WinSDK/RecompiledIdlHeaders/winrt/rometadata.h
index 82fdc8af6f7df389ae44157dde27eb99e2580de4..03da3020c47967d28442322361f1c18286f0d23d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/winrt/rometadata.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/winrt/rometadata.h
@@ -5,6 +5,7 @@
 
 #ifndef __RO_METADATA_H__
 #define __RO_METADATA_H__
+#include "win32metadata_annotations.h"
 
 #if NTDDI_VERSION >= NTDDI_WIN8
 
@@ -12,6 +13,8 @@
 // Arguments:
 //    rclsid - Only CLSID_CorMetaDataDispenser supported.
 //    riid   - Interfaces supported: IID_IMetaDataDispenser, IID_IMetaDataDispenserEx.
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 STDAPI MetaDataGetDispenser(
         __in REFCLSID            rclsid, 
         __in REFIID              riid, 
