diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/RTWorkQ.h b/generation/WinSDK/RecompiledIdlHeaders/um/RTWorkQ.h
index b3de755bda..c07df5efa7 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/RTWorkQ.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/RTWorkQ.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //*@@@+++@@@@******************************************************************
 //
 // RTWorkQ
@@ -36,27 +37,35 @@ typedef unsigned __int64 RTWQWORKITEM_KEY;
 #define RTWQ_E_SHUTDOWN            RTWQ_E_ERROR(16005)
 
 
-typedef enum
+typedef enum 
 {
     RTWQ_STANDARD_WORKQUEUE = 0,        // single threaded MTA
     RTWQ_WINDOW_WORKQUEUE = 1,          // Message loop that calls PeekMessage() / DispatchMessage()..
     RTWQ_MULTITHREADED_WORKQUEUE = 2,   // multithreaded MTA
 }   RTWQ_WORKQUEUE_TYPE;
 
-interface DECLSPEC_UUID("ac6b7889-0740-4d51-8619-905994a55cc6") DECLSPEC_NOVTABLE
+interface _Windows_SupportedOS_Windows81_  DECLSPEC_UUID("ac6b7889-0740-4d51-8619-905994a55cc6") DECLSPEC_NOVTABLE
     IRtwqAsyncResult : public IUnknown
 {
+    
     STDMETHOD(GetState)( _Out_ IUnknown** ppunkState)=0;
+    
+    
     STDMETHOD(GetStatus)()=0;
+    
     STDMETHOD(SetStatus)( HRESULT hrStatus)=0;
+    
     STDMETHOD(GetObject)( _Out_ IUnknown ** ppObject)=0;
+    
     STDMETHOD_(IUnknown *, GetStateNoAddRef)()=0;
 };
 
-interface DECLSPEC_UUID("a27003cf-2354-4f2a-8d6a-ab7cff15437e") DECLSPEC_NOVTABLE
+interface _Windows_SupportedOS_Windows81_  DECLSPEC_UUID("a27003cf-2354-4f2a-8d6a-ab7cff15437e") DECLSPEC_NOVTABLE
     IRtwqAsyncCallback : public IUnknown
 {
+    
     STDMETHOD(GetParameters)( _Out_ DWORD* pdwFlags, _Out_ DWORD* pdwQueue )=0;
+    _Windows_SupportedOS_Windows81_
     STDMETHOD(Invoke)( _In_ IRtwqAsyncResult* pAsyncResult )=0;
 };
 
@@ -69,63 +78,98 @@ typedef struct tagRTWQASYNCRESULT : public IRtwqAsyncResult
     HANDLE hEvent;
 }   RTWQASYNCRESULT;
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqStartup();
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqShutdown();
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqLockWorkQueue(_In_ DWORD workQueueId);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqUnlockWorkQueue(_In_ DWORD workQueueId);
-STDAPI RtwqLockSharedWorkQueue(_In_ PCWSTR usageClass, _In_ LONG basePriority, _Inout_ DWORD* taskId, _Out_ DWORD* id);
+_Windows_SupportedOS_Windows81_
+STDAPI RtwqLockSharedWorkQueue(_In_ PCWSTR usageClass, _In_ LONG basePriority, _Inout_ DWORD* taskId, _Out_  DWORD* id);
 
-STDAPI RtwqJoinWorkQueue(DWORD workQueueId, HANDLE hFile, _Out_ HANDLE* out);
+_Windows_SupportedOS_Windows81_
+STDAPI RtwqJoinWorkQueue(DWORD workQueueId, HANDLE hFile, _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ HANDLE* out);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqUnjoinWorkQueue(DWORD workQueueId, HANDLE hFile);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqCreateAsyncResult(_In_ IUnknown* appObject, _In_ IRtwqAsyncCallback* callback, _In_ IUnknown* appState, _Out_ IRtwqAsyncResult** asyncResult);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqInvokeCallback(IRtwqAsyncResult* result);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqLockPlatform();
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqUnlockPlatform();
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqRegisterPlatformWithMMCSS(_In_ PCWSTR usageClass, _Inout_ DWORD* taskId, _In_ LONG lPriority);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqUnregisterPlatformFromMMCSS();
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqPutWorkItem(DWORD dwQueue, LONG lPriority, _In_ IRtwqAsyncResult* result);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqPutWaitingWorkItem(HANDLE hEvent, LONG lPriority, _In_ IRtwqAsyncResult* result, _Out_opt_ RTWQWORKITEM_KEY* key);
-STDAPI RtwqAllocateSerialWorkQueue(_In_ DWORD workQueueIdIn, _Out_ DWORD* workQueueIdOut);
+_Windows_SupportedOS_Windows81_
+STDAPI RtwqAllocateSerialWorkQueue(_In_ DWORD workQueueIdIn, _Out_  DWORD* workQueueIdOut);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqScheduleWorkItem(IRtwqAsyncResult* result, INT64 Timeout, _Out_opt_ RTWQWORKITEM_KEY* key);
 
-STDAPI RtwqAddPeriodicCallback(RTWQPERIODICCALLBACK Callback, IUnknown* context, _Out_opt_ DWORD* key);
+_Windows_SupportedOS_Windows81_
+STDAPI RtwqAddPeriodicCallback( RTWQPERIODICCALLBACK Callback, IUnknown* context, _Out_opt_  DWORD* key);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqRemovePeriodicCallback(DWORD dwKey);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqCancelWorkItem(RTWQWORKITEM_KEY Key);
 
-STDAPI RtwqAllocateWorkQueue(_In_ RTWQ_WORKQUEUE_TYPE WorkQueueType, _Out_ DWORD* workQueueId);
+_Windows_SupportedOS_Windows81_
+STDAPI RtwqAllocateWorkQueue(_In_ RTWQ_WORKQUEUE_TYPE WorkQueueType, _Out_  DWORD* workQueueId);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqBeginRegisterWorkQueueWithMMCSS(DWORD workQueueId, _In_ PCWSTR usageClass, DWORD dwTaskId, LONG lPriority, _In_ IRtwqAsyncCallback* doneCallback, _In_ IUnknown* doneState);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqBeginUnregisterWorkQueueWithMMCSS(DWORD workQueueId, _In_ IRtwqAsyncCallback* doneCallback, _In_ IUnknown* doneState);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqEndRegisterWorkQueueWithMMCSS(_In_ IRtwqAsyncResult* result, _Out_ DWORD* taskId);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqGetWorkQueueMMCSSClass(DWORD workQueueId, _Out_writes_to_opt_(*usageClassLength,*usageClassLength)  PWSTR usageClass, _Inout_  DWORD* usageClassLength);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqGetWorkQueueMMCSSTaskId(DWORD workQueueId, _Out_ DWORD* taskId);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqGetWorkQueueMMCSSPriority(DWORD workQueueId, _Out_ LONG* priority);
 
-interface DECLSPEC_UUID("63d9255a-7ff1-4b61-8faf-ed6460dacf2b") DECLSPEC_NOVTABLE
+interface _Windows_SupportedOS_Windows81_  DECLSPEC_UUID("63d9255a-7ff1-4b61-8faf-ed6460dacf2b") DECLSPEC_NOVTABLE
     IRtwqPlatformEvents : public IUnknown
 {
+    
     STDMETHOD(InitializationComplete)()=0;
+    
     STDMETHOD(ShutdownStart)()=0;
+    
     STDMETHOD(ShutdownComplete)()=0;
 };
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqRegisterPlatformEvents(_In_ IRtwqPlatformEvents* platformEvents);
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqUnregisterPlatformEvents(_In_ IRtwqPlatformEvents* platformEvents);
 
+_Windows_SupportedOS_Windows81_
 STDAPI RtwqSetLongRunning(DWORD workQueueId, BOOL enable);
 
-STDAPI RtwqSetDeadline(DWORD workQueueId, LONGLONG deadlineInHNS, _Out_ HANDLE* pRequest);
+_Windows_SupportedOS_10240_  
+STDAPI RtwqSetDeadline(DWORD workQueueId, LONGLONG deadlineInHNS, _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_  HANDLE* pRequest);
 // TODO: switch to RS define once it exists (see: 5312604)
 #if (WINVER >= _WIN32_WINNT_WIN10)
-STDAPI RtwqSetDeadline2(DWORD workQueueId, LONGLONG deadlineInHNS, LONGLONG preDeadlineInHNS, _Out_ HANDLE* pRequest);
+_Windows_SupportedOS_10240_  
+STDAPI RtwqSetDeadline2(DWORD workQueueId, LONGLONG deadlineInHNS, LONGLONG preDeadlineInHNS, _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_  HANDLE* pRequest);
 #endif // (WINVER > _WIN32_WINNT_WIN10)
 STDAPI RtwqCancelDeadline(_In_ HANDLE pRequest);
 
