diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/sapiddk.h b/generation/WinSDK/RecompiledIdlHeaders/um/sapiddk.h
index 9a23390bc3..be8d16846f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/sapiddk.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/sapiddk.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -677,28 +678,28 @@ EXTERN_C const IID IID_ISpObjectTokenEnumBuilder;
 /* [local] */ 
 
 #if 0
-typedef void *SPWORDHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPWORDHANDLE;
 
-typedef void *SPRULEHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPRULEHANDLE;
 
-typedef void *SPGRAMMARHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPGRAMMARHANDLE;
 
-typedef void *SPRECOCONTEXTHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPRECOCONTEXTHANDLE;
 
-typedef void *SPPHRASERULEHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPPHRASERULEHANDLE;
 
-typedef void *SPPHRASEPROPERTYHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPPHRASEPROPERTYHANDLE;
 
-typedef void *SPTRANSITIONID;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPTRANSITIONID;
 
 #else
-DECLARE_HANDLE(SPWORDHANDLE);
-DECLARE_HANDLE(SPRULEHANDLE);
-DECLARE_HANDLE(SPGRAMMARHANDLE);
-DECLARE_HANDLE(SPRECOCONTEXTHANDLE);
-DECLARE_HANDLE(SPPHRASERULEHANDLE);
-DECLARE_HANDLE(SPPHRASEPROPERTYHANDLE);
-DECLARE_HANDLE(SPTRANSITIONID);
+DECLARE_HANDLE(SPWORDHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPRULEHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPGRAMMARHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPRECOCONTEXTHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPPHRASERULEHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPPHRASEPROPERTYHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
+DECLARE_HANDLE(SPTRANSITIONID) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
 #endif
 
 
@@ -1454,7 +1455,7 @@ EXTERN_C const IID IID_ISpPhraseBuilder;
 interface ISpTask
 {
 virtual HRESULT STDMETHODCALLTYPE Execute(
-                 void *pvTaskData,
+                 _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void *pvTaskData,
                  volatile const BOOL* pfContinueProcessing) = 0;
 };
 #else
@@ -1465,16 +1466,16 @@ typedef void *ISpTask;
 interface ISpThreadTask
 {
 virtual HRESULT STDMETHODCALLTYPE InitThread(
-                 void* pvTaskData,
+                 _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void* pvTaskData,
                  HWND hwnd) = 0;
 virtual HRESULT STDMETHODCALLTYPE ThreadProc(
-                 void *pvTaskData,
+                 _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void *pvTaskData,
                  HANDLE hExitThreadEvent,
                  HANDLE hNotifyEvent,
                  HWND hwndWorker,
                  volatile const BOOL* pfContinueProcessing) = 0;
 virtual LRESULT STDMETHODCALLTYPE WindowMessage(
-                 void *pvTaskData,
+                 _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void *pvTaskData,
                  HWND hWnd,
                  UINT Msg,
                  WPARAM wParam,
@@ -1520,17 +1521,17 @@ EXTERN_C const IID IID_ISpThreadControl;
         
         virtual HRESULT STDMETHODCALLTYPE TerminateThread( void) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE ThreadHandle( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE ThreadHandle( void) = 0;
         
         virtual DWORD STDMETHODCALLTYPE ThreadId( void) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE NotifyEvent( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE NotifyEvent( void) = 0;
         
         virtual HWND STDMETHODCALLTYPE WindowHandle( void) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE ThreadCompleteEvent( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE ThreadCompleteEvent( void) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE ExitThreadEvent( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE ExitThreadEvent( void) = 0;
         
     };
     
@@ -2612,7 +2613,7 @@ typedef struct tagSPPHRASEALT
     ULONG cbAltExtra;
     } 	SPPHRASEALT;
 
-typedef struct SPRECORESULTINFO
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SPRECORESULTINFO
     {
     ULONG cbSize;
     SPRESULTTYPE eResultType;
@@ -2667,7 +2668,7 @@ enum SPRULEINFOOPT
         SPRIO_NONE	= 0
     } 	SPRULEINFOOPT;
 
-typedef struct SPPARSEINFO
+typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SPPARSEINFO
     {
     ULONG cbSize;
     SPRULEHANDLE hRule;
