diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/sapi53.h b/generation/WinSDK/RecompiledIdlHeaders/um/sapi53.h
index 10cd1740df..042bce452c 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/sapi53.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/sapi53.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -1357,7 +1358,7 @@ EXTERN_C const IID IID_ISpNotifySource;
             /* [in] */ LPARAM lParam) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE SetNotifyCallbackFunction( 
-            /* [in] */ SPNOTIFYCALLBACK *pfnCallback,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") SPNOTIFYCALLBACK *pfnCallback,
             /* [in] */ WPARAM wParam,
             /* [in] */ LPARAM lParam) = 0;
         
@@ -1371,7 +1372,7 @@ EXTERN_C const IID IID_ISpNotifySource;
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE WaitForNotifyEvent( 
             /* [in] */ DWORD dwMilliseconds) = 0;
         
-        virtual /* [local] */ HANDLE STDMETHODCALLTYPE GetNotifyEventHandle( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual /* [local] */ HANDLE STDMETHODCALLTYPE GetNotifyEventHandle( void) = 0;
         
     };
     
@@ -1596,7 +1597,7 @@ EXTERN_C const IID IID_ISpNotifyTranslator;
             /* [in] */ LPARAM lParam) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE InitCallback( 
-            /* [in] */ SPNOTIFYCALLBACK *pfnCallback,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") SPNOTIFYCALLBACK *pfnCallback,
             /* [in] */ WPARAM wParam,
             /* [in] */ LPARAM lParam) = 0;
         
@@ -1612,7 +1613,7 @@ EXTERN_C const IID IID_ISpNotifyTranslator;
         virtual HRESULT STDMETHODCALLTYPE Wait( 
             /* [in] */ DWORD dwMilliseconds) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE GetEventHandle( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE GetEventHandle( void) = 0;
         
     };
     
@@ -1757,8 +1758,8 @@ EXTERN_C const IID IID_ISpDataKey;
         
         virtual HRESULT STDMETHODCALLTYPE GetData( 
             /* [in] */ LPCWSTR pszValueName,
-            /* [in] */ ULONG *pcbData,
-            /* [out] */ BYTE *pData) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") ULONG *pcbData,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BYTE *pData) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetStringValue( 
             /* [in][annotation] */ 
@@ -1777,7 +1778,7 @@ EXTERN_C const IID IID_ISpDataKey;
         
         virtual HRESULT STDMETHODCALLTYPE GetDWORD( 
             /* [in] */ LPCWSTR pszValueName,
-            /* [out] */ DWORD *pdwValue) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") DWORD *pdwValue) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE OpenKey( 
             /* [in] */ LPCWSTR pszSubKeyName,
@@ -2486,7 +2487,7 @@ EXTERN_C const IID IID_ISpObjectToken;
             /* [in] */ IUnknown *pUnkOuter,
             /* [in] */ DWORD dwClsContext,
             /* [in] */ REFIID riid,
-            /* [iid_is][out] */ void **ppvObject) = 0;
+            /* [iid_is][out] */ _WIN32META_ANNOTATION_("win32metadata:in") void **ppvObject) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetStorageFileName( 
             /* [in] */ REFCLSID clsidCaller,
@@ -2510,22 +2511,22 @@ EXTERN_C const IID IID_ISpObjectToken;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE IsUISupported( 
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData,
             /* [in] */ IUnknown *punkObject,
-            /* [out] */ BOOL *pfSupported) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *pfSupported) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE DisplayUI( 
             /* [in] */ HWND hwndParent,
             /* [in] */ LPCWSTR pszTitle,
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData,
             /* [in] */ IUnknown *punkObject) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE MatchesAttributes( 
             /* [in] */ LPCWSTR pszAttributes,
-            /* [out] */ BOOL *pfMatches) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *pfMatches) = 0;
         
     };
     
@@ -3158,7 +3159,7 @@ EXTERN_C const IID IID_IEnumSpObjectTokens;
             /* [out] */ ISpObjectToken **ppToken) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetCount( 
-            /* [out] */ ULONG *pCount) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pCount) = 0;
         
     };
     
@@ -3720,11 +3721,11 @@ EXTERN_C const IID IID_ISpEventSource;
         
         virtual HRESULT STDMETHODCALLTYPE GetEvents( 
             /* [in] */ ULONG ulCount,
-            /* [size_is][out] */ SPEVENT *pEventArray,
-            /* [out] */ ULONG *pulFetched) = 0;
+            /* [size_is][out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPEVENT *pEventArray,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pulFetched) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetInfo( 
-            /* [out] */ SPEVENTSOURCEINFO *pInfo) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPEVENTSOURCEINFO *pInfo) = 0;
         
     };
     
@@ -3890,8 +3891,8 @@ EXTERN_C const IID IID_ISpEventSource2;
     public:
         virtual HRESULT STDMETHODCALLTYPE GetEventsEx( 
             /* [in] */ ULONG ulCount,
-            /* [size_is][out] */ SPEVENTEX *pEventArray,
-            /* [out] */ ULONG *pulFetched) = 0;
+            /* [size_is][out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPEVENTEX *pEventArray,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pulFetched) = 0;
         
     };
     
@@ -4071,7 +4072,7 @@ EXTERN_C const IID IID_ISpEventSink;
             /* [in] */ ULONG ulCount) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetEventInterest( 
-            /* [out] */ ULONGLONG *pullEventInterest) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONGLONG *pullEventInterest) = 0;
         
     };
     
@@ -4952,28 +4953,28 @@ EXTERN_C const IID IID_ISpAudio;
             /* [in] */ const WAVEFORMATEX *pWaveFormatEx) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetStatus( 
-            /* [out] */ SPAUDIOSTATUS *pStatus) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPAUDIOSTATUS *pStatus) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetBufferInfo( 
             /* [in] */ const SPAUDIOBUFFERINFO *pBuffInfo) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetBufferInfo( 
-            /* [out] */ SPAUDIOBUFFERINFO *pBuffInfo) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPAUDIOBUFFERINFO *pBuffInfo) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetDefaultFormat( 
-            /* [out] */ GUID *pFormatId,
-            /* [out] */ WAVEFORMATEX **ppCoMemWaveFormatEx) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") GUID *pFormatId,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") WAVEFORMATEX **ppCoMemWaveFormatEx) = 0;
         
-        virtual HANDLE STDMETHODCALLTYPE EventHandle( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual HANDLE STDMETHODCALLTYPE EventHandle( void) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetVolumeLevel( 
-            /* [out] */ ULONG *pLevel) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pLevel) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetVolumeLevel( 
             /* [in] */ ULONG Level) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetBufferNotifySize( 
-            /* [out] */ ULONG *pcbSize) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pcbSize) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetBufferNotifySize( 
             /* [in] */ ULONG cbSize) = 0;
@@ -5265,16 +5266,16 @@ EXTERN_C const IID IID_ISpMMSysAudio;
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetDeviceId( 
-            /* [out] */ UINT *puDeviceId) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") UINT *puDeviceId) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetDeviceId( 
             /* [in] */ UINT uDeviceId) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetMMHandle( 
-            /* [out] */ void **pHandle) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") void **pHandle) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetLineId( 
-            /* [out] */ UINT *puLineId) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") UINT *puLineId) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetLineId( 
             /* [in] */ UINT uLineId) = 0;
@@ -5795,7 +5796,7 @@ enum SPSEMANTICFORMAT
         SPSMF_SRGS_SEMANTICINTERPRETATION_W3C	= 8
     } 	SPSEMANTICFORMAT;
 
-typedef /* [hidden][restricted] */ struct SPPHRASE_50
+typedef /* [hidden][restricted] */ struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  SPPHRASE_50
     {
     ULONG cbSize;
     WORD LangID;
@@ -5888,10 +5889,10 @@ enum SPPHRASERNG
 #define SP_GETWHOLEPHRASE SPPR_ALL_ELEMENTS
 #define SPRR_ALL_ELEMENTS SPPR_ALL_ELEMENTS
 #if 0
-typedef void *SPSTATEHANDLE;
+_WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1") _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") typedef void *SPSTATEHANDLE;
 
 #else
-DECLARE_HANDLE(SPSTATEHANDLE);
+DECLARE_HANDLE(SPSTATEHANDLE) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
 #endif
 typedef /* [hidden] */ 
 enum SPRECOEVENTFLAGS
@@ -6045,7 +6046,7 @@ EXTERN_C const IID IID_ISpLexicon;
             _In_opt_  PCSPPHONEID pszPronunciation) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetGeneration( 
-            /* [out] */ DWORD *pdwGeneration) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") DWORD *pdwGeneration) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetGenerationChange( 
             /* [in] */ DWORD dwFlags,
@@ -7081,38 +7082,38 @@ EXTERN_C const IID IID_ISpVoice;
             _Out_opt_  ULONG *pulStreamNumber) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetStatus( 
-            /* [out] */ SPVOICESTATUS *pStatus,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPVOICESTATUS *pStatus,
             /* [out][annotation] */ 
             _Outptr_  LPWSTR *ppszLastBookmark) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Skip( 
             /* [string][in] */ LPCWSTR pItemType,
             /* [in] */ long lNumItems,
-            /* [out] */ ULONG *pulNumSkipped) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pulNumSkipped) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetPriority( 
             /* [in] */ SPVPRIORITY ePriority) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetPriority( 
-            /* [out] */ SPVPRIORITY *pePriority) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPVPRIORITY *pePriority) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetAlertBoundary( 
             /* [in] */ SPEVENTENUM eBoundary) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAlertBoundary( 
-            /* [out] */ SPEVENTENUM *peBoundary) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPEVENTENUM *peBoundary) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetRate( 
             /* [in] */ long RateAdjust) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetRate( 
-            /* [out] */ long *pRateAdjust) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") long *pRateAdjust) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetVolume( 
             /* [in] */ USHORT usVolume) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetVolume( 
-            /* [out] */ USHORT *pusVolume) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") USHORT *pusVolume) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE WaitUntilDone( 
             /* [in] */ ULONG msTimeout) = 0;
@@ -7121,21 +7122,21 @@ EXTERN_C const IID IID_ISpVoice;
             /* [in] */ ULONG msTimeout) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetSyncSpeakTimeout( 
-            /* [out] */ ULONG *pmsTimeout) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pmsTimeout) = 0;
         
-        virtual /* [local] */ HANDLE STDMETHODCALLTYPE SpeakCompleteEvent( void) = 0;
+        _Win32_RAIIFree_(CloseHandle, -1, 0) virtual /* [local] */ HANDLE STDMETHODCALLTYPE SpeakCompleteEvent( void) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE IsUISupported( 
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData,
-            /* [out] */ BOOL *pfSupported) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *pfSupported) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE DisplayUI( 
             /* [in] */ HWND hwndParent,
             /* [in] */ LPCWSTR pszTitle,
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData) = 0;
         
     };
@@ -7654,9 +7655,9 @@ EXTERN_C const IID IID_ISpPhraseAlt;
     public:
         virtual HRESULT STDMETHODCALLTYPE GetAltInfo( 
             /* [out] */ ISpPhrase **ppParent,
-            /* [out] */ ULONG *pulStartElementInParent,
-            /* [out] */ ULONG *pcElementsInParent,
-            /* [out] */ ULONG *pcElementsInAlt) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pulStartElementInParent,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pcElementsInParent,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pcElementsInAlt) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Commit( void) = 0;
         
@@ -7813,7 +7814,7 @@ EXTERN_C const IID IID_ISpPhrase2;
             /* [in] */ SPXMLRESULTOPTIONS Options) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetXMLErrorInfo( 
-            /* [out] */ SPSEMANTICERRORINFO *pSemanticErrorInfo) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPSEMANTICERRORINFO *pSemanticErrorInfo) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAudio( 
             /* [in] */ ULONG ulStartElement,
@@ -7986,7 +7987,7 @@ EXTERN_C const IID IID_ISpRecoResult;
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetResultTimes( 
-            /* [out] */ SPRECORESULTTIMES *pTimes) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPRECORESULTTIMES *pTimes) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAlternates( 
             /* [in] */ ULONG ulStartElement,
@@ -7994,7 +7995,7 @@ EXTERN_C const IID IID_ISpRecoResult;
             /* [in] */ ULONG ulRequestCount,
             /* [out][annotation] */ 
             _Out_writes_to_(ulRequestCount,*pcPhrasesReturned)  ISpPhraseAlt **ppPhrases,
-            /* [out] */ ULONG *pcPhrasesReturned) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pcPhrasesReturned) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAudio( 
             /* [in] */ ULONG ulStartElement,
@@ -8005,10 +8006,10 @@ EXTERN_C const IID IID_ISpRecoResult;
             /* [in] */ ULONG ulStartElement,
             /* [in] */ ULONG cElements,
             /* [in] */ DWORD dwFlags,
-            /* [out] */ ULONG *pulStreamNumber) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pulStreamNumber) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Serialize( 
-            /* [out] */ SPSERIALIZEDRESULT **ppCoMemSerializedResult) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPSERIALIZEDRESULT **ppCoMemSerializedResult) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE ScaleAudio( 
             /* [in] */ const GUID *pAudioFormatId,
@@ -8447,7 +8448,7 @@ EXTERN_C const IID IID_ISpXMLRecoResult;
             /* [in] */ SPXMLRESULTOPTIONS Options) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetXMLErrorInfo( 
-            /* [out] */ SPSEMANTICERRORINFO *pSemanticErrorInfo) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPSEMANTICERRORINFO *pSemanticErrorInfo) = 0;
         
     };
     
@@ -8743,14 +8744,14 @@ EXTERN_C const IID IID_ISpGrammarBuilder;
             /* [in] */ DWORD dwRuleId,
             /* [in] */ DWORD dwAttributes,
             /* [in] */ BOOL fCreateIfNotExist,
-            /* [out] */ SPSTATEHANDLE *phInitialState) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPSTATEHANDLE *phInitialState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE ClearRule( 
             /* [in] */ SPSTATEHANDLE hState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE CreateNewState( 
             /* [in] */ SPSTATEHANDLE hState,
-            /* [out] */ SPSTATEHANDLE *phState) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPSTATEHANDLE *phState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE AddWordTransition( 
             /* [in] */ SPSTATEHANDLE hFromState,
@@ -8952,7 +8953,7 @@ EXTERN_C const IID IID_ISpRecoGrammar;
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetGrammarId( 
-            /* [out] */ ULONGLONG *pullGrammarId) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONGLONG *pullGrammarId) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetRecoContext( 
             /* [out][annotation] */ 
@@ -8988,7 +8989,7 @@ EXTERN_C const IID IID_ISpRecoGrammar;
         virtual HRESULT STDMETHODCALLTYPE SetRuleState( 
             /* [string][in][annotation] */ 
             _In_opt_  LPCWSTR pszName,
-            /* [in] */ void *pReserved,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pReserved,
             /* [in] */ SPRULESTATE NewState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetRuleIdState( 
@@ -9016,7 +9017,7 @@ EXTERN_C const IID IID_ISpRecoGrammar;
         
         virtual HRESULT STDMETHODCALLTYPE IsPronounceable( 
             /* [string][in] */ LPCWSTR pszWord,
-            /* [out] */ SPWORDPRONOUNCEABLE *pWordPronounceable) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPWORDPRONOUNCEABLE *pWordPronounceable) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetGrammarState( 
             /* [in] */ SPGRAMMARSTATE eGrammarState) = 0;
@@ -9027,7 +9028,7 @@ EXTERN_C const IID IID_ISpRecoGrammar;
             _Outptr_opt_  LPWSTR *ppszCoMemErrorText) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetGrammarState( 
-            /* [out] */ SPGRAMMARSTATE *peGrammarState) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPGRAMMARSTATE *peGrammarState) = 0;
         
     };
     
@@ -9492,8 +9493,8 @@ EXTERN_C const IID IID_ISpRecoGrammar2;
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetRules( 
-            /* [out] */ SPRULE **ppCoMemRules,
-            /* [out] */ UINT *puNumRules) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPRULE **ppCoMemRules,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") UINT *puNumRules) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE LoadCmdFromFile2( 
             /* [string][in] */ LPCWSTR pszFileName,
@@ -9687,18 +9688,18 @@ EXTERN_C const IID IID_ISpeechResourceLoader;
             /* [in] */ VARIANT_BOOL fAlwaysReload,
             /* [out] */ IUnknown **pStream,
             /* [out][annotation] */ 
-            _Out_opt_  BSTR *pbstrMIMEType,
-            /* [out] */ VARIANT_BOOL *pfModified,
+            _Win32_RAIIFree_(SysFreeString) _Out_opt_  BSTR *pbstrMIMEType,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") VARIANT_BOOL *pfModified,
             /* [out][annotation] */ 
-            _Out_opt_  BSTR *pbstrRedirectUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_opt_  BSTR *pbstrRedirectUrl) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE GetLocalCopy( 
             /* [in] */ BSTR bstrResourceUri,
-            /* [out] */ BSTR *pbstrLocalPath,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BSTR *pbstrLocalPath,
             /* [out][annotation] */ 
-            _Out_opt_  BSTR *pbstrMIMEType,
+            _Win32_RAIIFree_(SysFreeString) _Out_opt_  BSTR *pbstrMIMEType,
             /* [out][annotation] */ 
-            _Out_opt_  BSTR *pbstrRedirectUrl) = 0;
+            _Win32_RAIIFree_(SysFreeString) _Out_opt_  BSTR *pbstrRedirectUrl) = 0;
         
         virtual /* [id] */ HRESULT STDMETHODCALLTYPE ReleaseLocalCopy( 
             /* [in] */ BSTR pbstrLocalPath) = 0;
@@ -9906,10 +9907,10 @@ EXTERN_C const IID IID_ISpRecoContext;
             /* [out] */ ISpRecoGrammar **ppGrammar) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetStatus( 
-            /* [out] */ SPRECOCONTEXTSTATUS *pStatus) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPRECOCONTEXTSTATUS *pStatus) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetMaxAlternates( 
-            /* [in] */ ULONG *pcAlternates) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") ULONG *pcAlternates) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetMaxAlternates( 
             /* [in] */ ULONG cAlternates) = 0;
@@ -9920,9 +9921,9 @@ EXTERN_C const IID IID_ISpRecoContext;
             /* [in] */ const WAVEFORMATEX *pWaveFormatEx) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAudioOptions( 
-            /* [in] */ SPAUDIOOPTIONS *pOptions,
-            /* [out] */ GUID *pAudioFormatId,
-            /* [out] */ WAVEFORMATEX **ppCoMemWFEX) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") SPAUDIOOPTIONS *pOptions,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") GUID *pAudioFormatId,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") WAVEFORMATEX **ppCoMemWFEX) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE DeserializeResult( 
             /* [in] */ const SPSERIALIZEDRESULT *pSerializedResult,
@@ -9954,13 +9955,13 @@ EXTERN_C const IID IID_ISpRecoContext;
             /* [in] */ ULONGLONG ullEventInterest) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetVoicePurgeEvent( 
-            /* [out] */ ULONGLONG *pullEventInterest) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONGLONG *pullEventInterest) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetContextState( 
             /* [in] */ SPCONTEXTSTATE eContextState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetContextState( 
-            /* [out] */ SPCONTEXTSTATE *peContextState) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPCONTEXTSTATE *peContextState) = 0;
         
     };
     
@@ -10329,7 +10330,7 @@ EXTERN_C const IID IID_ISpRecoContext2;
             /* [in] */ DWORD eGrammarOptions) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetGrammarOptions( 
-            /* [out] */ DWORD *peGrammarOptions) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") DWORD *peGrammarOptions) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetAdaptationData2( 
             /* [in] */ LPCWSTR pAdaptationData,
@@ -10445,7 +10446,7 @@ EXTERN_C const IID IID_ISpProperties;
         
         virtual HRESULT STDMETHODCALLTYPE GetPropertyNum( 
             /* [in] */ LPCWSTR pName,
-            /* [out] */ LONG *plValue) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *plValue) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetPropertyString( 
             /* [in] */ LPCWSTR pName,
@@ -10638,32 +10639,32 @@ EXTERN_C const IID IID_ISpRecognizer;
         virtual HRESULT STDMETHODCALLTYPE IsSharedInstance( void) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetRecoState( 
-            /* [out] */ SPRECOSTATE *pState) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPRECOSTATE *pState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetRecoState( 
             /* [in] */ SPRECOSTATE NewState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetStatus( 
-            /* [out] */ SPRECOGNIZERSTATUS *pStatus) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPRECOGNIZERSTATUS *pStatus) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetFormat( 
             /* [in] */ SPSTREAMFORMATTYPE WaveFormatType,
-            /* [out] */ GUID *pFormatId,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") GUID *pFormatId,
             /* [out][annotation] */ 
             _Outptr_  WAVEFORMATEX **ppCoMemWFEX) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE IsUISupported( 
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData,
-            /* [out] */ BOOL *pfSupported) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *pfSupported) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE DisplayUI( 
             /* [in] */ HWND hwndParent,
             /* [in][annotation] */ 
             _In_opt_  LPCWSTR pszTitle,
             /* [in] */ LPCWSTR pszTypeOfUI,
-            /* [in] */ void *pvExtraData,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") void *pvExtraData,
             /* [in] */ ULONG cbExtraData) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE EmulateRecognition( 
@@ -11154,7 +11155,7 @@ EXTERN_C const IID IID_ISpEnginePronunciation;
             /* [string][in] */ LPCWSTR pszLeftContext,
             /* [string][in] */ LPCWSTR pszRightContext,
             /* [in] */ WORD LangID,
-            /* [out] */ SPNORMALIZATIONLIST *pNormalizationList) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPNORMALIZATIONLIST *pNormalizationList) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetPronunciations( 
             /* [string][in] */ LPCWSTR pszWord,
@@ -11284,8 +11285,8 @@ EXTERN_C const IID IID_ISpDisplayAlternates;
         virtual HRESULT STDMETHODCALLTYPE GetDisplayAlternates( 
             /* [in] */ const SPDISPLAYPHRASE *pPhrase,
             /* [in] */ ULONG cRequestCount,
-            /* [out] */ SPDISPLAYPHRASE **ppCoMemPhrases,
-            /* [out] */ ULONG *pcPhrasesReturned) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") SPDISPLAYPHRASE **ppCoMemPhrases,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") ULONG *pcPhrasesReturned) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetFullStopTrailSpace( 
             /* [in] */ ULONG ulTrailSpace) = 0;
@@ -12498,7 +12499,7 @@ EXTERN_C const IID IID_ISpeechDataKey;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetStringValue( 
             /* [in] */ __RPC__in const BSTR ValueName,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Value) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Value) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE SetLongValue( 
             /* [in] */ __RPC__in const BSTR ValueName,
@@ -12524,11 +12525,11 @@ EXTERN_C const IID IID_ISpeechDataKey;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE EnumKeys( 
             /* [in] */ long Index,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *SubKeyName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *SubKeyName) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE EnumValues( 
             /* [in] */ long Index,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *ValueName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ValueName) = 0;
         
     };
     
@@ -12764,7 +12765,7 @@ EXTERN_C const IID IID_ISpeechObjectToken;
     {
     public:
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Id( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *ObjectId) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ObjectId) = 0;
         
         virtual /* [hidden][id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_DataKey( 
             /* [retval][out] */ __RPC__deref_out_opt ISpeechDataKey **DataKey) = 0;
@@ -12774,7 +12775,7 @@ EXTERN_C const IID IID_ISpeechObjectToken;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetDescription( 
             /* [defaultvalue][in] */ long Locale,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Description) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Description) = 0;
         
         virtual /* [id][hidden][helpstring] */ HRESULT STDMETHODCALLTYPE SetId( 
             /* [in] */ __RPC__in BSTR Id,
@@ -12783,7 +12784,7 @@ EXTERN_C const IID IID_ISpeechObjectToken;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetAttribute( 
             /* [in] */ __RPC__in BSTR AttributeName,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *AttributeValue) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *AttributeValue) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE CreateInstance( 
             /* [defaultvalue][in] */ __RPC__in_opt IUnknown *pUnkOuter,
@@ -12798,7 +12799,7 @@ EXTERN_C const IID IID_ISpeechObjectToken;
             /* [in] */ __RPC__in BSTR KeyName,
             /* [in] */ __RPC__in BSTR FileName,
             /* [in] */ SpeechTokenShellFolder Folder,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *FilePath) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *FilePath) = 0;
         
         virtual /* [id][hidden][helpstring] */ HRESULT STDMETHODCALLTYPE RemoveStorageFileName( 
             /* [in] */ __RPC__in BSTR ObjectStorageCLSID,
@@ -13236,13 +13237,13 @@ EXTERN_C const IID IID_ISpeechObjectTokenCategory;
     {
     public:
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Id( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Id) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Id) = 0;
         
         virtual /* [id][helpstring][propput] */ HRESULT STDMETHODCALLTYPE put_Default( 
             /* [in] */ __RPC__in const BSTR TokenId) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Default( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *TokenId) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *TokenId) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE SetId( 
             /* [in] */ __RPC__in const BSTR Id,
@@ -13818,7 +13819,7 @@ EXTERN_C const IID IID_ISpeechAudioFormat;
             /* [in] */ SpeechAudioFormatType AudioFormat) = 0;
         
         virtual /* [id][helpstring][hidden][propget] */ HRESULT STDMETHODCALLTYPE get_Guid( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Guid) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Guid) = 0;
         
         virtual /* [id][helpstring][hidden][propput] */ HRESULT STDMETHODCALLTYPE put_Guid( 
             /* [in] */ __RPC__in BSTR Guid) = 0;
@@ -16169,7 +16170,7 @@ EXTERN_C const IID IID_ISpeechVoiceStatus;
             /* [retval][out] */ __RPC__out long *Length) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_LastBookmark( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Bookmark) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Bookmark) = 0;
         
         virtual /* [hidden][id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_LastBookmarkId( 
             /* [retval][out] */ __RPC__out long *BookmarkId) = 0;
@@ -16995,7 +16996,7 @@ EXTERN_C const IID IID_ISpeechRecognizerStatus;
             /* [retval][out] */ __RPC__out long *NumberOfActiveRules) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_ClsidEngine( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *ClsidEngine) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ClsidEngine) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_SupportedLanguages( 
             /* [retval][out] */ __RPC__out VARIANT *SupportedLanguages) = 0;
@@ -17182,7 +17183,7 @@ EXTERN_C const IID IID_ISpeechRecoContext;
             /* [retval][out] */ __RPC__out SpeechInterference *Interference) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_RequestedUIType( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *UIType) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *UIType) = 0;
         
         virtual /* [id][helpstring][propputref] */ HRESULT STDMETHODCALLTYPE putref_Voice( 
             /* [in] */ __RPC__in_opt ISpeechVoice *Voice) = 0;
@@ -18078,7 +18079,7 @@ EXTERN_C const IID IID_ISpeechGrammarRule;
             /* [retval][out] */ __RPC__deref_out_opt ISpeechGrammarRuleState **State) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Name( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Name) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Name) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Id( 
             /* [retval][out] */ __RPC__out long *Id) = 0;
@@ -18301,7 +18302,7 @@ EXTERN_C const IID IID_ISpeechGrammarRules;
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE Commit( void) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE CommitAndSave( 
-            /* [out] */ __RPC__deref_out_opt BSTR *ErrorText,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ErrorText,
             /* [retval][out] */ __RPC__out VARIANT *SaveStream) = 0;
         
     };
@@ -18720,7 +18721,7 @@ EXTERN_C const IID IID_ISpeechGrammarRuleStateTransition;
             /* [retval][out] */ __RPC__out SpeechGrammarRuleStateTransitionType *Type) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Text( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Text) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Text) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Rule( 
             /* [retval][out] */ __RPC__deref_out_opt ISpeechGrammarRule **Rule) = 0;
@@ -18729,7 +18730,7 @@ EXTERN_C const IID IID_ISpeechGrammarRuleStateTransition;
             /* [retval][out] */ __RPC__out VARIANT *Weight) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_PropertyName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *PropertyName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *PropertyName) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_PropertyId( 
             /* [retval][out] */ __RPC__out long *PropertyId) = 0;
@@ -20337,7 +20338,7 @@ EXTERN_C const IID IID_ISpeechPhraseInfo;
             /* [retval][out] */ __RPC__deref_out_opt ISpeechPhraseReplacements **Replacements) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_EngineId( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *EngineIdGuid) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *EngineIdGuid) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_EnginePrivateData( 
             /* [retval][out] */ __RPC__out VARIANT *PrivateData) = 0;
@@ -20349,7 +20350,7 @@ EXTERN_C const IID IID_ISpeechPhraseInfo;
             /* [defaultvalue][in] */ long StartElement,
             /* [defaultvalue][in] */ long Elements,
             /* [defaultvalue][in] */ VARIANT_BOOL UseReplacements,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Text) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Text) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetDisplayAttributes( 
             /* [defaultvalue][in] */ long StartElement,
@@ -20637,10 +20638,10 @@ EXTERN_C const IID IID_ISpeechPhraseElement;
             /* [retval][out] */ __RPC__out long *RetainedSizeBytes) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_DisplayText( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *DisplayText) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *DisplayText) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_LexicalForm( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *LexicalForm) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *LexicalForm) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Pronunciation( 
             /* [retval][out] */ __RPC__out VARIANT *Pronunciation) = 0;
@@ -21054,7 +21055,7 @@ EXTERN_C const IID IID_ISpeechPhraseReplacement;
             /* [retval][out] */ __RPC__out SpeechDisplayAttributes *DisplayAttributes) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Text( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Text) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Text) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_FirstElement( 
             /* [retval][out] */ __RPC__out long *FirstElement) = 0;
@@ -21384,7 +21385,7 @@ EXTERN_C const IID IID_ISpeechPhraseProperty;
     {
     public:
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Name( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Name) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Name) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Id( 
             /* [retval][out] */ __RPC__out long *Id) = 0;
@@ -21772,7 +21773,7 @@ EXTERN_C const IID IID_ISpeechPhraseRule;
     {
     public:
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Name( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Name) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Name) = 0;
         
         virtual /* [id][helpstring][propget] */ HRESULT STDMETHODCALLTYPE get_Id( 
             /* [retval][out] */ __RPC__out long *Id) = 0;
@@ -22568,7 +22569,7 @@ EXTERN_C const IID IID_ISpeechLexiconWord;
             /* [retval][out] */ __RPC__out SpeechWordType *WordType) = 0;
         
         virtual /* [propget][id] */ HRESULT STDMETHODCALLTYPE get_Word( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Word) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Word) = 0;
         
         virtual /* [propget][id] */ HRESULT STDMETHODCALLTYPE get_Pronunciations( 
             /* [retval][out] */ __RPC__deref_out_opt ISpeechLexiconPronunciations **Pronunciations) = 0;
@@ -22907,7 +22908,7 @@ EXTERN_C const IID IID_ISpeechLexiconPronunciation;
             /* [retval][out] */ __RPC__out VARIANT *PhoneIds) = 0;
         
         virtual /* [helpstring][propget][id] */ HRESULT STDMETHODCALLTYPE get_Symbolic( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Symbolic) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Symbolic) = 0;
         
     };
     
@@ -23182,13 +23183,13 @@ EXTERN_C const IID IID_ISpeechXMLRecoResult;
     public:
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetXMLResult( 
             /* [in] */ SPXMLRESULTOPTIONS Options,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pResult) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pResult) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetXMLErrorInfo( 
             /* [out] */ __RPC__out long *LineNumber,
-            /* [out] */ __RPC__deref_out_opt BSTR *ScriptLine,
-            /* [out] */ __RPC__deref_out_opt BSTR *Source,
-            /* [out] */ __RPC__deref_out_opt BSTR *Description,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ScriptLine,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Source,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Description,
             /* [out] */ __RPC__out long *ResultCode,
             /* [retval][out] */ __RPC__out VARIANT_BOOL *IsError) = 0;
         
@@ -23470,13 +23471,13 @@ EXTERN_C const IID IID_ISpeechRecoResultDispatch;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetXMLResult( 
             /* [in] */ SPXMLRESULTOPTIONS Options,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pResult) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pResult) = 0;
         
         virtual /* [id][helpstring] */ HRESULT STDMETHODCALLTYPE GetXMLErrorInfo( 
             /* [out] */ __RPC__out long *LineNumber,
-            /* [out] */ __RPC__deref_out_opt BSTR *ScriptLine,
-            /* [out] */ __RPC__deref_out_opt BSTR *Source,
-            /* [out] */ __RPC__deref_out_opt BSTR *Description,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *ScriptLine,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Source,
+            /* [out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Description,
             /* [out] */ __RPC__out HRESULT *ResultCode,
             /* [retval][out] */ __RPC__out VARIANT_BOOL *IsError) = 0;
         
@@ -23882,7 +23883,7 @@ EXTERN_C const IID IID_ISpeechPhoneConverter;
         
         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE IdToPhone( 
             /* [in] */ const VARIANT IdArray,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *Phonemes) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *Phonemes) = 0;
         
     };
     
