diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/sensorsutils.h b/generation/WinSDK/RecompiledIdlHeaders/shared/sensorsutils.h
index a27db07739..2d01afb897 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/sensorsutils.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/sensorsutils.h
@@ -17,6 +17,7 @@ Abstract:
 #endif
 
 #include <SensorsStructures.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -236,7 +237,7 @@ CollectionsListUpdateMarshalledPointer(
 NTSTATUS
 SerializationBufferAllocate(
     _In_ ULONG SizeInBytes,
-    _Outptr_result_bytebuffer_(SizeInBytes) BYTE** pBuffer
+     _Outptr_result_bytebuffer_(SizeInBytes) BYTE** pBuffer
     );
 
 VOID
@@ -260,7 +261,7 @@ NTSTATUS
 CollectionsListAllocateBufferAndSerialize(
     _In_ const PSENSOR_COLLECTION_LIST SourceCollection,
     _Out_ ULONG* pTargetBufferSizeInBytes,
-    _Outptr_result_bytebuffer_(*pTargetBufferSizeInBytes) BYTE** pTargetBuffer
+     _Outptr_result_bytebuffer_(*pTargetBufferSizeInBytes) BYTE** pTargetBuffer
     );
 
 NTSTATUS
