diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/SessionPortListener.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/SessionPortListener.h
index ea09cd585a..abb1ca26af 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/SessionPortListener.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/SessionPortListener.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * alljoyn_sessionportlistener is a collection of functions implemented by users
@@ -94,7 +95,7 @@ typedef struct {
  *
  * @return Handle to newly allocated alljoyn_sessionportlistener.
  */
-extern AJ_API alljoyn_sessionportlistener AJ_CALL alljoyn_sessionportlistener_create(const alljoyn_sessionportlistener_callbacks* callbacks,
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_sessionportlistener AJ_CALL alljoyn_sessionportlistener_create(const alljoyn_sessionportlistener_callbacks* callbacks,
                                                                                      const void* context);
 
 /**
@@ -102,7 +103,7 @@ extern AJ_API alljoyn_sessionportlistener AJ_CALL alljoyn_sessionportlistener_cr
  *
  * @param listener alljoyn_sessionportlistener to destroy.
  */
-extern AJ_API void AJ_CALL alljoyn_sessionportlistener_destroy(alljoyn_sessionportlistener listener);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionportlistener_destroy(alljoyn_sessionportlistener listener);
 
 #ifdef __cplusplus
 } /* extern "C" */
