diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/Session.h b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/Session.h
index 7965cfa2d4..7ed63dbd94 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/Session.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/alljoyn_c/Session.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /**
  * @file
  * AllJoyn session related data types.
@@ -78,15 +79,15 @@ typedef uint32_t alljoyn_sessionid;
  *
  * @return the allocated alljoyn_sessionopts
  */
-extern AJ_API alljoyn_sessionopts AJ_CALL alljoyn_sessionopts_create(uint8_t traffic, QCC_BOOL isMultipoint,
-                                                                     uint8_t proximity, alljoyn_transportmask transports);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_sessionopts AJ_CALL alljoyn_sessionopts_create(_WIN32META_ANNOTATION_("win32metadata:in") uint8_t traffic, _WIN32META_ANNOTATION_("win32metadata:in") QCC_BOOL isMultipoint,
+                                                                     _WIN32META_ANNOTATION_("win32metadata:in") uint8_t proximity, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_transportmask transports);
 
 /**
  * Destroy an alljoyn_sessionopts created with alljoyn_sessionopts_create.
  *
  * @param opts alljoyn_sessionopts to destroy
  */
-extern AJ_API void AJ_CALL alljoyn_sessionopts_destroy(alljoyn_sessionopts opts);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionopts_destroy(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionopts opts);
 
 /**
  * Get accessor for the traffic member of alljoyn_sessionopts.
@@ -95,7 +96,7 @@ extern AJ_API void AJ_CALL alljoyn_sessionopts_destroy(alljoyn_sessionopts opts)
  *
  * @return Traffic type specified by the specified alljoyn_sessionopts.
  */
-extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_traffic(const alljoyn_sessionopts opts);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_traffic(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts opts);
 
 /**
  * Set accessor for the traffic member of alljoyn_sessionopts.
@@ -103,7 +104,7 @@ extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_traffic(const alljoyn_sess
  * @param opts alljoyn_sessionopts
  * @param traffic Traffic type to set in the alljoyn_sessionopts.
  */
-extern AJ_API void AJ_CALL alljoyn_sessionopts_set_traffic(alljoyn_sessionopts opts, uint8_t traffic);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionopts_set_traffic(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionopts opts, _WIN32META_ANNOTATION_("win32metadata:in") uint8_t traffic);
 
 /**
  * Get accessor for the isMultipoint member of alljoyn_sessionopts.
@@ -112,7 +113,7 @@ extern AJ_API void AJ_CALL alljoyn_sessionopts_set_traffic(alljoyn_sessionopts o
  *
  * @return Multipoint value specified by the specified alljoyn_sessionopts.
  */
-extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_get_multipoint(const alljoyn_sessionopts opts);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_get_multipoint(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts opts);
 
 /**
  * Set accessor for the isMultipoint member of alljoyn_sessionopts.
@@ -120,7 +121,7 @@ extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_get_multipoint(const alljoyn_
  * @param opts alljoyn_sessionopts
  * @param isMultipoint Multipoint value to set in the alljoyn_sessionopts.
  */
-extern AJ_API void AJ_CALL alljoyn_sessionopts_set_multipoint(alljoyn_sessionopts opts, QCC_BOOL isMultipoint);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionopts_set_multipoint(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionopts opts, _WIN32META_ANNOTATION_("win32metadata:in") QCC_BOOL isMultipoint);
 
 /**
  * Get accessor for the proximity member of alljoyn_sessionopts.
@@ -129,7 +130,7 @@ extern AJ_API void AJ_CALL alljoyn_sessionopts_set_multipoint(alljoyn_sessionopt
  *
  * @return Proximity specified by the specified alljoyn_sessionopts.
  */
-extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_proximity(const alljoyn_sessionopts opts);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_proximity(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts opts);
 
 /**
  * Set accessor for the proximity member of alljoyn_sessionopts.
@@ -137,7 +138,7 @@ extern AJ_API uint8_t AJ_CALL alljoyn_sessionopts_get_proximity(const alljoyn_se
  * @param opts alljoyn_sessionopts
  * @param proximity Proximity value to set in the alljoyn_sessionopts.
  */
-extern AJ_API void AJ_CALL alljoyn_sessionopts_set_proximity(alljoyn_sessionopts opts, uint8_t proximity);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionopts_set_proximity(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionopts opts, _WIN32META_ANNOTATION_("win32metadata:in") uint8_t proximity);
 
 /**
  * Get accessor for the transports member of alljoyn_sessionopts.
@@ -146,7 +147,7 @@ extern AJ_API void AJ_CALL alljoyn_sessionopts_set_proximity(alljoyn_sessionopts
  *
  * @return Transports allowed by the specified alljoyn_sessionopts.
  */
-extern AJ_API alljoyn_transportmask AJ_CALL alljoyn_sessionopts_get_transports(const alljoyn_sessionopts opts);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API alljoyn_transportmask AJ_CALL alljoyn_sessionopts_get_transports(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts opts);
 
 /**
  * Set accessor for the transports member of alljoyn_sessionopts.
@@ -154,7 +155,7 @@ extern AJ_API alljoyn_transportmask AJ_CALL alljoyn_sessionopts_get_transports(c
  * @param opts alljoyn_sessionopts
  * @param transports change the transports allowed by the specified alljoyn_sessionopts.
  */
-extern AJ_API void AJ_CALL alljoyn_sessionopts_set_transports(alljoyn_sessionopts opts, alljoyn_transportmask transports);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API void AJ_CALL alljoyn_sessionopts_set_transports(_WIN32META_ANNOTATION_("win32metadata:in") alljoyn_sessionopts opts, _WIN32META_ANNOTATION_("win32metadata:in") alljoyn_transportmask transports);
 
 /**
  * Determine whether one alljoyn_sessionopts is compatible with the alljoyn_sessionopts
@@ -171,7 +172,7 @@ extern AJ_API void AJ_CALL alljoyn_sessionopts_set_transports(alljoyn_sessionopt
  * @param other  Options to be compared against one.
  * @return QCC_TRUE iff this alljoyn_sessionopts can use the option set offered by other.
  */
-extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_iscompatible(const alljoyn_sessionopts one, const alljoyn_sessionopts other);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_iscompatible(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts one, _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts other);
 
 /**
  * Compare two alljoyn_sessionopts.
@@ -180,7 +181,7 @@ extern AJ_API QCC_BOOL AJ_CALL alljoyn_sessionopts_iscompatible(const alljoyn_se
  * @param other  Options to be compared against one.
  * @return 0 if the alljoyn_sessionopts are equal, 1 if one > other, -1 if one < other.
  */
-extern AJ_API int32_t AJ_CALL alljoyn_sessionopts_cmp(const alljoyn_sessionopts one, const alljoyn_sessionopts other);
+_Win32_ImportLibrary_("msajapi.dll") extern AJ_API int32_t AJ_CALL alljoyn_sessionopts_cmp(_WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts one, _WIN32META_ANNOTATION_("win32metadata:in") const alljoyn_sessionopts other);
 
 #ifdef __cplusplus
 } /* extern "C" */
