diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/shdeprecated.h b/generation/WinSDK/RecompiledIdlHeaders/um/shdeprecated.h
index 0ad95321b8..192e01a4bb 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/shdeprecated.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/shdeprecated.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -526,9 +527,9 @@ class CIE4ConnectionPoint : public IConnectionPoint
 {
     // This is how you actually fire the events
     // Those called by shell32 are virtual
-    virtual HRESULT DoInvokeIE4(BOOL *pf, void **ppv, DISPID dispid, DISPPARAMS *pdispparams) PURE;
+    virtual HRESULT DoInvokeIE4(_WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") BOOL *pf, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void **ppv, DISPID dispid, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") DISPPARAMS *pdispparams) PURE;
     // This helper function does work that callers of DoInvoke often need done
-    virtual HRESULT DoInvokePIDLIE4(DISPID dispid, PCIDLIST_ABSOLUTE pidl, BOOL fCanCancel) PURE;
+    virtual HRESULT DoInvokePIDLIE4(DISPID dispid, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PCIDLIST_ABSOLUTE pidl, BOOL fCanCancel) PURE;
 };
 #else // !__cplusplus
 typedef void *CIE4ConnectionPoint;
@@ -550,7 +551,7 @@ EXTERN_C const IID IID_IExpDispSupportXP;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("2f0dd58c-f789-4f14-99fb-9293b3c9c212")
+    MIDL_INTERFACE("2f0dd58c-f789-4f14-99fb-9293b3c9c212") _Windows_SupportedOS_WindowsVista_ 
     IExpDispSupportXP : public IUnknown
     {
     public:
@@ -1044,11 +1045,11 @@ EXTERN_C const IID IID_IBrowserService;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE GetPalette( 
             /* [annotation][out] */ 
-            _Out_  HPALETTE *hpal) = 0;
+            _Win32_RAIIFree_(DeleteObject, -1, 0) _Out_  HPALETTE *hpal) = 0;
         
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE RegisterWindow( 
             BOOL fForceRegister,
-            int swc) = 0;
+            _Win32_AssociatedEnum_(ShellWindowTypeConstants) int swc) = 0;
         
     };
     
@@ -1608,7 +1609,7 @@ typedef struct tagFolderSetData *LPFOLDERSETDATA;
 
 #if !defined(HMONITOR_DECLARED) && !defined(HMONITOR) && (WINVER < 0x0500)
 #define HMONITOR_DECLARED
-DECLARE_HANDLE(HMONITOR);
+DECLARE_HANDLE(HMONITOR) _WIN32META_ANNOTATION_("win32metadata:invalid_handle=-1")  _WIN32META_ANNOTATION_("win32metadata:invalid_handle=0") ;
 #endif
 #if 0
 typedef void *LPCREATESTRUCT;
@@ -1645,7 +1646,7 @@ EXTERN_C const IID IID_IBrowserService2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("68BD21CC-438B-11d2-A560-00A0C92DBFE8")
+    MIDL_INTERFACE("68BD21CC-438B-11d2-A560-00A0C92DBFE8") _Windows_SupportedOS_Windows2000_ 
     IBrowserService2 : public IBrowserService
     {
     public:
@@ -1811,7 +1812,7 @@ EXTERN_C const IID IID_IBrowserService2;
             _In_  LPARAM lParam) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetFolderSetData( 
-            /* [out] */ struct tagFolderSetData *pfsd) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") struct tagFolderSetData *pfsd) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE _OnFocusChange( 
             /* [annotation][in] */ 
@@ -1832,7 +1833,7 @@ EXTERN_C const IID IID_IBrowserService2;
             _In_  UINT uState) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE _GetViewBorderRect( 
-            /* [out] */ RECT *prc) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") RECT *prc) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE _UpdateViewRectSize( void) = 0;
         
@@ -1849,7 +1850,7 @@ EXTERN_C const IID IID_IBrowserService2;
             _In_  HMONITOR hmon) = 0;
         
         virtual IStream *STDMETHODCALLTYPE v_GetViewStream( 
-            /* [in] */ PCIDLIST_ABSOLUTE pidl,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCIDLIST_ABSOLUTE pidl,
             /* [annotation][in] */ 
             _In_  DWORD grfMode,
             /* [annotation][in] */ 
@@ -1912,7 +1913,7 @@ EXTERN_C const IID IID_IBrowserService2;
             _In_  LPMSG lpMsg) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE v_MayTranslateAccelerator( 
-            /* [in] */ MSG *pmsg) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") MSG *pmsg) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE _GetBorderDWHelper( 
             /* [annotation][in] */ 
@@ -2880,7 +2881,7 @@ EXTERN_C const IID IID_IBrowserService3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("27D7CE21-762D-48F3-86F3-40E2FD3749C4")
+    MIDL_INTERFACE("27D7CE21-762D-48F3-86F3-40E2FD3749C4") _Windows_SupportedOS_WindowsXP_ 
     IBrowserService3 : public IBrowserService2
     {
     public:
@@ -3884,7 +3885,7 @@ EXTERN_C const IID IID_IBrowserService4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("639f1bff-e135-4096-abd8-e0f504d649a4")
+    MIDL_INTERFACE("639f1bff-e135-4096-abd8-e0f504d649a4") _Windows_SupportedOS_WindowsVista_ 
     IBrowserService4 : public IBrowserService3
     {
     public:
@@ -4903,7 +4904,7 @@ EXTERN_C const IID IID_ITrackShellMenu;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("8278F932-2A3E-11d2-838F-00C04FD918D0")
+    MIDL_INTERFACE("8278F932-2A3E-11d2-838F-00C04FD918D0") _Windows_SupportedOS_WindowsVista_ 
     ITrackShellMenu : public IShellMenu
     {
     public:
@@ -4914,8 +4915,8 @@ EXTERN_C const IID IID_ITrackShellMenu;
         
         virtual HRESULT STDMETHODCALLTYPE Popup( 
             /* [in] */ HWND hwnd,
-            /* [in] */ POINTL *ppt,
-            /* [in] */ RECTL *prcExclude,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") POINTL *ppt,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") RECTL *prcExclude,
             MP_POPUPFLAGS dwFlags) = 0;
         
     };
@@ -5154,7 +5155,7 @@ extern RPC_IF_HANDLE __MIDL_itf_shdeprecated_0000_0011_v0_0_s_ifspec;
 
 /* Additional Prototypes for ALL interfaces */
 
-unsigned long             __RPC_USER  HMENU_UserSize(     __RPC__in unsigned long *, unsigned long            , __RPC__in HMENU * ); 
+_Win32_ImportLibrary_("ole32.dll") unsigned long             __RPC_USER  HMENU_UserSize(     __RPC__in unsigned long *, unsigned long            , __RPC__in HMENU * ); 
 unsigned char * __RPC_USER  HMENU_UserMarshal(  __RPC__in unsigned long *, __RPC__inout_xcount(0) unsigned char *, __RPC__in HMENU * ); 
 unsigned char * __RPC_USER  HMENU_UserUnmarshal(__RPC__in unsigned long *, __RPC__in_xcount(0) unsigned char *, __RPC__out HMENU * ); 
 void                      __RPC_USER  HMENU_UserFree(     __RPC__in unsigned long *, __RPC__in HMENU * ); 
