diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/shtypes.h b/generation/WinSDK/RecompiledIdlHeaders/um/shtypes.h
index ed9335e9ef..18a29f3949 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/shtypes.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/shtypes.h
@@ -28,6 +28,7 @@
 
 #ifndef __shtypes_h__
 #define __shtypes_h__
+#include "win32metadata_annotations.h"
 
 #if defined(_MSC_VER) && (_MSC_VER >= 1020)
 #pragma once
@@ -124,7 +125,7 @@ extern "C"{
 //     BYTE        abID[];         // The item ID (variable length)
 //
 #include <pshpack1.h>
-typedef struct _SHITEMID
+typedef struct   _SHITEMID
     {
     USHORT cb;
     BYTE abID[ 1 ];
@@ -309,7 +310,7 @@ typedef STRRET *LPSTRRET;
 //  str;            // String information
 //
 #include <pshpack1.h>
-typedef struct _SHELLDETAILS
+typedef struct  _SHELLDETAILS
     {
     int fmt;
     int cxChar;
@@ -322,7 +323,7 @@ typedef struct _SHELLDETAILS *LPSHELLDETAILS;
 
 #if (_WIN32_IE >= _WIN32_IE_IE60SP2)
 typedef /* [v1_enum] */ 
-enum tagPERCEIVED
+enum  tagPERCEIVED
     {
         PERCEIVED_TYPE_FIRST	= -3,
         PERCEIVED_TYPE_CUSTOM	= -3,
@@ -354,7 +355,7 @@ typedef DWORD PERCEIVEDFLAG;
 #endif  // _WIN32_IE_IE60SP2
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
-typedef struct _COMDLG_FILTERSPEC
+typedef struct   _COMDLG_FILTERSPEC
     {
     /* [string] */ LPCWSTR pszName;
     /* [string] */ LPCWSTR pszSpec;
@@ -478,7 +479,7 @@ typedef PROPERTYKEY SHCOLUMNID;
 typedef const SHCOLUMNID *LPCSHCOLUMNID;
 
 typedef /* [v1_enum] */ 
-enum DEVICE_SCALE_FACTOR
+enum   DEVICE_SCALE_FACTOR
     {
         DEVICE_SCALE_FACTOR_INVALID	= 0,
         SCALE_100_PERCENT	= 100,
