diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/slpublic.h b/generation/WinSDK/RecompiledIdlHeaders/um/slpublic.h
index cd708ed5df9aed45f3d72c674c621363f93c24be..5cde423b0c163cafd2ad66e42670a1ce0bcaf27e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/slpublic.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/slpublic.h
@@ -17,6 +17,7 @@ Abstract:
 #define _SLPUBLIC_H_
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -35,7 +36,7 @@ typedef PVOID HSLC;
  
 typedef PVOID HSLP;
     
-typedef enum _tagSLDATATYPE
+typedef enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ _tagSLDATATYPE
 {
     SL_DATA_NONE        = REG_NONE,
     SL_DATA_SZ          = REG_SZ,
@@ -48,7 +49,7 @@ typedef enum _tagSLDATATYPE
 //
 // Type of SL Id
 //
-typedef enum _tagSLIDTYPE
+typedef enum _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSLIDTYPE
 {
     SL_ID_APPLICATION = 0,
     SL_ID_PRODUCT_SKU,
@@ -64,7 +65,7 @@ typedef enum _tagSLIDTYPE
 //
 // Licensing status
 //
-typedef enum _tagSLLICENSINGSTATUS
+typedef enum _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSLLICENSINGSTATUS
 {
     SL_LICENSING_STATUS_UNLICENSED,
     SL_LICENSING_STATUS_LICENSED,
@@ -76,7 +77,7 @@ typedef enum _tagSLLICENSINGSTATUS
 //
 // Licensing status
 //
-typedef struct _tagSL_LICENSING_STATUS
+typedef struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSL_LICENSING_STATUS
 {
     SLID SkuId;                          // SKU id
     SLLICENSINGSTATUS eStatus;           // licensing status, see SLLICENSINGSTATUS
@@ -89,13 +90,13 @@ typedef struct _tagSL_LICENSING_STATUS
 //
 // Product activation structures
 //
-typedef enum _tagSL_ACTIVATION_TYPE
+typedef enum _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSL_ACTIVATION_TYPE
 {
     SL_ACTIVATION_TYPE_DEFAULT          = 0,
     SL_ACTIVATION_TYPE_ACTIVE_DIRECTORY = 1,
 } SL_ACTIVATION_TYPE;
 
-typedef struct _tagSL_ACTIVATION_INFO_HEADER
+typedef struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSL_ACTIVATION_INFO_HEADER
 {
     DWORD                       cbSize;
     SL_ACTIVATION_TYPE          type;
@@ -104,7 +105,7 @@ typedef struct _tagSL_ACTIVATION_INFO_HEADER
 //
 // Active directory activation (header.type == SL_ACTIVATION_TYPE_ACTIVE_DIRECTORY)
 //
-typedef struct _tagSL_AD_ACTIVATION_INFO
+typedef struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _tagSL_AD_ACTIVATION_INFO
 {
     SL_ACTIVATION_INFO_HEADER   header;
     PCWSTR                      pwszProductKey;
@@ -199,7 +200,7 @@ typedef enum
 #define SPP_MIGRATION_GATHER_ALL                            0xFFFFFFFF
 
 
-typedef enum _SL_GENUINE_STATE
+typedef enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ _SL_GENUINE_STATE
 {
     SL_GEN_STATE_IS_GENUINE         = 0,
     SL_GEN_STATE_INVALID_LICENSE,
@@ -208,7 +209,7 @@ typedef enum _SL_GENUINE_STATE
     SL_GEN_STATE_LAST, 
 } SL_GENUINE_STATE;
 
-typedef struct _tagSL_NONGENUINE_UI_OPTIONS
+typedef struct _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ _tagSL_NONGENUINE_UI_OPTIONS
 {
     DWORD                       cbSize;
     CONST SLID*                 pComponentId;
