diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/socketapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/socketapi.h
index 59c12438772cc67f7bf635825b7bde3ecba4028f..5bf3f401827b5c0391dc233480e25802489aea30 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/socketapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/socketapi.h
@@ -24,12 +24,15 @@ extern "C"
 #endif
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
 #if NTDDI_VERSION >= NTDDI_WIN8
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 HRESULT WINAPI SetSocketMediaStreamingMode(_In_ BOOL value);
 
 #endif // NTDDI_VERSION >= NTDDI_WIN8
