diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/statehelpers.h b/generation/WinSDK/RecompiledIdlHeaders/um/statehelpers.h
index 126ecdff47..b2672195d1 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/statehelpers.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/statehelpers.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /************************************************************************************
 *                                                                                   *
 * statehelpers.h -- ApiSet Contract for api-ms-win-core-state-helpers-l1            *
@@ -31,7 +32,7 @@ extern "C" {
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN10)
 
-WINBASEAPI
+_Win32_AssociatedEnum_(WIN32_ERROR) WINBASEAPI
 LSTATUS
 WINAPI
 GetRegistryValueWithFallbackW(
