diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/SubsMgr.h b/generation/WinSDK/RecompiledIdlHeaders/um/SubsMgr.h
index 66b7b85c33..fbd7cd8b24 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/SubsMgr.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/SubsMgr.h
@@ -118,6 +118,7 @@ extern "C"{
 //=--------------------------------------------------------------------------=
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -141,7 +142,7 @@ extern RPC_IF_HANDLE __MIDL_itf_subsmgr_0000_0000_v0_0_s_ifspec;
 
 typedef /* [unique] */  __RPC_unique_pointer IEnumItemProperties *LPENUMITEMPROPERTIES;
 
-typedef struct _tagITEMPROP
+typedef struct  _tagITEMPROP
     {
     VARIANT variantValue;
     LPWSTR pwszName;
