diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/swdevicedef.h b/generation/WinSDK/RecompiledIdlHeaders/um/swdevicedef.h
index 1b1db0b47b..10b79db4fc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/swdevicedef.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/swdevicedef.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation
@@ -36,7 +37,7 @@ typedef struct _SW_DEVICE_CREATE_INFO
     __in_opt PCZZWSTR pszzHardwareIds;
     __in_opt PCZZWSTR pszzCompatibleIds;
     __in_opt const GUID *pContainerId;
-    ULONG CapabilityFlags;
+     ULONG CapabilityFlags;
     __in_opt PCWSTR pszDeviceDescription;
     __in_opt PCWSTR pszDeviceLocation;
     __in_opt const SECURITY_DESCRIPTOR *pSecurityDescriptor;
