diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/swdevice.h b/generation/WinSDK/RecompiledIdlHeaders/um/swdevice.h
index b1f5c27c34..d9211d2ec4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/swdevice.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/swdevice.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation
@@ -38,42 +39,48 @@ typedef VOID (WINAPI *SW_DEVICE_CREATE_CALLBACK)(
     _In_opt_ PVOID pContext,
     _In_opt_ PCWSTR pszDeviceInstanceId);
 
-_Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+ 
 SwDeviceCreate(
     _In_ PCWSTR pszEnumeratorName,
     _In_ PCWSTR pszParentDeviceInstance,
     _In_ const SW_DEVICE_CREATE_INFO *pCreateInfo,
     _In_ ULONG cPropertyCount,
     _In_reads_opt_(cPropertyCount) const DEVPROPERTY *pProperties,
-    _In_ SW_DEVICE_CREATE_CALLBACK pCallback,
-    _In_opt_ PVOID pContext,
-    _Out_ PHSWDEVICE phSwDevice);
+    _In_  SW_DEVICE_CREATE_CALLBACK pCallback,
+    _In_opt_  PVOID pContext,
+    _Win32_RAIIFree_(SwDeviceClose, -1, 0) _Out_  PHSWDEVICE phSwDevice);
 
-VOID WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") VOID WINAPI
+ 
 SwDeviceClose(
     _In_ HSWDEVICE hSwDevice);
 
 #if (NTDDI_VERSION >= NTDDI_WINBLUE)
 
-_Check_return_ HRESULT WINAPI
+_Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows81_
 SwDeviceSetLifetime(
     _In_ HSWDEVICE hSwDevice,
     _In_ SW_DEVICE_LIFETIME Lifetime);
 
-_Check_return_ HRESULT WINAPI
+_Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows81_
 SwDeviceGetLifetime(
     _In_ HSWDEVICE hSwDevice,
     _Out_ PSW_DEVICE_LIFETIME pLifetime);
 
 #endif
 
-_Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+ 
 SwDevicePropertySet(
     _In_ HSWDEVICE hSwDevice,
     _In_ ULONG cPropertyCount,
     _In_reads_(cPropertyCount) const DEVPROPERTY *pProperties);
 
-_Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+ 
 SwDeviceInterfaceRegister(
     _In_ HSWDEVICE hSwDevice,
     _In_ const GUID *pInterfaceClassGuid,
@@ -81,19 +88,22 @@ SwDeviceInterfaceRegister(
     _In_ ULONG cPropertyCount,
     _In_reads_opt_(cPropertyCount) const DEVPROPERTY *pProperties,
     _In_ BOOL fEnabled,
-    _Outptr_result_maybenull_ PWSTR* ppszDeviceInterfaceId);
+    _Outptr_result_maybenull_  PWSTR* ppszDeviceInterfaceId);
 
-VOID WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") VOID WINAPI
+ 
 SwMemFree(
     _In_ PVOID pMem);
 
-_Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+ 
 SwDeviceInterfaceSetState(
     _In_ HSWDEVICE hSwDevice,
     _In_ PCWSTR pszDeviceInterfaceId,
     _In_ BOOL fEnabled);
 
-_Check_return_ HRESULT WINAPI
+_Windows_SupportedOS_Windows8_ _Win32_ImportLibrary_("cfgmgr32.dll") _Check_return_ HRESULT WINAPI
+ 
 SwDeviceInterfacePropertySet(
     _In_ HSWDEVICE hSwDevice,
     _In_ PCWSTR pszDeviceInterfaceId,
