diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h b/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/tbs.h
@@ -18,6 +18,7 @@ Contents:   Definitions and prototypes for the TBS (TPM Base Services) for
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family or SecureStartup Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_SECURESTARTUP)
@@ -88,14 +89,14 @@ typedef UINT32 TBS_HANDLE;
 #define TBS_OWNERAUTH_TYPE_ENDORSEMENT_20  12
 #define TBS_OWNERAUTH_TYPE_STORAGE_20      13
 
-typedef struct tdTBS_CONTEXT_PARAMS
+typedef struct _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ tdTBS_CONTEXT_PARAMS
 {
     UINT32 version;
 } TBS_CONTEXT_PARAMS, *PTBS_CONTEXT_PARAMS;
 typedef const TBS_CONTEXT_PARAMS *PCTBS_CONTEXT_PARAMS;
 
 TBS_RESULT WINAPI
-Tbsi_Context_Create(
+_Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ Tbsi_Context_Create(
     _In_ PCTBS_CONTEXT_PARAMS pContextParams,
     _Out_ PTBS_HCONTEXT phContext);
 
@@ -105,11 +106,11 @@ Tbsi_Tpm_Vendor_Maintenance_Mode(
     _Out_ PTBS_HCONTEXT phContext);
 
 TBS_RESULT WINAPI
-Tbsip_Context_Close(
+_Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ Tbsip_Context_Close(
     _In_ TBS_HCONTEXT   hContext);
 
 TBS_RESULT WINAPI
-Tbsip_Submit_Command(
+_Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ Tbsip_Submit_Command(
     _In_ TBS_HCONTEXT hContext,
     _In_ TBS_COMMAND_LOCALITY Locality,
     _In_ TBS_COMMAND_PRIORITY Priority,
@@ -119,11 +120,11 @@ Tbsip_Submit_Command(
     _Inout_ PUINT32 pcbResult);
 
 TBS_RESULT WINAPI
-Tbsip_Cancel_Commands(
+_Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ Tbsip_Cancel_Commands(
     _In_ TBS_HCONTEXT hContext);
 
 TBS_RESULT WINAPI
-Tbsi_Physical_Presence_Command(
+_Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ Tbsi_Physical_Presence_Command(
     _In_ TBS_HCONTEXT hContext,
     _In_reads_bytes_(cbInput) PCBYTE pabInput,
     _In_ UINT32 cbInput,
@@ -136,7 +137,7 @@ Tbsi_Physical_Presence_Command(
 
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
-Tbsi_Get_TCG_Log(
+_Windows_SupportedOS_WindowsVistaSP1_ _Windows_SupportedOS_Server2008_ Tbsi_Get_TCG_Log(
     _In_ TBS_HCONTEXT hContext,
     _Out_writes_bytes_to_opt_(*pOutputBufLen, *pOutputBufLen) PBYTE pOutputBuf,
     _Inout_ PUINT32 pOutputBufLen);
@@ -202,7 +203,7 @@ typedef struct tdTPM_WNF_PROVISIONING
 #define TPM_IFTYPE_EMULATOR       4 // 2.0: SW-emulator
 #define TPM_IFTYPE_SPB            5 // 2.0: SPB attached
 
-typedef struct _TPM_DEVICE_INFO
+typedef struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _TPM_DEVICE_INFO
 {
     UINT32      structVersion;      // = 1 for now
     UINT32      tpmVersion;         // 1.2 / 2.0
@@ -214,20 +215,20 @@ typedef const TPM_DEVICE_INFO *PCTPM_DEVICE_INFO;
 #endif //TPM_VERSION_UNKNOWN
 
 TBS_RESULT WINAPI
-Tbsi_GetDeviceInfo(
+_Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ Tbsi_GetDeviceInfo(
     _In_ UINT32 Size,
     _Out_writes_bytes_(Size) PVOID Info);
 
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
-Tbsi_Get_OwnerAuth(
+_Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ Tbsi_Get_OwnerAuth(
     _In_ TBS_HCONTEXT hContext,
     _In_ TBS_OWNERAUTH_TYPE ownerauthType,
     _Out_writes_bytes_to_opt_(*pOutputBufLen, *pOutputBufLen) PBYTE pOutputBuf,
     _Inout_  PUINT32 pOutputBufLen);
 
 TBS_RESULT WINAPI
-Tbsi_Revoke_Attestation();
+_Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ Tbsi_Revoke_Attestation();
 
 #endif // (NTDDI_VERSION >= NTDDI_WIN8)
 
@@ -284,7 +285,7 @@ Tbsi_Create_Windows_Key(
 
 _Success_(return == TBS_SUCCESS)
 TBS_RESULT WINAPI
-Tbsi_Get_TCG_Log_Ex(
+_Windows_SupportedOS_17134_ Tbsi_Get_TCG_Log_Ex(
     _In_ UINT32 logType,
     _Out_writes_bytes_opt_(*pcbOutput) PBYTE pbOutput,
     _Inout_ PUINT32 pcbOutput);
