diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/thumbnailstreamcache.h b/generation/WinSDK/RecompiledIdlHeaders/um/thumbnailstreamcache.h
index a608750572..d97d124e3b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/thumbnailstreamcache.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/thumbnailstreamcache.h
@@ -80,10 +80,11 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region App Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 typedef /* [v1_enum] */ 
-enum ThumbnailStreamCacheOptions
+enum   ThumbnailStreamCacheOptions
     {
         ExtractIfNotCached	= 0,
         ReturnOnlyIfCached	= 0x1,
@@ -108,8 +109,8 @@ EXTERN_C const IID IID_IThumbnailStreamCache;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("90E11430-9569-41D8-AE75-6D4D2AE7CCA0")
-    IThumbnailStreamCache : public IUnknown
+    MIDL_INTERFACE("90E11430-9569-41D8-AE75-6D4D2AE7CCA0") _Windows_SupportedOS_10240_ 
+      IThumbnailStreamCache : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetThumbnailStream( 
@@ -192,7 +193,7 @@ EXTERN_C const IID IID_IThumbnailStreamCache;
         END_INTERFACE
     } IThumbnailStreamCacheVtbl;
 
-    interface IThumbnailStreamCache
+    interface _Windows_SupportedOS_10240_ IThumbnailStreamCache
     {
         CONST_VTBL struct IThumbnailStreamCacheVtbl *lpVtbl;
     };
