diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/thumbcache.h b/generation/WinSDK/RecompiledIdlHeaders/um/thumbcache.h
index ab597d65987785b43c12082ea15c3fa48a649d8d..70f730ef8a2fb9f43f473c3dc1315f9e414b6c72 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/thumbcache.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/thumbcache.h
@@ -120,10 +120,11 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 typedef /* [v1_enum] */ 
-enum WTS_FLAGS
+enum _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ WTS_FLAGS
     {
         WTS_NONE	= 0,
         WTS_EXTRACT	= 0,
@@ -155,7 +156,7 @@ enum WTS_CACHEFLAGS
 
 DEFINE_ENUM_FLAG_OPERATORS(WTS_CACHEFLAGS)
 typedef /* [v1_enum] */ 
-enum WTS_CONTEXTFLAGS
+enum _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ WTS_CONTEXTFLAGS
     {
         WTSCF_DEFAULT	= 0,
         WTSCF_APPSTYLE	= 0x1,
@@ -173,7 +174,7 @@ enum WTS_ALPHATYPE
         WTSAT_ARGB	= 2
     } 	WTS_ALPHATYPE;
 
-typedef struct WTS_THUMBNAILID
+typedef struct _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ WTS_THUMBNAILID
     {
     BYTE rgbKey[ 16 ];
     } 	WTS_THUMBNAILID;
@@ -203,7 +204,7 @@ EXTERN_C const IID IID_ISharedBitmap;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("091162a4-bc96-411f-aae8-c5122cd03363")
-    ISharedBitmap : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ ISharedBitmap : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetSharedBitmap( 
@@ -288,7 +289,7 @@ EXTERN_C const IID IID_ISharedBitmap;
         END_INTERFACE
     } ISharedBitmapVtbl;
 
-    interface ISharedBitmap
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ ISharedBitmap
     {
         CONST_VTBL struct ISharedBitmapVtbl *lpVtbl;
     };
@@ -346,7 +347,7 @@ EXTERN_C const IID IID_IThumbnailCache;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("F676C15D-596A-4ce2-8234-33996F445DB1")
-    IThumbnailCache : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IThumbnailCache : public IUnknown
     {
     public:
         virtual /* [local] */ HRESULT STDMETHODCALLTYPE GetThumbnail( 
@@ -429,7 +430,7 @@ EXTERN_C const IID IID_IThumbnailCache;
         END_INTERFACE
     } IThumbnailCacheVtbl;
 
-    interface IThumbnailCache
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IThumbnailCache
     {
         CONST_VTBL struct IThumbnailCacheVtbl *lpVtbl;
     };
@@ -520,7 +521,7 @@ EXTERN_C const IID IID_IThumbnailProvider;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("e357fccd-a995-4576-b01f-234630154e96")
-    IThumbnailProvider : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IThumbnailProvider : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetThumbnail( 
@@ -569,7 +570,7 @@ EXTERN_C const IID IID_IThumbnailProvider;
         END_INTERFACE
     } IThumbnailProviderVtbl;
 
-    interface IThumbnailProvider
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IThumbnailProvider
     {
         CONST_VTBL struct IThumbnailProviderVtbl *lpVtbl;
     };
@@ -615,7 +616,7 @@ EXTERN_C const IID IID_IThumbnailSettings;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("F4376F00-BEF5-4d45-80F3-1E023BBF1209")
-    IThumbnailSettings : public IUnknown
+    _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ IThumbnailSettings : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE SetContext( 
@@ -656,7 +657,7 @@ EXTERN_C const IID IID_IThumbnailSettings;
         END_INTERFACE
     } IThumbnailSettingsVtbl;
 
-    interface IThumbnailSettings
+    interface _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ IThumbnailSettings
     {
         CONST_VTBL struct IThumbnailSettingsVtbl *lpVtbl;
     };
@@ -702,7 +703,7 @@ EXTERN_C const IID IID_IThumbnailCachePrimer;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("0f03f8fe-2b26-46f0-965a-212aa8d66b76")
-    IThumbnailCachePrimer : public IUnknown
+    _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ IThumbnailCachePrimer : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE PageInThumbnail( 
@@ -751,7 +752,7 @@ EXTERN_C const IID IID_IThumbnailCachePrimer;
         END_INTERFACE
     } IThumbnailCachePrimerVtbl;
 
-    interface IThumbnailCachePrimer
+    interface _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ IThumbnailCachePrimer
     {
         CONST_VTBL struct IThumbnailCachePrimerVtbl *lpVtbl;
     };
