diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/ual.h b/generation/WinSDK/RecompiledIdlHeaders/um/ual.h
index c54fe7f617..c397f53bd7 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/ual.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/ual.h
@@ -10,6 +10,9 @@
 
 #include <winapifamily.h>
 #include <winsock2.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 #if defined (_MSC_VER)
 #pragma once
@@ -33,21 +36,25 @@ typedef struct tagUAL_DATA_BLOB
     WCHAR               UserName[MAX_PATH];
 } UAL_DATA_BLOB, *PUAL_DATA_BLOB;
 
+_Windows_SupportedOS_Windows8_
 HRESULT
 WINAPI
 UalStart(
     _In_ PUAL_DATA_BLOB Data);
 
+_Windows_SupportedOS_Windows8_
 HRESULT
 WINAPI
 UalStop(
     _In_ PUAL_DATA_BLOB Data);
     
+_Windows_SupportedOS_Windows8_
 HRESULT
 WINAPI
 UalInstrument(
     _In_ PUAL_DATA_BLOB Data);
     
+_Windows_SupportedOS_Windows8_
 HRESULT
 WINAPI
 UalRegisterProduct(
