diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UIAnimation.h b/generation/WinSDK/RecompiledIdlHeaders/um/UIAnimation.h
index c064cba291..7f25014584 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UIAnimation.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UIAnimation.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -454,7 +455,7 @@ EXTERN_C const IID IID_IUIAnimationManager;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("9169896C-AC8D-4e7d-94E5-67FA4DC2F2E8")
+    MIDL_INTERFACE("9169896C-AC8D-4e7d-94E5-67FA4DC2F2E8") _Windows_SupportedOS_Windows7_ 
     IUIAnimationManager : public IUnknown
     {
     public:
@@ -883,7 +884,7 @@ EXTERN_C const IID IID_IUIAnimationVariable;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("8CEEB155-2849-4ce5-9448-91FF70E1E4D9")
+    MIDL_INTERFACE("8CEEB155-2849-4ce5-9448-91FF70E1E4D9") _Windows_SupportedOS_Windows7_ 
     IUIAnimationVariable : public IUnknown
     {
     public:
@@ -1247,7 +1248,7 @@ EXTERN_C const IID IID_IUIAnimationStoryboard;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("A8FF128F-9BF9-4af1-9E67-E5E410DEFB84")
+    MIDL_INTERFACE("A8FF128F-9BF9-4af1-9E67-E5E410DEFB84") _Windows_SupportedOS_Windows7_ 
     IUIAnimationStoryboard : public IUnknown
     {
     public:
@@ -1658,7 +1659,7 @@ EXTERN_C const IID IID_IUIAnimationTransition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("DC6CE252-F731-41cf-B610-614B6CA049AD")
+    MIDL_INTERFACE("DC6CE252-F731-41cf-B610-614B6CA049AD") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTransition : public IUnknown
     {
     public:
@@ -1796,7 +1797,7 @@ EXTERN_C const IID IID_IUIAnimationManagerEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("783321ED-78A3-4366-B574-6AF607A64788")
+    MIDL_INTERFACE("783321ED-78A3-4366-B574-6AF607A64788") _Windows_SupportedOS_Windows7_ 
     IUIAnimationManagerEventHandler : public IUnknown
     {
     public:
@@ -1891,7 +1892,7 @@ EXTERN_C const IID IID_IUIAnimationVariableChangeHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("6358B7BA-87D2-42d5-BF71-82E919DD5862")
+    MIDL_INTERFACE("6358B7BA-87D2-42d5-BF71-82E919DD5862") _Windows_SupportedOS_Windows7_ 
     IUIAnimationVariableChangeHandler : public IUnknown
     {
     public:
@@ -1994,7 +1995,7 @@ EXTERN_C const IID IID_IUIAnimationVariableIntegerChangeHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("BB3E1550-356E-44b0-99DA-85AC6017865E")
+    MIDL_INTERFACE("BB3E1550-356E-44b0-99DA-85AC6017865E") _Windows_SupportedOS_Windows7_ 
     IUIAnimationVariableIntegerChangeHandler : public IUnknown
     {
     public:
@@ -2097,7 +2098,7 @@ EXTERN_C const IID IID_IUIAnimationStoryboardEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("3D5C9008-EC7C-4364-9F8A-9AF3C58CBAE6")
+    MIDL_INTERFACE("3D5C9008-EC7C-4364-9F8A-9AF3C58CBAE6") _Windows_SupportedOS_Windows7_ 
     IUIAnimationStoryboardEventHandler : public IUnknown
     {
     public:
@@ -2228,7 +2229,7 @@ EXTERN_C const IID IID_IUIAnimationPriorityComparison;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("83FA9B74-5F86-4618-BC6A-A2FAC19B3F44")
+    MIDL_INTERFACE("83FA9B74-5F86-4618-BC6A-A2FAC19B3F44") _Windows_SupportedOS_Windows7_ 
     IUIAnimationPriorityComparison : public IUnknown
     {
     public:
@@ -2342,7 +2343,7 @@ EXTERN_C const IID IID_IUIAnimationTransitionLibrary;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("CA5A14B1-D24F-48b8-8FE4-C78169BA954E")
+    MIDL_INTERFACE("CA5A14B1-D24F-48b8-8FE4-C78169BA954E") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTransitionLibrary : public IUnknown
     {
     public:
@@ -2755,7 +2756,7 @@ EXTERN_C const IID IID_IUIAnimationInterpolator;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("7815CBBA-DDF7-478c-A46C-7B6C738B7978")
+    MIDL_INTERFACE("7815CBBA-DDF7-478c-A46C-7B6C738B7978") _Windows_SupportedOS_Windows7_ 
     IUIAnimationInterpolator : public IUnknown
     {
     public:
@@ -2968,7 +2969,7 @@ EXTERN_C const IID IID_IUIAnimationTransitionFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("FCD91E03-3E3B-45ad-BBB1-6DFC8153743D")
+    MIDL_INTERFACE("FCD91E03-3E3B-45ad-BBB1-6DFC8153743D") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTransitionFactory : public IUnknown
     {
     public:
@@ -3078,7 +3079,7 @@ EXTERN_C const IID IID_IUIAnimationTimer;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("6B0EFAD1-A053-41d6-9085-33A689144665")
+    MIDL_INTERFACE("6B0EFAD1-A053-41d6-9085-33A689144665") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTimer : public IUnknown
     {
     public:
@@ -3263,7 +3264,7 @@ EXTERN_C const IID IID_IUIAnimationTimerUpdateHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("195509B7-5D5E-4e3e-B278-EE3759B367AD")
+    MIDL_INTERFACE("195509B7-5D5E-4e3e-B278-EE3759B367AD") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTimerUpdateHandler : public IUnknown
     {
     public:
@@ -3403,7 +3404,7 @@ EXTERN_C const IID IID_IUIAnimationTimerClientEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("BEDB4DB6-94FA-4bfb-A47F-EF2D9E408C25")
+    MIDL_INTERFACE("BEDB4DB6-94FA-4bfb-A47F-EF2D9E408C25") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTimerClientEventHandler : public IUnknown
     {
     public:
@@ -3498,7 +3499,7 @@ EXTERN_C const IID IID_IUIAnimationTimerEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("274A7DEA-D771-4095-ABBD-8DF7ABD23CE3")
+    MIDL_INTERFACE("274A7DEA-D771-4095-ABBD-8DF7ABD23CE3") _Windows_SupportedOS_Windows7_ 
     IUIAnimationTimerEventHandler : public IUnknown
     {
     public:
@@ -3625,7 +3626,7 @@ EXTERN_C const IID IID_IUIAnimationManager2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("D8B6F7D4-4109-4d3f-ACEE-879926968CB1")
+    MIDL_INTERFACE("D8B6F7D4-4109-4d3f-ACEE-879926968CB1") _Windows_SupportedOS_Windows8_ 
     IUIAnimationManager2 : public IUnknown
     {
     public:
@@ -4084,7 +4085,7 @@ EXTERN_C const IID IID_IUIAnimationVariable2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("4914B304-96AB-44d9-9E77-D5109B7E7466")
+    MIDL_INTERFACE("4914B304-96AB-44d9-9E77-D5109B7E7466") _Windows_SupportedOS_Windows8_ 
     IUIAnimationVariable2 : public IUnknown
     {
     public:
@@ -4662,7 +4663,7 @@ EXTERN_C const IID IID_IUIAnimationTransition2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("62FF9123-A85A-4e9b-A218-435A93E268FD")
+    MIDL_INTERFACE("62FF9123-A85A-4e9b-A218-435A93E268FD") _Windows_SupportedOS_Windows8_ 
     IUIAnimationTransition2 : public IUnknown
     {
     public:
@@ -4859,7 +4860,7 @@ EXTERN_C const IID IID_IUIAnimationManagerEventHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("F6E022BA-BFF3-42EC-9033-E073F33E83C3")
+    MIDL_INTERFACE("F6E022BA-BFF3-42EC-9033-E073F33E83C3") _Windows_SupportedOS_Windows8_ 
     IUIAnimationManagerEventHandler2 : public IUnknown
     {
     public:
@@ -4954,7 +4955,7 @@ EXTERN_C const IID IID_IUIAnimationVariableChangeHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("63ACC8D2-6EAE-4bb0-B879-586DD8CFBE42")
+    MIDL_INTERFACE("63ACC8D2-6EAE-4bb0-B879-586DD8CFBE42") _Windows_SupportedOS_Windows8_ 
     IUIAnimationVariableChangeHandler2 : public IUnknown
     {
     public:
@@ -5061,7 +5062,7 @@ EXTERN_C const IID IID_IUIAnimationVariableIntegerChangeHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("829B6CF1-4F3A-4412-AE09-B243EB4C6B58")
+    MIDL_INTERFACE("829B6CF1-4F3A-4412-AE09-B243EB4C6B58") _Windows_SupportedOS_Windows8_ 
     IUIAnimationVariableIntegerChangeHandler2 : public IUnknown
     {
     public:
@@ -5168,7 +5169,7 @@ EXTERN_C const IID IID_IUIAnimationVariableCurveChangeHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("72895E91-0145-4C21-9192-5AAB40EDDF80")
+    MIDL_INTERFACE("72895E91-0145-4C21-9192-5AAB40EDDF80") _Windows_SupportedOS_Windows8_ 
     IUIAnimationVariableCurveChangeHandler2 : public IUnknown
     {
     public:
@@ -5259,7 +5260,7 @@ EXTERN_C const IID IID_IUIAnimationStoryboardEventHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("BAC5F55A-BA7C-414C-B599-FBF850F553C6")
+    MIDL_INTERFACE("BAC5F55A-BA7C-414C-B599-FBF850F553C6") _Windows_SupportedOS_Windows8_ 
     IUIAnimationStoryboardEventHandler2 : public IUnknown
     {
     public:
@@ -5385,7 +5386,7 @@ EXTERN_C const IID IID_IUIAnimationLoopIterationChangeHandler2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("2D3B15A4-4762-47AB-A030-B23221DF3AE0")
+    MIDL_INTERFACE("2D3B15A4-4762-47AB-A030-B23221DF3AE0") _Windows_SupportedOS_Windows8_ 
     IUIAnimationLoopIterationChangeHandler2 : public IUnknown
     {
     public:
@@ -5488,7 +5489,7 @@ EXTERN_C const IID IID_IUIAnimationPriorityComparison2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("5B6D7A37-4621-467C-8B05-70131DE62DDB")
+    MIDL_INTERFACE("5B6D7A37-4621-467C-8B05-70131DE62DDB") _Windows_SupportedOS_Windows8_ 
     IUIAnimationPriorityComparison2 : public IUnknown
     {
     public:
@@ -5587,7 +5588,7 @@ EXTERN_C const IID IID_IUIAnimationTransitionLibrary2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("03CFAE53-9580-4ee3-B363-2ECE51B4AF6A")
+    MIDL_INTERFACE("03CFAE53-9580-4ee3-B363-2ECE51B4AF6A") _Windows_SupportedOS_Windows8_ 
     IUIAnimationTransitionLibrary2 : public IUnknown
     {
     public:
@@ -6216,7 +6217,7 @@ EXTERN_C const IID IID_IUIAnimationPrimitiveInterpolation;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("BAB20D63-4361-45DA-A24F-AB8508846B5B")
+    MIDL_INTERFACE("BAB20D63-4361-45DA-A24F-AB8508846B5B") _Windows_SupportedOS_Windows8_ 
     IUIAnimationPrimitiveInterpolation : public IUnknown
     {
     public:
@@ -6360,7 +6361,7 @@ EXTERN_C const IID IID_IUIAnimationInterpolator2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("EA76AFF8-EA22-4a23-A0EF-A6A966703518")
+    MIDL_INTERFACE("EA76AFF8-EA22-4a23-A0EF-A6A966703518") _Windows_SupportedOS_Windows8_ 
     IUIAnimationInterpolator2 : public IUnknown
     {
     public:
@@ -6627,7 +6628,7 @@ EXTERN_C const IID IID_IUIAnimationTransitionFactory2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("937D4916-C1A6-42d5-88D8-30344D6EFE31")
+    MIDL_INTERFACE("937D4916-C1A6-42d5-88D8-30344D6EFE31") _Windows_SupportedOS_Windows8_ 
     IUIAnimationTransitionFactory2 : public IUnknown
     {
     public:
