diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCoreApi.h b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCoreApi.h
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCoreApi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCoreApi.h
@@ -16,6 +16,7 @@ extern "C" {
 #endif
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
@@ -476,7 +477,7 @@ DEFINE_GUID(IsDialog_Property_GUID,                                     0x9d0dfb
 //
 // --------------------------------------------------------------------------
 
-BOOL WINAPI UiaGetErrorDescription(BSTR * pDescription);
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetErrorDescription(BSTR * pDescription);
 
 // --------------------------------------------------------------------------
 //
@@ -484,7 +485,7 @@ BOOL WINAPI UiaGetErrorDescription(BSTR * pDescription);
 //
 // --------------------------------------------------------------------------
 
-enum ConditionType
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ConditionType
 {
     ConditionType_True = 0,
     ConditionType_False = 1,
@@ -494,7 +495,7 @@ enum ConditionType
     ConditionType_Not = 5,
 };
 
-struct UiaCondition
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaCondition
 {
     enum ConditionType ConditionType;
 };
@@ -508,7 +509,7 @@ enum PropertyConditionFlags
 };
 #endif
 
-struct UiaPropertyCondition
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaPropertyCondition
 {
     enum ConditionType ConditionType;
     PROPERTYID PropertyId;
@@ -516,14 +517,14 @@ struct UiaPropertyCondition
     enum PropertyConditionFlags Flags;
 };
 
-struct UiaAndOrCondition
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaAndOrCondition
 {
     enum ConditionType ConditionType;
     struct UiaCondition ** ppConditions; // ptr to array-of-ptrs to conditions
     int cConditions;
 };
 
-struct UiaNotCondition
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNotCondition
 {
     enum ConditionType ConditionType;
     struct UiaCondition * pCondition;
@@ -544,7 +545,7 @@ enum AutomationElementMode
 };
 #endif
 
-struct UiaCacheRequest
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaCacheRequest
 {
     struct UiaCondition *  pViewCondition;
     enum TreeScope  Scope;
@@ -556,9 +557,9 @@ struct UiaCacheRequest
     enum AutomationElementMode automationElementMode;
 };
 
-HRESULT WINAPI UiaHUiaNodeFromVariant(VARIANT * pvar, HUIANODE *phnode);
-HRESULT WINAPI UiaHPatternObjectFromVariant(VARIANT * pvar, HUIAPATTERNOBJECT *phobj);
-HRESULT WINAPI UiaHTextRangeFromVariant(VARIANT * pvar, HUIATEXTRANGE *phtextrange);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaHUiaNodeFromVariant(VARIANT * pvar, HUIANODE *phnode);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaHPatternObjectFromVariant(VARIANT * pvar, HUIAPATTERNOBJECT *phobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaHTextRangeFromVariant(VARIANT * pvar, HUIATEXTRANGE *phtextrange);
 
 // --------------------------------------------------------------------------
 //
@@ -567,7 +568,7 @@ HRESULT WINAPI UiaHTextRangeFromVariant(VARIANT * pvar, HUIATEXTRANGE *phtextran
 // --------------------------------------------------------------------------
 
 // Used by UiaGetUpdatedCache
-enum NormalizeState
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ NormalizeState
 {
     NormalizeState_None,    // Don't normalize
     NormalizeState_View,    // Normalize against condition in UiaCacheRequest
@@ -597,7 +598,7 @@ enum TreeTraversalOptions
 };
 #endif  // __uiautomationclient_h__
 
-struct UiaFindParams
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaFindParams
 {
     int MaxDepth;
     BOOL FindFirst;
@@ -605,21 +606,21 @@ struct UiaFindParams
     struct UiaCondition * pFindCondition;
 };
 
-BOOL WINAPI UiaNodeRelease(HUIANODE hnode);
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNodeRelease(HUIANODE hnode);
 
-HRESULT WINAPI UiaGetPropertyValue(HUIANODE hnode, PROPERTYID propertyId, VARIANT * pValue);
-HRESULT WINAPI UiaGetPatternProvider(HUIANODE hnode, PATTERNID patternId, HUIAPATTERNOBJECT * phobj);
-HRESULT WINAPI UiaGetRuntimeId(HUIANODE hnode, SAFEARRAY ** pruntimeId);
-HRESULT WINAPI UiaSetFocus(HUIANODE hnode);
-HRESULT WINAPI UiaNavigate(HUIANODE hnode, enum NavigateDirection direction, struct UiaCondition * pCondition, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
-HRESULT WINAPI UiaGetUpdatedCache(HUIANODE hnode, struct UiaCacheRequest * pRequest, enum NormalizeState normalizeState, struct UiaCondition * pNormalizeCondition, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
-HRESULT WINAPI UiaFind(HUIANODE hnode, struct UiaFindParams * pParams, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, SAFEARRAY ** ppOffsets, SAFEARRAY ** ppTreeStructures);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetPropertyValue(HUIANODE hnode, PROPERTYID propertyId, VARIANT * pValue);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetPatternProvider(HUIANODE hnode, PATTERNID patternId, HUIAPATTERNOBJECT * phobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetRuntimeId(HUIANODE hnode, SAFEARRAY ** pruntimeId);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaSetFocus(HUIANODE hnode);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNavigate(HUIANODE hnode, enum NavigateDirection direction, struct UiaCondition * pCondition, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetUpdatedCache(HUIANODE hnode, struct UiaCacheRequest * pRequest, enum NormalizeState normalizeState, struct UiaCondition * pNormalizeCondition, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaFind(HUIANODE hnode, struct UiaFindParams * pParams, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, SAFEARRAY ** ppOffsets, SAFEARRAY ** ppTreeStructures);
 
-HRESULT WINAPI UiaNodeFromPoint(double x, double y, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
-HRESULT WINAPI UiaNodeFromFocus(struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
-HRESULT WINAPI UiaNodeFromHandle(HWND hwnd, HUIANODE * phnode);
-HRESULT WINAPI UiaNodeFromProvider(IRawElementProviderSimple * pProvider, HUIANODE * phnode);
-HRESULT WINAPI UiaGetRootNode(HUIANODE * phnode);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNodeFromPoint(double x, double y, struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNodeFromFocus(struct UiaCacheRequest * pRequest, SAFEARRAY ** ppRequestedData, BSTR * ppTreeStructure);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNodeFromHandle(HWND hwnd, HUIANODE * phnode);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaNodeFromProvider(IRawElementProviderSimple * pProvider, HUIANODE * phnode);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetRootNode(HUIANODE * phnode);
 
 
 #ifdef __cplusplus
@@ -643,7 +644,7 @@ inline bool UiaRectIsEmpty(const UiaRect & rc)
 //
 // --------------------------------------------------------------------------
 
-enum ProviderType
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ProviderType
 {
     ProviderType_BaseHwnd,
     ProviderType_Proxy,
@@ -652,7 +653,7 @@ enum ProviderType
 
 typedef SAFEARRAY * WINAPI UiaProviderCallback(HWND hwnd, enum ProviderType providerType);
 
-void WINAPI UiaRegisterProviderCallback(UiaProviderCallback * pCallback);
+void WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRegisterProviderCallback(UiaProviderCallback * pCallback);
 
 
 // --------------------------------------------------------------------------
@@ -661,7 +662,7 @@ void WINAPI UiaRegisterProviderCallback(UiaProviderCallback * pCallback);
 //
 // --------------------------------------------------------------------------
 
-enum AutomationIdentifierType
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ AutomationIdentifierType
 {
     AutomationIdentifierType_Property,
     AutomationIdentifierType_Pattern,
@@ -674,7 +675,7 @@ enum AutomationIdentifierType
     AutomationIdentifierType_Style,
 };
 
-int WINAPI UiaLookupId(enum AutomationIdentifierType type, const GUID* pGuid);
+int WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaLookupId(enum AutomationIdentifierType type, const GUID* pGuid);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
@@ -682,8 +683,8 @@ int WINAPI UiaLookupId(enum AutomationIdentifierType type, const GUID* pGuid);
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
-HRESULT WINAPI UiaGetReservedNotSupportedValue(IUnknown **punkNotSupportedValue);
-HRESULT WINAPI UiaGetReservedMixedAttributeValue(IUnknown **punkMixedAttributeValue);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetReservedNotSupportedValue(IUnknown **punkNotSupportedValue);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaGetReservedMixedAttributeValue(IUnknown **punkMixedAttributeValue);
 
 // --------------------------------------------------------------------------
 //
@@ -695,7 +696,7 @@ HRESULT WINAPI UiaGetReservedMixedAttributeValue(IUnknown **punkMixedAttributeVa
 // Event structs and enums
 //
 
-enum EventArgsType
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ EventArgsType
 {
     EventArgsType_Simple,
     EventArgsType_PropertyChanged,
@@ -709,20 +710,20 @@ enum EventArgsType
     EventArgsType_StructuredMarkup,
 };
 
-enum AsyncContentLoadedState
+enum _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ AsyncContentLoadedState
 {
     AsyncContentLoadedState_Beginning,
     AsyncContentLoadedState_Progress,
     AsyncContentLoadedState_Completed
 };
 
-struct UiaEventArgs
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaEventArgs
 {
     enum EventArgsType Type;
     int EventId;
 };
 
-struct UiaPropertyChangedEventArgs
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaPropertyChangedEventArgs
 {
     enum EventArgsType Type;
     int EventId;
@@ -731,7 +732,7 @@ struct UiaPropertyChangedEventArgs
     VARIANT NewValue;
 };
 
-struct UiaStructureChangedEventArgs
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaStructureChangedEventArgs
 {
     enum EventArgsType Type;
     int EventId;
@@ -755,7 +756,7 @@ struct UiaChangesEventArgs {
     struct UiaChangeInfo * pUiaChanges;
 };
 
-struct UiaAsyncContentLoadedEventArgs
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaAsyncContentLoadedEventArgs
 {
     enum EventArgsType Type;
     int EventId;
@@ -763,7 +764,7 @@ struct UiaAsyncContentLoadedEventArgs
     double PercentComplete;
 };
 
-struct UiaWindowClosedEventArgs
+struct _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaWindowClosedEventArgs
 {
     enum EventArgsType Type;
     int EventId;
@@ -774,20 +775,20 @@ struct UiaWindowClosedEventArgs
 //
 // Provider Event APIs
 //
-BOOL WINAPI UiaClientsAreListening();
-HRESULT WINAPI UiaRaiseAutomationPropertyChangedEvent(IRawElementProviderSimple * pProvider, PROPERTYID id, VARIANT oldValue, VARIANT newValue);
-HRESULT WINAPI UiaRaiseAutomationEvent(IRawElementProviderSimple * pProvider, EVENTID id);
-HRESULT WINAPI UiaRaiseStructureChangedEvent(IRawElementProviderSimple * pProvider, enum StructureChangeType structureChangeType, int * pRuntimeId, int cRuntimeIdLen);
-HRESULT WINAPI UiaRaiseAsyncContentLoadedEvent(IRawElementProviderSimple * pProvider, enum AsyncContentLoadedState asyncContentLoadedState, double percentComplete);
-HRESULT WINAPI UiaRaiseTextEditTextChangedEvent(IRawElementProviderSimple * pProvider, enum TextEditChangeType textEditChangeType, SAFEARRAY *pChangedData);
-HRESULT WINAPI UiaRaiseChangesEvent(IRawElementProviderSimple * pProvider, int eventIdCount, struct UiaChangeInfo * pUiaChanges);
-HRESULT WINAPI UiaRaiseNotificationEvent(
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaClientsAreListening();
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRaiseAutomationPropertyChangedEvent(IRawElementProviderSimple * pProvider, PROPERTYID id, VARIANT oldValue, VARIANT newValue);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRaiseAutomationEvent(IRawElementProviderSimple * pProvider, EVENTID id);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRaiseStructureChangedEvent(IRawElementProviderSimple * pProvider, enum StructureChangeType structureChangeType, int * pRuntimeId, int cRuntimeIdLen);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRaiseAsyncContentLoadedEvent(IRawElementProviderSimple * pProvider, enum AsyncContentLoadedState asyncContentLoadedState, double percentComplete);
+HRESULT WINAPI _Windows_SupportedOS_Windows81_ UiaRaiseTextEditTextChangedEvent(IRawElementProviderSimple * pProvider, enum TextEditChangeType textEditChangeType, SAFEARRAY *pChangedData);
+HRESULT WINAPI _Windows_SupportedOS_10240_ _Windows_SupportedOS_Server2016_ UiaRaiseChangesEvent(IRawElementProviderSimple * pProvider, int eventIdCount, struct UiaChangeInfo * pUiaChanges);
+HRESULT WINAPI _Windows_SupportedOS_16299_ _Windows_SupportedOS_Server2016_ UiaRaiseNotificationEvent(
     _In_ IRawElementProviderSimple* provider,
     enum NotificationKind notificationKind,
     enum NotificationProcessing notificationProcessing,
     _In_opt_ BSTR displayString,
     _In_ BSTR activityId);
-HRESULT WINAPI UiaRaiseActiveTextPositionChangedEvent(
+HRESULT WINAPI _Windows_SupportedOS_Windows81_ UiaRaiseActiveTextPositionChangedEvent(
     _In_ IRawElementProviderSimple* provider,
     _In_opt_ ITextRangeProvider* textRange);
 
@@ -801,10 +802,10 @@ HRESULT WINAPI UiaRaiseActiveTextPositionChangedEvent(
 // Client Event APIs
 //
 typedef void WINAPI UiaEventCallback(struct UiaEventArgs * pArgs, SAFEARRAY * pRequestedData, BSTR pTreeStructure);
-HRESULT WINAPI UiaAddEvent(HUIANODE hnode, EVENTID eventId, UiaEventCallback * pCallback, enum TreeScope scope, PROPERTYID * pProperties, int cProperties, struct UiaCacheRequest * pRequest, HUIAEVENT * phEvent);
-HRESULT WINAPI UiaRemoveEvent(HUIAEVENT hEvent);
-HRESULT WINAPI UiaEventAddWindow(HUIAEVENT hEvent, HWND hwnd);
-HRESULT WINAPI UiaEventRemoveWindow(HUIAEVENT hEvent, HWND hwnd);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaAddEvent(HUIANODE hnode, EVENTID eventId, UiaEventCallback * pCallback, enum TreeScope scope, PROPERTYID * pProperties, int cProperties, struct UiaCacheRequest * pRequest, HUIAEVENT * phEvent);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaRemoveEvent(HUIAEVENT hEvent);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaEventAddWindow(HUIAEVENT hEvent, HWND hwnd);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaEventRemoveWindow(HUIAEVENT hEvent, HWND hwnd);
 
 
 // --------------------------------------------------------------------------
@@ -813,53 +814,53 @@ HRESULT WINAPI UiaEventRemoveWindow(HUIAEVENT hEvent, HWND hwnd);
 //
 // --------------------------------------------------------------------------
 
-HRESULT WINAPI DockPattern_SetDockPosition(HUIAPATTERNOBJECT hobj, enum DockPosition dockPosition);
-HRESULT WINAPI ExpandCollapsePattern_Collapse(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI ExpandCollapsePattern_Expand(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI GridPattern_GetItem(HUIAPATTERNOBJECT hobj, int row, int column, HUIANODE * pResult);
-HRESULT WINAPI InvokePattern_Invoke(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI MultipleViewPattern_GetViewName(HUIAPATTERNOBJECT hobj, int viewId, BSTR * ppStr);
-HRESULT WINAPI MultipleViewPattern_SetCurrentView(HUIAPATTERNOBJECT hobj, int viewId);
-HRESULT WINAPI RangeValuePattern_SetValue(HUIAPATTERNOBJECT hobj, double val);
-HRESULT WINAPI ScrollItemPattern_ScrollIntoView(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI ScrollPattern_Scroll(HUIAPATTERNOBJECT hobj, enum ScrollAmount horizontalAmount, enum ScrollAmount verticalAmount);
-HRESULT WINAPI ScrollPattern_SetScrollPercent(HUIAPATTERNOBJECT hobj, double horizontalPercent, double verticalPercent);
-HRESULT WINAPI SelectionItemPattern_AddToSelection(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI SelectionItemPattern_RemoveFromSelection(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI SelectionItemPattern_Select(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI TogglePattern_Toggle(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI TransformPattern_Move(HUIAPATTERNOBJECT hobj, double x, double y);
-HRESULT WINAPI TransformPattern_Resize(HUIAPATTERNOBJECT hobj, double width, double height);
-HRESULT WINAPI TransformPattern_Rotate(HUIAPATTERNOBJECT hobj, double degrees);
-HRESULT WINAPI ValuePattern_SetValue(HUIAPATTERNOBJECT hobj, LPCWSTR pVal);
-HRESULT WINAPI WindowPattern_Close(HUIAPATTERNOBJECT hobj);
-HRESULT WINAPI WindowPattern_SetWindowVisualState(HUIAPATTERNOBJECT hobj, enum WindowVisualState state);
-HRESULT WINAPI WindowPattern_WaitForInputIdle(HUIAPATTERNOBJECT hobj, int milliseconds, BOOL * pResult);
-
-HRESULT WINAPI TextPattern_GetSelection(HUIAPATTERNOBJECT hobj, SAFEARRAY** pRetVal);
-HRESULT WINAPI TextPattern_GetVisibleRanges(HUIAPATTERNOBJECT hobj, SAFEARRAY** pRetVal);
-HRESULT WINAPI TextPattern_RangeFromChild(HUIAPATTERNOBJECT hobj, HUIANODE hnodeChild, HUIATEXTRANGE* pRetVal);
-HRESULT WINAPI TextPattern_RangeFromPoint(HUIAPATTERNOBJECT hobj, struct UiaPoint point, HUIATEXTRANGE* pRetVal);
-HRESULT WINAPI TextPattern_get_DocumentRange(HUIAPATTERNOBJECT hobj, HUIATEXTRANGE* pRetVal);
-HRESULT WINAPI TextPattern_get_SupportedTextSelection(HUIAPATTERNOBJECT hobj, enum SupportedTextSelection* pRetVal);
-HRESULT WINAPI TextRange_Clone(HUIATEXTRANGE hobj, HUIATEXTRANGE* pRetVal);
-HRESULT WINAPI TextRange_Compare(HUIATEXTRANGE hobj, HUIATEXTRANGE range, BOOL* pRetVal);
-HRESULT WINAPI TextRange_CompareEndpoints(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, HUIATEXTRANGE targetRange, enum TextPatternRangeEndpoint targetEndpoint, int* pRetVal);
-HRESULT WINAPI TextRange_ExpandToEnclosingUnit(HUIATEXTRANGE hobj, enum TextUnit unit);
-HRESULT WINAPI TextRange_GetAttributeValue(HUIATEXTRANGE hobj, TEXTATTRIBUTEID attributeId, VARIANT * pRetVal);
-HRESULT WINAPI TextRange_FindAttribute(HUIATEXTRANGE hobj, TEXTATTRIBUTEID attributeId, VARIANT val, BOOL backward, HUIATEXTRANGE * pRetVal);
-HRESULT WINAPI TextRange_FindText(HUIATEXTRANGE hobj, BSTR text, BOOL backward, BOOL ignoreCase, HUIATEXTRANGE* pRetVal);
-HRESULT WINAPI TextRange_GetBoundingRectangles(HUIATEXTRANGE hobj, SAFEARRAY** pRetVal);
-HRESULT WINAPI TextRange_GetEnclosingElement(HUIATEXTRANGE hobj, HUIANODE* pRetVal);
-HRESULT WINAPI TextRange_GetText(HUIATEXTRANGE hobj, int maxLength, BSTR* pRetVal);
-HRESULT WINAPI TextRange_Move(HUIATEXTRANGE hobj, enum TextUnit unit, int count, int* pRetVal);
-HRESULT WINAPI TextRange_MoveEndpointByUnit(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, enum TextUnit unit, int count, int* pRetVal);
-HRESULT WINAPI TextRange_MoveEndpointByRange(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, HUIATEXTRANGE targetRange, enum TextPatternRangeEndpoint targetEndpoint);
-HRESULT WINAPI TextRange_Select(HUIATEXTRANGE hobj);
-HRESULT WINAPI TextRange_AddToSelection(HUIATEXTRANGE hobj);
-HRESULT WINAPI TextRange_RemoveFromSelection(HUIATEXTRANGE hobj);
-HRESULT WINAPI TextRange_ScrollIntoView(HUIATEXTRANGE hobj, BOOL alignToTop);
-HRESULT WINAPI TextRange_GetChildren(HUIATEXTRANGE hobj, SAFEARRAY** pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ DockPattern_SetDockPosition(HUIAPATTERNOBJECT hobj, enum DockPosition dockPosition);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ExpandCollapsePattern_Collapse(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ExpandCollapsePattern_Expand(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ GridPattern_GetItem(HUIAPATTERNOBJECT hobj, int row, int column, HUIANODE * pResult);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ InvokePattern_Invoke(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ MultipleViewPattern_GetViewName(HUIAPATTERNOBJECT hobj, int viewId, BSTR * ppStr);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ MultipleViewPattern_SetCurrentView(HUIAPATTERNOBJECT hobj, int viewId);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ RangeValuePattern_SetValue(HUIAPATTERNOBJECT hobj, double val);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ScrollItemPattern_ScrollIntoView(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ScrollPattern_Scroll(HUIAPATTERNOBJECT hobj, enum ScrollAmount horizontalAmount, enum ScrollAmount verticalAmount);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ScrollPattern_SetScrollPercent(HUIAPATTERNOBJECT hobj, double horizontalPercent, double verticalPercent);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ SelectionItemPattern_AddToSelection(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ SelectionItemPattern_RemoveFromSelection(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ SelectionItemPattern_Select(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TogglePattern_Toggle(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TransformPattern_Move(HUIAPATTERNOBJECT hobj, double x, double y);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TransformPattern_Resize(HUIAPATTERNOBJECT hobj, double width, double height);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TransformPattern_Rotate(HUIAPATTERNOBJECT hobj, double degrees);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ ValuePattern_SetValue(HUIAPATTERNOBJECT hobj, LPCWSTR pVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ WindowPattern_Close(HUIAPATTERNOBJECT hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ WindowPattern_SetWindowVisualState(HUIAPATTERNOBJECT hobj, enum WindowVisualState state);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ WindowPattern_WaitForInputIdle(HUIAPATTERNOBJECT hobj, int milliseconds, BOOL * pResult);
+
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_GetSelection(HUIAPATTERNOBJECT hobj, SAFEARRAY** pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_GetVisibleRanges(HUIAPATTERNOBJECT hobj, SAFEARRAY** pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_RangeFromChild(HUIAPATTERNOBJECT hobj, HUIANODE hnodeChild, HUIATEXTRANGE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_RangeFromPoint(HUIAPATTERNOBJECT hobj, struct UiaPoint point, HUIATEXTRANGE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_get_DocumentRange(HUIAPATTERNOBJECT hobj, HUIATEXTRANGE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextPattern_get_SupportedTextSelection(HUIAPATTERNOBJECT hobj, enum SupportedTextSelection* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_Clone(HUIATEXTRANGE hobj, HUIATEXTRANGE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_Compare(HUIATEXTRANGE hobj, HUIATEXTRANGE range, BOOL* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_CompareEndpoints(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, HUIATEXTRANGE targetRange, enum TextPatternRangeEndpoint targetEndpoint, int* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_ExpandToEnclosingUnit(HUIATEXTRANGE hobj, enum TextUnit unit);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_GetAttributeValue(HUIATEXTRANGE hobj, TEXTATTRIBUTEID attributeId, VARIANT * pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_FindAttribute(HUIATEXTRANGE hobj, TEXTATTRIBUTEID attributeId, VARIANT val, BOOL backward, HUIATEXTRANGE * pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_FindText(HUIATEXTRANGE hobj, BSTR text, BOOL backward, BOOL ignoreCase, HUIATEXTRANGE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_GetBoundingRectangles(HUIATEXTRANGE hobj, SAFEARRAY** pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_GetEnclosingElement(HUIATEXTRANGE hobj, HUIANODE* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_GetText(HUIATEXTRANGE hobj, int maxLength, BSTR* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_Move(HUIATEXTRANGE hobj, enum TextUnit unit, int count, int* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_MoveEndpointByUnit(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, enum TextUnit unit, int count, int* pRetVal);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_MoveEndpointByRange(HUIATEXTRANGE hobj, enum TextPatternRangeEndpoint endpoint, HUIATEXTRANGE targetRange, enum TextPatternRangeEndpoint targetEndpoint);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_Select(HUIATEXTRANGE hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_AddToSelection(HUIATEXTRANGE hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_RemoveFromSelection(HUIATEXTRANGE hobj);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_ScrollIntoView(HUIATEXTRANGE hobj, BOOL alignToTop);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ TextRange_GetChildren(HUIATEXTRANGE hobj, SAFEARRAY** pRetVal);
 
 HRESULT WINAPI ItemContainerPattern_FindItemByProperty(HUIAPATTERNOBJECT hobj, HUIANODE hnodeStartAfter, PROPERTYID propertyId, VARIANT value, HUIANODE *pFound);
 HRESULT WINAPI LegacyIAccessiblePattern_Select(HUIAPATTERNOBJECT hobj, long flagsSelect);
@@ -871,8 +872,8 @@ HRESULT WINAPI SynchronizedInputPattern_Cancel(HUIAPATTERNOBJECT hobj);
 HRESULT WINAPI VirtualizedItemPattern_Realize(HUIAPATTERNOBJECT hobj);
 HRESULT WINAPI CustomNavigationPattern_Navigate(HUIAPATTERNOBJECT hobj, enum NavigateDirection direction, HUIANODE * pResult);
 
-BOOL WINAPI UiaPatternRelease(HUIAPATTERNOBJECT hobj);
-BOOL WINAPI UiaTextRangeRelease(HUIATEXTRANGE hobj);
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaPatternRelease(HUIAPATTERNOBJECT hobj);
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaTextRangeRelease(HUIATEXTRANGE hobj);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
@@ -887,9 +888,9 @@ BOOL WINAPI UiaTextRangeRelease(HUIATEXTRANGE hobj);
 // 
 // --------------------------------------------------------------------------
 
-LRESULT WINAPI UiaReturnRawElementProvider(HWND hwnd, WPARAM wParam, LPARAM lParam, IRawElementProviderSimple * el);
-HRESULT WINAPI UiaHostProviderFromHwnd(HWND hwnd, IRawElementProviderSimple ** ppProvider);
-HRESULT WINAPI UiaProviderForNonClient(HWND hwnd, long idObject, long idChild, IRawElementProviderSimple ** ppProvider);
+LRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaReturnRawElementProvider(HWND hwnd, WPARAM wParam, LPARAM lParam, IRawElementProviderSimple * el);
+HRESULT WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaHostProviderFromHwnd(HWND hwnd, IRawElementProviderSimple ** ppProvider);
+HRESULT WINAPI _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ UiaProviderForNonClient(HWND hwnd, long idObject, long idChild, IRawElementProviderSimple ** ppProvider);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP) */
 #pragma endregion
@@ -900,7 +901,7 @@ HRESULT WINAPI UiaProviderForNonClient(HWND hwnd, long idObject, long idChild, I
 #define UIA_IAFP_DEFAULT        0x0000
 #define UIA_IAFP_UNWRAP_BRIDGE  0x0001
 
-HRESULT WINAPI UiaIAccessibleFromProvider(
+HRESULT WINAPI _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ UiaIAccessibleFromProvider(
     _In_ IRawElementProviderSimple * pProvider, 
     _In_ DWORD dwFlags, 
     _Outptr_ IAccessible ** ppAccessible, 
@@ -915,7 +916,7 @@ HRESULT WINAPI UiaIAccessibleFromProvider(
 #define UIA_PFIA_DEFAULT        0x0000
 #define UIA_PFIA_UNWRAP_BRIDGE  0x0001
 
-HRESULT WINAPI UiaProviderFromIAccessible(
+HRESULT WINAPI _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ UiaProviderFromIAccessible(
     _In_ IAccessible * pAccessible, 
     long idChild, 
     _In_ DWORD dwFlags, 
@@ -927,8 +928,8 @@ HRESULT WINAPI UiaProviderFromIAccessible(
 // 
 // --------------------------------------------------------------------------
 
-HRESULT WINAPI UiaDisconnectAllProviders();
-HRESULT WINAPI UiaDisconnectProvider(IRawElementProviderSimple * pProvider);
+HRESULT WINAPI _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ UiaDisconnectAllProviders();
+HRESULT WINAPI _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ UiaDisconnectProvider(IRawElementProviderSimple * pProvider);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP) */
 #pragma endregion
@@ -942,7 +943,7 @@ HRESULT WINAPI UiaDisconnectProvider(IRawElementProviderSimple * pProvider);
 //
 // --------------------------------------------------------------------------
 
-BOOL WINAPI UiaHasServerSideProvider(HWND hwnd);
+BOOL WINAPI _Windows_SupportedOS_WindowsXP_ _Windows_SupportedOS_Server2003_ UiaHasServerSideProvider(HWND hwnd);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
