diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationClient.h b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationClient.h
index 38f461d2a4..aa94006971 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationClient.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationClient.h
@@ -9,6 +9,10 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
 #endif
@@ -1586,8 +1590,9 @@ const long UIA_SayAsInterpretAsMetadataId	=	100000;
 EXTERN_C const IID IID_IUIAutomationElement;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d22108aa-8ac5-49a5-837b-37bbb3d7591e")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationElement : public IUnknown
     {
     public:
@@ -2818,8 +2823,9 @@ EXTERN_C const IID IID_IUIAutomationElement;
 EXTERN_C const IID IID_IUIAutomationElementArray;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("14314595-b4bc-4055-95f2-58f2e42c9855")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationElementArray : public IUnknown
     {
     public:
@@ -2922,8 +2928,9 @@ EXTERN_C const IID IID_IUIAutomationElementArray;
 EXTERN_C const IID IID_IUIAutomationCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("352ffba8-0973-437c-a61f-f64cafd81df9")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationCondition : public IUnknown
     {
     public:
@@ -2996,8 +3003,9 @@ EXTERN_C const IID IID_IUIAutomationCondition;
 EXTERN_C const IID IID_IUIAutomationBoolCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("1b4e1f2e-75eb-4d0b-8952-5a69988e2307")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationBoolCondition : public IUIAutomationCondition
     {
     public:
@@ -3084,8 +3092,9 @@ EXTERN_C const IID IID_IUIAutomationBoolCondition;
 EXTERN_C const IID IID_IUIAutomationPropertyCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("99ebf2cb-5578-4267-9ad4-afd6ea77e94b")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationPropertyCondition : public IUIAutomationCondition
     {
     public:
@@ -3198,8 +3207,9 @@ EXTERN_C const IID IID_IUIAutomationPropertyCondition;
 EXTERN_C const IID IID_IUIAutomationAndCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a7d0af36-b912-45fe-9855-091ddc174aec")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationAndCondition : public IUIAutomationCondition
     {
     public:
@@ -3316,8 +3326,9 @@ EXTERN_C const IID IID_IUIAutomationAndCondition;
 EXTERN_C const IID IID_IUIAutomationOrCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("8753f032-3db1-47b5-a1fc-6e34a266c712")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationOrCondition : public IUIAutomationCondition
     {
     public:
@@ -3434,8 +3445,9 @@ EXTERN_C const IID IID_IUIAutomationOrCondition;
 EXTERN_C const IID IID_IUIAutomationNotCondition;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("f528b657-847b-498c-8896-d52b565407a1")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationNotCondition : public IUIAutomationCondition
     {
     public:
@@ -3522,8 +3534,9 @@ EXTERN_C const IID IID_IUIAutomationNotCondition;
 EXTERN_C const IID IID_IUIAutomationCacheRequest;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("b32a92b5-bc25-4078-9c08-d7ee95c48e03")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationCacheRequest : public IUnknown
     {
     public:
@@ -3713,8 +3726,9 @@ EXTERN_C const IID IID_IUIAutomationCacheRequest;
 EXTERN_C const IID IID_IUIAutomationTreeWalker;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("4042c624-389c-4afc-a630-9df854a541fc")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTreeWalker : public IUnknown
     {
     public:
@@ -4028,8 +4042,9 @@ EXTERN_C const IID IID_IUIAutomationTreeWalker;
 EXTERN_C const IID IID_IUIAutomationEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("146c3c17-f12e-4e22-8c27-f894b9b79c69")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationEventHandler : public IUnknown
     {
     public:
@@ -4119,8 +4134,9 @@ EXTERN_C const IID IID_IUIAutomationEventHandler;
 EXTERN_C const IID IID_IUIAutomationPropertyChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("40cd37d4-c756-4b0c-8c6f-bddfeeb13b50")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationPropertyChangedEventHandler : public IUnknown
     {
     public:
@@ -4214,8 +4230,9 @@ EXTERN_C const IID IID_IUIAutomationPropertyChangedEventHandler;
 EXTERN_C const IID IID_IUIAutomationStructureChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("e81d1b4e-11c5-42f8-9754-e7036c79f054")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationStructureChangedEventHandler : public IUnknown
     {
     public:
@@ -4309,8 +4326,9 @@ EXTERN_C const IID IID_IUIAutomationStructureChangedEventHandler;
 EXTERN_C const IID IID_IUIAutomationFocusChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("c270f6b5-5c69-4290-9745-7a7f97169468")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationFocusChangedEventHandler : public IUnknown
     {
     public:
@@ -4396,8 +4414,9 @@ EXTERN_C const IID IID_IUIAutomationFocusChangedEventHandler;
 EXTERN_C const IID IID_IUIAutomationTextEditTextChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("92FAA680-E704-4156-931A-E32D5BB38F3F")
+    
+    _Windows_SupportedOS_Windows81_
     IUIAutomationTextEditTextChangedEventHandler : public IUnknown
     {
     public:
@@ -4491,8 +4510,9 @@ EXTERN_C const IID IID_IUIAutomationTextEditTextChangedEventHandler;
 EXTERN_C const IID IID_IUIAutomationChangesEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("58EDCA55-2C3E-4980-B1B9-56C17F27A2A0")
+    
+    _Windows_SupportedOS_15063_
     IUIAutomationChangesEventHandler : public IUnknown
     {
     public:
@@ -4586,8 +4606,9 @@ EXTERN_C const IID IID_IUIAutomationChangesEventHandler;
 EXTERN_C const IID IID_IUIAutomationNotificationEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("C7CB2637-E6C2-4D0C-85DE-4948C02175C7")
+    
+    _Windows_SupportedOS_16299_
     IUIAutomationNotificationEventHandler : public IUnknown
     {
     public:
@@ -4685,8 +4706,9 @@ EXTERN_C const IID IID_IUIAutomationNotificationEventHandler;
 EXTERN_C const IID IID_IUIAutomationInvokePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("fb377fbe-8ea6-46d5-9c73-6499642d3059")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationInvokePattern : public IUnknown
     {
     public:
@@ -4768,8 +4790,9 @@ EXTERN_C const IID IID_IUIAutomationInvokePattern;
 EXTERN_C const IID IID_IUIAutomationDockPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("fde5ef97-1464-48f6-90bf-43d0948e86ec")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationDockPattern : public IUnknown
     {
     public:
@@ -4881,8 +4904,9 @@ EXTERN_C const IID IID_IUIAutomationDockPattern;
 EXTERN_C const IID IID_IUIAutomationExpandCollapsePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("619be086-1f4e-4ee4-bafa-210128738730")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationExpandCollapsePattern : public IUnknown
     {
     public:
@@ -4999,8 +5023,9 @@ EXTERN_C const IID IID_IUIAutomationExpandCollapsePattern;
 EXTERN_C const IID IID_IUIAutomationGridPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("414c3cdc-856b-4f5b-8538-3131c6302550")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationGridPattern : public IUnknown
     {
     public:
@@ -5146,8 +5171,9 @@ EXTERN_C const IID IID_IUIAutomationGridPattern;
 EXTERN_C const IID IID_IUIAutomationGridItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("78f8ef57-66c3-4e09-bd7c-e79b2004894d")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationGridItemPattern : public IUnknown
     {
     public:
@@ -5350,8 +5376,9 @@ EXTERN_C const IID IID_IUIAutomationGridItemPattern;
 EXTERN_C const IID IID_IUIAutomationMultipleViewPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("8d253c91-1dc5-4bb5-b18f-ade16fa495e8")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationMultipleViewPattern : public IUnknown
     {
     public:
@@ -5506,8 +5533,9 @@ EXTERN_C const IID IID_IUIAutomationMultipleViewPattern;
 EXTERN_C const IID IID_IUIAutomationObjectModelPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("71c284b3-c14d-4d14-981e-19751b0d756d")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationObjectModelPattern : public IUnknown
     {
     public:
@@ -5593,8 +5621,9 @@ EXTERN_C const IID IID_IUIAutomationObjectModelPattern;
 EXTERN_C const IID IID_IUIAutomationRangeValuePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("59213f4f-7346-49e5-b120-80555987a148")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationRangeValuePattern : public IUnknown
     {
     public:
@@ -5836,8 +5865,9 @@ EXTERN_C const IID IID_IUIAutomationRangeValuePattern;
 EXTERN_C const IID IID_IUIAutomationScrollPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("88f4d42a-e881-459d-a77c-73bbbb7e02dc")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationScrollPattern : public IUnknown
     {
     public:
@@ -6100,8 +6130,9 @@ EXTERN_C const IID IID_IUIAutomationScrollPattern;
 EXTERN_C const IID IID_IUIAutomationScrollItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("b488300f-d015-4f19-9c29-bb595e3645ef")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationScrollItemPattern : public IUnknown
     {
     public:
@@ -6183,8 +6214,9 @@ EXTERN_C const IID IID_IUIAutomationScrollItemPattern;
 EXTERN_C const IID IID_IUIAutomationSelectionPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("5ed5202e-b2ac-47a6-b638-4b0bf140d78e")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationSelectionPattern : public IUnknown
     {
     public:
@@ -6335,8 +6367,9 @@ EXTERN_C const IID IID_IUIAutomationSelectionPattern;
 EXTERN_C const IID IID_IUIAutomationSelectionPattern2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("0532bfae-c011-4e32-a343-6d642d798555")
+    
+    _Windows_SupportedOS_16299_
     IUIAutomationSelectionPattern2 : public IUIAutomationSelectionPattern
     {
     public:
@@ -6568,8 +6601,9 @@ EXTERN_C const IID IID_IUIAutomationSelectionPattern2;
 EXTERN_C const IID IID_IUIAutomationSelectionItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a8efa66a-0fda-421a-9194-38021f3578ea")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationSelectionItemPattern : public IUnknown
     {
     public:
@@ -6721,8 +6755,9 @@ EXTERN_C const IID IID_IUIAutomationSelectionItemPattern;
 EXTERN_C const IID IID_IUIAutomationSynchronizedInputPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("2233be0b-afb7-448b-9fda-3b378aa5eae1")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationSynchronizedInputPattern : public IUnknown
     {
     public:
@@ -6817,8 +6852,9 @@ EXTERN_C const IID IID_IUIAutomationSynchronizedInputPattern;
 EXTERN_C const IID IID_IUIAutomationTablePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("620e691c-ea96-4710-a850-754b24ce2417")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTablePattern : public IUnknown
     {
     public:
@@ -6969,8 +7005,9 @@ EXTERN_C const IID IID_IUIAutomationTablePattern;
 EXTERN_C const IID IID_IUIAutomationTableItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("0b964eb3-ef2e-4464-9c79-61d61737a27e")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTableItemPattern : public IUnknown
     {
     public:
@@ -7095,8 +7132,9 @@ EXTERN_C const IID IID_IUIAutomationTableItemPattern;
 EXTERN_C const IID IID_IUIAutomationTogglePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("94cf8058-9b8d-4ab9-8bfd-4cd0a33c8c70")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTogglePattern : public IUnknown
     {
     public:
@@ -7204,8 +7242,9 @@ EXTERN_C const IID IID_IUIAutomationTogglePattern;
 EXTERN_C const IID IID_IUIAutomationTransformPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a9b55844-a55d-4ef0-926d-569c16ff89bb")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTransformPattern : public IUnknown
     {
     public:
@@ -7403,8 +7442,9 @@ EXTERN_C const IID IID_IUIAutomationTransformPattern;
 EXTERN_C const IID IID_IUIAutomationValuePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a94cd8b1-0844-4cd6-9d2d-640537ab39e9")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationValuePattern : public IUnknown
     {
     public:
@@ -7542,8 +7582,9 @@ EXTERN_C const IID IID_IUIAutomationValuePattern;
 EXTERN_C const IID IID_IUIAutomationWindowPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("0faef453-9208-43ef-bbb2-3b485177864f")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationWindowPattern : public IUnknown
     {
     public:
@@ -7811,8 +7852,9 @@ EXTERN_C const IID IID_IUIAutomationWindowPattern;
 EXTERN_C const IID IID_IUIAutomationTextRange;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a543cc6a-f4ae-494b-8239-c814481187a8")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTextRange : public IUnknown
     {
     public:
@@ -8183,8 +8225,9 @@ EXTERN_C const IID IID_IUIAutomationTextRange;
 EXTERN_C const IID IID_IUIAutomationTextRange2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("BB9B40E0-5E04-46BD-9BE0-4B601B9AFAD4")
+    
+    _Windows_SupportedOS_Windows81_
     IUIAutomationTextRange2 : public IUIAutomationTextRange
     {
     public:
@@ -8461,8 +8504,9 @@ EXTERN_C const IID IID_IUIAutomationTextRange2;
 EXTERN_C const IID IID_IUIAutomationTextRange3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6A315D69-5512-4C2E-85F0-53FCE6DD4BC2")
+    
+    _Windows_SupportedOS_15063_
     IUIAutomationTextRange3 : public IUIAutomationTextRange2
     {
     public:
@@ -8793,8 +8837,9 @@ EXTERN_C const IID IID_IUIAutomationTextRange3;
 EXTERN_C const IID IID_IUIAutomationTextRangeArray;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("ce4ae76a-e717-4c98-81ea-47371d028eb6")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTextRangeArray : public IUnknown
     {
     public:
@@ -8897,8 +8942,9 @@ EXTERN_C const IID IID_IUIAutomationTextRangeArray;
 EXTERN_C const IID IID_IUIAutomationTextPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("32eba289-3583-42c9-9c59-3b6d9a1e9b6a")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationTextPattern : public IUnknown
     {
     public:
@@ -9057,8 +9103,9 @@ EXTERN_C const IID IID_IUIAutomationTextPattern;
 EXTERN_C const IID IID_IUIAutomationTextPattern2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("506a921a-fcc9-409f-b23b-37eb74106872")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationTextPattern2 : public IUIAutomationTextPattern
     {
     public:
@@ -9224,8 +9271,9 @@ EXTERN_C const IID IID_IUIAutomationTextPattern2;
 EXTERN_C const IID IID_IUIAutomationTextEditPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("17E21576-996C-4870-99D9-BFF323380C06")
+    
+    _Windows_SupportedOS_Windows81_
     IUIAutomationTextEditPattern : public IUIAutomationTextPattern
     {
     public:
@@ -9383,8 +9431,9 @@ EXTERN_C const IID IID_IUIAutomationTextEditPattern;
 EXTERN_C const IID IID_IUIAutomationCustomNavigationPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("01EA217A-1766-47ED-A6CC-ACF492854B1F")
+    
+    _Windows_SupportedOS_10240_
     IUIAutomationCustomNavigationPattern : public IUnknown
     {
     public:
@@ -9474,8 +9523,9 @@ EXTERN_C const IID IID_IUIAutomationCustomNavigationPattern;
 EXTERN_C const IID IID_IUIAutomationActiveTextPositionChangedEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("F97933B0-8DAE-4496-8997-5BA015FE0D82")
+    
+    _Windows_SupportedOS_17763_
     IUIAutomationActiveTextPositionChangedEventHandler : public IUnknown
     {
     public:
@@ -9565,8 +9615,9 @@ EXTERN_C const IID IID_IUIAutomationActiveTextPositionChangedEventHandler;
 EXTERN_C const IID IID_IUIAutomationLegacyIAccessiblePattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("828055ad-355b-4435-86d5-3b51c14a9b1b")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationLegacyIAccessiblePattern : public IUnknown
     {
     public:
@@ -9943,8 +9994,9 @@ EXTERN_C const IID IID_IUIAutomationLegacyIAccessiblePattern;
 EXTERN_C const IID IID_IUIAutomationItemContainerPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("c690fdb2-27a8-423c-812d-429773c9084e")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationItemContainerPattern : public IUnknown
     {
     public:
@@ -10042,8 +10094,9 @@ EXTERN_C const IID IID_IUIAutomationItemContainerPattern;
 EXTERN_C const IID IID_IUIAutomationVirtualizedItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6ba3d7a6-04cf-4f11-8793-a8d1cde9969f")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationVirtualizedItemPattern : public IUnknown
     {
     public:
@@ -10125,8 +10178,9 @@ EXTERN_C const IID IID_IUIAutomationVirtualizedItemPattern;
 EXTERN_C const IID IID_IUIAutomationAnnotationPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("9a175b21-339e-41b1-8e8b-623f6b681098")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationAnnotationPattern : public IUnknown
     {
     public:
@@ -10329,8 +10383,9 @@ EXTERN_C const IID IID_IUIAutomationAnnotationPattern;
 EXTERN_C const IID IID_IUIAutomationStylesPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("85b5f0a2-bd79-484a-ad2b-388c9838d5fb")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationStylesPattern : public IUnknown
     {
     public:
@@ -10619,8 +10674,9 @@ EXTERN_C const IID IID_IUIAutomationStylesPattern;
 EXTERN_C const IID IID_IUIAutomationSpreadsheetPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("7517a7c8-faae-4de9-9f08-29b91e8595c1")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationSpreadsheetPattern : public IUnknown
     {
     public:
@@ -10710,8 +10766,9 @@ EXTERN_C const IID IID_IUIAutomationSpreadsheetPattern;
 EXTERN_C const IID IID_IUIAutomationSpreadsheetItemPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("7d4fb86c-8d34-40e1-8e83-62c15204e335")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationSpreadsheetItemPattern : public IUnknown
     {
     public:
@@ -10862,8 +10919,9 @@ EXTERN_C const IID IID_IUIAutomationSpreadsheetItemPattern;
 EXTERN_C const IID IID_IUIAutomationTransformPattern2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6d74d017-6ecb-4381-b38b-3c17a48ff1c2")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationTransformPattern2 : public IUIAutomationTransformPattern
     {
     public:
@@ -11152,8 +11210,9 @@ EXTERN_C const IID IID_IUIAutomationTransformPattern2;
 EXTERN_C const IID IID_IUIAutomationTextChildPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6552b038-ae05-40c8-abfd-aa08352aab86")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationTextChildPattern : public IUnknown
     {
     public:
@@ -11252,8 +11311,9 @@ EXTERN_C const IID IID_IUIAutomationTextChildPattern;
 EXTERN_C const IID IID_IUIAutomationDragPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("1dc7b570-1f54-4bad-bcda-d36a722fb7bd")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationDragPattern : public IUnknown
     {
     public:
@@ -11430,8 +11490,9 @@ EXTERN_C const IID IID_IUIAutomationDragPattern;
 EXTERN_C const IID IID_IUIAutomationDropTargetPattern;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("69a095f7-eee4-430e-a46b-fb73b1ae39a5")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationDropTargetPattern : public IUnknown
     {
     public:
@@ -11556,8 +11617,9 @@ EXTERN_C const IID IID_IUIAutomationDropTargetPattern;
 EXTERN_C const IID IID_IUIAutomationElement2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6749c683-f70d-4487-a698-5f79d55290d6")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomationElement2 : public IUIAutomationElement
     {
     public:
@@ -12493,8 +12555,9 @@ EXTERN_C const IID IID_IUIAutomationElement2;
 EXTERN_C const IID IID_IUIAutomationElement3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("8471DF34-AEE0-4A01-A7DE-7DB9AF12C296")
+    
+    _Windows_SupportedOS_Windows81_
     IUIAutomationElement3 : public IUIAutomationElement2
     {
     public:
@@ -13442,8 +13505,9 @@ EXTERN_C const IID IID_IUIAutomationElement3;
 EXTERN_C const IID IID_IUIAutomationElement4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("3B6E233C-52FB-4063-A4C9-77C075C2A06B")
+    
+    _Windows_SupportedOS_10240_
     IUIAutomationElement4 : public IUIAutomationElement3
     {
     public:
@@ -14512,8 +14576,9 @@ EXTERN_C const IID IID_IUIAutomationElement4;
 EXTERN_C const IID IID_IUIAutomationElement5;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("98141C1D-0D0E-4175-BBE2-6BFF455842A7")
+    
+    _Windows_SupportedOS_15063_
     IUIAutomationElement5 : public IUIAutomationElement4
     {
     public:
@@ -15595,8 +15660,9 @@ EXTERN_C const IID IID_IUIAutomationElement5;
 EXTERN_C const IID IID_IUIAutomationElement6;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("4780d450-8bca-4977-afa5-a4a517f555e3")
+    
+    _Windows_SupportedOS_15063_
     IUIAutomationElement6 : public IUIAutomationElement5
     {
     public:
@@ -16689,8 +16755,9 @@ EXTERN_C const IID IID_IUIAutomationElement6;
 EXTERN_C const IID IID_IUIAutomationElement7;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("204e8572-cfc3-4c11-b0c8-7da7420750b7")
+    
+    _Windows_SupportedOS_15063_
     IUIAutomationElement7 : public IUIAutomationElement6
     {
     public:
@@ -17921,8 +17988,9 @@ EXTERN_C const IID IID_IUIAutomationElement7;
 EXTERN_C const IID IID_IUIAutomationElement8;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("8C60217D-5411-4CDE-BCC0-1CEDA223830C")
+    
+    _Windows_SupportedOS_17134_
     IUIAutomationElement8 : public IUIAutomationElement7
     {
     public:
@@ -19120,8 +19188,9 @@ EXTERN_C const IID IID_IUIAutomationElement8;
 EXTERN_C const IID IID_IUIAutomationElement9;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("39325fac-039d-440e-a3a3-5eb81a5cecc3")
+    
+    _Windows_SupportedOS_17763_
     IUIAutomationElement9 : public IUIAutomationElement8
     {
     public:
@@ -20338,8 +20407,9 @@ EXTERN_C const IID IID_IUIAutomationElement9;
 EXTERN_C const IID IID_IUIAutomationProxyFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("85b94ecd-849d-42b6-b94d-d6db23fdf5a4")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationProxyFactory : public IUnknown
     {
     public:
@@ -20450,8 +20520,9 @@ EXTERN_C const IID IID_IUIAutomationProxyFactory;
 EXTERN_C const IID IID_IUIAutomationProxyFactoryEntry;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d50e472e-b64b-490c-bca1-d30696f9f289")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationProxyFactoryEntry : public IUnknown
     {
     public:
@@ -20709,8 +20780,9 @@ EXTERN_C const IID IID_IUIAutomationProxyFactoryEntry;
 EXTERN_C const IID IID_IUIAutomationProxyFactoryMapping;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("09e31e18-872d-4873-93d1-1e541ec133fd")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationProxyFactoryMapping : public IUnknown
     {
     public:
@@ -20904,8 +20976,9 @@ EXTERN_C const IID IID_IUIAutomationProxyFactoryMapping;
 EXTERN_C const IID IID_IUIAutomationEventHandlerGroup;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("C9EE12F2-C13B-4408-997C-639914377F4E")
+    
+    _Windows_SupportedOS_17763_
     IUIAutomationEventHandlerGroup : public IUnknown
     {
     public:
@@ -21149,8 +21222,9 @@ EXTERN_C const IID IID_IUIAutomationEventHandlerGroup;
 EXTERN_C const IID IID_IUIAutomation;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("30cbe57d-d9d0-452a-ab13-7ac5ac4825ee")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomation : public IUnknown
     {
     public:
@@ -22218,8 +22292,9 @@ EXTERN_C const IID IID_IUIAutomation;
 EXTERN_C const IID IID_IUIAutomation2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("34723aff-0c9d-49d0-9896-7ab52df8cd8a")
+    
+    _Windows_SupportedOS_Windows8_
     IUIAutomation2 : public IUIAutomation
     {
     public:
@@ -23006,8 +23081,9 @@ EXTERN_C const IID IID_IUIAutomation2;
 EXTERN_C const IID IID_IUIAutomation3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("73D768DA-9B51-4B89-936E-C209290973E7")
+    
+    _Windows_SupportedOS_Windows81_
     IUIAutomation3 : public IUIAutomation2
     {
     public:
@@ -23817,8 +23893,9 @@ EXTERN_C const IID IID_IUIAutomation3;
 EXTERN_C const IID IID_IUIAutomation4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("1189C02A-05F8-4319-8E21-E817E3DB2860")
+    
+    _Windows_SupportedOS_14393_
     IUIAutomation4 : public IUIAutomation3
     {
     public:
@@ -24661,8 +24738,9 @@ EXTERN_C const IID IID_IUIAutomation4;
 EXTERN_C const IID IID_IUIAutomation5;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("25F700C8-D816-4057-A9DC-3CBDEE77E256")
+    
+    _Windows_SupportedOS_14393_
     IUIAutomation5 : public IUIAutomation4
     {
     public:
@@ -25528,8 +25606,9 @@ EXTERN_C const IID IID_IUIAutomation5;
 EXTERN_C const IID IID_IUIAutomation6;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("AAE072DA-29E3-413D-87A7-192DBF81ED10")
+    
+    _Windows_SupportedOS_17763_
     IUIAutomation6 : public IUIAutomation5
     {
     public:
