diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCore.h b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCore.h
index 8730f908c4..45f521fda0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCore.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UIAutomationCore.h
@@ -9,6 +9,10 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
 #endif
@@ -967,8 +971,9 @@ const double UIA_ScrollPatternNoScroll	=	-1;
 EXTERN_C const IID IID_IRawElementProviderSimple;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d6dd68d1-86fd-4332-8666-9abedea2d24c")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRawElementProviderSimple : public IUnknown
     {
     public:
@@ -1101,8 +1106,9 @@ EXTERN_C const IID IID_IRawElementProviderSimple;
 EXTERN_C const IID IID_IAccessibleEx;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("f8b80ada-2c44-48d0-89be-5ff23c9cd875")
+    
+    _Windows_SupportedOS_Windows7_
     IAccessibleEx : public IUnknown
     {
     public:
@@ -1239,8 +1245,9 @@ EXTERN_C const IID IID_IAccessibleEx;
 EXTERN_C const IID IID_IRawElementProviderSimple2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("A0A839A9-8DA1-4A82-806A-8E0D44E79F56")
+    
+    _Windows_SupportedOS_Windows81_
     IRawElementProviderSimple2 : public IRawElementProviderSimple
     {
     public:
@@ -1363,8 +1370,9 @@ EXTERN_C const IID IID_IRawElementProviderSimple2;
 EXTERN_C const IID IID_IRawElementProviderSimple3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("fcf5d820-d7ec-4613-bdf6-42a84ce7daaf")
+    
+    _Windows_SupportedOS_15063_
     IRawElementProviderSimple3 : public IRawElementProviderSimple2
     {
     public:
@@ -1507,8 +1515,9 @@ EXTERN_C const IID IID_IRawElementProviderSimple3;
 EXTERN_C const IID IID_IRawElementProviderFragmentRoot;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("620ce2a5-ab8f-40a9-86cb-de3c75599b58")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRawElementProviderFragmentRoot : public IUnknown
     {
     public:
@@ -1615,8 +1624,9 @@ EXTERN_C const IID IID_IRawElementProviderFragmentRoot;
 EXTERN_C const IID IID_IRawElementProviderFragment;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("f7063da8-8359-439c-9297-bbc5299a7d87")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRawElementProviderFragment : public IUnknown
     {
     public:
@@ -1767,8 +1777,9 @@ EXTERN_C const IID IID_IRawElementProviderFragment;
 EXTERN_C const IID IID_IRawElementProviderAdviseEvents;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("a407b27b-0f6d-4427-9292-473c7bf93258")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRawElementProviderAdviseEvents : public IUnknown
     {
     public:
@@ -1875,8 +1886,9 @@ EXTERN_C const IID IID_IRawElementProviderAdviseEvents;
 EXTERN_C const IID IID_IRawElementProviderHwndOverride;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("1d5df27c-8947-4425-b8d9-79787bb460b8")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRawElementProviderHwndOverride : public IUnknown
     {
     public:
@@ -1966,8 +1978,9 @@ EXTERN_C const IID IID_IRawElementProviderHwndOverride;
 EXTERN_C const IID IID_IProxyProviderWinEventSink;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("4fd82b78-a43e-46ac-9803-0a6969c7c183")
+    
+    _Windows_SupportedOS_Windows7_
     IProxyProviderWinEventSink : public IUnknown
     {
     public:
@@ -2099,8 +2112,9 @@ EXTERN_C const IID IID_IProxyProviderWinEventSink;
 EXTERN_C const IID IID_IProxyProviderWinEventHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("89592ad4-f4e0-43d5-a3b6-bad7e111b435")
+    
+    _Windows_SupportedOS_Windows7_
     IProxyProviderWinEventHandler : public IUnknown
     {
     public:
@@ -2202,8 +2216,9 @@ EXTERN_C const IID IID_IProxyProviderWinEventHandler;
 EXTERN_C const IID IID_IRawElementProviderWindowlessSite;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("0a2a93cc-bfad-42ac-9b2e-0991fb0d3ea0")
+    
+    _Windows_SupportedOS_Windows8_
     IRawElementProviderWindowlessSite : public IUnknown
     {
     public:
@@ -2306,8 +2321,9 @@ EXTERN_C const IID IID_IRawElementProviderWindowlessSite;
 EXTERN_C const IID IID_IAccessibleHostingElementProviders;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("33AC331B-943E-4020-B295-DB37784974A3")
+    
+    _Windows_SupportedOS_Windows8_
     IAccessibleHostingElementProviders : public IUnknown
     {
     public:
@@ -2410,8 +2426,9 @@ EXTERN_C const IID IID_IAccessibleHostingElementProviders;
 EXTERN_C const IID IID_IRawElementProviderHostingAccessibles;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("24BE0B07-D37D-487A-98CF-A13ED465E9B3")
+    
+    _Windows_SupportedOS_Windows8_
     IRawElementProviderHostingAccessibles : public IUnknown
     {
     public:
@@ -2497,8 +2514,9 @@ EXTERN_C const IID IID_IRawElementProviderHostingAccessibles;
 EXTERN_C const IID IID_IDockProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("159bc72c-4ad3-485e-9637-d7052edf0146")
+    
+    _Windows_SupportedOS_WindowsXP_
     IDockProvider : public IUnknown
     {
     public:
@@ -2597,8 +2615,9 @@ EXTERN_C const IID IID_IDockProvider;
 EXTERN_C const IID IID_IExpandCollapseProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d847d3a5-cab0-4a98-8c32-ecb45c59ad24")
+    
+    _Windows_SupportedOS_WindowsXP_
     IExpandCollapseProvider : public IUnknown
     {
     public:
@@ -2702,8 +2721,9 @@ EXTERN_C const IID IID_IExpandCollapseProvider;
 EXTERN_C const IID IID_IGridProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("b17d6187-0907-464b-a168-0ef17a1572b1")
+    
+    _Windows_SupportedOS_WindowsXP_
     IGridProvider : public IUnknown
     {
     public:
@@ -2823,8 +2843,9 @@ EXTERN_C const IID IID_IGridProvider;
 EXTERN_C const IID IID_IGridItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d02541f1-fb81-4d64-ae32-f520f8a6dbd1")
+    
+    _Windows_SupportedOS_WindowsXP_
     IGridItemProvider : public IUnknown
     {
     public:
@@ -2962,8 +2983,9 @@ EXTERN_C const IID IID_IGridItemProvider;
 EXTERN_C const IID IID_IInvokeProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("54fcb24b-e18e-47a2-b4d3-eccbe77599a2")
+    
+    _Windows_SupportedOS_WindowsXP_
     IInvokeProvider : public IUnknown
     {
     public:
@@ -3045,8 +3067,9 @@ EXTERN_C const IID IID_IInvokeProvider;
 EXTERN_C const IID IID_IMultipleViewProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6278cab1-b556-4a1a-b4e0-418acc523201")
+    
+    _Windows_SupportedOS_WindowsXP_
     IMultipleViewProvider : public IUnknown
     {
     public:
@@ -3175,8 +3198,9 @@ EXTERN_C const IID IID_IMultipleViewProvider;
 EXTERN_C const IID IID_IRangeValueProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("36dc7aef-33e6-4691-afe1-2be7274b3d33")
+    
+    _Windows_SupportedOS_WindowsXP_
     IRangeValueProvider : public IUnknown
     {
     public:
@@ -3340,8 +3364,9 @@ EXTERN_C const IID IID_IRangeValueProvider;
 EXTERN_C const IID IID_IScrollItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("2360c714-4bf1-4b26-ba65-9b21316127eb")
+    
+    _Windows_SupportedOS_WindowsXP_
     IScrollItemProvider : public IUnknown
     {
     public:
@@ -3423,8 +3448,9 @@ EXTERN_C const IID IID_IScrollItemProvider;
 EXTERN_C const IID IID_ISelectionProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("fb8b03af-3bdf-48d4-bd36-1a65793be168")
+    
+    _Windows_SupportedOS_WindowsXP_
     ISelectionProvider : public IUnknown
     {
     public:
@@ -3536,8 +3562,9 @@ EXTERN_C const IID IID_ISelectionProvider;
 EXTERN_C const IID IID_ISelectionProvider2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("14f68475-ee1c-44f6-a869-d239381f0fe7")
+    
+    _Windows_SupportedOS_16299_
     ISelectionProvider2 : public ISelectionProvider
     {
     public:
@@ -3690,8 +3717,9 @@ EXTERN_C const IID IID_ISelectionProvider2;
 EXTERN_C const IID IID_IScrollProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("b38b8077-1fc3-42a5-8cae-d40c2215055a")
+    
+    _Windows_SupportedOS_WindowsXP_
     IScrollProvider : public IUnknown
     {
     public:
@@ -3876,8 +3904,9 @@ EXTERN_C const IID IID_IScrollProvider;
 EXTERN_C const IID IID_ISelectionItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("2acad808-b2d4-452d-a407-91ff1ad167b2")
+    
+    _Windows_SupportedOS_WindowsXP_
     ISelectionItemProvider : public IUnknown
     {
     public:
@@ -4003,8 +4032,9 @@ EXTERN_C const IID IID_ISelectionItemProvider;
 EXTERN_C const IID IID_ISynchronizedInputProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("29db1a06-02ce-4cf7-9b42-565d4fab20ee")
+    
+    _Windows_SupportedOS_Windows7_
     ISynchronizedInputProvider : public IUnknown
     {
     public:
@@ -4099,8 +4129,9 @@ EXTERN_C const IID IID_ISynchronizedInputProvider;
 EXTERN_C const IID IID_ITableProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("9c860395-97b3-490a-b52a-858cc22af166")
+    
+    _Windows_SupportedOS_WindowsXP_
     ITableProvider : public IUnknown
     {
     public:
@@ -4212,8 +4243,9 @@ EXTERN_C const IID IID_ITableProvider;
 EXTERN_C const IID IID_ITableItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("b9734fa6-771f-4d78-9c90-2517999349cd")
+    
+    _Windows_SupportedOS_WindowsXP_
     ITableItemProvider : public IUnknown
     {
     public:
@@ -4312,8 +4344,9 @@ EXTERN_C const IID IID_ITableItemProvider;
 EXTERN_C const IID IID_IToggleProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("56d00bd0-c4f4-433c-a836-1a52a57e0892")
+    
+    _Windows_SupportedOS_WindowsXP_
     IToggleProvider : public IUnknown
     {
     public:
@@ -4408,8 +4441,9 @@ EXTERN_C const IID IID_IToggleProvider;
 EXTERN_C const IID IID_ITransformProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6829ddc4-4f91-4ffa-b86f-bd3e2987cb4c")
+    
+    _Windows_SupportedOS_WindowsXP_
     ITransformProvider : public IUnknown
     {
     public:
@@ -4568,8 +4602,9 @@ EXTERN_C const IID IID_ITransformProvider;
 EXTERN_C const IID IID_IValueProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("c7935180-6fb3-4201-b174-7df73adbf64a")
+    
+    _Windows_SupportedOS_WindowsXP_
     IValueProvider : public IUnknown
     {
     public:
@@ -4681,8 +4716,9 @@ EXTERN_C const IID IID_IValueProvider;
 EXTERN_C const IID IID_IWindowProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("987df77b-db06-4d77-8f8a-86a9c3bb90b9")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWindowProvider : public IUnknown
     {
     public:
@@ -4872,8 +4908,9 @@ EXTERN_C const IID IID_IWindowProvider;
 EXTERN_C const IID IID_ILegacyIAccessibleProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("e44c3566-915d-4070-99c6-047bff5a08f5")
+    
+    _Windows_SupportedOS_Windows7_
     ILegacyIAccessibleProvider : public IUnknown
     {
     public:
@@ -5120,8 +5157,9 @@ EXTERN_C const IID IID_ILegacyIAccessibleProvider;
 EXTERN_C const IID IID_IItemContainerProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("e747770b-39ce-4382-ab30-d8fb3f336f24")
+    
+    _Windows_SupportedOS_Windows7_
     IItemContainerProvider : public IUnknown
     {
     public:
@@ -5219,8 +5257,9 @@ EXTERN_C const IID IID_IItemContainerProvider;
 EXTERN_C const IID IID_IVirtualizedItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("cb98b665-2d35-4fac-ad35-f3c60d0c0b8b")
+    
+    _Windows_SupportedOS_Windows7_
     IVirtualizedItemProvider : public IUnknown
     {
     public:
@@ -5302,8 +5341,9 @@ EXTERN_C const IID IID_IVirtualizedItemProvider;
 EXTERN_C const IID IID_IObjectModelProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("3ad86ebd-f5ef-483d-bb18-b1042a475d64")
+    
+    _Windows_SupportedOS_Windows8_
     IObjectModelProvider : public IUnknown
     {
     public:
@@ -5389,8 +5429,9 @@ EXTERN_C const IID IID_IObjectModelProvider;
 EXTERN_C const IID IID_IAnnotationProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("f95c7e80-bd63-4601-9782-445ebff011fc")
+    
+    _Windows_SupportedOS_Windows8_
     IAnnotationProvider : public IUnknown
     {
     public:
@@ -5528,8 +5569,9 @@ EXTERN_C const IID IID_IAnnotationProvider;
 EXTERN_C const IID IID_IStylesProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("19b6b649-f5d7-4a6d-bdcb-129252be588a")
+    
+    _Windows_SupportedOS_Windows8_
     IStylesProvider : public IUnknown
     {
     public:
@@ -5693,8 +5735,9 @@ EXTERN_C const IID IID_IStylesProvider;
 EXTERN_C const IID IID_ISpreadsheetProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6f6b5d35-5525-4f80-b758-85473832ffc7")
+    
+    _Windows_SupportedOS_Windows8_
     ISpreadsheetProvider : public IUnknown
     {
     public:
@@ -5784,8 +5827,9 @@ EXTERN_C const IID IID_ISpreadsheetProvider;
 EXTERN_C const IID IID_ISpreadsheetItemProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("eaed4660-7b3d-4879-a2e6-365ce603f3d0")
+    
+    _Windows_SupportedOS_Windows8_
     ISpreadsheetItemProvider : public IUnknown
     {
     public:
@@ -5897,8 +5941,9 @@ EXTERN_C const IID IID_ISpreadsheetItemProvider;
 EXTERN_C const IID IID_ITransformProvider2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("4758742f-7ac2-460c-bc48-09fc09308a93")
+    
+    _Windows_SupportedOS_Windows8_
     ITransformProvider2 : public ITransformProvider
     {
     public:
@@ -6108,8 +6153,9 @@ EXTERN_C const IID IID_ITransformProvider2;
 EXTERN_C const IID IID_IDragProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("6aa7bbbb-7ff9-497d-904f-d20b897929d8")
+    
+    _Windows_SupportedOS_Windows8_
     IDragProvider : public IUnknown
     {
     public:
@@ -6234,8 +6280,9 @@ EXTERN_C const IID IID_IDragProvider;
 EXTERN_C const IID IID_IDropTargetProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("bae82bfd-358a-481c-85a0-d8b4d90a5d61")
+    
+    _Windows_SupportedOS_Windows8_
     IDropTargetProvider : public IUnknown
     {
     public:
@@ -6334,8 +6381,9 @@ EXTERN_C const IID IID_IDropTargetProvider;
 EXTERN_C const IID IID_ITextRangeProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("5347ad7b-c355-46f8-aff5-909033582f63")
+    
+    _Windows_SupportedOS_WindowsXP_
     ITextRangeProvider : public IUnknown
     {
     public:
@@ -6706,8 +6754,9 @@ EXTERN_C const IID IID_ITextRangeProvider;
 EXTERN_C const IID IID_ITextProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("3589c92c-63f3-4367-99bb-ada653b77cf2")
+    
+    _Windows_SupportedOS_WindowsXP_
     ITextProvider : public IUnknown
     {
     public:
@@ -6866,8 +6915,9 @@ EXTERN_C const IID IID_ITextProvider;
 EXTERN_C const IID IID_ITextProvider2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("0dc5e6ed-3e16-4bf1-8f9a-a979878bc195")
+    
+    _Windows_SupportedOS_Windows8_
     ITextProvider2 : public ITextProvider
     {
     public:
@@ -7033,8 +7083,9 @@ EXTERN_C const IID IID_ITextProvider2;
 EXTERN_C const IID IID_ITextEditProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("EA3605B4-3A05-400E-B5F9-4E91B40F6176")
+    
+    _Windows_SupportedOS_Windows81_
     ITextEditProvider : public ITextProvider
     {
     public:
@@ -7192,8 +7243,9 @@ EXTERN_C const IID IID_ITextEditProvider;
 EXTERN_C const IID IID_ITextRangeProvider2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("9BBCE42C-1921-4F18-89CA-DBA1910A0386")
+    
+    _Windows_SupportedOS_Windows81_
     ITextRangeProvider2 : public ITextRangeProvider
     {
     public:
@@ -7470,8 +7522,9 @@ EXTERN_C const IID IID_ITextRangeProvider2;
 EXTERN_C const IID IID_ITextChildProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("4c2de2b9-c88f-4f88-a111-f1d336b7d1a9")
+    
+    _Windows_SupportedOS_Windows8_
     ITextChildProvider : public IUnknown
     {
     public:
@@ -7661,8 +7714,9 @@ EXTERN_C const IID IID_ICustomNavigationProvider;
 EXTERN_C const IID IID_IUIAutomationPatternInstance;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("c03a7fe4-9431-409f-bed8-ae7c2299bc8d")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationPatternInstance : public IUnknown
     {
     public:
@@ -7781,8 +7835,9 @@ EXTERN_C const IID IID_IUIAutomationPatternInstance;
 EXTERN_C const IID IID_IUIAutomationPatternHandler;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("d97022f3-a947-465e-8b2a-ac4315fa54e8")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationPatternHandler : public IUnknown
     {
     public:
@@ -7897,8 +7952,9 @@ EXTERN_C const IID IID_IUIAutomationPatternHandler;
 EXTERN_C const IID IID_IUIAutomationRegistrar;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("8609c4ec-4a1a-4d88-a357-5a66e060e1cf")
+    
+    _Windows_SupportedOS_Windows7_
     IUIAutomationRegistrar : public IUnknown
     {
     public:
