diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/lmstats.h b/generation/WinSDK/RecompiledIdlHeaders/um/lmstats.h
index 3b4bd799f9..a5aaa741c0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/lmstats.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/lmstats.h
@@ -30,6 +30,7 @@ Notes:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -45,13 +46,15 @@ extern "C" {
 // Function Prototypes - Statistics
 //
 
+_Windows_SupportedOS_WindowsXP_ 
+
 NET_API_STATUS NET_API_FUNCTION
 NetStatisticsGet (
     _In_        LPTSTR  ServerName,
     _In_        LPTSTR  Service,
     _In_        DWORD   Level,
     _In_        DWORD   Options,
-    _Outptr_ LPBYTE *Buffer
+    _Outptr_  LPBYTE *Buffer
     );
 
 //
@@ -158,7 +161,7 @@ typedef struct _STAT_WORKSTATION_0 {
 } STAT_WORKSTATION_0, *PSTAT_WORKSTATION_0, *LPSTAT_WORKSTATION_0;
 #endif
 
-typedef struct _STAT_SERVER_0 {
+typedef struct   _STAT_SERVER_0 {
      DWORD          sts0_start;
      DWORD          sts0_fopens;
      DWORD          sts0_devopens;
