diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WinNls.h b/generation/WinSDK/RecompiledIdlHeaders/um/WinNls.h
index 2a104ac5a0..0f08d8fe0b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WinNls.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WinNls.h
@@ -18,6 +18,7 @@ Abstract:
 #define _WINNLS_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 
 #ifdef __cplusplus
@@ -1045,6 +1046,177 @@ extern "C" {
 
 #endif
 
+#if defined(WIN32METADATA) && !defined(RC_INVOKED)
+enum [[clang::flag_enum]]
+#if defined(MAP_COMPOSITE)
+    
+#endif
+#if defined(MAP_EXPAND_LIGATURES)
+    
+#endif
+#if defined(MAP_FOLDCZONE)
+    
+#endif
+#if defined(MAP_FOLDDIGITS)
+    
+#endif
+#if defined(MAP_PRECOMPOSED)
+    
+#endif
+    FOLD_STRING_MAP_FLAGS : unsigned long
+{
+};
+
+enum
+#if defined(DATE_SHORTDATE)
+    
+#endif
+#if defined(DATE_LONGDATE)
+    
+#endif
+#if defined(DATE_YEARMONTH)
+    
+#endif
+#if defined(DATE_MONTHDAY)
+    
+#endif
+#if defined(DATE_AUTOLAYOUT)
+    
+#endif
+#if defined(DATE_LTRREADING)
+    
+#endif
+#if defined(DATE_RTLREADING)
+    
+#endif
+#if defined(DATE_USE_ALT_CALENDAR)
+    
+#endif
+    ENUM_DATE_FORMATS_FLAGS : unsigned long
+{
+};
+
+enum [[clang::flag_enum]]
+#if defined(TIME_NOMINUTESORSECONDS)
+    
+#endif
+#if defined(TIME_NOSECONDS)
+    
+#endif
+#if defined(TIME_NOTIMEMARKER)
+    
+#endif
+#if defined(TIME_FORCE24HOURFORMAT)
+    
+#endif
+#if defined(LOCALE_USE_CP_ACP)
+    
+#endif
+    TIME_FORMAT_FLAGS : unsigned long
+{
+};
+
+enum
+#if defined(LGRPID_INSTALLED)
+    
+#endif
+#if defined(LGRPID_SUPPORTED)
+    
+#endif
+    ENUM_SYSTEM_LANGUAGE_GROUPS_FLAGS : unsigned long
+{
+};
+
+enum [[clang::flag_enum]]
+#if defined(MB_COMPOSITE)
+    
+#endif
+#if defined(MB_ERR_INVALID_CHARS)
+    
+#endif
+#if defined(MB_PRECOMPOSED)
+    
+#endif
+#if defined(MB_USEGLYPHCHARS)
+    
+#endif
+    MULTI_BYTE_TO_WIDE_CHAR_FLAGS : unsigned long
+{
+};
+
+enum [[clang::flag_enum]]
+#if defined(LINGUISTIC_IGNORECASE)
+    
+#endif
+#if defined(LINGUISTIC_IGNOREDIACRITIC)
+    
+#endif
+#if defined(NORM_IGNORECASE)
+    
+#endif
+#if defined(NORM_IGNOREKANATYPE)
+    
+#endif
+#if defined(NORM_IGNORENONSPACE)
+    
+#endif
+#if defined(NORM_IGNORESYMBOLS)
+    
+#endif
+#if defined(NORM_IGNOREWIDTH)
+    
+#endif
+#if defined(NORM_LINGUISTIC_CASING)
+    
+#endif
+#if defined(SORT_DIGITSASNUMBERS)
+    
+#endif
+#if defined(SORT_STRINGSORT)
+    
+#endif
+    COMPARE_STRING_FLAGS : unsigned long
+{
+};
+
+enum
+#if defined(LCID_INSTALLED)
+    
+#endif
+#if defined(LCID_SUPPORTED)
+    
+#endif
+    IS_VALID_LOCALE_FLAGS : unsigned long
+{
+};
+
+enum
+#if defined(CP_INSTALLED)
+    
+#endif
+#if defined(CP_SUPPORTED)
+    
+#endif
+    ENUM_SYSTEM_CODE_PAGES_FLAGS : unsigned long
+{
+};
+
+enum
+#if defined(CSTR_LESS_THAN)
+    
+#endif
+#if defined(CSTR_EQUAL)
+    
+#endif
+#if defined(CSTR_GREATER_THAN)
+    
+#endif
+    COMPARESTRING_RESULT : int
+{
+};
+
+#endif
+
 ////////////////////////////////////////////////////////////////////////////
 //
 //  Typedefs
@@ -1611,12 +1783,16 @@ typedef struct _FILEMUIINFO {
 //  Applications should use Unicode (WCHAR / UTF-16 &/or UTF-8)
 //
 
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 BOOL
 WINAPI
 IsValidCodePage(
     _In_ UINT  CodePage);
 
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 UINT
 WINAPI
@@ -1628,6 +1804,8 @@ GetACP(void);
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 UINT
 WINAPI
@@ -1639,7 +1817,10 @@ GetOEMCP(void);
 #pragma region Desktop or Pc Family or OneCore or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP |WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accuratedly and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Windows_SupportedOS_Windows2000_ DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accuratedly and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1653,7 +1834,10 @@ GetCPInfo(
 #pragma region Desktop or Pc Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP |WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accurately and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accurately and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1661,7 +1845,10 @@ GetCPInfoExA(
     _In_ UINT          CodePage,
     _In_ DWORD         dwFlags,
     _Out_ LPCPINFOEXA  lpCPInfoEx);
-DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accurately and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ DEPRECATED("Use Unicode. The information in this structure cannot represent all encodings accurately and may be unreliable on many machines. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1685,7 +1872,11 @@ GetCPInfoExW(
 #pragma region Desktop or OneCore or Application or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
-DEPRECATED("CompareStringEx is preferred. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ DEPRECATED("CompareStringEx is preferred. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Win32_AssociatedEnum_(COMPARESTRING_RESULT)
+
+
+
 WINBASEAPI
 int
 WINAPI
@@ -1739,6 +1930,9 @@ CompareString(
 #if (WINVER >= 0x0600)
 
 // DEPRECATED: FindNLSStringEx is preferred
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1760,6 +1954,9 @@ FindNLSString(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // DEPRECATED: LCMapStringEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1775,6 +1972,9 @@ LCMapStringW(
 #endif
 
 // DEPRECATED: Use Unicode, LCMapStringEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1796,6 +1996,9 @@ LCMapStringA(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // DEPRECATED: GetLocaleInfoEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1810,6 +2013,9 @@ GetLocaleInfoW(
 #endif
 
 // DEPRECATED: Use Unicode. GetLocaleInfoEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int 
 WINAPI 
@@ -1830,6 +2036,9 @@ GetLocaleInfoA(
 #pragma region Application or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1837,6 +2046,9 @@ SetLocaleInfoA(
     _In_ LCID     Locale,
     _In_ LCTYPE   LCType,
     _In_ LPCSTR  lpLCData);
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1853,6 +2065,9 @@ SetLocaleInfoW(
 #if (WINVER >= 0x040A)
 
 // DEPRECATED: GetCalendarInfoEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1864,6 +2079,9 @@ GetCalendarInfoA(
     _In_ int      cchData,
     _Out_opt_ LPDWORD  lpValue);
 // DEPRECATED: GetCalendarInfoEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1880,6 +2098,9 @@ GetCalendarInfoW(
 #define GetCalendarInfo  GetCalendarInfoA
 #endif // !UNICODE
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1888,6 +2109,9 @@ SetCalendarInfoA(
     _In_ CALID    Calendar,
     _In_ CALTYPE  CalType,
     _In_ LPCSTR  lpCalData);
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1940,7 +2164,10 @@ LoadStringByReference(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
-DEPRECATED("Use Unicode. The information provided by this structure is inaccurate for some encodings and may be unreliable on many machines.")
+_Windows_SupportedOS_Windows2000_ DEPRECATED("Use Unicode. The information provided by this structure is inaccurate for some encodings and may be unreliable on many machines.")
+
+
+_Win32_SetLastError_
 WINBASEAPI         
 BOOL               
 WINAPI             
@@ -1948,7 +2175,10 @@ IsDBCSLeadByte(
     _In_ BYTE  TestChar 
     );
 
-DEPRECATED("Use Unicode. The information provided by this structure is inaccurate for some encodings and may be unreliable on many machines.")
+_Windows_SupportedOS_Windows2000_ DEPRECATED("Use Unicode. The information provided by this structure is inaccurate for some encodings and may be unreliable on many machines.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1967,6 +2197,9 @@ IsDBCSLeadByteEx(
 
 // Use of Locale Names is preferred, LCIDs are deprecated.
 // This function is provided to enable compatibility with legacy data sets only.
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 LCID
 WINAPI
@@ -1976,6 +2209,9 @@ LocaleNameToLCID(
 
 // Use of Locale Names is preferred, LCIDs are deprecated.
 // This function is provided to enable compatibility with legacy data sets only.
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -1995,6 +2231,9 @@ LCIDToLocaleName(
 
 // DEPRECATED: GetDurationFormatEx is preferred
 #if (WINVER >= 0x0600)
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2015,6 +2254,9 @@ GetDurationFormat(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // DEPRECATED: GetNumberFormatEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2026,6 +2268,9 @@ GetNumberFormatA(
     _Out_writes_opt_(cchNumber) LPSTR          lpNumberStr,
     _In_ int              cchNumber);
 // DEPRECATED: GetNumberFormatEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2043,6 +2288,9 @@ GetNumberFormatW(
 #endif // !UNICODE
 
 // DEPRECATED: GetCurrencyFormatEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2054,6 +2302,9 @@ GetCurrencyFormatA(
     _Out_writes_opt_(cchCurrency) LPSTR            lpCurrencyStr,
     _In_ int                cchCurrency);
 // DEPRECATED: GetCurrencyFormatEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2077,6 +2328,9 @@ GetCurrencyFormatW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 // DEPRECATED: EnumCalendarInfoExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2086,6 +2340,9 @@ EnumCalendarInfoA(
     _In_ CALID             Calendar,
     _In_ CALTYPE           CalType);
 // DEPRECATED: EnumCalendarInfoExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2102,6 +2359,9 @@ EnumCalendarInfoW(
 
 #if(WINVER >= 0x0500)
 // DEPRECATED: EnumCalendarInfoExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2111,6 +2371,9 @@ EnumCalendarInfoExA(
     _In_ CALID               Calendar,
     _In_ CALTYPE             CalType);
 // DEPRECATED: EnumCalendarInfoExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2127,21 +2390,27 @@ EnumCalendarInfoExW(
 #endif /* WINVER >= 0x0500 */
 
 // DEPRECATED: EnumTimeFormatsEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumTimeFormatsA(
     _In_ TIMEFMT_ENUMPROCA lpTimeFmtEnumProc,
     _In_ LCID              Locale,
-    _In_ DWORD             dwFlags);
+    _In_ _Win32_AssociatedEnum_(TIME_FORMAT_FLAGS) DWORD             dwFlags);
 // DEPRECATED: EnumTimeFormatsEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumTimeFormatsW(
     _In_ TIMEFMT_ENUMPROCW lpTimeFmtEnumProc,
     _In_ LCID              Locale,
-    _In_ DWORD             dwFlags);
+    _In_ _Win32_AssociatedEnum_(TIME_FORMAT_FLAGS) DWORD             dwFlags);
 #ifdef UNICODE
 #define EnumTimeFormats  EnumTimeFormatsW
 #else
@@ -2149,6 +2418,9 @@ EnumTimeFormatsW(
 #endif // !UNICODE
 
 // DEPRECATED: EnumDateFormatsExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2157,6 +2429,9 @@ EnumDateFormatsA(
     _In_ LCID              Locale,
     _In_ DWORD             dwFlags);
 // DEPRECATED: EnumDateFormatsExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2172,6 +2447,9 @@ EnumDateFormatsW(
 
 #if(WINVER >= 0x0500)
 // DEPRECATED: EnumDateFormatsExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2180,6 +2458,9 @@ EnumDateFormatsExA(
     _In_ LCID                Locale,
     _In_ DWORD               dwFlags);
 // DEPRECATED: EnumDateFormatsExEx is preferred
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2201,16 +2482,21 @@ EnumDateFormatsExW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 #if(WINVER >= 0x0500)
-DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Windows_SupportedOS_Windows2000_ DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
 WINBASEAPI
 BOOL
 WINAPI
 IsValidLanguageGroup(
     _In_ LGRPID  LanguageGroup,
-    _In_ DWORD   dwFlags);
+    _In_ _Win32_AssociatedEnum_(ENUM_SYSTEM_LANGUAGE_GROUPS_FLAGS) DWORD   dwFlags);
 #endif /* WINVER >= 0x0500 */
 
 // DEPRECATED: GetNLSVersionEx is preferred
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2226,12 +2512,14 @@ GetNLSVersion(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 // DEPRECATED: IsValidLocaleName is preferred
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 BOOL
 WINAPI
 IsValidLocale(
     _In_ LCID   Locale,
-    _In_ DWORD  dwFlags);
+    _In_ _Win32_AssociatedEnum_(IS_VALID_LOCALE_FLAGS) DWORD  dwFlags);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM) */
 #pragma endregion
@@ -2240,24 +2528,30 @@ IsValidLocale(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // GetGeoInfoEx is preferred where available
-DEPRECATED("The GeoID concept is obsolete. Use GetGeoInfoEx instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ DEPRECATED("The GeoID concept is obsolete. Use GetGeoInfoEx instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
 GetGeoInfoA(
     _In_ GEOID       Location,
-    _In_ GEOTYPE     GeoType,
+    _In_ _Win32_AssociatedEnum_(SYSGEOTYPE) GEOTYPE     GeoType,
     _Out_writes_opt_(cchData) LPSTR     lpGeoData,
     _In_ int         cchData,
     _In_ LANGID      LangId);
 // GetGeoInfoEx is preferred where available
-DEPRECATED("The GeoID concept is obsolete. Use GetGeoInfoEx instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ DEPRECATED("The GeoID concept is obsolete. Use GetGeoInfoEx instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
 GetGeoInfoW(
     _In_ GEOID       Location,
-    _In_ GEOTYPE     GeoType,
+    _In_ _Win32_AssociatedEnum_(SYSGEOTYPE) GEOTYPE     GeoType,
     _Out_writes_opt_(cchData) LPWSTR     lpGeoData,
     _In_ int         cchData,
     _In_ LANGID      LangId);
@@ -2268,12 +2562,15 @@ GetGeoInfoW(
 #endif // !UNICODE
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS3)
+_Windows_SupportedOS_16299_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
 GetGeoInfoEx(
     _In_ PWSTR       location,
-    _In_ GEOTYPE     geoType,
+    _In_ _Win32_AssociatedEnum_(SYSGEOTYPE) GEOTYPE     geoType,
     _Out_writes_opt_(geoDataCount) PWSTR    geoData,
     _In_ int         geoDataCount);
 #endif
@@ -2285,7 +2582,10 @@ GetGeoInfoEx(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM)
 
 // EnumSystemGeoNames is preferred where available
-DEPRECATED("The GeoID concept is obsolete. Use EnumSystemGoNames instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Windows_SupportedOS_WindowsXP_ DEPRECATED("The GeoID concept is obsolete. Use EnumSystemGoNames instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2295,6 +2595,9 @@ EnumSystemGeoID(
     _In_ GEO_ENUMPROC    lpGeoEnumProc);
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS3)
+_Windows_SupportedOS_16299_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2311,18 +2614,23 @@ EnumSystemGeoNames(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 // GetUserDefaultGeoName is preferred where available
-DEPRECATED("The GeoID concept is obsolete. Use GetUserDefaultGeoName instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Windows_SupportedOS_WindowsXP_ DEPRECATED("The GeoID concept is obsolete. Use GetUserDefaultGeoName instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
 WINBASEAPI
 GEOID
 WINAPI
 GetUserGeoID(
-    _In_ GEOCLASS    GeoClass);
+    _In_ _Win32_AssociatedEnum_(SYSGEOCLASS) GEOCLASS    GeoClass);
 
 /**
  * Note: This API was added in the Windows 10 Fall Creators Update.
  * (Please use this API instead of calling GetUserGeoID.)
  *
  */
+_Windows_SupportedOS_16299_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2339,7 +2647,9 @@ GetUserDefaultGeoName(
 
 // GetUserDefaultGeoName is preferred where available
 // Applications are recommended to not change user settings themselves.
-DEPRECATED("The GeoID concept is obsolete. Use SetUserGeoName instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ DEPRECATED("The GeoID concept is obsolete. Use SetUserGeoName instead. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
 WINBASEAPI
 BOOL
 WINAPI
@@ -2348,6 +2658,9 @@ SetUserGeoID(
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS3)
 // Applications are recommended to not change user settings themselves.
+_Windows_SupportedOS_16299_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2356,6 +2669,8 @@ SetUserGeoName(
 #endif
 
 // DEPRECATED: Please use ResolveLocaleName
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LCID
 WINAPI
@@ -2364,6 +2679,8 @@ ConvertDefaultLocale(
 
 #if(WINVER >= 0x0500)
 // DEPRECATED: Please use the user's language profile.
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LANGID
 WINAPI
@@ -2377,11 +2694,15 @@ GetSystemDefaultUILanguage(void);
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LCID
 WINAPI
 GetThreadLocale(void);
 
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 BOOL
 WINAPI
@@ -2397,6 +2718,8 @@ SetThreadLocale(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 #if(WINVER >= 0x0500)
 // DEPRECATED: Please use the user's language profile.
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LANGID
 WINAPI
@@ -2404,6 +2727,8 @@ GetUserDefaultUILanguage(void);
 #endif /* WINVER >= 0x0500 */
 
 // DEPRECATED: Please use GetUserDefaultLocaleName
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LANGID
 WINAPI
@@ -2416,18 +2741,24 @@ GetUserDefaultLangID(void);
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
 // DEPRECATED: Please use GetUserDefaultLocaleName or the user's Language Profile
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LANGID
 WINAPI
 GetSystemDefaultLangID(void);
 
 // DEPRECATED: Please use GetUserDefaultLocaleName or the user's Language Profile
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LCID
 WINAPI
 GetSystemDefaultLCID(void);
 
 // DEPRECATED: Please use GetUserDefaultLocaleName
+_Windows_SupportedOS_Windows2000_ 
+
 WINBASEAPI
 LCID
 WINAPI
@@ -2446,6 +2777,9 @@ GetUserDefaultLCID(void);
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsXP_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 LANGID
 WINAPI
@@ -2454,28 +2788,32 @@ SetThreadUILanguage(_In_ LANGID LangId);
 
 #if (WINVER >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 LANGID
 WINAPI
 GetThreadUILanguage(void);
 
+_Windows_SupportedOS_Windows7_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetProcessPreferredUILanguages(
     _In_ DWORD dwFlags,
     _Out_ PULONG pulNumLanguages,
-    _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
     _Inout_ PULONG pcchLanguagesBuffer
 );
 
 
+_Windows_SupportedOS_Windows7_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetProcessPreferredUILanguages(
     _In_        DWORD dwFlags,
-    _In_opt_    PCZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _In_opt_    PCZZWSTR pwszLanguagesBuffer,
     _Out_opt_   PULONG pulNumLanguages
 );
 
@@ -2487,13 +2825,16 @@ SetProcessPreferredUILanguages(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_PHONE_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if(WINVER >= 0x0600)
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetUserPreferredUILanguages (
     _In_ DWORD dwFlags,
     _Out_ PULONG pulNumLanguages,
-    _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
     _Inout_ PULONG pcchLanguagesBuffer
 );
 
@@ -2507,37 +2848,48 @@ GetUserPreferredUILanguages (
 
 #if(WINVER >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetSystemPreferredUILanguages (
     _In_ DWORD dwFlags,
     _Out_ PULONG pulNumLanguages,
-    _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
     _Inout_ PULONG pcchLanguagesBuffer
 );
 
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetThreadPreferredUILanguages(
     _In_ DWORD dwFlags,
     _Out_ PULONG pulNumLanguages,
-    _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _Out_writes_opt_(*pcchLanguagesBuffer) PZZWSTR pwszLanguagesBuffer,
     _Inout_ PULONG pcchLanguagesBuffer
 );
 
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 BOOL
 WINAPI
 SetThreadPreferredUILanguages(
     _In_        DWORD dwFlags,
-    _In_opt_    PCZZWSTR pwszLanguagesBuffer,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _In_opt_    PCZZWSTR pwszLanguagesBuffer,
     _Out_opt_   PULONG pulNumLanguages
 );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return!=FALSE)
 BOOL
@@ -2548,6 +2900,9 @@ GetFileMUIInfo(
     _Inout_updates_bytes_to_opt_(*pcbFileMUIInfo,*pcbFileMUIInfo) PFILEMUIINFO    pFileMUIInfo,
     _Inout_             DWORD*          pcbFileMUIInfo);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2562,13 +2917,16 @@ GetFileMUIPath(
 );
 
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 GetUILanguageInfo(
     _In_ DWORD dwFlags,
-    _In_ PCZZWSTR pwmszLanguage,
-    _Out_writes_opt_(*pcchFallbackLanguages) PZZWSTR pwszFallbackLanguages,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _In_ PCZZWSTR pwmszLanguage,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _Out_writes_opt_(*pcchFallbackLanguages) PZZWSTR pwszFallbackLanguages,
     _Inout_opt_ PDWORD pcchFallbackLanguages,
     _Out_ PDWORD pAttributes
 );
@@ -2590,7 +2948,7 @@ BOOL
 WINAPI
 SetThreadPreferredUILanguages2(
     _In_       ULONG flags,
-    _In_opt_   PCZZWSTR languages,
+    _WIN32META_ANNOTATION_("win32metadata:not_null_terminated") _In_opt_   PCZZWSTR languages,
     _Out_opt_  PULONG numLanguagesSet,
     _Out_opt_  HSAVEDUILANGUAGES* snapshot);
 
@@ -2609,14 +2967,14 @@ RestoreThreadPreferredUILanguages(_In_ const HSAVEDUILANGUAGES snapshot);
 
 #if(WINVER >= 0x0600)
 
-WINBASEAPI
+_Windows_SupportedOS_Windows7_ WINBASEAPI
 BOOL
 WINAPI
 NotifyUILanguageChange(
     _In_        DWORD dwFlags,
     _In_opt_    PCWSTR pcwstrNewLanguage,
     _In_opt_    PCWSTR pcwstrPreviousLanguage,
-    _In_        DWORD dwReserved,
+    _Reserved_ _In_        DWORD dwReserved,
     _Out_opt_   PDWORD pdwStatusRtrn
 );
 
@@ -2633,7 +2991,10 @@ NotifyUILanguageChange(
 #pragma region Desktop or OneCore or Application or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP | WINAPI_PARTITION_GAMES)
 
-WINBASEAPI
+
+
+
+_WIN32META_ANNOTATION_("win32metadata:ansi") WINBASEAPI
 BOOL
 WINAPI
 GetStringTypeExA(
@@ -2658,6 +3019,9 @@ GetStringTypeExA(
 //
 //        GetStringTypeEx (above) should be used instead.
 //
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2668,11 +3032,14 @@ GetStringTypeA(
     _In_ int      cchSrc,
     _Out_ LPWORD  lpCharType);
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
 FoldStringA(
-    _In_ DWORD    dwMapFlags,
+    _In_ _Win32_AssociatedEnum_(FOLD_STRING_MAP_FLAGS) DWORD    dwMapFlags,
     _In_reads_(cchSrc) LPCSTR  lpSrcStr,
     _In_ int      cchSrc,
     _Out_writes_opt_(cchDest) LPSTR  lpDestStr,
@@ -2690,6 +3057,9 @@ FoldStringA(
 #if(WINVER >= 0x0500)
 
 // DEPRECATED, please use Locale Names and call EnumSystemLocalesEx
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2697,6 +3067,9 @@ EnumSystemLocalesA(
     _In_ LOCALE_ENUMPROCA lpLocaleEnumProc,
     _In_ DWORD            dwFlags);
 // DEPRECATED, please use Locale Names and call EnumSystemLocalesEx
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2719,21 +3092,27 @@ EnumSystemLocalesW(
 
 #if(WINVER >= 0x0500)
 
-DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumSystemLanguageGroupsA(
     _In_ LANGUAGEGROUP_ENUMPROCA lpLanguageGroupEnumProc,
-    _In_ DWORD                   dwFlags,
+    _In_ _Win32_AssociatedEnum_(ENUM_SYSTEM_LANGUAGE_GROUPS_FLAGS) DWORD                   dwFlags,
     _In_ LONG_PTR                lParam);
-DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumSystemLanguageGroupsW(
     _In_ LANGUAGEGROUP_ENUMPROCW lpLanguageGroupEnumProc,
-    _In_ DWORD                   dwFlags,
+    _In_ _Win32_AssociatedEnum_(ENUM_SYSTEM_LANGUAGE_GROUPS_FLAGS) DWORD                   dwFlags,
     _In_ LONG_PTR                lParam);
 #ifdef UNICODE
 #define EnumSystemLanguageGroups  EnumSystemLanguageGroupsW
@@ -2741,7 +3120,10 @@ EnumSystemLanguageGroupsW(
 #define EnumSystemLanguageGroups  EnumSystemLanguageGroupsA
 #endif // !UNICODE
 
-DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2750,7 +3132,10 @@ EnumLanguageGroupLocalesA(
     _In_ LGRPID                    LanguageGroup,
     _In_ DWORD                     dwFlags,
     _In_ LONG_PTR                  lParam);
-DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ DEPRECATED("The Language Group concept is obsolete and no longer supported. Set DISABLE_NLS_DEPRECATION to disable this warning.")
+
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2772,6 +3157,9 @@ EnumLanguageGroupLocalesW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 // DEPRECATED: use the user language profile instead.
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2780,6 +3168,9 @@ EnumUILanguagesA(
     _In_ DWORD                dwFlags,
     _In_ LONG_PTR             lParam);
 // DEPRECATED: use the user language profile instead.
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2802,19 +3193,25 @@ EnumUILanguagesW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP  | WINAPI_PARTITION_SYSTEM)
 
 // Please use Unicode instead.  Use of other code pages/encodings is discouraged.
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumSystemCodePagesA(
     _In_ CODEPAGE_ENUMPROCA lpCodePageEnumProc,
-    _In_ DWORD              dwFlags);
+    _In_ _Win32_AssociatedEnum_(ENUM_SYSTEM_CODE_PAGES_FLAGS) DWORD              dwFlags);
 // Please use Unicode instead.  Use of other code pages/encodings is discouraged.
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows2000_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumSystemCodePagesW(
     _In_ CODEPAGE_ENUMPROCW lpCodePageEnumProc,
-    _In_ DWORD              dwFlags);
+    _In_ _Win32_AssociatedEnum_(ENUM_SYSTEM_CODE_PAGES_FLAGS) DWORD              dwFlags);
 #ifdef UNICODE
 #define EnumSystemCodePages  EnumSystemCodePagesW
 #else
@@ -2836,6 +3233,7 @@ EnumSystemCodePagesW(
 //
 // IDN (International Domain Name) Functions
 //
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINNORMALIZEAPI
 int
 WINAPI IdnToAscii(_In_                           DWORD    dwFlags,
@@ -2844,6 +3242,7 @@ WINAPI IdnToAscii(_In_                           DWORD    dwFlags,
                   _Out_writes_opt_(cchASCIIChar) LPWSTR   lpASCIICharStr,
                   _In_                        	 int      cchASCIIChar);
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINNORMALIZEAPI
 int
 WINAPI IdnToUnicode(_In_                         	 DWORD   dwFlags,
@@ -2852,6 +3251,7 @@ WINAPI IdnToUnicode(_In_                         	 DWORD   dwFlags,
                     _Out_writes_opt_(cchUnicodeChar) LPWSTR  lpUnicodeCharStr,
                     _In_                         	 int     cchUnicodeChar);
 
+_Windows_SupportedOS_WindowsVista_ _Win32_ImportLibrary_("kernel32.dll") _Win32_SetLastError_
 WINNORMALIZEAPI
 int
 WINAPI IdnToNameprepUnicode(_In_                            	DWORD   dwFlags,
@@ -2870,6 +3270,7 @@ WINAPI IdnToNameprepUnicode(_In_                            	DWORD   dwFlags,
 
 #if (WINVER >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_ImportLibrary_("kernel32.dll") _Win32_SetLastError_
 WINNORMALIZEAPI
 int
 WINAPI NormalizeString( _In_                          NORM_FORM NormForm,
@@ -2878,12 +3279,16 @@ WINAPI NormalizeString( _In_                          NORM_FORM NormForm,
                         _Out_writes_opt_(cwDstLength) LPWSTR    lpDstString,
                         _In_                          int       cwDstLength );
 
+_Windows_SupportedOS_WindowsVista_ _Win32_ImportLibrary_("kernel32.dll") _Win32_SetLastError_
 WINNORMALIZEAPI
 BOOL
 WINAPI IsNormalizedString( _In_                   NORM_FORM NormForm,
                            _In_reads_(cwLength)  LPCWSTR   lpString,
                            _In_                   int       cwLength );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI VerifyScripts(
@@ -2893,6 +3298,9 @@ WINAPI VerifyScripts(
     _In_    LPCWSTR lpTestScripts,      // test scripts string
     _In_    int     cchTestScripts);    // size of test list string
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI GetStringScripts(
@@ -2920,6 +3328,9 @@ WINAPI GetStringScripts(
 #define LOCALE_NAME_INVARIANT               L""
 #define LOCALE_NAME_SYSTEM_DEFAULT          L"!x-sys-default-locale"
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2936,13 +3347,14 @@ GetLocaleInfoEx(
 #pragma region Desktop or PC Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
 GetCalendarInfoEx(
     _In_opt_ LPCWSTR lpLocaleName,
     _In_ CALID Calendar,
-    _In_opt_ LPCWSTR lpReserved,
+    _Reserved_ _In_opt_ LPCWSTR lpReserved,
     _In_ CALTYPE CalType,
     _Out_writes_opt_(cchData) LPWSTR lpCalData,
     _In_ int cchData,
@@ -2956,6 +3368,9 @@ GetCalendarInfoEx(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
 #ifndef GetDurationFormatEx_DEFINED
+_Windows_SupportedOS_WindowsVista_
+_Windows_SupportedOS_Server2008_
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2976,6 +3391,9 @@ GetDurationFormatEx(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -2988,6 +3406,7 @@ GetNumberFormatEx(
     _In_ int cchNumber
 );
 
+_Windows_SupportedOS_Windows2000_ _Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -3000,6 +3419,9 @@ GetCurrencyFormatEx(
     _In_ int cchCurrency
 );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -3014,6 +3436,9 @@ GetUserDefaultLocaleName(
 #pragma region Desktop or PC Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -3022,6 +3447,9 @@ GetSystemDefaultLocaleName(
     _In_ int cchLocaleName
 );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3032,6 +3460,9 @@ IsNLSDefinedString(
     _In_reads_(cchStr) LPCWSTR          lpString,
     _In_ INT              cchStr);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3042,6 +3473,8 @@ GetNLSVersionEx(
 );
 
 #if (WINVER >= _WIN32_WINNT_WIN8)
+_Windows_SupportedOS_Windows8_ 
+
 WINBASEAPI
 DWORD
 WINAPI
@@ -3058,6 +3491,9 @@ IsValidNLSVersion(
 #pragma region Application Family or OneCore or Gamaes Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -3070,7 +3506,7 @@ FindNLSStringEx(
     _In_ int cchValue,
     _Out_opt_ LPINT pcchFound,
     _In_opt_ LPNLSVERSIONINFO lpVersionInformation,
-    _In_opt_ LPVOID lpReserved,
+    _Reserved_ _In_opt_ LPVOID lpReserved,
     _In_opt_ LPARAM sortHandle
 );
 
@@ -3081,6 +3517,9 @@ _When_((dwMapFlags & (LCMAP_SORTKEY | LCMAP_BYTEREV)) != 0, _At_((LPBYTE) lpDest
 #endif
 _When_(cchSrc != -1,  _At_((WCHAR *) lpSrcStr, _Out_writes_opt_(cchSrc)))
 _When_(cchDest != -1, _At_((WCHAR *) lpDestStr, _Out_writes_opt_(cchDest)))
+
+
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
@@ -3092,10 +3531,12 @@ LCMapStringEx(
     _Out_writes_opt_(cchDest) LPWSTR lpDestStr,
     _In_ int cchDest,
     _In_opt_ LPNLSVERSIONINFO lpVersionInformation,
-    _In_opt_ LPVOID lpReserved,
+    _Reserved_ _In_opt_ LPVOID lpReserved,
     _In_opt_ LPARAM sortHandle
 );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 WINBASEAPI
 BOOL
 WINAPI
@@ -3111,6 +3552,9 @@ IsValidLocaleName(
 
 typedef BOOL (CALLBACK* CALINFO_ENUMPROCEXEX)(LPWSTR, CALID, LPWSTR, LPARAM);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3118,25 +3562,31 @@ EnumCalendarInfoExEx(
     _In_ CALINFO_ENUMPROCEXEX pCalInfoEnumProcExEx,
     _In_opt_ LPCWSTR lpLocaleName,
     _In_ CALID Calendar,
-    _In_opt_ LPCWSTR lpReserved,
+    _Reserved_ _In_opt_ LPCWSTR lpReserved,
     _In_ CALTYPE CalType,
     _In_ LPARAM lParam
 );
 
 typedef BOOL (CALLBACK* DATEFMT_ENUMPROCEXEX)(LPWSTR, CALID, LPARAM);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 EnumDateFormatsExEx(
     _In_ DATEFMT_ENUMPROCEXEX lpDateFmtEnumProcExEx,
     _In_opt_ LPCWSTR lpLocaleName,
-    _In_ DWORD dwFlags,
+    _In_ _Win32_AssociatedEnum_(ENUM_DATE_FORMATS_FLAGS) DWORD dwFlags,
     _In_ LPARAM lParam
 );
 
 typedef BOOL (CALLBACK* TIMEFMT_ENUMPROCEX)(LPWSTR, LPARAM);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3155,6 +3605,9 @@ EnumTimeFormatsEx(
 
 typedef BOOL (CALLBACK* LOCALE_ENUMPROCEX)(LPWSTR, DWORD, LPARAM);
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3162,7 +3615,7 @@ EnumSystemLocalesEx(
     _In_ LOCALE_ENUMPROCEX lpLocaleEnumProcEx,
     _In_ DWORD dwFlags,
     _In_ LPARAM lParam,
-    _In_opt_ LPVOID lpReserved
+    _Reserved_ _In_opt_ LPVOID lpReserved
 );
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PC_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
@@ -3175,6 +3628,7 @@ EnumSystemLocalesEx(
 #pragma region Application Family or OneCore or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Windows_SupportedOS_Windows7_ _Win32_SetLastError_
 WINBASEAPI
 int
 WINAPI
