diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h b/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
index dc9d2dad8d..7d8c663812 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/UserEnv.h
@@ -13,6 +13,9 @@
 #define _INC_USERENV
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -70,12 +73,16 @@ extern "C" {
 #define PI_NOUI         0x00000001      // Prevents displaying of messages
 #define PI_APPLYPOLICY  0x00000002      // Apply NT4 style policy
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
 LoadUserProfileA(
     _In_    HANDLE          hToken,
     _Inout_ LPPROFILEINFOA  lpProfileInfo);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -104,6 +111,8 @@ LoadUserProfileW(
 //
 //=============================================================================
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -130,12 +139,16 @@ UnloadUserProfile(
 //
 //=============================================================================
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
 GetProfilesDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -176,6 +189,8 @@ GetProfilesDirectoryW(
 #define PT_MANDATORY                 0x00000004      // The loaded profile is mandatory.
 #define PT_ROAMING_PREEXISTING       0x00000008      // The loaded roaming profile is not a brand new one - it was obtained from roamed location
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -202,6 +217,8 @@ GetProfileType(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -209,6 +226,8 @@ DeleteProfileA (
     _In_        LPCSTR    lpSidString,
     _In_opt_    LPCSTR    lpProfilePath,
     _In_opt_    LPCSTR    lpComputerName);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -249,6 +268,7 @@ DeleteProfileW (
 
 #if(WINVER >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 HRESULT
 WINAPI
@@ -281,12 +301,16 @@ CreateProfile(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
 GetDefaultUserProfileDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -322,12 +346,16 @@ GetDefaultUserProfileDirectoryW(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
 GetAllUsersProfileDirectoryA(
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -362,6 +390,8 @@ GetAllUsersProfileDirectoryW(
 //
 //=============================================================================
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -369,6 +399,8 @@ GetUserProfileDirectoryA(
     _In_                            HANDLE  hToken,
     _Out_writes_opt_(*lpcchSize)    LPSTR lpProfileDir,
     _Inout_                         LPDWORD lpcchSize);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -409,6 +441,8 @@ GetUserProfileDirectoryW(
 //
 //=============================================================================
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 BOOL
 WINAPI
 CreateEnvironmentBlock(
@@ -429,6 +463,8 @@ CreateEnvironmentBlock(
 //
 //=============================================================================
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 BOOL
 WINAPI
 DestroyEnvironmentBlock(
@@ -456,6 +492,8 @@ DestroyEnvironmentBlock(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -464,6 +502,8 @@ ExpandEnvironmentStringsForUserA(
     _In_                    LPCSTR    lpSrc,
     _Out_writes_(dwSize)    LPSTR     lpDest,
     _In_                    DWORD       dwSize);
+_Windows_SupportedOS_Windows2000_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -495,6 +535,8 @@ ExpandEnvironmentStringsForUserW(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -522,6 +564,8 @@ RefreshPolicy(
 #define RP_FORCE            1      // Refresh policies without any optimisations.
 #define RP_SYNC             2      // Refresh Policy sync; the call does not return till the time policy processing is completed
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -560,6 +604,8 @@ RefreshPolicyEx(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 HANDLE
 WINAPI
@@ -586,6 +632,8 @@ EnterCriticalPolicySection(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -621,6 +669,8 @@ LeaveCriticalPolicySection(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -645,6 +695,8 @@ RegisterGPNotification(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -833,6 +885,8 @@ typedef PGROUP_POLICY_OBJECTA PGROUP_POLICY_OBJECT;
 #define GPO_LIST_FLAG_NO_WMIFILTERS  0x00000004  // Ignore WMI filters when filtering GPO's
 #define GPO_LIST_FLAG_NO_SECURITYFILTERS 0x00000008 // Ignore security filters
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -843,6 +897,8 @@ GetGPOListA (
     _In_opt_    LPCSTR    lpComputerName,
     _In_        DWORD       dwFlags,
     _Outptr_ PGROUP_POLICY_OBJECTA * pGPOList);
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -878,11 +934,15 @@ GetGPOListW (
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
 FreeGPOListA (
     _In_        PGROUP_POLICY_OBJECTA   pGPOList);
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 BOOL
 WINAPI
@@ -924,6 +984,7 @@ FreeGPOListW (
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 DWORD
 WINAPI
@@ -933,6 +994,7 @@ GetAppliedGPOListA (
     _In_opt_    PSID        pSidUser,
     _In_        GUID        *pGuidExtension,
     _Outptr_ PGROUP_POLICY_OBJECTA   *ppGPOList);
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 DWORD
 WINAPI
@@ -1103,6 +1165,7 @@ typedef DWORD(*PFNGENERATEGROUPPOLICY)(
 
 typedef GUID *REFGPEXTENSIONID;
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 DWORD
 WINAPI
@@ -1130,6 +1193,7 @@ ProcessGroupPolicyCompleted(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 DWORD
 WINAPI
@@ -1172,6 +1236,8 @@ ProcessGroupPolicyCompletedEx(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 USERENVAPI
 HRESULT 
 WINAPI
@@ -1214,6 +1280,7 @@ RsopAccessCheckByType(
 
 #if(WINVER >= 0x0500)
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 HRESULT 
 WINAPI
@@ -1278,6 +1345,7 @@ typedef struct _POLICYSETTINGSTATUSINFO
 //
 //=============================================================================
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1303,6 +1371,7 @@ RsopSetPolicySettingStatus( _In_ DWORD                       dwFlags,
 //
 //=============================================================================
 
+_Windows_SupportedOS_WindowsVista_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1392,6 +1461,7 @@ GenerateGPNotification (
 
 #if(WINVER >= 0x0602)
 
+_Windows_SupportedOS_Windows8_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1423,6 +1493,7 @@ CreateAppContainerProfile(
 
 #if(WINVER >= 0x0602)
 
+_Windows_SupportedOS_Windows8_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1451,6 +1522,7 @@ DeleteAppContainerProfile(
 
 #if(WINVER >= 0x0602)
 
+_Windows_SupportedOS_Windows8_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1480,6 +1552,7 @@ GetAppContainerRegistryLocation(
 
 #if(WINVER >= 0x0602)
 
+_Windows_SupportedOS_Windows8_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1508,6 +1581,7 @@ GetAppContainerFolderPath(
 
 #if(WINVER >= 0x0603)
 
+_Windows_SupportedOS_10240_
 USERENVAPI
 HRESULT
 WINAPI
@@ -1544,6 +1618,7 @@ DeriveRestrictedAppContainerSidFromAppContainerSidAndRestrictedName(
 
 #if(WINVER >= 0x0602)
 
+_Windows_SupportedOS_Windows8_
 USERENVAPI
 HRESULT
 WINAPI
