diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/vpconfig.h b/generation/WinSDK/RecompiledIdlHeaders/um/vpconfig.h
index aa02d53e37..4ccb67c94e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/vpconfig.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/vpconfig.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //------------------------------------------------------------------------------
 // File: VPConfig.h
 //
@@ -26,17 +27,20 @@ DECLARE_INTERFACE_(IVPBaseConfig, IUnknown)
     // gets the various connection information structures (guid, portwidth)
     // in an array of structures. If the pointer to the array is NULL, first 
     // parameter returns the total number of formats supported.
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (GetConnectInfo)(THIS_
 			       _Inout_ LPDWORD pdwNumConnectInfo,
 			       _Out_writes_to_opt_(*pdwNumConnectInfo, *pdwNumConnectInfo) LPDDVIDEOPORTCONNECT pddVPConnectInfo
 			      ) PURE;
 
     // sets the connection entry chosen (0, 1, .. ,(dwNumProposedEntries-1))
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetConnectInfo)(THIS_
 			       IN DWORD dwChosenEntry
 			      ) PURE;
 
     // gets various data parameters, includes dimensionnal info
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (GetVPDataInfo)(THIS_
 			      IN OUT LPAMVPDATAINFO pamvpDataInfo
 			     ) PURE;
@@ -45,12 +49,15 @@ DECLARE_INTERFACE_(IVPBaseConfig, IUnknown)
     // format and a given scaling factor. If decoder does not support 
     // those scaling factors, then it gives the rate and the nearest 
     // scaling factors.
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (GetMaxPixelRate)(THIS_
 				IN OUT LPAMVPSIZE pamvpSize,
 				OUT LPDWORD pdwMaxPixelsPerSecond
 			       ) PURE;
 
     // informs the callee of the videoformats supported by the videoport
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
+    _Win32_PreserveResult_
     STDMETHOD (InformVPInputFormats)(THIS_
 				     IN DWORD dwNumFormats,
 				     IN LPDDPIXELFORMAT pDDPixelFormats
@@ -59,37 +66,44 @@ DECLARE_INTERFACE_(IVPBaseConfig, IUnknown)
     // gets the various formats supported by the decoder in an array
     // of structures. If the pointer to the array is NULL, first parameter
     // returns the total number of formats supported.
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (GetVideoFormats)(THIS_
 				_Inout_ LPDWORD pdwNumFormats,
 				_Out_writes_to_opt_(*pdwNumFormats, *pdwNumFormats) LPDDPIXELFORMAT pddPixelFormats
 			       ) PURE;
 
     // sets the format entry chosen (0, 1, .. ,(dwNumProposedEntries-1))
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetVideoFormat)(THIS_
 			       IN DWORD dwChosenEntry
 			      ) PURE;
 
     // asks the decoder to treat even fields like odd fields and visa versa
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetInvertPolarity)(THIS
 				 ) PURE;
 
     // the mixer uses this function to determine if the callee wants
     // the vpmixer to use its overlay surface and if so to get a pointer to it
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (GetOverlaySurface)(THIS_
 				  OUT LPDIRECTDRAWSURFACE* ppddOverlaySurface
 				 ) PURE;
 
     // sets the direct draw kernel handle
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetDirectDrawKernelHandle)(THIS_
 					  IN ULONG_PTR dwDDKernelHandle
 					 ) PURE;
 
     // sets the video port id
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetVideoPortID)(THIS_
 			       IN DWORD dwVideoPortID
 			      ) PURE;
 
     // sets the direct draw surface kernel handle
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetDDSurfaceKernelHandles)(THIS_
 					  IN DWORD cHandles,
 					  IN ULONG_PTR *rgDDKernelHandles
@@ -100,6 +114,7 @@ DECLARE_INTERFACE_(IVPBaseConfig, IUnknown)
     // dwPitch is the pitch of the surface (distance in pixels between the start 
     // pixels of two consecutive lines of the surface). (dwXOrigin, dwYOrigin) 
     // are the (X, Y) coordinates of the pixel at which valid data starts.  
+    _Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
     STDMETHOD (SetSurfaceParameters)(THIS_
                     IN DWORD dwPitch,
                     IN DWORD dwXOrigin,
@@ -112,12 +127,14 @@ DECLARE_INTERFACE_(IVPConfig, IVPBaseConfig)
 {
 	// the mixer uses this function to determine if the callee wants
 	// the mixer to decimate VIDEO data at its own descrition
+	_Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
 	STDMETHOD (IsVPDecimationAllowed)(THIS_
 					  OUT LPBOOL pbIsDecimationAllowed
 					 ) PURE;
 
 	// sets the scaling factors. If decoder does not support these,
 	// then it sets the values to the nearest factors it can support
+	_Windows_SupportedOS_Windows2000_ _Windows_SupportedOS_Server2000_
 	STDMETHOD (SetScalingFactors)(THIS_
 				      IN LPAMVPSIZE pamvpSize
 				     ) PURE;
