diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/vpnotify.h b/generation/WinSDK/RecompiledIdlHeaders/um/vpnotify.h
index e1d5d309c3..d7d0089a95 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/vpnotify.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/vpnotify.h
@@ -11,6 +11,7 @@
 #define __IVPNotify__
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -23,6 +24,8 @@ extern "C" {
 DECLARE_INTERFACE_(IVPBaseNotify, IUnknown)
 {
 	// this function initializes the reconnection to the decoder. 
+	
+	
 	STDMETHOD (RenegotiateVPParameters)(THIS
 					   ) PURE;
 
@@ -32,11 +35,15 @@ DECLARE_INTERFACE_(IVPBaseNotify, IUnknown)
 DECLARE_INTERFACE_(IVPNotify, IVPBaseNotify)
 {
     // function to set the mode (bob, weave etc)
+    
+    
     STDMETHOD (SetDeinterlaceMode)(THIS_ 
 				   IN AMVP_MODE mode
 				  ) PURE;
 
     // function to get the mode (bob, weave etc)
+    
+    
     STDMETHOD (GetDeinterlaceMode)(THIS_ 
 				   OUT AMVP_MODE *pMode
 				  ) PURE;
@@ -46,11 +53,15 @@ DECLARE_INTERFACE_(IVPNotify, IVPBaseNotify)
 DECLARE_INTERFACE_(IVPNotify2, IVPNotify)
 {
 // function to set the mode (bob, weave etc)
+    
+    
     STDMETHOD (SetVPSyncMaster)(THIS_ 
 				   IN BOOL bVPSyncMaster
 				  ) PURE;
 
     // function to get the mode (bob, weave etc)
+    
+    
     STDMETHOD (GetVPSyncMaster)(THIS_ 
 				   OUT BOOL *pbVPSyncMaster
 				  ) PURE;
