diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h b/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
index f71f149570..ace972cf75 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/vsbackup.h
@@ -15,6 +15,7 @@ Abstract:
 #ifndef _VSBACKUP_H_
 #define _VSBACKUP_H_
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family or VSS Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_VSS)
@@ -88,7 +89,7 @@ const IID IID_IVssBackupComponentsEx4       = // f434c2fd-b553-4961-a9f9-a8e90b6
 
 
 // description of a component
-typedef struct _VSS_COMPONENTINFO
+typedef struct   _VSS_COMPONENTINFO
     {
     VSS_COMPONENT_TYPE type;    // either VSS_CT_DATABASE or VSS_CT_FILEGROUP
     BSTR bstrLogicalPath;        // logical path to component
@@ -164,7 +165,7 @@ public:
         (
         _Out_ VSS_ID *pidInstance,
         _Out_ VSS_ID *pidWriter,
-        _Out_ BSTR *pbstrWriterName,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrWriterName,
         _Out_ VSS_USAGE_TYPE *pUsage,
         _Out_ VSS_SOURCE_TYPE *pSource
         ) = 0;
@@ -202,8 +203,8 @@ public:
     STDMETHOD(GetRestoreMethod)
         (
         _Out_ VSS_RESTOREMETHOD_ENUM *pMethod,
-        _Out_ BSTR *pbstrService,
-        _Out_ BSTR *pbstrUserProcedure,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrService,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrUserProcedure,
         _Out_ VSS_WRITERRESTORE_ENUM *pwriterRestore,
         _Out_ bool *pbRebootRequired,
         _Out_ UINT *pcMappings
@@ -250,8 +251,8 @@ public:
         (
         _Out_ VSS_ID *pidInstance,
         _Out_ VSS_ID *pidWriter,
-        _Out_ BSTR *pbstrWriterName,
-        _Out_ BSTR* pbstrInstanceName,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrWriterName,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR* pbstrInstanceName,
         _Out_ VSS_USAGE_TYPE *pUsage,
         _Out_ VSS_SOURCE_TYPE *pSource
         ) = 0;
@@ -392,7 +393,7 @@ public:
         _In_ UINT iWriter,
         _Out_ VSS_ID *pidInstance,
         _Out_ VSS_ID *pidWriter,
-        _Out_ BSTR *pbstrWriter,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrWriter,
         _Out_ VSS_WRITER_STATE *pnStatus,
         _Out_ HRESULT *phResultFailure
         ) = 0;
@@ -747,11 +748,11 @@ public:
         _In_ UINT iWriter,
         _Out_ VSS_ID *pidInstance,
         _Out_ VSS_ID *pidWriter,
-        _Out_ BSTR *pbstrWriter,
+        _Win32_RAIIFree_(SysFreeString) _Out_ BSTR *pbstrWriter,
         _Out_ VSS_WRITER_STATE *pnStatus,
         _Out_ HRESULT *phrFailureWriter,
         _Out_opt_ HRESULT *phrApplication = NULL,
-        _Out_opt_ BSTR *pbstrApplicationMessage = NULL
+        _Win32_RAIIFree_(SysFreeString) _Out_opt_ BSTR *pbstrApplicationMessage = NULL
         ) = 0;
 
     STDMETHOD(AddSnapshotToRecoverySet)
@@ -791,19 +792,19 @@ public:
 #define VSS_SW_BOOTABLE_STATE   (1 << 0)
 
 extern "C"  {
-    HRESULT STDAPICALLTYPE CreateVssBackupComponentsInternal
+    _Windows_SupportedOS_WindowsXP_ HRESULT STDAPICALLTYPE CreateVssBackupComponentsInternal
         (
         _Out_ IVssBackupComponents **ppBackup
         );
 
-    HRESULT STDAPICALLTYPE CreateVssExamineWriterMetadataInternal
+    _Windows_SupportedOS_WindowsXP_ HRESULT STDAPICALLTYPE CreateVssExamineWriterMetadataInternal
         (
         _In_ BSTR bstrXML,
         _Out_ IVssExamineWriterMetadata **ppMetadata
         );
 
 
-    HRESULT APIENTRY IsVolumeSnapshottedInternal
+    _Windows_SupportedOS_WindowsXP_ HRESULT APIENTRY IsVolumeSnapshottedInternal
         (
         _In_z_ VSS_PWSZ  pwszVolumeName,
         _Out_ BOOL    *pbSnapshotsPresent,
@@ -813,6 +814,8 @@ extern "C"  {
 /////////////////////////////////////////////////////////////////////
 // Life-management methods for structure members
 
+_Windows_SupportedOS_WindowsXP_ 
+
     void APIENTRY VssFreeSnapshotPropertiesInternal(
         _In_ VSS_SNAPSHOT_PROP*  pProp
         );
