diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WabDefs.h b/generation/WinSDK/RecompiledIdlHeaders/um/WabDefs.h
index 5b7a5d86ec..db4fa267e9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WabDefs.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WabDefs.h
@@ -7,6 +7,7 @@
  */
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -314,7 +315,7 @@ typedef struct _MAPIUID
 
 /* Property Tag Array */
 
-typedef struct _SPropTagArray
+typedef struct   _SPropTagArray
 {
     ULONG   cValues;
     _Field_size_(cValues) 
