diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WcnTypes.h b/generation/WinSDK/RecompiledIdlHeaders/um/WcnTypes.h
index 343e6b2fa1..b38094a57e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WcnTypes.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WcnTypes.h
@@ -65,6 +65,7 @@ Abstract:
 
 --*/
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 //
@@ -94,7 +95,7 @@ Abstract:
 // New attributes to WPS 2.0 have been appended to the enum.
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_ATTRIBUTE_TYPE
+enum   tagWCN_ATTRIBUTE_TYPE
     {
         WCN_TYPE_AP_CHANNEL	= 0,
         WCN_TYPE_ASSOCIATION_STATE	= ( WCN_TYPE_AP_CHANNEL + 1 ) ,
@@ -204,7 +205,7 @@ enum tagWCN_ATTRIBUTE_TYPE
 // Values for WCN_TYPE_VERSION/WCN_TYPE_VERSION2 (these refer to the WPS protocol version)
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_VERSION
+enum   tagWCN_VALUE_TYPE_VERSION
     {
         WCN_VALUE_VERSION_1_0	= 0x10,
         WCN_VALUE_VERSION_2_0	= 0x20
@@ -215,7 +216,7 @@ enum tagWCN_VALUE_TYPE_VERSION
 // Values for all boolean types (example: WCN_TYPE_AP_SETUP_LOCKED)
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_BOOLEAN
+enum   tagWCN_VALUE_TYPE_BOOLEAN
     {
         WCN_VALUE_FALSE	= 0,
         WCN_VALUE_TRUE	= 1
@@ -226,7 +227,7 @@ enum tagWCN_VALUE_TYPE_BOOLEAN
 // Values for WCN_TYPE_ASSOCIATION_STATE
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_ASSOCIATION_STATE
+enum   tagWCN_VALUE_TYPE_ASSOCIATION_STATE
     {
         WCN_VALUE_AS_NOT_ASSOCIATED	= 0,
         WCN_VALUE_AS_CONNECTION_SUCCESS	= 1,
@@ -241,7 +242,7 @@ enum tagWCN_VALUE_TYPE_ASSOCIATION_STATE
 // and WCN_TYPE_AUTHENTICATION_TYPE_FLAGS
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_AUTHENTICATION_TYPE
+enum   tagWCN_VALUE_TYPE_AUTHENTICATION_TYPE
     {
         WCN_VALUE_AT_OPEN	= 0x1,
         WCN_VALUE_AT_WPAPSK	= 0x2,
@@ -257,7 +258,7 @@ enum tagWCN_VALUE_TYPE_AUTHENTICATION_TYPE
 // Values for WCN_TYPE_CONFIG_METHODS
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_CONFIG_METHODS
+enum   tagWCN_VALUE_TYPE_CONFIG_METHODS
     {
         WCN_VALUE_CM_USBA	= 0x1,
         WCN_VALUE_CM_ETHERNET	= 0x2,
@@ -279,7 +280,7 @@ enum tagWCN_VALUE_TYPE_CONFIG_METHODS
 // Values for WCN_TYPE_CONFIGURATION_ERROR
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_CONFIGURATION_ERROR
+enum   tagWCN_VALUE_TYPE_CONFIGURATION_ERROR
     {
         WCN_VALUE_CE_NO_ERROR	= 0,
         WCN_VALUE_CE_OOB_INTERFACE_READ_ERROR	= 1,
@@ -308,7 +309,7 @@ enum tagWCN_VALUE_TYPE_CONFIGURATION_ERROR
 // and WCN_TYPE_CONNECTION_TYPE_FLAGS
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_CONNECTION_TYPE
+enum   tagWCN_VALUE_TYPE_CONNECTION_TYPE
     {
         WCN_VALUE_CT_ESS	= 1,
         WCN_VALUE_CT_IBSS	= 2
@@ -319,7 +320,7 @@ enum tagWCN_VALUE_TYPE_CONNECTION_TYPE
 // Values for WCN_TYPE_DEVICE_PASSWORD_ID
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_DEVICE_PASSWORD_ID
+enum   tagWCN_VALUE_TYPE_DEVICE_PASSWORD_ID
     {
         WCN_VALUE_DP_DEFAULT	= 0,
         WCN_VALUE_DP_USER_SPECIFIED	= 0x1,
@@ -339,7 +340,7 @@ enum tagWCN_VALUE_TYPE_DEVICE_PASSWORD_ID
 // WCN_TYPE_ENCRYPTION_TYPE_FLAGS
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_ENCRYPTION_TYPE
+enum   tagWCN_VALUE_TYPE_ENCRYPTION_TYPE
     {
         WCN_VALUE_ET_NONE	= 0x1,
         WCN_VALUE_ET_WEP	= 0x2,
@@ -403,7 +404,7 @@ enum tagWCN_VALUE_TYPE_RESPONSE_TYPE
 // Values for WCN_TYPE_RF_BANDS
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_RF_BANDS
+enum   tagWCN_VALUE_TYPE_RF_BANDS
     {
         WCN_VALUE_RB_24GHZ	= 0x1,
         WCN_VALUE_RB_50GHZ	= 0x2
@@ -414,7 +415,7 @@ enum tagWCN_VALUE_TYPE_RF_BANDS
 // Values for WCN_TYPE_WI_FI_PROTECTED_SETUP_STATE
 //
 typedef /* [v1_enum] */ 
-enum tagWCN_VALUE_TYPE_WI_FI_PROTECTED_SETUP_STATE
+enum   tagWCN_VALUE_TYPE_WI_FI_PROTECTED_SETUP_STATE
     {
         WCN_VALUE_SS_RESERVED00	= 0,
         WCN_VALUE_SS_NOT_CONFIGURED	= 0x1,
@@ -603,7 +604,7 @@ typedef unsigned short WCN_VALUE_TYPE_DEVICE_TYPE_SUBCATEGORY;
 //
 #ifdef  _MSC_VER
 #pragma pack(push, 1)
-typedef struct tagWCN_VALUE_TYPE_PRIMARY_DEVICE_TYPE
+typedef struct   tagWCN_VALUE_TYPE_PRIMARY_DEVICE_TYPE
     {
     WCN_VALUE_TYPE_DEVICE_TYPE_CATEGORY Category;
     WCN_VALUE_TYPE_DEVICE_TYPE_SUBCATEGORY_OUI SubCategoryOUI;
