diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wct.h b/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
index 84abdf2d78..ade1f82da7 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wct.h
@@ -21,6 +21,7 @@ History:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -68,7 +69,7 @@ typedef enum _WCT_OBJECT_STATUS
 
 #define WCT_OBJNAME_LENGTH 128
 
-typedef struct _WAITCHAIN_NODE_INFO
+typedef struct   _WAITCHAIN_NODE_INFO
 {
     WCT_OBJECT_TYPE ObjectType;
     WCT_OBJECT_STATUS ObjectStatus;
@@ -93,26 +94,49 @@ typedef struct _WAITCHAIN_NODE_INFO
 
 typedef LPVOID HWCT;
 
+
+
 typedef VOID (CALLBACK *PWAITCHAINCALLBACK) (
-    HWCT WctHandle,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") HWCT WctHandle,
     DWORD_PTR Context,
     DWORD CallbackStatus,
-    LPDWORD NodeCount,
-    PWAITCHAIN_NODE_INFO NodeInfoArray,
-    LPBOOL IsCycle
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPDWORD NodeCount,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PWAITCHAIN_NODE_INFO NodeInfoArray,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPBOOL IsCycle
     );
 
 #define WCT_ASYNC_OPEN_FLAG 0x1
 #define WCTP_OPEN_ALL_FLAGS (WCT_ASYNC_OPEN_FLAG)
 
+#ifdef WIN32METADATA
+#pragma push_macro("WCT_ASYNC_OPEN_FLAG")
+#undef WCT_ASYNC_OPEN_FLAG
+enum 
+#ifndef WIN32METADATA
+[[clang::flag_enum]]
+#endif
+ OPEN_THREAD_WAIT_CHAIN_SESSION_FLAGS : DWORD {
+    WCT_ASYNC_OPEN_FLAG = 0x1,
+};
+#pragma pop_macro("WCT_ASYNC_OPEN_FLAG")
+#endif
+
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
+
+_Win32_ImportLibrary_("Advapi32.dll")
 _Check_return_
 WINADVAPI
 HWCT WINAPI
 OpenThreadWaitChainSession (
-    _In_ DWORD Flags,
-    _In_opt_ PWAITCHAINCALLBACK callback
+    _In_ _Win32_AssociatedEnum_(OPEN_THREAD_WAIT_CHAIN_SESSION_FLAGS) DWORD Flags,
+    _In_opt_  PWAITCHAINCALLBACK callback
     );
 
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_ImportLibrary_("Advapi32.dll")
 WINADVAPI
 VOID WINAPI
 CloseThreadWaitChainSession (
@@ -125,27 +149,56 @@ CloseThreadWaitChainSession (
 #define WCT_NETWORK_IO_FLAG      0x8
 #define WCTP_GETINFO_ALL_FLAGS (WCT_OUT_OF_PROC_FLAG|WCT_OUT_OF_PROC_COM_FLAG|WCT_OUT_OF_PROC_CS_FLAG)
 
+#ifdef WIN32METADATA
+#pragma push_macro("WCT_OUT_OF_PROC_COM_FLAG")
+#pragma push_macro("WCT_OUT_OF_PROC_CS_FLAG")
+#pragma push_macro("WCT_OUT_OF_PROC_FLAG")
+#undef WCT_OUT_OF_PROC_COM_FLAG
+#undef WCT_OUT_OF_PROC_CS_FLAG
+#undef WCT_OUT_OF_PROC_FLAG
+enum 
+#ifndef WIN32METADATA
+[[clang::flag_enum]]
+#endif
+ WAIT_CHAIN_THREAD_OPTIONS : DWORD {
+    WCT_OUT_OF_PROC_COM_FLAG = 0x2,
+    WCT_OUT_OF_PROC_CS_FLAG = 0x4,
+    WCT_OUT_OF_PROC_FLAG = 0x1,
+};
+#pragma pop_macro("WCT_OUT_OF_PROC_FLAG")
+#pragma pop_macro("WCT_OUT_OF_PROC_CS_FLAG")
+#pragma pop_macro("WCT_OUT_OF_PROC_COM_FLAG")
+#endif
+
+_Windows_SupportedOS_WindowsVista_ 
+
+_Win32_SetLastError_
+_Win32_PreserveResult_
+_Win32_ImportLibrary_("Advapi32.dll")
 _Check_return_
 WINADVAPI
 BOOL WINAPI
 GetThreadWaitChain (
     _In_ HWCT WctHandle,
     _In_opt_ DWORD_PTR Context,
-    _In_ DWORD Flags,
+    _In_ _Win32_AssociatedEnum_(WAIT_CHAIN_THREAD_OPTIONS) DWORD Flags,
     _In_ DWORD ThreadId,
-    _Inout_ LPDWORD NodeCount,
-    _Out_writes_(*NodeCount) PWAITCHAIN_NODE_INFO NodeInfoArray,
-    _Out_ LPBOOL IsCycle
+    _Inout_  LPDWORD NodeCount,
+    _Out_writes_(*NodeCount)  PWAITCHAIN_NODE_INFO NodeInfoArray,
+    _Out_  LPBOOL IsCycle
     );
 
-typedef HRESULT (*PCOGETCALLSTATE)(int, PULONG);
-typedef HRESULT (*PCOGETACTIVATIONSTATE)(GUID, DWORD, DWORD*);
+typedef HRESULT (*PCOGETCALLSTATE)(int, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PULONG);
+typedef HRESULT (*PCOGETACTIVATIONSTATE)(GUID, DWORD, _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") DWORD*);
+
+_Windows_SupportedOS_WindowsVista_ 
 
+_Win32_ImportLibrary_("Advapi32.dll")
 WINADVAPI
 VOID WINAPI
 RegisterWaitChainCOMCallback (
-    _In_ PCOGETCALLSTATE CallStateCallback,
-    _In_ PCOGETACTIVATIONSTATE ActivationStateCallback
+    _In_  PCOGETCALLSTATE CallStateCallback,
+    _In_  PCOGETACTIVATIONSTATE ActivationStateCallback
     );
 
 #ifdef __cplusplus
