diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WdsClientApi.h b/generation/WinSDK/RecompiledIdlHeaders/um/WdsClientApi.h
index 1f3aeba9cd..0a36438ed9 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WdsClientApi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WdsClientApi.h
@@ -22,6 +22,7 @@ Environment:
 #pragma once
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -43,7 +44,7 @@ extern "C"
 
 // Credentials structure.
 //
-typedef struct tagWDS_CLI_CRED
+typedef struct   tagWDS_CLI_CRED
 {
     PCWSTR pwszUserName,
            pwszDomain,
@@ -54,12 +55,15 @@ typedef struct tagWDS_CLI_CRED
 //
 // General functions
 //
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliClose( 
     _In_ HANDLE Handle );
 
 
+_Windows_SupportedOS_Server2008_
 typedef 
 VOID
 (WDSCLIAPI *PFN_WdsCliTraceFunction)(
@@ -68,12 +72,14 @@ VOID
 );
 
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliRegisterTrace(
-    _In_opt_ PFN_WdsCliTraceFunction pfn );
+     _In_opt_ PFN_WdsCliTraceFunction pfn );
 
 
+_Windows_SupportedOS_Windows7_
 HRESULT
 WDSCLIAPI
 WdsCliFreeStringArray(
@@ -83,34 +89,42 @@ WdsCliFreeStringArray(
 //
 // Image enumeration functions
 //
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliFindFirstImage(
     _In_ HANDLE hSession,
-    _Out_ PHANDLE phFindHandle );
+    _Win32_RAIIFree_(CloseHandle, -1, 0)  _Out_ PHANDLE phFindHandle );
+
+_Windows_SupportedOS_WindowsVista_ 
 
 HRESULT
 WDSCLIAPI
 WdsCliFindNextImage(
     _In_ HANDLE Handle );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetEnumerationFlags(
     _In_ HANDLE Handle,
     _Out_ PDWORD pdwFlags );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliGetImageHandleFromFindHandle(
     _In_ HANDLE FindHandle,
-    _Out_ PHANDLE phImageHandle );
+    _Win32_RAIIFree_(CloseHandle, -1, 0)  _Out_ PHANDLE phImageHandle );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliGetImageHandleFromTransferHandle(
     _In_ HANDLE hTransfer,
-    _Out_ PHANDLE phImageHandle );
+    _Win32_RAIIFree_(CloseHandle, -1, 0)  _Out_ PHANDLE phImageHandle );
 
 
 //
@@ -160,12 +174,16 @@ typedef enum _WDS_CLI_IMAGE_PARAM_TYPE
 //
 // Session functions
 //
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliCreateSession(
     _In_ PWSTR pwszServer,
     _In_opt_ PWDS_CLI_CRED pCred,
-    _Out_ PHANDLE phSession );
+    _Win32_RAIIFree_(CloseHandle, -1, 0)  _Out_ PHANDLE phSession );
+
+_Windows_SupportedOS_WindowsVista_ 
 
 HRESULT
 WDSCLIAPI
@@ -176,6 +194,8 @@ WdsCliAuthorizeSession(
 //
 // Client -> Server Logging functions
 //
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliInitializeLog(
@@ -184,6 +204,8 @@ WdsCliInitializeLog(
     _In_ PWSTR pwszClientId,
     _In_ PWSTR pwszClientAddress );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliLog(
@@ -242,12 +264,16 @@ enum
 // Image information functions
 //
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageName(
     _In_ HANDLE hIfh, 
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageDescription(
@@ -267,12 +293,16 @@ WdsCliGetImageFiles(
     _Outptr_result_buffer_(*pdwCount) LPWSTR **pppwszFiles,
    _Out_ PDWORD pdwCount );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageLanguage(
     _In_ HANDLE hIfh, 
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageLanguages(
@@ -280,54 +310,72 @@ WdsCliGetImageLanguages(
     _Outptr_result_buffer_(*pdwNumValues) PTSTR **pppszValues,
     _Out_ PDWORD pdwNumValues );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageVersion(
     _In_ HANDLE hIfh, 
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImagePath(
     _In_ HANDLE hIfh,
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageIndex(
     _In_ HANDLE hIfh,
     _Out_ PDWORD pdwValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageArchitecture(
     _In_ HANDLE hIfh,
     _Out_ PDWORD pdwValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageLastModifiedTime(
     _In_ HANDLE hIfh,
     _Out_ PSYSTEMTIME *ppSysTimeValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageSize(
     _In_ HANDLE hIfh,
     _Out_ PULONGLONG pullValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageHalName(
     _In_ HANDLE hIfh,
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageGroup(
     _In_ HANDLE hIfh,
     _Out_ PWSTR *ppwszValue );
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetImageNamespace(
@@ -349,16 +397,38 @@ WdsCliGetImageParameter(
 // Callback function
 //
 
+_Windows_SupportedOS_Server2008_
 typedef
 VOID
 (WDSCLIAPI *PFN_WdsCliCallback)(
-    DWORD dwMessageId,
+    _Win32_AssociatedEnum_(PFN_WDS_CLI_CALLBACK_MESSAGE_ID) DWORD dwMessageId,
     _In_opt_ WPARAM wParam,
     _In_opt_ LPARAM lParam,
     _In_opt_ PVOID pvUserData );
 
 // Callback messages.
 //
+#ifdef WIN32METADATA
+#pragma push_macro("WDS_CLI_MSG_START")
+#undef WDS_CLI_MSG_START
+#pragma push_macro("WDS_CLI_MSG_COMPLETE")
+#undef WDS_CLI_MSG_COMPLETE
+#pragma push_macro("WDS_CLI_MSG_PROGRESS")
+#undef WDS_CLI_MSG_PROGRESS
+#pragma push_macro("WDS_CLI_MSG_TEXT")
+#undef WDS_CLI_MSG_TEXT
+enum PFN_WDS_CLI_CALLBACK_MESSAGE_ID : unsigned int
+{
+    WDS_CLI_MSG_START = 0U,
+    WDS_CLI_MSG_COMPLETE = 1U,
+    WDS_CLI_MSG_PROGRESS = 2U,
+    WDS_CLI_MSG_TEXT = 3U,
+};
+#pragma pop_macro("WDS_CLI_MSG_TEXT")
+#pragma pop_macro("WDS_CLI_MSG_PROGRESS")
+#pragma pop_macro("WDS_CLI_MSG_COMPLETE")
+#pragma pop_macro("WDS_CLI_MSG_START")
+#else
 enum
 {
     WDS_CLI_MSG_START = 0,
@@ -366,8 +436,11 @@ enum
     WDS_CLI_MSG_PROGRESS,
     WDS_CLI_MSG_TEXT,
 };
+#endif
 
 
+_Windows_SupportedOS_WindowsVista_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetTransferSize(
@@ -386,6 +459,7 @@ WDSCLIAPI
 WdsCliSetTransferBufferSize(
     _In_ ULONG ulSizeInBytes );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliTransferImage(
@@ -393,10 +467,11 @@ WdsCliTransferImage(
     _In_ PWSTR pwszLocalPath,
     _In_ DWORD dwFlags,
     _In_ DWORD dwReserved,
-    _In_opt_ PFN_WdsCliCallback pfnWdsCliCallback,
-    _In_opt_ PVOID pvUserData,
-    _Out_ PHANDLE phTransfer );
+     _In_opt_ PFN_WdsCliCallback pfnWdsCliCallback,
+     _In_opt_ PVOID pvUserData,
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE phTransfer );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliTransferFile(
@@ -406,15 +481,17 @@ WdsCliTransferFile(
     _In_ PCWSTR pwszLocalFilePath,
     _In_ DWORD dwFlags,
     _In_ DWORD dwReserved,
-    _In_opt_ PFN_WdsCliCallback pfnWdsCliCallback,
-    _In_opt_ PVOID pvUserData,
-    _Out_ PHANDLE phTransfer );
+     _In_opt_ PFN_WdsCliCallback pfnWdsCliCallback,
+     _In_opt_ PVOID pvUserData,
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_ PHANDLE phTransfer );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliCancelTransfer(
     _In_ HANDLE hTransfer );
 
+_Windows_SupportedOS_WindowsVista_ 
 HRESULT
 WDSCLIAPI
 WdsCliWaitForTransfer(
@@ -423,6 +500,7 @@ WdsCliWaitForTransfer(
 //
 // Dynamic Driver Provisioning functions
 //
+_Windows_SupportedOS_Windows7_
 HRESULT
 WDSCLIAPI
 WdsCliObtainDriverPackages(
@@ -432,6 +510,8 @@ WdsCliObtainDriverPackages(
     _Out_ ULONG *pulCount    
     );
 
+_Windows_SupportedOS_Windows8_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliObtainDriverPackagesEx (
@@ -442,6 +522,8 @@ WdsCliObtainDriverPackagesEx (
     _Out_ ULONG* pulCount
     );
 
+_Windows_SupportedOS_Windows8_ 
+
 HRESULT
 WDSCLIAPI
 WdsCliGetDriverQueryXml (
@@ -459,4 +541,6 @@ WdsCliGetDriverQueryXml (
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
+
+
 #endif  // __WDSCLIENTAPI_H__
