diff --git a/generation/WinSDK/RecompiledIdlHeaders/winrt/WeakReference.h b/generation/WinSDK/RecompiledIdlHeaders/winrt/WeakReference.h
index ea6c5c42ac..b132f9e5e0 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/winrt/WeakReference.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/winrt/WeakReference.h
@@ -19,6 +19,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -90,10 +91,12 @@ extern RPC_IF_HANDLE __MIDL_itf_weakreference_0000_0000_v0_0_s_ifspec;
     EXTERN_C const IID IID_IWeakReference;
     extern "C++"
     {
-        MIDL_INTERFACE("00000037-0000-0000-C000-000000000046")
+        MIDL_INTERFACE("00000037-0000-0000-C000-000000000046") _Windows_SupportedOS_Windows8_ 
         IWeakReference : public IUnknown
         {
         public:
+            
+            
             virtual HRESULT STDMETHODCALLTYPE Resolve( 
                 /* [in] */ __RPC__in REFIID riid,
                 /* [iid_is][out] */ __RPC__deref_out IInspectable **objectReference) = 0;
@@ -114,10 +117,12 @@ EXTERN_C const IID IID_IWeakReference;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("00000037-0000-0000-C000-000000000046")
+    MIDL_INTERFACE("00000037-0000-0000-C000-000000000046") _Windows_SupportedOS_Windows8_ 
     IWeakReference : public IUnknown
     {
     public:
+        
+        
         virtual HRESULT STDMETHODCALLTYPE Resolve( 
             /* [in] */ __RPC__in REFIID riid,
             /* [iid_is][retval][out] */ __RPC__deref_out_opt IInspectable **objectReference) = 0;
@@ -155,7 +160,7 @@ EXTERN_C const IID IID_IWeakReference;
         END_INTERFACE
     } IWeakReferenceVtbl;
 
-    interface IWeakReference
+    interface _Windows_SupportedOS_Windows8_  IWeakReference
     {
         CONST_VTBL struct IWeakReferenceVtbl *lpVtbl;
     };
@@ -209,10 +214,12 @@ EXTERN_C const IID IID_IWeakReferenceSource;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("00000038-0000-0000-C000-000000000046")
+    MIDL_INTERFACE("00000038-0000-0000-C000-000000000046") _Windows_SupportedOS_Windows8_ 
     IWeakReferenceSource : public IUnknown
     {
     public:
+        
+        
         virtual HRESULT STDMETHODCALLTYPE GetWeakReference( 
             /* [retval][out] */ __RPC__deref_out_opt IWeakReference **weakReference) = 0;
         
