diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/websocket.h b/generation/WinSDK/RecompiledIdlHeaders/um/websocket.h
index eaf1937156a11749f8e5cf97c1c2f46ec8b0d614..96fae620073f414fba1bad381803b329f9f7fa5f 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/websocket.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/websocket.h
@@ -25,6 +25,7 @@ extern "C"
 #endif
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
@@ -89,7 +90,7 @@ typedef enum _WEB_SOCKET_ACTION
     WEB_SOCKET_INDICATE_RECEIVE_COMPLETE_ACTION     = 4,
 } WEB_SOCKET_ACTION, *PWEB_SOCKET_ACTION;
 
-typedef struct _WEB_SOCKET_PROPERTY
+typedef struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ _WEB_SOCKET_PROPERTY
 {
     WEB_SOCKET_PROPERTY_TYPE Type;
     PVOID pvValue;
@@ -120,11 +121,15 @@ typedef union _WEB_SOCKET_BUFFER
     } CloseStatus;
 } WEB_SOCKET_BUFFER, *PWEB_SOCKET_BUFFER;
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 HRESULT WINAPI WebSocketCreateClientHandle(
     _In_reads_(ulPropertyCount) const PWEB_SOCKET_PROPERTY pProperties,
     _In_ ULONG ulPropertyCount,
     _Out_ WEB_SOCKET_HANDLE *phWebSocket);
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 HRESULT WINAPI WebSocketBeginClientHandshake(
     _In_ WEB_SOCKET_HANDLE hWebSocket,
     _In_reads_opt_(ulSubprotocolCount) PCSTR *pszSubprotocols,
@@ -136,6 +141,8 @@ HRESULT WINAPI WebSocketBeginClientHandshake(
     _Out_writes_(*pulAdditionalHeaderCount) PWEB_SOCKET_HTTP_HEADER *pAdditionalHeaders,
     _Out_ ULONG *pulAdditionalHeaderCount);
 
+_Windows_SupportedOS_Windows8_
+_Windows_SupportedOS_Server2012_
 HRESULT WINAPI WebSocketEndClientHandshake(
     _In_ WEB_SOCKET_HANDLE hWebSocket,
     _In_reads_(ulReponseHeaderCount) const PWEB_SOCKET_HTTP_HEADER pResponseHeaders,
