diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WebServices.h b/generation/WinSDK/RecompiledIdlHeaders/um/WebServices.h
index dda241ddccb2f276f358a923276769bec99da46e..629389dbf65b4de883cd496d6feafe5bd1f87e3d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WebServices.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WebServices.h
@@ -9,6 +9,7 @@
 #include <wtypes.h>
 #include <wincrypt.h>
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 
 #ifdef __cplusplus
 extern "C"
@@ -8016,6 +8017,7 @@ HRESULT WINAPI WsGetOperationContextProperty(
 
 //  Dictionaries function
 
+_Windows_SupportedOS_Windows7_
 HRESULT WINAPI WsGetDictionary(
     _In_ WS_ENCODING encoding, 
     _Outptr_result_maybenull_ WS_XML_DICTIONARY** dictionary, 
