diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WinBase.h b/generation/WinSDK/RecompiledIdlHeaders/um/WinBase.h
index 9afdd463..bed358f4 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WinBase.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WinBase.h
@@ -1,6 +1,9 @@
 //@[contract("ntoskrnl-winbase"), comment("MVI_tracked - https://osgwiki.com/wiki/Microsoft_Virus_Initiative")];
 
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 /************************************************************************
 *                                                                       *
@@ -372,10 +375,10 @@ typedef enum FILE_FLUSH_MODE
 //
 
 #if(_WIN32_WINNT >= 0x0400)
-typedef VOID (WINAPI *PFIBER_START_ROUTINE)(
+typedef VOID (WINAPI *LPFIBER_START_ROUTINE)(
     LPVOID lpFiberParameter
     );
-typedef PFIBER_START_ROUTINE LPFIBER_START_ROUTINE;
+typedef LPFIBER_START_ROUTINE PFIBER_START_ROUTINE;
 
 typedef LPVOID (WINAPI *PFIBER_CALLOUT_ROUTINE)(
     LPVOID lpParameter
@@ -1373,6 +1376,7 @@ LocalCompact(
 # define SCS_THIS_PLATFORM_BINARY SCS_32BIT_BINARY
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1380,6 +1384,7 @@ GetBinaryTypeA(
     _In_  LPCSTR lpApplicationName,
     _Out_ LPDWORD  lpBinaryType
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1393,6 +1398,7 @@ GetBinaryTypeW(
 #define GetBinaryType  GetBinaryTypeA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < cchBuffer)
 DWORD
@@ -1408,6 +1414,7 @@ GetShortPathNameA(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < cchBuffer)
 DWORD
@@ -1418,6 +1425,7 @@ GetLongPathNameTransactedA(
     _In_     DWORD cchBuffer,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < cchBuffer)
 DWORD
@@ -1442,6 +1450,7 @@ GetLongPathNameTransactedW(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1451,6 +1460,7 @@ GetProcessAffinityMask(
     _Out_ PDWORD_PTR lpSystemAffinityMask
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1465,6 +1475,7 @@ SetProcessAffinityMask(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1788,6 +1799,7 @@ SetThreadAffinityMask(
 #define PROCESS_DEP_ENABLE                          0x00000001
 #define PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION     0x00000002
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1795,6 +1807,7 @@ SetProcessDEPPolicy(
     _In_ DWORD dwFlags
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1820,6 +1833,7 @@ IsSystemResumeAutomatic(
     VOID
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1844,6 +1858,7 @@ SetThreadExecutionState(
 
 typedef REASON_CONTEXT POWER_REQUEST_CONTEXT, *PPOWER_REQUEST_CONTEXT, *LPPOWER_REQUEST_CONTEXT;
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -1851,6 +1866,7 @@ PowerCreateRequest (
     _In_ PREASON_CONTEXT Context
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1859,6 +1875,7 @@ PowerSetRequest (
     _In_ POWER_REQUEST_TYPE RequestType
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1930,6 +1947,7 @@ typedef VOID (WINAPI* PRESTORE_LAST_ERROR)(DWORD);
 //
 #define FILE_SKIP_SET_EVENT_ON_HANDLE           0x2
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1955,6 +1973,7 @@ SetFileCompletionNotificationModes(
 
 #if (_WIN32_WINNT >= 0x0601)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -1998,6 +2017,8 @@ DebugBreakProcess (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2064,6 +2085,7 @@ SetMessageWaitingIndicator(
     );
 
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2071,6 +2093,7 @@ SetFileShortNameA(
     _In_ HANDLE hFile,
     _In_ LPCSTR lpShortName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2307,67 +2330,82 @@ GetCommPorts(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+#ifdef WIN32METADATA
+enum GET_TAPE_DRIVE_PARAMETERS_OPERATION : DWORD;
+enum TAPE_INFORMATION_TYPE : DWORD;
+#else
+typedef DWORD GET_TAPE_DRIVE_PARAMETERS_OPERATION;
+typedef DWORD TAPE_INFORMATION_TYPE;
+#endif
+
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 SetTapePosition(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwPositionMethod,
+    _In_ TAPE_POSITION_METHOD dwPositionMethod,
     _In_ DWORD dwPartition,
     _In_ DWORD dwOffsetLow,
     _In_ DWORD dwOffsetHigh,
     _In_ BOOL bImmediate
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 GetTapePosition(
     _In_  HANDLE hDevice,
-    _In_  DWORD dwPositionType,
+    _In_  TAPE_POSITION_TYPE dwPositionType,
     _Out_ LPDWORD lpdwPartition,
     _Out_ LPDWORD lpdwOffsetLow,
     _Out_ LPDWORD lpdwOffsetHigh
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 PrepareTape(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwOperation,
+    _In_ PREPARE_TAPE_OPERATION dwOperation,
     _In_ BOOL bImmediate
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 EraseTape(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwEraseType,
+    _In_ ERASE_TAPE_TYPE dwEraseType,
     _In_ BOOL bImmediate
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 CreateTapePartition(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwPartitionMethod,
+    _In_ CREATE_TAPE_PARTITION_METHOD dwPartitionMethod,
     _In_ DWORD dwCount,
     _In_ DWORD dwSize
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 WriteTapemark(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwTapemarkType,
+    _In_ TAPEMARK_TYPE dwTapemarkType,
     _In_ DWORD dwTapemarkCount,
     _In_ BOOL bImmediate
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
@@ -2375,12 +2413,13 @@ GetTapeStatus(
     _In_ HANDLE hDevice
     );
 
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 GetTapeParameters(
     _In_    HANDLE hDevice,
-    _In_    DWORD dwOperation,
+    _In_    GET_TAPE_DRIVE_PARAMETERS_OPERATION dwOperation,
     _Inout_ LPDWORD lpdwSize,
     _Out_writes_bytes_(*lpdwSize) LPVOID lpTapeInformation
     );
@@ -2388,18 +2427,47 @@ GetTapeParameters(
 #define GET_TAPE_MEDIA_INFORMATION 0
 #define GET_TAPE_DRIVE_INFORMATION 1
 
+#ifdef WIN32METADATA
+#pragma push_macro("GET_TAPE_MEDIA_INFORMATION")
+#pragma push_macro("GET_TAPE_DRIVE_INFORMATION")
+#undef GET_TAPE_MEDIA_INFORMATION
+#undef GET_TAPE_DRIVE_INFORMATION
+enum GET_TAPE_DRIVE_PARAMETERS_OPERATION : DWORD
+{
+    GET_TAPE_MEDIA_INFORMATION = 0,
+    GET_TAPE_DRIVE_INFORMATION = 1,
+};
+#pragma pop_macro("GET_TAPE_DRIVE_INFORMATION")
+#pragma pop_macro("GET_TAPE_MEDIA_INFORMATION")
+#endif
+
+_Windows_SupportedOS_WindowsXP_
 WINBASEAPI
 DWORD
 WINAPI
 SetTapeParameters(
     _In_ HANDLE hDevice,
-    _In_ DWORD dwOperation,
+    _In_ TAPE_INFORMATION_TYPE dwOperation,
     _In_ LPVOID lpTapeInformation
     );
 
 #define SET_TAPE_MEDIA_INFORMATION 0
 #define SET_TAPE_DRIVE_INFORMATION 1
 
+#ifdef WIN32METADATA
+#pragma push_macro("SET_TAPE_MEDIA_INFORMATION")
+#pragma push_macro("SET_TAPE_DRIVE_INFORMATION")
+#undef SET_TAPE_MEDIA_INFORMATION
+#undef SET_TAPE_DRIVE_INFORMATION
+enum TAPE_INFORMATION_TYPE : DWORD
+{
+    SET_TAPE_MEDIA_INFORMATION = 0,
+    SET_TAPE_DRIVE_INFORMATION = 1,
+};
+#pragma pop_macro("SET_TAPE_DRIVE_INFORMATION")
+#pragma pop_macro("SET_TAPE_MEDIA_INFORMATION")
+#endif
+
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
 
@@ -2442,6 +2510,7 @@ GetSystemDEPPolicy(
 
 #if _WIN32_WINNT >= 0x0501
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2456,6 +2525,7 @@ GetSystemRegistryQuota(
 // Routines to convert back and forth between system time and file time
 //
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2465,6 +2535,7 @@ FileTimeToDosDateTime(
     _Out_ LPWORD lpFatTime
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2564,6 +2635,7 @@ FormatMessage(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -2573,6 +2645,7 @@ CreateMailslotA(
     _In_     DWORD lReadTimeout,
     _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -2588,6 +2661,7 @@ CreateMailslotW(
 #define CreateMailslot  CreateMailslotA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2599,6 +2673,7 @@ GetMailslotInfo(
     _Out_opt_ LPDWORD lpReadTimeout
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -2611,12 +2686,14 @@ SetMailslotInfo(
 // File Encryption API
 //
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
 EncryptFileA(
     _In_ LPCSTR lpFileName
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -2629,6 +2706,7 @@ EncryptFileW(
 #define EncryptFile  EncryptFileA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -2636,6 +2714,7 @@ DecryptFileA(
     _In_       LPCSTR lpFileName,
     _Reserved_ DWORD dwReserved
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -2664,6 +2743,7 @@ DecryptFileW(
 #define FILE_READ_ONLY                  8
 #define FILE_DIR_DISALLOWED             9
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -2671,6 +2751,7 @@ FileEncryptionStatusA(
     _In_  LPCSTR lpFileName,
     _Out_ LPDWORD  lpStatus
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -2930,6 +3011,7 @@ lstrlenW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 HFILE
 WINAPI
@@ -2947,6 +3029,7 @@ _lopen(
     _In_ int iReadWrite
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 HFILE
 WINAPI
@@ -2964,6 +3047,7 @@ _lread(
     _In_ UINT uBytes
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 UINT
 WINAPI
@@ -2998,6 +3082,7 @@ _lclose(
     _In_ HFILE hFile
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 LONG
 WINAPI
@@ -3040,6 +3125,8 @@ SignalObjectAndWait(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3053,6 +3140,8 @@ BackupRead(
     _Inout_ LPVOID *lpContext
     );
 
+_Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3065,6 +3154,7 @@ BackupSeek(
     _Inout_ LPVOID *lpContext
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3082,8 +3172,14 @@ BackupWrite(
 //
 //  Stream id structure
 //
+#ifdef WIN32METADATA
+enum WIN_STREAM_ID : DWORD;
+#else
+typedef DWORD WIN_STREAM_ID;
+#endif
+
 typedef struct _WIN32_STREAM_ID {
-        DWORD          dwStreamId ;
+        WIN_STREAM_ID  dwStreamId ;
         DWORD          dwStreamAttributes ;
         LARGE_INTEGER  Size ;
         DWORD          dwStreamNameSize ;
@@ -3107,6 +3203,60 @@ typedef struct _WIN32_STREAM_ID {
 #define BACKUP_TXFS_DATA        0x0000000a
 #define BACKUP_GHOSTED_FILE_EXTENTS 0x0000000b
 
+#ifdef WIN32METADATA
+#pragma push_macro("BACKUP_INVALID")
+#pragma push_macro("BACKUP_DATA")
+#pragma push_macro("BACKUP_EA_DATA")
+#pragma push_macro("BACKUP_SECURITY_DATA")
+#pragma push_macro("BACKUP_ALTERNATE_DATA")
+#pragma push_macro("BACKUP_LINK")
+#pragma push_macro("BACKUP_PROPERTY_DATA")
+#pragma push_macro("BACKUP_OBJECT_ID")
+#pragma push_macro("BACKUP_REPARSE_DATA")
+#pragma push_macro("BACKUP_SPARSE_BLOCK")
+#pragma push_macro("BACKUP_TXFS_DATA")
+#pragma push_macro("BACKUP_GHOSTED_FILE_EXTENTS")
+#undef BACKUP_INVALID
+#undef BACKUP_DATA
+#undef BACKUP_EA_DATA
+#undef BACKUP_SECURITY_DATA
+#undef BACKUP_ALTERNATE_DATA
+#undef BACKUP_LINK
+#undef BACKUP_PROPERTY_DATA
+#undef BACKUP_OBJECT_ID
+#undef BACKUP_REPARSE_DATA
+#undef BACKUP_SPARSE_BLOCK
+#undef BACKUP_TXFS_DATA
+#undef BACKUP_GHOSTED_FILE_EXTENTS
+enum WIN_STREAM_ID : DWORD
+{
+    BACKUP_INVALID = 0x00000000,
+    BACKUP_DATA = 0x00000001,
+    BACKUP_EA_DATA = 0x00000002,
+    BACKUP_SECURITY_DATA = 0x00000003,
+    BACKUP_ALTERNATE_DATA = 0x00000004,
+    BACKUP_LINK = 0x00000005,
+    BACKUP_PROPERTY_DATA = 0x00000006,
+    BACKUP_OBJECT_ID = 0x00000007,
+    BACKUP_REPARSE_DATA = 0x00000008,
+    BACKUP_SPARSE_BLOCK = 0x00000009,
+    BACKUP_TXFS_DATA = 0x0000000a,
+    BACKUP_GHOSTED_FILE_EXTENTS = 0x0000000b,
+};
+#pragma pop_macro("BACKUP_GHOSTED_FILE_EXTENTS")
+#pragma pop_macro("BACKUP_TXFS_DATA")
+#pragma pop_macro("BACKUP_SPARSE_BLOCK")
+#pragma pop_macro("BACKUP_REPARSE_DATA")
+#pragma pop_macro("BACKUP_OBJECT_ID")
+#pragma pop_macro("BACKUP_PROPERTY_DATA")
+#pragma pop_macro("BACKUP_LINK")
+#pragma pop_macro("BACKUP_ALTERNATE_DATA")
+#pragma pop_macro("BACKUP_SECURITY_DATA")
+#pragma pop_macro("BACKUP_EA_DATA")
+#pragma pop_macro("BACKUP_DATA")
+#pragma pop_macro("BACKUP_INVALID")
+#endif
+
 //
 //  Stream Attributes
 //
@@ -3318,6 +3468,7 @@ CreateWaitableTimerExA(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -3342,6 +3493,7 @@ CreateFileMappingA(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -3368,6 +3520,7 @@ CreateFileMappingNumaA(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -3386,6 +3539,7 @@ OpenFileMappingA(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return <= nBufferLength)
 DWORD
@@ -3413,6 +3567,7 @@ GetLogicalDriveStringsA(
 
 #if (_WIN32_WINNT >= 0x0602)
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HMODULE
@@ -3466,6 +3621,7 @@ LoadPackagedLibrary (
 
 #define PROCESS_NAME_NATIVE     0x00000001
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -3475,6 +3631,7 @@ QueryFullProcessImageNameA(
     _Out_writes_to_(*lpdwSize, *lpdwSize) LPSTR lpExeName,
     _Inout_ PDWORD lpdwSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4115,6 +4272,7 @@ GetStartupInfoA(
 #pragma warning(disable: 6103)
 #endif /* _MSC_VER >= 1400 */
 
+_Win32_SetLastError_
 _Success_(return != 0 && return < nSize)
 __inline
 DWORD
@@ -4143,6 +4301,7 @@ GetEnvironmentVariable(
 
 #if defined(_M_CEE)
 #undef SetEnvironmentVariable
+_Win32_SetLastError_
 __inline
 BOOL
 SetEnvironmentVariable(
@@ -4167,6 +4326,7 @@ SetEnvironmentVariable(
 #pragma region OneCore Family or App Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_APP)
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4176,6 +4336,7 @@ GetFirmwareEnvironmentVariableA(
     _Out_writes_bytes_to_opt_(nSize, return) PVOID pBuffer,
     _In_ DWORD    nSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4193,6 +4354,7 @@ GetFirmwareEnvironmentVariableW(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4203,6 +4365,7 @@ GetFirmwareEnvironmentVariableExA(
     _In_ DWORD    nSize,
     _Out_opt_ PDWORD pdwAttribubutes
     );
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4221,6 +4384,7 @@ GetFirmwareEnvironmentVariableExW(
 
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4230,6 +4394,7 @@ SetFirmwareEnvironmentVariableA(
     _In_reads_bytes_opt_(nSize) PVOID pValue,
     _In_ DWORD    nSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4247,6 +4412,7 @@ SetFirmwareEnvironmentVariableW(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4257,6 +4423,7 @@ SetFirmwareEnvironmentVariableExA(
     _In_ DWORD    nSize,
     _In_ DWORD    dwAttributes
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4283,6 +4450,7 @@ SetFirmwareEnvironmentVariableExW(
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4291,6 +4459,7 @@ GetFirmwareType (
     );
 
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4300,6 +4469,7 @@ IsNativeVhdBoot (
 
 #endif // _WIN32_WINNT >= _WIN32_WINNT_WIN8
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HRSRC
@@ -4313,6 +4483,7 @@ FindResourceA(
 #define FindResource  FindResourceA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HRSRC
@@ -4327,6 +4498,7 @@ FindResourceExA(
 #define FindResourceEx  FindResourceExA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4335,6 +4507,7 @@ EnumResourceTypesA(
     _In_     ENUMRESTYPEPROCA lpEnumFunc,
     _In_     LONG_PTR lParam
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4349,6 +4522,7 @@ EnumResourceTypesW(
 #define EnumResourceTypes  EnumResourceTypesA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4359,6 +4533,7 @@ EnumResourceLanguagesA(
     _In_     ENUMRESLANGPROCA lpEnumFunc,
     _In_     LONG_PTR lParam
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4375,6 +4550,7 @@ EnumResourceLanguagesW(
 #define EnumResourceLanguages  EnumResourceLanguagesA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -4382,6 +4558,7 @@ BeginUpdateResourceA(
     _In_ LPCSTR pFileName,
     _In_ BOOL bDeleteExistingResources
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -4395,6 +4572,7 @@ BeginUpdateResourceW(
 #define BeginUpdateResource  BeginUpdateResourceA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4406,6 +4584,7 @@ UpdateResourceA(
     _In_reads_bytes_opt_(cb) LPVOID lpData,
     _In_ DWORD cb
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4423,6 +4602,7 @@ UpdateResourceW(
 #define UpdateResource  UpdateResourceA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4430,6 +4610,7 @@ EndUpdateResourceA(
     _In_ HANDLE hUpdate,
     _In_ BOOL   fDiscard
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4641,6 +4822,7 @@ GetProfileStringW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4649,6 +4831,7 @@ WriteProfileStringA(
     _In_opt_ LPCSTR lpKeyName,
     _In_opt_ LPCSTR lpString
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4697,6 +4880,7 @@ GetProfileSectionW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4704,6 +4888,7 @@ WriteProfileSectionA(
     _In_ LPCSTR lpAppName,
     _In_ LPCSTR lpString
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4771,6 +4956,7 @@ GetPrivateProfileInt(
 }
 #endif  /* _M_CEE */
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4782,6 +4968,7 @@ GetPrivateProfileStringA(
     _In_     DWORD nSize,
     _In_opt_ LPCSTR lpFileName
     );
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -4801,6 +4988,7 @@ GetPrivateProfileStringW(
 
 #if defined(_M_CEE)
 #undef GetPrivateProfileString
+_Win32_SetLastError_
 __inline
 DWORD
 GetPrivateProfileString(
@@ -4827,6 +5015,7 @@ GetPrivateProfileString(
 }
 #endif  /* _M_CEE */
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4836,6 +5025,7 @@ WritePrivateProfileStringA(
     _In_opt_ LPCSTR lpString,
     _In_opt_ LPCSTR lpFileName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4899,6 +5089,7 @@ GetPrivateProfileSection(
 }
 #endif  /* _M_CEE */
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -4907,6 +5098,7 @@ WritePrivateProfileSectionA(
     _In_opt_ LPCSTR lpString,
     _In_opt_ LPCSTR lpFileName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5023,6 +5215,7 @@ GetPrivateProfileStruct(
 }
 #endif  /* _M_CEE */
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5033,6 +5226,7 @@ WritePrivateProfileStructA(
     _In_     UINT     uSizeStruct,
     _In_opt_ LPCSTR szFile
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5051,6 +5245,7 @@ WritePrivateProfileStructW(
 
 #if defined(_M_CEE)
 #undef GetTempFileName
+_Win32_SetLastError_
 __inline
 UINT
 GetTempFileName(
@@ -5109,6 +5304,7 @@ typedef UINT (WINAPI* PGET_SYSTEM_WOW64_DIRECTORY_W)(_Out_writes_to_opt_(uSize,
 
 #if defined(_M_CEE)
 #undef SetCurrentDirectory
+_Win32_SetLastError_
 __inline
 BOOL
 SetCurrentDirectory(
@@ -5127,6 +5323,7 @@ SetCurrentDirectory(
 
 #if defined(_M_CEE)
 #undef GetCurrentDirectory
+_Win32_SetLastError_
 __inline
 DWORD
 GetCurrentDirectory(
@@ -5153,12 +5350,14 @@ GetCurrentDirectory(
 
 #if _WIN32_WINNT >= 0x0502
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 SetDllDirectoryA(
     _In_opt_ LPCSTR lpPathName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5173,6 +5372,7 @@ SetDllDirectoryW(
 #endif // _WIN32_WINNT >= 0x0502
 
 #if _WIN32_WINNT >= 0x0502
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < nBufferLength)
 DWORD
@@ -5181,6 +5381,7 @@ GetDllDirectoryA(
     _In_ DWORD nBufferLength,
     _Out_writes_to_opt_(nBufferLength, return + 1) LPSTR lpBuffer
     );
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < nBufferLength)
 DWORD
@@ -5208,6 +5409,7 @@ GetDllDirectoryW(
 #define BASE_SEARCH_PATH_PERMANENT 0x8000
 #define BASE_SEARCH_PATH_INVALID_FLAGS ~0x18001
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5223,6 +5425,7 @@ SetSearchPathMode (
 
 #if defined(_M_CEE)
 #undef CreateDirectory
+_Win32_SetLastError_
 __inline
 BOOL
 CreateDirectory(
@@ -5247,6 +5450,7 @@ CreateDirectory(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5255,6 +5459,7 @@ CreateDirectoryExA(
     _In_     LPCSTR lpNewDirectory,
     _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5277,6 +5482,7 @@ CreateDirectoryExW(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5286,6 +5492,7 @@ CreateDirectoryTransactedA(
     _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5301,6 +5508,7 @@ CreateDirectoryTransactedW(
 #define CreateDirectoryTransacted  CreateDirectoryTransactedA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5308,6 +5516,7 @@ RemoveDirectoryTransactedA(
     _In_ LPCSTR lpPathName,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5321,6 +5530,7 @@ RemoveDirectoryTransactedW(
 #define RemoveDirectoryTransacted  RemoveDirectoryTransactedA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < nBufferLength)
 DWORD
@@ -5332,6 +5542,7 @@ GetFullPathNameTransactedA(
     _Outptr_opt_ LPSTR *lpFilePart,
     _In_            HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0 && return < nBufferLength)
 DWORD
@@ -5363,6 +5574,7 @@ GetFullPathNameTransactedW(
 #define DDD_NO_BROADCAST_SYSTEM     0x00000008
 #define DDD_LUID_BROADCAST_DRIVE    0x00000010
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5375,6 +5587,7 @@ DefineDosDeviceA(
 #define DefineDosDevice  DefineDosDeviceA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -5397,6 +5610,7 @@ QueryDosDeviceA(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -5412,6 +5626,7 @@ CreateFileTransactedA(
     _In_opt_   PUSHORT pusMiniVersion,
     _Reserved_ PVOID  lpExtendedParameter
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -5443,6 +5658,7 @@ CreateFileTransactedW(
 
 #if _WIN32_WINNT >= 0x0502
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -5464,6 +5680,7 @@ ReOpenFile(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5472,6 +5689,7 @@ SetFileAttributesTransactedA(
     _In_     DWORD dwFileAttributes,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5486,6 +5704,7 @@ SetFileAttributesTransactedW(
 #define SetFileAttributesTransacted  SetFileAttributesTransactedA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5495,6 +5714,7 @@ GetFileAttributesTransactedA(
     _Out_writes_bytes_(sizeof(WIN32_FILE_ATTRIBUTE_DATA)) LPVOID lpFileInformation,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5510,6 +5730,7 @@ GetFileAttributesTransactedW(
 #define GetFileAttributesTransacted  GetFileAttributesTransactedA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -5518,6 +5739,7 @@ GetCompressedFileSizeTransactedA(
     _Out_opt_ LPDWORD  lpFileSizeHigh,
     _In_      HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -5532,6 +5754,7 @@ GetCompressedFileSizeTransactedW(
 #define GetCompressedFileSizeTransacted  GetCompressedFileSizeTransactedA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5539,6 +5762,7 @@ DeleteFileTransactedA(
     _In_     LPCSTR lpFileName,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5562,6 +5786,7 @@ DeleteFileTransactedW(
 
 #if defined(_M_CEE)
 #undef DeleteFile
+_Win32_SetLastError_
 __inline
 BOOL
 DeleteFile(
@@ -5586,6 +5811,7 @@ DeleteFile(
 
 #if _WIN32_WINNT >= 0x0501
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5596,6 +5822,7 @@ CheckNameLegalDOS8Dot3A(
     _Out_opt_ PBOOL pbNameContainsSpaces OPTIONAL,
     _Out_     PBOOL pbNameLegal
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5624,6 +5851,7 @@ CheckNameLegalDOS8Dot3W(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -5636,6 +5864,7 @@ FindFirstFileTransactedA(
     _In_       DWORD dwAdditionalFlags,
     _In_       HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -5665,6 +5894,7 @@ FindFirstFileTransactedW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5673,6 +5903,7 @@ CopyFileA(
     _In_ LPCSTR lpNewFileName,
     _In_ BOOL bFailIfExists
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5689,6 +5920,7 @@ CopyFileW(
 
 #if defined(_M_CEE)
 #undef CopyFile
+_Win32_SetLastError_
 __inline
 BOOL
 CopyFile(
@@ -5731,6 +5963,7 @@ DWORD
     _In_opt_ LPVOID lpData
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5743,6 +5976,7 @@ CopyFileExA(
     _Inout_opt_ LPBOOL pbCancel,
     _In_        DWORD dwCopyFlags
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5769,6 +6003,7 @@ CopyFileExW(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -5781,6 +6016,7 @@ CopyFileTransactedA(
     _In_     DWORD dwCopyFlags,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6073,6 +6309,7 @@ CopyFile2(
 #pragma region Desktop Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6080,6 +6317,7 @@ MoveFileA(
     _In_ LPCSTR lpExistingFileName,
     _In_ LPCSTR lpNewFileName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6095,6 +6333,7 @@ MoveFileW(
 
 #if defined(_M_CEE)
 #undef MoveFile
+_Win32_SetLastError_
 __inline
 BOOL
 MoveFile(
@@ -6119,6 +6358,7 @@ MoveFile(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6127,6 +6367,7 @@ MoveFileExA(
     _In_opt_ LPCSTR lpNewFileName,
     _In_     DWORD    dwFlags
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6148,6 +6389,7 @@ MoveFileExW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
 #if (_WIN32_WINNT >= 0x0500)
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6158,6 +6400,7 @@ MoveFileWithProgressA(
     _In_opt_ LPVOID lpData,
     _In_     DWORD dwFlags
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6182,6 +6425,7 @@ MoveFileWithProgressW(
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
 #if (_WIN32_WINNT >= 0x0600)
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6193,6 +6437,7 @@ MoveFileTransactedA(
     _In_     DWORD dwFlags,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6241,6 +6486,7 @@ MoveFileTransactedW(
 
 #if (_WIN32_WINNT >= 0x0500)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6252,6 +6498,7 @@ ReplaceFileA(
     _Reserved_ LPVOID   lpExclude,
     _Reserved_ LPVOID  lpReserved
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6275,6 +6522,7 @@ ReplaceFileW(
 // API call to create hard links.
 //
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6283,6 +6531,7 @@ CreateHardLinkA(
     _In_       LPCSTR lpExistingFileName,
     _Reserved_ LPSECURITY_ATTRIBUTES lpSecurityAttributes
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6310,6 +6559,7 @@ CreateHardLinkW(
 // API call to create hard links.
 //
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6319,6 +6569,7 @@ CreateHardLinkTransactedA(
     _Reserved_ LPSECURITY_ATTRIBUTES lpSecurityAttributes,
     _In_       HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6338,6 +6589,7 @@ CreateHardLinkTransactedW(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -6349,6 +6601,7 @@ FindFirstStreamTransactedW (
     _In_       HANDLE hTransaction
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -6368,6 +6621,7 @@ FindFirstFileNameTransactedW (
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -6391,6 +6645,7 @@ CreateNamedPipeA(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6407,6 +6662,7 @@ GetNamedPipeHandleStateA(
 #define GetNamedPipeHandleState  GetNamedPipeHandleStateA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6430,6 +6686,7 @@ CallNamedPipeA(
 #pragma region Desktop Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6456,6 +6713,7 @@ WaitNamedPipeA(
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6475,6 +6733,7 @@ GetNamedPipeClientComputerNameA(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6483,6 +6742,7 @@ GetNamedPipeClientProcessId(
     _Out_ PULONG ClientProcessId
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6491,6 +6751,7 @@ GetNamedPipeClientSessionId(
     _Out_ PULONG ClientSessionId
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6499,6 +6760,7 @@ GetNamedPipeServerProcessId(
     _Out_ PULONG ServerProcessId
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6515,6 +6777,7 @@ GetNamedPipeServerSessionId(
 #pragma region Application Family or Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6522,6 +6785,7 @@ SetVolumeLabelA(
     _In_opt_ LPCSTR lpRootPathName,
     _In_opt_ LPCSTR lpVolumeName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6544,6 +6808,7 @@ SetVolumeLabelW(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6556,6 +6821,7 @@ SetFileBandwidthReservation(
     _Out_ LPDWORD lpNumOutstandingRequests
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -6574,6 +6840,7 @@ GetFileBandwidthReservation(
 // Legacy Event logging APIs
 //
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6582,6 +6849,7 @@ ClearEventLogA (
     _In_     HANDLE hEventLog,
     _In_opt_ LPCSTR lpBackupFileName
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6596,6 +6864,7 @@ ClearEventLogW (
 #define ClearEventLog  ClearEventLogA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6604,6 +6873,7 @@ BackupEventLogA (
     _In_ HANDLE hEventLog,
     _In_ LPCSTR lpBackupFileName
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6618,6 +6888,7 @@ BackupEventLogW (
 #define BackupEventLog  BackupEventLogA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6632,6 +6903,7 @@ CloseEventLog (
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6646,6 +6918,7 @@ DeregisterEventSource (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6655,6 +6928,7 @@ NotifyChangeEventLog(
     _In_ HANDLE  hEvent
     );
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6664,6 +6938,7 @@ GetNumberOfEventLogRecords (
     _Out_ PDWORD NumberOfRecords
     );
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6673,6 +6948,7 @@ GetOldestEventLogRecord (
     _Out_ PDWORD OldestRecord
     );
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6681,6 +6957,7 @@ OpenEventLogA (
     _In_opt_ LPCSTR lpUNCServerName,
     _In_     LPCSTR lpSourceName
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6701,6 +6978,7 @@ OpenEventLogW (
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6709,6 +6987,7 @@ RegisterEventSourceA (
     _In_opt_ LPCSTR lpUNCServerName,
     _In_     LPCSTR lpSourceName
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6729,6 +7008,7 @@ RegisterEventSourceW (
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6737,6 +7017,7 @@ OpenBackupEventLogA (
     _In_opt_ LPCSTR lpUNCServerName,
     _In_     LPCSTR lpFileName
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 HANDLE
@@ -6751,6 +7032,7 @@ OpenBackupEventLogW (
 #define OpenBackupEventLog  OpenBackupEventLogA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6764,6 +7046,7 @@ ReadEventLogA (
     _Out_ DWORD      *pnBytesRead,
     _Out_ DWORD      *pnMinNumberOfBytesNeeded
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6789,6 +7072,7 @@ ReadEventLogW (
 #pragma region Application Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6804,6 +7088,7 @@ ReportEventA (
     _In_reads_opt_(wNumStrings) LPCSTR *lpStrings,
     _In_reads_bytes_opt_(dwDataSize) LPVOID lpRawData
     );
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6839,6 +7124,7 @@ typedef struct _EVENTLOG_FULL_INFORMATION
 }
 EVENTLOG_FULL_INFORMATION, *LPEVENTLOG_FULL_INFORMATION;
 
+_Win32_SetLastError_
 _Success_(return != FALSE)
 WINADVAPI
 BOOL
@@ -6906,6 +7192,7 @@ OperationEnd (
 //
 
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -6928,6 +7215,7 @@ AccessCheckAndAuditAlarmA (
 
 #if(_WIN32_WINNT >= 0x0500)
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -6953,6 +7241,7 @@ AccessCheckByTypeAndAuditAlarmA (
 #define AccessCheckByTypeAndAuditAlarm  AccessCheckByTypeAndAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -6978,6 +7267,7 @@ AccessCheckByTypeResultListAndAuditAlarmA (
 #define AccessCheckByTypeResultListAndAuditAlarm  AccessCheckByTypeResultListAndAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7005,6 +7295,7 @@ AccessCheckByTypeResultListAndAuditAlarmByHandleA (
 #endif
 #endif //(_WIN32_WINNT >= 0x0500)
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7026,6 +7317,7 @@ ObjectOpenAuditAlarmA (
 #define ObjectOpenAuditAlarm  ObjectOpenAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7041,6 +7333,7 @@ ObjectPrivilegeAuditAlarmA (
 #define ObjectPrivilegeAuditAlarm  ObjectPrivilegeAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7053,6 +7346,7 @@ ObjectCloseAuditAlarmA (
 #define ObjectCloseAuditAlarm  ObjectCloseAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7065,6 +7359,7 @@ ObjectDeleteAuditAlarmA (
 #define ObjectDeleteAuditAlarm  ObjectDeleteAuditAlarmA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7080,6 +7375,7 @@ PrivilegedServiceAuditAlarmA (
 #endif
 
 #if(_WIN32_WINNT >= 0x0601)
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7101,6 +7397,7 @@ AddConditionalAce (
 #pragma region Desktop Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7113,6 +7410,7 @@ SetFileSecurityA (
 #define SetFileSecurity  SetFileSecurityA
 #endif
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7134,6 +7432,7 @@ GetFileSecurityA (
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
 #if(_WIN32_WINNT >= 0x0400)
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7149,6 +7448,7 @@ ReadDirectoryChangesW(
     );
 
 #if (NTDDI_VERSION >= NTDDI_WIN10_RS3)
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7174,6 +7474,7 @@ ReadDirectoryChangesExW(
 
 #if _WIN32_WINNT >= 0x0600
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_ __out_data_source(FILE)
 LPVOID
@@ -7255,6 +7556,7 @@ IsBadStringPtrW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7267,6 +7569,7 @@ LookupAccountSidA(
     _Inout_ LPDWORD cchReferencedDomainName,
     _Out_ PSID_NAME_USE peUse
     );
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7285,6 +7588,7 @@ LookupAccountSidW(
 #define LookupAccountSid  LookupAccountSidA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7297,6 +7601,7 @@ LookupAccountNameA(
     _Inout_  LPDWORD cchReferencedDomainName,
     _Out_    PSID_NAME_USE peUse
     );
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7351,6 +7656,7 @@ LookupAccountNameLocalW(
 #define LookupAccountNameLocal  LookupAccountNameLocalA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7362,6 +7668,7 @@ LookupAccountSidLocalA(
     _Inout_ LPDWORD cchReferencedDomainName,
     _Out_ PSID_NAME_USE peUse
     );
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7409,6 +7716,7 @@ LookupAccountSidLocalW(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7417,6 +7725,7 @@ LookupPrivilegeValueA(
     _In_     LPCSTR lpName,
     _Out_    PLUID   lpLuid
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7431,6 +7740,7 @@ LookupPrivilegeValueW(
 #define LookupPrivilegeValue  LookupPrivilegeValueA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7440,6 +7750,7 @@ LookupPrivilegeNameA(
     _Out_writes_to_opt_(*cchName, *cchName + 1) LPSTR lpName,
     _Inout_  LPDWORD cchName
     );
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7455,6 +7766,7 @@ LookupPrivilegeNameW(
 #define LookupPrivilegeName  LookupPrivilegeNameA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7465,6 +7777,7 @@ LookupPrivilegeDisplayNameA(
     _Inout_  LPDWORD cchDisplayName,
     _Out_    LPDWORD lpLanguageId
     );
+_Win32_SetLastError_
 WINADVAPI
 _Success_(return != FALSE) BOOL
 WINAPI
@@ -7487,6 +7800,7 @@ LookupPrivilegeDisplayNameW(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7494,6 +7808,7 @@ BuildCommDCBA(
     _In_  LPCSTR lpDef,
     _Out_ LPDCB lpDCB
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7507,6 +7822,7 @@ BuildCommDCBW(
 #define BuildCommDCB  BuildCommDCBA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7515,6 +7831,7 @@ BuildCommDCBAndTimeoutsA(
     _Out_ LPDCB lpDCB,
     _Out_ LPCOMMTIMEOUTS lpCommTimeouts
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7529,6 +7846,7 @@ BuildCommDCBAndTimeoutsW(
 #define BuildCommDCBAndTimeouts  BuildCommDCBAndTimeoutsA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7537,6 +7855,7 @@ CommConfigDialogA(
     _In_opt_ HWND hWnd,
     _Inout_  LPCOMMCONFIG lpCC
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7551,6 +7870,7 @@ CommConfigDialogW(
 #define CommConfigDialog  CommConfigDialogA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7559,6 +7879,7 @@ GetDefaultCommConfigA(
     _Out_writes_bytes_to_(*lpdwSize, *lpdwSize) LPCOMMCONFIG lpCC,
     _Inout_ LPDWORD lpdwSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7573,6 +7894,7 @@ GetDefaultCommConfigW(
 #define GetDefaultCommConfig  GetDefaultCommConfigA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7581,6 +7903,7 @@ SetDefaultCommConfigA(
     _In_reads_bytes_(dwSize) LPCOMMCONFIG lpCC,
     _In_ DWORD dwSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -7607,6 +7930,7 @@ SetDefaultCommConfigW(
 #define MAX_COMPUTERNAME_LENGTH 31
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0)
 BOOL
@@ -7615,6 +7939,7 @@ GetComputerNameA (
     _Out_writes_to_opt_(*nSize, *nSize + 1) LPSTR lpBuffer,
     _Inout_ LPDWORD nSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != 0)
 BOOL
@@ -7638,6 +7963,7 @@ GetComputerNameW (
 #if (_WIN32_WINNT >= 0x0500)
 
 
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -7647,6 +7973,7 @@ DnsHostnameToComputerNameA (
     _Out_writes_to_opt_(*nSize, *nSize + 1) LPSTR ComputerName,
     _Inout_ LPDWORD nSize
     );
+_Win32_SetLastError_
 WINBASEAPI
 _Success_(return != FALSE)
 BOOL
@@ -7664,6 +7991,7 @@ DnsHostnameToComputerNameW (
 
 #endif // _WIN32_WINNT
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7671,6 +7999,7 @@ GetUserNameA (
     _Out_writes_to_opt_(*pcbBuffer, *pcbBuffer) LPSTR lpBuffer,
     _Inout_ LPDWORD pcbBuffer
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7712,6 +8041,7 @@ GetUserNameW (
 
 
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7723,6 +8053,7 @@ LogonUserA (
     _In_        DWORD dwLogonProvider,
     _Outptr_ PHANDLE phToken
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7740,6 +8071,7 @@ LogonUserW (
 #define LogonUser  LogonUserA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7755,6 +8087,7 @@ LogonUserExA (
     _Out_opt_       LPDWORD pdwProfileLength,
     _Out_opt_       PQUOTA_LIMITS pQuotaLimits
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -7796,6 +8129,7 @@ LogonUserExW (
 #define LOGON_NETCREDENTIALS_ONLY       0x00000002
 #define LOGON_ZERO_PASSWORD_BUFFER      0x80000000
 
+_Win32_SetLastError_
 WINADVAPI
 _Must_inspect_result_ BOOL
 WINAPI
@@ -7813,6 +8147,7 @@ CreateProcessWithLogonW(
     _Out_       LPPROCESS_INFORMATION lpProcessInformation
       );
 
+_Win32_SetLastError_
 WINADVAPI
 _Must_inspect_result_ BOOL
 WINAPI
@@ -8124,12 +8459,14 @@ typedef LPHW_PROFILE_INFOA LPHW_PROFILE_INFO;
 #endif // UNICODE
 
 
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
 GetCurrentHwProfileA (
     _Out_ LPHW_PROFILE_INFOA  lpHwProfileInfo
     );
+_Win32_SetLastError_
 WINADVAPI
 BOOL
 WINAPI
@@ -8149,6 +8486,7 @@ GetCurrentHwProfileW (
 #pragma region Desktop Family or Gaming Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8157,6 +8495,7 @@ VerifyVersionInfoA(
     _In_    DWORD dwTypeMask,
     _In_    DWORDLONG dwlConditionMask
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8196,6 +8535,7 @@ VerifyVersionInfoW(
 // Power Management APIs
 //
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8244,6 +8584,7 @@ typedef struct _SYSTEM_POWER_STATUS {
     DWORD BatteryFullLifeTime;
 }   SYSTEM_POWER_STATUS, *LPSYSTEM_POWER_STATUS;
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8279,6 +8620,7 @@ MapUserPhysicalPagesScatter(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -8294,6 +8636,7 @@ CreateJobObjectA(
 #define CreateJobObject  CreateJobObjectA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 _Ret_maybenull_
 HANDLE
@@ -8330,6 +8673,7 @@ CreateJobSet (
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -8341,6 +8685,7 @@ FindFirstVolumeA(
 #define FindFirstVolume FindFirstVolumeA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8359,6 +8704,7 @@ FindNextVolumeA(
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -8367,6 +8713,7 @@ FindFirstVolumeMountPointA(
     _Out_writes_(cchBufferLength) LPSTR lpszVolumeMountPoint,
     _In_ DWORD cchBufferLength
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -8381,6 +8728,7 @@ FindFirstVolumeMountPointW(
 #define FindFirstVolumeMountPoint FindFirstVolumeMountPointA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8389,6 +8737,7 @@ FindNextVolumeMountPointA(
     _Out_writes_(cchBufferLength) LPSTR lpszVolumeMountPoint,
     _In_ DWORD cchBufferLength
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8403,6 +8752,7 @@ FindNextVolumeMountPointW(
 #define FindNextVolumeMountPoint FindNextVolumeMountPointA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8410,6 +8760,7 @@ FindVolumeMountPointClose(
     _In_ HANDLE hFindVolumeMountPoint
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8417,6 +8768,7 @@ SetVolumeMountPointA(
     _In_ LPCSTR lpszVolumeMountPoint,
     _In_ LPCSTR lpszVolumeName
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8436,6 +8788,7 @@ SetVolumeMountPointW(
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8456,6 +8809,7 @@ DeleteVolumeMountPointA(
 #define GetVolumeNameForVolumeMountPoint  GetVolumeNameForVolumeMountPointA
 #endif
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8471,6 +8825,7 @@ GetVolumeNameForVolumeMountPointA(
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8493,6 +8848,7 @@ GetVolumePathNameA(
 
 #if(_WIN32_WINNT >= 0x0501)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8560,12 +8916,14 @@ typedef PCACTCTXA PCACTCTX;
 
 
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
 CreateActCtxA(
     _In_ PCACTCTXA pActCtx
     );
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -8593,6 +8951,7 @@ ReleaseActCtx(
     _Inout_ HANDLE hActCtx
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8601,6 +8960,7 @@ ZombifyActCtx(
     );
 
 
+_Win32_SetLastError_
 _Success_(return)
 WINBASEAPI
 BOOL
@@ -8613,6 +8973,7 @@ ActivateActCtx(
 
 #define DEACTIVATE_ACTCTX_FLAG_FORCE_EARLY_DEACTIVATION (0x00000001)
 
+_Win32_SetLastError_
 _Success_(return)
 WINBASEAPI
 BOOL
@@ -8622,6 +8983,7 @@ DeactivateActCtx(
     _In_ ULONG_PTR ulCookie
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8675,6 +9037,7 @@ typedef const ACTCTX_SECTION_KEYED_DATA * PCACTCTX_SECTION_KEYED_DATA;
 
 
 
+_Win32_SetLastError_
 _Success_(return)
 WINBASEAPI
 BOOL
@@ -8686,6 +9049,7 @@ FindActCtxSectionStringA(
     _In_       LPCSTR lpStringToFind,
     _Out_      PACTCTX_SECTION_KEYED_DATA ReturnedData
     );
+_Win32_SetLastError_
 _Success_(return)
 WINBASEAPI
 BOOL
@@ -8703,6 +9067,7 @@ FindActCtxSectionStringW(
 #define FindActCtxSectionString  FindActCtxSectionStringA
 #endif // !UNICODE
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8765,6 +9130,7 @@ typedef const struct _ACTIVATION_CONTEXT_BASIC_INFORMATION *PCACTIVATION_CONTEXT
 //
 // String are placed after the structs.
 //
+_Win32_SetLastError_
 _Success_(return)
 WINBASEAPI
 BOOL
@@ -8843,6 +9209,7 @@ GetMaximumProcessorGroupCount(
     VOID
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -8850,6 +9217,7 @@ GetActiveProcessorCount(
     _In_ WORD GroupNumber
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
@@ -8863,6 +9231,7 @@ GetMaximumProcessorCount(
 // NUMA Information routines.
 //
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8873,6 +9242,7 @@ GetNumaProcessorNode(
 
 #if _WIN32_WINNT >= 0x0601
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8885,6 +9255,7 @@ GetNumaNodeNumberFromHandle(
 
 #if _WIN32_WINNT >= 0x0601
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8895,6 +9266,7 @@ GetNumaProcessorNodeEx(
 
 #endif // (_WIN32_WINNT >=0x0601)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8903,6 +9275,7 @@ GetNumaNodeProcessorMask(
     _Out_ PULONGLONG ProcessorMask
     );
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8913,6 +9286,7 @@ GetNumaAvailableMemoryNode(
 
 #if _WIN32_WINNT >= 0x0601
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -8925,6 +9299,7 @@ GetNumaAvailableMemoryNodeEx(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -9380,6 +9755,7 @@ typedef struct _FILE_REMOTE_PROTOCOL_INFO
 
 } FILE_REMOTE_PROTOCOL_INFO, *PFILE_REMOTE_PROTOCOL_INFO;
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -9428,6 +9804,7 @@ typedef struct FILE_ID_DESCRIPTOR {
     } DUMMYUNIONNAME;
 } FILE_ID_DESCRIPTOR, *LPFILE_ID_DESCRIPTOR;
 
+_Win32_SetLastError_
 WINBASEAPI
 HANDLE
 WINAPI
@@ -9469,6 +9846,7 @@ OpenFileById (
 #define SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE    (0x2)
 
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOLEAN
 APIENTRY
@@ -9477,6 +9855,7 @@ CreateSymbolicLinkA (
     _In_ LPCSTR lpTargetFileName,
     _In_ DWORD dwFlags
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOLEAN
 APIENTRY
@@ -9495,6 +9874,7 @@ CreateSymbolicLinkW (
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -9518,6 +9898,7 @@ QueryActCtxSettingsW(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Win32_SetLastError_
 WINBASEAPI
 BOOLEAN
 APIENTRY
@@ -9527,6 +9908,7 @@ CreateSymbolicLinkTransactedA (
     _In_     DWORD dwFlags,
     _In_     HANDLE hTransaction
     );
+_Win32_SetLastError_
 WINBASEAPI
 BOOLEAN
 APIENTRY
