diff --git a/generation/WinSDK/RecompiledIdlHeaders/shared/winbio_ioctl.h b/generation/WinSDK/RecompiledIdlHeaders/shared/winbio_ioctl.h
index ab046e2ffc..f02538f405 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/shared/winbio_ioctl.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/shared/winbio_ioctl.h
@@ -20,6 +20,7 @@ Revision History:
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -137,7 +138,7 @@ DEFINE_GUID( GUID_DEVINTERFACE_BIOMETRIC_READER,
 //
 // OUT payload for IOCTL_BIOMETRIC_GET_ATTRIBUTES
 //
-typedef struct _WINBIO_SENSOR_ATTRIBUTES {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_SENSOR_ATTRIBUTES {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     WINBIO_VERSION WinBioVersion;
@@ -156,7 +157,7 @@ typedef struct _WINBIO_SENSOR_ATTRIBUTES {
 // The WINBIO_DATA structure associates a length, in
 // bytes, with an arbitrary block of contiguous memory.
 //
-typedef struct _WINBIO_DATA{
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_DATA{
     DWORD Size;
     BYTE Data[1]; // Defined as a length of 1 to avoid compiler warning.
 } WINBIO_DATA, *PWINBIO_DATA;
@@ -165,7 +166,7 @@ typedef struct _WINBIO_DATA{
 //
 // IN payload for IOCTL_BIOMETRIC_UPDATE_FIRMWARE
 //
-typedef struct _WINBIO_UPDATE_FIRMWARE {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_UPDATE_FIRMWARE {
     DWORD PayloadSize;
     WINBIO_DATA FirmwareData;
 } WINBIO_UPDATE_FIRMWARE, *PWINBIO_UPDATE_FIRMWARE;
@@ -174,7 +175,7 @@ typedef struct _WINBIO_UPDATE_FIRMWARE {
 //
 // OUT payload for IOCTL_BIOMETRIC_CALIBRATE
 //
-typedef struct _WINBIO_CALIBRATION_INFO {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_CALIBRATION_INFO {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     WINBIO_DATA CalibrationData;
@@ -184,7 +185,7 @@ typedef struct _WINBIO_CALIBRATION_INFO {
 //
 // OUT payload for IOCTL_BIOMETRIC_GET_SENSOR_STATUS
 //
-typedef struct _WINBIO_DIAGNOSTICS {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_DIAGNOSTICS {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     WINBIO_SENSOR_STATUS SensorStatus;
@@ -196,7 +197,7 @@ typedef struct _WINBIO_DIAGNOSTICS {
 // OUT payload for IOCTL_BIOMETRIC_RESET
 // OUT payload for IOCTL_BIOMETRIC_UPDATE_FIRMWARE
 //
-typedef struct _WINBIO_BLANK_PAYLOAD {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_BLANK_PAYLOAD {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
 } WINBIO_BLANK_PAYLOAD, *PWINBIO_BLANK_PAYLOAD;
@@ -207,7 +208,7 @@ typedef struct _WINBIO_BLANK_PAYLOAD {
 // IN payload for IOCTL_BIOMETRIC_CAPTURE_DATA
 //
 
-typedef struct _WINBIO_CAPTURE_PARAMETERS {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_CAPTURE_PARAMETERS {
     DWORD PayloadSize;
     WINBIO_BIR_PURPOSE Purpose;
     WINBIO_REGISTERED_FORMAT Format;
@@ -219,7 +220,7 @@ typedef struct _WINBIO_CAPTURE_PARAMETERS {
 // A container for biometric captured data.
 // OUT payload for IOCTL_BIOMETRIC_CAPTURE_DATA
 //
-typedef struct _WINBIO_CAPTURE_DATA {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_CAPTURE_DATA {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     WINBIO_SENSOR_STATUS SensorStatus;
@@ -231,7 +232,7 @@ typedef struct _WINBIO_CAPTURE_DATA {
 //
 // OUT payload for IOCTL_BIOMETRIC_GET_SUPPORTED_ALGORITHMS
 //
-typedef struct _WINBIO_SUPPORTED_ALGORITHMS {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_SUPPORTED_ALGORITHMS {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     DWORD NumberOfAlgorithms;
@@ -242,7 +243,7 @@ typedef struct _WINBIO_SUPPORTED_ALGORITHMS {
 //
 // OUT payload for IOCTL_GET_INDICATOR
 //
-typedef struct _WINBIO_GET_INDICATOR {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_GET_INDICATOR {
     DWORD PayloadSize;
     HRESULT WinBioHresult;
     WINBIO_INDICATOR_STATUS IndicatorStatus;
@@ -252,7 +253,7 @@ typedef struct _WINBIO_GET_INDICATOR {
 //
 // IN payload for IOCTL_SET_INDICATOR
 //
-typedef struct _WINBIO_SET_INDICATOR {
+typedef struct _Windows_SupportedOS_Windows7_ _WINBIO_SET_INDICATOR {
     DWORD PayloadSize;
     WINBIO_INDICATOR_STATUS IndicatorStatus;
 } WINBIO_SET_INDICATOR, *PWINBIO_SET_INDICATOR;
