diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wincodecsdk.h b/generation/WinSDK/RecompiledIdlHeaders/um/wincodecsdk.h
index 776892c852..0ea738df78 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wincodecsdk.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wincodecsdk.h
@@ -9,8 +9,13 @@
 
 
 /* verify that the <rpcndr.h> version is high enough to compile this file*/
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
+
 #ifndef __REQUIRED_RPCNDR_H_VERSION__
 #define __REQUIRED_RPCNDR_H_VERSION__ 500
+
 #endif
 
 /* verify that the <rpcsal.h> version is high enough to compile this file*/
@@ -307,8 +312,9 @@ extern RPC_IF_HANDLE __MIDL_itf_wincodecsdk_0000_0000_v0_0_s_ifspec;
 EXTERN_C const IID IID_IWICMetadataBlockReader;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("FEAA2A8D-B3F3-43E4-B25C-D1DE990A1AE1")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataBlockReader : public IUnknown
     {
     public:
@@ -437,8 +443,9 @@ EXTERN_C const IID IID_IWICMetadataBlockReader;
 EXTERN_C const IID IID_IWICMetadataBlockWriter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("08FB9676-B444-41E8-8DBE-6A53A542BFF1")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataBlockWriter : public IWICMetadataBlockReader
     {
     public:
@@ -623,8 +630,9 @@ EXTERN_C const IID IID_IWICMetadataBlockWriter;
 EXTERN_C const IID IID_IWICMetadataReader;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("9204FE99-D8FC-4FD5-A001-9536B067A899")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataReader : public IUnknown
     {
     public:
@@ -795,8 +803,9 @@ EXTERN_C const IID IID_IWICMetadataReader;
 EXTERN_C const IID IID_IWICMetadataWriter;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("F7836E16-3BE0-470B-86BB-160D0AECD7DE")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataWriter : public IWICMetadataReader
     {
     public:
@@ -1010,8 +1019,9 @@ EXTERN_C const IID IID_IWICMetadataWriter;
 EXTERN_C const IID IID_IWICStreamProvider;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("449494BC-B468-4927-96D7-BA90D31AB505")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICStreamProvider : public IUnknown
     {
     public:
@@ -1132,8 +1142,9 @@ EXTERN_C const IID IID_IWICStreamProvider;
 EXTERN_C const IID IID_IWICPersistStream;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("00675040-6908-45F8-86A3-49C7DFD6D9AD")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICPersistStream : public IPersistStream
     {
     public:
@@ -1295,8 +1306,9 @@ EXTERN_C const IID IID_IWICPersistStream;
 EXTERN_C const IID IID_IWICMetadataHandlerInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("ABA958BF-C672-44D1-8D61-CE6DF2E682C2")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataHandlerInfo : public IWICComponentInfo
     {
     public:
@@ -1590,8 +1602,9 @@ extern RPC_IF_HANDLE __MIDL_itf_wincodecsdk_0000_0007_v0_0_s_ifspec;
 EXTERN_C const IID IID_IWICMetadataReaderInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("EEBF1F5B-07C1-4447-A3AB-22ACAF78A804")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataReaderInfo : public IWICMetadataHandlerInfo
     {
     public:
@@ -1925,8 +1938,9 @@ extern RPC_IF_HANDLE __MIDL_itf_wincodecsdk_0000_0008_v0_0_s_ifspec;
 EXTERN_C const IID IID_IWICMetadataWriterInfo;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("B22E3FBA-3925-4323-B5C1-9EBFC430F236")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICMetadataWriterInfo : public IWICMetadataHandlerInfo
     {
     public:
@@ -2217,8 +2231,9 @@ void __RPC_STUB IWICMetadataWriterInfo_Remote_GetHeader_Stub(
 EXTERN_C const IID IID_IWICComponentFactory;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
-    
     MIDL_INTERFACE("412D0C3A-9650-44FA-AF5B-DD2A06C8E8FB")
+    
+    _Windows_SupportedOS_WindowsXP_
     IWICComponentFactory : public IWICImagingFactory
     {
     public:
@@ -2757,18 +2772,21 @@ EXTERN_C const IID IID_IWICComponentFactory;
 /* interface __MIDL_itf_wincodecsdk_0000_0010 */
 /* [local] */ 
 
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI WICMatchMetadataContent(
     _In_ REFGUID guidContainerFormat,
     _In_opt_  const GUID *pguidVendor,
     _In_  IStream *pIStream,
     _Out_ GUID *pguidMetadataFormat
     );
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI WICSerializeMetadataContent(
     _In_ REFGUID guidContainerFormat,
     _In_ IWICMetadataWriter *pIWriter,
     _In_ DWORD dwPersistOptions,
     _In_ IStream *pIStream
     );
+_Windows_SupportedOS_WindowsXP_
 HRESULT WINAPI WICGetMetadataContentSize(
     _In_ REFGUID guidContainerFormat,
     _In_ IWICMetadataWriter *pIWriter,
