diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wincodec.h b/generation/WinSDK/RecompiledIdlHeaders/um/wincodec.h
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wincodec.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wincodec.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 
 /* this ALWAYS GENERATED file contains the definitions for the interfaces */
@@ -529,7 +530,7 @@ typedef interface ID2D1Device ID2D1Device;
 typedef interface ID2D1Image ID2D1Image;
 typedef UINT32 WICColor;
 
-typedef /* [public] */ struct WICRect
+typedef /* [public] */ struct _Windows_SupportedOS_Server2008_ WICRect
     {
     INT X;
     INT Y;
@@ -543,7 +544,7 @@ typedef /* [wire_marshal] */ void *WICInProcPointer;
 
 #endif
 typedef /* [public] */ 
-enum WICColorContextType
+enum _Windows_SupportedOS_Server2008_ WICColorContextType
     {
         WICColorContextUninitialized	= 0,
         WICColorContextProfile	= 0x1,
@@ -680,7 +681,7 @@ DEFINE_GUID(GUID_WICPixelFormat16bppYQuantizedDctCoefficients,           0xA355F
 DEFINE_GUID(GUID_WICPixelFormat16bppCbQuantizedDctCoefficients,          0xD2C4FF61, 0x56A5, 0x49C2, 0x8B, 0x5C, 0x4C, 0x19, 0x25, 0x96, 0x48, 0x37);
 DEFINE_GUID(GUID_WICPixelFormat16bppCrQuantizedDctCoefficients,          0x2FE354F0, 0x1680, 0x42D8, 0x92, 0x31, 0xE7, 0x3C, 0x05, 0x65, 0xBF, 0xC1);
 typedef /* [public] */ 
-enum WICBitmapCreateCacheOption
+enum _Windows_SupportedOS_Server2008_ WICBitmapCreateCacheOption
     {
         WICBitmapNoCache	= 0,
         WICBitmapCacheOnDemand	= 0x1,
@@ -689,7 +690,7 @@ enum WICBitmapCreateCacheOption
     } 	WICBitmapCreateCacheOption;
 
 typedef /* [public] */ 
-enum WICDecodeOptions
+enum _Windows_SupportedOS_Server2008_ WICDecodeOptions
     {
         WICDecodeMetadataCacheOnDemand	= 0,
         WICDecodeMetadataCacheOnLoad	= 0x1,
@@ -697,7 +698,7 @@ enum WICDecodeOptions
     } 	WICDecodeOptions;
 
 typedef /* [public] */ 
-enum WICBitmapEncoderCacheOption
+enum _Windows_SupportedOS_Server2008_ WICBitmapEncoderCacheOption
     {
         WICBitmapEncoderCacheInMemory	= 0,
         WICBitmapEncoderCacheTempFile	= 0x1,
@@ -706,7 +707,7 @@ enum WICBitmapEncoderCacheOption
     } 	WICBitmapEncoderCacheOption;
 
 typedef /* [public] */ 
-enum WICComponentType
+enum _Windows_SupportedOS_Server2008_ WICComponentType
     {
         WICDecoder	= 0x1,
         WICEncoder	= 0x2,
@@ -719,7 +720,7 @@ enum WICComponentType
     } 	WICComponentType;
 
 typedef /* [public] */ 
-enum WICComponentEnumerateOptions
+enum _Windows_SupportedOS_Server2008_ WICComponentEnumerateOptions
     {
         WICComponentEnumerateDefault	= 0,
         WICComponentEnumerateRefresh	= 0x1,
@@ -729,7 +730,7 @@ enum WICComponentEnumerateOptions
         WICCOMPONENTENUMERATEOPTIONS_FORCE_DWORD	= 0x7fffffff
     } 	WICComponentEnumerateOptions;
 
-typedef /* [public] */ struct WICBitmapPattern
+typedef /* [public] */ struct _Windows_SupportedOS_Server2008_ WICBitmapPattern
     {
     ULARGE_INTEGER Position;
     ULONG Length;
@@ -739,7 +740,7 @@ typedef /* [public] */ struct WICBitmapPattern
     } 	WICBitmapPattern;
 
 typedef /* [public] */ 
-enum WICBitmapInterpolationMode
+enum _Windows_SupportedOS_Server2008_ WICBitmapInterpolationMode
     {
         WICBitmapInterpolationModeNearestNeighbor	= 0,
         WICBitmapInterpolationModeLinear	= 0x1,
@@ -750,7 +751,7 @@ enum WICBitmapInterpolationMode
     } 	WICBitmapInterpolationMode;
 
 typedef /* [public] */ 
-enum WICBitmapPaletteType
+enum _Windows_SupportedOS_Server2008_ WICBitmapPaletteType
     {
         WICBitmapPaletteTypeCustom	= 0,
         WICBitmapPaletteTypeMedianCut	= 0x1,
@@ -770,7 +771,7 @@ enum WICBitmapPaletteType
     } 	WICBitmapPaletteType;
 
 typedef /* [public] */ 
-enum WICBitmapDitherType
+enum _Windows_SupportedOS_Server2008_ WICBitmapDitherType
     {
         WICBitmapDitherTypeNone	= 0,
         WICBitmapDitherTypeSolid	= 0,
@@ -786,7 +787,7 @@ enum WICBitmapDitherType
     } 	WICBitmapDitherType;
 
 typedef /* [public] */ 
-enum WICBitmapAlphaChannelOption
+enum _Windows_SupportedOS_Server2008_ WICBitmapAlphaChannelOption
     {
         WICBitmapUseAlpha	= 0,
         WICBitmapUsePremultipliedAlpha	= 0x1,
@@ -795,7 +796,7 @@ enum WICBitmapAlphaChannelOption
     } 	WICBitmapAlphaChannelOption;
 
 typedef /* [public] */ 
-enum WICBitmapTransformOptions
+enum _Windows_SupportedOS_Server2008_ WICBitmapTransformOptions
     {
         WICBitmapTransformRotate0	= 0,
         WICBitmapTransformRotate90	= 0x1,
@@ -807,7 +808,7 @@ enum WICBitmapTransformOptions
     } 	WICBitmapTransformOptions;
 
 typedef /* [public] */ 
-enum WICBitmapLockFlags
+enum _Windows_SupportedOS_Server2008_ WICBitmapLockFlags
     {
         WICBitmapLockRead	= 0x1,
         WICBitmapLockWrite	= 0x2,
@@ -815,7 +816,7 @@ enum WICBitmapLockFlags
     } 	WICBitmapLockFlags;
 
 typedef /* [public] */ 
-enum WICBitmapDecoderCapabilities
+enum _Windows_SupportedOS_Server2008_ WICBitmapDecoderCapabilities
     {
         WICBitmapDecoderCapabilitySameEncoder	= 0x1,
         WICBitmapDecoderCapabilityCanDecodeAllImages	= 0x2,
@@ -826,7 +827,7 @@ enum WICBitmapDecoderCapabilities
     } 	WICBitmapDecoderCapabilities;
 
 typedef /* [public] */ 
-enum WICProgressOperation
+enum _Windows_SupportedOS_Server2008_ WICProgressOperation
     {
         WICProgressOperationCopyPixels	= 0x1,
         WICProgressOperationWritePixels	= 0x2,
@@ -835,7 +836,7 @@ enum WICProgressOperation
     } 	WICProgressOperation;
 
 typedef /* [public] */ 
-enum WICProgressNotification
+enum _Windows_SupportedOS_Server2008_ WICProgressNotification
     {
         WICProgressNotificationBegin	= 0x10000,
         WICProgressNotificationEnd	= 0x20000,
@@ -845,7 +846,7 @@ enum WICProgressNotification
     } 	WICProgressNotification;
 
 typedef /* [public] */ 
-enum WICComponentSigning
+enum _Windows_SupportedOS_Server2008_ WICComponentSigning
     {
         WICComponentSigned	= 0x1,
         WICComponentUnsigned	= 0x2,
@@ -1056,7 +1057,7 @@ enum WICPngTimeProperties
     } 	WICPngTimeProperties;
 
 typedef /* [public] */ 
-enum WICHeifProperties
+enum _Windows_SupportedOS_17763_ WICHeifProperties
     {
         WICHeifOrientation	= 0x1,
         WICHeifLayeredImageCanvasColor	= 0x2,
@@ -1065,7 +1066,7 @@ enum WICHeifProperties
     } 	WICHeifProperties;
 
 typedef /* [public] */ 
-enum WICHeifHdrProperties
+enum _Windows_SupportedOS_18362_ WICHeifHdrProperties
     {
         WICHeifHdrMaximumLuminanceLevel	= 0x1,
         WICHeifHdrMaximumFrameAverageLuminanceLevel	= 0x2,
@@ -1076,14 +1077,14 @@ enum WICHeifHdrProperties
     } 	WICHeifHdrProperties;
 
 typedef /* [public] */ 
-enum WICWebpAnimProperties
+enum _Windows_SupportedOS_17763_ WICWebpAnimProperties
     {
         WICWebpAnimLoopCount	= 0x1,
         WICWebpAnimProperties_FORCE_DWORD	= 0x7fffffff
     } 	WICWebpAnimProperties;
 
 typedef /* [public] */ 
-enum WICWebpAnmfProperties
+enum _Windows_SupportedOS_17763_ WICWebpAnmfProperties
     {
         WICWebpAnmfFrameDuration	= 0x1,
         WICWebpAnmfProperties_FORCE_DWORD	= 0x7fffffff
@@ -1134,7 +1135,7 @@ enum WICPixelFormatNumericRepresentation
     } 	WICPixelFormatNumericRepresentation;
 
 typedef /* [public] */ 
-enum WICPlanarOptions
+enum _Windows_SupportedOS_Windows81_ WICPlanarOptions
     {
         WICPlanarOptionsDefault	= 0,
         WICPlanarOptionsPreserveSubsampling	= 0x1,
@@ -1167,7 +1168,7 @@ enum WICJpegScanType
     } 	WICJpegScanType;
 
 #if (_WIN32_WINNT >= _WIN32_WINNT_WIN8) || defined(_WIN7_PLATFORM_UPDATE)
-typedef /* [public] */ struct WICImageParameters
+typedef /* [public] */ struct _Windows_SupportedOS_Windows8_ _Windows_SupportedOS_Server2012_ WICImageParameters
     {
     D2D1_PIXEL_FORMAT PixelFormat;
     FLOAT DpiX;
@@ -1179,14 +1180,14 @@ typedef /* [public] */ struct WICImageParameters
     } 	WICImageParameters;
 
 #endif
-typedef /* [public] */ struct WICBitmapPlaneDescription
+typedef /* [public] */ struct _Windows_SupportedOS_Windows81_ WICBitmapPlaneDescription
     {
     WICPixelFormatGUID Format;
     UINT Width;
     UINT Height;
     } 	WICBitmapPlaneDescription;
 
-typedef /* [public] */ struct WICBitmapPlane
+typedef /* [public] */ struct _Windows_SupportedOS_Windows81_ WICBitmapPlane
     {
     WICPixelFormatGUID Format;
     /* [size_is] */ BYTE *pbBuffer;
@@ -9957,12 +9958,12 @@ EXTERN_C const IID IID_IWICImagingFactory3;
 /* [local] */ 
 
 #endif
-HRESULT WINAPI WICConvertBitmapSource(
+HRESULT WINAPI _Windows_SupportedOS_Server2008_ WICConvertBitmapSource(
      _In_ REFWICPixelFormatGUID dstFormat, // Destination pixel format
      _In_ IWICBitmapSource  *pISrc,    // Source bitmap
      _Outptr_ IWICBitmapSource **ppIDst   // Destination bitmap, a copy or addrefed source
      );
-HRESULT WINAPI WICCreateBitmapFromSection(
+HRESULT WINAPI _Windows_SupportedOS_Server2008_ WICCreateBitmapFromSection(
      _In_ UINT width,
      _In_ UINT height,
      _In_ REFWICPixelFormatGUID pixelFormat,
@@ -9981,17 +9982,17 @@ HRESULT WINAPI WICCreateBitmapFromSectionEx(
      _In_ WICSectionAccessLevel desiredAccessLevel,
      _Outptr_ IWICBitmap **ppIBitmap
      );
-HRESULT WINAPI WICMapGuidToShortName(
+HRESULT WINAPI _Windows_SupportedOS_Server2008_ WICMapGuidToShortName(
     _In_ REFGUID guid,
     _In_ UINT cchName,
     _Inout_updates_opt_(cchName) WCHAR *wzName,
     _Out_ UINT *pcchActual
    );
-HRESULT WINAPI WICMapShortNameToGuid(
+HRESULT WINAPI _Windows_SupportedOS_Server2008_ WICMapShortNameToGuid(
     _In_ PCWSTR wzName,
     _Out_ GUID *pguid
    );
-HRESULT WINAPI WICMapSchemaToName(
+HRESULT WINAPI _Windows_SupportedOS_Server2008_ WICMapSchemaToName(
     _In_ REFGUID guidMetadataFormat,
     _In_ LPWSTR pwzSchema,
     _In_ UINT cchName,
@@ -10010,7 +10011,7 @@ HRESULT WINAPI WICMapSchemaToName(
 #define WINCODEC_ERR_ACCESSDENIED                     E_ACCESSDENIED
 #define WINCODEC_ERR_VALUEOVERFLOW                    INTSAFE_E_ARITHMETIC_OVERFLOW
 typedef /* [public] */ 
-enum WICTiffCompressionOption
+enum _Windows_SupportedOS_Server2008_ WICTiffCompressionOption
     {
         WICTiffCompressionDontCare	= 0,
         WICTiffCompressionNone	= 0x1,
@@ -10061,7 +10062,7 @@ enum WICHeifCompressionOption
     } 	WICHeifCompressionOption;
 
 typedef /* [public] */ 
-enum WICNamedWhitePoint
+enum _Windows_SupportedOS_Server2008_ WICNamedWhitePoint
     {
         WICWhitePointDefault	= 0x1,
         WICWhitePointDaylight	= 0x2,
@@ -10078,7 +10079,7 @@ enum WICNamedWhitePoint
     } 	WICNamedWhitePoint;
 
 typedef /* [public] */ 
-enum WICRawCapabilities
+enum _Windows_SupportedOS_Server2008_ WICRawCapabilities
     {
         WICRawCapabilityNotSupported	= 0,
         WICRawCapabilityGetSupported	= 0x1,
@@ -10087,7 +10088,7 @@ enum WICRawCapabilities
     } 	WICRawCapabilities;
 
 typedef /* [public] */ 
-enum WICRawRotationCapabilities
+enum _Windows_SupportedOS_Server2008_ WICRawRotationCapabilities
     {
         WICRawRotationCapabilityNotSupported	= 0,
         WICRawRotationCapabilityGetSupported	= 0x1,
@@ -10096,7 +10097,7 @@ enum WICRawRotationCapabilities
         WICRAWROTATIONCAPABILITIES_FORCE_DWORD	= 0x7fffffff
     } 	WICRawRotationCapabilities;
 
-typedef /* [public] */ struct WICRawCapabilitiesInfo
+typedef /* [public] */ struct _Windows_SupportedOS_Server2008_ WICRawCapabilitiesInfo
     {
     UINT cbSize;
     UINT CodecMajorVersion;
@@ -10119,7 +10120,7 @@ typedef /* [public] */ struct WICRawCapabilitiesInfo
     } 	WICRawCapabilitiesInfo;
 
 typedef /* [public] */ 
-enum WICRawParameterSet
+enum _Windows_SupportedOS_Server2008_ WICRawParameterSet
     {
         WICAsShotParameterSet	= 0x1,
         WICUserAdjustedParameterSet	= 0x2,
@@ -10128,7 +10129,7 @@ enum WICRawParameterSet
     } 	WICRawParameterSet;
 
 typedef /* [public] */ 
-enum WICRawRenderMode
+enum _Windows_SupportedOS_Server2008_ WICRawRenderMode
     {
         WICRawRenderModeDraft	= 0x1,
         WICRawRenderModeNormal	= 0x2,
@@ -10136,13 +10137,13 @@ enum WICRawRenderMode
         WICRAWRENDERMODE_FORCE_DWORD	= 0x7fffffff
     } 	WICRawRenderMode;
 
-typedef /* [public] */ struct WICRawToneCurvePoint
+typedef /* [public] */ struct _Windows_SupportedOS_Server2008_ WICRawToneCurvePoint
     {
     double Input;
     double Output;
     } 	WICRawToneCurvePoint;
 
-typedef /* [public] */ struct WICRawToneCurve
+typedef /* [public] */ struct _Windows_SupportedOS_Server2008_ WICRawToneCurve
     {
     UINT cPoints;
     WICRawToneCurvePoint aPoints[ 1 ];
@@ -10915,7 +10916,7 @@ void __RPC_STUB IWICDevelopRaw_Remote_GetToneCurve_Stub(
 /* [local] */ 
 
 typedef /* [public] */ 
-enum WICDdsDimension
+enum _Windows_SupportedOS_Windows81_ WICDdsDimension
     {
         WICDdsTexture1D	= 0,
         WICDdsTexture2D	= 0x1,
@@ -10925,7 +10926,7 @@ enum WICDdsDimension
     } 	WICDdsDimension;
 
 typedef /* [public] */ 
-enum WICDdsAlphaMode
+enum _Windows_SupportedOS_Windows81_ WICDdsAlphaMode
     {
         WICDdsAlphaModeUnknown	= 0,
         WICDdsAlphaModeStraight	= 0x1,
@@ -10935,7 +10936,7 @@ enum WICDdsAlphaMode
         WICDDSALPHAMODE_FORCE_DWORD	= 0x7fffffff
     } 	WICDdsAlphaMode;
 
-typedef /* [public] */ struct WICDdsParameters
+typedef /* [public] */ struct _Windows_SupportedOS_Windows81_ WICDdsParameters
     {
     UINT Width;
     UINT Height;
@@ -11186,7 +11187,7 @@ EXTERN_C const IID IID_IWICDdsEncoder;
 /* interface __MIDL_itf_wincodec_0000_0048 */
 /* [local] */ 
 
-typedef /* [public] */ struct WICDdsFormatInfo
+typedef /* [public] */ struct _Windows_SupportedOS_Windows81_ WICDdsFormatInfo
     {
     DXGI_FORMAT DxgiFormat;
     UINT BytesPerBlock;
