diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsSideShow.h b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsSideShow.h
index e81272f085..e75e1861ba 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsSideShow.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsSideShow.h
@@ -15,6 +15,7 @@
 
 #pragma once
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
@@ -130,7 +131,7 @@ typedef struct tagSCF_NAVIGATION_EVENT
 {
     CONTENT_ID  PreviousPage;
     CONTENT_ID  TargetPage;
-    UINT32      Button;
+    _Win32_AssociatedEnum_(SCF_BUTTON_IDS) UINT32      Button;
 } SCF_NAVIGATION_EVENT, *PSCF_NAVIGATION_EVENT;
 
 // The data passed with an SCF_EVENT_MENUACTION ApplicationEvent
@@ -138,7 +139,7 @@ typedef struct tagSCF_MENUACTION_EVENT
 {
     CONTENT_ID  PreviousPage;
     CONTENT_ID  TargetPage;
-    UINT32      Button;
+    _Win32_AssociatedEnum_(SCF_BUTTON_IDS) UINT32      Button;
     UINT32      ItemId;
 } SCF_MENUACTION_EVENT, *PSCF_MENUACTION_EVENT;
 
