diff --git a/generation/WinSDK/RecompiledIdlHeaders/winrt/windows.ui.composition.interop.h b/generation/WinSDK/RecompiledIdlHeaders/winrt/windows.ui.composition.interop.h
index dfc256ac31..7a6c5b2b8d 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/winrt/windows.ui.composition.interop.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/winrt/windows.ui.composition.interop.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 //---------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation.  All rights reserved.
@@ -60,6 +61,7 @@ DECLARE_INTERFACE_IID_(ICompositionDrawingSurfaceInterop, IUnknown, "FD04E6E3-FE
 #define INTERFACE ICompositionDrawingSurfaceInterop2
 DECLARE_INTERFACE_IID_(ICompositionDrawingSurfaceInterop2, ICompositionDrawingSurfaceInterop, "41E64AAE-98C0-4239-8E95-A330DD6AA18B")
 {
+    _Windows_SupportedOS_19041_
     IFACEMETHOD(CopySurface)(
         _In_ IUnknown* destinationResource,
         _In_ int destinationOffsetX,
