diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WindowsCeip.h
@@ -15,6 +15,7 @@ Abstract:
 #endif // _MSC_VER
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -25,6 +26,7 @@ extern "C" {
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
+_Windows_SupportedOS_Windows8_
 BOOL
 WINAPI
 CeipIsOptedIn();
