diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winenclaveapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/winenclaveapi.h
index 27c3586661..a0254a4b11 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winenclaveapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winenclaveapi.h
@@ -16,6 +16,8 @@ Abstract:
 #ifndef _WINENCLAVEAPI_
 #define _WINENCLAVEAPI_
 
+#include <win32metadata_annotations.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -27,6 +29,8 @@ extern "C" {
 // Attestation APIs.
 //
 
+_Windows_SupportedOS_16299_
+_Windows_SupportedOS_Server2016_
 STDAPI
 EnclaveGetAttestationReport(
     _In_opt_ const UINT8 EnclaveData[ENCLAVE_REPORT_DATA_LENGTH],
@@ -35,6 +39,8 @@ EnclaveGetAttestationReport(
     _Out_ UINT32 *OutputSize
     );
 
+_Windows_SupportedOS_16299_
+_Windows_SupportedOS_Server2016_
 STDAPI
 EnclaveVerifyAttestationReport(
     _In_ UINT32 EnclaveType,
@@ -46,6 +52,8 @@ EnclaveVerifyAttestationReport(
 // Sealing/unsealing APIs.
 //
 
+_Windows_SupportedOS_16299_
+_Windows_SupportedOS_Server2016_
 STDAPI
 EnclaveSealData(
     _In_reads_bytes_(DataToEncryptSize) const VOID *DataToEncrypt,
@@ -57,6 +65,8 @@ EnclaveSealData(
     _Out_ UINT32 *ProtectedBlobSize
     );
 
+_Windows_SupportedOS_16299_
+_Windows_SupportedOS_Server2016_
 STDAPI
 EnclaveUnsealData(
     _In_reads_bytes_(ProtectedBlobSize) const VOID *ProtectedBlob,
@@ -82,6 +92,8 @@ EnclaveEncryptDataForTrustlet(
 // Enclave properties.
 //
 
+_Windows_SupportedOS_16299_
+_Windows_SupportedOS_Server2016_
 STDAPI
 EnclaveGetEnclaveInformation(
     _In_ UINT32 InformationSize,
