diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winhttp.h b/generation/WinSDK/RecompiledIdlHeaders/um/winhttp.h
index ddcdcb381a..c5728bea76 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winhttp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winhttp.h
@@ -16,6 +16,7 @@ Abstract:
 #define _WINHTTPX_
 
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
@@ -168,7 +169,7 @@ typedef struct _WINHTTP_URL_COMPONENTS
     DWORD   dwStructSize;       // size of this structure. Used in version check
     LPWSTR  lpszScheme;         // pointer to scheme name
     DWORD   dwSchemeLength;     // length of scheme name
-    INTERNET_SCHEME nScheme;    // enumerated scheme type (if known)
+    _Win32_AssociatedEnum_(WINHTTP_INTERNET_SCHEME) INTERNET_SCHEME nScheme;    // enumerated scheme type (if known)
     LPWSTR  lpszHostName;       // pointer to host name
     DWORD   dwHostNameLength;   // length of host name
     INTERNET_PORT nPort;        // converted port number
@@ -196,7 +197,7 @@ typedef LPURL_COMPONENTS LPURL_COMPONENTSW;
 
 typedef struct _WINHTTP_PROXY_INFO
 {
-    DWORD  dwAccessType;      // see WINHTTP_ACCESS_* types below
+    _Win32_AssociatedEnum_(WINHTTP_ACCESS_TYPE) DWORD  dwAccessType;      // see WINHTTP_ACCESS_* types below
     LPWSTR lpszProxy;         // proxy server list
     LPWSTR lpszProxyBypass;   // proxy bypass list
 } WINHTTP_PROXY_INFO, *LPWINHTTP_PROXY_INFO, *PWINHTTP_PROXY_INFO;
@@ -249,16 +250,16 @@ typedef struct _WINHTTP_AUTOPROXY_OPTIONS
 // see WinHttpGetProxyForUrlEx and WinHttpGetProxyResult, use WinHttpFreeProxyResult to free its members.
 //
 
-typedef struct _WINHTTP_PROXY_RESULT_ENTRY
+typedef struct   _WINHTTP_PROXY_RESULT_ENTRY
 {
     BOOL            fProxy;                // Is this a proxy or DIRECT?
     BOOL            fBypass;               // If DIRECT, is it bypassing a proxy (intranet) or is all traffic DIRECT (internet)
-    INTERNET_SCHEME ProxyScheme;           // The scheme of the proxy, SOCKS, HTTP (CERN Proxy), HTTPS (SSL through Proxy)
+    _Win32_AssociatedEnum_(WINHTTP_INTERNET_SCHEME) INTERNET_SCHEME ProxyScheme;           // The scheme of the proxy, SOCKS, HTTP (CERN Proxy), HTTPS (SSL through Proxy)
     PWSTR           pwszProxy;             // Hostname of the proxy.
     INTERNET_PORT   ProxyPort;             // Port of the proxy.
 } WINHTTP_PROXY_RESULT_ENTRY;
 
-typedef struct _WINHTTP_PROXY_RESULT
+typedef struct   _WINHTTP_PROXY_RESULT
 {
     DWORD cEntries;
     WINHTTP_PROXY_RESULT_ENTRY *pEntries;
@@ -939,24 +940,24 @@ typedef struct _WINHTTP_FAILED_CONNECTION_RETRIES
 #define WINHTTP_ENABLE_SPN_SERVER_PORT            0x00000001
 #define WINHTTP_OPTION_SPN_MASK                   WINHTTP_ENABLE_SPN_SERVER_PORT
 
-typedef struct tagWINHTTP_CREDS
+typedef struct   tagWINHTTP_CREDS
 {
     LPSTR lpszUserName;
     LPSTR lpszPassword;
     LPSTR lpszRealm;
-    DWORD dwAuthScheme;
+    _Win32_AssociatedEnum_(WINHTTP_CREDS_AUTHSCHEME) DWORD dwAuthScheme;
     LPSTR lpszHostName;
     DWORD dwPort;
 } WINHTTP_CREDS, *PWINHTTP_CREDS;
 
 // structure for WINHTTP_OPTION_GLOBAL_SERVER_CREDS and
 // WINHTTP_OPTION_GLOBAL_PROXY_CREDS
-typedef struct tagWINHTTP_CREDS_EX
+typedef struct   tagWINHTTP_CREDS_EX
 {
     LPSTR lpszUserName;
     LPSTR lpszPassword;
     LPSTR lpszRealm;
-    DWORD dwAuthScheme;
+    _Win32_AssociatedEnum_(WINHTTP_CREDS_AUTHSCHEME) DWORD dwAuthScheme;
     LPSTR lpszHostName;
     DWORD dwPort;
     LPSTR lpszUrl;
@@ -1584,7 +1585,7 @@ typedef struct _WINHTTP_CURRENT_USER_IE_PROXY_CONFIG
 // prototypes
 //
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ WINHTTPAPI
 WINHTTP_STATUS_CALLBACK
 WINAPI
 WinHttpSetStatusCallback
@@ -1595,28 +1596,28 @@ WinHttpSetStatusCallback
     IN DWORD_PTR dwReserved
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpTimeFromSystemTime
+_Win32_SetLastError_ WinHttpTimeFromSystemTime
 (
     _In_ CONST SYSTEMTIME *pst,  // input GMT time
     _Out_writes_bytes_(WINHTTP_TIME_FORMAT_BUFSIZE) LPWSTR pwszTime // output string buffer
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpTimeToSystemTime
+_Win32_SetLastError_ WinHttpTimeToSystemTime
 (
     _In_z_ LPCWSTR pwszTime,        // NULL terminated string
     _Out_ SYSTEMTIME *pst           // output in GMT time
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpCrackUrl
+_Win32_SetLastError_ WinHttpCrackUrl
 (
     _In_reads_(dwUrlLength) LPCWSTR pwszUrl,
     _In_ DWORD dwUrlLength,
@@ -1624,14 +1625,14 @@ WinHttpCrackUrl
     _Inout_ LPURL_COMPONENTS lpUrlComponents
 );
 
-_Success_(return != FALSE)
+_Windows_SupportedOS_WindowsXP_ _Success_(return != FALSE)
 WINHTTPAPI
 BOOL
 WINAPI
-WinHttpCreateUrl
+_Win32_SetLastError_ WinHttpCreateUrl
 (
     _In_ LPURL_COMPONENTS lpUrlComponents,
-    _In_ DWORD dwFlags,
+    _Win32_AssociatedEnum_(WIN_HTTP_CREATE_URL_FLAGS) _In_ DWORD dwFlags,
     _Out_writes_to_opt_(*pdwUrlLength, *pdwUrlLength) LPWSTR pwszUrl,
     _Inout_ LPDWORD pdwUrlLength
 );
@@ -1642,10 +1643,10 @@ WinHttpCreateUrl
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
-WINHTTPAPI BOOL WINAPI WinHttpCheckPlatform(void);
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ WINHTTPAPI BOOL WINAPI WinHttpCheckPlatform(void);
 
-WINHTTPAPI BOOL WINAPI WinHttpGetDefaultProxyConfiguration( IN OUT WINHTTP_PROXY_INFO * pProxyInfo);
-WINHTTPAPI BOOL WINAPI WinHttpSetDefaultProxyConfiguration( IN WINHTTP_PROXY_INFO * pProxyInfo);
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI BOOL WINAPI _Win32_SetLastError_ WinHttpGetDefaultProxyConfiguration( IN OUT WINHTTP_PROXY_INFO * pProxyInfo);
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI BOOL WINAPI _Win32_SetLastError_ WinHttpSetDefaultProxyConfiguration( IN WINHTTP_PROXY_INFO * pProxyInfo);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM) */
 #pragma endregion
@@ -1653,27 +1654,27 @@ WINHTTPAPI BOOL WINAPI WinHttpSetDefaultProxyConfiguration( IN WINHTTP_PROXY_INF
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ WINHTTPAPI
 HINTERNET
 WINAPI
 WinHttpOpen
 (
     _In_opt_z_ LPCWSTR pszAgentW,
-    _In_ DWORD dwAccessType,
+    _Win32_AssociatedEnum_(WINHTTP_ACCESS_TYPE) _In_ DWORD dwAccessType,
     _In_opt_z_ LPCWSTR pszProxyW,
     _In_opt_z_ LPCWSTR pszProxyBypassW,
     _In_ DWORD dwFlags
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpCloseHandle
+_Win32_SetLastError_ WinHttpCloseHandle
 (
     IN HINTERNET hInternet
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ WINHTTPAPI
 HINTERNET
 WINAPI
 WinHttpConnect
@@ -1684,10 +1685,10 @@ WinHttpConnect
     IN DWORD dwReserved
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpReadData
+_Win32_SetLastError_ WinHttpReadData
 (
     IN HINTERNET hRequest,
     _Out_writes_bytes_to_(dwNumberOfBytesToRead, *lpdwNumberOfBytesRead) __out_data_source(NETWORK) LPVOID lpBuffer,
@@ -1709,10 +1710,10 @@ WinHttpReadDataEx
     _In_reads_bytes_opt_(cbProperty) PVOID pvProperty
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpWriteData
+_Win32_SetLastError_ WinHttpWriteData
 (
     IN HINTERNET hRequest,
     _In_reads_bytes_opt_(dwNumberOfBytesToWrite) LPCVOID lpBuffer,
@@ -1720,20 +1721,20 @@ WinHttpWriteData
     OUT LPDWORD lpdwNumberOfBytesWritten
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpQueryDataAvailable
+_Win32_SetLastError_ WinHttpQueryDataAvailable
 (
     IN HINTERNET hRequest,
     __out_data_source(NETWORK) LPDWORD lpdwNumberOfBytesAvailable
 );
 
-_Success_(return != FALSE)
+_Windows_SupportedOS_WindowsXP_ _Success_(return != FALSE)
 WINHTTPAPI
 BOOL
 WINAPI
-WinHttpQueryOption
+_Win32_SetLastError_ WinHttpQueryOption
 (
     IN HINTERNET hInternet,
     IN DWORD dwOption,
@@ -1741,10 +1742,10 @@ WinHttpQueryOption
     IN OUT LPDWORD lpdwBufferLength
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpSetOption
+_Win32_SetLastError_ WinHttpSetOption
 (
     _In_opt_ HINTERNET hInternet,
     _In_ DWORD dwOption,
@@ -1767,10 +1768,10 @@ WinHttpSetOption
     _In_ DWORD dwBufferLength
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpSetTimeouts
+_Win32_SetLastError_ WinHttpSetTimeouts
 (
     IN HINTERNET    hInternet,           // Session/Request handle.
     IN int          nResolveTimeout,
@@ -1807,7 +1808,7 @@ WinHttpIsHostInProxyBypassList
 // prototypes
 //
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ _Win32_SetLastError_ WINHTTPAPI
 HINTERNET
 WINAPI
 WinHttpOpenRequest
@@ -1818,13 +1819,13 @@ WinHttpOpenRequest
     IN LPCWSTR pwszVersion,
     IN LPCWSTR pwszReferrer OPTIONAL,
     IN LPCWSTR FAR * ppwszAcceptTypes OPTIONAL,
-    IN DWORD dwFlags
+    _Win32_AssociatedEnum_(WINHTTP_OPEN_REQUEST_FLAGS) IN DWORD dwFlags
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpAddRequestHeaders
+_Win32_SetLastError_ WinHttpAddRequestHeaders
 (
     IN HINTERNET hRequest,
     _When_(dwHeadersLength == (DWORD)-1, _In_z_)
@@ -1847,10 +1848,10 @@ WinHttpAddRequestHeadersEx
     _In_reads_(cHeaders) WINHTTP_EXTENDED_HEADER *pHeaders
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpSendRequest
+_Win32_SetLastError_ WinHttpSendRequest
 (
     IN HINTERNET hRequest,
     _In_reads_opt_(dwHeadersLength) LPCWSTR lpszHeaders,
@@ -1861,10 +1862,10 @@ WinHttpSendRequest
     IN DWORD_PTR dwContext
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpSetCredentials
+_Win32_SetLastError_ WinHttpSetCredentials
 (
     IN HINTERNET   hRequest,        // HINTERNET handle returned by WinHttpOpenRequest.
     IN DWORD       AuthTargets,     // Only WINHTTP_AUTH_TARGET_SERVER and
@@ -1881,10 +1882,10 @@ WinHttpSetCredentials
     IN LPVOID      pAuthParams
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpQueryAuthSchemes
+_Win32_SetLastError_ WinHttpQueryAuthSchemes
 (
     IN  HINTERNET   hRequest,             // HINTERNET handle returned by WinHttpOpenRequest
     OUT LPDWORD     lpdwSupportedSchemes, // a bitmap of available Authentication Schemes
@@ -1914,20 +1915,20 @@ WinHttpQueryAuthParams
 #pragma region Application Family or OneCore Family or Games Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpReceiveResponse
+_Win32_SetLastError_ WinHttpReceiveResponse
 (
     IN HINTERNET hRequest,
     IN LPVOID lpReserved
 );
 
-_Success_(return != FALSE)
+_Windows_SupportedOS_WindowsXP_ _Success_(return != FALSE)
 WINHTTPAPI
 BOOL
 WINAPI
-WinHttpQueryHeaders
+_Win32_SetLastError_ WinHttpQueryHeaders
 (
     IN     HINTERNET hRequest,
     IN     DWORD     dwInfoLevel,
@@ -1979,19 +1980,19 @@ WinHttpFreeQueryConnectionGroupResult
 #pragma region Application Family or OneCore Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpDetectAutoProxyConfigUrl
+_Win32_SetLastError_ WinHttpDetectAutoProxyConfigUrl
 (
     DWORD dwAutoDetectFlags,
     _Outptr_result_maybenull_ LPWSTR * ppwstrAutoConfigUrl
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpGetProxyForUrl
+_Win32_SetLastError_ WinHttpGetProxyForUrl
 (
     IN  HINTERNET                   hSession,
     IN  LPCWSTR                     lpcwszUrl,
@@ -1999,19 +2000,19 @@ WinHttpGetProxyForUrl
     OUT WINHTTP_PROXY_INFO *        pProxyInfo
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpCreateProxyResolver
+  WinHttpCreateProxyResolver
 (
     _In_ HINTERNET hSession,
     _Out_ HINTERNET *phResolver
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpGetProxyForUrlEx
+  WinHttpGetProxyForUrlEx
 (
     _In_ HINTERNET hResolver,
     _In_ PCWSTR pcwszUrl,
@@ -2032,10 +2033,10 @@ WinHttpGetProxyForUrlEx2
     _In_opt_ DWORD_PTR pContext
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpGetProxyResult
+  WinHttpGetProxyResult
 (
     _In_ HINTERNET hResolver,
     _Out_ WINHTTP_PROXY_RESULT *pProxyResult
@@ -2050,10 +2051,10 @@ WinHttpGetProxyResultEx
     _Out_ WINHTTP_PROXY_RESULT_EX *pProxyResultEx
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 VOID
 WINAPI
-WinHttpFreeProxyResult
+  WinHttpFreeProxyResult
 (
     _Inout_ WINHTTP_PROXY_RESULT *pProxyResult
 );
@@ -2066,19 +2067,19 @@ WinHttpFreeProxyResultEx
     _Inout_ WINHTTP_PROXY_RESULT_EX *pProxyResultEx
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpResetAutoProxy
+  WinHttpResetAutoProxy
 (
     _In_ HINTERNET hSession,
     _In_ DWORD dwFlags
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_WindowsXP_ WINHTTPAPI
 BOOL
 WINAPI
-WinHttpGetIEProxyConfigForCurrentUser
+_Win32_SetLastError_ WinHttpGetIEProxyConfigForCurrentUser
 (
     IN OUT WINHTTP_CURRENT_USER_IE_PROXY_CONFIG * pProxyConfig
 );
@@ -2166,13 +2167,13 @@ typedef enum _WINHTTP_WEB_SOCKET_CLOSE_STATUS
     WINHTTP_WEB_SOCKET_SECURE_HANDSHAKE_ERROR_CLOSE_STATUS = 1015
 } WINHTTP_WEB_SOCKET_CLOSE_STATUS;
 
-typedef struct _WINHTTP_WEB_SOCKET_ASYNC_RESULT
+typedef struct   _WINHTTP_WEB_SOCKET_ASYNC_RESULT
 {
     WINHTTP_ASYNC_RESULT AsyncResult;
     WINHTTP_WEB_SOCKET_OPERATION Operation;
 } WINHTTP_WEB_SOCKET_ASYNC_RESULT;
 
-typedef struct _WINHTTP_WEB_SOCKET_STATUS
+typedef struct   _WINHTTP_WEB_SOCKET_STATUS
 {
     DWORD dwBytesTransferred;
     WINHTTP_WEB_SOCKET_BUFFER_TYPE eBufferType;
@@ -2181,19 +2182,19 @@ typedef struct _WINHTTP_WEB_SOCKET_STATUS
 #define WINHTTP_WEB_SOCKET_MAX_CLOSE_REASON_LENGTH 123
 #define WINHTTP_WEB_SOCKET_MIN_KEEPALIVE_VALUE 15000
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ _Win32_SetLastError_ WINHTTPAPI
 HINTERNET
 WINAPI
-WinHttpWebSocketCompleteUpgrade
+  WinHttpWebSocketCompleteUpgrade
 (
     _In_ HINTERNET hRequest,
     _In_opt_ DWORD_PTR pContext
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpWebSocketSend
+  WinHttpWebSocketSend
 (
     _In_ HINTERNET hWebSocket,
     _In_ WINHTTP_WEB_SOCKET_BUFFER_TYPE eBufferType,
@@ -2201,10 +2202,10 @@ WinHttpWebSocketSend
     _In_ DWORD dwBufferLength
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpWebSocketReceive
+  WinHttpWebSocketReceive
 (
     _In_ HINTERNET hWebSocket,
     _Out_writes_bytes_to_(dwBufferLength, *pdwBytesRead) PVOID pvBuffer,
@@ -2213,10 +2214,10 @@ WinHttpWebSocketReceive
     _Out_ WINHTTP_WEB_SOCKET_BUFFER_TYPE *peBufferType
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpWebSocketShutdown
+  WinHttpWebSocketShutdown
 (
     _In_ HINTERNET hWebSocket,
     _In_ USHORT usStatus,
@@ -2224,10 +2225,10 @@ WinHttpWebSocketShutdown
     _In_range_(0, WINHTTP_WEB_SOCKET_MAX_CLOSE_REASON_LENGTH) DWORD dwReasonLength
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpWebSocketClose
+  WinHttpWebSocketClose
 (
     _In_ HINTERNET hWebSocket,
     _In_ USHORT usStatus,
@@ -2235,10 +2236,10 @@ WinHttpWebSocketClose
     _In_range_(0, WINHTTP_WEB_SOCKET_MAX_CLOSE_REASON_LENGTH) DWORD dwReasonLength
 );
 
-WINHTTPAPI
+_Windows_SupportedOS_Windows8_ WINHTTPAPI
 DWORD
 WINAPI
-WinHttpWebSocketQueryCloseStatus
+  WinHttpWebSocketQueryCloseStatus
 (
     _In_ HINTERNET hWebSocket,
     _Out_ USHORT *pusStatus,
@@ -2524,5 +2525,127 @@ typedef struct _WINHTTP_FAST_FORWARDING_STATUS
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) */
 #pragma endregion
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("WINHTTP_FLAG_BYPASS_PROXY_CACHE")
+#undef WINHTTP_FLAG_BYPASS_PROXY_CACHE
+#pragma push_macro("WINHTTP_FLAG_ESCAPE_DISABLE")
+#undef WINHTTP_FLAG_ESCAPE_DISABLE
+#pragma push_macro("WINHTTP_FLAG_ESCAPE_DISABLE_QUERY")
+#undef WINHTTP_FLAG_ESCAPE_DISABLE_QUERY
+#pragma push_macro("WINHTTP_FLAG_ESCAPE_PERCENT")
+#undef WINHTTP_FLAG_ESCAPE_PERCENT
+#pragma push_macro("WINHTTP_FLAG_NULL_CODEPAGE")
+#undef WINHTTP_FLAG_NULL_CODEPAGE
+#pragma push_macro("WINHTTP_FLAG_REFRESH")
+#undef WINHTTP_FLAG_REFRESH
+#pragma push_macro("WINHTTP_FLAG_SECURE")
+#undef WINHTTP_FLAG_SECURE
+enum [[clang::flag_enum]] WINHTTP_OPEN_REQUEST_FLAGS : unsigned int
+{
+    WINHTTP_FLAG_BYPASS_PROXY_CACHE = 256U,
+    WINHTTP_FLAG_ESCAPE_DISABLE = 64U,
+    WINHTTP_FLAG_ESCAPE_DISABLE_QUERY = 128U,
+    WINHTTP_FLAG_ESCAPE_PERCENT = 4U,
+    WINHTTP_FLAG_NULL_CODEPAGE = 8U,
+    WINHTTP_FLAG_REFRESH = 256U,
+    WINHTTP_FLAG_SECURE = 8388608U,
+};
+#pragma pop_macro("WINHTTP_FLAG_SECURE")
+#pragma pop_macro("WINHTTP_FLAG_REFRESH")
+#pragma pop_macro("WINHTTP_FLAG_NULL_CODEPAGE")
+#pragma pop_macro("WINHTTP_FLAG_ESCAPE_PERCENT")
+#pragma pop_macro("WINHTTP_FLAG_ESCAPE_DISABLE_QUERY")
+#pragma pop_macro("WINHTTP_FLAG_ESCAPE_DISABLE")
+#pragma pop_macro("WINHTTP_FLAG_BYPASS_PROXY_CACHE")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("ICU_ESCAPE")
+#undef ICU_ESCAPE
+#pragma push_macro("ICU_REJECT_USERPWD")
+#undef ICU_REJECT_USERPWD
+#pragma push_macro("ICU_DECODE")
+#undef ICU_DECODE
+enum WIN_HTTP_CREATE_URL_FLAGS : unsigned int
+{
+    ICU_ESCAPE = 2147483648U,
+    ICU_REJECT_USERPWD = 16384U,
+    ICU_DECODE = 268435456U,
+};
+#pragma pop_macro("ICU_DECODE")
+#pragma pop_macro("ICU_REJECT_USERPWD")
+#pragma pop_macro("ICU_ESCAPE")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("WINHTTP_ACCESS_TYPE_NO_PROXY")
+#undef WINHTTP_ACCESS_TYPE_NO_PROXY
+#pragma push_macro("WINHTTP_ACCESS_TYPE_DEFAULT_PROXY")
+#undef WINHTTP_ACCESS_TYPE_DEFAULT_PROXY
+#pragma push_macro("WINHTTP_ACCESS_TYPE_NAMED_PROXY")
+#undef WINHTTP_ACCESS_TYPE_NAMED_PROXY
+#pragma push_macro("WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY")
+#undef WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY
+enum WINHTTP_ACCESS_TYPE : unsigned int
+{
+    WINHTTP_ACCESS_TYPE_NO_PROXY = 1U,
+    WINHTTP_ACCESS_TYPE_DEFAULT_PROXY = 0U,
+    WINHTTP_ACCESS_TYPE_NAMED_PROXY = 3U,
+    WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY = 4U,
+};
+#pragma pop_macro("WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY")
+#pragma pop_macro("WINHTTP_ACCESS_TYPE_NAMED_PROXY")
+#pragma pop_macro("WINHTTP_ACCESS_TYPE_DEFAULT_PROXY")
+#pragma pop_macro("WINHTTP_ACCESS_TYPE_NO_PROXY")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("WINHTTP_AUTH_SCHEME_BASIC")
+#undef WINHTTP_AUTH_SCHEME_BASIC
+#pragma push_macro("WINHTTP_AUTH_SCHEME_NTLM")
+#undef WINHTTP_AUTH_SCHEME_NTLM
+#pragma push_macro("WINHTTP_AUTH_SCHEME_NEGOTIATE")
+#undef WINHTTP_AUTH_SCHEME_NEGOTIATE
+enum WINHTTP_CREDS_AUTHSCHEME : unsigned int
+{
+    WINHTTP_AUTH_SCHEME_BASIC = 1U,
+    WINHTTP_AUTH_SCHEME_NTLM = 2U,
+    WINHTTP_AUTH_SCHEME_NEGOTIATE = 16U,
+};
+#pragma pop_macro("WINHTTP_AUTH_SCHEME_NEGOTIATE")
+#pragma pop_macro("WINHTTP_AUTH_SCHEME_NTLM")
+#pragma pop_macro("WINHTTP_AUTH_SCHEME_BASIC")
+#endif
+
 #endif // !defined(_WINHTTPX_)
 
+
+#ifdef WIN32METADATA
+#ifndef WIN32METADATA_ENUM_WINHTTP_INTERNET_SCHEME
+#define WIN32METADATA_ENUM_WINHTTP_INTERNET_SCHEME
+#pragma push_macro("WINHTTP_INTERNET_SCHEME_HTTP")
+#undef WINHTTP_INTERNET_SCHEME_HTTP
+#pragma push_macro("WINHTTP_INTERNET_SCHEME_HTTPS")
+#undef WINHTTP_INTERNET_SCHEME_HTTPS
+#pragma push_macro("WINHTTP_INTERNET_SCHEME_FTP")
+#undef WINHTTP_INTERNET_SCHEME_FTP
+#pragma push_macro("WINHTTP_INTERNET_SCHEME_SOCKS")
+#undef WINHTTP_INTERNET_SCHEME_SOCKS
+namespace Windows { namespace MetadataEnumValues {
+namespace WINHTTP_INTERNET_SCHEMEValues {
+enum WINHTTP_INTERNET_SCHEME : int
+{
+    WINHTTP_INTERNET_SCHEME_HTTP = 1,
+    WINHTTP_INTERNET_SCHEME_HTTPS = 2,
+    WINHTTP_INTERNET_SCHEME_FTP = 3,
+    WINHTTP_INTERNET_SCHEME_SOCKS = 4,
+};
+}
+} }
+#pragma pop_macro("WINHTTP_INTERNET_SCHEME_SOCKS")
+#pragma pop_macro("WINHTTP_INTERNET_SCHEME_FTP")
+#pragma pop_macro("WINHTTP_INTERNET_SCHEME_HTTPS")
+#pragma pop_macro("WINHTTP_INTERNET_SCHEME_HTTP")
+#endif
+#endif
