diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h b/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
index 294f7aebb4..2ca27bb7f8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winppi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++ 
 Copyright (c) 1985-1999, Microsoft Corporation
 All rights reserved
@@ -25,9 +26,10 @@ typedef int (CALLBACK* EMFPLAYPROC)( HDC, INT, HANDLE );
 
 #define  EMF_PP_COLOR_OPTIMIZATION  0x01
 
+_Win32_RAIIFree_(CloseHandle, -1, 0) 
 HANDLE WINAPI GdiGetSpoolFileHandle(
     LPWSTR     pwszPrinterName,
-    LPDEVMODEW pDevmode,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPDEVMODEW pDevmode,
     LPWSTR     pwszDocName);
 
 BOOL WINAPI GdiDeleteSpoolFileHandle(
@@ -39,41 +41,46 @@ DWORD WINAPI GdiGetPageCount(
 HDC WINAPI GdiGetDC(
     HANDLE     SpoolFileHandle);
 
-HANDLE WINAPI GdiGetPageHandle(
+_Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE WINAPI GdiGetPageHandle(
     HANDLE     SpoolFileHandle,
     DWORD      Page,
-    LPDWORD    pdwPageType);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") LPDWORD    pdwPageType);
+
 
 BOOL WINAPI GdiStartDocEMF(
     HANDLE     SpoolFileHandle,
-    DOCINFOW   *pDocInfo);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") DOCINFOW   *pDocInfo);
+
 
 BOOL WINAPI GdiStartPageEMF(
     HANDLE     SpoolFileHandle);
 
+
 BOOL WINAPI GdiPlayPageEMF(
     HANDLE     SpoolFileHandle,
     HANDLE     hemf,
-    RECT       *prectDocument,
-    RECT       *prectBorder,
-    RECT       *prectClip);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") RECT       *prectDocument,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") RECT       *prectBorder,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") RECT       *prectClip);
+
 
 BOOL WINAPI GdiEndPageEMF(
     HANDLE     SpoolFileHandle,
     DWORD      dwOptimization);
 
+
 BOOL WINAPI GdiEndDocEMF(
     HANDLE     SpoolFileHandle);
 
 BOOL WINAPI GdiGetDevmodeForPage(
     HANDLE     SpoolFileHandle,
     DWORD      dwPageNumber,
-    PDEVMODEW  *pCurrDM,
-    PDEVMODEW  *pLastDM);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PDEVMODEW  *pCurrDM,
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PDEVMODEW  *pLastDM);
 
 BOOL WINAPI GdiResetDCEMF(
     HANDLE     SpoolFileHandle,
-    PDEVMODEW  pCurrDM);
+    _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") PDEVMODEW  pCurrDM);
 
 #endif /* WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) */
 #pragma endregion
