diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winscard.h b/generation/WinSDK/RecompiledIdlHeaders/um/winscard.h
index c7969d4d8e..5c4a00a1e2 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winscard.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winscard.h
@@ -34,6 +34,7 @@ Notes:
 #include "SCardErr.h"
 #endif
 #include <winapifamily.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -108,19 +109,19 @@ typedef SCARDHANDLE *PSCARDHANDLE, *LPSCARDHANDLE;
                                 // application must have appropriate access
                                 // permissions for any database actions.)
 
-extern WINSCARDAPI LONG WINAPI
-SCardEstablishContext(
-    _In_  DWORD dwScope,
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardEstablishContext(
+    _Win32_AssociatedEnum_(SCARD_SCOPE) _In_  DWORD dwScope,
     _Reserved_  LPCVOID pvReserved1,
     _Reserved_  LPCVOID pvReserved2,
     _Out_ LPSCARDCONTEXT phContext);
 
-extern WINSCARDAPI LONG WINAPI
-SCardReleaseContext(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardReleaseContext(
     _In_      SCARDCONTEXT hContext);
 
-extern WINSCARDAPI LONG WINAPI
-SCardIsValidContext(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIsValidContext(
     _In_      SCARDCONTEXT hContext);
 
 
@@ -146,13 +147,13 @@ SCardIsValidContext(
 // Database Reader routines
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardListReaderGroupsA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardListReaderGroupsA(
     _In_    SCARDCONTEXT hContext,
     _Out_writes_opt_(*pcchGroups) _Post_ _NullNull_terminated_  LPSTR mszGroups,
     _Inout_ LPDWORD pcchGroups);
-extern WINSCARDAPI LONG WINAPI
-SCardListReaderGroupsW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardListReaderGroupsW(
     _In_    SCARDCONTEXT hContext,
     _Out_writes_opt_(*pcchGroups) _Post_ _NullNull_terminated_  LPWSTR mszGroups,
     _Inout_ LPDWORD pcchGroups);
@@ -162,18 +163,18 @@ SCardListReaderGroupsW(
 #define SCardListReaderGroups  SCardListReaderGroupsA
 #endif // !UNICODE
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListReadersA(
+  SCardListReadersA(
     _In_     SCARDCONTEXT hContext,
     _In_opt_ LPCSTR mszGroups,
     _When_(_Old_(*pcchReaders) == SCARD_AUTOALLOCATE, _At_((LPSTR *)mszReaders, _Outptr_result_buffer_maybenull_(*pcchReaders) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
     _When_(_Old_(*pcchReaders) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchReaders) _Post_ _NullNull_terminated_)
              LPSTR mszReaders,
     _Inout_  LPDWORD pcchReaders);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListReadersW(
+  SCardListReadersW(
     _In_     SCARDCONTEXT hContext,
     _In_opt_ LPCWSTR mszGroups,
     _When_(_Old_(*pcchReaders) == SCARD_AUTOALLOCATE, _At_((LPWSTR *)mszReaders, _Outptr_result_buffer_maybenull_(*pcchReaders) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
@@ -186,9 +187,9 @@ SCardListReadersW(
 #define SCardListReaders  SCardListReadersA
 #endif // !UNICODE
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListCardsA(
+  SCardListCardsA(
     _In_      SCARDCONTEXT hContext,
     _In_opt_  LPCBYTE pbAtr,
     _In_reads_opt_(cguidInterfaceCount)  LPCGUID rgquidInterfaces,
@@ -197,9 +198,9 @@ SCardListCardsA(
     _When_(_Old_(*pcchCards) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchCards) _Post_ _NullNull_terminated_)
               CHAR *mszCards,
     _Inout_   LPDWORD pcchCards);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListCardsW(
+  SCardListCardsW(
     _In_      SCARDCONTEXT hContext,
     _In_opt_  LPCBYTE pbAtr,
     _In_reads_opt_(cguidInterfaceCount)  LPCGUID rgquidInterfaces,
@@ -229,14 +230,14 @@ SCardListCardsW(
 //          Here's a work-around MACRO:
 #define SCardListCardTypes SCardListCards
 
-extern WINSCARDAPI LONG WINAPI
-SCardListInterfacesA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardListInterfacesA(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCSTR szCard,
     _Out_    LPGUID pguidInterfaces,
     _Inout_  LPDWORD pcguidInterfaces);
-extern WINSCARDAPI LONG WINAPI
-SCardListInterfacesW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardListInterfacesW(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCWSTR szCard,
     _Out_    LPGUID pguidInterfaces,
@@ -247,13 +248,13 @@ SCardListInterfacesW(
 #define SCardListInterfaces  SCardListInterfacesA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardGetProviderIdA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardGetProviderIdA(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCSTR szCard,
     _Out_    LPGUID pguidProviderId);
-extern WINSCARDAPI LONG WINAPI
-SCardGetProviderIdW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardGetProviderIdW(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCWSTR szCard,
     _Out_    LPGUID pguidProviderId);
@@ -267,9 +268,9 @@ SCardGetProviderIdW(
 //          The PC/SC definition uses BYTEs.
 //
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardGetCardTypeProviderNameA(
+  SCardGetCardTypeProviderNameA(
     _In_      SCARDCONTEXT hContext,
     _In_      LPCSTR szCardName,
     _In_      DWORD dwProviderId,
@@ -277,9 +278,9 @@ SCardGetCardTypeProviderNameA(
     _When_(_Old_(*pcchProvider) != SCARD_AUTOALLOCATE, _Out_writes_to_(*pcchProvider, *pcchProvider) _Post_z_)
               CHAR *szProvider,
     _Inout_   LPDWORD pcchProvider);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardGetCardTypeProviderNameW(
+  SCardGetCardTypeProviderNameW(
     _In_      SCARDCONTEXT hContext,
     _In_      LPCWSTR szCardName,
     _In_      DWORD dwProviderId,
@@ -301,12 +302,12 @@ SCardGetCardTypeProviderNameW(
 // Database Writer routines
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceReaderGroupA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceReaderGroupA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szGroupName);
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceReaderGroupW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceReaderGroupW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szGroupName);
 #ifdef UNICODE
@@ -315,12 +316,12 @@ SCardIntroduceReaderGroupW(
 #define SCardIntroduceReaderGroup  SCardIntroduceReaderGroupA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardForgetReaderGroupA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetReaderGroupA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szGroupName);
-extern WINSCARDAPI LONG WINAPI
-SCardForgetReaderGroupW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetReaderGroupW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szGroupName);
 #ifdef UNICODE
@@ -329,13 +330,13 @@ SCardForgetReaderGroupW(
 #define SCardForgetReaderGroup  SCardForgetReaderGroupA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceReaderA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceReaderA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szReaderName,
     _In_ LPCSTR szDeviceName);
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceReaderW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceReaderW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szReaderName,
     _In_ LPCWSTR szDeviceName);
@@ -345,12 +346,12 @@ SCardIntroduceReaderW(
 #define SCardIntroduceReader  SCardIntroduceReaderA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardForgetReaderA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetReaderA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szReaderName);
-extern WINSCARDAPI LONG WINAPI
-SCardForgetReaderW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetReaderW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szReaderName);
 #ifdef UNICODE
@@ -359,13 +360,13 @@ SCardForgetReaderW(
 #define SCardForgetReader  SCardForgetReaderA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardAddReaderToGroupA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardAddReaderToGroupA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szReaderName,
     _In_ LPCSTR szGroupName);
-extern WINSCARDAPI LONG WINAPI
-SCardAddReaderToGroupW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardAddReaderToGroupW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szReaderName,
     _In_ LPCWSTR szGroupName);
@@ -375,13 +376,13 @@ SCardAddReaderToGroupW(
 #define SCardAddReaderToGroup  SCardAddReaderToGroupA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardRemoveReaderFromGroupA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardRemoveReaderFromGroupA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szReaderName,
     _In_ LPCSTR szGroupName);
-extern WINSCARDAPI LONG WINAPI
-SCardRemoveReaderFromGroupW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardRemoveReaderFromGroupW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szReaderName,
     _In_ LPCWSTR szGroupName);
@@ -391,8 +392,8 @@ SCardRemoveReaderFromGroupW(
 #define SCardRemoveReaderFromGroup  SCardRemoveReaderFromGroupA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceCardTypeA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceCardTypeA(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCSTR szCardName,
     _In_opt_ LPCGUID pguidPrimaryProvider,
@@ -401,8 +402,8 @@ SCardIntroduceCardTypeA(
     _In_     LPCBYTE pbAtr,
     _In_     LPCBYTE pbAtrMask,
     _In_     DWORD cbAtrLen);
-extern WINSCARDAPI LONG WINAPI
-SCardIntroduceCardTypeW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardIntroduceCardTypeW(
     _In_     SCARDCONTEXT hContext,
     _In_     LPCWSTR szCardName,
     _In_opt_ LPCGUID pguidPrimaryProvider,
@@ -435,14 +436,14 @@ SCardIntroduceCardTypeW(
 #define PCSCardIntroduceCardType(hContext, szCardName, pbAtr, pbAtrMask, cbAtrLen, pguidPrimaryProvider, rgguidInterfaces, dwInterfaceCount) \
           SCardIntroduceCardType(hContext, szCardName, pguidPrimaryProvider, rgguidInterfaces, dwInterfaceCount, pbAtr, pbAtrMask, cbAtrLen)
 
-extern WINSCARDAPI LONG WINAPI
-SCardSetCardTypeProviderNameA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardSetCardTypeProviderNameA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szCardName,
     _In_ DWORD dwProviderId,
     _In_ LPCSTR szProvider);
-extern WINSCARDAPI LONG WINAPI
-SCardSetCardTypeProviderNameW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardSetCardTypeProviderNameW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szCardName,
     _In_ DWORD dwProviderId,
@@ -456,12 +457,12 @@ SCardSetCardTypeProviderNameW(
 // NOTE:    This routine is an extention to the PC/SC specifications.
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardForgetCardTypeA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetCardTypeA(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCSTR szCardName);
-extern WINSCARDAPI LONG WINAPI
-SCardForgetCardTypeW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardForgetCardTypeW(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCWSTR szCardName);
 #ifdef UNICODE
@@ -480,17 +481,17 @@ SCardForgetCardTypeW(
 //      Manager API.
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardFreeMemory(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardFreeMemory(
     _In_ SCARDCONTEXT hContext,
     _In_ LPCVOID pvMem);
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
-extern WINSCARDAPI HANDLE WINAPI
-SCardAccessStartedEvent(void);
+_Windows_SupportedOS_WindowsXP_ _Win32_RAIIFree_(CloseHandle, -1, 0) extern WINSCARDAPI HANDLE WINAPI
+  _Win32_SetLastError_ SCardAccessStartedEvent(void);
 
-extern WINSCARDAPI void WINAPI
-SCardReleaseStartedEvent(void);
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI void WINAPI
+  SCardReleaseStartedEvent(void);
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
 //
@@ -501,19 +502,19 @@ SCardReleaseStartedEvent(void);
 //      The following services supply means for tracking cards within readers.
 //
 
-typedef struct {
+typedef struct   {
     LPCSTR      szReader;       // reader name
     LPVOID      pvUserData;     // user defined data
-    DWORD       dwCurrentState; // current state of reader at time of call
-    DWORD       dwEventState;   // state of reader after state change
+    _Win32_AssociatedEnum_(SCARD_STATE) DWORD       dwCurrentState; // current state of reader at time of call
+    _Win32_AssociatedEnum_(SCARD_STATE) DWORD       dwEventState;   // state of reader after state change
     DWORD       cbAtr;          // Number of bytes in the returned ATR.
     BYTE        rgbAtr[36];     // Atr of inserted card, (extra alignment bytes)
 } SCARD_READERSTATEA, *PSCARD_READERSTATEA, *LPSCARD_READERSTATEA;
-typedef struct {
+typedef struct   {
     LPCWSTR     szReader;       // reader name
     LPVOID      pvUserData;     // user defined data
-    DWORD       dwCurrentState; // current state of reader at time of call
-    DWORD       dwEventState;   // state of reader after state change
+    _Win32_AssociatedEnum_(SCARD_STATE) DWORD       dwCurrentState; // current state of reader at time of call
+    _Win32_AssociatedEnum_(SCARD_STATE) DWORD       dwEventState;   // state of reader after state change
     DWORD       cbAtr;          // Number of bytes in the returned ATR.
     BYTE        rgbAtr[36];     // Atr of inserted card, (extra alignment bytes)
 } SCARD_READERSTATEW, *PSCARD_READERSTATEW, *LPSCARD_READERSTATEW;
@@ -597,14 +598,14 @@ typedef LPSCARD_READERSTATEA LPSCARD_READERSTATE;
 #define SCARD_STATE_UNPOWERED   0x00000400  // This implies that the card in the
                                             // reader has not been powered up.
 
-extern WINSCARDAPI LONG WINAPI
-SCardLocateCardsA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardLocateCardsA(
     _In_    SCARDCONTEXT hContext,
     _In_    LPCSTR mszCards,
     _Inout_ LPSCARD_READERSTATEA rgReaderStates,
     _In_    DWORD cReaders);
-extern WINSCARDAPI LONG WINAPI
-SCardLocateCardsW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardLocateCardsW(
     _In_    SCARDCONTEXT hContext,
     _In_    LPCWSTR mszCards,
     _Inout_ LPSCARD_READERSTATEW rgReaderStates,
@@ -616,22 +617,22 @@ SCardLocateCardsW(
 #endif // !UNICODE
 
 #if (NTDDI_VERSION >= NTDDI_WINXP)
-typedef struct _SCARD_ATRMASK {
+typedef struct   _SCARD_ATRMASK {
     DWORD       cbAtr;          // Number of bytes in the ATR and the mask.
     BYTE        rgbAtr[36];     // Atr of card (extra alignment bytes)
     BYTE        rgbMask[36];    // Mask for the Atr (extra alignment bytes)
 } SCARD_ATRMASK, *PSCARD_ATRMASK, *LPSCARD_ATRMASK;
 
 
-extern WINSCARDAPI LONG WINAPI
-SCardLocateCardsByATRA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardLocateCardsByATRA(
     _In_    SCARDCONTEXT hContext,
     _In_    LPSCARD_ATRMASK rgAtrMasks,
     _In_    DWORD cAtrs,
     _Inout_ LPSCARD_READERSTATEA rgReaderStates,
     _In_    DWORD cReaders);
-extern WINSCARDAPI LONG WINAPI
-SCardLocateCardsByATRW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardLocateCardsByATRW(
     _In_    SCARDCONTEXT hContext,
     _In_    LPSCARD_ATRMASK rgAtrMasks,
     _In_    DWORD cAtrs,
@@ -644,14 +645,14 @@ SCardLocateCardsByATRW(
 #endif // !UNICODE
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
-extern WINSCARDAPI LONG WINAPI
-SCardGetStatusChangeA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardGetStatusChangeA(
     _In_    SCARDCONTEXT hContext,
     _In_    DWORD dwTimeout,
     _Inout_ LPSCARD_READERSTATEA rgReaderStates,
     _In_    DWORD cReaders);
-extern WINSCARDAPI LONG WINAPI
-SCardGetStatusChangeW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardGetStatusChangeW(
     _In_    SCARDCONTEXT hContext,
     _In_    DWORD dwTimeout,
     _Inout_ LPSCARD_READERSTATEW rgReaderStates,
@@ -662,8 +663,8 @@ SCardGetStatusChangeW(
 #define SCardGetStatusChange  SCardGetStatusChangeA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardCancel(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardCancel(
     _In_    SCARDCONTEXT hContext);
 
 
@@ -688,16 +689,16 @@ SCardCancel(
 #define SCARD_UNPOWER_CARD    2 // Power down the card on close
 #define SCARD_EJECT_CARD      3 // Eject the card on close
 
-extern WINSCARDAPI LONG WINAPI
-SCardConnectA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardConnectA(
     _In_    SCARDCONTEXT hContext,
     _In_    LPCSTR szReader,
     _In_    DWORD dwShareMode,
     _In_    DWORD dwPreferredProtocols,
     _Out_   LPSCARDHANDLE phCard,
     _Out_   LPDWORD pdwActiveProtocol);
-extern WINSCARDAPI LONG WINAPI
-SCardConnectW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardConnectW(
     _In_    SCARDCONTEXT hContext,
     _In_    LPCWSTR szReader,
     _In_    DWORD dwShareMode,
@@ -710,25 +711,25 @@ SCardConnectW(
 #define SCardConnect  SCardConnectA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardReconnect(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardReconnect(
     _In_      SCARDHANDLE hCard,
     _In_      DWORD dwShareMode,
     _In_      DWORD dwPreferredProtocols,
     _In_      DWORD dwInitialization,
     _Out_opt_ LPDWORD pdwActiveProtocol);
 
-extern WINSCARDAPI LONG WINAPI
-SCardDisconnect(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardDisconnect(
     _In_    SCARDHANDLE hCard,
     _In_    DWORD dwDisposition);
 
-extern WINSCARDAPI LONG WINAPI
-SCardBeginTransaction(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardBeginTransaction(
     _In_    SCARDHANDLE hCard);
 
-extern WINSCARDAPI LONG WINAPI
-SCardEndTransaction(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardEndTransaction(
     _In_    SCARDHANDLE hCard,
     _In_    DWORD dwDisposition);
 
@@ -753,8 +754,8 @@ SCardState(
 //          SCardStatus.
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardStatusA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardStatusA(
     _In_        SCARDHANDLE hCard,
     _When_(_Old_(*pcchReaderLen) == SCARD_AUTOALLOCATE, _At_((LPSTR *)mszReaderNames, _Outptr_result_buffer_maybenull_(*pcchReaderLen) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
     _When_(_Old_(*pcchReaderLen) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchReaderLen) _Post_ _NullNull_terminated_)
@@ -766,8 +767,8 @@ SCardStatusA(
     _When_(_Old_(*pcbAtrLen) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcbAtrLen) _Post_ _NullNull_terminated_)
                 LPBYTE pbAtr,
     _Inout_opt_ LPDWORD pcbAtrLen);
-extern WINSCARDAPI LONG WINAPI
-SCardStatusW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardStatusW(
     _In_        SCARDHANDLE hCard,
     _When_(_Old_(*pcchReaderLen) == SCARD_AUTOALLOCATE, _At_((LPWSTR *)mszReaderNames, _Outptr_result_buffer_maybenull_(*pcchReaderLen) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
     _When_(_Old_(*pcchReaderLen) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchReaderLen) _Post_ _NullNull_terminated_)
@@ -785,8 +786,8 @@ SCardStatusW(
 #define SCardStatus  SCardStatusA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardTransmit(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardTransmit(
     _In_        SCARDHANDLE hCard,
     _In_        LPCSCARD_IO_REQUEST pioSendPci,
     _In_reads_bytes_(cbSendLength) LPCBYTE pbSendBuffer,
@@ -796,8 +797,8 @@ SCardTransmit(
     _Inout_     LPDWORD pcbRecvLength);
 
 #if (NTDDI_VERSION >= NTDDI_VISTA)
-extern WINSCARDAPI LONG WINAPI
-SCardGetTransmitCount(
+_Windows_SupportedOS_WindowsVista_ extern WINSCARDAPI LONG WINAPI
+  SCardGetTransmitCount(
     _In_ SCARDHANDLE hCard,
     _Out_ LPDWORD pcTransmitCount);
 #endif // (NTDDI_VERSION >= NTDDI_VISTA)
@@ -812,8 +813,8 @@ SCardGetTransmitCount(
 //      attributes of the communications with the card.
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardControl(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardControl(
     _In_    SCARDHANDLE hCard,
     _In_    DWORD dwControlCode,
     _In_reads_bytes_(cbInBufferSize) LPCVOID lpInBuffer,
@@ -822,8 +823,8 @@ SCardControl(
     _In_    DWORD cbOutBufferSize,
     _Out_   LPDWORD lpBytesReturned);
 
-extern WINSCARDAPI LONG WINAPI
-SCardGetAttrib(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardGetAttrib(
     _In_    SCARDHANDLE hCard,
     _In_    DWORD dwAttrId,
     _Out_writes_bytes_opt_(*pcbAttrLen) LPBYTE pbAttr,
@@ -842,8 +843,8 @@ SCardGetAttrib(
 //          Here's a work-around MACRO:
 #define SCardGetReaderCapabilities SCardGetAttrib
 
-extern WINSCARDAPI LONG WINAPI
-SCardSetAttrib(
+_Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardSetAttrib(
     _In_ SCARDHANDLE hCard,
     _In_ DWORD dwAttrId,
     _In_reads_bytes_(cbAttrLen) LPCBYTE pbAttr,
@@ -901,7 +902,7 @@ typedef void (WINAPI *LPOCNDSCPROC) (_In_ SCARDCONTEXT, _In_ SCARDHANDLE, _In_ P
 // will be used.
 //
 
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;
     LPSTR           lpstrGroupNames;        // OPTIONAL reader groups to include in
     DWORD           nMaxGroupNames;         //          search.  NULL defaults to
@@ -917,7 +918,7 @@ typedef struct {
     DWORD           dwShareMode;            // OPTIONAL must be set if lpfnCheck is not null
     DWORD           dwPreferredProtocols;   // OPTIONAL
 } OPENCARD_SEARCH_CRITERIAA, *POPENCARD_SEARCH_CRITERIAA, *LPOPENCARD_SEARCH_CRITERIAA;
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;
     LPWSTR          lpstrGroupNames;        // OPTIONAL reader groups to include in
     DWORD           nMaxGroupNames;         //          search.  NULL defaults to
@@ -948,7 +949,7 @@ typedef LPOPENCARD_SEARCH_CRITERIAA LPOPENCARD_SEARCH_CRITERIA;
 // OPENCARDNAME_EX: used by SCardUIDlgSelectCard; replaces obsolete OPENCARDNAME
 //
 
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;           // REQUIRED
     SCARDCONTEXT    hSCardContext;          // REQUIRED
     HWND            hwndOwner;              // OPTIONAL
@@ -969,7 +970,7 @@ typedef struct {
     DWORD           dwActiveProtocol;       // [OUT] set only if dwShareMode not NULL
     SCARDHANDLE     hCardHandle;            // [OUT] set if a card connection was indicated
 } OPENCARDNAME_EXA, *POPENCARDNAME_EXA, *LPOPENCARDNAME_EXA;
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;           // REQUIRED
     SCARDCONTEXT    hSCardContext;          // REQUIRED
     HWND            hwndOwner;              // OPTIONAL
@@ -1142,11 +1143,11 @@ typedef struct {
 // SCardUIDlgSelectCard replaces GetOpenCardName
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardUIDlgSelectCardA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardUIDlgSelectCardA(
     LPOPENCARDNAMEA_EX);
-extern WINSCARDAPI LONG WINAPI
-SCardUIDlgSelectCardW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  SCardUIDlgSelectCardW(
     LPOPENCARDNAMEW_EX);
 #ifdef UNICODE
 #define SCardUIDlgSelectCard  SCardUIDlgSelectCardW
@@ -1160,7 +1161,7 @@ SCardUIDlgSelectCardW(
 //  with the Smart Card Base Services SDK version 1.0
 //
 
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;
     HWND            hwndOwner;
     SCARDCONTEXT    hSCardContext;
@@ -1185,7 +1186,7 @@ typedef struct {
     LPOCNDSCPROC    lpfnDisconnect;
     SCARDHANDLE     hCardHandle;
 } OPENCARDNAMEA, *POPENCARDNAMEA, *LPOPENCARDNAMEA;
-typedef struct {
+typedef struct   {
     DWORD           dwStructSize;
     HWND            hwndOwner;
     SCARDCONTEXT    hSCardContext;
@@ -1228,11 +1229,11 @@ typedef LPOPENCARDNAMEA LPOPENCARDNAME;
 #define LPOPENCARDNAME_A LPOPENCARDNAMEA
 #define LPOPENCARDNAME_W LPOPENCARDNAMEW
 
-extern WINSCARDAPI LONG WINAPI
-GetOpenCardNameA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  GetOpenCardNameA(
     LPOPENCARDNAMEA);
-extern WINSCARDAPI LONG WINAPI
-GetOpenCardNameW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_ extern WINSCARDAPI LONG WINAPI
+  GetOpenCardNameW(
     LPOPENCARDNAMEW);
 #ifdef UNICODE
 #define GetOpenCardName  GetOpenCardNameW
@@ -1249,16 +1250,16 @@ SCardDlgExtendedError (void);
 // Smartcard Caching API
 //
 
-extern WINSCARDAPI LONG WINAPI
-SCardReadCacheA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ extern WINSCARDAPI LONG WINAPI
+  SCardReadCacheA(
     _In_  SCARDCONTEXT hContext,
     _In_  UUID *CardIdentifier,
     _In_  DWORD FreshnessCounter,
     _In_  LPSTR LookupName,
     _Out_writes_bytes_(*DataLen) PBYTE Data,
     _Out_ DWORD *DataLen);
-extern WINSCARDAPI LONG WINAPI
-SCardReadCacheW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ extern WINSCARDAPI LONG WINAPI
+  SCardReadCacheW(
     _In_  SCARDCONTEXT hContext,
     _In_  UUID *CardIdentifier,
     _In_  DWORD FreshnessCounter,
@@ -1271,16 +1272,16 @@ SCardReadCacheW(
 #define SCardReadCache  SCardReadCacheA
 #endif // !UNICODE
 
-extern WINSCARDAPI LONG WINAPI
-SCardWriteCacheA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsVista_ extern WINSCARDAPI LONG WINAPI
+  SCardWriteCacheA(
     _In_ SCARDCONTEXT hContext,
     _In_ UUID *CardIdentifier,
     _In_ DWORD FreshnessCounter,
     _In_ LPSTR LookupName,
     _In_reads_bytes_(DataLen) PBYTE Data,
     _In_ DWORD DataLen);
-extern WINSCARDAPI LONG WINAPI
-SCardWriteCacheW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsVista_ extern WINSCARDAPI LONG WINAPI
+  SCardWriteCacheW(
     _In_ SCARDCONTEXT hContext,
     _In_ UUID *CardIdentifier,
     _In_ DWORD FreshnessCounter,
@@ -1297,18 +1298,18 @@ SCardWriteCacheW(
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI 
-SCardGetReaderIconA(
+  SCardGetReaderIconA(
   _In_     SCARDCONTEXT hContext,
   _In_     LPCSTR szReaderName, 
   _When_(_Old_(*pcbIcon) == SCARD_AUTOALLOCATE, _At_((LPBYTE *)pbIcon, _Outptr_result_bytebuffer_all_maybenull_(*pcbIcon)))
   _When_(_Old_(*pcbIcon) != SCARD_AUTOALLOCATE, _Out_writes_bytes_to_(*pcbIcon, *pcbIcon) _Post_z_)
            LPBYTE pbIcon,
   _Inout_  LPDWORD pcbIcon);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI 
-SCardGetReaderIconW(
+  SCardGetReaderIconW(
   _In_     SCARDCONTEXT hContext,
   _In_     LPCWSTR szReaderName, 
   _When_(_Old_(*pcbIcon) == SCARD_AUTOALLOCATE, _At_((LPBYTE *)pbIcon, _Outptr_result_bytebuffer_all_maybenull_(*pcbIcon)))
@@ -1321,15 +1322,15 @@ SCardGetReaderIconW(
 #define SCardGetReaderIcon  SCardGetReaderIconA
 #endif // !UNICODE
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI 
-SCardGetDeviceTypeIdA(
+  SCardGetDeviceTypeIdA(
   _In_     SCARDCONTEXT hContext,
   _In_     LPCSTR szReaderName, 
   _Inout_  LPDWORD pdwDeviceTypeId);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI 
-SCardGetDeviceTypeIdW(
+  SCardGetDeviceTypeIdW(
   _In_     SCARDCONTEXT hContext,
   _In_     LPCWSTR szReaderName, 
   _Inout_  LPDWORD pdwDeviceTypeId);
@@ -1339,18 +1340,18 @@ SCardGetDeviceTypeIdW(
 #define SCardGetDeviceTypeId  SCardGetDeviceTypeIdA
 #endif // !UNICODE
   
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardGetReaderDeviceInstanceIdA(
+  SCardGetReaderDeviceInstanceIdA(
   _In_    SCARDCONTEXT hContext,
   _In_    LPCSTR szReaderName,
   _When_(_Old_(*pcchDeviceInstanceId) == SCARD_AUTOALLOCATE, _At_((LPSTR *)szDeviceInstanceId, _Outptr_result_buffer_maybenull_(*pcchDeviceInstanceId) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
   _When_(_Old_(*pcchDeviceInstanceId) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchDeviceInstanceId) _Post_ _NullNull_terminated_)
           LPSTR szDeviceInstanceId,
   _Inout_ LPDWORD pcchDeviceInstanceId);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardGetReaderDeviceInstanceIdW(
+  SCardGetReaderDeviceInstanceIdW(
   _In_    SCARDCONTEXT hContext,
   _In_    LPCWSTR szReaderName,
   _When_(_Old_(*pcchDeviceInstanceId) == SCARD_AUTOALLOCATE, _At_((LPWSTR *)szDeviceInstanceId, _Outptr_result_buffer_maybenull_(*pcchDeviceInstanceId) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
@@ -1363,18 +1364,18 @@ SCardGetReaderDeviceInstanceIdW(
 #define SCardGetReaderDeviceInstanceId  SCardGetReaderDeviceInstanceIdA
 #endif // !UNICODE
 
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListReadersWithDeviceInstanceIdA(
+  SCardListReadersWithDeviceInstanceIdA(
   _In_    SCARDCONTEXT hContext,
   _In_    LPCSTR szDeviceInstanceId,
   _When_(_Old_(*pcchReaders) == SCARD_AUTOALLOCATE, _At_((LPSTR *)mszReaders, _Outptr_result_buffer_maybenull_(*pcchReaders) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
   _When_(_Old_(*pcchReaders) != SCARD_AUTOALLOCATE, _Out_writes_opt_(*pcchReaders) _Post_ _NullNull_terminated_)
           LPSTR mszReaders,
   _Inout_ LPDWORD pcchReaders);
-_Success_(return == SCARD_S_SUCCESS)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardListReadersWithDeviceInstanceIdW(
+  SCardListReadersWithDeviceInstanceIdW(
   _In_    SCARDCONTEXT hContext,
   _In_    LPCWSTR szDeviceInstanceId,
   _When_(_Old_(*pcchReaders) == SCARD_AUTOALLOCATE, _At_((LPWSTR *)mszReaders, _Outptr_result_buffer_maybenull_(*pcchReaders) _At_(*_Curr_, _Post_z_ _Post_ _NullNull_terminated_)))
@@ -1399,9 +1400,9 @@ SCardListReadersWithDeviceInstanceIdW(
 #define SCARD_AUDIT_CHV_SUCCESS 0x1 // A smart card holder verification (CHV)
                                     // attempt succeeded.
 
-_Success_(return == SCARD_S_SUCCESS)
+_Windows_SupportedOS_Windows8_ _Success_(return == SCARD_S_SUCCESS)
 extern WINSCARDAPI LONG WINAPI
-SCardAudit(
+  SCardAudit(
   _In_ SCARDCONTEXT hContext,
   _In_ DWORD dwEvent);
 
@@ -1417,5 +1418,70 @@ SCardAudit(
 #ifdef __cplusplus
 }
 #endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SCARD_SCOPE_USER")
+#undef SCARD_SCOPE_USER
+#pragma push_macro("SCARD_SCOPE_SYSTEM")
+#undef SCARD_SCOPE_SYSTEM
+enum SCARD_SCOPE : unsigned int
+{
+    SCARD_SCOPE_USER = 0U,
+    SCARD_SCOPE_SYSTEM = 2U,
+};
+#pragma pop_macro("SCARD_SCOPE_SYSTEM")
+#pragma pop_macro("SCARD_SCOPE_USER")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("SCARD_STATE_UNAWARE")
+#undef SCARD_STATE_UNAWARE
+#pragma push_macro("SCARD_STATE_IGNORE")
+#undef SCARD_STATE_IGNORE
+#pragma push_macro("SCARD_STATE_UNAVAILABLE")
+#undef SCARD_STATE_UNAVAILABLE
+#pragma push_macro("SCARD_STATE_EMPTY")
+#undef SCARD_STATE_EMPTY
+#pragma push_macro("SCARD_STATE_PRESENT")
+#undef SCARD_STATE_PRESENT
+#pragma push_macro("SCARD_STATE_ATRMATCH")
+#undef SCARD_STATE_ATRMATCH
+#pragma push_macro("SCARD_STATE_EXCLUSIVE")
+#undef SCARD_STATE_EXCLUSIVE
+#pragma push_macro("SCARD_STATE_INUSE")
+#undef SCARD_STATE_INUSE
+#pragma push_macro("SCARD_STATE_MUTE")
+#undef SCARD_STATE_MUTE
+#pragma push_macro("SCARD_STATE_CHANGED")
+#undef SCARD_STATE_CHANGED
+#pragma push_macro("SCARD_STATE_UNKNOWN")
+#undef SCARD_STATE_UNKNOWN
+enum SCARD_STATE : unsigned int
+{
+    SCARD_STATE_UNAWARE = 0U,
+    SCARD_STATE_IGNORE = 1U,
+    SCARD_STATE_UNAVAILABLE = 8U,
+    SCARD_STATE_EMPTY = 16U,
+    SCARD_STATE_PRESENT = 32U,
+    SCARD_STATE_ATRMATCH = 64U,
+    SCARD_STATE_EXCLUSIVE = 128U,
+    SCARD_STATE_INUSE = 256U,
+    SCARD_STATE_MUTE = 512U,
+    SCARD_STATE_CHANGED = 2U,
+    SCARD_STATE_UNKNOWN = 4U,
+};
+#pragma pop_macro("SCARD_STATE_UNKNOWN")
+#pragma pop_macro("SCARD_STATE_CHANGED")
+#pragma pop_macro("SCARD_STATE_MUTE")
+#pragma pop_macro("SCARD_STATE_INUSE")
+#pragma pop_macro("SCARD_STATE_EXCLUSIVE")
+#pragma pop_macro("SCARD_STATE_ATRMATCH")
+#pragma pop_macro("SCARD_STATE_PRESENT")
+#pragma pop_macro("SCARD_STATE_EMPTY")
+#pragma pop_macro("SCARD_STATE_UNAVAILABLE")
+#pragma pop_macro("SCARD_STATE_IGNORE")
+#pragma pop_macro("SCARD_STATE_UNAWARE")
+#endif
+
 #endif // _WINSCARD_H_
 
