diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
index a2a8b352f8..f949000ce8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winspool.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 
 /*++
 
@@ -1208,12 +1209,12 @@ typedef LPDATATYPES_INFO_1A LPDATATYPES_INFO_1;
 typedef struct _PRINTER_DEFAULTSA{
     LPSTR         pDatatype;
     LPDEVMODEA pDevMode;
-    ACCESS_MASK DesiredAccess;
+    _Win32_AssociatedEnum_(PRINTER_ACCESS_RIGHTS) ACCESS_MASK DesiredAccess;
 } PRINTER_DEFAULTSA, *PPRINTER_DEFAULTSA, *LPPRINTER_DEFAULTSA;
 typedef struct _PRINTER_DEFAULTSW{
     LPWSTR        pDatatype;
     LPDEVMODEW pDevMode;
-    ACCESS_MASK DesiredAccess;
+    _Win32_AssociatedEnum_(PRINTER_ACCESS_RIGHTS) ACCESS_MASK DesiredAccess;
 } PRINTER_DEFAULTSW, *PPRINTER_DEFAULTSW, *LPPRINTER_DEFAULTSW;
 #ifdef UNICODE
 typedef PRINTER_DEFAULTSW PRINTER_DEFAULTS;
@@ -1249,12 +1250,12 @@ typedef PPRINTER_ENUM_VALUESA PPRINTER_ENUM_VALUES;
 typedef LPPRINTER_ENUM_VALUESA LPPRINTER_ENUM_VALUES;
 #endif // UNICODE
 
-_Success_(return != FALSE)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ _Success_(return != FALSE)
 BOOL
 WINAPI
 EnumPrintersA(
 _In_          DWORD   Flags,
-_In_opt_      LPSTR Name,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_      LPSTR Name,
               DWORD   Level,
 _Out_writes_bytes_to_opt_(cbBuf, *pcbNeeded)
               LPBYTE  pPrinterEnum,
@@ -1262,12 +1263,12 @@ _Out_writes_bytes_to_opt_(cbBuf, *pcbNeeded)
 _Out_         LPDWORD pcbNeeded,
 _Out_         LPDWORD pcReturned
 );
-_Success_(return != FALSE)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") _Success_(return != FALSE)
 BOOL
 WINAPI
 EnumPrintersW(
 _In_          DWORD   Flags,
-_In_opt_      LPWSTR Name,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_      LPWSTR Name,
               DWORD   Level,
 _Out_writes_bytes_to_opt_(cbBuf, *pcbNeeded)
               LPBYTE  pPrinterEnum,
@@ -1313,13 +1314,13 @@ _Out_         LPDWORD pcReturned
 #define SPOOL_FILE_PERSISTENT    0x00000001
 #define SPOOL_FILE_TEMPORARY     0x00000002
 
-HANDLE
+_Win32_SetLastError_ _Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE
 WINAPI
 GetSpoolFileHandle(
 _In_    HANDLE  hPrinter
 );
 
-HANDLE
+_Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE
 WINAPI
 CommitSpoolData(
 _In_    HANDLE  hPrinter,
@@ -1334,18 +1335,18 @@ _In_    HANDLE  hPrinter,
 _In_    HANDLE  hSpoolFile
 );
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 OpenPrinterA(
-_In_opt_    LPSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR             pPrinterName,
+_Win32_RAIIFree_(ClosePrinter, -1, 0) _Out_       LPHANDLE            phPrinter,
 _In_opt_    LPPRINTER_DEFAULTSA pDefault
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 OpenPrinterW(
-_In_opt_    LPWSTR             pPrinterName,
-_Out_       LPHANDLE            phPrinter,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR             pPrinterName,
+_Win32_RAIIFree_(ClosePrinter, -1, 0) _Out_       LPHANDLE            phPrinter,
 _In_opt_    LPPRINTER_DEFAULTSW pDefault
 );
 #ifdef UNICODE
@@ -1354,13 +1355,13 @@ _In_opt_    LPPRINTER_DEFAULTSW pDefault
 #define OpenPrinter  OpenPrinterA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 ResetPrinterA(
 _In_        HANDLE                hPrinter,
 _In_opt_    LPPRINTER_DEFAULTSA   pDefault
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 ResetPrinterW(
 _In_        HANDLE                hPrinter,
@@ -1372,7 +1373,7 @@ _In_opt_    LPPRINTER_DEFAULTSW   pDefault
 #define ResetPrinter  ResetPrinterA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 SetJobA(
 _In_             HANDLE hPrinter,
@@ -1386,7 +1387,7 @@ _When_(Level == 3, _In_reads_bytes_opt_(sizeof(JOB_INFO_3)))
                  LPBYTE pJob,
                  DWORD  Command
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 SetJobW(
 _In_             HANDLE hPrinter,
@@ -1406,7 +1407,7 @@ _When_(Level == 3, _In_reads_bytes_opt_(sizeof(JOB_INFO_3)))
 #define SetJob  SetJobA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetJobA(
 _In_  HANDLE   hPrinter,
@@ -1417,7 +1418,7 @@ _Out_writes_bytes_opt_(cbBuf)
       DWORD    cbBuf,
 _Out_ LPDWORD  pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetJobW(
 _In_  HANDLE   hPrinter,
@@ -1434,7 +1435,7 @@ _Out_ LPDWORD  pcbNeeded
 #define GetJob  GetJobA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 EnumJobsA(
 _In_  HANDLE  hPrinter,
@@ -1447,7 +1448,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_ LPDWORD pcbNeeded,
 _Out_ LPDWORD pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumJobsW(
 _In_  HANDLE  hPrinter,
@@ -1466,20 +1467,20 @@ _Out_ LPDWORD pcReturned
 #define EnumJobs  EnumJobsA
 #endif // !UNICODE
 
-HANDLE
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ _Win32_RAIIFree_(ClosePrinter, -1, 0) HANDLE
 WINAPI
 AddPrinterA(
-_In_opt_    LPSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR    pName,
 _In_range_(1, 2)
             DWORD      Level,
 _When_(Level == 1, _In_reads_bytes_(sizeof(PRINTER_INFO_1)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(PRINTER_INFO_2)))
             LPBYTE     pPrinter
 );
-HANDLE
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") _Win32_RAIIFree_(ClosePrinter, -1, 0) HANDLE
 WINAPI
 AddPrinterW(
-_In_opt_    LPWSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR    pName,
 _In_range_(1, 2)
             DWORD      Level,
 _When_(Level == 1, _In_reads_bytes_(sizeof(PRINTER_INFO_1)))
@@ -1492,13 +1493,13 @@ _When_(Level == 2, _In_reads_bytes_(sizeof(PRINTER_INFO_2)))
 #define AddPrinter  AddPrinterA
 #endif // !UNICODE
 
-BOOL
+_Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrinter(
 _Inout_ HANDLE   hPrinter
 );
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 SetPrinterA(
     _In_             HANDLE  hPrinter,
@@ -1517,7 +1518,7 @@ SetPrinterA(
                      LPBYTE  pPrinter,
     _In_             DWORD   Command
     );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 SetPrinterW(
     _In_             HANDLE  hPrinter,
@@ -1542,7 +1543,7 @@ SetPrinterW(
 #define SetPrinter  SetPrinterA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 GetPrinterA(
 _In_  HANDLE  hPrinter,
@@ -1552,7 +1553,7 @@ _Out_writes_bytes_opt_(cbBuf)
       DWORD   cbBuf,
 _Out_ LPDWORD pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetPrinterW(
 _In_  HANDLE  hPrinter,
@@ -1568,17 +1569,17 @@ _Out_ LPDWORD pcbNeeded
 #define GetPrinter  GetPrinterA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 AddPrinterDriverA(
-_In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
             DWORD     Level,
 _In_        LPBYTE    pDriverInfo
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPrinterDriverW(
-_In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
             DWORD     Level,
 _In_        LPBYTE    pDriverInfo
 );
@@ -1588,10 +1589,10 @@ _In_        LPBYTE    pDriverInfo
 #define AddPrinterDriver  AddPrinterDriverA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddPrinterDriverExA(
-_In_opt_         LPSTR  pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_         LPSTR  pName,
 _In_range_(2, 8) DWORD    Level,
 _When_(Level == 2, _In_reads_bytes_(sizeof(DRIVER_INFO_2)))
 _When_(Level == 3, _In_reads_bytes_(sizeof(DRIVER_INFO_3)))
@@ -1601,10 +1602,10 @@ _When_(Level == 8, _In_reads_bytes_(sizeof(DRIVER_INFO_8)))
                  PBYTE    lpbDriverInfo,
 _In_             DWORD    dwFileCopyFlags
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPrinterDriverExW(
-_In_opt_         LPWSTR  pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_         LPWSTR  pName,
 _In_range_(2, 8) DWORD    Level,
 _When_(Level == 2, _In_reads_bytes_(sizeof(DRIVER_INFO_2)))
 _When_(Level == 3, _In_reads_bytes_(sizeof(DRIVER_INFO_3)))
@@ -1620,11 +1621,11 @@ _In_             DWORD    dwFileCopyFlags
 #define AddPrinterDriverEx  AddPrinterDriverExA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 EnumPrinterDriversA(
-_In_opt_    LPSTR        pName,
-_In_opt_    LPSTR        pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR        pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR        pEnvironment,
             DWORD          Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE        pDriverInfo,
@@ -1632,11 +1633,11 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_       LPDWORD        pcbNeeded,
 _Out_       LPDWORD        pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumPrinterDriversW(
-_In_opt_    LPWSTR        pName,
-_In_opt_    LPWSTR        pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR        pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR        pEnvironment,
             DWORD          Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE        pDriverInfo,
@@ -1650,22 +1651,22 @@ _Out_       LPDWORD        pcReturned
 #define EnumPrinterDrivers  EnumPrinterDriversA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetPrinterDriverA(
 _In_        HANDLE  hPrinter,
-_In_opt_    LPSTR pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR pEnvironment,
             DWORD   Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE  pDriverInfo,
             DWORD   cbBuf,
 _Out_       LPDWORD pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetPrinterDriverW(
 _In_        HANDLE  hPrinter,
-_In_opt_    LPWSTR pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR pEnvironment,
             DWORD   Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE  pDriverInfo,
@@ -1678,22 +1679,22 @@ _Out_       LPDWORD pcbNeeded
 #define GetPrinterDriver  GetPrinterDriverA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetPrinterDriverDirectoryA(
-_In_opt_    LPSTR   pName,
-_In_opt_    LPSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pEnvironment,
             DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE    pDriverDirectory,
             DWORD     cbBuf,
 _Out_       LPDWORD   pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetPrinterDriverDirectoryW(
-_In_opt_    LPWSTR   pName,
-_In_opt_    LPWSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pEnvironment,
             DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE    pDriverDirectory,
@@ -1706,19 +1707,19 @@ _Out_       LPDWORD   pcbNeeded
 #define GetPrinterDriverDirectory  GetPrinterDriverDirectoryA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeletePrinterDriverA(
-_In_opt_    LPSTR    pName,
-_In_opt_    LPSTR    pEnvironment,
-_In_        LPSTR    pDriverName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR    pDriverName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrinterDriverW(
-_In_opt_    LPWSTR    pName,
-_In_opt_    LPWSTR    pEnvironment,
-_In_        LPWSTR    pDriverName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR    pDriverName
 );
 #ifdef UNICODE
 #define DeletePrinterDriver  DeletePrinterDriverW
@@ -1726,21 +1727,21 @@ _In_        LPWSTR    pDriverName
 #define DeletePrinterDriver  DeletePrinterDriverA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeletePrinterDriverExA(
-_In_opt_    LPSTR    pName,
-_In_opt_    LPSTR    pEnvironment,
-_In_        LPSTR    pDriverName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR    pDriverName,
             DWORD      dwDeleteFlag,
             DWORD      dwVersionFlag
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrinterDriverExW(
-_In_opt_    LPWSTR    pName,
-_In_opt_    LPWSTR    pEnvironment,
-_In_        LPWSTR    pDriverName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR    pDriverName,
             DWORD      dwDeleteFlag,
             DWORD      dwVersionFlag
 );
@@ -1752,21 +1753,21 @@ _In_        LPWSTR    pDriverName,
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddPrintProcessorA(
-_In_opt_   LPSTR   pName,
-_In_opt_   LPSTR   pEnvironment,
-_In_       LPSTR   pPathName,
-_In_       LPSTR   pPrintProcessorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_       LPSTR   pPathName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_       LPSTR   pPrintProcessorName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPrintProcessorW(
-_In_opt_   LPWSTR   pName,
-_In_opt_   LPWSTR   pEnvironment,
-_In_       LPWSTR   pPathName,
-_In_       LPWSTR   pPrintProcessorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_       LPWSTR   pPathName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_       LPWSTR   pPrintProcessorName
 );
 #ifdef UNICODE
 #define AddPrintProcessor  AddPrintProcessorW
@@ -1774,11 +1775,11 @@ _In_       LPWSTR   pPrintProcessorName
 #define AddPrintProcessor  AddPrintProcessorA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 EnumPrintProcessorsA(
-_In_opt_   LPSTR   pName,
-_In_opt_   LPSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR   pEnvironment,
            DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
            LPBYTE    pPrintProcessorInfo,
@@ -1786,11 +1787,11 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_      LPDWORD   pcbNeeded,
 _Out_      LPDWORD   pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumPrintProcessorsW(
-_In_opt_   LPWSTR   pName,
-_In_opt_   LPWSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR   pEnvironment,
            DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
            LPBYTE    pPrintProcessorInfo,
@@ -1806,22 +1807,22 @@ _Out_      LPDWORD   pcReturned
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetPrintProcessorDirectoryA(
-_In_opt_   LPSTR    pName,
-_In_opt_   LPSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPSTR    pEnvironment,
            DWORD      Level,
 _Out_writes_bytes_opt_(cbBuf)
            LPBYTE     pPrintProcessorInfo,
            DWORD      cbBuf,
 _Out_      LPDWORD    pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetPrintProcessorDirectoryW(
-_In_opt_   LPWSTR    pName,
-_In_opt_   LPWSTR    pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR    pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_   LPWSTR    pEnvironment,
            DWORD      Level,
 _Out_writes_bytes_opt_(cbBuf)
            LPBYTE     pPrintProcessorInfo,
@@ -1834,12 +1835,12 @@ _Out_      LPDWORD    pcbNeeded
 #define GetPrintProcessorDirectory  GetPrintProcessorDirectoryA
 #endif // !UNICODE
 
-_Success_(return != FALSE)
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Success_(return != FALSE)
 BOOL
 WINAPI
 EnumPrintProcessorDatatypesA(
-_In_opt_    LPSTR   pName,
-_In_        LPSTR   pPrintProcessorName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR   pPrintProcessorName,
             DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE    pDatatypes,
@@ -1847,12 +1848,12 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_       LPDWORD   pcbNeeded,
 _Out_       LPDWORD   pcReturned
 );
-_Success_(return != FALSE)
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") _Success_(return != FALSE)
 BOOL
 WINAPI
 EnumPrintProcessorDatatypesW(
-_In_opt_    LPWSTR   pName,
-_In_        LPWSTR   pPrintProcessorName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR   pPrintProcessorName,
             DWORD     Level,
 _Out_writes_bytes_opt_(cbBuf)
             LPBYTE    pDatatypes,
@@ -1866,19 +1867,19 @@ _Out_       LPDWORD   pcReturned
 #define EnumPrintProcessorDatatypes  EnumPrintProcessorDatatypesA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeletePrintProcessorA(
-_In_opt_    LPSTR   pName,
-_In_opt_    LPSTR   pEnvironment,
-_In_        LPSTR   pPrintProcessorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR   pPrintProcessorName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrintProcessorW(
-_In_opt_    LPWSTR   pName,
-_In_opt_    LPWSTR   pEnvironment,
-_In_        LPWSTR   pPrintProcessorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR   pPrintProcessorName
 );
 #ifdef UNICODE
 #define DeletePrintProcessor  DeletePrintProcessorW
@@ -1886,7 +1887,7 @@ _In_        LPWSTR   pPrintProcessorName
 #define DeletePrintProcessor  DeletePrintProcessorA
 #endif // !UNICODE
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 StartDocPrinterA(
 _In_             HANDLE  hPrinter,
@@ -1895,7 +1896,7 @@ _When_(Level == 1, _In_reads_bytes_(sizeof(DOC_INFO_1)))
 _When_(Level == 3, _In_reads_bytes_(sizeof(DOC_INFO_3)))
                  LPBYTE  pDocInfo
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 StartDocPrinterW(
 _In_             HANDLE  hPrinter,
@@ -1910,13 +1911,13 @@ _When_(Level == 3, _In_reads_bytes_(sizeof(DOC_INFO_3)))
 #define StartDocPrinter  StartDocPrinterA
 #endif // !UNICODE
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 StartPagePrinter(
 _In_    HANDLE  hPrinter
 );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 WritePrinter(
 _In_    HANDLE  hPrinter,
@@ -1927,7 +1928,7 @@ _Out_   LPDWORD pcWritten
 );
 
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 FlushPrinter(
 _In_    HANDLE  hPrinter,
@@ -1938,19 +1939,19 @@ _Out_   LPDWORD pcWritten,
         DWORD   cSleep
 );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EndPagePrinter(
 _In_    HANDLE   hPrinter
 );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AbortPrinter(
 _In_    HANDLE   hPrinter
 );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 ReadPrinter(
 _In_    HANDLE  hPrinter,
@@ -1960,13 +1961,13 @@ _Out_writes_bytes_(cbBuf)
 _Out_   LPDWORD pNoBytesRead
 );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EndDocPrinter(
     _In_  HANDLE   hPrinter
     );
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddJobA(
 _In_    HANDLE  hPrinter,
@@ -1976,7 +1977,7 @@ _Out_writes_bytes_opt_(cbBuf)
         DWORD   cbBuf,
 _Out_   LPDWORD pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddJobW(
 _In_    HANDLE  hPrinter,
@@ -1992,7 +1993,7 @@ _Out_   LPDWORD pcbNeeded
 #define AddJob  AddJobA
 #endif // !UNICODE
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 ScheduleJob(
 _In_    HANDLE  hPrinter,
@@ -2006,7 +2007,7 @@ _In_    HWND    hWnd,
 _In_    HANDLE  hPrinter
 );
 
-_When_(fMode == 0, _Post_satisfies_(return >= sizeof(DEVMODE)))
+_WIN32META_ANNOTATION_("win32metadata:ansi") _When_(fMode == 0, _Post_satisfies_(return >= sizeof(DEVMODE)))
 LONG
 WINAPI
 DocumentPropertiesA(
@@ -2017,7 +2018,7 @@ _Out_opt_       PDEVMODEA pDevModeOutput,
 _In_opt_        PDEVMODEA pDevModeInput,
                 DWORD     fMode
 );
-_When_(fMode == 0, _Post_satisfies_(return >= sizeof(DEVMODE)))
+_WIN32META_ANNOTATION_("win32metadata:unicode") _When_(fMode == 0, _Post_satisfies_(return >= sizeof(DEVMODE)))
 LONG
 WINAPI
 DocumentPropertiesW(
@@ -2034,7 +2035,7 @@ _In_opt_        PDEVMODEW pDevModeInput,
 #define DocumentProperties  DocumentPropertiesA
 #endif // !UNICODE
 
-LONG
+_WIN32META_ANNOTATION_("win32metadata:ansi") LONG
 WINAPI
 AdvancedDocumentPropertiesA(
 _In_            HWND        hWnd,
@@ -2043,7 +2044,7 @@ _In_            LPSTR     pDeviceName,
 _Inout_opt_     PDEVMODEA   pDevModeOutput,
 _In_opt_        PDEVMODEA   pDevModeInput
 );
-LONG
+_WIN32META_ANNOTATION_("win32metadata:unicode") LONG
 WINAPI
 AdvancedDocumentPropertiesW(
 _In_            HWND        hWnd,
@@ -2073,22 +2074,22 @@ _In_opt_        PDEVMODEW   pDevModeInput
 #endif // (NTDDI_VERSION >= NTDDI_WINXPSP1)
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 GetPrinterDataA(
 _In_         HANDLE   hPrinter,
-_In_         LPSTR  pValueName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_         LPSTR  pValueName,
 _Out_opt_    LPDWORD  pType,
 _Out_writes_bytes_opt_(nSize)
              LPBYTE   pData,
              DWORD    nSize,
 _Out_        LPDWORD  pcbNeeded
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 GetPrinterDataW(
 _In_         HANDLE   hPrinter,
-_In_         LPWSTR  pValueName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_         LPWSTR  pValueName,
 _Out_opt_    LPDWORD  pType,
 _Out_writes_bytes_opt_(nSize)
              LPBYTE   pData,
@@ -2101,7 +2102,7 @@ _Out_        LPDWORD  pcbNeeded
 #define GetPrinterData  GetPrinterDataA
 #endif // !UNICODE
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 GetPrinterDataExA(
 _In_         HANDLE   hPrinter,
@@ -2113,7 +2114,7 @@ _Out_writes_bytes_opt_(nSize)
              DWORD    nSize,
 _Out_        LPDWORD  pcbNeeded
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 GetPrinterDataExW(
 _In_         HANDLE   hPrinter,
@@ -2131,7 +2132,7 @@ _Out_        LPDWORD  pcbNeeded
 #define GetPrinterDataEx  GetPrinterDataExA
 #endif // !UNICODE
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 EnumPrinterDataA(
 _In_        HANDLE   hPrinter,
@@ -2147,7 +2148,7 @@ _Out_writes_to_opt_(cbData, *pcbData)
 _When_(pData != NULL, _Pre_notnull_)
 _Out_opt_   LPDWORD  pcbData
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 EnumPrinterDataW(
 _In_        HANDLE   hPrinter,
@@ -2169,7 +2170,7 @@ _Out_opt_   LPDWORD  pcbData
 #define EnumPrinterData  EnumPrinterDataA
 #endif // !UNICODE
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 EnumPrinterDataExA(
 _In_    HANDLE    hPrinter,
@@ -2180,7 +2181,7 @@ _Out_writes_bytes_opt_(cbEnumValues)
 _Out_   LPDWORD   pcbEnumValues,
 _Out_   LPDWORD   pnEnumValues
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 EnumPrinterDataExW(
 _In_    HANDLE    hPrinter,
@@ -2197,7 +2198,7 @@ _Out_   LPDWORD   pnEnumValues
 #define EnumPrinterDataEx  EnumPrinterDataExA
 #endif // !UNICODE
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 EnumPrinterKeyA(
 _In_    HANDLE   hPrinter,
@@ -2207,7 +2208,7 @@ _Out_writes_bytes_opt_(cbSubkey)
         DWORD    cbSubkey,
 _Out_   LPDWORD  pcbSubkey
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 EnumPrinterKeyW(
 _In_    HANDLE   hPrinter,
@@ -2224,21 +2225,21 @@ _Out_   LPDWORD  pcbSubkey
 #endif // !UNICODE
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 SetPrinterDataA(
 _In_    HANDLE  hPrinter,
-_In_    LPSTR pValueName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPSTR pValueName,
         DWORD   Type,
 _In_reads_bytes_(cbData)
         LPBYTE  pData,
         DWORD   cbData
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 SetPrinterDataW(
 _In_    HANDLE  hPrinter,
-_In_    LPWSTR pValueName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPWSTR pValueName,
         DWORD   Type,
 _In_reads_bytes_(cbData)
         LPBYTE  pData,
@@ -2251,7 +2252,7 @@ _In_reads_bytes_(cbData)
 #endif // !UNICODE
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 SetPrinterDataExA(
 _In_    HANDLE   hPrinter,
@@ -2262,7 +2263,7 @@ _In_reads_bytes_(cbData)
         LPBYTE   pData,
         DWORD    cbData
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 SetPrinterDataExW(
 _In_    HANDLE   hPrinter,
@@ -2281,17 +2282,17 @@ _In_reads_bytes_(cbData)
 
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 DeletePrinterDataA(
 _In_    HANDLE  hPrinter,
-_In_    LPSTR pValueName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPSTR pValueName
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 DeletePrinterDataW(
 _In_    HANDLE  hPrinter,
-_In_    LPWSTR pValueName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPWSTR pValueName
 );
 #ifdef UNICODE
 #define DeletePrinterData  DeletePrinterDataW
@@ -2300,14 +2301,14 @@ _In_    LPWSTR pValueName
 #endif // !UNICODE
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 DeletePrinterDataExA(
 _In_    HANDLE   hPrinter,
 _In_    LPCSTR pKeyName,
 _In_    LPCSTR pValueName
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 DeletePrinterDataExW(
 _In_    HANDLE   hPrinter,
@@ -2321,13 +2322,13 @@ _In_    LPCWSTR pValueName
 #endif // !UNICODE
 
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 DeletePrinterKeyA(
 _In_    HANDLE   hPrinter,
 _In_    LPCSTR pKeyName
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 DeletePrinterKeyW(
 _In_    HANDLE   hPrinter,
@@ -2555,14 +2556,14 @@ typedef struct _PRINTER_NOTIFY_INFO {
     #define ERROR_BIDI_UNSUPPORTED_RESOURCE_FORMAT  (ERROR_BIDI_ERROR_BASE + 18)
 #endif // (NTDDI_VERSION >= NTDDI_WINXP)
 
-DWORD
+_Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 WaitForPrinterChange(
 _In_ HANDLE  hPrinter,
      DWORD   Flags
 );
 
-HANDLE
+_Win32_RAIIFree_(FindClosePrinterChangeNotification, -1, 0) HANDLE
 WINAPI
 FindFirstPrinterChangeNotification(
 _In_     HANDLE hPrinter,
@@ -2587,7 +2588,7 @@ FreePrinterNotifyInfo(
 _In_   PPRINTER_NOTIFY_INFO pPrinterNotifyInfo
     );
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 FindClosePrinterChangeNotification(
 _In_   HANDLE hChange
@@ -2629,7 +2630,7 @@ _In_   HANDLE hChange
 #define PRINTER_CHANGE_ALL                      0x7777FFFF
 #endif
 
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:ansi") DWORD
 WINAPI
 PrinterMessageBoxA(
 _In_   HANDLE    hPrinter,
@@ -2639,7 +2640,7 @@ _In_   LPSTR   pText,
 _In_   LPSTR   pCaption,
        DWORD     dwType
 );
-DWORD
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 PrinterMessageBoxW(
 _In_   HANDLE    hPrinter,
@@ -2665,13 +2666,13 @@ _In_   LPWSTR   pCaption,
 #define PRINTER_ERROR_JAM           0x00000002
 #define PRINTER_ERROR_OUTOFTONER    0x00000004
 
-BOOL
+_Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 ClosePrinter(
 _In_ HANDLE hPrinter
 );
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddFormA(
 _In_             HANDLE hPrinter,
@@ -2680,7 +2681,7 @@ _When_(Level == 1, _In_reads_bytes_(sizeof(FORM_INFO_1A)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(FORM_INFO_2A)))
                  LPBYTE pForm
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddFormW(
 _In_             HANDLE hPrinter,
@@ -2695,17 +2696,17 @@ _When_(Level == 2, _In_reads_bytes_(sizeof(FORM_INFO_2W)))
 #define AddForm  AddFormA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeleteFormA(
 _In_    HANDLE    hPrinter,
-_In_    LPSTR   pFormName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPSTR   pFormName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeleteFormW(
 _In_    HANDLE    hPrinter,
-_In_    LPWSTR   pFormName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPWSTR   pFormName
 );
 #ifdef UNICODE
 #define DeleteForm  DeleteFormW
@@ -2715,22 +2716,22 @@ _In_    LPWSTR   pFormName
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetFormA(
 _In_    HANDLE        hPrinter,
-_In_    LPSTR       pFormName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPSTR       pFormName,
         DWORD         Level,
 _Out_writes_bytes_opt_(cbBuf)
         LPBYTE        pForm,
         DWORD         cbBuf,
 _Out_   LPDWORD       pcbNeeded
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 GetFormW(
 _In_    HANDLE        hPrinter,
-_In_    LPWSTR       pFormName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_    LPWSTR       pFormName,
         DWORD         Level,
 _Out_writes_bytes_opt_(cbBuf)
         LPBYTE        pForm,
@@ -2745,21 +2746,21 @@ _Out_   LPDWORD       pcbNeeded
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 SetFormA(
 _In_             HANDLE  hPrinter,
-_In_             LPSTR pFormName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_             LPSTR pFormName,
 _In_range_(1, 2) DWORD   Level,
 _When_(Level == 1, _In_reads_bytes_(sizeof(FORM_INFO_1A)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(FORM_INFO_2A)))
                  LPBYTE  pForm
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 SetFormW(
 _In_             HANDLE  hPrinter,
-_In_             LPWSTR pFormName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_             LPWSTR pFormName,
 _In_range_(1, 2) DWORD   Level,
 _When_(Level == 1, _In_reads_bytes_(sizeof(FORM_INFO_1W)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(FORM_INFO_2W)))
@@ -2771,7 +2772,7 @@ _When_(Level == 2, _In_reads_bytes_(sizeof(FORM_INFO_2W)))
 #define SetForm  SetFormA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 EnumFormsA(
 _In_    HANDLE  hPrinter,
@@ -2782,7 +2783,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_   LPDWORD pcbNeeded,
 _Out_   LPDWORD pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumFormsW(
 _In_    HANDLE  hPrinter,
@@ -2800,7 +2801,7 @@ _Out_   LPDWORD pcReturned
 #endif // !UNICODE
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 EnumMonitorsA(
 _In_opt_    LPSTR   pName,
@@ -2811,7 +2812,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_       LPDWORD    pcbNeeded,
 _Out_       LPDWORD    pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumMonitorsW(
 _In_opt_    LPWSTR   pName,
@@ -2830,18 +2831,18 @@ _Out_       LPDWORD    pcReturned
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 AddMonitorA(
-_In_opt_    LPSTR pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR pName,
             DWORD   Level,
 _In_reads_bytes_opt_(sizeof(MONITOR_INFO_2))
             LPBYTE  pMonitors
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddMonitorW(
-_In_opt_    LPWSTR pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR pName,
             DWORD   Level,
 _In_reads_bytes_opt_(sizeof(MONITOR_INFO_2))
             LPBYTE  pMonitors
@@ -2854,19 +2855,19 @@ _In_reads_bytes_opt_(sizeof(MONITOR_INFO_2))
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 DeleteMonitorA(
-_In_opt_    LPSTR     pName,
-_In_opt_    LPSTR     pEnvironment,
-_In_        LPSTR     pMonitorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR     pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR     pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR     pMonitorName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeleteMonitorW(
-_In_opt_    LPWSTR     pName,
-_In_opt_    LPWSTR     pEnvironment,
-_In_        LPWSTR     pMonitorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR     pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR     pEnvironment,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR     pMonitorName
 );
 #ifdef UNICODE
 #define DeleteMonitor  DeleteMonitorW
@@ -2876,7 +2877,7 @@ _In_        LPWSTR     pMonitorName
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 EnumPortsA(
 _In_opt_    LPSTR pName,
@@ -2887,7 +2888,7 @@ _Out_writes_bytes_opt_(cbBuf)
 _Out_       LPDWORD pcbNeeded,
 _Out_       LPDWORD pcReturned
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 EnumPortsW(
 _In_opt_    LPWSTR pName,
@@ -2905,19 +2906,19 @@ _Out_       LPDWORD pcReturned
 #endif // !UNICODE
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 AddPortA(
-_In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
 _In_        HWND      hWnd,
-_In_        LPSTR   pMonitorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR   pMonitorName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPortW(
-_In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
 _In_        HWND      hWnd,
-_In_        LPWSTR   pMonitorName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR   pMonitorName
 );
 #ifdef UNICODE
 #define AddPort  AddPortW
@@ -2927,19 +2928,19 @@ _In_        LPWSTR   pMonitorName
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 ConfigurePortA(
-_In_opt_    LPSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR   pName,
 _In_        HWND      hWnd,
-_In_        LPSTR   pPortName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR   pPortName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 ConfigurePortW(
-_In_opt_    LPWSTR   pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR   pName,
 _In_        HWND      hWnd,
-_In_        LPWSTR   pPortName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR   pPortName
 );
 #ifdef UNICODE
 #define ConfigurePort  ConfigurePortW
@@ -2947,19 +2948,19 @@ _In_        LPWSTR   pPortName
 #define ConfigurePort  ConfigurePortA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 DeletePortA(
-_In_opt_    LPSTR pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPSTR pName,
 _In_        HWND    hWnd,
-_In_        LPSTR pPortName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPSTR pPortName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePortW(
-_In_opt_    LPWSTR pName,
+_WIN32META_ANNOTATION_("win32metadata:const") _In_opt_    LPWSTR pName,
 _In_        HWND    hWnd,
-_In_        LPWSTR pPortName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_        LPWSTR pPortName
 );
 #ifdef UNICODE
 #define DeletePort  DeletePortW
@@ -2967,7 +2968,7 @@ _In_        LPWSTR pPortName
 #define DeletePort  DeletePortA
 #endif // !UNICODE
 
-BOOL
+_Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 XcvDataW(
 _In_        HANDLE  hXcv,
@@ -2983,13 +2984,13 @@ _Out_opt_   PDWORD  pdwStatus
 );
 #define XcvData  XcvDataW
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 GetDefaultPrinterA(
     _Out_writes_opt_(*pcchBuffer)   LPSTR   pszBuffer,
     _Inout_                         LPDWORD   pcchBuffer
     );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") BOOL
 WINAPI
 GetDefaultPrinterW(
     _Out_writes_opt_(*pcchBuffer)   LPWSTR   pszBuffer,
@@ -3001,12 +3002,12 @@ GetDefaultPrinterW(
 #define GetDefaultPrinter  GetDefaultPrinterA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 SetDefaultPrinterA(
 _In_opt_    LPCSTR pszPrinter
     );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") BOOL
 WINAPI
 SetDefaultPrinterW(
 _In_opt_    LPCWSTR pszPrinter
@@ -3018,7 +3019,7 @@ _In_opt_    LPCWSTR pszPrinter
 #endif // !UNICODE
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
 WINAPI
 SetPortA(
 _In_opt_    LPSTR     pName,
@@ -3027,7 +3028,7 @@ _In_        LPSTR     pPortName,
 _In_reads_bytes_(sizeof(PORT_INFO_3))
             LPBYTE      pPortInfo
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 SetPortW(
 _In_opt_    LPWSTR     pName,
@@ -3044,15 +3045,15 @@ _In_reads_bytes_(sizeof(PORT_INFO_3))
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddPrinterConnectionA(
-_In_ LPSTR   pName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_ LPSTR   pName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPrinterConnectionW(
-_In_ LPWSTR   pName
+_WIN32META_ANNOTATION_("win32metadata:const") _In_ LPWSTR   pName
 );
 #ifdef UNICODE
 #define AddPrinterConnection  AddPrinterConnectionW
@@ -3062,12 +3063,12 @@ _In_ LPWSTR   pName
 
 
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeletePrinterConnectionA(
 _In_ LPSTR   pName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrinterConnectionW(
 _In_ LPWSTR   pName
@@ -3080,7 +3081,7 @@ _In_ LPWSTR   pName
 
 
 
-HANDLE
+_Win32_RAIIFree_(CloseHandle, -1, 0) HANDLE
 WINAPI
 ConnectToPrinterDlg(
 _In_    HWND    hwnd,
@@ -3123,7 +3124,7 @@ typedef PPROVIDOR_INFO_2A PPROVIDOR_INFO_2;
 typedef LPPROVIDOR_INFO_2A LPPROVIDOR_INFO_2;
 #endif // UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 AddPrintProvidorA(
 _In_opt_         LPSTR  pName,
@@ -3132,7 +3133,7 @@ _When_(Level == 1, _In_reads_bytes_(sizeof(PROVIDOR_INFO_1)))
 _When_(Level == 2, _In_reads_bytes_(sizeof(PROVIDOR_INFO_2)))
 _In_             LPBYTE   pProvidorInfo
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 AddPrintProvidorW(
 _In_opt_         LPWSTR  pName,
@@ -3147,14 +3148,14 @@ _In_             LPBYTE   pProvidorInfo
 #define AddPrintProvidor  AddPrintProvidorA
 #endif // !UNICODE
 
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
 WINAPI
 DeletePrintProvidorA(
 _In_opt_    LPSTR   pName,
 _In_opt_    LPSTR   pEnvironment,
 _In_        LPSTR   pPrintProvidorName
 );
-BOOL
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_ImportLibrary_("winspool.drv") BOOL
 WINAPI
 DeletePrintProvidorW(
 _In_opt_    LPWSTR   pName,
@@ -3168,13 +3169,13 @@ _In_        LPWSTR   pPrintProvidorName
 #endif // !UNICODE
 
 #if (NTDDI_VERSION >= NTDDI_WINXPSP2)
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
     WINAPI
     IsValidDevmodeA(
         _In_opt_    PDEVMODEA   pDevmode,
                     size_t      DevmodeSize
         );
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:unicode") BOOL
     WINAPI
     IsValidDevmodeW(
         _In_opt_    PDEVMODEW   pDevmode,
@@ -3421,15 +3422,15 @@ _In_        LPWSTR   pPrintProvidorName
     } PRINTER_OPTION_FLAGS;
 
 
-    typedef struct _PRINTER_OPTIONSA
+    typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _PRINTER_OPTIONSA
     {
         UINT            cbSize;
-        DWORD           dwFlags;
+        _Win32_AssociatedEnum_(PRINTER_OPTION_FLAGS) DWORD           dwFlags;
     } PRINTER_OPTIONSA, *PPRINTER_OPTIONSA, *LPPRINTER_OPTIONSA;
-    typedef struct _PRINTER_OPTIONSW
+    typedef struct _WIN32META_ANNOTATION_("win32metadata:struct_size_field=cbSize")  _PRINTER_OPTIONSW
     {
         UINT            cbSize;
-        DWORD           dwFlags;
+        _Win32_AssociatedEnum_(PRINTER_OPTION_FLAGS) DWORD           dwFlags;
     } PRINTER_OPTIONSW, *PPRINTER_OPTIONSW, *LPPRINTER_OPTIONSW;
 #ifdef UNICODE
 typedef PRINTER_OPTIONSW PRINTER_OPTIONS;
@@ -3441,19 +3442,19 @@ typedef PPRINTER_OPTIONSA PPRINTER_OPTIONS;
 typedef LPPRINTER_OPTIONSA LPPRINTER_OPTIONS;
 #endif // UNICODE
 
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:ansi") _Win32_SetLastError_ BOOL
     WINAPI
     OpenPrinter2A(
     _In_opt_      LPCSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _Win32_RAIIFree_(ClosePrinter, -1, 0) _Out_         LPHANDLE                phPrinter,
     _In_opt_      PPRINTER_DEFAULTSA      pDefault,
     _In_opt_      PPRINTER_OPTIONSA       pOptions
     );
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:unicode") _Win32_SetLastError_ _Win32_ImportLibrary_("winspool.drv") BOOL
     WINAPI
     OpenPrinter2W(
     _In_opt_      LPCWSTR                pPrinterName,
-    _Out_         LPHANDLE                phPrinter,
+    _Win32_RAIIFree_(ClosePrinter, -1, 0) _Out_         LPHANDLE                phPrinter,
     _In_opt_      PPRINTER_DEFAULTSW      pDefault,
     _In_opt_      PPRINTER_OPTIONSW       pOptions
     );
@@ -3484,7 +3485,7 @@ typedef PRINTER_CONNECTION_INFO_1A PRINTER_CONNECTION_INFO_1;
 typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
 #endif // UNICODE
 
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
     WINAPI
     AddPrinterConnection2A(
     _In_opt_    HWND            hWnd,
@@ -3492,7 +3493,7 @@ typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
                 DWORD           dwLevel,
     _In_        PVOID           pConnectionInfo
         );
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:unicode") BOOL
     WINAPI
     AddPrinterConnection2W(
     _In_opt_    HWND            hWnd,
@@ -3513,7 +3514,7 @@ typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
 //
 #define IPDFP_COPY_ALL_FILES                0x00000001
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     InstallPrinterDriverFromPackageA(
     _In_opt_   LPCSTR    pszServer,
@@ -3522,7 +3523,7 @@ typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
     _In_opt_   LPCSTR    pszEnvironment,
     _In_       DWORD       dwFlags
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     InstallPrinterDriverFromPackageW(
     _In_opt_   LPCWSTR    pszServer,
@@ -3542,7 +3543,7 @@ typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
                                             // driver package is already present on remote machine.
 #define UPDP_CHECK_DRIVERSTORE  0x00000004  // Check whether the given package exists in driver store or not.
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     UploadPrinterDriverPackageA(
     _In_opt_   LPCSTR    pszServer,
@@ -3554,7 +3555,7 @@ typedef PPRINTER_CONNECTION_INFO_1A PPRINTER_CONNECTION_INFO_1;
                LPSTR     pszDestInfPath,
     _Inout_    PULONG      pcchDestInfPath
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     UploadPrinterDriverPackageW(
     _In_opt_   LPCWSTR    pszServer,
@@ -3594,7 +3595,7 @@ typedef CORE_PRINTER_DRIVERA CORE_PRINTER_DRIVER;
 typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
 #endif // UNICODE
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     GetCorePrinterDriversA(
     _In_opt_                          LPCSTR               pszServer,
@@ -3603,7 +3604,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
     _In_                              DWORD                  cCorePrinterDrivers,
     _Out_writes_(cCorePrinterDrivers) PCORE_PRINTER_DRIVERA  pCorePrinterDrivers
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     GetCorePrinterDriversW(
     _In_opt_                          LPCWSTR               pszServer,
@@ -3618,7 +3619,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
 #define GetCorePrinterDrivers  GetCorePrinterDriversA
 #endif // !UNICODE
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     CorePrinterDriverInstalledA(
     _In_opt_            LPCSTR     pszServer,
@@ -3628,7 +3629,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
     _In_                DWORDLONG    dwlDriverVersion,
     _Out_               BOOL         *pbDriverInstalled
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     CorePrinterDriverInstalledW(
     _In_opt_            LPCWSTR     pszServer,
@@ -3644,7 +3645,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
 #define CorePrinterDriverInstalled  CorePrinterDriverInstalledA
 #endif // !UNICODE
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     GetPrinterDriverPackagePathA(
     _In_opt_                                   LPCSTR      pszServer,
@@ -3655,7 +3656,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
     _In_opt_                                   DWORD         cchDriverPackageCab,
     _Out_                                      LPDWORD       pcchRequiredSize
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     GetPrinterDriverPackagePathW(
     _In_opt_                                   LPCWSTR      pszServer,
@@ -3672,14 +3673,14 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
 #define GetPrinterDriverPackagePath  GetPrinterDriverPackagePathA
 #endif // !UNICODE
 
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:ansi") HRESULT
     WINAPI
     DeletePrinterDriverPackageA(
     _In_opt_   LPCSTR    pszServer,
     _In_       LPCSTR    pszInfPath,
     _In_opt_   LPCSTR    pszEnvironment
     );
-    HRESULT
+    _WIN32META_ANNOTATION_("win32metadata:unicode") HRESULT
     WINAPI
     DeletePrinterDriverPackageW(
     _In_opt_   LPCWSTR    pszServer,
@@ -3760,7 +3761,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
 
     }PrintPropertiesCollection;
 
-    HRESULT
+    _Win32_ImportLibrary_("winspool.drv") HRESULT
     WINAPI
     ReportJobProcessingProgress(
     _In_        HANDLE                         printerHandle,
@@ -3769,7 +3770,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
                 EPrintXPSJobProgress           jobProgress
         );
 
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:ansi") BOOL
     WINAPI
     GetPrinterDriver2A(
         _In_opt_    HWND    hWnd,
@@ -3781,7 +3782,7 @@ typedef PCORE_PRINTER_DRIVERA PCORE_PRINTER_DRIVER;
                     DWORD   cbBuf,
         _Out_       LPDWORD pcbNeeded
     );
-    BOOL
+    _WIN32META_ANNOTATION_("win32metadata:unicode") BOOL
     WINAPI
     GetPrinterDriver2W(
         _In_opt_    HWND    hWnd,
@@ -3832,7 +3833,7 @@ GetPrintExecutionData(
 
 #if (NTDDI_VERSION >= NTDDI_WIN8)
 
-DWORD
+_Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 GetJobNamedPropertyValue(
     _In_  HANDLE             hPrinter,
@@ -3841,7 +3842,7 @@ GetJobNamedPropertyValue(
     _Out_ PrintPropertyValue *pValue
     );
 
-void
+_Win32_ImportLibrary_("winspool.drv") void
 WINAPI
 FreePrintPropertyValue(
     _Inout_ PrintPropertyValue *pValue
@@ -3856,7 +3857,7 @@ FreePrintNamedPropertyArray(
                                       PrintNamedProperty **ppProperties
     );
 
-DWORD
+_Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 SetJobNamedProperty(
     _In_  HANDLE                   hPrinter,
@@ -3864,7 +3865,7 @@ SetJobNamedProperty(
     _In_  const PrintNamedProperty *pProperty
     );
 
-DWORD
+_Win32_ImportLibrary_("winspool.drv") DWORD
 WINAPI
 DeleteJobNamedProperty(
     _In_  HANDLE             hPrinter,
@@ -3887,7 +3888,7 @@ WINAPI
 GetPrintOutputInfo(
     _In_        HWND   hWnd,
     _In_        PCWSTR pszPrinter,
-    _Outptr_ HANDLE *phFile,
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Outptr_ HANDLE *phFile,
     _Outptr_ PWSTR  *ppszOutputFile
     );
 
@@ -3914,5 +3915,100 @@ GetPrintOutputInfo(
 #endif
 
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("PRINTER_ALL_ACCESS")
+#undef PRINTER_ALL_ACCESS
+#pragma push_macro("PRINTER_READ")
+#undef PRINTER_READ
+#pragma push_macro("PRINTER_WRITE")
+#undef PRINTER_WRITE
+#pragma push_macro("PRINTER_EXECUTE")
+#undef PRINTER_EXECUTE
+#pragma push_macro("SERVER_ALL_ACCESS")
+#undef SERVER_ALL_ACCESS
+#pragma push_macro("SERVER_READ")
+#undef SERVER_READ
+#pragma push_macro("SERVER_WRITE")
+#undef SERVER_WRITE
+#pragma push_macro("SERVER_EXECUTE")
+#undef SERVER_EXECUTE
+#pragma push_macro("PRINTER_DELETE")
+#undef PRINTER_DELETE
+#pragma push_macro("PRINTER_READ_CONTROL")
+#undef PRINTER_READ_CONTROL
+#pragma push_macro("PRINTER_WRITE_DAC")
+#undef PRINTER_WRITE_DAC
+#pragma push_macro("PRINTER_WRITE_OWNER")
+#undef PRINTER_WRITE_OWNER
+#pragma push_macro("PRINTER_SYNCHRONIZE")
+#undef PRINTER_SYNCHRONIZE
+#pragma push_macro("PRINTER_STANDARD_RIGHTS_REQUIRED")
+#undef PRINTER_STANDARD_RIGHTS_REQUIRED
+#pragma push_macro("PRINTER_STANDARD_RIGHTS_READ")
+#undef PRINTER_STANDARD_RIGHTS_READ
+#pragma push_macro("PRINTER_STANDARD_RIGHTS_WRITE")
+#undef PRINTER_STANDARD_RIGHTS_WRITE
+#pragma push_macro("PRINTER_STANDARD_RIGHTS_EXECUTE")
+#undef PRINTER_STANDARD_RIGHTS_EXECUTE
+#pragma push_macro("SERVER_ACCESS_ADMINISTER")
+#undef SERVER_ACCESS_ADMINISTER
+#pragma push_macro("SERVER_ACCESS_ENUMERATE")
+#undef SERVER_ACCESS_ENUMERATE
+#pragma push_macro("PRINTER_ACCESS_ADMINISTER")
+#undef PRINTER_ACCESS_ADMINISTER
+#pragma push_macro("PRINTER_ACCESS_USE")
+#undef PRINTER_ACCESS_USE
+#pragma push_macro("PRINTER_ACCESS_MANAGE_LIMITED")
+#undef PRINTER_ACCESS_MANAGE_LIMITED
+enum [[clang::flag_enum]] PRINTER_ACCESS_RIGHTS : unsigned int
+{
+    PRINTER_ALL_ACCESS = 983052U,
+    PRINTER_READ = 131080U,
+    PRINTER_WRITE = 131080U,
+    PRINTER_EXECUTE = 131080U,
+    SERVER_ALL_ACCESS = 983043U,
+    SERVER_READ = 131074U,
+    SERVER_WRITE = 131075U,
+    SERVER_EXECUTE = 131074U,
+    PRINTER_DELETE = 65536U,
+    PRINTER_READ_CONTROL = 131072U,
+    PRINTER_WRITE_DAC = 262144U,
+    PRINTER_WRITE_OWNER = 524288U,
+    PRINTER_SYNCHRONIZE = 1048576U,
+    PRINTER_STANDARD_RIGHTS_REQUIRED = 983040U,
+    PRINTER_STANDARD_RIGHTS_READ = 131072U,
+    PRINTER_STANDARD_RIGHTS_WRITE = 131072U,
+    PRINTER_STANDARD_RIGHTS_EXECUTE = 131072U,
+    SERVER_ACCESS_ADMINISTER = 1U,
+    SERVER_ACCESS_ENUMERATE = 2U,
+    PRINTER_ACCESS_ADMINISTER = 4U,
+    PRINTER_ACCESS_USE = 8U,
+    PRINTER_ACCESS_MANAGE_LIMITED = 64U,
+};
+#pragma pop_macro("PRINTER_ACCESS_MANAGE_LIMITED")
+#pragma pop_macro("PRINTER_ACCESS_USE")
+#pragma pop_macro("PRINTER_ACCESS_ADMINISTER")
+#pragma pop_macro("SERVER_ACCESS_ENUMERATE")
+#pragma pop_macro("SERVER_ACCESS_ADMINISTER")
+#pragma pop_macro("PRINTER_STANDARD_RIGHTS_EXECUTE")
+#pragma pop_macro("PRINTER_STANDARD_RIGHTS_WRITE")
+#pragma pop_macro("PRINTER_STANDARD_RIGHTS_READ")
+#pragma pop_macro("PRINTER_STANDARD_RIGHTS_REQUIRED")
+#pragma pop_macro("PRINTER_SYNCHRONIZE")
+#pragma pop_macro("PRINTER_WRITE_OWNER")
+#pragma pop_macro("PRINTER_WRITE_DAC")
+#pragma pop_macro("PRINTER_READ_CONTROL")
+#pragma pop_macro("PRINTER_DELETE")
+#pragma pop_macro("SERVER_EXECUTE")
+#pragma pop_macro("SERVER_WRITE")
+#pragma pop_macro("SERVER_READ")
+#pragma pop_macro("SERVER_ALL_ACCESS")
+#pragma pop_macro("PRINTER_EXECUTE")
+#pragma pop_macro("PRINTER_WRITE")
+#pragma pop_macro("PRINTER_READ")
+#pragma pop_macro("PRINTER_ALL_ACCESS")
+#endif
+
 #endif // _WINSPOOL_
 
