diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/winsock.h b/generation/WinSDK/RecompiledIdlHeaders/um/winsock.h
index f580e7af..e5f2d6cc 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/winsock.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/winsock.h
@@ -1,4 +1,7 @@
 #include <winapifamily.h>
+#if defined(WIN32METADATA)
+#include <win32metadata_annotations.h>
+#endif
 
 
 /* WINSOCK.H--definitions to be used with the WINSOCK.DLL
@@ -736,38 +739,46 @@ struct  linger {
 extern "C" {
 #endif
 
+_Win32_SetLastError_
 SOCKET PASCAL FAR accept (
                           _In_ SOCKET s,
                           _Out_writes_bytes_opt_(*addrlen) struct sockaddr FAR *addr,
                           _Inout_opt_ int FAR *addrlen);
 
+_Win32_SetLastError_
 int PASCAL FAR bind (
                      _In_ SOCKET s,
                      _In_reads_bytes_(namelen) const struct sockaddr FAR *addr,
                      _In_ int namelen);
 
+_Win32_SetLastError_
 int PASCAL FAR closesocket ( IN SOCKET s);
 
+_Win32_SetLastError_
 int PASCAL FAR connect (
                         _In_ SOCKET s,
                         _In_reads_bytes_(namelen) const struct sockaddr FAR *name,
                         _In_ int namelen);
 
+_Win32_SetLastError_
 int PASCAL FAR ioctlsocket (
                             _In_ SOCKET s,
                             _In_ long cmd,
                             _Inout_ u_long FAR *argp);
 
+_Win32_SetLastError_
 int PASCAL FAR getpeername (
                             _In_ SOCKET s,
                             _Out_writes_bytes_to_(*namelen, *namelen) struct sockaddr FAR *name,
                             _Inout_ int FAR * namelen);
 
+_Win32_SetLastError_
 int PASCAL FAR getsockname (
                             _In_ SOCKET s,
                             _Out_writes_bytes_to_(*namelen, *namelen) struct sockaddr FAR *name,
                             _Inout_ int FAR * namelen);
 
+_Win32_SetLastError_
 int PASCAL FAR getsockopt (
                            _In_ SOCKET s,
                            _In_ int level,
@@ -775,28 +786,37 @@ int PASCAL FAR getsockopt (
                            _Out_writes_bytes_(*optlen) char FAR * optval,
                            _Inout_ int FAR *optlen);
 
+_Win32_SetLastError_
 u_long PASCAL FAR htonl ( _In_ u_long hostlong);
 
+_Win32_SetLastError_
 u_short PASCAL FAR htons (_In_ u_short hostshort);
 
+_Win32_SetLastError_
 unsigned long PASCAL FAR inet_addr (_In_z_ const char FAR * cp);
 
+_Win32_SetLastError_
 char FAR * PASCAL FAR inet_ntoa (_In_ struct in_addr in);
 
+_Win32_SetLastError_
 int PASCAL FAR listen (
                        _In_ SOCKET s,
                        _In_ int backlog);
 
+_Win32_SetLastError_
 u_long PASCAL FAR ntohl (_In_ u_long netlong);
 
+_Win32_SetLastError_
 u_short PASCAL FAR ntohs (_In_ u_short netshort);
 
+_Win32_SetLastError_
 int PASCAL FAR recv (
                      _In_ SOCKET s,
                      _Out_writes_bytes_to_(len, return) __out_data_source(NETWORK) char FAR * buf,
                      _In_ int len,
                      _In_ int flags);
 
+_Win32_SetLastError_
 int PASCAL FAR recvfrom (
                          _In_ SOCKET s,
                          _Out_writes_bytes_to_(len, return) __out_data_source(NETWORK) char FAR * buf,
@@ -805,6 +825,7 @@ int PASCAL FAR recvfrom (
                          _Out_writes_bytes_to_opt_(*fromlen, *fromlen) struct sockaddr FAR * from,
                          _Inout_opt_ int FAR * fromlen);
 
+_Win32_SetLastError_
 int PASCAL FAR select (
                         _In_ int nfds,
                         _Inout_opt_ fd_set FAR *readfds,
@@ -812,12 +833,14 @@ int PASCAL FAR select (
                         _Inout_opt_ fd_set FAR *exceptfds,
                         _In_opt_  const struct timeval FAR *timeout);
 
+_Win32_SetLastError_
 int PASCAL FAR send (
                      _In_ SOCKET s,
                      _In_reads_bytes_(len) const char FAR * buf,
                      _In_ int len,
                      _In_ int flags);
 
+_Win32_SetLastError_
 int PASCAL FAR sendto (
                        _In_ SOCKET s,
                        _In_reads_bytes_(len) const char FAR * buf,
@@ -826,6 +849,7 @@ int PASCAL FAR sendto (
                        _In_reads_bytes_opt_(tolen) const struct sockaddr FAR *to,
                        _In_ int tolen);
 
+_Win32_SetLastError_
 int PASCAL FAR setsockopt (
                            _In_ SOCKET s,
                            _In_ int level,
@@ -833,10 +857,12 @@ int PASCAL FAR setsockopt (
                            _In_reads_bytes_opt_(optlen) const char FAR * optval,
                            _In_ int optlen);
 
+_Win32_SetLastError_
 int PASCAL FAR shutdown (
                          _In_ SOCKET s,
                          _In_ int how);
 
+_Win32_SetLastError_
 SOCKET PASCAL FAR socket (
                           _In_ int af,
                           _In_ int type,
@@ -844,49 +870,65 @@ SOCKET PASCAL FAR socket (
 
 /* Database function prototypes */
 
+_Win32_SetLastError_
 struct hostent FAR * PASCAL FAR gethostbyaddr(
                                               _In_reads_bytes_(len) const char FAR * addr,
                                               _In_ int len,
                                               _In_ int type);
 
+_Win32_SetLastError_
 struct hostent FAR * PASCAL FAR gethostbyname(_In_z_ const char FAR * name);
 
+_Win32_SetLastError_
 int PASCAL FAR gethostname (
                             _Out_writes_bytes_to_(namelen, return) char FAR * name,
                             _In_ int namelen);
 
+_Win32_SetLastError_
 struct servent FAR * PASCAL FAR getservbyport(
                                               _In_ int port,
                                               _In_z_ const char FAR * proto);
 
+_Win32_SetLastError_
 struct servent FAR * PASCAL FAR getservbyname(
                                               _In_z_ const char FAR * name,
                                               _In_z_ const char FAR * proto);
 
+_Win32_SetLastError_
 struct protoent FAR * PASCAL FAR getprotobynumber(_In_ int proto);
 
+_Win32_SetLastError_
 struct protoent FAR * PASCAL FAR getprotobyname(_In_z_ const char FAR * name);
 
 /* Microsoft Windows Extension function prototypes */
 
+_Win32_SetLastError_
 int PASCAL FAR WSAStartup(
                           _In_ WORD wVersionRequired,
                           _Out_ LPWSADATA lpWSAData);
 
+_Win32_SetLastError_
 int PASCAL FAR WSACleanup(void);
 
+_Win32_SetLastError_
 void PASCAL FAR WSASetLastError(_In_ int iError);
 
+_Win32_SetLastError_
 int PASCAL FAR WSAGetLastError(void);
 
+_Win32_SetLastError_
 BOOL PASCAL FAR WSAIsBlocking(void);
 
+_Win32_SetLastError_
 int PASCAL FAR WSAUnhookBlockingHook(void);
 
+_Win32_SetLastError_
 FARPROC PASCAL FAR WSASetBlockingHook(_In_ FARPROC lpBlockFunc);
 
+_Win32_SetLastError_
 int PASCAL FAR WSACancelBlockingCall(void);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetServByName(
                                         _In_ HWND hWnd,
                                         _In_ u_int wMsg,
@@ -895,6 +937,7 @@ HANDLE PASCAL FAR WSAAsyncGetServByName(
                                         _Out_writes_bytes_(buflen) char FAR * buf,
                                         _In_ int buflen);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetServByPort(
                                         _In_ HWND hWnd,
                                         _In_ u_int wMsg,
@@ -903,6 +946,7 @@ HANDLE PASCAL FAR WSAAsyncGetServByPort(
                                         _Out_writes_bytes_(buflen) char FAR * buf,
                                         _In_ int buflen);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetProtoByName(
                                          _In_ HWND hWnd,
                                          _In_ u_int wMsg,
@@ -910,6 +954,7 @@ HANDLE PASCAL FAR WSAAsyncGetProtoByName(
                                          _Out_writes_bytes_(buflen) char FAR * buf,
                                          _In_ int buflen);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetProtoByNumber(
                                            _In_ HWND hWnd,
                                            _In_ u_int wMsg,
@@ -917,6 +962,7 @@ HANDLE PASCAL FAR WSAAsyncGetProtoByNumber(
                                            _Out_writes_bytes_(buflen) char FAR * buf,
                                            _In_ int buflen);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetHostByName(
                                         _In_ HWND hWnd,
                                         _In_ u_int wMsg,
@@ -924,6 +970,7 @@ HANDLE PASCAL FAR WSAAsyncGetHostByName(
                                         _Out_writes_bytes_(buflen) char FAR * buf,
                                         _In_ int buflen);
 
+_Win32_SetLastError_
 HANDLE PASCAL FAR WSAAsyncGetHostByAddr(
                                         _In_ HWND hWnd,
                                         _In_ u_int wMsg,
@@ -933,14 +980,17 @@ HANDLE PASCAL FAR WSAAsyncGetHostByAddr(
                                         _Out_writes_bytes_(buflen) char FAR * buf,
                                         _In_ int buflen);
 
+_Win32_SetLastError_
 int PASCAL FAR WSACancelAsyncRequest(_In_ HANDLE hAsyncTaskHandle);
 
+_Win32_SetLastError_
 int PASCAL FAR WSAAsyncSelect(
                               _In_ SOCKET s,
                               _In_ HWND hWnd,
                               _In_ u_int wMsg,
                               _In_ long lEvent);
 
+_Win32_SetLastError_
 int PASCAL FAR WSARecvEx (
                           _In_ SOCKET s,
                           _Out_writes_bytes_to_(len, return) __out_data_source(NETWORK) char FAR * buf,
