diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wlanihv.h b/generation/WinSDK/RecompiledIdlHeaders/um/wlanihv.h
index 54e97a0aee..9a2cac831b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wlanihv.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wlanihv.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) 2005 Microsoft Corporation
@@ -510,7 +511,7 @@ DWORD
 (WINAPI *DOT11EXT_REQUEST_VIRTUAL_STATION)
 (
     _In_opt_    HANDLE  hDot11PrimaryHandle,
-    _Reserved_  LPVOID  pvReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_  LPVOID  pvReserved
 );
 
 
@@ -519,7 +520,7 @@ DWORD
 (WINAPI *DOT11EXT_RELEASE_VIRTUAL_STATION)
 (
     _In_opt_    HANDLE  hDot11PrimaryHandle,
-    _Reserved_  LPVOID  pvReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_  LPVOID  pvReserved
 );
 
 
@@ -531,7 +532,7 @@ DWORD
     _In_opt_    HANDLE  hDot11SvcHandle,
     _Out_       BOOL*   pbIsVirtualStation,
     _Out_       GUID*   pgPrimary,
-    _Reserved_  LPVOID  pvReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_  LPVOID  pvReserved
 );
 
 
@@ -544,7 +545,7 @@ DWORD
     _In_opt_                        HANDLE                                  hConnectSession,
                                     DWORD                                   dwNumProperties,
     _In_reads_(dwNumProperties)    PDOT11EXT_VIRTUAL_STATION_AP_PROPERTY   pProperties,
-    _Reserved_                      LPVOID                                  pvReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_                      LPVOID                                  pvReserved
 );
 
 
@@ -572,7 +573,7 @@ DWORD
 (
                 DWORD                      dwVerNumUsed,
     _In_        PDOT11EXT_APIS             pDot11ExtAPI,
-    _Reserved_  LPVOID                     pvReserved,
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_  LPVOID                     pvReserved,
     _Out_       PDOT11EXT_IHV_HANDLERS     pDot11IHVHandlers
 );
 
@@ -584,7 +585,7 @@ DWORD
 (WINAPI *DOT11EXTIHV_INIT_VIRTUAL_STATION)
 (
     _In_        PDOT11EXT_VIRTUAL_STATION_APIS      pDot11ExtVSAPI,
-    _Reserved_  LPVOID                              pvReserved
+    _WIN32META_ANNOTATION_("win32metadata:optional") _Reserved_  LPVOID                              pvReserved
 );
 
 
@@ -606,7 +607,7 @@ DWORD
 (
     _In_        PDOT11_ADAPTER    pDot11Adapter,
     _In_opt_    HANDLE            hDot11SvcHandle,
-    _Out_       PHANDLE           phIhvExtAdapter
+    _Win32_RAIIFree_(CloseHandle, -1, 0) _Out_       PHANDLE           phIhvExtAdapter
 );
 
 
