diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h b/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
index 5ccb585fc241fb952390841e9726a4254a1f43d8..566937b9c5d487a188f3e33e2b839867bc2073de 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/WMIUtils.h
@@ -113,6 +113,7 @@ extern "C"{
 /*                                                                             */
 /*******************************************************************************/
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family or WinMgmt Package
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_PKG_WINMGMT)
 
@@ -193,7 +194,7 @@ EXTERN_C const IID IID_IWbemPathKeyList;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("9AE62877-7544-4bb0-AA26-A13824659ED6")
-    IWbemPathKeyList : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IWbemPathKeyList : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE GetCount( 
@@ -408,7 +409,7 @@ EXTERN_C const IID IID_IWbemPathKeyList;
         END_INTERFACE
     } IWbemPathKeyListVtbl;
 
-    interface IWbemPathKeyList
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IWbemPathKeyList
     {
         CONST_VTBL struct IWbemPathKeyListVtbl *lpVtbl;
     };
@@ -481,7 +482,7 @@ EXTERN_C const IID IID_IWbemPath;
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
     MIDL_INTERFACE("3BC15AF2-736C-477e-9E51-238AF8667DCC")
-    IWbemPath : public IUnknown
+    _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IWbemPath : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE SetText( 
@@ -844,7 +845,7 @@ EXTERN_C const IID IID_IWbemPath;
         END_INTERFACE
     } IWbemPathVtbl;
 
-    interface IWbemPath
+    interface _Windows_SupportedOS_WindowsVista_ _Windows_SupportedOS_Server2008_ IWbemPath
     {
         CONST_VTBL struct IWbemPathVtbl *lpVtbl;
     };
