diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wmpplug.h b/generation/WinSDK/RecompiledIdlHeaders/um/wmpplug.h
index 769df188fd..75cfb2f8a8 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wmpplug.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wmpplug.h
@@ -15,6 +15,7 @@
 #endif
 
 #include "rpc.h"
+#include <win32metadata_annotations.h>
 #include "rpcndr.h"
 
 #ifndef __RPCNDR_H_VERSION__
@@ -157,6 +158,7 @@ EXTERN_C const IID IID_IWMPPluginUI;
             /* [in] */ const WCHAR *pwszName,
             /* [in] */ const VARIANT *pvarProperty) = 0;
         
+        _Win32_PreserveResult_
         virtual HRESULT STDMETHODCALLTYPE TranslateAccelerator( 
             /* [in] */ LPMSG lpmsg) = 0;
         
