diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h b/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
index 292b851272..6eb819d608 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wnvapi.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 /*++
 
 Copyright (c) Microsoft Corporation. All rights reserved.
@@ -27,14 +28,14 @@ Abstract:
 #define WNV_API_MAJOR_VERSION_1 1
 #define WNV_API_MINOR_VERSION_0 0
 
-typedef enum _WNV_NOTIFICATION_TYPE {
+typedef enum _Windows_SupportedOS_Server2012_ _WNV_NOTIFICATION_TYPE {
     WnvPolicyMismatchType,
     WnvRedirectType,
     WnvObjectChangeType,
     WnvNotificationTypeMax
 } WNV_NOTIFICATION_TYPE, *PWNV_NOTIFICATION_TYPE;
 
-typedef enum _WNV_OBJECT_TYPE {
+typedef enum _Windows_SupportedOS_Server2012_ _WNV_OBJECT_TYPE {
     WnvProviderAddressType,
     WnvCustomerAddressType,
     WnvObjectTypeMax
@@ -47,20 +48,20 @@ typedef enum _WNV_CA_NOTIFICATION_TYPE {
     WnvCustomerAddressMax
 } WNV_CA_NOTIFICATION_TYPE, *PWNV_CA_NOTIFICATION_TYPE;
 
-typedef struct _WNV_OBJECT_HEADER {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_OBJECT_HEADER {
     UCHAR MajorVersion;
     UCHAR MinorVersion;
     ULONG Size;
 } WNV_OBJECT_HEADER, *PWNV_OBJECT_HEADER;
 
-typedef struct _WNV_NOTIFICATION_PARAM {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_NOTIFICATION_PARAM {
     WNV_OBJECT_HEADER Header;
     WNV_NOTIFICATION_TYPE NotificationType;
     ULONG PendingNotifications;
     PUCHAR Buffer;
 } WNV_NOTIFICATION_PARAM, *PWNV_NOTIFICATION_PARAM;
 
-typedef struct _WNV_IP_ADDRESS {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_IP_ADDRESS {
     union {
         IN_ADDR v4;
         IN6_ADDR v6;
@@ -68,7 +69,7 @@ typedef struct _WNV_IP_ADDRESS {
     } IP;
 } WNV_IP_ADDRESS, *PWNV_IP_ADDRESS;
 
-typedef struct _WNV_POLICY_MISMATCH_PARAM {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_POLICY_MISMATCH_PARAM {
     ADDRESS_FAMILY CAFamily;
     ADDRESS_FAMILY PAFamily;
     ULONG VirtualSubnetId;
@@ -76,7 +77,7 @@ typedef struct _WNV_POLICY_MISMATCH_PARAM {
     WNV_IP_ADDRESS PA;
 } WNV_POLICY_MISMATCH_PARAM, *PWNV_POLICY_MISMATCH_PARAM;
 
-typedef struct _WNV_PROVIDER_ADDRESS_CHANGE_PARAM {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_PROVIDER_ADDRESS_CHANGE_PARAM {
     ADDRESS_FAMILY PAFamily;
     WNV_IP_ADDRESS PA;
     NL_DAD_STATE AddressState;
@@ -92,7 +93,7 @@ typedef struct _WNV_CUSTOMER_ADDRESS_CHANGE_PARAM {
     WNV_CA_NOTIFICATION_TYPE NotificationReason;
 } WNV_CUSTOMER_ADDRESS_CHANGE_PARAM, *PWNV_CUSTOMER_ADDRESS_CHANGE_PARAM;
 
-typedef struct _WNV_OBJECT_CHANGE_PARAM {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_OBJECT_CHANGE_PARAM {
     WNV_OBJECT_TYPE ObjectType;
     union {
         WNV_PROVIDER_ADDRESS_CHANGE_PARAM ProviderAddressChange;
@@ -100,7 +101,7 @@ typedef struct _WNV_OBJECT_CHANGE_PARAM {
     } ObjectParam;
 } WNV_OBJECT_CHANGE_PARAM, *PWNV_OBJECT_CHANGE_PARAM;
 
-typedef struct _WNV_REDIRECT_PARAM {
+typedef struct _Windows_SupportedOS_Server2012_ _WNV_REDIRECT_PARAM {
     ADDRESS_FAMILY CAFamily;
     ADDRESS_FAMILY PAFamily;
     ADDRESS_FAMILY NewPAFamily;
@@ -116,15 +117,17 @@ extern "C" {
 #endif  // __cplusplus
 HANDLE
 WINAPI
+_Windows_SupportedOS_Server2012_ _Win32_RAIIFree_(CloseHandle, 0)
 WnvOpen(
     );
 
 ULONG
 WINAPI
+_Windows_SupportedOS_Server2012_
 WnvRequestNotification (
     HANDLE WnvHandle,
     PWNV_NOTIFICATION_PARAM NotificationParam,
-    LPOVERLAPPED Overlapped,
+    _Win32_Retained_ LPOVERLAPPED Overlapped,
     PULONG BytesTransferred
     );
 
