diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/wow64apiset.h b/generation/WinSDK/RecompiledIdlHeaders/um/wow64apiset.h
index 55b1a9c0b6..835a36105e 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/wow64apiset.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/wow64apiset.h
@@ -17,6 +17,7 @@
 #include <apisetcconv.h>
 #include <minwindef.h>
 #include <minwinbase.h>
+#include <win32metadata_annotations.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -29,6 +30,7 @@ extern "C" {
 #if !defined(RC_INVOKED)
 #if _WIN32_WINNT >= 0x0501 || defined(WINBASE_DECLARE_GET_SYSTEM_WOW64_DIRECTORY)
 
+_Windows_SupportedOS_WindowsVista_
 WINBASEAPI
 BOOLEAN
 WINAPI
@@ -36,6 +38,7 @@ Wow64EnableWow64FsRedirection(
     _In_ BOOLEAN Wow64FsEnableRedirection
     );
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -43,6 +46,7 @@ Wow64DisableWow64FsRedirection(
     _Out_ PVOID* OldValue
     );
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -61,6 +65,7 @@ Wow64RevertWow64FsRedirection(
 
 #if (_WIN32_WINNT >= 0x0501)
 
+_Windows_SupportedOS_WindowsVista_ _Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -81,6 +86,8 @@ IsWow64Process(
 #if !defined(RC_INVOKED)
 #if _WIN32_WINNT >= 0x0501 || defined(WINBASE_DECLARE_GET_SYSTEM_WOW64_DIRECTORY)
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -91,6 +98,8 @@ GetSystemWow64DirectoryA(
     _In_ UINT uSize
     );
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_WindowsXP_
+_Win32_SetLastError_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -129,13 +138,15 @@ Wow64SetThreadDefaultGuestMachine(
 
 #if (_WIN32_WINNT >= 0x0A00)
 
+_Windows_SupportedOS_10586_ 
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
 IsWow64Process2(
     _In_ HANDLE hProcess,
-    _Out_ USHORT* pProcessMachine,
-    _Out_opt_ USHORT* pNativeMachine
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) _Out_ USHORT* pProcessMachine,
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) _Out_opt_ USHORT* pNativeMachine
     );
 
 #endif // _WIN32_WINNT >= 0x0A00
@@ -148,6 +159,8 @@ IsWow64Process2(
 
 #if (_WIN32_WINNT >= 0x0A00)
 
+_WIN32META_ANNOTATION_("win32metadata:ansi") _Windows_SupportedOS_10586_
+_Win32_SetLastError_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -156,9 +169,11 @@ WINAPI
 GetSystemWow64Directory2A(
     _Out_writes_to_opt_(uSize,return + 1) LPSTR lpBuffer,
     _In_ UINT uSize,
-    _In_ WORD ImageFileMachineType
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) _In_ WORD ImageFileMachineType
     );
 
+_WIN32META_ANNOTATION_("win32metadata:unicode") _Windows_SupportedOS_10586_
+_Win32_SetLastError_
 WINBASEAPI
 _Must_inspect_result_
 _Success_(return != 0 && return < uSize)
@@ -167,7 +182,7 @@ WINAPI
 GetSystemWow64Directory2W(
     _Out_writes_to_opt_(uSize,return + 1) LPWSTR lpBuffer,
     _In_ UINT uSize,
-    _In_ WORD ImageFileMachineType
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) _In_ WORD ImageFileMachineType
     );
 
 #ifdef UNICODE
@@ -180,12 +195,14 @@ GetSystemWow64Directory2W(
 
 #if (_WIN32_WINNT >= 0x0A00)
 
+_Windows_SupportedOS_16299_
+_Win32_SetLastError_
 WINBASEAPI
 _Must_inspect_result_
 HRESULT
 WINAPI
 IsWow64GuestMachineSupported(
-    _In_ USHORT WowGuestMachine,
+    _Win32_AssociatedEnum_(IMAGE_FILE_MACHINE) _In_ USHORT WowGuestMachine,
     _Out_ BOOL* MachineIsSupported
     );
 
@@ -193,6 +210,8 @@ IsWow64GuestMachineSupported(
 
 #if (_WIN32_WINNT >= 0x0600)
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -201,6 +220,8 @@ Wow64GetThreadContext(
     _Inout_ PWOW64_CONTEXT lpContext
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 BOOL
 WINAPI
@@ -209,6 +230,8 @@ Wow64SetThreadContext(
     _In_ CONST WOW64_CONTEXT* lpContext
     );
 
+_Windows_SupportedOS_WindowsVista_
+_Win32_SetLastError_
 WINBASEAPI
 DWORD
 WINAPI
