diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdutil.h b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdutil.h
index b305efdfb9..867966a7e5 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdutil.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdutil.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //+-------------------------------------------------------------------------
 //
 //  Microsoft Windows
@@ -43,13 +44,15 @@ extern "C" {
 #define WSDAPI_OPTION_TRACE_XML_TO_FILE 0x0003
 #endif
     
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDSetConfigurationOption(
     DWORD dwOption,
     _In_reads_bytes_(cbInBuffer) LPVOID pVoid,
     DWORD cbInBuffer);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDGetConfigurationOption(
     DWORD dwOption,
     _Out_writes_bytes_(cbOutBuffer) LPVOID pVoid,
@@ -59,21 +62,25 @@ WSDGetConfigurationOption(
 // Linked memory
 //
 
-void * WINAPI 
+_Windows_SupportedOS_WindowsVista_ void * WINAPI 
+ 
 WSDAllocateLinkedMemory(
     void* pParent, 
     size_t cbSize);
 
-void WINAPI
+_Windows_SupportedOS_WindowsVista_ void WINAPI
+ 
 WSDFreeLinkedMemory(
     void *pVoid);
 
-void WINAPI
+_Windows_SupportedOS_WindowsVista_ void WINAPI
+ 
 WSDAttachLinkedMemory(
     void* pParent,
-    void* pChild);
+     void* pChild);
 
-void WINAPI
+_Windows_SupportedOS_WindowsVista_ void WINAPI
+ 
 WSDDetachLinkedMemory(
     void* pVoid);
 
@@ -83,63 +90,72 @@ WSDDetachLinkedMemory(
 
 typedef struct _WSDXML_ELEMENT WSDXML_ELEMENT; 
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDXMLBuildAnyForSingleElement(
     WSDXML_NAME* pElementName, 
     _In_opt_ LPCWSTR pszText, 
-    _Outptr_ WSDXML_ELEMENT** ppAny);
+    _Outptr_  WSDXML_ELEMENT** ppAny);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDXMLGetValueFromAny(
     _In_ LPCWSTR pszNamespace,
     _In_ LPCWSTR pszName, 
     WSDXML_ELEMENT* pAny, 
     _Outptr_ LPCWSTR* ppszValue);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDXMLAddSibling(
     WSDXML_ELEMENT* pFirst,
-    WSDXML_ELEMENT* pSecond);
+     WSDXML_ELEMENT* pSecond);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDXMLAddChild(
     WSDXML_ELEMENT* pParent,
-    WSDXML_ELEMENT* pChild);
+     WSDXML_ELEMENT* pChild);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDXMLCleanupElement(
     WSDXML_ELEMENT* pAny);    
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDGenerateFault(
     _In_ LPCWSTR pszCode,
     _In_opt_ LPCWSTR pszSubCode,
     _In_ LPCWSTR pszReason,
     _In_opt_ LPCWSTR pszDetail,
     _In_ IWSDXMLContext* pContext,
-    _Outptr_ WSD_SOAP_FAULT** ppFault);
+    _Outptr_  WSD_SOAP_FAULT** ppFault);
 
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
+ 
 WSDGenerateFaultEx(
     _In_ WSDXML_NAME* pCode, 
     _In_opt_ WSDXML_NAME* pSubCode, 
     _In_ WSD_LOCALIZED_STRING_LIST* pReasons, 
     _In_opt_ LPCWSTR pszDetail, 
-    _Outptr_ WSD_SOAP_FAULT** ppFault);
+    _Outptr_  WSD_SOAP_FAULT** ppFault);
 
 #if (WINVER >= _WIN32_WINNT_WIN7)
 HRESULT WINAPI
+_Windows_SupportedOS_Windows7_
 WSDUriEncode(
     _In_reads_(cchSource) LPCWSTR source,
     _In_ DWORD cchSource,
-    _Out_writes_(*cchDestOut) LPWSTR *destOut,
+    _Out_writes_(*cchDestOut)  LPWSTR *destOut,
     _Out_opt_ DWORD *cchDestOut);
 
 HRESULT WINAPI
+_Windows_SupportedOS_Windows7_
 WSDUriDecode(
     _In_reads_(cchSource) LPCWSTR source,
     _In_ DWORD cchSource,
-    _Out_writes_(*cchDestOut) LPWSTR *destOut,
+    _Out_writes_(*cchDestOut)  LPWSTR *destOut,
     _Out_opt_ DWORD *cchDestOut);
 #endif
 
