diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdxml.h b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdxml.h
index b3e27f7c97..e60859e8d3 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/Wsdxml.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/Wsdxml.h
@@ -67,6 +67,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -92,7 +93,7 @@ typedef const WSDXML_NAMESPACE *PCWSDXML_NAMESPACE;
 
 typedef const WSDXML_TYPE *PCWSDXML_TYPE;
 
-typedef struct _WSD_DATETIME
+typedef struct   _WSD_DATETIME
     {
     BOOL isPositive;
     ULONG year;
@@ -108,7 +109,7 @@ typedef struct _WSD_DATETIME
     UCHAR TZMinute;
     } 	WSD_DATETIME;
 
-typedef struct _WSD_DURATION
+typedef struct   _WSD_DURATION
     {
     BOOL isPositive;
     ULONG year;
@@ -210,12 +211,12 @@ enum __MIDL___MIDL_itf_wsdxml_0000_0000_0001
 #define WSDXML_TYPE_ENCODING(typeIndex,layerNumber) ((((DWORD)layerNumber) << 28) | typeIndex)
 #define WSDXML_NAMESPACE_ENCODING(namespaceIndex,layerNumber) ((((WORD)layerNumber) << 12) | namespaceIndex)
 #define WSDXML_NAME_ENCODING(nameIndex,nameSpaceEncoding) ((((DWORD)nameSpaceEncoding) << 16) | nameIndex)
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
 WSDXMLGetNameFromBuiltinNamespace(
     _In_ LPCWSTR pszNamespace,
     _In_ LPCWSTR pszName,
     _Outptr_ WSDXML_NAME** ppName);
-HRESULT WINAPI
+_Windows_SupportedOS_WindowsVista_ HRESULT WINAPI
 WSDXMLCreateContext(
     _Outptr_ IWSDXMLContext** ppContext);
 
@@ -234,8 +235,8 @@ EXTERN_C const IID IID_IWSDXMLContext;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("75d8f3ee-3e5a-43b4-a15a-bcf6887460c0")
-    IWSDXMLContext : public IUnknown
+    MIDL_INTERFACE("75d8f3ee-3e5a-43b4-a15a-bcf6887460c0") _Windows_SupportedOS_WindowsVista_ 
+      IWSDXMLContext : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE AddNamespace( 
@@ -338,7 +339,7 @@ EXTERN_C const IID IID_IWSDXMLContext;
         END_INTERFACE
     } IWSDXMLContextVtbl;
 
-    interface IWSDXMLContext
+    interface _Windows_SupportedOS_WindowsVista_ IWSDXMLContext
     {
         CONST_VTBL struct IWSDXMLContextVtbl *lpVtbl;
     };
@@ -399,6 +400,8 @@ extern RPC_IF_HANDLE __MIDL_itf_wsdxml_0000_0001_v0_0_s_ifspec;
 }
 #endif
 
+
+
 #endif
 
 
