diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/xenroll.h b/generation/WinSDK/RecompiledIdlHeaders/um/xenroll.h
index eda8c1dbb8..098349827b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/xenroll.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/xenroll.h
@@ -134,6 +134,7 @@ extern "C"{
 /* [local] */ 
 
 #include <winapifamily.h>
+#include "win32metadata_annotations.h"
 #pragma region Desktop Family
 #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
 
@@ -152,8 +153,8 @@ EXTERN_C const IID IID_ICEnroll;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("43F8F288-7A20-11D0-8F06-00C04FC295E1")
-    ICEnroll : public IDispatch
+    MIDL_INTERFACE("43F8F288-7A20-11D0-8F06-00C04FC295E1") _Windows_SupportedOS_WindowsXP_ 
+      ICEnroll : public IDispatch
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE createFilePKCS10( 
@@ -167,35 +168,35 @@ EXTERN_C const IID IID_ICEnroll;
         virtual HRESULT STDMETHODCALLTYPE createPKCS10( 
             /* [in] */ __RPC__in BSTR DNName,
             /* [in] */ __RPC__in BSTR Usage,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pPKCS10) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pPKCS10) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE acceptPKCS7( 
             /* [in] */ __RPC__in BSTR PKCS7) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE getCertFromPKCS7( 
             /* [in] */ __RPC__in BSTR wszPKCS7,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrCert) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrCert) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE enumProviders( 
             /* [in] */ LONG dwIndex,
             /* [in] */ LONG dwFlags,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrProvName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrProvName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE enumContainers( 
             /* [in] */ LONG dwIndex,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstr) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstr) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE freeRequestInfo( 
             /* [in] */ __RPC__in BSTR PKCS7OrPKCS10) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_MyStoreName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_MyStoreName( 
             /* [in] */ __RPC__in BSTR bstrName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_MyStoreType( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrType) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_MyStoreType( 
             /* [in] */ __RPC__in BSTR bstrType) = 0;
@@ -207,13 +208,13 @@ EXTERN_C const IID IID_ICEnroll;
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_CAStoreName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_CAStoreName( 
             /* [in] */ __RPC__in BSTR bstrName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_CAStoreType( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrType) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_CAStoreType( 
             /* [in] */ __RPC__in BSTR bstrType) = 0;
@@ -225,13 +226,13 @@ EXTERN_C const IID IID_ICEnroll;
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RootStoreName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RootStoreName( 
             /* [in] */ __RPC__in BSTR bstrName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RootStoreType( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrType) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RootStoreType( 
             /* [in] */ __RPC__in BSTR bstrType) = 0;
@@ -243,13 +244,13 @@ EXTERN_C const IID IID_ICEnroll;
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RequestStoreName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrName) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RequestStoreName( 
             /* [in] */ __RPC__in BSTR bstrName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RequestStoreType( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrType) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RequestStoreType( 
             /* [in] */ __RPC__in BSTR bstrType) = 0;
@@ -261,13 +262,13 @@ EXTERN_C const IID IID_ICEnroll;
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ContainerName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrContainer) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrContainer) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ContainerName( 
             /* [in] */ __RPC__in BSTR bstrContainer) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ProviderName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrProvider) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrProvider) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ProviderName( 
             /* [in] */ __RPC__in BSTR bstrProvider) = 0;
@@ -315,19 +316,19 @@ EXTERN_C const IID IID_ICEnroll;
             /* [in] */ BOOL fBool) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_SPCFileName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstr) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstr) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_SPCFileName( 
             /* [in] */ __RPC__in BSTR bstr) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_PVKFileName( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstr) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstr) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_PVKFileName( 
             /* [in] */ __RPC__in BSTR bstr) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_HashAlgorithm( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstr) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstr) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_HashAlgorithm( 
             /* [in] */ __RPC__in BSTR bstr) = 0;
@@ -688,7 +689,7 @@ EXTERN_C const IID IID_ICEnroll;
         END_INTERFACE
     } ICEnrollVtbl;
 
-    interface ICEnroll
+    interface _Windows_SupportedOS_WindowsXP_ ICEnroll
     {
         CONST_VTBL struct ICEnrollVtbl *lpVtbl;
     };
@@ -911,8 +912,8 @@ EXTERN_C const IID IID_ICEnroll2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("704ca730-c90b-11d1-9bec-00c04fc295e1")
-    ICEnroll2 : public ICEnroll
+    MIDL_INTERFACE("704ca730-c90b-11d1-9bec-00c04fc295e1") _Windows_SupportedOS_WindowsXP_ 
+      ICEnroll2 : public ICEnroll
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE addCertTypeToRequest( 
@@ -1321,7 +1322,7 @@ EXTERN_C const IID IID_ICEnroll2;
         END_INTERFACE
     } ICEnroll2Vtbl;
 
-    interface ICEnroll2
+    interface _Windows_SupportedOS_WindowsXP_ ICEnroll2
     {
         CONST_VTBL struct ICEnroll2Vtbl *lpVtbl;
     };
@@ -1563,8 +1564,8 @@ EXTERN_C const IID IID_ICEnroll3;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("c28c2d95-b7de-11d2-a421-00c04f79fe8e")
-    ICEnroll3 : public ICEnroll2
+    MIDL_INTERFACE("c28c2d95-b7de-11d2-a421-00c04f79fe8e") _Windows_SupportedOS_WindowsXP_ 
+      ICEnroll3 : public ICEnroll2
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE InstallPKCS7( 
@@ -1587,7 +1588,7 @@ EXTERN_C const IID IID_ICEnroll3;
         
         virtual HRESULT STDMETHODCALLTYPE GetAlgName( 
             /* [in] */ LONG algID,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstr) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstr) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ReuseHardwareKeyIfUnableToGenNew( 
             /* [in] */ BOOL fReuseHardwareKeyIfUnableToGenNew) = 0;
@@ -2074,7 +2075,7 @@ EXTERN_C const IID IID_ICEnroll3;
         END_INTERFACE
     } ICEnroll3Vtbl;
 
-    interface ICEnroll3
+    interface _Windows_SupportedOS_WindowsXP_ ICEnroll3
     {
         CONST_VTBL struct ICEnroll3Vtbl *lpVtbl;
     };
@@ -2359,31 +2360,31 @@ EXTERN_C const IID IID_ICEnroll4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("c1f1188a-2eb5-4a80-841b-7e729a356d90")
-    ICEnroll4 : public ICEnroll3
+    MIDL_INTERFACE("c1f1188a-2eb5-4a80-841b-7e729a356d90") _Windows_SupportedOS_WindowsXP_ 
+      ICEnroll4 : public ICEnroll3
     {
     public:
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_PrivateKeyArchiveCertificate( 
             /* [in] */ __RPC__in BSTR bstrCert) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_PrivateKeyArchiveCertificate( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrCert) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrCert) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ThumbPrint( 
             /* [in] */ __RPC__in BSTR bstrThumbPrint) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ThumbPrint( 
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pbstrThumbPrint) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pbstrThumbPrint) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE binaryToString( 
             /* [in] */ LONG Flags,
             /* [in] */ __RPC__in BSTR strBinary,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrEncoded) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrEncoded) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE stringToBinary( 
             /* [in] */ LONG Flags,
             /* [in] */ __RPC__in BSTR strEncoded,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrBinary) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrBinary) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addExtensionToRequest( 
             /* [in] */ LONG Flags,
@@ -2405,13 +2406,13 @@ EXTERN_C const IID IID_ICEnroll4;
         virtual HRESULT STDMETHODCALLTYPE resetAttributes( void) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createRequest( 
-            /* [in] */ LONG Flags,
+            /* [in] */ _Win32_AssociatedEnum_(CERT_CREATE_REQUEST_FLAGS) LONG Flags,
             /* [in] */ __RPC__in BSTR strDNName,
             /* [in] */ __RPC__in BSTR Usage,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrRequest) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrRequest) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createFileRequest( 
-            /* [in] */ LONG Flags,
+            /* [in] */ _Win32_AssociatedEnum_(CERT_CREATE_REQUEST_FLAGS) LONG Flags,
             /* [in] */ __RPC__in BSTR strDNName,
             /* [in] */ __RPC__in BSTR strUsage,
             /* [in] */ __RPC__in BSTR strRequestFileName) = 0;
@@ -2424,15 +2425,15 @@ EXTERN_C const IID IID_ICEnroll4;
         
         virtual HRESULT STDMETHODCALLTYPE getCertFromResponse( 
             /* [in] */ __RPC__in BSTR strResponse,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrCert) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrCert) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE getCertFromFileResponse( 
             /* [in] */ __RPC__in BSTR strResponseFileName,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrCert) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrCert) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createPFX( 
             /* [in] */ __RPC__in BSTR strPassword,
-            /* [retval][out] */ __RPC__deref_out_opt BSTR *pstrPFX) = 0;
+            /* [retval][out] */ _Win32_RAIIFree_(SysFreeString) __RPC__deref_out_opt BSTR *pstrPFX) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createFilePFX( 
             /* [in] */ __RPC__in BSTR strPassword,
@@ -2446,15 +2447,15 @@ EXTERN_C const IID IID_ICEnroll4;
         
         virtual HRESULT STDMETHODCALLTYPE enumPendingRequest( 
             /* [in] */ LONG lIndex,
-            /* [in] */ LONG lDesiredProperty,
+            /* [in] */ _Win32_AssociatedEnum_(PENDING_REQUEST_DESIRED_PROPERTY) LONG lDesiredProperty,
             /* [retval][out] */ __RPC__out VARIANT *pvarProperty) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE removePendingRequest( 
             /* [in] */ __RPC__in BSTR strThumbprint) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetKeyLenEx( 
-            /* [in] */ LONG lSizeSpec,
-            /* [in] */ LONG lKeySpec,
+            /* [in] */ _Win32_AssociatedEnum_(XEKL_KEYSIZE) LONG lSizeSpec,
+            /* [in] */ _Win32_AssociatedEnum_(XEKL_KEYSPEC) LONG lKeySpec,
             /* [retval][out] */ __RPC__out LONG *pdwKeySize) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE InstallPKCS7Ex( 
@@ -2462,7 +2463,7 @@ EXTERN_C const IID IID_ICEnroll4;
             /* [retval][out] */ __RPC__out LONG *plCertInstalled) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addCertTypeToRequestEx( 
-            /* [in] */ LONG lType,
+            /* [in] */ _Win32_AssociatedEnum_(ADDED_CERT_TYPE) LONG lType,
             /* [in] */ __RPC__in BSTR bstrOIDOrName,
             /* [in] */ LONG lMajorVersion,
             /* [in] */ BOOL fMinorVersion,
@@ -3152,7 +3153,7 @@ EXTERN_C const IID IID_ICEnroll4;
         END_INTERFACE
     } ICEnroll4Vtbl;
 
-    interface ICEnroll4
+    interface _Windows_SupportedOS_WindowsXP_ ICEnroll4
     {
         CONST_VTBL struct ICEnroll4Vtbl *lpVtbl;
     };
@@ -3537,8 +3538,8 @@ EXTERN_C const IID IID_IEnroll;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("acaa7838-4585-11d1-ab57-00c04fc295e1")
-    IEnroll : public IUnknown
+    MIDL_INTERFACE("acaa7838-4585-11d1-ab57-00c04fc295e1") _Windows_SupportedOS_WindowsXP_ 
+      IEnroll : public IUnknown
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE createFilePKCS10WStr( 
@@ -3552,13 +3553,13 @@ EXTERN_C const IID IID_IEnroll;
         virtual HRESULT STDMETHODCALLTYPE createPKCS10WStr( 
             /* [in] */ LPCWSTR DNName,
             /* [in] */ LPCWSTR Usage,
-            /* [out] */ PCRYPT_DATA_BLOB pPkcs10Blob) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB pPkcs10Blob) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE acceptPKCS7Blob( 
-            /* [in] */ PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
         
         virtual PCCERT_CONTEXT STDMETHODCALLTYPE getCertContextFromPKCS7( 
-            /* [in] */ PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
         
         virtual HCERTSTORE STDMETHODCALLTYPE getMyStore( void) = 0;
         
@@ -3569,173 +3570,173 @@ EXTERN_C const IID IID_IEnroll;
         virtual HRESULT STDMETHODCALLTYPE enumProvidersWStr( 
             /* [in] */ LONG dwIndex,
             /* [in] */ LONG dwFlags,
-            /* [out] */ LPWSTR *pbstrProvName) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *pbstrProvName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE enumContainersWStr( 
             /* [in] */ LONG dwIndex,
-            /* [out] */ LPWSTR *pbstr) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *pbstr) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE freeRequestInfoBlob( 
             /* [in] */ CRYPT_DATA_BLOB pkcs7OrPkcs10) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_MyStoreNameWStr( 
-            /* [out] */ LPWSTR *szwName) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_MyStoreNameWStr( 
             /* [in] */ LPWSTR szwName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_MyStoreTypeWStr( 
-            /* [out] */ LPWSTR *szwType) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_MyStoreTypeWStr( 
             /* [in] */ LPWSTR szwType) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_MyStoreFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_MyStoreFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_CAStoreNameWStr( 
-            /* [out] */ LPWSTR *szwName) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_CAStoreNameWStr( 
             /* [in] */ LPWSTR szwName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_CAStoreTypeWStr( 
-            /* [out] */ LPWSTR *szwType) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_CAStoreTypeWStr( 
             /* [in] */ LPWSTR szwType) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_CAStoreFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_CAStoreFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RootStoreNameWStr( 
-            /* [out] */ LPWSTR *szwName) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RootStoreNameWStr( 
             /* [in] */ LPWSTR szwName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RootStoreTypeWStr( 
-            /* [out] */ LPWSTR *szwType) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RootStoreTypeWStr( 
             /* [in] */ LPWSTR szwType) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RootStoreFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RootStoreFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RequestStoreNameWStr( 
-            /* [out] */ LPWSTR *szwName) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwName) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RequestStoreNameWStr( 
             /* [in] */ LPWSTR szwName) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RequestStoreTypeWStr( 
-            /* [out] */ LPWSTR *szwType) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RequestStoreTypeWStr( 
             /* [in] */ LPWSTR szwType) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RequestStoreFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RequestStoreFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ContainerNameWStr( 
-            /* [out] */ LPWSTR *szwContainer) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwContainer) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ContainerNameWStr( 
             /* [in] */ LPWSTR szwContainer) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ProviderNameWStr( 
-            /* [out] */ LPWSTR *szwProvider) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szwProvider) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ProviderNameWStr( 
             /* [in] */ LPWSTR szwProvider) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ProviderType( 
-            /* [out] */ LONG *pdwType) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwType) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ProviderType( 
             /* [in] */ LONG dwType) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_KeySpec( 
-            /* [out] */ LONG *pdw) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdw) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_KeySpec( 
             /* [in] */ LONG dw) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ProviderFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ProviderFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_UseExistingKeySet( 
-            /* [out] */ BOOL *fUseExistingKeys) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fUseExistingKeys) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_UseExistingKeySet( 
             /* [in] */ BOOL fUseExistingKeys) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_GenKeyFlags( 
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_GenKeyFlags( 
             /* [in] */ LONG dwFlags) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_DeleteRequestCert( 
-            /* [out] */ BOOL *fDelete) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fDelete) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_DeleteRequestCert( 
             /* [in] */ BOOL fDelete) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_WriteCertToUserDS( 
-            /* [out] */ BOOL *fBool) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fBool) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_WriteCertToUserDS( 
             /* [in] */ BOOL fBool) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_EnableT61DNEncoding( 
-            /* [out] */ BOOL *fBool) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fBool) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_EnableT61DNEncoding( 
             /* [in] */ BOOL fBool) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_WriteCertToCSP( 
-            /* [out] */ BOOL *fBool) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fBool) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_WriteCertToCSP( 
             /* [in] */ BOOL fBool) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_SPCFileNameWStr( 
-            /* [out] */ LPWSTR *szw) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szw) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_SPCFileNameWStr( 
             /* [in] */ LPWSTR szw) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_PVKFileNameWStr( 
-            /* [out] */ LPWSTR *szw) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szw) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_PVKFileNameWStr( 
             /* [in] */ LPWSTR szw) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_HashAlgorithmWStr( 
-            /* [out] */ LPWSTR *szw) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *szw) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_HashAlgorithmWStr( 
             /* [in] */ LPWSTR szw) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_RenewalCertificate( 
-            /* [out] */ PCCERT_CONTEXT *ppCertContext) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCCERT_CONTEXT *ppCertContext) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_RenewalCertificate( 
             /* [in] */ PCCERT_CONTEXT pCertContext) = 0;
@@ -3748,15 +3749,15 @@ EXTERN_C const IID IID_IEnroll;
             /* [in] */ LPWSTR Value) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE AddExtensionsToRequest( 
-            /* [in] */ PCERT_EXTENSIONS pCertExtensions) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCERT_EXTENSIONS pCertExtensions) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE AddAuthenticatedAttributesToPKCS7Request( 
-            /* [in] */ PCRYPT_ATTRIBUTES pAttributes) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_ATTRIBUTES pAttributes) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE CreatePKCS7RequestFromRequest( 
-            /* [in] */ PCRYPT_DATA_BLOB pRequest,
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pRequest,
             /* [in] */ PCCERT_CONTEXT pSigningCertContext,
-            /* [out] */ PCRYPT_DATA_BLOB pPkcs7Blob) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB pPkcs7Blob) = 0;
         
     };
     
@@ -4142,7 +4143,7 @@ EXTERN_C const IID IID_IEnroll;
         END_INTERFACE
     } IEnrollVtbl;
 
-    interface IEnroll
+    interface _Windows_SupportedOS_WindowsXP_ IEnroll
     {
         CONST_VTBL struct IEnrollVtbl *lpVtbl;
     };
@@ -4394,43 +4395,43 @@ EXTERN_C const IID IID_IEnroll2;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("c080e199-b7df-11d2-a421-00c04f79fe8e")
-    IEnroll2 : public IEnroll
+    MIDL_INTERFACE("c080e199-b7df-11d2-a421-00c04f79fe8e") _Windows_SupportedOS_WindowsXP_ 
+      IEnroll2 : public IEnroll
     {
     public:
         virtual HRESULT STDMETHODCALLTYPE InstallPKCS7Blob( 
-            /* [in] */ PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pBlobPKCS7) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE Reset( void) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetSupportedKeySpec( 
-            /* [out] */ LONG *pdwKeySpec) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwKeySpec) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetKeyLen( 
             /* [in] */ BOOL fMin,
             /* [in] */ BOOL fExchange,
-            /* [out] */ LONG *pdwKeySize) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwKeySize) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE EnumAlgs( 
             /* [in] */ LONG dwIndex,
             /* [in] */ LONG algClass,
-            /* [out] */ LONG *pdwAlgID) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwAlgID) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetAlgNameWStr( 
             /* [in] */ LONG algID,
-            /* [out] */ LPWSTR *ppwsz) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *ppwsz) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ReuseHardwareKeyIfUnableToGenNew( 
             /* [in] */ BOOL fReuseHardwareKeyIfUnableToGenNew) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ReuseHardwareKeyIfUnableToGenNew( 
-            /* [out] */ BOOL *fReuseHardwareKeyIfUnableToGenNew) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fReuseHardwareKeyIfUnableToGenNew) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_HashAlgID( 
             /* [in] */ LONG hashAlgID) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_HashAlgID( 
-            /* [retval][out] */ LONG *hashAlgID) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *hashAlgID) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetHStoreMy( 
             /* [in] */ HCERTSTORE hStore) = 0;
@@ -4448,13 +4449,13 @@ EXTERN_C const IID IID_IEnroll2;
             /* [in] */ BOOL fLimitExchangeKeyToEncipherment) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_LimitExchangeKeyToEncipherment( 
-            /* [retval][out] */ BOOL *fLimitExchangeKeyToEncipherment) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fLimitExchangeKeyToEncipherment) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_EnableSMIMECapabilities( 
             /* [in] */ BOOL fEnableSMIMECapabilities) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_EnableSMIMECapabilities( 
-            /* [retval][out] */ BOOL *fEnableSMIMECapabilities) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *fEnableSMIMECapabilities) = 0;
         
     };
     
@@ -4934,7 +4935,7 @@ EXTERN_C const IID IID_IEnroll2;
         END_INTERFACE
     } IEnroll2Vtbl;
 
-    interface IEnroll2
+    interface _Windows_SupportedOS_WindowsXP_ IEnroll2
     {
         CONST_VTBL struct IEnroll2Vtbl *lpVtbl;
     };
@@ -5241,15 +5242,15 @@ EXTERN_C const IID IID_IEnroll4;
 
 #if defined(__cplusplus) && !defined(CINTERFACE)
     
-    MIDL_INTERFACE("f8053fe5-78f4-448f-a0db-41d61b73446b")
-    IEnroll4 : public IEnroll2
+    MIDL_INTERFACE("f8053fe5-78f4-448f-a0db-41d61b73446b") _Windows_SupportedOS_WindowsXP_ 
+      IEnroll4 : public IEnroll2
     {
     public:
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_ThumbPrintWStr( 
             /* [in] */ CRYPT_DATA_BLOB thumbPrintBlob) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ThumbPrintWStr( 
-            /* [retval][out] */ PCRYPT_DATA_BLOB thumbPrintBlob) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB thumbPrintBlob) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetPrivateKeyArchiveCertificate( 
             /* [in] */ PCCERT_CONTEXT pPrivateKeyArchiveCert) = 0;
@@ -5258,25 +5259,25 @@ EXTERN_C const IID IID_IEnroll4;
         
         virtual HRESULT STDMETHODCALLTYPE binaryBlobToString( 
             /* [in] */ LONG Flags,
-            /* [in] */ PCRYPT_DATA_BLOB pblobBinary,
-            /* [out] */ LPWSTR *ppwszString) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pblobBinary,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPWSTR *ppwszString) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE stringToBinaryBlob( 
             /* [in] */ LONG Flags,
             /* [in] */ LPCWSTR pwszString,
-            /* [out] */ PCRYPT_DATA_BLOB pblobBinary,
-            /* [out] */ LONG *pdwSkip,
-            /* [out] */ LONG *pdwFlags) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB pblobBinary,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwSkip,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwFlags) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addExtensionToRequestWStr( 
             /* [in] */ LONG Flags,
             /* [in] */ LPCWSTR pwszName,
-            /* [in] */ PCRYPT_DATA_BLOB pblobValue) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pblobValue) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addAttributeToRequestWStr( 
             /* [in] */ LONG Flags,
             /* [in] */ LPCWSTR pwszName,
-            /* [in] */ PCRYPT_DATA_BLOB pblobValue) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pblobValue) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addNameValuePairToRequestWStr( 
             /* [in] */ LONG Flags,
@@ -5288,34 +5289,34 @@ EXTERN_C const IID IID_IEnroll4;
         virtual HRESULT STDMETHODCALLTYPE resetAttributes( void) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createRequestWStr( 
-            /* [in] */ LONG Flags,
+            /* [in] */ _Win32_AssociatedEnum_(CERT_CREATE_REQUEST_FLAGS) LONG Flags,
             /* [in] */ LPCWSTR pwszDNName,
             /* [in] */ LPCWSTR pwszUsage,
-            /* [out] */ PCRYPT_DATA_BLOB pblobRequest) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB pblobRequest) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createFileRequestWStr( 
-            /* [in] */ LONG Flags,
+            /* [in] */ _Win32_AssociatedEnum_(CERT_CREATE_REQUEST_FLAGS) LONG Flags,
             /* [in] */ LPCWSTR pwszDNName,
             /* [in] */ LPCWSTR pwszUsage,
             /* [in] */ LPCWSTR pwszRequestFileName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE acceptResponseBlob( 
-            /* [in] */ PCRYPT_DATA_BLOB pblobResponse) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pblobResponse) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE acceptFileResponseWStr( 
             /* [in] */ LPCWSTR pwszResponseFileName) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE getCertContextFromResponseBlob( 
-            /* [in] */ PCRYPT_DATA_BLOB pblobResponse,
-            /* [out] */ PCCERT_CONTEXT *ppCertContext) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pblobResponse,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCCERT_CONTEXT *ppCertContext) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE getCertContextFromFileResponseWStr( 
             /* [in] */ LPCWSTR pwszResponseFileName,
-            /* [out] */ PCCERT_CONTEXT *ppCertContext) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCCERT_CONTEXT *ppCertContext) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createPFXWStr( 
             /* [in] */ LPCWSTR pwszPassword,
-            /* [out] */ PCRYPT_DATA_BLOB pblobPFX) = 0;
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") PCRYPT_DATA_BLOB pblobPFX) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE createFilePFXWStr( 
             /* [in] */ LPCWSTR pwszPassword,
@@ -5329,23 +5330,23 @@ EXTERN_C const IID IID_IEnroll4;
         
         virtual HRESULT STDMETHODCALLTYPE enumPendingRequestWStr( 
             /* [in] */ LONG lIndex,
-            /* [in] */ LONG lDesiredProperty,
-            /* [out] */ LPVOID ppProperty) = 0;
+            /* [in] */ _Win32_AssociatedEnum_(PENDING_REQUEST_DESIRED_PROPERTY) LONG lDesiredProperty,
+            /* [out] */ _WIN32META_ANNOTATION_("win32metadata:in") LPVOID ppProperty) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE removePendingRequestWStr( 
             /* [in] */ CRYPT_DATA_BLOB thumbPrintBlob) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE GetKeyLenEx( 
-            /* [in] */ LONG lSizeSpec,
-            /* [in] */ LONG lKeySpec,
-            /* [retval][out] */ LONG *pdwKeySize) = 0;
+            /* [in] */ _Win32_AssociatedEnum_(XEKL_KEYSIZE) LONG lSizeSpec,
+            /* [in] */ _Win32_AssociatedEnum_(XEKL_KEYSPEC) LONG lKeySpec,
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *pdwKeySize) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE InstallPKCS7BlobEx( 
-            /* [in] */ PCRYPT_DATA_BLOB pBlobPKCS7,
-            /* [retval][out] */ LONG *plCertInstalled) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pBlobPKCS7,
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *plCertInstalled) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE AddCertTypeToRequestWStrEx( 
-            /* [in] */ LONG lType,
+            /* [in] */ _Win32_AssociatedEnum_(ADDED_CERT_TYPE) LONG lType,
             /* [in] */ LPCWSTR pwszOIDOrName,
             /* [in] */ LONG lMajorVersion,
             /* [in] */ BOOL fMinorVersion,
@@ -5353,12 +5354,12 @@ EXTERN_C const IID IID_IEnroll4;
         
         virtual HRESULT STDMETHODCALLTYPE getProviderTypeWStr( 
             /* [in] */ LPCWSTR pwszProvName,
-            /* [retval][out] */ LONG *plProvType) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *plProvType) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE addBlobPropertyToCertificateWStr( 
             /* [in] */ LONG lPropertyId,
             /* [in] */ LONG lReserved,
-            /* [in] */ PCRYPT_DATA_BLOB pBlobProperty) = 0;
+            /* [in] */ _WIN32META_ANNOTATION_("win32metadata:out") PCRYPT_DATA_BLOB pBlobProperty) = 0;
         
         virtual HRESULT STDMETHODCALLTYPE SetSignerCertificate( 
             /* [in] */ PCCERT_CONTEXT pSignerCert) = 0;
@@ -5367,13 +5368,13 @@ EXTERN_C const IID IID_IEnroll4;
             /* [in] */ LONG lClientId) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_ClientId( 
-            /* [retval][out] */ LONG *plClientId) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") LONG *plClientId) = 0;
         
         virtual /* [propput] */ HRESULT STDMETHODCALLTYPE put_IncludeSubjectKeyID( 
             /* [in] */ BOOL fInclude) = 0;
         
         virtual /* [propget] */ HRESULT STDMETHODCALLTYPE get_IncludeSubjectKeyID( 
-            /* [retval][out] */ BOOL *pfInclude) = 0;
+            /* [retval][out] */ _WIN32META_ANNOTATION_("win32metadata:in") BOOL *pfInclude) = 0;
         
     };
     
@@ -6047,7 +6048,7 @@ EXTERN_C const IID IID_IEnroll4;
         END_INTERFACE
     } IEnroll4Vtbl;
 
-    interface IEnroll4
+    interface _Windows_SupportedOS_WindowsXP_ IEnroll4
     {
         CONST_VTBL struct IEnroll4Vtbl *lpVtbl;
     };
@@ -6538,6 +6539,101 @@ void                      __RPC_USER  VARIANT_UserFree64(     __RPC__in unsigned
 }
 #endif
 
+
+#ifdef WIN32METADATA
+#pragma push_macro("XEPR_CADNS")
+#undef XEPR_CADNS
+#pragma push_macro("XEPR_CAFRIENDLYNAME")
+#undef XEPR_CAFRIENDLYNAME
+#pragma push_macro("XEPR_CANAME")
+#undef XEPR_CANAME
+#pragma push_macro("XEPR_HASH")
+#undef XEPR_HASH
+#pragma push_macro("XEPR_REQUESTID")
+#undef XEPR_REQUESTID
+enum PENDING_REQUEST_DESIRED_PROPERTY : int
+{
+    XEPR_CADNS = 1,
+    XEPR_CAFRIENDLYNAME = 3,
+    XEPR_CANAME = 2,
+    XEPR_HASH = 8,
+    XEPR_REQUESTID = 4,
+};
+#pragma pop_macro("XEPR_REQUESTID")
+#pragma pop_macro("XEPR_HASH")
+#pragma pop_macro("XEPR_CANAME")
+#pragma pop_macro("XEPR_CAFRIENDLYNAME")
+#pragma pop_macro("XEPR_CADNS")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("XEKL_KEYSIZE_MIN")
+#undef XEKL_KEYSIZE_MIN
+#pragma push_macro("XEKL_KEYSIZE_MAX")
+#undef XEKL_KEYSIZE_MAX
+#pragma push_macro("XEKL_KEYSIZE_INC")
+#undef XEKL_KEYSIZE_INC
+enum XEKL_KEYSIZE : int
+{
+    XEKL_KEYSIZE_MIN = 1,
+    XEKL_KEYSIZE_MAX = 2,
+    XEKL_KEYSIZE_INC = 3,
+};
+#pragma pop_macro("XEKL_KEYSIZE_INC")
+#pragma pop_macro("XEKL_KEYSIZE_MAX")
+#pragma pop_macro("XEKL_KEYSIZE_MIN")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("XECR_CMC")
+#undef XECR_CMC
+#pragma push_macro("XECR_PKCS10_V1_5")
+#undef XECR_PKCS10_V1_5
+#pragma push_macro("XECR_PKCS10_V2_0")
+#undef XECR_PKCS10_V2_0
+#pragma push_macro("XECR_PKCS7")
+#undef XECR_PKCS7
+enum CERT_CREATE_REQUEST_FLAGS : int
+{
+    XECR_CMC = 3,
+    XECR_PKCS10_V1_5 = 4,
+    XECR_PKCS10_V2_0 = 1,
+    XECR_PKCS7 = 2,
+};
+#pragma pop_macro("XECR_PKCS7")
+#pragma pop_macro("XECR_PKCS10_V2_0")
+#pragma pop_macro("XECR_PKCS10_V1_5")
+#pragma pop_macro("XECR_CMC")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("XECT_EXTENSION_V1")
+#undef XECT_EXTENSION_V1
+#pragma push_macro("XECT_EXTENSION_V2")
+#undef XECT_EXTENSION_V2
+enum ADDED_CERT_TYPE : int
+{
+    XECT_EXTENSION_V1 = 1,
+    XECT_EXTENSION_V2 = 2,
+};
+#pragma pop_macro("XECT_EXTENSION_V2")
+#pragma pop_macro("XECT_EXTENSION_V1")
+#endif
+
+#ifdef WIN32METADATA
+#pragma push_macro("XEKL_KEYSPEC_KEYX")
+#undef XEKL_KEYSPEC_KEYX
+#pragma push_macro("XEKL_KEYSPEC_SIG")
+#undef XEKL_KEYSPEC_SIG
+enum XEKL_KEYSPEC : int
+{
+    XEKL_KEYSPEC_KEYX = 1,
+    XEKL_KEYSPEC_SIG = 2,
+};
+#pragma pop_macro("XEKL_KEYSPEC_SIG")
+#pragma pop_macro("XEKL_KEYSPEC_KEYX")
+#endif
+
 #endif
 
 
