diff --git a/generation/WinSDK/RecompiledIdlHeaders/um/xoleHlp.h b/generation/WinSDK/RecompiledIdlHeaders/um/xoleHlp.h
index d0d411a9e6..3f332a810b 100644
--- a/generation/WinSDK/RecompiledIdlHeaders/um/xoleHlp.h
+++ b/generation/WinSDK/RecompiledIdlHeaders/um/xoleHlp.h
@@ -1,3 +1,4 @@
+#include <win32metadata_annotations.h>
 //  Copyright (C) 1995-1999 Microsoft Corporation.  All rights reserved.
 /* ----------------------------------------------------------------------------
  Microsoft   D.T.C (Distributed Transaction Coordinator)
@@ -90,7 +91,7 @@ EXTERN_C HRESULT __cdecl DtcGetTransactionManager(
                                     /* in */ DWORD i_dwReserved1,
                                     /* in */ WORD i_wcbReserved2,
                                     /* in */ _In_reads_bytes_opt_(i_wcbReserved2) void * i_pvReserved2,
-                                    /* out */ void** o_ppvObject
+                                    /* out */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void** o_ppvObject
                                     );
 EXTERN_C HRESULT __cdecl DtcGetTransactionManagerC(
                                     /* in */ _In_opt_ LPSTR i_pszHost,
@@ -99,26 +100,26 @@ EXTERN_C HRESULT __cdecl DtcGetTransactionManagerC(
                                     /* in */ DWORD i_dwReserved1,
                                     /* in */ WORD i_wcbReserved2,
                                     /* in */ _In_reads_bytes_opt_(i_wcbReserved2) void * i_pvReserved2,
-                                    /* out */ void ** o_ppvObject
+                                    /* out */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void ** o_ppvObject
                                     );
 
-EXTERN_C HRESULT __cdecl DtcGetTransactionManagerExA(
+_WIN32META_ANNOTATION_("win32metadata:ansi") EXTERN_C HRESULT __cdecl DtcGetTransactionManagerExA(
                                     /* in */ _In_opt_ LPSTR i_pszHost,
                                     /* in */ _In_opt_ LPSTR i_pszTmName,
                                     /* in */ REFIID i_riid,
                                     /* in */ DWORD i_grfOptions,
-                                    /* in */ void * i_pvConfigParams,
-                                    /* out */ void ** o_ppvObject
+                                    /* in */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void * i_pvConfigParams,
+                                    /* out */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void ** o_ppvObject
                                     );
 
 
-EXTERN_C HRESULT __cdecl DtcGetTransactionManagerExW(
+_WIN32META_ANNOTATION_("win32metadata:unicode") EXTERN_C HRESULT __cdecl DtcGetTransactionManagerExW(
                                     /* in */ _In_opt_ LPWSTR i_pwszHost,
                                     /* in */ _In_opt_ LPWSTR i_pwszTmName,
                                     /* in */ REFIID i_riid,
                                     /* in */ DWORD i_grfOptions,
-                                    /* in */ void * i_pvConfigParams,
-                                    /* out */ void ** o_ppvObject
+                                    /* in */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void * i_pvConfigParams,
+                                    /* out */ _WIN32META_ANNOTATION_("win32metadata:in") _WIN32META_ANNOTATION_("win32metadata:out") void ** o_ppvObject
                                     );
 #ifdef UNICODE
 #define DtcGetTransactionManagerEx      DtcGetTransactionManagerExW
